diff --git a/share/arpwarp_setup_base.bash b/share/arpwarp_setup_base.bash
index 29e769c..0b8a51e 100644
--- a/share/arpwarp_setup_base.bash
+++ b/share/arpwarp_setup_base.bash
@@ -13,10 +13,10 @@ else
     identifier=${nameprocessor}'-'${namesystem}
 fi
 #
-arpwarphome="$1X"
+arpwarphome="@GENTOO_PORTAGE_EPREFIX@/opt/arp-warp-bin"
 #
 export warpbin=${arpwarphome}/bin/bin-${identifier}
-export warpdoc=${arpwarphome}/manual
+export warpdoc=@GENTOO_PORTAGE_EPREFIX@/usr/share/doc/arp-warp-bin-7.1/html/
 export PATH=${warpbin}:${PATH}
 #
 # Checks for existence of $warpbin directory
diff --git a/share/arpwarp_setup_base.csh b/share/arpwarp_setup_base.csh
index 41b11e9..a4edcff 100644
--- a/share/arpwarp_setup_base.csh
+++ b/share/arpwarp_setup_base.csh
@@ -13,7 +13,7 @@ else
     set identifier = ${nameprocessor}'-'${namesystem}
 endif
 #
-set arpwarphome = '$1X'
+set arpwarphome = '@GENTOO_PORTAGE_EPREFIX@/opt/arp-warp-bin'
 #
 setenv warpbin ${arpwarphome}/bin/bin-${identifier}
 #
