--- samba-3.6.0rc2/examples/smb.conf.default~	2011-06-07 19:13:47.000000000 +0100
+++ samba-3.6.0rc2/examples/smb.conf.default	2011-06-30 16:12:42.388102480 +0100
@@ -63,7 +63,7 @@
 
 # this tells Samba to use a separate log file for each machine
 # that connects
-   log file = /usr/local/samba/var/log.%m
+   log file = /var/log/samba/log.%m
 
 # Put a capping on the size of the log files (in Kb).
    max log size = 50
@@ -81,7 +81,10 @@
 
 # Backend to store user information in. New installations should 
 # use either tdbsam or ldapsam. smbpasswd is available for backwards 
-# compatibility. tdbsam requires no further configuration.
+# compatibility. tdbsam requires no further configuration. If you're 
+# migrating from < samba 3.4, you'll have to convert your old user 
+# passwords to the new backend with the command:
+# pdbedit -i smbpasswd:/var/lib/samba/private/smbpasswd -e
 ;   passdb backend = tdbsam
 
 # Using the following line enables you to customise your configuration
@@ -89,7 +92,7 @@
 # of the machine that is connecting.
 # Note: Consider carefully the location in the configuration file of
 #       this line.  The included file is read at that point.
-;   include = /usr/local/samba/lib/smb.conf.%m
+;   include = /etc/samba/smb.conf.%m
 
 # Configure Samba to use multiple interfaces
 # If you have multiple network interfaces then you must list them
@@ -166,7 +169,7 @@
 # Un-comment the following and create the netlogon directory for Domain Logons
 ; [netlogon]
 ;   comment = Network Logon Service
-;   path = /usr/local/samba/lib/netlogon
+;   path = /var/lib/samba/netlogon
 ;   guest ok = yes
 ;   writable = no
 ;   share modes = no
@@ -175,7 +178,7 @@
 # Un-comment the following to provide a specific roving profile share
 # the default is to use the user's home directory
 ;[Profiles]
-;    path = /usr/local/samba/profiles
+;    path = /var/lib/samba/profiles
 ;    browseable = no
 ;    guest ok = yes
 
@@ -184,7 +187,7 @@
 # specifically define each individual printer
 [printers]
    comment = All Printers
-   path = /usr/spool/samba
+   path = /var/spool/samba
    browseable = no
 # Set public = yes to allow user 'guest account' to print
    guest ok = no
@@ -204,7 +207,7 @@
 ;   comment = Public Stuff
 ;   path = /home/samba
 ;   public = yes
-;   writable = no
+;   writable = yes
 ;   printable = no
 ;   write list = @staff
 
