From 44d231b5604480601bedc623ca2f5204bd4a3ac8 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Sun, 11 Sep 2022 15:55:44 +0200
Subject: [PATCH] Enable highdpi scaling and pixmaps

---
 src/Main.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Main.cpp b/src/Main.cpp
index 0af162a..4088d37 100644
--- a/src/Main.cpp
+++ b/src/Main.cpp
@@ -85,6 +85,9 @@
     */
 int main(int argc, char *argv[])
 {
+    QGuiApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);
+    QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+
     QApplication app(argc, argv);
     KLocalizedString::setApplicationDomain("kxstitch");
 
-- 
GitLab

