diff -ru sassena-1.4.2_org/src/main/sassena.cpp sassena-1.4.2/src/main/sassena.cpp
--- sassena-1.4.2_org/src/main/sassena.cpp	2014-07-29 11:02:12.650053749 +0200
+++ sassena-1.4.2/src/main/sassena.cpp	2014-07-29 12:52:56.978605357 +0200
@@ -62,6 +62,7 @@
 #include <boost/serialization/complex.hpp>
 #include <boost/serialization/map.hpp>
 #include <boost/serialization/vector.hpp>
+#include <boost/thread/thread.hpp>
 
 // other headers
 #include "exceptions/exceptions.hpp"
Nur in sassena-1.4.2/src/main: .sassena.cpp.swp.
diff -ru sassena-1.4.2_org/src/main/s_stage.cpp sassena-1.4.2/src/main/s_stage.cpp
--- sassena-1.4.2_org/src/main/s_stage.cpp	2014-07-29 11:02:12.650053749 +0200
+++ sassena-1.4.2/src/main/s_stage.cpp	2014-07-29 12:51:22.257611749 +0200
@@ -33,6 +33,7 @@
 #include <boost/serialization/complex.hpp>
 #include <boost/serialization/map.hpp>
 #include <boost/serialization/vector.hpp>
+#include <boost/thread/thread.hpp>
 
 // other headers
 #include "exceptions/exceptions.hpp"
diff -ru sassena-1.4.2_org/src/scatter_devices/abstract_scatter_device.cpp sassena-1.4.2/src/scatter_devices/abstract_scatter_device.cpp
--- sassena-1.4.2_org/src/scatter_devices/abstract_scatter_device.cpp	2014-07-29 11:02:12.650053749 +0200
+++ sassena-1.4.2/src/scatter_devices/abstract_scatter_device.cpp	2014-07-29 13:02:20.824567306 +0200
@@ -18,6 +18,7 @@
 #include <boost/accumulators/accumulators.hpp>
 #include <boost/accumulators/statistics.hpp>
 #include <boost/lexical_cast.hpp>
+#include <boost/thread/thread.hpp>
 
 // other headers
 #include "exceptions/exceptions.hpp"
Nur in sassena-1.4.2/src/scatter_devices: .abstract_scatter_device.cpp.swp.
diff -ru sassena-1.4.2_org/src/scatter_devices/all_vectors_scatter_device.cpp sassena-1.4.2/src/scatter_devices/all_vectors_scatter_device.cpp
--- sassena-1.4.2_org/src/scatter_devices/all_vectors_scatter_device.cpp	2014-07-29 11:02:12.649053750 +0200
+++ sassena-1.4.2/src/scatter_devices/all_vectors_scatter_device.cpp	2014-07-29 13:04:04.907560282 +0200
@@ -18,6 +18,7 @@
 #include <boost/accumulators/accumulators.hpp>
 #include <boost/accumulators/statistics.hpp>
 #include <boost/lexical_cast.hpp>
+#include <boost/thread/thread.hpp>
 
 // other headers
 #include "math/coor3d.hpp"
Nur in sassena-1.4.2/src/scatter_devices: .all_vectors_scatter_device.cpp.swp.
diff -ru sassena-1.4.2_org/src/scatter_devices/multipole_scatter_device.cpp sassena-1.4.2/src/scatter_devices/multipole_scatter_device.cpp
--- sassena-1.4.2_org/src/scatter_devices/multipole_scatter_device.cpp	2014-07-29 11:02:12.649053750 +0200
+++ sassena-1.4.2/src/scatter_devices/multipole_scatter_device.cpp	2014-07-29 13:01:16.038571678 +0200
@@ -19,6 +19,7 @@
 #include <boost/accumulators/statistics.hpp>
 #include <boost/lexical_cast.hpp>
 #include <boost/math/special_functions.hpp>
+#include <boost/thread/thread.hpp>
 
 // other headers
 #include "math/coor3d.hpp"
diff -ru sassena-1.4.2_org/src/scatter_devices/self_vectors_scatter_device.cpp sassena-1.4.2/src/scatter_devices/self_vectors_scatter_device.cpp
--- sassena-1.4.2_org/src/scatter_devices/self_vectors_scatter_device.cpp	2014-07-29 11:02:12.650053749 +0200
+++ sassena-1.4.2/src/scatter_devices/self_vectors_scatter_device.cpp	2014-07-29 13:01:45.753569673 +0200
@@ -18,6 +18,7 @@
 #include <boost/accumulators/accumulators.hpp>
 #include <boost/accumulators/statistics.hpp>
 #include <boost/lexical_cast.hpp>
+#include <boost/thread/thread.hpp>
 
 // other headers
 #include "math/coor3d.hpp"
diff -ru sassena-1.4.2_org/src/services/file_writer_service.cpp sassena-1.4.2/src/services/file_writer_service.cpp
--- sassena-1.4.2_org/src/services/file_writer_service.cpp	2014-07-29 11:02:12.650053749 +0200
+++ sassena-1.4.2/src/services/file_writer_service.cpp	2014-07-29 13:08:07.448543914 +0200
@@ -13,6 +13,7 @@
 
 #include <boost/lexical_cast.hpp>
 #include <boost/filesystem.hpp>
+#include <boost/thread/thread.hpp>
 #include <log.hpp>
 #include <control.hpp>
 #include <math/smath.hpp>
Nur in sassena-1.4.2/src/services: .file_writer_service.cpp.swp.
diff -ru sassena-1.4.2_org/src/services/monitor_service.cpp sassena-1.4.2/src/services/monitor_service.cpp
--- sassena-1.4.2_org/src/services/monitor_service.cpp	2014-07-29 11:02:12.650053749 +0200
+++ sassena-1.4.2/src/services/monitor_service.cpp	2014-07-29 13:07:06.617548019 +0200
@@ -13,6 +13,7 @@
 #include <boost/lexical_cast.hpp>
 #include <boost/filesystem.hpp>
 #include <boost/random/uniform_int.hpp>
+#include <boost/thread/thread.hpp>
 #include <log.hpp>
 #include <control.hpp>
 #include <report/timer.hpp>
