--- a/src/unix/lwt_unix.cppo.ml	2021-10-01 20:35:45.285987395 +0200
+++ b/src/unix/lwt_unix.cppo.ml	2021-10-01 20:36:19.570493202 +0200
@@ -1819,6 +1819,9 @@
   | SO_ACCEPTCONN
   | TCP_NODELAY
   | IPV6_ONLY
+#if OCAML_VERSION >= (4, 12, 0)
+  | SO_REUSEPORT
+#endif
 
 type socket_int_option =
   Unix.socket_int_option =
--- a/src/unix/lwt_unix.cppo.mli	2021-10-01 20:33:57.558539065 +0200
+++ b/src/unix/lwt_unix.cppo.mli	2021-10-01 20:35:30.393202008 +0200
@@ -1008,6 +1008,9 @@
   | SO_ACCEPTCONN
   | TCP_NODELAY
   | IPV6_ONLY
+#if OCAML_VERSION >= (4, 12, 0)
+  | SO_REUSEPORT
+#endif
 
 type socket_int_option =
     Unix.socket_int_option =
