From 67d2548f9e531650ee90e9518cf6559aabe98afd Mon Sep 17 00:00:00 2001
From: Jordan Guymon <jogu.sd@gmail.com>
Date: Fri, 31 Aug 2018 10:22:25 -0700
Subject: [PATCH] Add whence to mock readable seek to fix functional test

---
 tests/functional/test_manager.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/functional/test_manager.py b/tests/functional/test_manager.py
index 9d5702d..ec4175a 100644
--- a/tests/functional/test_manager.py
+++ b/tests/functional/test_manager.py
@@ -40,7 +40,7 @@ def signal_transferring(self):
     def signal_not_transferring(self):
         self.signal_not_transferring_call_count += 1
 
-    def seek(self, where):
+    def seek(self, where, whence=0):
         pass
 
     def tell(self):
