diff -ur db-3.2.9.old/dist/Makefile.in db-3.2.9/dist/Makefile.in
--- db-3.2.9.old/dist/Makefile.in	2003-10-18 02:21:18.000000000 -0700
+++ db-3.2.9/dist/Makefile.in	2003-10-18 02:22:14.000000000 -0700
@@ -72,7 +72,7 @@
 JAVA_DBDIR=	$(JAVA_SRCDIR)/$(JAVA_DBREL)
 JAVA_EXDIR=	$(JAVA_SRCDIR)/com/sleepycat/examples
 
-libj_jarfile=	db.jar
+libj_jarfile=	db-$(SOVERSION).jar
 libjso_base=	libdb_java
 libjso=		$(libjso_base)-$(SOVERSION).@SOSUFFIX@
 libjso_target=	$(libjso_base)-$(SOVERSION).la
