diff -ur db-4.1.25.old/dist/Makefile.in db-4.1.25/dist/Makefile.in
--- db-4.1.25.old/dist/Makefile.in	2003-10-18 02:15:45.000000000 -0700
+++ db-4.1.25/dist/Makefile.in	2003-10-18 02:13:47.000000000 -0700
@@ -107,9 +107,9 @@
 JAVA_EXDIR=	$(JAVA_SRCDIR)/$(JAVA_EXREL)
 JAVA_RPCDIR=	$(srcdir)/rpc_server/java
 
-libj_jarfile=	db.jar
-libj_exjarfile=	dbexamples.jar
-rpc_jarfile=	dbsvc.jar
+libj_jarfile=	db-4.3.jar
+libj_exjarfile=	dbexamples-4.3.jar
+rpc_jarfile=	dbsvc-4.3.jar
 libjso_base=	libdb_java
 libjso=		$(libjso_base)-$(SOVERSION).@JMODSUFFIX@
 libjso_static=	$(libjso_base)-$(SOVERSION).a
