--- a/dist/aclocal/types.m4
+++ b/dist/aclocal/types.m4
@@ -188,7 +188,7 @@ AC_SUBST(db_off_t_decl)
 AC_CHECK_TYPE(db_off_t,,
     [AM_SEARCH_SSIZES(db_off_t_decl, db_off_t, $ac_cv_sizeof_off_t)])
 AC_SUBST(db_off_t_max_decl)
-if test "$ac_cv_sizeof_off_t" == "4"; then
+if test "$ac_cv_sizeof_off_t" = "4"; then
 	db_off_t_max_decl="#define DB_OFF_T_MAX INT32_MAX"
 else
 	db_off_t_max_decl="#define DB_OFF_T_MAX INT64_MAX"
