--- CMakeLists.txt
+++ CMakeLists.txt
@@ -242,14 +242,11 @@
 
 install(TARGETS indi_lx200generic RUNTIME DESTINATION bin )
 
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/make_lx200generic_symlink.cmake
-"exec_program(\"${CMAKE_COMMAND}\" ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic \$ENV{DESTDIR}${BIN_INSTALL_DIR}/indi_lx200classic)\n
-exec_program(\"${CMAKE_COMMAND}\" ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic \$ENV{DESTDIR}${BIN_INSTALL_DIR}/indi_lx200autostar)\n
-exec_program(\"${CMAKE_COMMAND}\" ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic \$ENV{DESTDIR}${BIN_INSTALL_DIR}/indi_lx200_16)\n
-exec_program(\"${CMAKE_COMMAND}\" ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic \$ENV{DESTDIR}${BIN_INSTALL_DIR}/indi_lx200gps)\n
-exec_program(\"${CMAKE_COMMAND}\" ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic \$ENV{DESTDIR}${BIN_INSTALL_DIR}/indi_lx200ap)\n
-")
-set_target_properties(indi_lx200generic PROPERTIES POST_INSTALL_SCRIPT ${CMAKE_CURRENT_BINARY_DIR}/make_lx200generic_symlink.cmake)
+install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${BIN_INSTALL_DIR}/indi_lx200generic\" \"\$ENV{DESTDIR}/${BUILD_ROOT}${BIN_INSTALL_DIR}/indi_lx200classic\" )" )
+install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${BIN_INSTALL_DIR}/indi_lx200generic\" \"\$ENV{DESTDIR}/${BUILD_ROOT}${BIN_INSTALL_DIR}/indi_lx200autostar\" )" )
+install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${BIN_INSTALL_DIR}/indi_lx200generic\" \"\$ENV{DESTDIR}/${BUILD_ROOT}${BIN_INSTALL_DIR}/indi_lx200_16\" )" )
+install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${BIN_INSTALL_DIR}/indi_lx200generic\" \"\$ENV{DESTDIR}/${BUILD_ROOT}${BIN_INSTALL_DIR}/indi_lx20gps\" )" )
+install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${BIN_INSTALL_DIR}/indi_lx200generic\" \"\$ENV{DESTDIR}/${BUILD_ROOT}${BIN_INSTALL_DIR}/indi_lx200ap\" )" )
 #################################################################################
 
 ########### LX200 Generic Legacy ###########
