--- slime-2.15-orig/slime.el	2015-08-31 07:46:47.000000000 +1000
+++ slime-2.15/slime.el	2015-09-17 17:05:53.616659817 +1000
@@ -138,16 +138,7 @@
     "Return the datestring of the latest entry in the ChangeLog file.
 Return nil if the ChangeLog file cannot be found."
     (interactive "p")
-    (let ((changelog (slime--changelog-file-name))
-          (date nil))
-      (when (file-exists-p changelog)
-        (with-temp-buffer
-          (insert-file-contents-literally changelog nil 0 100)
-          (goto-char (point-min))
-          (setq date (symbol-name (read (current-buffer))))))
-      (when interactivep
-        (message "Slime ChangeLog dates %s." date))
-      date)))
+    "@SLIME-CHANGELOG-DATE@"))
 
 (defvar slime-protocol-version nil)
 (setq slime-protocol-version
