From 7db58a893a1448b7229de5464696b3cc6eb7663c Mon Sep 17 00:00:00 2001
From: Alexander Olofsson <ace@haxalot.com>
Date: Thu, 13 Feb 2020 19:32:17 +0100
Subject: [PATCH] Ensure osg/Version is included

---
 apps/openmw/mwrender/sky.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/apps/openmw/mwrender/sky.cpp b/apps/openmw/mwrender/sky.cpp
index e13f5a7b65..7cf2af88d0 100644
--- a/apps/openmw/mwrender/sky.cpp
+++ b/apps/openmw/mwrender/sky.cpp
@@ -16,6 +16,7 @@
 #include <osg/BlendFunc>
 #include <osg/AlphaFunc>
 #include <osg/PolygonOffset>
+#include <osg/Version>
 #include <osg/observer_ptr>
 
 #include <osgParticle/BoxPlacer>
