--- pytz/__init__.py.bak	2016-07-14 13:20:37.856381431 +0200
+++ pytz/__init__.py	2016-07-14 13:21:00.464512973 +0200
@@ -81,8 +81,7 @@
     for part in name_parts:
         if part == os.path.pardir or os.path.sep in part:
             raise ValueError('Bad path segment: %r' % part)
-    filename = os.path.join(os.path.dirname(__file__),
-                            'zoneinfo', *name_parts)
+    filename = os.path.join('/usr/share/zoneinfo', *name_parts)
     if not os.path.exists(filename):
         # http://bugs.launchpad.net/bugs/383171 - we avoid using this
         # unless absolutely necessary to help when a broken version of
