--- tqsl-2.3.1-orig/src/CMakeLists.txt
+++ tqsl-2.3.1/src/CMakeLists.txt
@@ -54,7 +54,7 @@ include_directories(${OPENSSL_INCLUDE_DIR} ${ZLIB_INCLUDE_DIR} ${EXPAT_INCLUDE_D
 if(NOT APPLE AND NOT WIN32)
 set_source_files_properties(location.cpp PROPERTIES COMPILE_DEFINITIONS CONFDIR="${CMAKE_INSTALL_PREFIX}/share/TrustedQSL/")
 set(HEADERS_TO_INSTALL tqsllib.h tqslerrno.h cabrillo.h adif.h tqslconvert.h)
-install(TARGETS tqsllib DESTINATION lib$(LIB_SUFFIX))
+install(TARGETS tqsllib DESTINATION lib${LIB_SUFFIX})
 install(FILES config.xml DESTINATION share/TrustedQSL)
 install(FILES ${HEADERS_TO_INSTALL} DESTINATION include)
 endif()
