diff --git a/share/cmake/modules/FindImath.cmake b/share/cmake/modules/FindImath.cmake
index 44a55e8..8a2d7ca 100644
--- a/share/cmake/modules/FindImath.cmake
+++ b/share/cmake/modules/FindImath.cmake
@@ -49,6 +49,7 @@ if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
         find_path(Imath_INCLUDE_DIR
             NAMES
                 Imath/ImathConfig.h
+                Imath-3/IMathConfig.h
             HINTS
                 ${Imath_ROOT}
                 ${PC_Imath_INCLUDE_DIRS}
@@ -90,8 +91,8 @@ if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
 
         # Get version from config header file
         if(Imath_INCLUDE_DIR)
-            if(EXISTS "${Imath_INCLUDE_DIR}/Imath/ImathConfig.h")
-                set(_Imath_CONFIG "${Imath_INCLUDE_DIR}/Imath/ImathConfig.h")
+            if(EXISTS "${Imath_INCLUDE_DIR}/Imath-3/ImathConfig.h")
+                set(_Imath_CONFIG "${Imath_INCLUDE_DIR}/Imath-3/ImathConfig.h")
             endif()
         endif()
 
diff --git a/src/utils/Half.h.in b/src/utils/Half.h.in
index 3784341..237f3ed 100644
--- a/src/utils/Half.h.in
+++ b/src/utils/Half.h.in
@@ -8,7 +8,7 @@
 #define OCIO_USE_IMATH_HALF @OCIO_USE_IMATH_HALF@
 
 #if OCIO_USE_IMATH_HALF
-#   include <Imath/half.h>
+#   include <Imath-3/half.h>
 #else
 #   include <OpenEXR/half.h>
 #endif
diff --git a/vendor/aftereffects/vc/vc15/utils/Half.h b/vendor/aftereffects/vc/vc15/utils/Half.h
index 5035750..5144e72 100644
--- a/vendor/aftereffects/vc/vc15/utils/Half.h
+++ b/vendor/aftereffects/vc/vc15/utils/Half.h
@@ -8,7 +8,7 @@
 #define OCIO_USE_IMATH_HALF 0
 
 #if OCIO_USE_IMATH_HALF
-#   include <Imath/half.h>
+#   include <Imath-3/half.h>
 #else
 #   include <OpenEXR/half.h>
 #endif
diff --git a/vendor/aftereffects/xcode/xcode12/utils/Half.h b/vendor/aftereffects/xcode/xcode12/utils/Half.h
index 5035750..5144e72 100644
--- a/vendor/aftereffects/xcode/xcode12/utils/Half.h
+++ b/vendor/aftereffects/xcode/xcode12/utils/Half.h
@@ -8,7 +8,7 @@
 #define OCIO_USE_IMATH_HALF 0
 
 #if OCIO_USE_IMATH_HALF
-#   include <Imath/half.h>
+#   include <Imath-3/half.h>
 #else
 #   include <OpenEXR/half.h>
 #endif
diff --git a/vendor/aftereffects/xcode/xcode9/utils/Half.h b/vendor/aftereffects/xcode/xcode9/utils/Half.h
index 5035750..5144e72 100644
--- a/vendor/aftereffects/xcode/xcode9/utils/Half.h
+++ b/vendor/aftereffects/xcode/xcode9/utils/Half.h
@@ -8,7 +8,7 @@
 #define OCIO_USE_IMATH_HALF 0
 
 #if OCIO_USE_IMATH_HALF
-#   include <Imath/half.h>
+#   include <Imath-3/half.h>
 #else
 #   include <OpenEXR/half.h>
 #endif
