diff --git a/meson.build b/meson.build
index 690e7d3d8aa..fb08e905ebd 100644
--- a/meson.build
+++ b/meson.build
@@ -906,7 +906,7 @@ if not cc.links('''#include <stdint.h>
                      return __sync_add_and_fetch(&v, (uint64_t)1);
                    }''',
                 name : 'GCC 64bit atomics')
-  pre_args += '-DMISSING_64_BIT_ATOMICS'
+  pre_args += '-DMISSING_64BIT_ATOMICS'
 endif
 
 # TODO: shared/static? Is this even worth doing?
