--- heirloom-devtools-070527/make/bsd/bsd.cc.old	2016-09-21 22:36:56.556036181 -0400
+++ heirloom-devtools-070527/make/bsd/bsd.cc	2016-09-21 22:52:09.819186461 -0400
@@ -50,7 +50,7 @@
 extern SIG_PF
 bsdsignal (int Signal, SIG_PF Handler)
 {
-  auto SIG_PF                   previous_handler;
+  SIG_PF                   previous_handler;
 #ifdef SUN5_0
 #ifdef sun
   previous_handler = sigset (Signal, Handler);
