diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9149a6c..cc1a03f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -16,7 +16,7 @@ if(WITH_GTK)
 			set(GRESOURCES "gtk_resources.c")
 			include_directories(${CMAKE_BINARY_DIR}/src)
 		endif(PORTABLE_BINARY)
-		add_library(gui_gtk
+		add_library(gui_gtk STATIC
 			gui_gtk.c
 			gui_gtk.h
 			gui_gtk_id.h
@@ -35,7 +35,7 @@ if(WITH_NCURSES)
 	if(NCURSES_FOUND)
 		include_directories(${NCURSES_INCLUDE_DIRS})
 		link_directories(${NCURSES_LIBRARY_DIRS})
-		add_library(tui_ncurses
+		add_library(tui_ncurses STATIC
 			tui_ncurses.c
 			tui_ncurses.h
 		)
