--- crwinfo.c.orig	2002-07-30 11:17:03.000000000 -0600
+++ crwinfo.c	2005-01-21 15:24:54.256888228 -0700
@@ -95,10 +95,10 @@
    fprintf( stderr, "-m        show misc. information (whitepoint, AF mode etc)\n" );
    fprintf( stderr, "-f        show flash information (flash mode etc)\n" );
    fprintf( stderr, "-cf       show custom function settings\n" );
-   fprintf( stderr, "-t        extract the thumbnail to the optional filename.
-	                       If no filename for the thumbnail is given, the
-			       name of the CRW-file is used with a .jpg
-			       extension\n" );
+   fprintf( stderr, "-t        extract the thumbnail to the optional filename. \n" );
+   fprintf( stderr, "          If no filename for the thumbnail is given, the\n" );
+   fprintf( stderr, "          name of the CRW-file is used with a .jpg\n" );
+   fprintf( stderr, "          extension\n" );
    fprintf( stderr, "-x        show experimental information (don't rely on it)\n" );
    exit(0);
 }
