--- src/nls.h	2009-01-01 16:45:39.000000000 -0800
+++ src/nls.h	2010-10-01 17:13:03.595458860 -0700
@@ -49,11 +49,6 @@
 #else
 #  define _(String) (String)
 #  define N_(String) (String)
-#  define textdomain(String) (String)
-#  define gettext(String) (String)
-#  define dgettext(Domain,String) (String)
-#  define dcgettext(Domain,String,Type) (String)
-#  define bindtextdomain(Domain,Directory) (Domain)
 #endif
 
 #endif
