diff -ruN automake-1.16.1.orig/t/instmany-python.sh automake-1.16.1/t/instmany-python.sh
--- automake-1.16.1.orig/t/instmany-python.sh	2018-02-26 21:38:28.000000000 +0100
+++ automake-1.16.1/t/instmany-python.sh	2020-01-20 02:44:37.192352310 +0100
@@ -21,7 +21,11 @@
 required='python'
 . test-init.sh
 
-limit=2500
+limit=3200
+# Let's increase the number here a bit, otherwise the __pycache__ dirs
+# make the test fail. I know this is a hack, but does anything conform
+# to the POSIX limit of 4096 still anyway?
+
 subdir=long_subdir_name_with_many_characters
 nfiles=81
 list=$(seq_ 1 $nfiles)
