diff -Naur aide-0.13.1.orig//src/be.c aide-0.13.1//src/be.c
--- aide-0.13.1.orig//src/be.c	2006-12-15 01:09:24.000000000 +0900
+++ aide-0.13.1//src/be.c	2010-05-26 20:08:10.000000000 +0900
@@ -161,7 +161,7 @@
 #endif
 #ifdef WITH_ZLIB
     if(iszipped && !inout){
-      fh=gzdopen(fd,"wb9+");
+      fh=gzdopen(fd,"wb9");
       if(fh==NULL){
 	error(0,_("Couldn't open file %s for %s"),u->value,
 	      inout?"reading\n":"writing\n");
