--- a/makefile
+++ b/makefile
@@ -10,25 +10,22 @@
 
 HEADERS = general.h clustalw.h
 
-CC	= cc
-CFLAGS  = -c -O
-LFLAGS	= -O -lm 
+# C99 for gets()
+CFLAGS	+= -std=gnu99
+LIBS	+= -lm
 
 clustalw : $(OBJECTS) amenu.o clustalw.o
-	$(CC) -o $@ $(OBJECTS) amenu.o clustalw.o $(LFLAGS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $(OBJECTS) amenu.o clustalw.o $(LIBS)
 
 interface.o : interface.c $(HEADERS) param.h
-	$(CC) $(CFLAGS) $*.c
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
 
 amenu.o : amenu.c $(HEADERS) param.h
-	$(CC) $(CFLAGS) $*.c
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
 
 readmat.o : readmat.c $(HEADERS) matrices.h
-	$(CC) $(CFLAGS) $*.c
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
 
 trees.o : trees.c $(HEADERS) dayhoff.h
-	$(CC) $(CFLAGS) $*.c
-
-.c.o :
-	$(CC) $(CFLAGS) $?
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
 
