--- gourmet/gtk_extras/dialog_extras.py	2013-10-31 09:03:08.456469166 -0500
+++ gourmet/gtk_extras/dialog_extras.py	2013-10-31 09:04:18.175067008 -0500
@@ -1096,7 +1096,9 @@
                   buttons=None
                   ):
         FileSelectorDialog.__init__(self, title, filename, filters, action, set_filter, buttons)
-        self.fsd.set_current_folder(get_user_special_dir(USER_DIRECTORY_PICTURES))
+        pictures_dir = get_user_special_dir(USER_DIRECTORY_PICTURES)
+        if not pictures_dir == None:
+            self.fsd.set_current_folder(pictures_dir)
 
     def post_dialog (self):
         self.preview = gtk.Image()
