From https://github.com/winterheart/aseprite/tree/v1.3.16.1-unbundle
From: "Azamat H. Hackimov" <azamat.hackimov@gmail.com>
Date: Fri, 9 Jan 2026 02:24:47 +0300
Subject: [PATCH 6/7] Use shared libjpeg-turbo library

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -341,7 +341,7 @@ endif()
 include_directories(${GIF_INCLUDE_DIRS})
 
 # libjpeg-turbo
-find_package(JpegTurbo)
+find_package(libjpeg-turbo)
 
 if(USE_SHARED_JSON11)
   find_package(PkgConfig QUIET)
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -786,7 +786,7 @@ target_link_libraries(app-lib
   ${GIF_LIBRARIES}
   ${PNG_LIBRARIES}
   ${ZLIB_LIBRARIES}
-  libjpeg-turbo
+  libjpeg-turbo::jpeg
   ${LibArchive_LIBRARIES}
   json11
   fmt
-- 
2.51.2

