--- a/cdist/preos.py
+++ b/cdist/preos.py
@@ -7,7 +7,7 @@ import cdist
 import logging
 import cdist.argparse
 import cdist.configuration
-import cdist.exec.util as util
+from cdist.exec import util
 

 _PREOS_CALL = "commandline"
--- a/cdist/exec/local.py
+++ b/cdist/exec/local.py
@@ -34,7 +34,7 @@ import datetime
 import cdist
 import cdist.message
 from cdist import core
-import cdist.exec.util as util
+from cdist.exec import util
 
 CONF_SUBDIRS_LINKED = ["explorer", "files", "manifest", "type", ]
 
--- a/cdist/exec/remote.py
+++ b/cdist/exec/remote.py
@@ -27,7 +27,7 @@ import logging
 import multiprocessing
 
 import cdist
-import cdist.exec.util as util
+from cdist.exec import util
 import cdist.util.ipaddr as ipaddr
 from cdist.mputil import mp_pool_run
 
--- a/cdist/info.py
+++ b/cdist/info.py
@@ -22,7 +22,7 @@
 import cdist
 import cdist.configuration
 import cdist.core
-import cdist.exec.util as util
+from cdist.exec import util
 import os
 import glob
 import fnmatch
