--- a/libwsmake/wsUtil.cpp
+++ b/libwsmake/wsUtil.cpp
@@ -710,13 +710,13 @@
   }
 }
 
-void std::__wsmake_print_it(FILE *out, const char *output, const va_list *ap)
+void std::__wsmake_print_it(FILE *out, const char *output, va_list *ap)
 {
   if(__wsmake_quiet) return;
 
   vfprintf(out,output,*ap);
   //BV: this does not compile:  va_end((void*&)*ap);
-  va_end((char*&)*ap);
+  va_end(*ap);
 }
 
 #ifdef DEBUG
--- a/libwsmake/wsUtil.h
+++ b/libwsmake/wsUtil.h
@@ -77,7 +77,7 @@
   void __wsmake_print_error(const char *, ...);
   void __wsmake_print(const char *, ...);
   void __wsmake_print(int, const char *, ...);
-  void __wsmake_print_it(FILE *, const char *, const va_list *);
+  void __wsmake_print_it(FILE *, const char *, va_list *);
 }
 
 #endif /* __WSUTIL_H__ */
