diff --git a/src/archive.rs b/src/archive.rs
index 63f4293..986a146 100644
--- a/src/archive.rs
+++ b/src/archive.rs
@@ -3,12 +3,13 @@ use std::{
     collections::VecDeque,
     path::Path,
     pin::Pin,
-    sync::{
-        atomic::{AtomicU64, Ordering},
-        Arc,
-    },
+    sync::Arc,
     task::{Context, Poll},
 };
+use portable_atomic::{
+    AtomicU64,
+    Ordering,
+};
 use tokio::{
     io::{self, AsyncRead as Read, AsyncReadExt},
     sync::Mutex,
diff -dup a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	2024-04-27 10:39:21.186708179 +0200
+++ b/Cargo.toml	2024-04-27 10:39:37.940402613 +0200
@@ -43,6 +43,9 @@ version = "0.2"
 [dependencies.futures-core]
 version = "0.3"
 
+[dependencies.portable-atomic]
+version = "1"
+
 [dependencies.tokio]
 version = "1"
 features = [
