Fix cstdint include for gcc 13

--- tg_owt-5098730b9eb6173f0b52068fe2555b7c1015123a.orig/src/modules/audio_processing/transient/file_utils.h
+++ tg_owt-5098730b9eb6173f0b52068fe2555b7c1015123a/src/modules/audio_processing/transient/file_utils.h
@@ -11,6 +11,7 @@
 #ifndef MODULES_AUDIO_PROCESSING_TRANSIENT_FILE_UTILS_H_
 #define MODULES_AUDIO_PROCESSING_TRANSIENT_FILE_UTILS_H_
 
+#include <cstdint>
 #include <string.h>
 
 #include "rtc_base/system/file_wrapper.h"
--- tg_owt-5098730b9eb6173f0b52068fe2555b7c1015123a.orig/src/rtc_base/third_party/base64/base64.h
+++ tg_owt-5098730b9eb6173f0b52068fe2555b7c1015123a/src/rtc_base/third_party/base64/base64.h
@@ -12,6 +12,7 @@
 #ifndef RTC_BASE_THIRD_PARTY_BASE64_BASE64_H_
 #define RTC_BASE_THIRD_PARTY_BASE64_BASE64_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
