From 71c7420e4b33b4bace8ed6d4fa45fa5e281d38bd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A9ven=20Car?= <meven.car@enioka.com>
Date: Sat, 27 Mar 2021 11:00:07 +0100
Subject: [PATCH] KNewFileMenu: use destination side to stat destination

BUG: 429541
---
 src/filewidgets/knewfilemenu.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/filewidgets/knewfilemenu.cpp b/src/filewidgets/knewfilemenu.cpp
index 64c4b1c2e..48fdf8622 100644
--- a/src/filewidgets/knewfilemenu.cpp
+++ b/src/filewidgets/knewfilemenu.cpp
@@ -1242,7 +1242,7 @@ void KNewFileMenuPrivate::_k_slotTextChanged(const QString &text)
         } else {
             url = QUrl(m_baseUrl.toString() + QLatin1Char('/') + text);
         }
-        KIO::StatJob *job = KIO::statDetails(url, KIO::StatJob::StatSide::SourceSide, KIO::StatDetail::StatBasic);
+        KIO::StatJob *job = KIO::statDetails(url, KIO::StatJob::StatSide::DestinationSide, KIO::StatDetail::StatBasic);
         QObject::connect(job, &KJob::result, q, [this](KJob *job) {
             _k_slotStatResult(job);
         });
-- 
GitLab

