From 0dd0e96c10f634974c63f85c2068e44019c27591 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A9ven=20Car?= <meven@kdemail.net>
Date: Wed, 29 Jun 2022 10:14:50 +0200
Subject: [PATCH] KRecentDocuments: Improve indentation in recentlyused.xbel

BUG: 456046
---
 src/core/krecentdocument.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/krecentdocument.cpp b/src/core/krecentdocument.cpp
index 23fcbbd09..87d276b06 100644
--- a/src/core/krecentdocument.cpp
+++ b/src/core/krecentdocument.cpp
@@ -270,6 +270,7 @@ static bool addToXbel(const QUrl &url, const QString &desktopEntryName, KRecentD
 
     if (!foundExisting) {
         // must create new bookmark tag
+        output.writeCharacters(QStringLiteral("  "));
         output.writeStartElement(bookmarkTag);
 
         output.writeAttribute(hrefAttribute, newUrl);
-- 
GitLab

