diff --git a/Makefile b/Makefile
index e6e4987..8acfe23 100644
--- a/Makefile
+++ b/Makefile
@@ -257,7 +257,7 @@ EXTRACFLAGS=
 
 # Extra linker flags (additional/alternative libraries, etc.)
 
-EXTRALDFLAGS=
+EXTRALDFLAGS=${LDFLAGS}
 
 
 # Special make flags (e.g. to override make environment variables)
diff --git a/src/mlock/Makefile b/src/mlock/Makefile
index d425da8..9c3020b 100644
--- a/src/mlock/Makefile
+++ b/src/mlock/Makefile
@@ -36,7 +36,7 @@ CFLAGS = -O2 -pipe -march=core2
 all:	mlock
 
 mlock:	mlock.o
-	$(CC) $(CFLAGS) -o mlock mlock.o
+	$(CC) $(CFLAGS) ${LDFLAGS} -o mlock mlock.o
 
 install: mlock
 	chgrp mail mlock
diff --git a/Makefile b/Makefile
index 8acfe23..70f889d 100644
--- a/Makefile
+++ b/Makefile
@@ -698,7 +698,7 @@ rebuild:
 	@echo Rebuilding c-client for `$(CAT) OSTYPE`...
 	@$(TOUCH) SPECIALS
 	$(CD) c-client;$(MAKE) all CC=`$(CAT) CCTYPE` \
-	 CFLAGS="`$(CAT) CFLAGS`" `$(CAT) SPECIALS`
+	 CFLAGS="`$(CAT) CFLAGS`" LDFLAGS=${LDFLAGS} `$(CAT) SPECIALS`
 
 rebuildclean:
 	$(SH) -c '$(RM) rebuild || true'
diff --git a/src/osdep/unix/Makefile b/src/osdep/unix/Makefile
index ef1d059..a53318c 100644
--- a/src/osdep/unix/Makefile
+++ b/src/osdep/unix/Makefile
@@ -969,7 +969,7 @@ onceenv:
 	 -DLOCKPGM3=\"$(LOCKPGM3)\" > OSCFLAGS
 	echo $(BASELDFLAGS) $(EXTRALDFLAGS) > LDFLAGS
 	echo "$(ARRC) $(ARCHIVE) $(BINARIES);$(RANLIB) $(ARCHIVE)" > ARCHIVE
-	echo "`$(CAT) CCTYPE` `$(CAT) CFLAGS` `$(CAT) OSFLAGS` -shared \
+	echo "`$(CAT) CCTYPE` `$(CAT) CFLAGS` `$(CAT) OSFLAGS` `$(CAT) LDFLAGS` -shared \
 	 -Wl,-soname,libc-client.so.1 -o libc-client.so.1.0.0 $(BINARIES)" \
 	 >> ARCHIVE
 	echo $(OS) > OSTYPE
