--- a/Makefile	2006-09-07 09:52:55.000000000 +0200
+++ b/Makefile	2006-09-07 09:54:54.000000000 +0200
@@ -21,7 +21,7 @@
 # ARCH=sun
 
 ${TGT}: Makefile ${TGT}.c 
-	${CC} -Wall -g -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
+	${CC} ${CFLAGS} -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS} ${LDFLAGS}
 
 install: ${TGT} 
 	if [ ! -d ${BINDIR} ]; then mkdir -p ${BINDIR};fi
