--- powerpc-utils-papr-1.1.0.orig/Makefile	2006-03-28 16:07:24.000000000 -0600
+++ powerpc-utils-papr-1.1.0/Makefile	2006-04-19 14:53:25.000000000 -0500
@@ -20,11 +20,9 @@
 	@$(MAKE) -C doc
 
 install: all
-	@$(call install_doc,$(FILES),$(DESTDIR))
 	@$(foreach d,$(SUBDIRS), $(MAKE) -C $d install;)
 
 uninstall: 
-	@$(call uninstall_doc,$(FILES),$(DESTDIR))
 	@$(foreach d,$(SUBDIRS), $(MAKE) -C $d uninstall;)
 
 rpm: all
