From f27f2945c8ca06e2eb0c2b95366a7a4c1363d857 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20Gr=C3=B6nholm?= <alex.gronholm@nextday.fi>
Date: Mon, 8 May 2017 21:20:01 +0300
Subject: [PATCH] Removed the test for the chained callback feature

---
 test/test_callback.py | 25 -------------------------
 1 file changed, 25 deletions(-)

diff --git a/test/test_callback.py b/test/test_callback.py
index 08d31e3..9c8ab99 100644
--- a/test/test_callback.py
+++ b/test/test_callback.py
@@ -59,31 +59,6 @@ def test_callback(framework):
     assert results[0] == "it worked"
 
 
-def test_chained_callback(framework):
-    """
-    Chain two callbacks where the first one alters the value.
-    """
-    calls = []
-
-    def callback0(arg):
-        calls.append(arg)
-        return arg + " pray I do not alter it futher"
-
-    def callback1(arg):
-        calls.append(arg)
-
-    f = txaio.create_future()
-    txaio.add_callbacks(f, callback0, None)
-    txaio.add_callbacks(f, callback1, None)
-    txaio.resolve(f, "the deal")
-
-    run_once()
-
-    assert len(calls) == 2
-    assert calls[0] == "the deal"
-    assert calls[1] == "the deal pray I do not alter it futher"
-
-
 def test_immediate_result(framework):
     f = txaio.create_future_success("it worked")
     results = []
-- 
2.13.0

