diff --git a/auto/lib/perl/make b/auto/lib/perl/make
index 260bd95..857ece6 100644
--- a/auto/lib/perl/make
+++ b/auto/lib/perl/make
@@ -35,6 +35,8 @@ $NGX_OBJS/src/http/modules/perl/Makefile: \\
 			NGX_DEPS="\$(CORE_DEPS) \$(HTTP_DEPS)" \\
 		$NGX_PERL Makefile.PL \\
 			LIB=$NGX_PERL_MODULES \\
+			PREFIX=$NGX_PREFIX \\
+			INSTALLDIRS=vendor \\
 			INSTALLSITEMAN3DIR=$NGX_PERL_MODULES_MAN
 
 END
