diff --git a/src/knemod/backends/netlinkbackend.cpp b/src/knemod/backends/netlinkbackend.cpp
index f749e43..5a82c13 100644
--- a/src/knemod/backends/netlinkbackend.cpp
+++ b/src/knemod/backends/netlinkbackend.cpp
@@ -23,6 +23,10 @@
 #include <netlink/route/link.h>
 #include <netlink/route/route.h>
 
+#if LIBNL_CURRENT >= 221
+#include <net/if.h>
+#endif
+
 #include <KLocale>
 #include <kio/global.h>
 
