diff --git a/check/exam_archive.cpp b/check/exam_archive.cpp
index 2ca13a8e..5821c97a 100644
--- a/check/exam_archive.cpp
+++ b/check/exam_archive.cpp
@@ -101,10 +101,10 @@ unsigned numeric_complex_bug()
 	const cl_I one(1);
 	const cl_R three_fp = cl_float(3.0);
 	std::vector<cl_N> numbers = {
-		complex(one, one),
-		complex(one, three_fp),
-		complex(three_fp, one),
-		complex(three_fp, three_fp)
+		cln::complex(one, one),
+		cln::complex(one, three_fp),
+		cln::complex(three_fp, one),
+		cln::complex(three_fp, three_fp)
 	};
 	for (auto & n : numbers) {
 		result += checker(n);
