http://svn.berlios.de/viewvc/docutils?view=revision&revision=7013

--- docutils/writers/odf_odt/__init__.py
+++ docutils/writers/odf_odt/__init__.py
@@ -300,7 +300,10 @@
 
 def ToString(et):
     outstream = StringIO.StringIO()
-    et.write(outstream)
+    if sys.version_info >= (3, 2):
+        et.write(outstream, encoding="unicode")
+    else:
+        et.write(outstream)
     s1 = outstream.getvalue()
     outstream.close()
     return s1
