https://github.com/dracut-ng/dracut-ng/pull/1250
diff --git a/modules.d/90mdraid/module-setup.sh b/modules.d/90mdraid/module-setup.sh
index 8ce5f2ee..b0ab8411 100755
--- a/modules.d/90mdraid/module-setup.sh
+++ b/modules.d/90mdraid/module-setup.sh
@@ -46,7 +46,7 @@ cmdline() {
         [[ ${host_fs_types[$dev]} != *_raid_member ]] && continue
 
         UUID=$(
-            /sbin/mdadm --examine --export "$dev" \
+            mdadm --examine --export "$dev" \
                 | while read -r line || [[ "$line" ]]; do
                     [[ ${line#MD_UUID=} == "$line" ]] && continue
                     printf "%s" "${line#MD_UUID=} "
-- 
2.48.1

