--- a/java/CMakeLists.txt	2017-10-25 22:26:12.920787580 +0900
+++ b/java/CMakeLists.txt	2017-10-25 22:27:21.423007891 +0900
@@ -9,7 +9,7 @@ find_package(Java)

 set(DATA_DIR "${CMAKE_INSTALL_PREFIX}/share")

-set(DEFAULT_JAVACFLAGS "-source 1.6 -target 1.6 -Xlint:all,-serial,-cast,-unchecked,-fallthrough,-dep-ann,-deprecation,-rawtypes")
+set(DEFAULT_JAVACFLAGS "-source 1.6 -target 1.6 -encoding UTF-8 -Xlint:all,-serial,-cast,-unchecked,-fallthrough,-dep-ann,-deprecation,-rawtypes")
 set(JAVACFLAGS ${DEFAULT_JAVACFLAGS} CACHE STRING
   "Java compiler flags (Default: ${DEFAULT_JAVACFLAGS})")
 message(STATUS "Java compiler flags = ${JAVACFLAGS}")
