commit f489d3c4ab9dc3ff6fe443a0e8a31b46ed73371d
Author: Louis Sautier <sautier.louis@gmail.com>
Date:   Sun Jul 23 22:37:01 2017 +0200

    Fix under-linking (pthread and dl), fixes #3

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c6c8f54..e0e2433 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -168,7 +168,11 @@ set_target_properties(openshot-audio
 			SOVERSION ${SO_VERSION}
 			INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
 
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+find_package(Threads REQUIRED)
 TARGET_LINK_LIBRARIES(openshot-audio
+	${CMAKE_DL_LIBS}
+	${CMAKE_THREAD_LIBS_INIT}
 	${JUCE_PLATFORM_SPECIFIC_LIBRARIES}
 )
 
