--- a/dbd/mysql/statement.c
+++ b/dbd/mysql/statement.c
@@ -226,7 +226,7 @@
 		*boolean = lua_toboolean(L, p);
 
 		bind[i].buffer_type = MYSQL_TYPE_LONG;
-		bind[i].is_null = (int*)0;
+		bind[i].is_null = false;
 		bind[i].buffer = (char *)boolean;
 		bind[i].length = 0;
 		break;
@@ -241,7 +241,7 @@
 		*num = lua_tonumber(L, p);
 
 		bind[i].buffer_type = MYSQL_TYPE_DOUBLE;
-		bind[i].is_null = (int*)0;
+		bind[i].is_null = false;
 		bind[i].buffer = (char *)num;
 		bind[i].length = 0;
 		break;
@@ -252,7 +252,7 @@
 		str = lua_tolstring(L, p, str_len);
 
 		bind[i].buffer_type = MYSQL_TYPE_STRING;
-		bind[i].is_null = (int*)0;
+		bind[i].is_null = false;
 		bind[i].buffer = (char *)str;
 		bind[i].length = str_len;
 		break;
