--- monkey-bubble-0.3.2/src/net/Makefile.am.orig
+++ monkey-bubble-0.3.2/src/net/Makefile.am
@@ -28,10 +28,10 @@
 monkey_srv_LDFLAGS =
 
 monkey_srv_LDADD = \
-	$(MONKEY_SRV_LIBS) \
 	libmonkeynet.a \
 	../monkey/libmonkey.a \
-	../util/libmonkeyutil.a 
+	../util/libmonkeyutil.a \
+	$(MONKEY_SRV_LIBS) -lm
 
 $(libmonkeynet_a_OBJECTS): $(marshal_sources)
 
--- monkey-bubble-0.3.2/src/ui/Makefile.am.orig
+++ monkey-bubble-0.3.2/src/ui/Makefile.am
@@ -50,10 +50,10 @@
 
 
 monkey_bubble_LDADD =  \
-	$(UI_LIBS) \
 	../monkey/libmonkey.a \
 	../view/libmonkeyview.a \
 	../util/libmonkeyutil.a \
 	../net/libmonkeynet.a \
 	../audio/libmonkeyaudio.a \
-	../input/libmbinput.a
+	../input/libmbinput.a \
+	$(UI_LIBS) -lm
