Disable tests that fail with network-sandbox

--- a/src/utils/utils_test.go
+++ b/src/utils/utils_test.go
@@ -7,7 +7,6 @@ import (
 	"log"
 	"math/rand"
 	"os"
-	"strings"
 	"testing"
 
 	"github.com/schollz/croc/v9/src/models"
@@ -189,19 +188,6 @@ func TestHashFile(t *testing.T) {
 	assert.Equal(t, "e66c561610ad51e2", fmt.Sprintf("%x", hashed))
 }
 
-func TestPublicIP(t *testing.T) {
-	ip, err := PublicIP()
-	fmt.Println(ip)
-	assert.True(t, strings.Contains(ip, ".") || strings.Contains(ip, ":"))
-	assert.Nil(t, err)
-}
-
-func TestLocalIP(t *testing.T) {
-	ip := LocalIP()
-	fmt.Println(ip)
-	assert.True(t, strings.Contains(ip, ".") || strings.Contains(ip, ":"))
-}
-
 func TestGetRandomName(t *testing.T) {
 	name := GetRandomName()
 	fmt.Println(name)
