--- jove4.16.0.73-orig/recover.c
+++ jove4.16.0.73/recover.c
@@ -776,7 +776,7 @@
 
 	last_update = ctime(&(rec->UpdTime));
 	/* Start up mail */
-	sprintf(mail_cmd, "/bin/mail %s", pw->pw_name);
+	sprintf(mail_cmd, "/usr/sbin/sendmail -t %s", pw->pw_name);
 	setuid(getuid());
 	if ((mail_pipe = popen(mail_cmd, "w")) == NULL)
 		return;
@@ -784,6 +784,7 @@
 	setbuf(mail_pipe, mail_cmd);
 	/* Let's be grammatically correct! */
 	buf_string = rec->Nbuffers == 1? "buffer" : "buffers";
+	fprintf(mail_pipe, "To: %s\n", pw->pw_name);
 	fprintf(mail_pipe, "Subject: Jove saved %d %s after \"%s\" crashed\n",
 		rec->Nbuffers, buf_string, hname());
 	fprintf(mail_pipe, " \n");
