--- configure
+++ configure
@@ -114,40 +114,6 @@
      echo 'no'
 fi
 
-if [ x$DEBUG = x ]; then
-    echo -n "Checking if $CC accepts -O2... "
-    if (
-         $CC -O2 __conftest.c -o __conftest
-       ) >/dev/null 2>&1; then
-         echo 'yes'
-         CFLAGS="$CFLAGS -O2"
-    else
-         echo 'no'
-         echo -n "Checking if $CC accepts -O... "
-         if (
-              $CC -O __conftest.c -o __conftest
-            ) >/dev/null 2>&1; then
-              echo 'yes'
-              CFLAGS="$CFLAGS -O"
-         else
-              echo 'no'
-         fi
-    fi
-
-else
-    echo -n "Checking if $CC accepts -g... "
-    if (
-         $CC -g __conftest.c -o __conftest
-       ) >/dev/null 2>&1; then
-         echo 'yes'
-         CFLAGS="$CFLAGS -g"
-    else
-         echo 'no'
-    fi
-
-fi
-
-LDFLAGS=
 LIBS=
 
 rm -f __conftest*
