diff -Naurp a/setup.py b/setup.py
--- a/setup.py	2020-08-20 06:35:06.000000000 +0200
+++ b/setup.py	2020-09-27 17:50:43.776476265 +0200
@@ -24,7 +24,7 @@ setup(
     keywords='locking locks lock',
     url='https://flufllock.readthedocs.io',
     download_url='https://pypi.python.org/pypi/flufl.lock',
-    packages=find_namespace_packages(where='.'),
+    packages=find_namespace_packages(where='.', exclude=['tests', 'docs']),
     namespace_packages=['flufl'],
     include_package_data=True,
     # readthedocs builds fail unless zip_safe is False.
