# Install template files in /usr/share/xnec2c
diff --git a/Makefile.am b/Makefile.am
index 9e8873a..6f6203e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -62,7 +62,7 @@ install-data-local:
 	  $(install_sh_DATA) doc/$(PACKAGE).1.gz $(DESTDIR)$(mandir)/man1/; \
 	fi;
 	if test -d .$(PACKAGE)/; then \
-	  $(MKDIR_P) $(DESTDIR)$(datarootdir)/examples/$(PACKAGE)/;\
-	  cp -r .$(PACKAGE)/* $(DESTDIR)$(datarootdir)/examples/$(PACKAGE)/; \
+	  $(MKDIR_P) $(DESTDIR)$(datarootdir)/$(PACKAGE)/;\
+	  cp -r .$(PACKAGE)/* $(DESTDIR)$(datarootdir)/$(PACKAGE)/; \
 	fi;
 
diff --git a/src/main.c b/src/main.c
index 4cb541e..6883ceb 100644
--- a/src/main.c
+++ b/src/main.c
@@ -88,7 +88,7 @@ main (int argc, char *argv[])
     /* Create file path to xnec2c examples directory */
     exe_path[bytes] = '\0';
     Strlcpy( file_path, exe_path, sizeof(file_path) );
-    Strlcat( file_path, "/share/examples/xnec2c", sizeof(file_path) );
+    Strlcat( file_path, "/share/xnec2c", sizeof(file_path) );
     fprintf( stderr, "xnec2c: creating xnec2c config directory from: %s\n", file_path );
 
     /* Create system command to copy examples/xnec2c to ~/.xnec2c */
