--- libsbsms-2.0.0/configure.in
+++ libsbsms-2.0.0/configure.in
@@ -69,8 +69,6 @@
 
 AC_ARG_ENABLE(universal_binary,[  --enable-universal_binary enable universal binary build: (default: disable)],[enable_universal_binary=$enableval],[enable_universal_binary=no])
 
-SBSMS_CFLAGS="-ffast-math -funroll-loops"
-
 if test x$enable_static = xno; then
   case "$target_os" in
      darwin*)
@@ -81,12 +79,6 @@
    esac
 fi
 
-if test x$enable_debug = xyes; then
-   SBSMS_CFLAGS="$SBSMS_CFLAGS -g -O3"
-else
-   SBSMS_CFLAGS="$SBSMS_CFLAGS -O3 -fomit-frame-pointer"
-fi
-
 if test x$enable_multithreaded = xyes; then
     AC_DEFINE(MULTITHREADED,1,[Define to compile multithreaded sbsms])
 fi
