From 63ca52af50f94b396182638224fe2c50b4cb7043 Mon Sep 17 00:00:00 2001
From: Allen Winter <allen.winter@kdab.com>
Date: Thu, 5 Apr 2018 10:07:01 -0400
Subject: [PATCH] src/core/model/element.h - fix compile on with XCode (include QPointF)

---
 src/core/model/element.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/model/element.h b/src/core/model/element.h
index ec32cfc..c20ac6f 100644
--- a/src/core/model/element.h
+++ b/src/core/model/element.h
@@ -30,6 +30,7 @@
 
 #include <QObject>
 #include <QMetaType>
+#include <QPointF>
 
 QT_BEGIN_NAMESPACE
 class QPainterPath;