diff --git i/libbase/accumulator.h w/libbase/accumulator.h
index 694cbd2..876c669 100644
--- i/libbase/accumulator.h
+++ w/libbase/accumulator.h
@@ -22,6 +22,7 @@
 
 #include <boost/program_options/value_semantic.hpp>
 #include <boost/any.hpp>
+#include <boost/version.hpp>
 #include <functional>
 #include <vector>
 #include <string>
@@ -61,6 +62,11 @@ public:
     virtual unsigned min_tokens() const { return 0; }
     virtual unsigned max_tokens() const { return 0; }
 
+#if BOOST_VERSION >= 105900
+    /// There are no tokens, so this is indifferent.
+    virtual bool adjacent_tokens_only() const { return false; }
+#endif
+
     /// Accumulating from different sources is silly.
     virtual bool is_composing() const { return false; }
 
