diff -Naur wmweather+-2.17.orig/wmgeneral/wmgeneral-x11.c wmweather+-2.17/wmgeneral/wmgeneral-x11.c
--- wmweather+-2.17.orig/wmgeneral/wmgeneral-x11.c	2014-02-07 20:17:44.000000000 +0100
+++ wmweather+-2.17/wmgeneral/wmgeneral-x11.c	2020-02-24 22:11:47.226905019 +0100
@@ -94,6 +94,7 @@
 GC          NormalGC;
 GC          RedrawGC;
 XpmIcon     wmgen;
+Display		*display;
 
   /***********************/
  /* Function Prototypes */
diff -Naur wmweather+-2.17.orig/wmgeneral/wmgeneral-x11.h wmweather+-2.17/wmgeneral/wmgeneral-x11.h
--- wmweather+-2.17.orig/wmgeneral/wmgeneral-x11.h	2014-02-07 20:17:44.000000000 +0100
+++ wmweather+-2.17/wmgeneral/wmgeneral-x11.h	2020-02-24 22:10:45.233884234 +0100
@@ -17,7 +17,7 @@
  /* Global variable */
 /*******************/
 
-Display		*display;
+extern Display		*display;
 
   /***********************/
  /* Function Prototypes */
