--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,7 @@ find_package( Threads    REQUIRED )
 find_package( GLM        REQUIRED )
 
 set_property(TARGET ${BIN_TARGET} PROPERTY CXX_STANDARD_REQUIRED ON)
-set_property(TARGET ${BIN_TARGET} PROPERTY CXX_STANDARD 11)
+set_property(TARGET ${BIN_TARGET} PROPERTY CXX_STANDARD 17)
 set_property(TARGET ${BIN_TARGET} PROPERTY CXX_EXTENSIONS OFF)
 
 # Includes
