Use system dev-db/sqlite:3 instead of a bundled version.

From: Lucio Sauer <watermanpaint@posteo.net>
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -108,7 +108,7 @@ pyo3 = { version = "0.20.0", features = ["extension-module", "abi3", "abi3-py39"
 rand = "0.8.5"
 regex = "1.10.2"
 reqwest = { version = "0.11.22", default-features = false, features = ["json", "socks", "stream", "multipart"] }
-rusqlite = { version = "0.30.0", features = ["trace", "functions", "collation", "bundled"] }
+rusqlite = { version = "0.30.0", features = ["trace", "functions", "collation"] }
 scopeguard = "1.2.0"
 serde = { version = "1.0.193", features = ["derive"] }
 serde-aux = "4.2.0"
