--- a/meshlab/mainwindow_Init.cpp
+++ b/meshlab/mainwindow_Init.cpp
@@ -607,7 +607,6 @@
 	helpMenu->addAction(onlineHelpAct);
 	helpMenu->addAction(onscreenHelpAct);
 	helpMenu->addAction(submitBugAct);
-	helpMenu->addAction(checkUpdatesAct);

 	fillEditMenu();
 	fillRenderMenu();
--- a/meshlab/mainwindow_RunTime.cpp
+++ b/meshlab/mainwindow_RunTime.cpp
@@ -2623,7 +2623,9 @@

 void MainWindow::helpOnline()
 {
+#if defined(__ENABLE_AUTO_STATS__)
 	checkForUpdates(false);
+#endif
 	QDesktopServices::openUrl(QUrl("http://www.meshlab.net/#support"));
 }

@@ -2882,6 +2884,7 @@
 void MainWindow::showEvent(QShowEvent * event)
 {
 	QWidget::showEvent(event);
+#if defined(__ENABLE_AUTO_STATS__)
 	QSettings settings;
 	QSettings::setDefaultFormat(QSettings::NativeFormat);
 	const QString versioncheckeddatestring("lastTimeMeshLabVersionCheckedOnStart");
@@ -2902,6 +2905,7 @@
 		settings.setValue(versioncheckeddatestring, todayStr);
 	}
 	sendUsAMail();
+#endif
 }

 void MainWindow::meshAdded(int mid)
