--- a/lib/crypto/configure.in
+++ b/lib/crypto/configure.in
@@ -26,10 +26,11 @@ dnl define([AC_CACHE_SAVE], )dnl
 AC_INIT(vsn.mk)
 
 if test -z "$ERL_TOP" || test ! -d "$ERL_TOP" ; then
-  AC_CONFIG_AUX_DIRS(autoconf)
+  :
+  AC_CONFIG_AUX_DIR(autoconf)
 else
   erl_top=${ERL_TOP}
-  AC_CONFIG_AUX_DIRS($erl_top/erts/autoconf)
+  AC_CONFIG_AUX_DIR(../../erts/autoconf)
 fi
 
 if test "X$host" != "Xfree_source" -a "X$host" != "Xwin32"; then
--- a/lib/megaco/configure.in
+++ b/lib/megaco/configure.in
@@ -26,10 +26,11 @@ dnl define([AC_CACHE_SAVE], )dnl
 AC_INIT(vsn.mk)
 
 if test -z "$ERL_TOP" || test ! -d $ERL_TOP ; then
-  AC_CONFIG_AUX_DIRS(autoconf)
+  :
+  AC_CONFIG_AUX_DIR(autoconf)
 else
   erl_top=${ERL_TOP}
-  AC_CONFIG_AUX_DIRS($erl_top/erts/autoconf)
+  AC_CONFIG_AUX_DIR(../../erts/autoconf)
 fi
 
 if test "X$host" != "Xfree_source" -a "X$host" != "Xwin32"; then
--- a/lib/odbc/configure.in
+++ b/lib/odbc/configure.in
@@ -26,10 +26,11 @@ dnl Process this file with autoconf to produce a configure script.
 AC_INIT(c_src/odbcserver.c)
 
 if test -z "$ERL_TOP" || test ! -d $ERL_TOP ; then
-  AC_CONFIG_AUX_DIRS(autoconf)
+  :
+  AC_CONFIG_AUX_DIR(autoconf)
 else
   erl_top=${ERL_TOP}
-  AC_CONFIG_AUX_DIRS($erl_top/erts/autoconf)
+  AC_CONFIG_AUX_DIR(../../erts/autoconf)
 fi
 
 if test "X$host" != "Xfree_source" -a "X$host" != "Xwin32"; then
--- a/lib/snmp/configure.in
+++ b/lib/snmp/configure.in
@@ -5,10 +5,11 @@ define([AC_CACHE_SAVE], )dnl
 AC_INIT(vsn.mk)
 
 if test -z "$ERL_TOP" || test ! -d $ERL_TOP ; then
-  AC_CONFIG_AUX_DIRS(autoconf)
+  :
+  AC_CONFIG_AUX_DIR(autoconf)
 else
   erl_top=${ERL_TOP}
-  AC_CONFIG_AUX_DIRS($erl_top/erts/autoconf)
+  AC_CONFIG_AUX_DIR(../../erts/autoconf)
 fi
 
 if test "X$host" != "Xfree_source" -a "X$host" != "Xwin32"; then
--- a/lib/wx/configure.in
+++ b/lib/wx/configure.in
@@ -20,7 +20,7 @@ dnl %CopyrightEnd%
 
 AC_INIT()
 
-AC_CONFIG_AUX_DIRS($srcdir/autoconf)
+AC_CONFIG_AUX_DIR($srcdir/autoconf)
 
 AC_PREREQ(2.59)
 
@@ -30,10 +30,11 @@ if test -f ./CONF_INFO; then
 fi
 
 if test -z "$ERL_TOP" || test ! -d $ERL_TOP ; then
+  :
   AC_MSG_ERROR([ERL_TOP is not set])
 else
   erl_top=${ERL_TOP}
-  AC_CONFIG_AUX_DIRS($erl_top/erts/autoconf)
+  AC_CONFIG_AUX_DIR(../../erts/autoconf)
   WX_BUILDING_INSIDE_ERLSRC=true
 fi
 
--- a/make/configure.in
+++ b/make/configure.in
@@ -82,7 +82,7 @@ AC_SUBST(ERL_TOP)
 dnl
 dnl Aux programs are found in erts/autoconf
 dnl
-AC_CONFIG_AUX_DIR(${srcdir}/erts/autoconf)
+AC_CONFIG_AUX_DIR(../erts/autoconf)
 
 dnl
 dnl Figure out what we are running on. And in violation of autoconf
