diff -Naur qrupdate-1.0.1/Makefile qrupdate-1.0.1.new/Makefile
--- qrupdate-1.0.1/Makefile	2009-02-06 04:12:00.000000000 -0500
+++ qrupdate-1.0.1.new/Makefile	2009-09-10 09:55:32.000000000 -0400
@@ -34,7 +34,7 @@
 lib:
 	make -C src/ lib
 solib:
-	make -C src/ solib
+	+make -C src/ solib
 test: lib
 	make -C test/
 
diff -Naur qrupdate-1.0.1/src/Makefile qrupdate-1.0.1.new/src/Makefile
--- qrupdate-1.0.1/src/Makefile	2009-02-06 04:12:50.000000000 -0500
+++ qrupdate-1.0.1.new/src/Makefile	2009-09-10 09:49:29.000000000 -0400
@@ -40,7 +40,7 @@
 	ar -cr $@ $(OBJS)
 
 ../libqrupdate.so: $(OBJS)
-	$(FC) $(FFLAGS) -shared -o $@ -Wl,-soname=libqrupdate.so.$(MAJOR) $(OBJS) \
+	$(FC) $(FFLAGS) $(LDFLAGS) -shared -o $@ -Wl,-soname=libqrupdate.so.$(MAJOR) $(OBJS) \
 		$(BLAS) $(LAPACK)
 
 $(OBJS): %.o: %.f
