--- a/bin/distro-install-desktop-integration	2018-10-29 20:55:29.000000000 +0100
+++ b/bin/distro-install-desktop-integration	2018-11-22 02:03:36.464866894 +0100
@@ -39,22 +39,20 @@
 
 create_man_link()
 {
-    echo "Install $MANDIR/man1/$1.1.gz"
+    echo "Install $MANDIR/man1/$1.1"
 
     mkdir -p $DESTDIR$MANDIR/man1
     echo ".so man1/$2.1" >| $DESTDIR$MANDIR/man1/$1.1
-    gzip -f $DESTDIR$MANDIR/man1/$1.1
-    test -f "$DESTDIR/$3" && echo "$MANDIR/man1/$1.1.gz" >>"$DESTDIR/$3"
+    test -f "$DESTDIR/$3" && echo "$MANDIR/man1/$1.1" >>"$DESTDIR/$3"
 }
 
 install_man()
 {
-    echo "Install $MANDIR/man1/$1.1.gz"
+    echo "Install $MANDIR/man1/$1.1"
 
     mkdir -p $DESTDIR$MANDIR/man1
     cp "${SRCDIR?}"/sysui/desktop/man/$1.1 $DESTDIR$MANDIR/man1 || exit 1;
-    gzip -f $DESTDIR$MANDIR/man1/$1.1
-    test -f "$DESTDIR/$2" && echo "$MANDIR/man1/$1.1.gz" >>"$DESTDIR/$2"
+    test -f "$DESTDIR/$2" && echo "$MANDIR/man1/$1.1" >>"$DESTDIR/$2"
 }
 
 
