Index: src/pptout/postscriptviewent.cc
===================================================================
--- src/pptout/postscriptviewent.cc	(revision 1517)
+++ src/pptout/postscriptviewent.cc	(arbetskopia)
@@ -143,7 +143,7 @@
     
     std::ostringstream tmp;
     tmp << psinterpreter //assuming it is ghostscript
-	<< " -q -dSAFER -dNOPAUSE -dBATCH"
+	<< " -q -dSAFE -dNOPAUSE -dBATCH"
 	<< " -sDEVICE=" << preferredDeviceName();
 
     // gs doesn't use fontconfig, so give font paths as parameters
