diff --git a/configure.ac b/configure.ac
index e59f631..ca329c1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6,7 +6,7 @@ AM_INIT_AUTOMAKE([foreign dist-bzip2])
 # For automake < 1.6 you need to use this line instead:
 #AM_INIT_AUTOMAKE(Passepartout, 0.7-PRE)
 
-AM_CONFIG_HEADER(src/defines.h)
+AC_CONFIG_HEADERS([src/defines.h])
 # AC_CONFIG_HEADER(src/defines.h)
 
 # We're just a few binaries...not worth doing shared by default?
