diff -urNad passepartout-0.7.0~/src/ps/misc.cc passepartout-0.7.0/src/ps/misc.cc
--- passepartout-0.7.0~/src/ps/misc.cc	2007-06-13 13:05:07.000000000 +0200
+++ passepartout-0.7.0/src/ps/misc.cc	2007-09-14 17:49:49.000000000 +0200
@@ -3,6 +3,8 @@
 ///
 #include "misc.h"
 
-std::ostream& PS::operator << (std::ostream& out, const PS::Concat& c) {
-  return out << '[' << c.matrix << "] concat\n";
+namespace PS {
+  std::ostream& operator << (std::ostream& out, const Concat& c) {
+    return out << '[' << c.matrix << "] concat\n";
+  }
 }
