diff -NurdpB passepartout-0.6/src/pptout/document/group.h passepartout-0.6-patched/src/pptout/document/group.h
--- passepartout-0.6/src/pptout/document/group.h	2004-10-02 16:18:45.000000000 -0500
+++ passepartout-0.6-patched/src/pptout/document/group.h	2007-03-21 19:00:13.000000000 -0500
@@ -30,7 +30,7 @@ public:
   void set_flow_around(bool);
   bool get_flow_around() const;
   Boundary get_obstacle_boundary() const;
-  BoundaryVect Group::obstacle_list() const;
+  BoundaryVect obstacle_list() const;
   
   void add(Pagent* obj);
   Pagent* ungroup(Pagent* obj);
@@ -55,7 +55,7 @@ public:
   /** Raised on direct changes to the group (i.e. new / ungrouped member) */
   SigC::Signal1<void, Pagent*> group_changed_signal; 
 private:
-  Boundary Group::get_untransformed_box() const;
+  Boundary get_untransformed_box() const;
   
   ChildVec childs;
 };
diff -NurdpB passepartout-0.6/src/pptout/propertiesdialog.h passepartout-0.6-patched/src/pptout/propertiesdialog.h
--- passepartout-0.6/src/pptout/propertiesdialog.h	2004-04-16 14:03:56.000000000 -0500
+++ passepartout-0.6-patched/src/pptout/propertiesdialog.h	2007-03-21 19:18:26.000000000 -0500
@@ -39,7 +39,7 @@ private:
   // no destructor
   ~PropertiesDialog();
 
-  void PropertiesDialog::show_page_contents(GtkNotebookPage*, guint index);
+  void show_page_contents(GtkNotebookPage*, guint index);
 
   Gtk::Notebook book;
   std::vector<PropBase*> pages;
diff -NurdpB passepartout-0.6/src/xml2ps/canvas.hh passepartout-0.6-patched/src/xml2ps/canvas.hh
--- passepartout-0.6/src/xml2ps/canvas.hh	2004-07-21 17:00:34.000000000 -0500
+++ passepartout-0.6-patched/src/xml2ps/canvas.hh	2007-03-21 18:47:12.000000000 -0500
@@ -52,7 +52,7 @@ namespace xml2ps {
     HBox hbox(bool span, float margin_left, float margin_right, 
 	      float ascender, float descender);
     
-    virtual void xml2ps::Canvas::addRelObstacle
+    virtual void addRelObstacle
     (const float left, const float bottom, const float right, const float top);
     
     // Check where the current column starts.
