--- main/__init__.py  2009-09-20 20:02:48 +0000
+++ main/__init__.py  2010-05-08 15:41:26 +0000
@@ -16,8 +16,11 @@
     # case the DISPLAY variable doesn't contain the screen number)
     DISPLAY = vfs.escape_path(gtk.gdk.Display(os.environ["DISPLAY"]).get_name()).replace("/", "_")
 except:
-    print "Error: could not open display", os.environ["DISPLAY"]
-    sys.exit(1)
+    try:
+        print >> sys.stderr, "Error: could not open display", os.environ["DISPLAY"]
+    except:
+        print >> sys.stderr, "(could not get DISPLAY environment variable)"
+    DISPLAY = "0"

