From eac425295227c5deda4536fd647ecf11b643e1ca Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Sun, 29 Aug 2021 09:22:33 -0400
Subject: [PATCH] Link bootctl against libshared_static

---
 meson.build | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 738879eb21..96042b6906 100644
--- a/meson.build
+++ b/meson.build
@@ -2150,9 +2150,8 @@ if conf.get('ENABLE_EFI') == 1 and conf.get('HAVE_BLKID') == 1
                 'bootctl',
                 'src/boot/bootctl.c',
                 include_directories : includes,
-                link_with : [libshared],
+                link_with : [libshared_static, libsystemd_static],
                 dependencies : [libblkid],
-                install_rpath : rootlibexecdir,
                 install : true)
 
         public_programs += executable(
-- 
2.33.0

