--- a/Makefile.in	Tue May 21 14:08:11 2002
+++ b/Makefile.in	Sun Aug 18 22:06:23 2002
@@ -591,7 +591,7 @@
 
 .8.8.html:
 	@rm -f $@
-	($(TBL) $< | $(NROFF) -mdoc2html > $@) || (rm -f $@; exit 1)
+	($(TBL) $< | $(NROFF) -mhtml > $@) || (rm -f $@; exit 1)
 
 .5.5.ps:
 	@rm -f $@
@@ -599,7 +599,7 @@
 
 .5.5.html:
 	@rm -f $@
-	($(TBL) $< | $(NROFF) -mdoc2html > $@) || (rm -f $@; exit 1)
+	($(TBL) $< | $(NROFF) -mhtml > $@) || (rm -f $@; exit 1)
 
 .cat5.cat5-dist:
 	@rm -f $@
