diff -uprN xml-xalan/c/src/xalanc/Makefile.in xml-xalan/c/src/xalanc/Makefile.in
--- xml-xalan/c/src/xalanc/Makefile.in	2007-03-31 14:54:16.106346968 -0400
+++ xml-xalan/c/src/xalanc/Makefile.in	2007-03-31 14:57:37.159195728 -0400
@@ -172,7 +172,7 @@ XalanExe: lib  "//'$(LOADSAMP)(${BATCH_X
 
 "//'$(LOADSAMP)(${BATCH_XALAN_UTIL_NAME})'": ${XSL_OBJ_DIR}/XalanExe.o
 	${LINK} $(XSL_BUILD_OPTIONS) ${PLATFORM_LIB_LINK_OPTIONS}  \
-	${EXTRA_LINK_OPTIONS}  $(XALAN_LIB) $(ALLLIBS) $^ -o $@	
+	${EXTRA_LINK_OPTIONS} $^ -o $@ $(XALAN_LIB) $(ALLLIBS)
 ${XSL_OBJ_DIR}/XalanExe.o: $(XALANEXE_DIR)/XalanExe.cpp                       
 	$(CC1) $(XSL_BUILD_OPTIONS) -c $(XSL_INCL) $(EXTRA_COMPILE_OPTIONS) -o $@ $<
 
@@ -180,7 +180,7 @@ testXSLT: lib  "//'$(LOADSAMP)(${BATCH_T
 
 "//'$(LOADSAMP)(${BATCH_TEST_UTIL_NAME})'": ${XSL_OBJ_DIR}/process.o
 	${LINK} $(XSL_BUILD_OPTIONS) ${PLATFORM_LIB_LINK_OPTIONS}  \
-	${EXTRA_LINK_OPTIONS} $(XALAN_LIB) $(ALLLIBS) $^ -o $@	
+	${EXTRA_LINK_OPTIONS} $^ -o $@ $(XALAN_LIB) $(ALLLIBS)
 ${XSL_OBJ_DIR}/process.o: $(TESTXSLT_DIR)/process.cpp                       
 	$(CC1) $(XSL_BUILD_OPTIONS) -c $(XSL_INCL) $(EXTRA_COMPILE_OPTIONS) -o $@ $<
 
@@ -190,7 +190,7 @@ XalanExe: lib  $(XSL_BIN_DIR)/Xalan
 
 $(XSL_BIN_DIR)/Xalan: ${XSL_OBJ_DIR}/XalanExe.o
 	${LINK} $(XSL_BUILD_OPTIONS) ${PLATFORM_LIB_LINK_OPTIONS}  \
-	${EXTRA_LINK_OPTIONS} $(XALAN_LIB) $(ALLLIBS) $^ -o $@	$(OTHER_LINK_PARAMETERS)
+	${EXTRA_LINK_OPTIONS} $^ -o $@ $(XALAN_LIB) $(ALLLIBS)	$(OTHER_LINK_PARAMETERS)
 ${XSL_OBJ_DIR}/XalanExe.o: $(XALANEXE_DIR)/XalanExe.cpp
 	$(CC1) $(XSL_BUILD_OPTIONS) -c $(XSL_INCL) $(EXTRA_COMPILE_OPTIONS) -o $@ $<
 endif
@@ -199,7 +199,7 @@ testXSLT: lib  $(XSL_BIN_DIR)/testXSLT
 
 $(XSL_BIN_DIR)/testXSLT: ${XSL_OBJ_DIR}/process.o
 	${LINK} $(XSL_BUILD_OPTIONS) ${PLATFORM_LIB_LINK_OPTIONS}  \
-	${EXTRA_LINK_OPTIONS} $(XALAN_LIB) $(ALLLIBS) $^ -o $@	$(OTHER_LINK_PARAMETERS)
+	${EXTRA_LINK_OPTIONS} $^ -o $@ $(XALAN_LIB) $(ALLLIBS)	$(OTHER_LINK_PARAMETERS)
 ${XSL_OBJ_DIR}/process.o: $(TESTXSLT_DIR)/process.cpp
 	$(CC1) $(XSL_BUILD_OPTIONS) -c $(XSL_INCL) $(EXTRA_COMPILE_OPTIONS) -o $@ $<
 
@@ -207,7 +207,7 @@ testXPath: lib  $(XSL_BIN_DIR)/testXPath
 
 $(XSL_BIN_DIR)/testXPath: ${XSL_OBJ_DIR}/TestXPath.o ${XSL_OBJ_DIR}/NodeNameTreeWalker.o
 	${LINK} $(XSL_BUILD_OPTIONS) ${PLATFORM_LIB_LINK_OPTIONS}  \
-	${EXTRA_LINK_OPTIONS} $(XALAN_LIB) $(ALLLIBS) $(CXXFLAGS) $^ -o $@	$(OTHER_LINK_PARAMETERS)
+	${EXTRA_LINK_OPTIONS} $^ -o $@ $(XALAN_LIB) $(ALLLIBS) $(CXXFLAGS)	$(OTHER_LINK_PARAMETERS)
 ${XSL_OBJ_DIR}/%.o:$(TESTXPATH_DIR)/%.cpp
 	$(CC1) $(XSL_BUILD_OPTIONS) -c $(XSL_INCL) $(EXTRA_COMPILE_OPTIONS) -o $@ $<
 
@@ -220,7 +220,7 @@ locale:        
 
 ${BATCH_TARGET}: $(ALL_OBJECTS)
 	@echo Building ${BATCH_TARGET}
-	$(MAKE_SHARED) $(ALLLIBS) $^ -o $@
+	$(MAKE_SHARED) $^ -o $@ $(ALLLIBS)
 
 else
 
@@ -230,7 +230,7 @@ locale:	
 	$(MAKE) -C Utils $@
 
 $(XSL_LIB_DIR)/$(LIB): $(ALL_OBJECTS)
-	$(MAKE_SHARED) $(PLATFORM_LIBRARIES) $(EXTRA_LINK_OPTIONS) $(LINK_SONAME) $(ALLLIBS) $^ -o $@ $(OTHER_LINK_LIB_PARAMERES)
+	$(MAKE_SHARED) $(PLATFORM_LIBRARIES) $(EXTRA_LINK_OPTIONS) $(LINK_SONAME) $^ -o $@ $(ALLLIBS) $(OTHER_LINK_LIB_PARAMERES)
 ifeq ($(PLATFORM), OS390)
 	mv *.x  ${XSL_LIB_DIR}        
 else
diff -uprN xml-xalan/c/src/xalanc/MsgCreator/Makefile.in xml-xalan/c/src/xalanc/MsgCreator/Makefile.in
--- xml-xalan/c/src/xalanc/Utils/MsgCreator/Makefile.in	2007-03-31 15:10:07.927334556 -0400
+++ xml-xalan/c/src/xalanc/Utils/MsgCreator/Makefile.in	2007-03-31 15:10:35.038804580 -0400
@@ -29,7 +29,7 @@ MSG_CREATOR_SRC=$(XALANCROOT)/src/xalanc
 
 $(XSL_BIN_DIR)/MsgCreator : $(MSG_CREATOR_OBJECTS)
 	${LINK} $(XSL_BUILD_OPTIONS) ${PLATFORM_LIB_LINK_OPTIONS} ${LIBRARY_SEARCH_PATHS} \
-	${EXTRA_LINK_OPTIONS}  $(XERCES_LIB) $^ -o $@ $(LOC_OTHER_LINK_PARAMETERS)
+	${EXTRA_LINK_OPTIONS} $^ -o $@ $(XERCES_LIB) $(LOC_OTHER_LINK_PARAMETERS)
 	
 $(XSL_OBJ_DIR)/%.o : $(MSG_CREATOR_SRC)/%.cpp    
 	$(CC1) $(XSL_BUILD_OPTIONS) -c $(XSL_INCL) $(EXTRA_COMPILE_OPTIONS) -o $@ $<
