diff -Nur httpunit-1.6.2/src/com/meterware/httpunit/javascript/JavaScript.java httpunit-1.6.2_fixed/src/com/meterware/httpunit/javascript/JavaScript.java
--- httpunit-1.6.2/src/com/meterware/httpunit/javascript/JavaScript.java	2006-03-26 22:16:46.000000000 +0300
+++ httpunit-1.6.2_fixed/src/com/meterware/httpunit/javascript/JavaScript.java	2007-10-14 20:34:03.000000000 +0300
@@ -838,8 +838,6 @@
         static ElementArray newElementArray( Scriptable parent ) {
             try {
                 return (ElementArray) Context.getCurrentContext().newObject( parent, "ElementArray" );
-            } catch (PropertyException e) {
-                throw new RhinoException( e );
             } catch (NotAFunctionException e) {
                 throw new RhinoException( e );
             } catch (JavaScriptException e) {
@@ -1144,10 +1142,10 @@
 
 class RhinoException extends RuntimeException {
 
-    private Exception _cause;
+    private Throwable _cause;
 
 
-    public RhinoException( Exception cause ) {
+    public RhinoException( Throwable cause ) {
         _cause = cause;
     }
 
