diff -ur guile-gtk-2.1.orig/glade/glade-support.c guile-gtk-2.1/glade/glade-support.c
--- guile-gtk-2.1.orig/glade/glade-support.c	2016-08-12 12:00:45.894290499 -0600
+++ guile-gtk-2.1/glade/glade-support.c	2016-08-12 12:02:33.186666585 -0600
@@ -17,7 +17,6 @@
 
 #include <glade/glade-xml.h>
 #include <libguile.h>
-#include <guile/gh.h>
 
 #include "config.h"
 #include "guile-gtk.h"
@@ -83,7 +82,7 @@
       sgtk_protshell *protshell;
       GClosure *closure;
 
-      cb = scm_internal_cwdr ((scm_t_catch_body) gh_eval_str, (char*) handler_name,
+      cb = scm_internal_cwdr ((scm_t_catch_body) scm_c_eval_string, (char*) handler_name,
 			      scm_handle_by_message_noexit, "glade",
 			      &stack_item);
 
diff -ur guile-gtk-2.1.orig/gtk-gl/gdk-gl-support.c guile-gtk-2.1/gtk-gl/gdk-gl-support.c
--- guile-gtk-2.1.orig/gtk-gl/gdk-gl-support.c	2016-08-12 12:00:45.894290499 -0600
+++ guile-gtk-2.1/gtk-gl/gdk-gl-support.c	2016-08-12 12:02:19.530121760 -0600
@@ -19,7 +19,6 @@
 #include <gtkgl/gdkgl.h>
 
 #include <libguile.h>
-#include <guile/gh.h>
 
 typedef struct {
   char*		name;
@@ -100,7 +99,7 @@
 	    if (! SCM_INUMP (s_value))
 	      goto error;
 	    
-	    *aptr++ = gh_scm2int (s_value);
+	    *aptr++ = scm_to_int (s_value);
 	  }
       }
     }
diff -ur guile-gtk-2.1.orig/guile-gtk.c guile-gtk-2.1/guile-gtk.c
--- guile-gtk-2.1.orig/guile-gtk.c	2016-08-12 12:00:45.890957197 -0600
+++ guile-gtk-2.1/guile-gtk.c	2016-08-12 12:01:59.346968496 -0600
@@ -28,7 +28,6 @@
 #include <gdk/gdkx.h>
 
 #include <libguile.h>
-#include <guile/gh.h>
 #include <libguile/dynl.h>
 #include <libguile/tags.h>
 
@@ -2196,7 +2195,7 @@
 #if 0
       /* Gone in Gtk 2 */
     case GTK_TYPE_CALLBACK:
-      return gh_procedure_p (obj);
+      return scm_is_true (scm_procedure_p (obj));
 #endif
     case G_TYPE_OBJECT:
       return sgtk_is_a_gtkobj (type, obj);
