--- libffado-2.4.1-old/support/mixer-qt4/ffado/mixer/globalmixer.py
+++ libffado-2.4.1/support/mixer-qt4/ffado/mixer/globalmixer.py
@@ -96,7 +96,6 @@ class GlobalMixer(QWidget):
     @pyqtSlot()
     def on_txtNickname_returnPressed( self ):
         if self.nickname.canChangeValue():
-            asciiData = self.txtNickname.text().toAscii()
-            self.nickname.setText( asciiData.data() )
+            self.nickname.setText( self.txtNickname.text().encode('utf-8') )
         else:
             self.txtNickname.setText( self.nickname.text() )
