--- libffado-2.4.1-old/support/mixer-qt4/SConscript	2015-04-12 20:18:15.000000000 +0900
+++ libffado-2.4.1/support/mixer-qt4/SConscript	2018-06-26 19:59:19.015719710 +0900
@@ -29,16 +29,12 @@
 if env['BUILD_MIXER'] == 'true':
     e = env.Clone()
 
-    def findfiles( arg, dirname, names ):
-        for name in names:
-            if name.startswith("."):
-                names.remove(name)
-        for name in names:
-            if '.pyc' not in name and '.in' not in name:
-                arg.append( os.path.join( dirname, name ) )
-
     pythonfiles = [ 'ffado/config.py' ]
-    os.path.walk( "ffado", findfiles, pythonfiles )
+    for root, dirs, files in os.walk( "ffado" ):
+        for name in files:
+            if name.endswith( '.pyc' ) or '.in' in name or name.startswith("."):
+                continue
+            pythonfiles.append( os.path.join( root, name ) )
 
     e.ScanReplace( "ffado/config.py.in" )
     e.Depends( "ffado/config.py", "#/SConstruct" )
