From 15e2351d40763acee5d246df7c725c3bd259c304 Mon Sep 17 00:00:00 2001
From: Alexey Sokolov <alexey+znc@asokolov.org>
Date: Wed, 26 May 2021 10:10:20 +0100
Subject: [PATCH] Switch --makeconf wizard from freenode to libera

---
 src/znc.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/znc.cpp b/src/znc.cpp
index c5ad17dc69..365367545a 100644
--- a/src/znc.cpp
+++ b/src/znc.cpp
@@ -778,7 +778,7 @@ bool CZNC::WriteNewConfig(const CString& sConfigFile) {
         CUtils::PrintMessage("");
 
         do {
-            CUtils::GetInput("Name", sNetwork, "freenode");
+            CUtils::GetInput("Name", sNetwork, "libera");
         } while (!CIRCNetwork::IsValidNetwork(sNetwork));
 
         vsLines.push_back("\t<Network " + sNetwork + ">");
@@ -795,8 +795,8 @@ bool CZNC::WriteNewConfig(const CString& sConfigFile) {
         bool bSSL = false;
         unsigned int uServerPort = 0;
 
-        if (sNetwork.Equals("freenode")) {
-            sHost = "chat.freenode.net";
+        if (sNetwork.Equals("libera")) {
+            sHost = "irc.libera.chat";
 #ifdef HAVE_LIBSSL
             bSSL = true;
 #endif
From 688645413c258f1fe42a39e42e5b5d1dead03d71 Mon Sep 17 00:00:00 2001
From: Alexey Sokolov <alexey+znc@asokolov.org>
Date: Fri, 18 Jun 2021 21:20:53 +0100
Subject: [PATCH] Fix integration test after switch to libera

---
 test/integration/framework/znctest.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/integration/framework/znctest.cpp b/test/integration/framework/znctest.cpp
index 195b6083a9..40dae85fc2 100644
--- a/test/integration/framework/znctest.cpp
+++ b/test/integration/framework/znctest.cpp
@@ -39,7 +39,7 @@ void WriteConfig(QString path) {
     p.ReadUntil("Real name");               p.Write();
     p.ReadUntil("Bind host");               p.Write();
     p.ReadUntil("Set up a network?");       p.Write();
-    p.ReadUntil("Name [freenode]");         p.Write("test");
+    p.ReadUntil("Name [libera]");           p.Write("test");
     p.ReadUntil("Server host (host only)"); p.Write("127.0.0.1");
     p.ReadUntil("Server uses SSL?");        p.Write();
     p.ReadUntil("6667");                    p.Write();
