--- FreeWnn-1.1.1-a021.orig/Wnn/jlib/Makefile.in	2004-09-12 18:56:55.000000000 +0200
+++ FreeWnn-1.1.1-a021/Wnn/jlib/Makefile.in	2009-12-29 20:29:18.086539501 +0100
@@ -101,7 +101,7 @@
 all:: libwnn.la
 
 libwnn.la: $(OBJS) $(EXTRALIBRARYDEPS)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJS:.o=.lo) -rpath $(WNNLIBDIR) -version-info $(WNNJLIBVERSION)
+	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJS:.o=.lo) @LIBS@ -rpath $(WNNLIBDIR) -version-info $(WNNJLIBVERSION)
 
 lintlib:: llib-lwnn.ln
 
