diff -ur wmsystray-0.1.1/Makefile wmsystray-0.1.1.new/Makefile
--- wmsystray-0.1.1/Makefile	2004-03-01 05:35:47.000000000 +0100
+++ wmsystray-0.1.1.new/Makefile	2007-06-30 18:09:59.000000000 +0200
@@ -1,13 +1,14 @@
 prefix=/usr
 bindir = ${prefix}/bin
 mandir = ${prefix}/share/man
-INSTALL = /bin/install
+INSTALL = install
 INSTALLPROG = ${INSTALL} -m 0755
 INSTALLMAN = ${INSTALL} -m 0644
+EXTRACFLAGS = 
 
 all:
-	make -C xembed xembed.o
-	make -C wmsystray
+	make EXTRACFLAGS="${EXTRACFLAGS}" -C xembed xembed.o
+	make EXTRACFLAGS="${EXTRACFLAGS}" -C wmsystray
 
 clean:
 	make -C xembed clean
diff -ur wmsystray-0.1.1/wmsystray/Makefile wmsystray-0.1.1.new/wmsystray/Makefile
--- wmsystray-0.1.1/wmsystray/Makefile	2004-03-01 04:54:44.000000000 +0100
+++ wmsystray-0.1.1.new/wmsystray/Makefile	2007-06-30 18:09:59.000000000 +0200
@@ -1,7 +1,7 @@
 include ../Rules.make
 
 CC = gcc
-CFLAGS = ${X11CFLAGS} -DTRACE_LEVEL=${TRACE_LEVEL} -I../xembed -Wall -g
+CFLAGS = ${X11CFLAGS} -DTRACE_LEVEL=${TRACE_LEVEL} -I../xembed -g ${EXTRACFLAGS}
 LDFLAGS = ${X11LDFLAGS} -lXpm
 
 OBJS = main.o ui.o systray.o xpms.o
diff -ur wmsystray-0.1.1/xembed/Makefile wmsystray-0.1.1.new/xembed/Makefile
--- wmsystray-0.1.1/xembed/Makefile	2004-03-01 04:54:44.000000000 +0100
+++ wmsystray-0.1.1.new/xembed/Makefile	2007-06-30 18:09:59.000000000 +0200
@@ -1,6 +1,6 @@
 include ../Rules.make
 
-CFLAGS = -DTRACE_LEVEL=${TRACE_LEVEL} ${X11CFLAGS} -Wall -g
+CFLAGS = -DTRACE_LEVEL=${TRACE_LEVEL} ${X11CFLAGS} -g ${EXTRACFLAGS}
 LDFLAGS = ${X11LDFLAGS}
 
 all: xembed.o
