--- a/configure.ac
+++ b/configure.ac
@@ -14,12 +14,12 @@
         [fastx_toolkit])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR([m4])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([dist-bzip2])
 
 AC_PROG_CC
 AC_PROG_CXX
-AC_PROG_LIBTOOL
+LT_INIT
 AX_C_LONG_LONG
 AX_CXX_HEADER_STDCXX_TR1
 AX_CXX_COMPILE_STDCXX_11([noext],[optional])
@@ -31,9 +31,9 @@
 PKG_CHECK_MODULES([GTEXTUTILS],[gtextutils])
 
 dnl --enable-wall
-EXTRA_CHECKS="-Wall -Wextra -Wformat-nonliteral -Wformat-security -Wswitch-default -Wswitch-enum -Wunused-parameter -Wfloat-equal -Werror"
+EXTRA_CHECKS="-Wall -Wextra -Wformat-nonliteral -Wformat-security -Wswitch-default -Wswitch-enum -Wunused-parameter -Wfloat-equal"
 AC_ARG_ENABLE(wall,
-[  --enable-wall          Enable many common GCC warnings (-Wall,-Wextra, -Werror etc., default enabled)],
+[  --enable-wall          Enable many common GCC warnings (-Wall,-Wextra etc., default enabled)],
 [case "${enableval}" in
   yes) wall=true ;;
   no)  wall=false ;;
@@ -45,22 +45,6 @@
   CXXFLAGS="${CXXFLAGS} ${EXTRA_CHECKS}"
 fi
 
-dnl --enable-debug
-AC_ARG_ENABLE(debug,
-[  --enable-debug          Enable debug mode (default enabled)],
-[case "${enableval}" in
-  yes) debug=true ;;
-  no)  debug=false ;;
-  *) AC_MSG_ERROR(bad value ${enableval} for --enable-debug) ;;
-esac],[debug=true])
-if test "$debug" = "true"
-then
-  CFLAGS="${CFLAGS} -DDEBUG -g -O1"
-  CXXFLAGS="${CXXFLAGS} -DDEBUG -g -O1"
-else
-  CFLAGS="${CFLAGS} -O3"
-  CXXFLAGS="${CXXFLAGS} -O3"
-fi
 
 dnl 'all-static' marco copied from subversion's configure.ac
 dnl Check for --enable-all-static option
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,7 +10,7 @@
 
 EXTRA_DIST = reconf configure README install_galaxy_files.sh
 
-SUBDIRS = m4 src doc galaxy scripts build_scripts
+SUBDIRS = src doc galaxy scripts build_scripts
 
 ACLOCAL_AMFLAGS = -I m4
 
