--- a/setup.py
+++ b/setup.py
@@ -18,7 +18,7 @@ setup(name=name,
       author="Harald Hoyer",
       author_email="augeas-devel@redhat.com",
       description="""Python bindings for Augeas""",
-      packages=find_packages(exclude=('test')),
+      packages=find_packages(exclude=('test',)),
       setup_requires=["cffi>=1.0.0"],
       cffi_modules=["augeas/ffi.py:ffi"],
       install_requires=["cffi>=1.0.0"],
