diff '--color=auto' -Naur a/SConstruct b/SConstruct
--- a/SConstruct
+++ b/SConstruct
@@ -182,7 +182,7 @@
 # Python Package Settings
 python_min_version = parse_version("3.8")
 # Newest Python version not supported/tested by Cantera
-python_max_version = parse_version("3.14")
+python_max_version = parse_version("3.15")
 # The string is used to set python_requires in setup.cfg.in
 py_requires_ver_str = f">={python_min_version},<{python_max_version}"
 
