diff --git a/src/Mixfix/Makefile.am b/src/Mixfix/Makefile.am
index d811631..295f0c6 100644
--- a/src/Mixfix/Makefile.am
+++ b/src/Mixfix/Makefile.am
@@ -72,7 +72,7 @@ libmixfix_a_SOURCES = \
 EXTRA_libmixfix_a_SOURCES = \
 	compiler.cc
 
-BUILT_SOURCES = surface.yy surface.cc surface.h lexer.cc
+BUILT_SOURCES = surface.cc lexer.cc
 CLEANFILES = surface.yy surface.cc surface.h lexer.cc
 
 surface.yy: \
@@ -87,7 +87,9 @@ surface.yy: \
 	$(srcdir)/bottom.yy \
 	> surface.yy
 
-surface.cc surface.h: surface.yy
+surface.h: surface.yy
+
+surface.cc: surface.h
 	$(BISON) -dv surface.yy -o surface.c
 	mv surface.c surface.cc
 
