--- src/instrument.c.old	2010-10-15 22:20:58.014091858 +0200
+++ src/instrument.c	2010-10-15 22:21:34.132271440 +0200
@@ -542,7 +542,7 @@ while (dbcp->c_get(dbcp, &key, &value, D
    
    tthen = (time_t)then;
 
-   snprintf(datebuf,CF_BUFSIZE-1,"%s",ctime(&tthen));
+   snprintf(datebuf,CF_MAXVARSIZE-1,"%s",ctime(&tthen));
    datebuf[strlen(datebuf)-9] = '\0';                     /* Chop off second and year */
 
    snprintf(addr,15,"%s",hostname+1);
