Description: Fix FTBFS due to glib gvariant type definitions
Forwarded: not-needed
Bug-Debian: http://bugs.debian.org/577329
Author: Sven Eckelmann <sven.eckelmann@gmx.de>

---
diff --git a/rice_video/typedefs.h b/rice_video/typedefs.h
index 4cded542b06445f043fa1fabf77e4d20c9d52920..b7e16e9c4ba3248f9fed9762f79349c137ee2026 100644
--- a/rice_video/typedefs.h
+++ b/rice_video/typedefs.h
@@ -20,10 +20,12 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 #ifndef _TYPEDEFS_H_
 #define _TYPEDEFS_H_
 
-#define uchar  unsigned char
-#define uint16 unsigned short
-#define uint32 unsigned int
-#define uint64 unsigned long long
+#include <stdint.h>
+
+typedef uint8_t uchar;
+typedef uint16_t uint16;
+typedef uint32_t uint32;
+typedef uint64_t uint64;
 
 typedef unsigned char               uint8;
 
