diff -urN kdelibs-4.6.3.old/kinit/CMakeLists.txt kdelibs-4.6.3/kinit/CMakeLists.txt
--- kdelibs-4.6.3.old/kinit/CMakeLists.txt	2011-05-02 00:19:10.000000000 +0200
+++ kdelibs-4.6.3/kinit/CMakeLists.txt	2011-05-02 00:20:32.000000000 +0200
@@ -170,15 +170,6 @@
   install(TARGETS start_kdeinit_wrapper DESTINATION ${LIBEXEC_INSTALL_DIR})
 endif (NOT WIN32)
 
-if (CMAKE_SYSTEM_NAME MATCHES Linux)
-    MESSAGE(STATUS "Using setuid root kdeinit wrapper in order to protect it from bad Linux OOM-killer")
-    set(KDEINIT_OOM_PROTECT 1)
-    install(CODE "
-        set(START_KDEINIT_PATH \"\$ENV{DESTDIR}${LIBEXEC_INSTALL_DIR}/start_kdeinit\")
-        EXECUTE_PROCESS(COMMAND sh -c \"chown 0 '\${START_KDEINIT_PATH}' && chmod u+s '\${START_KDEINIT_PATH}'\")
-    ")
-endif (CMAKE_SYSTEM_NAME MATCHES Linux)
-
 ########### install files ###############
 
 install( FILES ${CMAKE_CURRENT_BINARY_DIR}/org.kde.KLauncher.xml DESTINATION ${DBUS_INTERFACES_INSTALL_DIR} )
