--- Makefile.in.orig	2003-02-20 21:57:38.000000000 -0800
+++ Makefile.in	2003-02-20 22:27:10.000000000 -0800
@@ -2,10 +2,12 @@
 CFLAGS=-Wall -g
 BINS=cdctl sndreset
 OBJS=cdctl.o version.o ioctls.o iso_header.o
-DOCS=NEWS NUTSANDBOLTS LICENSE PUBLICKEY README SEGFAULTS
+DOCS=NEWS NUTSANDBOLTS LICENSE PUBLICKEY README
 MANS=cdctl.1
 MOS=cdctl.mo
 
+prefix = @prefix@
+exec_prefix = @exec_prefix@
 bindir = @bindir@
 mandir = @mandir@
 datadir = @datadir@
@@ -16,6 +18,10 @@
 INSTALL = @INSTALL@
 INSTALL_DATA = $(INSTALL) -m 644
 
+DESTDIR =
+
+man1dir = $(mandir)/man1
+
 all: $(BINS)
 
 cdctl: $(OBJS)
@@ -26,21 +32,21 @@
 version.o: version.c
 iso_header.o: iso_header.c
 
-install: install-bins install-docs install-mans install-mos
+install: install-bins install-mans install-mos
 
 install-bins: $(BINS)
-	$(INSTALL) -s $(BINS) $(bindir)
-
-install-docs: $(DOCS)
-	$(INSTALL_DATA) $(DOCS) $(datadir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) $(BINS) $(DESTDIR)$(bindir)
 
 install-mans: $(MANS)
-	$(INSTALL_DATA) $(MANS) $(mandir)
+	$(INSTALL) -d $(DESTDIR)$(man1dir)
+	$(INSTALL_DATA) $(MANS) $(DESTDIR)$(man1dir)
 
 # FIXME: need to deal with other languages besides German
 
 install-mos: cdctl.mo
-	$(INSTALL_DATA) $(MOS) $(localedir)/de/LC_MESSAGES
+	$(INSTALL) -d $(DESTDIR)$(localedir)/de/LC_MESSAGES
+	$(INSTALL_DATA) $(MOS) $(DESTDIR)$(localedir)/de/LC_MESSAGES
 
 cdctl.mo: cdctl.po
 	msgfmt cdctl.po -o cdctl.mo
