diff --git a/libimchooseui/imchooseui.c b/libimchooseui/imchooseui.c
index f135aff..dd5881a 100644
--- a/libimchooseui/imchooseui.c
+++ b/libimchooseui/imchooseui.c
@@ -41,14 +41,14 @@ static void     _imchoose_ui_switch_im_finish   (GObject      *source_object,
                                                  GAsyncResult *res,
                                                  gpointer      user_data);
 
-typedef struct _IMChooseUIPrivate {
+struct _IMChooseUIPrivate {
 	IMSettingsClient *client;
 	gchar            *default_im;
 	gchar            *initial_im;
 	gchar            *current_im;
 	guint             note_type;
 	gboolean          clicked:1;
-} IMChooseUIPrivate;
+};
 enum {
 	POS_ICON = 0,
 	POS_LABEL,
