diff -ur NVIDIA_GLX-1.0-4191/usr/include/GL/glx.h NVIDIA_GLX-1.0-4191.new/usr/include/GL/glx.h
--- NVIDIA_GLX-1.0-4191/usr/include/GL/glx.h	2002-12-09 21:26:55.000000000 +0100
+++ NVIDIA_GLX-1.0-4191.new/usr/include/GL/glx.h	2003-01-30 18:20:23.000000000 +0100
@@ -39,6 +39,7 @@
 typedef XID GLXPixmap;
 typedef XID GLXDrawable;
 typedef XID GLXPbuffer;
+typedef XID GLXPbufferSGIX;
 typedef XID GLXWindow;
 typedef XID GLXFBConfigID;
 
