Index: lxtask-0.1.4/configure.ac
===================================================================
--- lxtask-0.1.4.orig/configure.ac
+++ lxtask-0.1.4/configure.ac
@@ -44,7 +44,7 @@ AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [Gettext package.])
 
 #These codes are NOT maintained by Pootle ca el en_GB lv sq
-ALL_LINGUAS="af ar ca cs da de el en_GB es et eu fa fi fr gl hr hu id it ja ko lt lv ml nb nl nn pl ps pt pt_BR ru sk sl sq sv tr uk ur ur_PK vi zh_CN zh_TW"
+ALL_LINGUAS="af ar ca cs da de el es et eu fa fi fr gl hr hu id it ja ko lt ml nb nl nn pl ps pt pt_BR ru sk sl sv tr uk ur ur_PK vi zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 AC_OUTPUT([
