diff --git a/jinja2/environment.py b/jinja2/environment.py
index ebb5454..1b5dc40 100644
--- a/jinja2/environment.py
+++ b/jinja2/environment.py
@@ -570,6 +570,10 @@ class Environment(object):
             py_header = imp.get_magic() + \
                 u'\xff\xff\xff\xff'.encode('iso-8859-15')
 
+            # Python 3.3 added a source filesize to the header
+            if sys.version_info >= (3, 3):
+                py_header += u'\x00\x00\x00\x00'.encode('iso-8859-15')
+
         def write_file(filename, data, mode):
             if zip:
                 info = ZipInfo(filename)
