diff -ruN kdepim-runtime-4.4.11.1.orig/resourcetester/xmloperations.cpp kdepim-runtime-4.4.11.1/resourcetester/xmloperations.cpp
--- kdepim-runtime-4.4.11.1.orig/resourcetester/xmloperations.cpp	2013-12-15 15:41:43.629536906 +0100
+++ kdepim-runtime-4.4.11.1/resourcetester/xmloperations.cpp	2013-12-15 15:42:33.492539695 +0100
@@ -34,7 +34,6 @@
 #include <QStringList>
 
 using namespace Akonadi;
-using namespace AkonadiFuture;
 
 template <typename T> QTextStream& operator<<( QTextStream &s, const QSet<T> &set )
 {
diff -ruN kdepim-runtime-4.4.11.1.orig/resourcetester/xmloperations.h kdepim-runtime-4.4.11.1/resourcetester/xmloperations.h
--- kdepim-runtime-4.4.11.1.orig/resourcetester/xmloperations.h	2013-12-15 15:41:43.629536906 +0100
+++ kdepim-runtime-4.4.11.1/resourcetester/xmloperations.h	2013-12-15 15:42:22.195539063 +0100
@@ -114,7 +114,7 @@
 
   private:
     Akonadi::Collection::List mRoots;
-    AkonadiFuture::XmlDocument mDocument;
+    Akonadi::XmlDocument mDocument;
     QString mFileName;
     QString mErrorMsg;
     CollectionFields mCollectionFields;
