diff -u --new-file gkrellm-bluez-0.2-orig/src/decal_net_leds.xpm gkrellm-bluez-0.2/src/decal_net_leds.xpm
--- gkrellm-bluez-0.2-orig/src/decal_net_leds.xpm	1969-12-31 18:00:00.000000000 -0600
+++ gkrellm-bluez-0.2/src/decal_net_leds.xpm	2007-03-08 09:07:50.000000000 -0600
@@ -0,0 +1,66 @@
+/* XPM */
+static char * decal_net_leds_xpm[] = {
+"10 12 51 1",
+" 	c None",
+".	c #222323",
+"+	c #3A3A3A",
+"@	c #4C4F4F",
+"#	c #3E3F3F",
+"$	c #262727",
+"%	c #334735",
+"&	c #335E4F",
+"*	c #337568",
+"=	c #2C857C",
+"-	c #259790",
+";	c #1EA8A4",
+">	c #18B5B4",
+",	c #20A39E",
+"'	c #2A786D",
+")	c #314E3E",
+"!	c #526E55",
+"~	c #48846F",
+"{	c #3F9988",
+"]	c #36B0A2",
+"^	c #2CC8BF",
+"/	c #22E0DA",
+"(	c #1BF1EF",
+"_	c #25D8D2",
+":	c #3AA595",
+"<	c #4E765F",
+"[	c #141414",
+"}	c #272929",
+"|	c #3D3F3F",
+"1	c #2D2E2E",
+"2	c #171717",
+"3	c #5D421D",
+"4	c #84612F",
+"5	c #AD8643",
+"6	c #D0A456",
+"7	c #CDA255",
+"8	c #BA914B",
+"9	c #A68040",
+"0	c #936E37",
+"a	c #805F2E",
+"b	c #6E5125",
+"c	c #7F643C",
+"d	c #A48350",
+"e	c #CDA666",
+"f	c #F1C479",
+"g	c #EEC277",
+"h	c #DAB16D",
+"i	c #C6A062",
+"j	c #B39058",
+"k	c #A0814F",
+"l	c #8F7345",
+".+@@@@@@#$",
+".+@@@@@@#$",
+".+@@@@@@#$",
+"%&*=-;>,')",
+"!~{]^/(_:<",
+"%&*=-;>,')",
+"[}||||||12",
+"[}||||||12",
+"[}||||||12",
+"34567890ab",
+"cdefghijkl",
+"34567890ab"};
diff -u --new-file gkrellm-bluez-0.2-orig/src/gkrellm-bluez-chart.c gkrellm-bluez-0.2/src/gkrellm-bluez-chart.c
--- gkrellm-bluez-0.2-orig/src/gkrellm-bluez-chart.c	2006-11-14 13:50:24.000000000 -0600
+++ gkrellm-bluez-0.2/src/gkrellm-bluez-chart.c	2007-03-08 09:13:29.000000000 -0600
@@ -22,6 +22,8 @@
 
 #include "gkrellm-bluez-chart.h"
 
+#include "decal_net_leds.xpm"
+
 #define DECAL_NET_LEDS      "decal_net_leds"
 
 #define GRID_RESOLUTION_MIN 0
@@ -237,7 +239,7 @@
 
     /* Check for theme_dir/net/decal_net_leds.png.
      */
-    gkrellm_load_piximage(DECAL_NET_LEDS, NULL,
+    gkrellm_load_piximage(DECAL_NET_LEDS, decal_net_leds_xpm,
 			  &decal_net_led_piximage, NET_STYLE_NAME);
     
     w = gdk_pixbuf_get_width(decal_net_led_piximage->pixbuf);
