--- tcltk/Makefile.old	2011-04-26 17:40:36.000000000 +0000
+++ tcltk/Makefile	2011-04-26 17:41:05.000000000 +0000
@@ -26,7 +26,7 @@
 	${CP} irsimexec $(DESTDIR)${TCLDIR}/irsimexec
 
 irsimexec: irsimexec.c tclirsim.o
-	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} irsimexec.c -o irsimexec \
+	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} ${LDFLAGS} irsimexec.c -o irsimexec \
 		${LIBS} ${LIB_SPECS}
 
 irsim.tcl: irsim.tcl.in
--- irsim/Makefile.old	2011-04-26 17:46:38.000000000 +0000
+++ irsim/Makefile	2011-04-26 17:47:21.000000000 +0000
@@ -22,7 +22,7 @@
 tclirsim${SHDLIB_EXT}: ${EXTRA_LIBS}
 	@echo --- making irsim Tcl library \(tclirsim${SHDLIB_EXT}\)
 	${RM} tclirsim${SHDLIB_EXT}
-	${CC} ${CFLAGS} ${CPPFLAGS} -o $@ ${LDDL_FLAGS} ${LD_RUN_PATH} \
+	${CC} ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} -o $@ ${LDDL_FLAGS} ${LD_RUN_PATH} \
 		${EXTRA_LIBS} -lc ${LIBS} ${LD_EXTRA_LIBS}
 
 $(DESTDIR)${BINDIR}/irsim:
--- random/Makefile.old	2011-04-26 17:48:46.000000000 +0000
+++ random/Makefile	2011-04-26 17:49:02.000000000 +0000
@@ -21,7 +21,7 @@
 random${SHDLIB_EXT}: randwrap.o
 	@echo --- making random Tcl library
 	${RM} random${SHDLIB_EXT}
-	${CC} ${CFLAGS} ${CPPFLAGS} -o $@ ${LDDL_FLAGS} randwrap.o \
+	${CC} ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} -o $@ ${LDDL_FLAGS} randwrap.o \
 		${MAC_SHARED} -lc ${LIBS} ${LD_EXTRA_LIBS}
 
 install-tcl: $(DESTDIR)${TCLDIR}/random${SHDLIB_EXT}
--- tclsubckt/Makefile.old	2011-04-26 17:50:23.000000000 +0000
+++ tclsubckt/Makefile	2011-04-26 17:50:45.000000000 +0000
@@ -22,7 +22,7 @@
 diglib${SHDLIB_EXT}: diglib.o
 	@echo --- making diglib subcircuit shared library
 	${RM} diglib${SHDLIB_EXT}
-	${CC} ${CFLAGS} ${CPPFLAGS} -o $@ ${LDDL_FLAGS} diglib.o \
+	${CC} ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} -o $@ ${LDDL_FLAGS} diglib.o \
 		${MAC_SHARED} -lc ${LIBS} ${LD_EXTRA_LIBS}
 
 install-tcl: $(DESTDIR)${TCLDIR}/diglib${SHDLIB_EXT}
