--- a/contrib/sb-capstone/tests.lisp.ORIG	2018-12-04 17:44:02.000000000 +0100
+++ b/contrib/sb-capstone/tests.lisp	2019-01-03 10:22:33.649177090 +0100
@@ -65,10 +65,10 @@
                 t)
 
 
-(deftest ppc-little-endian (capstone-check '(#x03 #x10 #x40 #x3c #x00 #x81 #x42 #x38)
+(deftest ppc-little-endian (capstone-check '(#x03 #x10 #x40 #x3c #x00 #x71 #x42 #x38)
                                            '(:ppc64 :little-endian)
                                            '("LIS R2, 0X1003"
-                                             "ADDI R2, R2, -0X7F00"))
+                                             "ADDI R2, R2, 0X7100"))
                            t)
 
 (deftest ppc-big-endian (capstone-check '(#x7c #x08 #x02 #xa6)
