diff --git a/lib/matplotlib/testing/decorators.py b/lib/matplotlib/testing/decorators.py
index 5e738ad9b..78b3f8169 100644
--- a/lib/matplotlib/testing/decorators.py
+++ b/lib/matplotlib/testing/decorators.py
@@ -266,7 +266,7 @@ class ImageComparisonTest(CleanupTest):
 
                 yield do_test, fignum, actual_fname, expected_fname
 
-def image_comparison(baseline_images=None, extensions=None, tol=0,
+def image_comparison(baseline_images=None, extensions=None, tol=20,
                      freetype_version=None, remove_text=False,
                      savefig_kwarg=None, style='_classic_test'):
     """
diff --git a/lib/matplotlib/tests/test_mathtext.py b/lib/matplotlib/tests/test_mathtext.py
index 39f5d7ef1..7681e5cb9 100644
--- a/lib/matplotlib/tests/test_mathtext.py
+++ b/lib/matplotlib/tests/test_mathtext.py
@@ -159,7 +159,8 @@ for fonts, chars in font_test_specs:
 
 def make_set(basename, fontset, tests, extensions=None):
     def make_test(filename, test):
-        @image_comparison(baseline_images=[filename], extensions=extensions)
+        @image_comparison(baseline_images=[filename], extensions=extensions,
+                          tol=40)
         def single_test():
             matplotlib.rcParams['mathtext.fontset'] = fontset
             fig = plt.figure(figsize=(5.25, 0.75))
