 setup.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/setup.py b/setup.py
index ed8fd5f..ab648a3 100644
--- a/setup.py
+++ b/setup.py
@@ -9,11 +9,6 @@ try:
     from setuptools.command.test import test as TestCommand
 
     class PyTest(TestCommand):
-        def finalize_options(self):
-            TestCommand.finalize_options(self)
-            self.test_args = []
-            self.test_suite = True
-
         def run_tests(self):
             # import here, because outside the eggs aren't loaded
             import pytest
