From 24722cc47cc9e968eb05becf6056517b23e81e34 Mon Sep 17 00:00:00 2001
From: Sambhav Kothari <sambhavs.email@gmail.com>
Date: Sun, 12 Aug 2018 18:06:58 +0100
Subject: [PATCH] PICARD-1310: Fix crash on clearing the log, due to wrong var.
 name

---
 picard/log.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/picard/log.py b/picard/log.py
index 5974a8a8c..a388af554 100644
--- a/picard/log.py
+++ b/picard/log.py
@@ -90,7 +90,7 @@ def contents(self, prev=-1):
         return contents
 
     def clear(self):
-        with self.log_queue_lock:
+        with self._queue_lock:
             self._log_queue.clear()
 
 
