Index: dcraw/parse.c
===================================================================
--- dcraw.orig/parse.c
+++ dcraw/parse.c
@@ -1213,7 +1213,7 @@ void parse_qt (int level, int end)
   }
 }
 
-char *memmem (char *haystack, size_t haystacklen,
+static void *my_memmem (char *haystack, size_t haystacklen,
               char *needle, size_t needlelen)
 {
   char *c;
@@ -1239,8 +1239,8 @@ void identify()
   tread (head, 1, 32, ifp);
   tseek (ifp, 0, SEEK_END);
   fsize = ftell(ifp);
-  if ((cp = memmem (head, 32, "MMMM", 4)) ||
-      (cp = memmem (head, 32, "IIII", 4))) {
+  if ((cp = my_memmem (head, 32, "MMMM", 4)) ||
+      (cp = my_memmem (head, 32, "IIII", 4))) {
     parse_phase_one (cp-head);
     if (cp-head) parse_tiff (0,0);
   } else if (order == 0x4949 || order == 0x4d4d) {
