--- qwtplot3d/doc/Doxyfile.doxygen.orig	2007-11-26 22:30:57.000000000 +0000
+++ qwtplot3d/doc/Doxyfile.doxygen	2007-11-26 22:30:03.000000000 +0000
@@ -116,7 +116,7 @@
 HTML_ALIGN_MEMBERS     = YES
 GENERATE_HTMLHELP      = NO
 CHM_FILE               = qwtplot3d.chm
-HHC_LOCATION           = "c:\Programme\HTML Help Workshop\hhc.exe"
+#HHC_LOCATION           = "c:\Programme\HTML Help Workshop\hhc.exe"
 GENERATE_CHI           = NO
 BINARY_TOC             = YES
 TOC_EXPAND             = YES
@@ -181,7 +181,7 @@
 MACRO_EXPANSION        = NO
 EXPAND_ONLY_PREDEF     = NO
 SEARCH_INCLUDES        = YES
-INCLUDE_PATH           = V:/cvs/qwtplot3d/include/
+INCLUDE_PATH           = ../include
 INCLUDE_FILE_PATTERNS  = *.h
 PREDEFINED             = QWT3D_NOT_FOR_DOXYGEN \
                          QT_VERSION=0x040000
@@ -212,7 +212,6 @@
 GRAPHICAL_HIERARCHY    = YES
 DIRECTORY_GRAPH        = YES
 DOT_IMAGE_FORMAT       = png
-DOT_PATH               = V:/graphviz/Graphviz/bin/
 DOTFILE_DIRS           = 
 MAX_DOT_GRAPH_WIDTH    = 974
 MAX_DOT_GRAPH_HEIGHT   = 10000
