diff -up scipy-0.12.0/scipy/linalg/flapack.pyf.src.gerqf scipy-0.12.0/scipy/linalg/flapack.pyf.src
--- scipy-0.12.0/scipy/linalg/flapack.pyf.src.gerqf	2013-04-06 10:10:35.000000000 -0600
+++ scipy-0.12.0/scipy/linalg/flapack.pyf.src	2013-04-15 17:03:37.290059000 -0600
@@ -650,7 +650,7 @@ interface
      <ftype> dimension(m,n),intent(in,out,copy,out=qr,aligned8) :: a
      <ftype> dimension(MIN(m,n)),intent(out) :: tau
 
-     integer optional,intent(in),depend(n),check(lwork>=m||lwork==-1) :: lwork=3*m
+     integer optional,intent(in),depend(m),check(lwork>=m||lwork==-1) :: lwork=3*m
      <ftype> dimension(MAX(lwork,1)),intent(out),depend(lwork) :: work
      integer intent(out) :: info
    end subroutine <prefix>gerqf
