--- a/pserror.c
+++ b/pserror.c
@@ -22,11 +22,11 @@ extern char *program ;	/* Defined by mai
 #define MAX_COLUMN	78	/* maximum column to print upto */
 
 void message(int flags, char *format, ...)
 {
   va_list args ;
-  static column = 0 ;		/* current screen column for message wrap */
+  static int column = 0 ;	/* current screen column for message wrap */
   char msgbuf[MAX_MESSAGE] ;	/* buffer in which to put the message */
   char *bufptr = msgbuf ;	/* message buffer pointer */
 
   if ( (flags & MESSAGE_NL) && column != 0 ) {	/* new line if not already */
     putc('\n', stderr) ;

