--- SDL2_mixer-2.0.2/music_smpeg-orig.c	2017-12-06 20:47:21.691153833 -0500
+++ SDL2_mixer-2.0.2/music_smpeg.c	2017-12-06 20:47:30.032153222 -0500
@@ -138,7 +138,7 @@
         SDL_free(music);
         return NULL;
     }
-    smpeg.SMPEG_actualSpec(mp3, &music_spec);
+    smpeg.SMPEG_actualSpec(music->mp3, &music_spec);
 
     music->freesrc = freesrc;
     return music;
--- SDL2_mixer-2.0.2/music_smpeg-orig.c	2017-12-06 21:03:44.907081811 -0500
+++ SDL2_mixer-2.0.2/music_smpeg.c	2017-12-06 23:34:36.696418752 -0500
@@ -174,6 +174,7 @@
     int left = (len - smpeg.SMPEG_playAudio(music->mp3, stream, len));
     if (left > 0) {
         stream += (len - left);
+    }
     return left;
 }
 
