diff -ur alliance-5.0-orig/nero/src/MDefs.h alliance-5.0/nero/src/MDefs.h
--- alliance-5.0-orig/nero/src/MDefs.h	2008-11-26 16:41:06.000000000 -0500
+++ alliance-5.0/nero/src/MDefs.h	2008-11-26 16:44:40.000000000 -0500
@@ -24,6 +24,8 @@
 # define  __MDefs__  1
 
 
+# include  <limits.h>
+# include  <cstring>
 # include  "UDefs.h"
 
 
diff -ur alliance-5.0-orig/ocp/src/placer/PPlacement.h alliance-5.0/ocp/src/placer/PPlacement.h
--- alliance-5.0-orig/ocp/src/placer/PPlacement.h	2008-11-26 16:41:06.000000000 -0500
+++ alliance-5.0/ocp/src/placer/PPlacement.h	2008-11-26 16:46:06.000000000 -0500
@@ -43,6 +43,7 @@
 #include <fstream>
 #include <vector>
 #include <map>
+#include <cstring>
 using namespace std;
 
 #include "mut.h"

