--- a/configure.in
+++ b/configure.in
@@ -469,11 +469,6 @@
     LOCALEDIR="/usr/lib/locale"
 fi
 
-dnl> Add /usr/local/ /opt/local
-CFLAGS="${CFLAGS} -I/usr/local/include -I/opt/local/include"
-CPPFLAGS="${CPPFLAGS} -I/usr/local/include -I/opt/local/include"
-LDFLAGS="${LDFLAGS} -L/usr/local/lib -L/opt/local/lib"
-
 PWD=`pwd`
 
 echo
@@ -1716,18 +1711,6 @@
 if test ".${ac_cv_lib_GeoIP_GeoIP_record_by_ipnum}" = ".yes"; then
 LIBS="$LIBS -lGeoIP"
 
-if test -f "GeoLiteCity.dat"; then
-   echo "GeoLiteCity.dat already present"
-else
-   if test -f "3rd_party/GeoLiteCity.dat.gz"; then
-     cp 3rd_party/GeoLiteCity.dat.gz .
-   else
-     wget http://www.maxmind.com/download/geoip/database/GeoLiteCity.dat.gz
-   fi
-
-   gunzip GeoLiteCity.dat.gz
-fi
-
 dnl> --------
 
 if test -f "GeoIPASNum.dat"; then
@@ -1746,8 +1729,11 @@
 exit 1
 fi
 
-cp "3rd_party/oui.txt.gz" .
-cp "3rd_party/specialMAC.txt.gz" .
+gunzip 3rd_party/GeoLiteCity.dat.gz &&
+mv "3rd_party/etter.finger.os.gz" . &&
+mv "3rd_party/oui.txt.gz" . &&
+mv "3rd_party/specialMAC.txt.gz" . ||
+    exit 1
 
 if test -d ./configureextra; then
 
@@ -1908,7 +1894,6 @@
 AC_SUBST(SO_VERSION_PATCH)
 AC_SUBST(RRD_LIB)
 AC_SUBST(RRD_INC)
-AC_SUBST(GEO_DIR)
 AC_SUBST(GETOPT_H)
 AC_SUBST(GETOPT_C)
 
--- a/globals-core.c
+++ b/globals-core.c
@@ -367,7 +367,7 @@
   _mtuSize[DLT_ATM_RFC1483] = 9180                             /* LLC/SNAP encapsulated atm */;
   _headerSize[DLT_ATM_RFC1483] = 0;
 
-  /* _mtuSize[DLT_RAW] = ?                                        raw IP */
+  _mtuSize[DLT_RAW] = 1500;                                     /*   raw IP */
   _headerSize[DLT_RAW] = 0;
 
   /* Others defined in bpf.h at tcpdump.org as of the resync - it would be NICE
@@ -461,19 +461,18 @@
   struct stat statbuf;
 
   /* Initialize GeoIP databases */
-  for(i=0; myGlobals.configFileDirs[i] != NULL; i++) {
+  {
     char path[256];
     
     safe_snprintf(__FILE__, __LINE__, path, sizeof(path),
 		  "%s%c%s",
-		  myGlobals.configFileDirs[i], 
+		  myGlobals.dbPath,
 		  CONST_PATH_SEP, GEO_IP_FILE);
     revertSlashIfWIN32(path, 0);
 
     if(stat(path, &statbuf) == 0) {
       if((myGlobals.geo_ip_db = GeoIP_open(path, GEOIP_CHECK_CACHE)) != NULL) {
 	traceEvent(CONST_TRACE_INFO, "GeoIP: loaded config file %s", path);
-	break;
       }
     }
   }
@@ -482,20 +481,18 @@
     traceEvent(CONST_TRACE_ERROR, "GeoIP: unable to load file %s", GEO_IP_FILE);
   
   /* *************************** */
-
-  for(i=0; myGlobals.configFileDirs[i] != NULL; i++) {
+  {
     char path[256];
     
     safe_snprintf(__FILE__, __LINE__, path, sizeof(path),
 		  "%s%c%s",
-		  myGlobals.configFileDirs[i], 
+		  myGlobals.dbPath,
 		  CONST_PATH_SEP, GEO_IP_ASN_FILE);
     revertSlashIfWIN32(path, 0);
 
     if(stat(path, &statbuf) == 0) {
       if((myGlobals.geo_ip_asn_db = GeoIP_open(path, GEOIP_CHECK_CACHE)) != NULL) {
 	traceEvent(CONST_TRACE_INFO, "GeoIP: loaded ASN config file %s", path);
-	break;
       }
     }
   }
--- a/Makefile.am
+++ b/Makefile.am
@@ -36,7 +36,6 @@
 
 #AM_CFLAGS =
 #AM_LDFLAGS= 
-ACLOCAL_AMFLAGS=-I m4
 
 DISTCLEANFILES = trivialcompilemessage *.db #* *~ *.log
 CLEANFILES     = $(DISTCLEANFILES) stamp-h1
@@ -67,9 +66,7 @@
 NTOPDATA = ntop-cert.pem \
            $(ETTER_PASSIVE) \
            oui.txt.gz \
-           specialMAC.txt.gz \
-	   GeoIPASNum.dat \
-	   GeoLiteCity.dat
+           specialMAC.txt.gz
 
 NTOPHTML = html html/*.js html/*.html  html/*.gif html/*.jpg html/*.ico html/*.png \
            html/*.css html/*.dtd \
@@ -130,7 +127,7 @@
 
 libntop_la_DEPENDENCIES =  config.h
 libntop_la_LIBADD       = $(BASE_LIBS)
-libntop_la_LDFLAGS      = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic @DYN_FLAGS@
+libntop_la_LDFLAGS      = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic -shared @DYN_FLAGS@
 
 # Archive for http representation, or the 'viewer'
 libntopreport_la_SOURCES = emitter.c       globals-report.c  graph.c \
@@ -142,7 +139,7 @@
 
 libntopreport_la_DEPENDENCIES = libntop.la
 libntopreport_la_LIBADD       = libntop.la $(BASE_LIBS)
-libntopreport_la_LDFLAGS      = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic @DYN_FLAGS@
+libntopreport_la_LDFLAGS      = $(AM_LDFLAGS) -release $(VERSION) -export-dynamic -shared @DYN_FLAGS@
 
 man_MANS = ntop.8
 
@@ -362,30 +359,7 @@
 
 # download the vendor information table
 dnvt:
-	@echo ""
-	@echo ""
-	@echo -n "Preparing "
-	@if test -f oui.txt; then \
-	  rm -rf oui.txt; \
-	fi
-	@if test -f oui.txt.gz.old; then \
-	  rm -rf oui.txt.gz.old; \
-	fi;
-	@mv oui.txt.gz oui.txt.gz.old
-	@echo "(old oui.txt.gz file is now oui.txt.gz.old)"
-	@echo ""
-	@wget -c http://standards.ieee.org/regauth/oui/oui.txt
-	@gzip oui.txt
-	@echo ""
-	@echo ""
-	@echo -n "Old file lines were: "
-	@gunzip -c oui.txt.gz.old | wc -l
-	@echo -n "New file lines are:  "
-	@gunzip -c oui.txt.gz | wc -l
-	@echo ""
-	@echo ""
-	@echo "New file is:"
-	@ls -l oui.txt.gz
+	@echo "Use included oui.txt.gz"
 
 # download the Novell SAP Protocol information table
 #dnsapt:
@@ -393,45 +367,7 @@
 
 # download the passive ethernet fingerprint database
 dnetter:
-	@echo ""
-	@echo ""
-	@echo "Preparing "
-	@echo ""
-
-	@if test -f $(ETTER_PASSIVE).old; then \
-	  echo "...Deleting prior file, $(ETTER_PASSIVE).old..."; \
-	  rm -rf $(ETTER_PASSIVE).old; \
-	fi;
-	@if test -f $(ETTER_PASSIVE); then \
-	  echo "...Moving current $(ETTER_PASSIVE) to $(ETTER_PASSIVE).old"; \
-	  mv $(ETTER_PASSIVE) $(ETTER_PASSIVE).old; \
-	fi;
-	@for file in $(ETTER_PASSIVE) $(ETTER_PASSIVE).gz \
-		$(ETTER_PASSIVE_FILE) $(ETTER_PASSIVE_FILE).gz; do \
-          if test -f $$file; then \
-	     echo "...Deleting prior file, $$file..."; \
-             rm -rf $$file; \
-          fi; \
-        done
-	@echo ""
-	@echo "...Downloading new file..."
-	@wget -O $(ETTER_PASSIVE_FILE) \
-		$(ETTER_PASSIVE_DOWNLOAD_FROM)/$(ETTER_PASSIVE_FILE)?$(ETTER_PASSIVE_DOWNLOAD_PARMS)
-	@echo ""
-	@echo "gziping downloaded file..."
-	@gzip $(ETTER_PASSIVE_FILE)
-	@echo ""
-	@echo ""
-	@if test -f $(ETTER_PASSIVE).old; then \
-	  echo -n "Old file lines were: "; \
-	  gunzip -c $(ETTER_PASSIVE).old | wc -l; \
-	fi;
-	@echo -n "New file lines are:  "
-	@gunzip -c $(ETTER_PASSIVE) | wc -l
-	@echo ""
-	@echo ""
-	@echo "New file is:"
-	@ls -l $(ETTER_PASSIVE)
+	@echo "Use included $(ETTER_PASSIVE)"
 
 # ntop census
 census-fail:
--- a/report.c
+++ b/report.c
@@ -2091,7 +2091,7 @@
   if(fetchPrefsValue("dot.path", buf, sizeof(buf)) != -1) {
     snprintf(dotPath, sizeof(dotPath), "%s", buf);
   } else {
-    snprintf(dotPath, sizeof(dotPath), "/usr/local/bin/dot");
+    snprintf(dotPath, sizeof(dotPath), "/usr/bin/dot");
     storePrefsValue("dot.path", dotPath); /* Set the default */
   }
 
