From 0713a497ad56e6b0ff939c277e87db46c7763091 Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentnl@gentoo.org>
Date: Fri, 8 May 2020 01:28:54 +1200
Subject: Include '.' in @INC on perl 5.26+

---
 Makefile.PL | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.PL b/Makefile.PL
index 7c16d62..00324cb 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -1,3 +1,4 @@
+use lib '.';
 use inc::Module::Install 1.06;
 
 if ( -e 'MANIFEST.SKIP' ) {
-- 
2.26.2

