--- wxcore-0.90.1.1-orig/src/haskell/Graphics/UI/WXCore/Events.hs	2014-03-23 08:01:17.000000000 +1100
+++ wxcore-0.90.1.1/src/haskell/Graphics/UI/WXCore/Events.hs	2014-03-23 17:05:26.428738461 +1100
@@ -1,4 +1,4 @@
-{-# LANGUAGE ForeignFunctionInterface #-}
+{-# LANGUAGE CPP, ForeignFunctionInterface #-}
 -----------------------------------------------------------------------------------------
 {-|	Module      :  Events
 	Copyright   :  (c) Daan Leijen 2003
@@ -248,6 +248,9 @@
 import Foreign.Marshal.Alloc
 import Foreign.Marshal.Array
 import Foreign.Marshal.Utils
+#if (__GLASGOW_HASKELL__>=705)
+import Foreign.C.Types(CInt(..))
+#endif
 
 import Data.Char ( chr ) -- used in stc
 import Data.Maybe ( fromMaybe, fromJust )
