--- a/Setup.hs
+++ b/Setup.hs
@@ -7,7 +7,7 @@ import Data.List (foldl', intersperse, intercalate, nub, lookup, isPrefixOf, isI
 import Data.Maybe (fromJust)
 import Distribution.PackageDescription hiding (includeDirs)
 import qualified Distribution.PackageDescription as PD (includeDirs)
-import Distribution.InstalledPackageInfo(installedPackageId, sourcePackageId, includeDirs)
+import Distribution.InstalledPackageInfo({-installedPackageId,-} sourcePackageId, includeDirs)
 import Distribution.Simple
 import Distribution.Simple.LocalBuildInfo (LocalBuildInfo, localPkgDescr, installedPkgs, withPrograms, buildDir)
 import Distribution.Simple.PackageIndex(SearchResult (..), searchByName, allPackages )
