--- Makefile~	2007-04-28 22:04:23.000000000 +0200
+++ Makefile	2007-04-28 22:21:46.000000000 +0200
@@ -91,8 +91,8 @@
   sed 's/-D/-DLCF_/g'`
 LILO=-DLILO=0xbb920890
 KVER=`if [ -f /usr/include/linux/version.h ]; then echo -DHAS_VERSION_H; fi`
-HDM=`if [ -f /usr/include/libdevmapper.h ]; then echo -DHAS_LIBDEVMAPPER_H; fi`
-DEVMAPPER=`if [ -f /usr/include/libdevmapper.h ]; then echo "-ldevmapper"; fi`
+HDM=
+DEVMAPPER=
 
 TMP=_TmP_
 
