--- mplayer-1.0~svn34750.orig/libmpcodecs/vd_ffmpeg.c
+++ mplayer-1.0~svn34750/libmpcodecs/vd_ffmpeg.c
@@ -47,6 +47,18 @@
 
 #include "libavcodec/avcodec.h"
 
+#ifndef AV_EF_COMPLIANT
+#define AV_EF_COMPLIANT 0
+#endif
+
+#ifndef AV_EF_CAREFUL
+#define AV_EF_CAREFUL 0
+#endif
+
+#ifndef AV_EF_AGGRESSIVE
+#define AV_EF_AGGRESSIVE 0
+#endif
+
 #if AVPALETTE_SIZE > 1024
 #error palette too large, adapt libmpcodecs/vf.c:vf_get_image
 #endif
