diff --git a/flute-1.3/org/w3c/flute/parser/Parser.java b/flute-1.3/org/w3c/flute/parser/Parser.java
index 3048834..55fec8b 100644
--- a/flute-1.3/org/w3c/flute/parser/Parser.java
+++ b/flute-1.3/org/w3c/flute/parser/Parser.java
@@ -3497,8 +3497,8 @@ Token language;
         jj_expentry[i] = jj_lasttokens[i];
       }
       boolean exists = false;
-      for (java.util.Enumeration enum = jj_expentries.elements(); enum.hasMoreElements();) {
-        int[] oldentry = (int[])(enum.nextElement());
+      for (java.util.Enumeration enumeration = jj_expentries.elements(); enumeration.hasMoreElements();) {
+        int[] oldentry = (int[])(enumeration.nextElement());
         if (oldentry.length == jj_expentry.length) {
           exists = true;
           for (int i = 0; i < jj_expentry.length; i++) {
