--- a/_mssql.pyx
+++ b/_mssql.pyx
@@ -34,6 +34,10 @@ DEF EXCOMM = 9
 # Provide constants missing in FreeTDS 0.82 so that we can build against it
 DEF DBVERSION_71 = 5
 DEF DBVERSION_72 = 6
+# See /usr/include/sybdb.h - DBVERSION_80 is not defined in FreeTDS 1.00.13
+# (cf. http://www.freetds.org/userguide/choosingtdsprotocol.htm)
+# 8.0 was actually 7.1 so set it to the same value
+DEF DBVERSION_80 = 5
 
 ROW_FORMAT_TUPLE = 1
 ROW_FORMAT_DICT = 2
