 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 67cf344..c15f785 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -272,7 +272,7 @@ endif (NOT DATADIR)
 
 set (OPENVAS_STATE_DIR "${LOCALSTATEDIR}/lib/openvas")
 set (OPENVAS_LOG_DIR   "${LOCALSTATEDIR}/log/openvas")
-set (GSAD_PID_DIR   "${LOCALSTATEDIR}/run")
+set (GSAD_PID_DIR   "/run")
 set (GSA_DATA_DIR "${DATADIR}/openvas/gsa")
 
 # TODO: Eventually use own certificates
