http://bugs.gentoo.org/314423

--- configure.ac
+++ configure.ac
@@ -39,9 +39,6 @@
 AM_CONDITIONAL(ENABLE_ICON_FRAMING, test x$enable_framing = xyes)
 
 if test "x$enable_framing" = "xyes"; then
-   PKG_CHECK_EXISTS([ImageMagick >= 5.5.7],
-                    [enable_framing=yes], [enable_framing=no])
-   if test "x$enable_framing" = "xyes"; then
       AC_PATH_PROG(CONVERT, convert)
       if test -z "$CONVERT"; then
       	 AC_MSG_ERROR([ImageMagick is required to build tango-icon-theme])
@@ -50,9 +47,6 @@
       # Add the command line options we need here
       CONVERT="$CONVERT -bordercolor Transparent -border 1x1"
       AC_SUBST(CONVERT)
-   else
-      AC_MSG_ERROR([ImageMagick is required to build tango-icon-theme])
-   fi
 fi
 
 # Check for rsvg or ksvgtopng for creating large scale bitmaps from svg
