diff --git a/libopeniscsiusr/Makefile b/libopeniscsiusr/Makefile
index 52e453a..523518b 100644
--- a/libopeniscsiusr/Makefile
+++ b/libopeniscsiusr/Makefile
@@ -51,7 +51,7 @@ EXTRA_MAN_FILES = libopeniscsiusr.h.3
 OBJS = context.o misc.o session.o sysfs.o iface.o idbm.o node.o default.o
 
 CFLAGS ?= -O2 -g
-CFLAGS += -Wall -Werror -Wextra -fvisibility=hidden -fPIC
+CFLAGS += -Wall  -Wextra -fvisibility=hidden -fPIC
 CFLAGS += $(shell $(PKG_CONFIG) --cflags libkmod)
 CFLAGS += -DSBINDIR=\"$(SBINDIR)\"
 CFLAGS += -DISCSI_DB_ROOT=\"$(DBROOT)\"
diff --git a/usr/Makefile b/usr/Makefile
index 3fc2248..c3580a3 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -44,7 +44,7 @@ HOMEDIR ?= $(etcdir)/iscsi
 PKG_CONFIG ?= /usr/bin/pkg-config
 
 CFLAGS ?= -O2 -g
-WARNFLAGS ?= -Wall -Wextra -Werror -Wstrict-prototypes -fno-common
+WARNFLAGS ?= -Wall -Wextra  -Wstrict-prototypes -fno-common
 CFLAGS += $(WARNFLAGS) -I../include -I. -D_GNU_SOURCE \
 	  -I$(TOPDIR)/libopeniscsiusr
 CFLAGS += $(shell $(PKG_CONFIG) --cflags libkmod)
