--- a/4stattack.py
+++ b/4stattack.py
@@ -28,6 +28,9 @@
 images	= None
 screen	= None
 
+f = os.path.expanduser('~/.4stattackrc')
+if not os.path.exists(f):
+    os.system('cp settings.ini %s' % f)
 
 def playMusic(datadir):
 	if pygame.mixer.get_init():
@@ -78,7 +81,7 @@
 	if options.has_key('-ini'):
 		ini_settings = IniSettings(options['-ini'])
 	else:
-		ini_settings = IniSettings('settings.ini')
+		ini_settings = IniSettings(os.path.expanduser('~/.4stattackrc'))
 	
 	settings = ini_settings.settings
 	
