--- a/src/e3/os/platform.py	2021-11-23 08:50:05.077801145 +0100
+++ b/src/e3/os/platform.py	2021-11-23 08:50:40.646227489 +0100
@@ -85,12 +85,12 @@
 
         # Fetch linux distribution info on linux OS
         if cls.uname.system == "Linux":  # linux-only
-            import ld
+            import distro
 
             cls.ld_info = {
-                "name": ld.name(),
-                "major_version": ld.major_version(),
-                "version": ld.version(),
+                "name": distro.name(),
+                "major_version": distro.major_version(),
+                "version": distro.version(),
             }
 
         # Fetch network interfaces
--- a/setup.py	2021-11-23 20:36:57.056478696 +0100
+++ b/setup.py	2021-11-23 20:36:03.371303043 +0100
@@ -18,7 +18,7 @@
     platform_string = ":sys_platform=='%s'" % p
     extras_require[platform_string] = ["psutil"]
     if p in ("linux", "linux2"):
-        extras_require[platform_string].append("ld")
+        extras_require[platform_string].append("distro")
 
 # Get e3 version from the VERSION file.
 version_file = os.path.join(os.path.dirname(__file__), "VERSION")
