--- a/yatexhlp.el
+++ b/yatexhlp.el
@@ -9,12 +9,7 @@
 			 (cond (YaTeX-japan "jp")
 			       (t "eng"))))
       (help-dir
-       (cond
-	((and (boundp 'site-directory) site-directory) site-directory)
-	((string-match "\\.app/" doc-directory)	;For Emacs.app(Darwin)
-	 (expand-file-name "../site-lisp" doc-directory))
-	(YaTeX-emacs-19 (expand-file-name "../../site-lisp" doc-directory))
-	(t exec-directory))))
+       "/usr/share/emacs/etc/yatex"))
   (defvar YaTeX-help-file
     (expand-file-name help-file help-dir)
     "*Help file of LaTeX/TeX commands or macros.")
