--- tic98/ppmd_hash.c.orig	2005-11-03 11:09:22.000000000 +0100
+++ tic98/ppmd_hash.c	2005-11-03 11:09:49.000000000 +0100
@@ -2,7 +2,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
+#ifndef __APPLE__
 #include <malloc.h>
+#endif
 #include "ppmd_hash.h"
 
 #define HASH_NULL 0
--- tic98/ppmd_model.c.orig	2005-11-03 11:07:09.000000000 +0100
+++ tic98/ppmd_model.c	2005-11-03 11:09:04.000000000 +0100
@@ -1,7 +1,11 @@
 /* PPM* model for arithmetic encoder. */
 #include <stdio.h>
 #include <assert.h>
+#ifndef __APPLE__
 #include <malloc.h>
+#else
+#include <stdlib.h>
+#endif
 #include "arithcode.h"
 #include "ppmd_hash.h"
 
