From 84d5affecac79ea1a8641f7fecc9adf7691f4acd Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentnl@gentoo.org>
Date: Thu, 9 Jul 2020 17:46:23 +1200
Subject: Include '.' in @INC on perl 5.26+

---
 Makefile.PL | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.PL b/Makefile.PL
index 225f785..022d09d 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -1,3 +1,4 @@
+use lib '.';
 use inc::Module::Install;
 
 name 'DBICx-TestDatabase';
-- 
2.27.0

