--- a/aksetup_helper.py
+++ b/aksetup_helper.py
@@ -43,11 +43,8 @@
 
 
 def get_numpy_incpath():
-    from imp import find_module
-    # avoid actually importing numpy, it screws up distutils
-    file, pathname, descr = find_module("numpy")
-    from os.path import join
-    return join(pathname, "core", "include")
+    import numpy as np
+    return np.get_include()
 
 
 class NumpyExtension(Extension):
