diff -Naur NAMD_2.10_Source.orig/arch/Linux-x86_64.fftw NAMD_2.10_Source/arch/Linux-x86_64.fftw
--- NAMD_2.10_Source.orig/arch/Linux-x86_64.fftw	2015-03-06 17:53:44.170416075 -0700
+++ NAMD_2.10_Source/arch/Linux-x86_64.fftw	2015-03-06 17:54:46.641162629 -0700
@@ -1,7 +1,7 @@
 
-FFTDIR=/Projects/namd2/fftw/linux-x86_64
+FFTDIR=/usr
 FFTINCL=-I$(FFTDIR)/include
-FFTLIB=-L$(FFTDIR)/lib -lsrfftw -lsfftw
+FFTLIB=-L$(FFTDIR)/gentoo-libdir -lsrfftw -lsfftw
 FFTFLAGS=-DNAMD_FFTW
 FFT=$(FFTINCL) $(FFTFLAGS)
 
diff -Naur NAMD_2.10_Source.orig/arch/Linux-x86_64.tcl NAMD_2.10_Source/arch/Linux-x86_64.tcl
--- NAMD_2.10_Source.orig/arch/Linux-x86_64.tcl	2015-03-06 17:53:44.170416075 -0700
+++ NAMD_2.10_Source/arch/Linux-x86_64.tcl	2015-03-06 17:54:46.641162629 -0700
@@ -1,9 +1,9 @@
 
 #TCLDIR=/Projects/namd2/tcl/tcl8.5.9-linux-x86_64
-TCLDIR=/Projects/namd2/tcl/tcl8.5.9-linux-x86_64-threaded
+TCLDIR=/usr
 TCLINCL=-I$(TCLDIR)/include
 #TCLLIB=-L$(TCLDIR)/lib -ltcl8.5 -ldl
-TCLLIB=-L$(TCLDIR)/lib -ltcl8.5 -ldl -lpthread
+TCLLIB=-L$(TCLDIR)/gentoo-libdir -ltcl8.5 -ldl -lpthread
 TCLFLAGS=-DNAMD_TCL
 TCL=$(TCLINCL) $(TCLFLAGS)
 
diff -Naur NAMD_2.10_Source.orig/Makefile NAMD_2.10_Source/Makefile
--- NAMD_2.10_Source.orig/Makefile	2015-03-06 17:53:44.170416075 -0700
+++ NAMD_2.10_Source/Makefile	2015-03-06 18:01:32.645966590 -0700
@@ -389,10 +389,10 @@
 	$(DSTDIR)/stringhash.o
 
 # definitions for Charm routines
-CHARMC = $(CHARM)/bin/charmc
-CHARMXI = $(CHARM)/bin/charmc
-CHARMINC = $(CHARM)/include $(COPTD)CMK_OPTIMIZE=1
-CHARMLIB = $(CHARM)/lib
+CHARMC = /usr/bin/charmc
+CHARMXI = /usr/bin/charmc
+CHARMINC = /usr/include/gentoo-charm $(COPTD)CMK_OPTIMIZE=1
+CHARMLIB = /usr/gentoo-libdir
 CHARM_MODULES = -module NeighborLB -module HybridLB -module RefineLB -module GreedyLB -module CkMulticast
 #CHARM_MODULES = -module NeighborLB -module HybridLB -module RefineLB -module GreedyLB
 #CHARM_MODULES = -module msa -module NeighborLB -module HybridLB -module RefineLB -module GreedyLB
@@ -418,7 +418,7 @@
 
 # Add new executables here.
 
-BINARIES = namd2 psfgen sortreplicas flipdcd flipbinpdb charmrun
+BINARIES = namd2 psfgen sortreplicas flipdcd flipbinpdb
 
 # This should be rebuilt at every compile, but not on Win32.
 BUILDINFO = $(DSTDIR)/buildinfo
@@ -434,7 +434,7 @@
 
 namd2:	$(MKINCDIR) $(MKDSTDIR) $(OBJS) $(LIBS)
 	$(MAKEBUILDINFO)
-	$(CHARMC) -verbose -ld++-option \
+	$(CHARMC) $(LDFLAGS) -verbose -ld++-option \
 	'$(COPTI)$(CHARMINC) $(COPTI)$(INCDIR) $(COPTI)$(SRCDIR) $(CXXOPTS) $(CXXMICOPTS)' \
 	"$(CHARM_MODULES)" -language charm++ \
 	$(BUILDINFO).o \
@@ -484,11 +484,8 @@
 charmd_faceless.exe:
 	$(COPY) $(CHARM)/bin/charmd_faceless.exe charmd_faceless.exe
 
-charmrun.exe:
-	$(COPY) $(CHARM)/bin/charmrun.exe charmrun.exe
-
 psfgen:	$(MKDSTDIR) $(SBOBJS) $(PLUGINOBJS)
-	$(CC) $(SBCFLAGS) -o psfgen $(SBOBJS) $(PLUGINOBJS) $(TCLLIB) $(TCLAPPLIB) -lm
+	$(CC) $(SBCFLAGS) $(LDFLAGS) -o psfgen $(SBOBJS) $(PLUGINOBJS) $(TCLLIB) $(TCLAPPLIB) -lm
 
 psfgen.exe:	$(MKDSTDIR) $(SBOBJS) $(PLUGINOBJS) $(TCLDLL)
 	$(CC) $(SBCFLAGS) -o psfgen $(SBOBJS) $(PLUGINOBJS) $(TCLLIB) $(TCLAPPLIB) -lm
@@ -506,7 +503,7 @@
 	$(CC) $(CFLAGS) -o diffbinpdb $(SRCDIR)/diffbinpdb.c -lm
 
 flipdcd:	$(SRCDIR)/flipdcd.c
-	$(CC) $(CFLAGS) -o $@ $(SRCDIR)/flipdcd.c || \
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(SRCDIR)/flipdcd.c || \
 	echo "#!/bin/sh\necho unavailable on this platform" > $@; \
 	chmod +x $@
 
@@ -516,7 +513,7 @@
 	chmod +x $@
 
 fixdcd:	$(SRCDIR)/fixdcd.c
-	$(CC) $(CFLAGS) -o fixdcd $(SRCDIR)/fixdcd.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o fixdcd $(SRCDIR)/fixdcd.c
 
 dumpdcd:	$(SRCDIR)/dumpdcd.c
 	$(CC) $(CFLAGS) -o dumpdcd $(SRCDIR)/dumpdcd.c
@@ -711,9 +708,7 @@
 
 DOC_FILES = README.txt announce.txt license.txt notes.txt
 
-RELEASE_FILES = $(LIBCUDARTSO) flipdcd flipbinpdb sortreplicas psfgen charmrun namd2
-
-WINDOWS_RELEASE_FILES = $(LIBCUDARTSO) $(WINDOWSBINARIES) $(TCLDLL)
+RELEASE_FILES = $(LIBCUDARTSO) flipdcd flipbinpdb sortreplicas psfgen namd2
 
 release: all
 	$(ECHO) Creating release $(RELEASE_DIR_NAME)
@@ -724,17 +719,9 @@
 	for f in `find $(RELEASE_DIR_NAME)/lib -name CVS`; do \
 	  /bin/rm -rf $$f; \
 	done
-	if [ -r $(CHARM)/bin/charmd ]; then \
-	  $(COPY) $(CHARM)/bin/charmd $(RELEASE_DIR_NAME); \
-	fi
-	if [ -r $(CHARM)/bin/charmd_faceless ]; then \
-	  $(COPY) $(CHARM)/bin/charmd_faceless $(RELEASE_DIR_NAME); \
-	fi
 	chmod -R a+rX $(RELEASE_DIR_NAME)
 	tar cf $(RELEASE_DIR_NAME).tar $(RELEASE_DIR_NAME)
 	gzip $(RELEASE_DIR_NAME).tar
-	echo $(CHARM)
-	ls -l $(CHARM)/lib
 	-for f in $(RELEASE_FILES); do echo $$f; $(LDD) $(RELEASE_DIR_NAME)/$$f; done
 
 winrelease: winall
