--- a/src/core/player/phonon/engine_phonon.h	2017-11-04 17:06:43.000000000 +0100
+++ b/src/core/player/phonon/engine_phonon.h	2018-10-04 18:23:29.079923850 +0200
@@ -27,12 +27,12 @@
 #include <QPixmap>
 
 // Phonon
-#include <phonon/mediasource.h>
-#include <phonon/mediaobject.h>
-#include <phonon/audiooutput.h>
-#include <phonon/path.h>
-#include <phonon/effect.h>
-#include <phonon/volumefadereffect.h>
+#include <phonon4qt5/phonon/mediasource.h>
+#include <phonon4qt5/phonon/mediaobject.h>
+#include <phonon4qt5/phonon/audiooutput.h>
+#include <phonon4qt5/phonon/path.h>
+#include <phonon4qt5/phonon/effect.h>
+#include <phonon4qt5/phonon/volumefadereffect.h>
 
 #include "core/player/engine_base.h"
 #include "core/mediaitem/mediaitem.h"
--- a/src/core/player/phonon/engine_phonon.cpp	2017-11-11 07:29:42.000000000 +0100
+++ b/src/core/player/phonon/engine_phonon.cpp	2018-10-04 18:23:29.084923716 +0200
@@ -29,9 +29,9 @@
 #include <QtPlugin>
 
 // phonon
-#include <phonon/backendcapabilities.h>
-#include <phonon/effect.h>
-#include <phonon/effectparameter.h>
+#include <phonon4qt5/phonon/backendcapabilities.h>
+#include <phonon4qt5/phonon/effect.h>
+#include <phonon4qt5/phonon/effectparameter.h>
 
 #include <cmath>
 
