--- sxhkd-0.5.3/Makefile
+++ sxhkd-0.5.3/Makefile
@@ -5,7 +5,6 @@
 LIBS     = -lm -lxcb -lxcb-keysyms
 CFLAGS  += -std=c99 -pedantic -Wall -Wextra -I$(PREFIX)/include
 CFLAGS  += -D_POSIX_C_SOURCE=200112L -DVERSION=\"$(VERSION)\"
-LDFLAGS += -L$(PREFIX)/lib
 
 PREFIX    ?= /usr/local
 BINPREFIX  = $(PREFIX)/bin
@@ -14,11 +13,9 @@
 SRC = $(wildcard *.c)
 OBJ = $(SRC:.c=.o)
 
-all: CFLAGS  += -Os
-all: LDFLAGS += -s
 all: $(NAME)
 
-debug: CFLAGS += -O0 -g -DDEBUG
+debug: CFLAGS += -DDEBUG
 debug: $(NAME)
 
 include Sourcedeps
