diff --git a/plugins/actor/G-Force/Common/UI/LineXX.cpp b/plugins/actor/G-Force/Common/UI/LineXX.cpp
index 304eec8..5b18503 100644
--- a/plugins/actor/G-Force/Common/UI/LineXX.cpp
+++ b/plugins/actor/G-Force/Common/UI/LineXX.cpp
@@ -74,13 +74,13 @@
 
 		
 	#if CLR_INTERP && P_SZ != 1
-	int32_t len = sqrt( dx * dx + dy * dy ) + 1;
+	int32_t len = sqrt( (long double) dx * dx + dy * dy ) + 1;
 	dR /= len;
 	dG /= len;
 	dB /= len;
 	color = __Clr( R, G, B );
 	#elif CLR_INTERP && P_SZ == 1
-	int32_t len = sqrt( dx * dx + dy * dy ) + 1;
+	int32_t len = sqrt( (long double) dx * dx + dy * dy ) + 1;
 	dR /= len;
 	color = __Clr( R, G, B );
 	#endif
@@ -154,7 +154,7 @@
 		
 			for ( j = 0; j < tw; j++ ) {
 				int32_t tmp = j - halfW;
-				c_x = halfW - ( ( int32_t ) sqrt( halfW * halfW - tmp * tmp ) );
+				c_x = halfW - ( ( int32_t ) sqrt( (long double) halfW * halfW - tmp * tmp ) );
 				center = basePtr + (j-halfW) * mBytesPerRow;
 				for ( int k = c_x; k < tw - c_x; k++ ){
 					((PIXTYPE*) center)[k-halfW] = color;
diff --git a/plugins/morph/flash/morph_flash.c b/plugins/morph/flash/morph_flash.c
index fb5308a..c57f490 100644
--- a/plugins/morph/flash/morph_flash.c
+++ b/plugins/morph/flash/morph_flash.c
@@ -116,7 +116,7 @@ int lv_morph_flash_palette (VisPluginData *plugin, float rate, VisAudio *audio,
 	FlashPrivate *priv = visual_object_get_private (VISUAL_OBJECT (plugin));
 
 	if (src1->pal == NULL || src2->pal == NULL)
-		return;
+		return -1;
 
 	if (rate < 0.5)
 		visual_palette_blend (pal, src1->pal, &priv->whitepal, rate * 2);
