Bug: https://bugs.gentoo.org/911481
Upstream: https://github.com/capstone-engine/capstone/pull/2114
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@ project(capstone
 if (MSVC)
     add_compile_options(/W1 /w14189)
 else()
-    add_compile_options(-Werror -Wformat -Wmissing-braces -Wunused-function -Warray-bounds -Wunused-variable -Wparentheses -Wint-in-bool-context)
+    add_compile_options(-Wformat -Wmissing-braces -Wunused-function -Warray-bounds -Wunused-variable -Wparentheses -Wint-in-bool-context)
 endif()
 
 
