--- a/CMakeLists.txt	2013-10-18 18:04:24.000000000 +0200
+++ b/CMakeLists.txt	2015-12-19 14:20:50.523356947 +0100
@@ -3,9 +3,14 @@
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules )
 
 # search packages used by KDE
-find_package(KDE4 REQUIRED)
-include (KDE4Defaults)
-include (MacroLibrary)
+cmake_minimum_required(VERSION 2.8.12)
+
+find_package(ECM 0.0.9 REQUIRED NO_MODULE)
+set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
+
+include(KDEInstallDirs)
+include(KDECMakeSettings)
+include(ECMOptionalAddSubdirectory)
 
 add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 add_definitions( -DQT_NO_CAST_FROM_ASCII )
@@ -14,18 +19,17 @@
 include_directories (${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES})
 
 
-macro_optional_add_subdirectory(ColorSchemes)
-macro_optional_add_subdirectory(IconThemes)
-macro_optional_add_subdirectory(emoticons)
-macro_optional_add_subdirectory(kscreensaver)
+ecm_optional_add_subdirectory(ColorSchemes)
+ecm_optional_add_subdirectory(IconThemes)
+ecm_optional_add_subdirectory(emoticons)
+ecm_optional_add_subdirectory(kscreensaver)
 if (Q_WS_X11)
-   macro_optional_add_subdirectory(kwin-styles)
+   ecm_optional_add_subdirectory(kwin-styles)
 endif (Q_WS_X11)
-macro_optional_add_subdirectory(styles)
-macro_optional_add_subdirectory(wallpapers)
-macro_optional_add_subdirectory(HighResolutionWallpapers)
-macro_optional_add_subdirectory(WeatherWallpapers)
-macro_optional_add_subdirectory(desktopthemes)
+ecm_optional_add_subdirectory(styles)
+ecm_optional_add_subdirectory(wallpapers)
+ecm_optional_add_subdirectory(HighResolutionWallpapers)
+ecm_optional_add_subdirectory(WeatherWallpapers)
+ecm_optional_add_subdirectory(desktopthemes)
 
-macro_display_feature_log()
 
