--- doc/Makefile.am
+++ doc/Makefile.am
@@ -29,11 +29,10 @@
 
 if BUILD_DOC
 man_MANS		+= hb_report.8 ha_logd.8 ha_logger.1 stonith.8 meatclient.8
-endif
 
 EXTRA_DIST		= $(man_MANS)
 
-STYLESHEET_PREFIX 	?= http://docbook.sourceforge.net/release/xsl/current
+STYLESHEET_PREFIX 	?= /usr/share/sgml/docbook/xsl-stylesheets/
 MANPAGES_STYLESHEET 	?= $(STYLESHEET_PREFIX)/manpages/docbook.xsl
 HTML_STYLESHEET 	?= $(STYLESHEET_PREFIX)/xhtml/docbook.xsl
 FO_STYLESHEET 		?= $(STYLESHEET_PREFIX)/fo/docbook.xsl
@@ -47,3 +46,4 @@
 	$(XSLTPROC) \
 	$(XSLTPROC_MANPAGES_OPTIONS) \
 	$(MANPAGES_STYLESHEET) $<
+endif
--- configure.ac
+++ configure.ac
@@ -492,11 +492,13 @@
 AC_PATH_PROGS(PKGCONFIG, pkg-config)
 AC_PATH_PROGS(XML2CONFIG, xml2-config)
 
-AC_PATH_PROGS(XSLTPROC, xsltproc)
-AM_CONDITIONAL(BUILD_DOC, test "x$XSLTPROC" != "x" )
-if test "x$XSLTPROC" = "x"; then
-   AC_MSG_WARN([xsltproc not installed, unable to (re-)build manual pages])
+AC_ARG_ENABLE([doc],
+ [  --enable-doc 	build documentation, [default=no]],
+ [enable_doc=$withval], [enable_doc=no])
+if test "x$enable_doc" != "xno"; then
+	AC_PATH_PROGS(XSLTPROC, xsltproc)
 fi
+AM_CONDITIONAL(BUILD_DOC, test "x$enable_doc" != "xno" )
 
 AC_PATH_PROGS(VALGRIND_BIN, valgrind, /usr/bin/valgrind)
 AC_DEFINE_UNQUOTED(VALGRIND_BIN, "$VALGRIND_BIN", Valgrind command)
