--- libs/mpi/build/Jamfile.v2
+++ libs/mpi/build/Jamfile.v2
@@ -30,7 +30,7 @@
 python-id = [ option.get "python-buildid" ] ;
 if $(python-id)
 {
-    PYTHON_ID = [ regex.replace $(python-id) [*\\/:.\"\'] _ ] ;
+    PYTHON_ID = [ regex.replace $(python-id) [*\\/:\"\'] _ ] ;
 }
 
 rule tag ( name : type ? : property-set )
--- libs/python/build/Jamfile.v2
+++ libs/python/build/Jamfile.v2
@@ -36,7 +36,7 @@
 python-id = [ option.get "python-buildid" ] ;
 if $(python-id)
 {
-    PYTHON_ID = [ regex.replace $(python-id) [*\\/:.\"\'] _ ] ;
+    PYTHON_ID = [ regex.replace $(python-id) [*\\/:\"\'] _ ] ;
 }
 
 
