diff --git a/src/pybind/CMakeLists.txt b/src/pybind/CMakeLists.txt
index 664543172a..e74d73c084 100644
--- a/src/pybind/CMakeLists.txt
+++ b/src/pybind/CMakeLists.txt
@@ -62,12 +62,6 @@ foreach(python_version ${py_vers})
 endforeach()
 
 if(WITH_MGR)
-  if(NOT WITH_PYTHON2 AND MGR_PYTHON_VERSION_MAJOR EQUAL 2)
-    message(FATAL_ERROR "mgr plugins require python2 binding")
-  elseif(NOT WITH_PYTHON3 AND MGR_PYTHON_VERSION_MAJOR EQUAL 3)
-    message(FATAL_ERROR "mgr plugins require python3 binding")
-  endif()
-
   if(USE_OPENSSL)
     execute_process(
       COMMAND ${PYTHON${PYTHON_VERSION}_EXECUTABLE} -c "import ssl; print('.'.join(map(str,ssl.OPENSSL_VERSION_INFO[0:3])))"
