diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ed4bbc75d44..5f5e72918d2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -451,9 +451,7 @@ if(WITH_LIBRADOSSTRIPER)
   add_subdirectory(libradosstriper)
 endif()
 
-if(WITH_MGR)
-  add_subdirectory(mgr)
-endif()
+add_subdirectory(mgr)
 
 set(librados_config_srcs
   librados-config.cc)
diff --git a/src/mgr/CMakeLists.txt b/src/mgr/CMakeLists.txt
index e7caaeff099..b4f13c05184 100644
--- a/src/mgr/CMakeLists.txt
+++ b/src/mgr/CMakeLists.txt
@@ -1,6 +1,7 @@
 add_library(mgr_cap_obj OBJECT
   MgrCap.cc)
 
+if(WITH_MGR)
 set(mgr_srcs
   ${CMAKE_SOURCE_DIR}/src/ceph_mgr.cc
   ${CMAKE_SOURCE_DIR}/src/mon/PGMap.cc
@@ -36,3 +37,4 @@ target_link_libraries(ceph-mgr
 set_target_properties(ceph-mgr PROPERTIES
   POSITION_INDEPENDENT_CODE ${EXE_LINKER_USE_PIE})
 install(TARGETS ceph-mgr DESTINATION bin)
+endif()
