--- ceph-14.2.0.orig/src/common/CMakeLists.txt	2019-03-18 04:08:29.000000000 -0600
+++ ceph-14.2.0/src/common/CMakeLists.txt	2019-03-21 18:42:32.903182824 -0600
@@ -165,7 +165,7 @@
     crc32c_aarch64.c)
 endif(HAVE_INTEL)
 
-add_library(crc32 ${crc32_srcs})
+add_library(crc32 STATIC ${crc32_srcs})
 if(HAVE_ARMV8_CRC)
   set_target_properties(crc32 PROPERTIES
     COMPILE_FLAGS "${CMAKE_C_FLAGS} ${ARMV8_CRC_COMPILE_FLAGS}")
