From 77bdc407d0760a529e40914d16248dce01fd4778 Mon Sep 17 00:00:00 2001
From: O01eg <o01eg@yandex.ru>
Date: Wed, 8 Nov 2017 16:07:57 +0300
Subject: [PATCH] Fix #1843

---
 CMakeLists.txt    | 2 +-
 GG/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/CMakeLists.txt	2018-06-03 11:36:04.631832409 -0400
+++ b/CMakeLists.txt	2018-06-03 11:36:20.678821846 -0400
@@ -484,7 +484,7 @@
         ${Boost_INCLUDE_DIRS}
         ${OPENGL_INCLUDE_DIR}
         ${GLEW_INCLUDE_DIRS}
-        ${SDL_INCLUDE_DIRS}
+        ${SDL_INCLUDE_DIR}
         ${OPENAL_INCLUDE_DIR}
         ${OGG_INCLUDE_DIRS}
         ${VORBIS_INCLUDE_DIRS}
diff --git a/GG/CMakeLists.txt b/GG/CMakeLists.txt
index 5d3c157362..c5bf3825d8 100644
--- a/GG/CMakeLists.txt
+++ b/GG/CMakeLists.txt
@@ -184,7 +184,7 @@ target_include_directories(GiGiSDL SYSTEM
         ${Boost_INCLUDE_DIRS}
         ${OPENGL_INCLUDE_DIR}
         ${GLEW_INCLUDE_DIRS}
-        ${SDL_INCLUDE_DIRS}
+        ${SDL_INCLUDE_DIR}
 )
 
 target_link_libraries(GiGiSDL
