--- a/dmtcp/src/Makefile.am
+++ b/dmtcp/src/Makefile.am
@@ -66,7 +66,7 @@
 
 dmtcp_inspector_SOURCES = dmtcp_inspector.cpp
 
-dmtcphijack_so_LDFLAGS = -shared -avoid-version -module
+dmtcphijack_so_LDFLAGS = -shared -module
 
 #dmtcp_nocheckpoint_LDFLAGS = -static
 
--- a/dmtcp/src/dmtcpalloc.h
+++ b/dmtcp/src/dmtcpalloc.h
@@ -33,6 +33,7 @@
 #include <iostream>
 #include <sstream>
 #include <fstream>
+#include <cstddef>
 
 #define DMTCPSTRING    dmtcp::string
 #define DMTCPVECTOR(T) dmtcp::vector<T>
