--- pysize/main.py	2007-03-11 16:09:53.000000000 +0100
+++ pysize/main.py.new	2009-01-02 00:08:43.384827054 +0100
@@ -69,17 +69,8 @@
     stats.print_stats(40)
     os.remove(prof_file)
 
-def _try_psyco():
-    try:
-        # Try to use psyco if available
-        import psyco
-        psyco.full()
-    except ImportError:
-        pass
-
 def main():
     install_sigquit_traceback()
-    _try_psyco()
     locale.setlocale(locale.LC_ALL, '')
     usage = '%s [OPTIONS] [DIRECTORIES...]' % (sys.argv[0])
     parser = optparse.OptionParser(usage=usage, version='pysize ' + VERSION)
