diff -ur gcc-3.4.3/libffi/Makefile.in gcc-3.4.3-ffi-fixes/libffi/Makefile.in
--- gcc-3.4.3/libffi/Makefile.in	2003-11-22 08:41:32.000000000 -0500
+++ gcc-3.4.3-ffi-fixes/libffi/Makefile.in	2004-11-28 17:31:50.000000000 -0500
@@ -89,7 +89,7 @@
 libffi_basedir = @libffi_basedir@
 tool_include_dir = @tool_include_dir@
 toolexecdir = @toolexecdir@
-toolexeclibdir = @toolexeclibdir@
+toolexeclibdir = @toolexeclibdir@/@gcc_version@
 
 AUTOMAKE_OPTIONS = cygnus
 
diff -ur gcc-3.4.3/libffi/configure gcc-3.4.3-ffi-fixes/libffi/configure
--- gcc-3.4.3/libffi/configure	2004-05-18 05:08:39.000000000 -0400
+++ gcc-3.4.3-ffi-fixes/libffi/configure	2004-11-28 17:48:19.000000000 -0500
@@ -3800,8 +3800,8 @@
   toolexecdir='$(exec_prefix)/$(target_alias)'
   toolexeclibdir='$(toolexecdir)/lib'
 else
-  toolexecdir='$(libdir)/gcc-lib/$(target_alias)'
-  toolexeclibdir='$(libdir)'
+  toolexecdir='$(libdir)/gcc/$(target_alias)'
+  toolexeclibdir='$(libdir)/gcc/$(target_alias)/$(gcc-version)'
 fi
 multi_os_directory=`$CC -print-multi-os-directory`
 case $multi_os_directory in
diff -ur gcc-3.4.3/libffi/configure.in gcc-3.4.3-ffi-fixes/libffi/configure.in
--- gcc-3.4.3/libffi/configure.in	2004-04-27 01:10:19.000000000 -0400
+++ gcc-3.4.3-ffi-fixes/libffi/configure.in	2004-11-28 17:40:30.000000000 -0500
@@ -225,8 +225,8 @@
   toolexecdir='$(exec_prefix)/$(target_alias)'
   toolexeclibdir='$(toolexecdir)/lib'
 else
-  toolexecdir='$(libdir)/gcc-lib/$(target_alias)'
-  toolexeclibdir='$(libdir)'
+  toolexecdir='$(libdir)/gcc/$(target_alias)'
+  toolexeclibdir='$(libdir)/gcc/$(target_alias)/$(gcc_version)'
 fi
 multi_os_directory=`$CC -print-multi-os-directory`
 case $multi_os_directory in
