--- Isabelle2011-1-orig/etc/settings	2011-10-10 01:47:58.000000000 +1100
+++ Isabelle2011-1/etc/settings	2012-05-27 23:28:37.283028668 +1000
@@ -66,8 +66,8 @@
 ###
 
 ISABELLE_LINE_EDITOR=""
-[ -z "$ISABELLE_LINE_EDITOR" ] && ISABELLE_LINE_EDITOR="$(type -p rlwrap)"
 [ -z "$ISABELLE_LINE_EDITOR" ] && ISABELLE_LINE_EDITOR="$(type -p ledit)"
+[ -z "$ISABELLE_LINE_EDITOR" ] && ISABELLE_LINE_EDITOR="$(type -p rlwrap)"
 
 
 ###
