From d9eb2378092d39678d1e2cf834c555f83d85b1a1 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Fri, 22 Jun 2018 11:43:23 +0200
Subject: [PATCH 1/2] fix test for 7.3 (int vs integer)

---
 tests/004_WKBWriter.phpt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/004_WKBWriter.phpt b/tests/004_WKBWriter.phpt
index d2b3c85..3a72e2d 100644
--- a/tests/004_WKBWriter.phpt
+++ b/tests/004_WKBWriter.phpt
@@ -139,7 +139,7 @@ class WKBWriterTest extends GEOSTest
             $writer->write(1);
             $this->assertTrue(false);
         } catch (ErrorException $e) {
-            $this->assertContains('expects parameter 1 to be object, integer given', $e->getMessage());
+            $this->assertContains('expects parameter 1 to be object, int', $e->getMessage());
         }
     }
 
@@ -151,7 +151,7 @@ class WKBWriterTest extends GEOSTest
             $writer->writeHEX(1);
             $this->assertTrue(false);
         } catch (ErrorException $e) {
-            $this->assertContains('expects parameter 1 to be object, integer given', $e->getMessage());
+            $this->assertContains('expects parameter 1 to be object, int', $e->getMessage());
         }
     }
 }
-- 
2.14.4

