--- wezterm-891bed31b75f7a71b78e8f42ad07ae89bf99a7de-orig/Cargo.toml	2026-03-29 20:33:24.582050661 +1100
+++ wezterm-891bed31b75f7a71b78e8f42ad07ae89bf99a7de/Cargo.toml	2026-06-15 13:55:17.949999137 +1000
@@ -92,7 +92,7 @@
 filedescriptor = { version="0.8.3", path = "filedescriptor" }
 filenamegen = "0.2.6"
 filesystem = { path = "lua-api-crates/filesystem" }
-finl_unicode = { version = "1.3",  git="https://github.com/wez/finl_unicode.git", branch="no_std" , default-features=false, features=["categories", "grapheme_clusters"]}
+finl_unicode = { path="deps/finl_unicode", default-features=false, features=["categories", "grapheme_clusters"]}
 fixed = "1.23"
 fixedbitset = {version="0.4", default-features=false}
 flume = "0.11"
