diff --git a/source/visualization/pyG4VisManager.cc b/source/visualization/pyG4VisManager.cc
index 33bb0d4..00f86a5 100644
--- a/source/visualization/pyG4VisManager.cc
+++ b/source/visualization/pyG4VisManager.cc
@@ -41,7 +41,7 @@ using namespace boost::python;
 // ====================================================================
 class PyG4VisManager : public G4VisManager {
 public:
-  PyG4VisManager() { SetVerboseLevel(quiet); }
+  PyG4VisManager() { SetVerboseLevel(all); }
   ~PyG4VisManager() { }
 
   virtual void RegisterGraphicsSystems() { }
