--- libsub-9999/src/smpte_dcp_reader.cc
+++ libsub-9999/src/smpte_dcp_reader.cc
@@ -29,7 +29,6 @@
 
 using std::string;
 using std::list;
-using std::stringstream;
 using boost::shared_ptr;
 using namespace sub;
 
@@ -46,9 +45,7 @@
 	
 		string s;
 		reader.ReadTimedTextResource (s, 0, 0);
-		stringstream t;
-		t << s;
-		xml->read_stream (t);
+		xml->read_string (s);
 
 		ASDCP::WriterInfo info;
 		reader.FillWriterInfo (info);
