diff -uNr xsim-0.3.9.4.orig/configure xsim-0.3.9.4/configure
--- xsim-0.3.9.4.orig/configure	2002-10-24 20:53:57.000000000 +0200
+++ xsim-0.3.9.4/configure	2005-01-27 16:21:18.553956144 +0100
@@ -5067,13 +5067,13 @@
 
 
 if test "x$enable_debug" = "x1" ; then
-  CFLAGS="-g -Wall -O2 -DDEBUG -I$x_includes"
+  CFLAGS="-g -Wall -O2 -DDEBUG"
   CXXFLAGS=$CFLAGS
-  LDFLAGS="-g -L$x_libraries"
+  LDFLAGS="-g"
 else
-  CFLAGS="-Wall -O2 -I$x_includes"
+  CFLAGS="-Wall -O2"
   CXXFLAGS=$CFLAGS
-  LDFLAGS="-L$x_libraries"
+  LDFLAGS=""
 fi
 
 if test "x$enable_mdk_patch" = "x1" ; then
@@ -5085,6 +5085,7 @@
 
 if test "$bdbinc" = "NONE"; then
   bdb_include_dirs="\
+    /usr/include/db4.1 \
     /usr/include/db4 \
     /usr/include/db3 \
     /usr/local/include \
diff -uNr xsim-0.3.9.4.orig/plugins/IMs/jianpin/pyphraser.cpp xsim-0.3.9.4/plugins/IMs/jianpin/pyphraser.cpp
--- xsim-0.3.9.4.orig/plugins/IMs/jianpin/pyphraser.cpp	2002-09-29 05:03:29.000000000 +0200
+++ xsim-0.3.9.4/plugins/IMs/jianpin/pyphraser.cpp	2005-01-27 15:42:06.553514560 +0100
@@ -16,6 +16,8 @@
 	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <cassert>
+
 #include "wstring.h"
 #include "pyphraser.h"
 #include "simpleimc.h"
diff -uNr xsim-0.3.9.4.orig/plugins/IMs/jianpin/simpleim.cpp xsim-0.3.9.4/plugins/IMs/jianpin/simpleim.cpp
--- xsim-0.3.9.4.orig/plugins/IMs/jianpin/simpleim.cpp	2002-10-22 11:42:51.000000000 +0200
+++ xsim-0.3.9.4/plugins/IMs/jianpin/simpleim.cpp	2005-01-27 15:38:13.771902744 +0100
@@ -15,6 +15,8 @@
 	License along with this library; if not, write to the Free Software
 	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
+#include <cassert>
+
 #include <X11/keysym.h>
 
 #include "simpleim.h"
diff -uNr xsim-0.3.9.4.orig/plugins/IMs/wubi/wubiim.cpp xsim-0.3.9.4/plugins/IMs/wubi/wubiim.cpp
--- xsim-0.3.9.4.orig/plugins/IMs/wubi/wubiim.cpp	2002-10-22 11:22:13.000000000 +0200
+++ xsim-0.3.9.4/plugins/IMs/wubi/wubiim.cpp	2005-01-27 16:06:29.427123968 +0100
@@ -16,6 +16,8 @@
 	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <cassert>
+
 #include <X11/keysym.h>
 
 #include "wubiim.h"
diff -uNr xsim-0.3.9.4.orig/xsim/im.cpp xsim-0.3.9.4/xsim/im.cpp
--- xsim-0.3.9.4.orig/xsim/im.cpp	2002-09-17 09:55:26.000000000 +0200
+++ xsim-0.3.9.4/xsim/im.cpp	2005-01-27 16:07:52.252532592 +0100
@@ -16,6 +16,8 @@
 	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <cassert>
+
 #include <dlfcn.h>
 
 #include "im.h"
