diff -Naur xsim-0.3.9.4.orig/plugins/IMs/jianpin/dict/char2db.cpp xsim-0.3.9.4/plugins/IMs/jianpin/dict/char2db.cpp
--- xsim-0.3.9.4.orig/plugins/IMs/jianpin/dict/char2db.cpp	2002-09-23 13:22:39.000000000 +0900
+++ xsim-0.3.9.4/plugins/IMs/jianpin/dict/char2db.cpp	2008-09-09 18:56:56.000000000 +0900
@@ -21,6 +21,7 @@
 #include <clocale>
 #include <vector>
 #include <fstream>
+#include <cstdlib>
 
 #include <db_cxx.h>
 
diff -Naur xsim-0.3.9.4.orig/plugins/IMs/jianpin/dict/word2db.cpp xsim-0.3.9.4/plugins/IMs/jianpin/dict/word2db.cpp
--- xsim-0.3.9.4.orig/plugins/IMs/jianpin/dict/word2db.cpp	2002-09-23 13:24:17.000000000 +0900
+++ xsim-0.3.9.4/plugins/IMs/jianpin/dict/word2db.cpp	2008-09-09 18:59:33.000000000 +0900
@@ -21,6 +21,7 @@
 #include <string>
 #include <db_cxx.h>
 #include <locale.h>
+#include <cstdlib>
 
 #include "../simpleimc.h"
 
diff -Naur xsim-0.3.9.4.orig/plugins/IMs/table/dict/word2db.cpp xsim-0.3.9.4/plugins/IMs/table/dict/word2db.cpp
--- xsim-0.3.9.4.orig/plugins/IMs/table/dict/word2db.cpp	2002-09-23 13:34:06.000000000 +0900
+++ xsim-0.3.9.4/plugins/IMs/table/dict/word2db.cpp	2008-09-09 23:52:31.000000000 +0900
@@ -21,6 +21,7 @@
 #include <string>
 #include <db_cxx.h>
 #include <clocale>
+#include <cstdlib>
 
 #include "../tableimc.h"
 
diff -Naur xsim-0.3.9.4.orig/plugins/IMs/wubi/dict/word2db.cpp xsim-0.3.9.4/plugins/IMs/wubi/dict/word2db.cpp
--- xsim-0.3.9.4.orig/plugins/IMs/wubi/dict/word2db.cpp	2002-09-23 13:28:45.000000000 +0900
+++ xsim-0.3.9.4/plugins/IMs/wubi/dict/word2db.cpp	2008-09-09 23:19:27.000000000 +0900
@@ -21,6 +21,7 @@
 #include <string>
 #include <db_cxx.h>
 #include <locale.h>
+#include <cstdlib>
 
 #include "../wubiimc.h"
 
