--- configure.ac.orig	2014-02-10 00:55:32.716237174 +0100
+++ configure.ac	2014-02-10 00:58:37.390508206 +0100
@@ -44,8 +44,8 @@
 if ! test "$localesonly" = "yes"; then
 
   if test "X$GCC" = Xyes; then
-    CFLAGS="$CFLAGS -Wall -g -fexceptions"
-    CXXFLAGS="$CXXFLAGS -Wall -g -fexceptions"
+    CFLAGS="$CFLAGS -Wall -fexceptions"
+    CXXFLAGS="$CXXFLAGS -Wall -fexceptions"
   fi
 
   CHECK_CXX0X
