From a51acc0c353959e8af06a031724a79cfe6b60963 Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <leio@gentoo.org>
Date: Sun, 5 Apr 2020 00:05:49 +0300
Subject: [PATCH] meson: Find python explicitly to honor downstream python
 choice

---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index 461d8d2..a09aee9 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -308,7 +308,7 @@ gtk_vnc_sources = gtk_vnc_gir_sources + gvnc_marshal + [
   'vncdisplaykeymap.c',
 ]
 
-python = import('python').find_installation()
+python = import('python').find_installation('python3')
 keymaps = [
   'xorgevdev',
   'xorgkbd',
-- 
2.20.1

