diff --git a/sabnzbd/cfg.py b/sabnzbd/cfg.py
index 39e418b..509a962 100644
--- a/sabnzbd/cfg.py
+++ b/sabnzbd/cfg.py
@@ -238,8 +238,8 @@ no_ipv6 = OptionBool('misc', 'no_ipv6', False)
 
 growl_server = OptionStr('growl', 'growl_server')
 growl_password = OptionPassword('growl', 'growl_password')
-growl_enable = OptionBool('growl', 'growl_enable', not sabnzbd.DARWIN_ML)
-ntfosd_enable = OptionBool('growl', 'ntfosd_enable', not sabnzbd.WIN32 and not sabnzbd.DARWIN)
+growl_enable = OptionBool('growl', 'growl_enable', 0)
+ntfosd_enable = OptionBool('growl', 'ntfosd_enable', 0)
 ncenter_enable = OptionBool('growl', 'ncenter_enable', sabnzbd.DARWIN)
 notify_classes = OptionList('growl', 'notify_classes', NOTIFY_KEYS)
 
