https://bugs.gentoo.org/853121
https://bugs.gentoo.org/726200
--- a/libs/cairo/m4/float.m4
+++ b/libs/cairo/m4/float.m4
@@ -22,23 +22,24 @@ AC_DEFUN([AX_C_FLOAT_WORDS_BIGENDIAN],
 # file, the target platform stores float words in big-endian order. If grep
 # finds "seesnoon", float words are in little-endian order. If neither value
 # is found, the user is instructed to specify the ordering.
 
 ax_cv_c_float_words_bigendian=unknown
-AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
+AC_LINK_IFELSE([AC_LANG_SOURCE([[
 
-double d = 90904234967036810337470478905505011476211692735615632014797120844053488865816695273723469097858056257517020191247487429516932130503560650002327564517570778480236724525140520121371739201496540132640109977779420565776568942592.0;
+double d __attribute__((used)) = 90904234967036810337470478905505011476211692735615632014797120844053488865816695273723469097858056257517020191247487429516932130503560650002327564517570778480236724525140520121371739201496540132640109977779420565776568942592.0;
+int main() { return 0; }
 
 ]])], [
 
 # allow users to override default 'strings' with 'llvm-strings'
 # or ${CHOST}-strings.
 AC_CHECK_TOOL([STRINGS], [strings])
-if $STRINGS - conftest.$ac_objext | grep noonsees >/dev/null ; then
+if $STRINGS - conftest | grep noonsees >/dev/null ; then
   ax_cv_c_float_words_bigendian=yes
 fi
-if $STRINGS - conftest.$ac_objext | grep seesnoon >/dev/null ; then
+if $STRINGS - conftest | grep seesnoon >/dev/null ; then
   if test "$ax_cv_c_float_words_bigendian" = unknown; then
     ax_cv_c_float_words_bigendian=no
   else
     ax_cv_c_float_words_bigendian=unknown
   fi

