--- Lib/Minuit/Makefile.PL
+++ Lib/Minuit/Makefile.PL
@@ -149,7 +149,7 @@
 	    ("
 
 minuitlib/$_\$(OBJ_EXT): minuitlib/$_.f 
-	$mycompiler -c $hack_64bit -o minuitlib/$_\$(OBJ_EXT) $mycflags minuitlib/$_.f
+	\$(FC) \$(FFLAGS) -fPIC -c -o minuitlib/$_\$(OBJ_EXT) $mycflags minuitlib/$_.f
 " )} @minuitfiles;
 
 	if (!defined($PDL::Config{MINUIT_LIB})){
--- Lib/Slatec/Makefile.PL
+++ Lib/Slatec/Makefile.PL
@@ -136,7 +136,7 @@
 ("
 
 slatec/$_\$(OBJ_EXT): slatec/$_.f 
-	$mycompiler -c $hack_64bit -o slatec/$_\$(OBJ_EXT) $mycflags slatec/$_.f
+	\$(FC) \$(FFLAGS) -c -fPIC -o slatec/$_\$(OBJ_EXT) $mycflags slatec/$_.f
 " )} @slatecfiles;
  
 }
