--- a/Makefile	2017-12-08 12:22:59.000000000 +0300
+++ b/Makefile	2018-01-14 15:42:56.804047742 +0300
@@ -165,12 +165,12 @@
 # -MMD to generate header dependencies. (They cannot be
 #  generated if building universal binaries on OSX)
 ifeq ($(YQ2_OSTYPE), Darwin)
-CFLAGS := -O2 -fno-strict-aliasing -fomit-frame-pointer \
-		  -Wall -pipe -g -fwrapv
+CFLAGS += -fno-strict-aliasing -fomit-frame-pointer \
+		  -Wall -fwrapv
 CFLAGS += $(OSX_ARCH)
 else
-CFLAGS := -std=gnu99 -O2 -fno-strict-aliasing \
-		  -Wall -pipe -g -ggdb -MMD -fwrapv
+CFLAGS += -std=gnu99 -fno-strict-aliasing \
+		  -Wall -MMD -fwrapv
 endif
 
 # ----------
@@ -263,15 +263,15 @@
 
 # Base LDFLAGS.
 ifeq ($(YQ2_OSTYPE),Linux)
-LDFLAGS := -L/usr/lib -lm -ldl -rdynamic
+LDFLAGS += -lm -ldl -rdynamic
 else ifeq ($(YQ2_OSTYPE),FreeBSD)
-LDFLAGS := -L/usr/local/lib -lm
+LDFLAGS += -lm
 else ifeq ($(YQ2_OSTYPE),OpenBSD)
-LDFLAGS := -L/usr/local/lib -lm
+LDFLAGS += -lm
 else ifeq ($(YQ2_OSTYPE),Windows)
-LDFLAGS := -L/usr/lib -lws2_32 -lwinmm -static-libgcc
+LDFLAGS += -lws2_32 -lwinmm -static-libgcc
 else ifeq ($(YQ2_OSTYPE), Darwin)
-LDFLAGS := $(OSX_ARCH) -lm
+LDFLAGS += $(OSX_ARCH) -lm
 endif
 
 CFLAGS += -fvisibility=hidden
