libAfterStep/alpha_header.h:15: error: conflicting types for 'strlen'

--- libAfterStep/asapp.h
+++ libAfterStep/asapp.h
@@ -15,10 +15,6 @@
 #include "sun_headers.h"
 #endif
 
-#if defined(__alpha)
-#include "alpha_header.h"
-#endif /* NEEDS_ALPHA_HEADER */
-
 /* Some people say that AIX and AIXV3 need 3 preceding underscores, other say
  * no. I'll do both */
 #if defined ___AIX || defined _AIX || defined __QNX__ || defined ___AIXV3 || defined AIXV3 || defined _SEQUENT_