--- makefile.orig	2010-09-07 15:05:01.000000000 -0400
+++ makefile	2010-09-07 15:05:19.000000000 -0400
@@ -15,7 +15,7 @@
   CC = gcc -std=c99 -U__STRICT_ANSI__ -g $(OS_CCDEFS) -I .
   ifeq ($(USE_NETWORK),)
   else
-    NETWORK_OPT = -DUSE_NETWORK -isystem /usr/local/include /usr/local/lib/libpcap.a
+    NETWORK_OPT = -DUSE_NETWORK -DNEED_PCAP_SENDPACKET -lpcap
   endif
 else
   #Win32 Environments
