diff --git a/Makefile b/Makefile
index 55b65b1..b04f270 100644
--- a/Makefile
+++ b/Makefile
@@ -36,10 +36,9 @@ $(NAME): $(OBJS)
 strip: $(NAME)
 	$(STRIP) --strip-unneeded $(NAME)
 
-install: strip
+install:
 	$(INSTALL) -D -m 0755 $(NAME) $(DESTDIR)$(PREFIX)/sbin/$(NAME)
 	$(INSTALL) -D -m 0644 $(NAME).1 $(DESTDIR)$(PREFIX)/share/man/man1/$(NAME).1
-	gzip -9 $(DESTDIR)$(PREFIX)/share/man/man1/$(NAME).1
 	sed "s|@PATH@|$(PREFIX)/sbin|" jitterentropy.service.in > jitterentropy.service
 	$(INSTALL) -D -m 0644 jitterentropy.service $(DESTDIR)$(UNITDIR)/jitterentropy.service
 
