From 731519b5573ab503b78f2c892e2d5c4bcabef1ea Mon Sep 17 00:00:00 2001
From: Sobhan Mohammadpour <www.sobhan.mohammadpour@gmail.com>
Date: Wed, 22 Nov 2017 23:33:11 +0330
Subject: [PATCH] use-external-bzip

---
 meshlabplugins/edit_arc3D/edit_arc3D.pro | 31 -------------------------------
 1 file changed, 31 deletions(-)

diff --git a/meshlabplugins/edit_arc3D/edit_arc3D.pro b/meshlabplugins/edit_arc3D/edit_arc3D.pro
index 08b5860..75db42c 100644
--- a/meshlabplugins/edit_arc3D/edit_arc3D.pro
+++ b/meshlabplugins/edit_arc3D/edit_arc3D.pro
@@ -28,34 +28,3 @@ SOURCES       += edit_arc3D.cpp   \
 TARGET        = edit_arc3D
 RESOURCES     = edit_arc3D.qrc
 
-win32-msvc2005:LIBS	   += ../../external/lib/win32-msvc2005/bz2.lib
-win32-msvc2008:LIBS	   += ../../external/lib/win32-msvc2008/bz2.lib
-win32-msvc2010:LIBS	   += ../../external/lib/win32-msvc2010/bz2.lib
-win32-msvc2012:LIBS	   += ../../external/lib/win32-msvc2012/bz2.lib
-win32-msvc2013:LIBS	   += ../../external/lib/win32-msvc2013/bz2.lib
-win32-msvc2015:LIBS	   += ../../external/lib/win32-msvc2015/bz2.lib
-win32-g++:LIBS	+= ../../external/lib/win32-gcc/libbz2.a
-linux-g++:LIBS	+= ../../external/lib/linux-g++/libbz2.a
-linux-g++-32:LIBS	+= ../../external/lib/linux-g++-32/libbz2.a
-linux-g++-64:LIBS	+= ../../external/lib/linux-g++-64/libbz2.a
-mac:LIBS   += -lbz2
-
-win32-msvc2005:INCLUDEPATH += ../../external/bzip2-1.0.5
-win32-msvc2008:INCLUDEPATH += ../../external/bzip2-1.0.5
-win32-msvc2010:INCLUDEPATH += ../../external/bzip2-1.0.5
-win32-msvc2012:INCLUDEPATH += ../../external/bzip2-1.0.5
-win32-msvc2013:INCLUDEPATH += ../../external/bzip2-1.0.5
-win32-msvc2015:INCLUDEPATH += ../../external/bzip2-1.0.5
-
-win32-g++:INCLUDEPATH  += ../../external/bzip2-1.0.5
-linux-g++:INCLUDEPATH	+= ../../external/bzip2-1.0.5
-linux-g++-32:INCLUDEPATH	+= ../../external/bzip2-1.0.5
-linux-g++-64:INCLUDEPATH	+= ../../external/bzip2-1.0.5
-
-
-CONFIG(release, debug|release) {
-	win32-g++:release:QMAKE_CXXFLAGS -= -O2
-	win32-g++:release:QMAKE_CFLAGS -= -O2
-	win32-g++:release:QMAKE_CXXFLAGS += -O3 -mtune=pentium3 -ffast-math
-}
-
-- 
2.15.0

