--- a/meshlabplugins/edit_align/align/AlignGlobal.cpp	2016-12-23 13:24:59.000000000 +0300
+++ b/meshlabplugins/edit_align/align/AlignGlobal.cpp	2017-11-26 03:05:11.010113483 +0300
@@ -429,11 +429,11 @@
 
 ******************************/
 
-bool AlignGlobal::GlobalAlign(const std::map<int,string> &Names, 	const double epsilon, int maxiter, bool Rigid, FILE *elfp, CallBack* cb )
+bool AlignGlobal::GlobalAlign(const std::map<int,string> &Names, 	const double epsilon, int maxiter, bool Rigid, FILE *elfp, CallBackPos* cb )
 {
   double change;
   int step, localmaxiter;
-  cb("Global Alignment...");
+  cb(0,"Global Alignment...");
   LOG(elfp,"----------------\n----------------\nGlobalAlignment (target eps %7.3f)\n",epsilon);
 
   queue<AlignGlobal::Node *>	Q;
