diff --git a/include/HTTP.h b/include/HTTP.h
index 6be27f9..916193b 100644
--- a/include/HTTP.h
+++ b/include/HTTP.h
@@ -21,7 +21,12 @@
 
 #include <string>
 #include <curl/curl.h>
-#include <curl/types.h>
+#if ! defined(LIBCURL_VERSION_PATCH) \
+    || (LIBCURL_VERSION_MAJOR < 7) \
+    || ((LIBCURL_VERSION_MAJOR == 7) && (LIBCURL_VERSION_MINOR < 21)) \
+    || ((LIBCURL_VERSION_MAJOR == 7) && (LIBCURL_VERSION_MINOR == 21) && (LIBCURL_VERSION_PATCH < 7))
+# include <curl/types.h>
+#endif
 #include <curl/easy.h>
 
 #include "Networking.h"
diff --git a/src/breakpad/external/src/common/linux/http_upload.cc b/src/breakpad/external/src/common/linux/http_upload.cc
index 8748d11..c079aef 100644
--- a/src/breakpad/external/src/common/linux/http_upload.cc
+++ b/src/breakpad/external/src/common/linux/http_upload.cc
@@ -31,7 +31,12 @@
 #include <dlfcn.h>
 #include <curl/curl.h>
 #include <curl/easy.h>
-#include <curl/types.h>
+#if ! defined(LIBCURL_VERSION_PATCH) \
+    || (LIBCURL_VERSION_MAJOR < 7) \
+    || ((LIBCURL_VERSION_MAJOR == 7) && (LIBCURL_VERSION_MINOR < 21)) \
+    || ((LIBCURL_VERSION_MAJOR == 7) && (LIBCURL_VERSION_MINOR == 21) && (LIBCURL_VERSION_PATCH < 7))
+# include <curl/types.h>
+#endif
 
 #include "common/linux/http_upload.h"
 
diff --git a/src/breakpad/external/src/common/linux/libcurl_wrapper.cc b/src/breakpad/external/src/common/linux/libcurl_wrapper.cc
index 5bea3af..1fa6d24 100644
--- a/src/breakpad/external/src/common/linux/libcurl_wrapper.cc
+++ b/src/breakpad/external/src/common/linux/libcurl_wrapper.cc
@@ -29,7 +29,12 @@
 
 #include <curl/curl.h>
 #include <curl/easy.h>
-#include <curl/types.h>
+#if ! defined(LIBCURL_VERSION_PATCH) \
+    || (LIBCURL_VERSION_MAJOR < 7) \
+    || ((LIBCURL_VERSION_MAJOR == 7) && (LIBCURL_VERSION_MINOR < 21)) \
+    || ((LIBCURL_VERSION_MAJOR == 7) && (LIBCURL_VERSION_MINOR == 21) && (LIBCURL_VERSION_PATCH < 7))
+# include <curl/types.h>
+#endif
 #include <dlfcn.h>
 
 #include <string>
diff --git a/src/common/HTTP.cpp b/src/common/HTTP.cpp
index 0782f9c..7f7a852 100644
--- a/src/common/HTTP.cpp
+++ b/src/common/HTTP.cpp
@@ -25,7 +25,12 @@
 	#include <stdlib.h>
 #endif
 #include <curl/curl.h>
-#include <curl/types.h>
+#if ! defined(LIBCURL_VERSION_PATCH) \
+    || (LIBCURL_VERSION_MAJOR < 7) \
+    || ((LIBCURL_VERSION_MAJOR == 7) && (LIBCURL_VERSION_MINOR < 21)) \
+    || ((LIBCURL_VERSION_MAJOR == 7) && (LIBCURL_VERSION_MINOR == 21) && (LIBCURL_VERSION_PATCH < 7))
+# include <curl/types.h>
+#endif
 #include <curl/easy.h>
 
 #include "LieroX.h"
