diff -ur lib/libgeom.old/Makefile lib/libgeom/Makefile
--- lib/libgeom.old/Makefile	2011-07-10 14:23:51.000000000 +0000
+++ lib/libgeom/Makefile	2011-07-10 15:12:39.000000000 +0000
@@ -13,8 +13,8 @@
 
 WARNS?=	3
 
-DPADD=	${LIBBSDXML} ${LIBSBUF}
-LDADD=	-lbsdxml -lsbuf
+DPADD=	${LIBSBUF}
+LDADD=	-lexpat -lsbuf
 
 MAN=	libgeom.3
 
diff -ur lib/libgeom.old/geom_xml2tree.c lib/libgeom/geom_xml2tree.c
--- lib/libgeom.old/geom_xml2tree.c	2010-12-21 17:09:25.000000000 +0000
+++ lib/libgeom/geom_xml2tree.c	2011-07-10 15:12:52.000000000 +0000
@@ -43,7 +43,7 @@
 #include <sys/sbuf.h>
 #include <sys/sysctl.h>
 #include <err.h>
-#include <bsdxml.h>
+#include <expat.h>
 #include <libgeom.h>
 
 struct mystate {
