Index: fbsd-6/lib/libipsec/policy_token.l
===================================================================
--- fbsd-6.orig/lib/libipsec/policy_token.l
+++ fbsd-6/lib/libipsec/policy_token.l
@@ -138,8 +138,6 @@ void
 __policy__strbuffer__init__(msg)
 	char *msg;
 {
-	if (yy_current_buffer)
-		yy_delete_buffer(yy_current_buffer);
 	strbuffer = (YY_BUFFER_STATE)yy_scan_string(msg);
 	yy_switch_to_buffer(strbuffer);
 
