diff --git a/lib/libgeom/Makefile b/lib/libgeom/Makefile
index bbfb57b..a7c17e3 100644
--- a/lib/libgeom/Makefile
+++ b/lib/libgeom/Makefile
@@ -13,7 +13,7 @@ CFLAGS += -I${.CURDIR}
 
 WARNS?=	3
 
-DPADD=	${LIBBSDXML} ${LIBSBUF}
+DPADD=	${LIBSBUF}
 LDADD=	-lbsdxml -lsbuf
 
 MAN=	libgeom.3
diff --git a/lib/libgeom/geom_xml2tree.c b/lib/libgeom/geom_xml2tree.c
index 9dedb8e..d96420c 100644
--- a/lib/libgeom/geom_xml2tree.c
+++ b/lib/libgeom/geom_xml2tree.c
@@ -43,7 +43,7 @@
 #include <sys/sbuf.h>
 #include <sys/sysctl.h>
 #include <err.h>
-#include <bsdxml.h>
+#include <expat.h>
 #include <libgeom.h>
 
 struct mystate {
diff --git a/lib/libmt/mtlib.c b/lib/libmt/mtlib.c
index d1fc0b1..89ac55e 100644
--- a/lib/libmt/mtlib.c
+++ b/lib/libmt/mtlib.c
@@ -48,7 +48,7 @@ __FBSDID("$FreeBSD: stable/10/lib/libmt/mtlib.c 280438 2015-03-24 14:36:10Z ken
 #include <unistd.h>
 #include <stdint.h>
 #include <errno.h>
-#include <bsdxml.h>
+#include <expat.h>
 #include <mtlib.h>
 
 /*
