--- include/Makefile.orig	2013-06-16 10:47:52.000000000 +0200
+++ include/Makefile	2013-06-16 10:47:37.000000000 +0200
@@ -6,11 +6,11 @@
 .include <bsd.own.mk>
 
 CLEANFILES= osreldate.h version vers.c
+SUBDIR= arpa protocols rpcsvc rpc xlocale
-SUBDIR= arpa gssapi protocols rpcsvc rpc xlocale
 INCS=	a.out.h ar.h assert.h bitstring.h complex.h cpio.h _ctype.h ctype.h \
 	db.h \
 	dirent.h dlfcn.h elf.h elf-hints.h err.h fmtmsg.h fnmatch.h fstab.h \
+	fts.h ftw.h getopt.h glob.h grp.h \
-	fts.h ftw.h getopt.h glob.h grp.h gssapi.h \
 	ieeefp.h ifaddrs.h \
 	inttypes.h iso646.h kenv.h langinfo.h libgen.h limits.h link.h \
 	locale.h malloc.h malloc_np.h memory.h monetary.h mpool.h mqueue.h \
