From bf4c1ab651a82eb369ffa3dc4b20246d043d797a Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@magic.io>
Date: Wed, 28 Mar 2018 12:02:53 -0400
Subject: [PATCH] Fix app restart

---
 src/main-process/atom-application.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main-process/atom-application.js b/src/main-process/atom-application.js
index 10b791761..d16a68fba 100644
--- a/src/main-process/atom-application.js
+++ b/src/main-process/atom-application.js
@@ -1344,6 +1344,7 @@ class AtomApplication extends EventEmitter {
 
   restart () {
     const args = []
+    args.push("--app=" + this.resourcePath);
     if (this.safeMode) args.push('--safe')
     if (this.logFile != null) args.push(`--log-file=${this.logFile}`)
     if (this.socketPath != null) args.push(`--socket-path=${this.socketPath}`)
-- 
2.16.1

