diff -Naur a/tools/cross-wine32.in b/tools/cross-wine32.in
--- a/tools/cross-wine32.in	2019-01-12 16:45:09.190471883 +0000
+++ b/tools/cross-wine32.in	2019-01-12 16:48:22.569404982 +0000
@@ -5,8 +5,8 @@
 pkgconfig = '@PKG_CONFIG@'
 
 [properties]
-c_args = ['-m32', '--no-gnu-unique']
-c_link_args = ['-m32', '-mwindows']
+c_args = @CFLAGS@ + ['-m32', '--no-gnu-unique']
+c_link_args = @LDFLAGS@ + ['-m32', '-mwindows']
 needs_exe_wrapper = true
 
 [host_machine]
diff -Naur a/tools/cross-wine64.in b/tools/cross-wine64.in
--- a/tools/cross-wine64.in	2019-01-12 16:45:09.190471883 +0000
+++ b/tools/cross-wine64.in	2019-01-12 16:48:19.865349306 +0000
@@ -5,8 +5,8 @@
 pkgconfig = '@PKG_CONFIG@'
 
 [properties]
-c_args = ['-m64', '--no-gnu-unique']
-c_link_args = ['-m64', '-mwindows']
+c_args = @CFLAGS@ + ['-m64', '--no-gnu-unique']
+c_link_args = @LDFLAGS@ + ['-m64', '-mwindows']
 needs_exe_wrapper = true
 
 [host_machine]
