--- tools/buildsharedlib.sh
+++ tools/buildsharedlib.sh
@@ -115,13 +115,13 @@
 	*)
 		if [ `$LD -v 2>&1 | grep -c gcc` -gt 0 -a \
 			`gcc -Wl,-Bsymbolic 2>&1 | grep -c unrecognized` = 0 ] ; then
-			$LD -shared -Wl,-Bsymbolic -o $LIBNAME `cat $LINKFILE` `./tools/getlibs.sh autodetect` ;
+			$LD -shared $LDFLAGS -Wl,-soname,$LIBNAME -Wl,-Bsymbolic -o $LIBNAME `cat $LINKFILE` `./tools/getlibs.sh autodetect` ;
 		else
-			$LD -shared -o $LIBNAME `cat $LINKFILE` `./tools/getlibs.sh autodetect` ;
+			$LD -shared $LDFLAGS -Wl,-soname,$LIBNAME -o $LIBNAME `cat $LINKFILE` `./tools/getlibs.sh autodetect` ;
 		fi
 		if [ `which objdump` -a `objdump -p $LIBNAME | grep -c TEXTREL` -gt '0' ] ; then
 			echo "Warning: Shared library still contains TEXTREL records." ;
 		fi
-		strip $LIBNAME ;;
+		;;
 esac
 rm -f $LINKFILE
