--- ./g15daemon-1.9.5.3/g15daemon/utility_funcs.c	2008-01-26 06:02:07.000000000 +0100
+++ ./g15daemon-1.9.5.3/g15daemon/utility_funcs.c	2009-02-01 17:00:55.203009264 +0100
@@ -356,7 +356,7 @@
     config_items_t * item=NULL;
     char line[1024];
     
-    config_fd = open(filename,O_CREAT|O_RDWR|O_TRUNC);
+    config_fd = open(filename,O_CREAT|O_RDWR|O_TRUNC,0664);
     if(config_fd){
     snprintf(line,1024,"# G15Daemon Configuration File\n# any items entered before a [section] header\n# will be in the Global config space\n# comments you wish to keep should start with a semicolon';'\n");
     write(config_fd,line,strlen(line));
