--- a/epan/Makefile.am
+++ b/epan/Makefile.am
@@ -137,6 +137,7 @@
 # Add the object files for missing routines, if any.
 #
 libwireshark_la_LIBADD = \
+	${top_builddir}/wiretap/libwiretap.la ${top_builddir}/wsutil/libwsutil.la \
 	libwireshark_generated.la \
 	libwireshark_asmopt.la crypt/libairpdcap.la \
 	ftypes/libftypes.la dfilter/libdfilter.la dissectors/libdissectors.la \
@@ -144,8 +145,7 @@
 	wmem/libwmem.la $(wslua_lib) $(wspython_lib) @SOCKET_LIBS@ @NSL_LIBS@ \
 	@C_ARES_LIBS@ @ADNS_LIBS@ @LIBGCRYPT_LIBS@ @LIBGNUTLS_LIBS@ \
 	@KRB5_LIBS@ @SSL_LIBS@ @LIBSMI_LDFLAGS@ @GEOIP_LIBS@ \
-	${top_builddir}/wiretap/libwiretap.la @GLIB_LIBS@ \
-	${top_builddir}/wsutil/libwsutil.la -lm
+	@GLIB_LIBS@ -lm
 
 libwireshark_la_DEPENDENCIES = \
 	libwireshark_generated.la \
