--- root/io/io/Module.mk.orig	2012-02-29 19:46:38.000000000 +0400
+++ root/io/io/Module.mk	2012-03-09 02:29:19.905665572 +0400
@@ -67,15 +67,5 @@
 distclean::     distclean-$(MODNAME)
 
 ##### extra rules ######
-#ifeq ($(GCC_VERS_FULL),gcc-4.4.0)
-ifeq ($(GCC_VERS),gcc-4.4)
-ifneq ($(filter -O%,$(OPT)),)
    $(call stripsrc,$(IODIRS)/TStreamerInfoReadBuffer.o): CXXFLAGS += -DR__EXPLICIT_FUNCTION_INSTANTIATION
-endif
-endif
-ifeq ($(GCC_VERS),gcc-4.5)
-ifneq ($(filter -O%,$(OPT)),)
    $(call stripsrc,$(IODIRS)/TStreamerInfoReadBuffer.o): CXXFLAGS += -DR__EXPLICIT_FUNCTION_INSTANTIATION
-endif
-endif
-
