diff -ur root-trunk.orig/math/unuran/Module.mk root-trunk/math/unuran/Module.mk
--- root.orig/math/unuran/Module.mk	2010-11-25 20:41:32.000000000 +0100
+++ root/math/unuran/Module.mk	2010-11-25 20:51:47.190704977 +0100
@@ -20,7 +20,7 @@
 UNURANETAG   := $(call stripsrc,$(UNURANDIRS)/headers.d)
 UNRCFG       := $(call stripsrc,$(UNURANDIRS)/$(UNRVERS)/config.h)
 
-UNRTARCONTENT:=$(subst $(UNRVERS),$(UNRDIRS),$(shell mkdir -p $(UNRDIR); cd $(UNRDIR); gunzip -c $(UNRSRCS) | tar tf -))
+UNRTARCONTENT:=
 UNRS         := $(filter %.c, \
                 $(filter $(UNRDIRS)/src/utils/%,$(UNRTARCONTENT)) \
                 $(filter $(UNRDIRS)/src/methods/%,$(UNRTARCONTENT)) \
@@ -69,10 +69,9 @@
 ##### local rules #####
 .PHONY:         all-$(MODNAME) clean-$(MODNAME) distclean-$(MODNAME)
 
-include/%.h: 	$(UNURANDIRI)/%.h $(UNURANETAG)
+include/%.h: 	$(UNURANDIRI)/%.h
 		cp $< $@
 
-$(UNURANDEP):   $(UNRCFG)
 $(UNRS):        $(UNURANETAG)
 
 $(UNURANETAG):	$(UNRSRCS)
@@ -125,12 +124,12 @@
 		GNUMAKE=$(MAKE) ./configure  CC="$$ACC"  \
 		CFLAGS="$$ACFLAGS");
 
-$(UNURANLIB):   $(UNRCFG) $(UNRO) $(UNURANO) $(UNURANDO) $(ORDER_) \
+$(UNURANLIB):   $(UNURANO) $(UNURANDO) $(ORDER_) \
                 $(MAINLIBS) $(UNURANLIBDEP)
 		@$(MAKELIB) $(PLATFORM) $(LD) "$(LDFLAGS)"  \
 		   "$(SOFLAGS)" libUnuran.$(SOEXT) $@     \
 		   "$(UNURANO) $(UNURANDO)"             \
-		   "$(UNURANLIBEXTRA) $(UNRO)"
+		   "$(UNURANLIBEXTRA) -lunuran"
 
 $(UNURANDS):    $(UNRINIT) $(UNURANDH1) $(UNURANL) $(ROOTCINTTMPDEP)
 		$(MAKEDIR)
