--- root.orig/configure	2010-12-20 07:17:54.000000000 +0000
+++ root/configure	2010-12-20 07:16:05.000000000 +0000
@@ -709,7 +709,7 @@
        cat <<EOF > conftest.mk
 include ${ac_srcdir}/config/Makefile.${arch}
 conftest: conftest.c
-	\$(CC) \$(CFLAGS) \$(LDFLAGS) $linkdir $linklib \$< -o \$@
+	\$(CC) \$(CFLAGS) \$(LDFLAGS) \$< $linkdir $linklib  -o \$@
 
 conftest.c:
 	echo "int main() { return 0; }" > \$@
@@ -723,7 +723,7 @@
        cat <<EOF > conftest.mk
 include ${ac_srcdir}/config/Makefile.${arch}
 conftest:conftest.c
-	\$(CC) \$(CFLAGS) \$(LDFLAGS) $linkdir $linklib \$< -o \$@
+	\$(CC) \$(CFLAGS) \$(LDFLAGS) \$< $linkdir $linklib -o \$@
 
 conftest.c:
 	echo "extern int $linksymbol (); " > \$@
