diff -r 035eded55c4d lib-python/2.7/sre_constants.py
--- a/lib-python/2.7/sre_constants.py	Wed Apr 16 18:30:13 2014 -0600
+++ b/lib-python/2.7/sre_constants.py	Fri Jul 25 10:31:27 2014 -0700
@@ -15,7 +15,11 @@
 
 MAGIC = 20031017
 
-from _sre import MAXREPEAT
+try:
+    from _sre import MAXREPEAT
+except ImportError:
+    import _sre
+    MAXREPEAT = _sre.MAXREPEAT = 65535
 
 # SRE standard exception (access as sre.error)
 # should this really be here?
