diff --git a/cmake/FindR.cmake b/cmake/FindR.cmake
index 6303417..6494884 100644
--- a/cmake/FindR.cmake
+++ b/cmake/FindR.cmake
@@ -31,7 +31,7 @@ IF(R_EXECUTABLE)
         OUTPUT_VARIABLE R_INCLUDE_DIR)
     ELSE(WIN32)
       EXECUTE_PROCESS(
-        COMMAND ${R_EXECUTABLE} CMD sh -c "echo -n $R_INCLUDE_DIR"
+        COMMAND ${R_EXECUTABLE} CMD sh -c "printf $R_INCLUDE_DIR"
         OUTPUT_VARIABLE R_INCLUDE_DIR)
     ENDIF(WIN32)
   ENDIF(NOT R_INCLUDE_DIR)
