diff -Nur ColPack-1.0.6.orig/Makefile.am ColPack-1.0.6/Makefile.am
--- ColPack-1.0.6.orig/Makefile.am	2012-04-13 03:25:37.000000000 +0100
+++ ColPack-1.0.6/Makefile.am	2012-04-13 03:30:35.000000000 +0100
@@ -1,5 +1,5 @@
 ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS}
-COLPACK_COMMON_FLAGS = $(EXTRA_FLAGS)  -O5  
+COLPACK_COMMON_FLAGS = $(EXTRA_FLAGS)
 
 INCLUDES =  $(COLPACK_COMMON_FLAGS) \
 		-I$(top_srcdir)/Main \
