diff --git a/ccp4i/crank/plugins/prep/crank_prep.tcl b/ccp4i/crank/plugins/prep/crank_prep.tcl
index 7d05905..d01f1dc 100755
--- a/ccp4i/crank/plugins/prep/crank_prep.tcl
+++ b/ccp4i/crank/plugins/prep/crank_prep.tcl
@@ -226,7 +226,7 @@ proc run_truncate { i j } {
     set script "$script\n$labout"
     set script "$script\nEND\n"
 	
-    set run_command "truncate $command << \"$script\""
+    set run_command "ftruncate $command << \"$script\""
     puts $run_command
     catch {eval exec $run_command } output 
     puts $output
diff --git a/ccp4i/scripts/dtrek2mtz.script b/ccp4i/scripts/dtrek2mtz.script
index e74aa0d..a2a1adb 100755
--- a/ccp4i/scripts/dtrek2mtz.script
+++ b/ccp4i/scripts/dtrek2mtz.script
@@ -36,7 +36,7 @@
     set HKL_TRUNCATE [GetTmpFileName -ext mtz]
 
     CreateComScript truncate truncate_script
-    set cmd "[BinPath truncate] HKLIN \"$HKLTMP\" HKLOUT \"$HKL_TRUNCATE\""
+    set cmd "[BinPath ftruncate] HKLIN \"$HKLTMP\" HKLOUT \"$HKL_TRUNCATE\""
     set status [Execute $cmd $truncate_script program_status report]
 
     HandleHarvestFile $HARVEST_MODE $HARVEST_PNAME $HARVEST_DNAME truncate
diff --git a/ccp4i/scripts/import.script b/ccp4i/scripts/import.script
index 90cba83..1c90c33 100755
--- a/ccp4i/scripts/import.script
+++ b/ccp4i/scripts/import.script
@@ -144,7 +144,7 @@ if { $FORMAT ==  "MTZ" } {
 
     set truncatemtz [GetTmpFileName -ext mtz ]
     CreateComScript truncate truncate_script
-    set cmd "[BinPath truncate] HKLIN \"$tmpmtz\" HKLOUT \"$truncatemtz\""
+    set cmd "[BinPath ftruncate] HKLIN \"$tmpmtz\" HKLOUT \"$truncatemtz\""
     set status [Execute $cmd $truncate_script program_status report]
 
 
diff --git a/ccp4i/scripts/import_scaled.script b/ccp4i/scripts/import_scaled.script
index dcc619b..7250513 100755
--- a/ccp4i/scripts/import_scaled.script
+++ b/ccp4i/scripts/import_scaled.script
@@ -62,7 +62,7 @@
 
     } else {
       CreateComScript truncate truncate_script
-      set cmd "[BinPath truncate] HKLIN \"$HKLTMP\" HKLOUT \"$HKL_TRUNCATE\""
+      set cmd "[BinPath ftruncate] HKLIN \"$HKLTMP\" HKLOUT \"$HKL_TRUNCATE\""
       set status [Execute $cmd $truncate_script program_status report]
 
       HandleHarvestFile $HARVEST_MODE $HARVEST_PNAME $HARVEST_DNAME truncate
diff --git a/ccp4i/scripts/scala.script b/ccp4i/scripts/scala.script
index 6940ccd..bde10d6 100755
--- a/ccp4i/scripts/scala.script
+++ b/ccp4i/scripts/scala.script
@@ -275,7 +275,7 @@ END"
 
       } else {
         CreateComScript truncate truncate_script
-        set cmd "[BinPath truncate] HKLIN \"$HKLTMP\" HKLOUT \"$HKL_TRUNCATE\""  
+        set cmd "[BinPath ftruncate] HKLIN \"$HKLTMP\" HKLOUT \"$HKL_TRUNCATE\""  
         set status [Execute $cmd $truncate_script $LOG_FILE  program_status report  ]
       }
           
diff --git a/ccp4i/scripts/scalepack2mtz.script b/ccp4i/scripts/scalepack2mtz.script
index 986e06c..a1947a6 100755
--- a/ccp4i/scripts/scalepack2mtz.script
+++ b/ccp4i/scripts/scalepack2mtz.script
@@ -35,7 +35,7 @@
 
     set HKL_TRUNCATE [GetTmpFileName -ext mtz]
     CreateComScript truncate truncate_script
-    set cmd "[BinPath truncate] HKLIN \"$HKLTMP\" HKLOUT \"$HKL_TRUNCATE\""
+    set cmd "[BinPath ftruncate] HKLIN \"$HKLTMP\" HKLOUT \"$HKL_TRUNCATE\""
     set status [Execute $cmd $truncate_script program_status report]
 
     HandleHarvestFile $HARVEST_MODE $HARVEST_PNAME $HARVEST_DNAME truncate
diff --git a/ccp4i/scripts/truncate.script b/ccp4i/scripts/truncate.script
index e0e09ce..2d448a2 100755
--- a/ccp4i/scripts/truncate.script
+++ b/ccp4i/scripts/truncate.script
@@ -45,7 +45,7 @@
 
   CreateComScript truncate truncate_script
 
-  set cmd "[BinPath truncate] HKLIN \"$HKLIN\""
+  set cmd "[BinPath ftruncate] HKLIN \"$HKLIN\""
   if { ![StringSame $INPUT_DATA AMPLITUDES] } {
     append cmd " HKLOUT \"$HKL_TRUNCATE\""
   }
