--- ccp4i/etc/configure.def.dist	2009-01-09 16:45:08.000000000 +0100
+++ ccp4i/etc/configure.def.dist.new	2009-07-07 23:17:26.000000000 +0200
@@ -53,8 +53,8 @@
 RUN_IMOSFLM               _text "ccp4iwish [file join [GetEnvPath CCP4I_TOP] imosflm imosflm.tcl] project [GetCurrentProjectDir [GetCurrentProject]]"
 RUN_CCP4MG                _text                     "ccp4mg"
 RUN_COOT                  _text                     "coot"
-RUN_BAUBLES               _text                     "python [file join [GetEnvPath CCP4] share smartie baubles.py]"
-RUN_DBVIEWER              _text                     "wish [file join [GetEnvPath CCP4] share dbccp4i application viewer.tcl]"
+RUN_BAUBLES               _text                     "python [file join [GetEnvPath CCP4] share ccp4 smartie baubles.py]"
+RUN_DBVIEWER              _text                     "wish [file join [GetEnvPath DBCCP4I_TOP] application viewer.tcl]"
 ENABLE_BUBBLE_HELP        _logical                  1
 DISABLE_TASKS             _logical                  1
 CCP4_SETUP_COMMAND        _text                     setup_ccp4
@@ -62,7 +62,7 @@
 BLT_LIBRARY               _text                     ""
 MENU_LENGTH               _positiveint              25
 HYPERTEXT_VIEWER          _text                     firefox
-START_NETSCAPE	          _text			    netscape
+START_NETSCAPE	          _text			    firefox
 O_MAPMAN                  _text                     mapman
 MAPMAN_MAXSIZE		  _positiveint		    4194304
 QUANTA_MBKALL             _text                     mbkall
