--- a/utilities/CMakeLists.txt
+++ b/utilities/CMakeLists.txt
@@ -110,21 +110,16 @@ MYSQL_ADD_EXECUTABLE(perror
   DEPENDENCIES GenError
   LINK_LIBRARIES mysys
   )
-IF(BUILD_BUNDLED_LZ4)
-  MYSQL_ADD_EXECUTABLE(lz4_decompress
-    lz4_decompress.cc
-    COMPONENT Server
-    LINK_LIBRARIES ${LZ4_LIBRARY} mysys
-    )
-ENDIF()
-
-IF(BUILD_BUNDLED_ZLIB OR NOT OPENSSL_EXECUTABLE_HAS_ZLIB)
-  MYSQL_ADD_EXECUTABLE(zlib_decompress
-    zlib_decompress.cc
-    COMPONENT Server
-    LINK_LIBRARIES ${ZLIB_LIBRARY} mysys
-    )
-ENDIF()
+MYSQL_ADD_EXECUTABLE(lz4_decompress
+  lz4_decompress.cc
+  COMPONENT Server
+  LINK_LIBRARIES ${LZ4_LIBRARY} mysys
+  )
+MYSQL_ADD_EXECUTABLE(zlib_decompress
+  zlib_decompress.cc
+  COMPONENT Server
+  LINK_LIBRARIES ${ZLIB_LIBRARY} mysys
+  )
 
 # All targets below belong to COMPONENT Server and depend on InnoDB.
 IF(WITHOUT_SERVER)
