--- a/utilities/CMakeLists.txt
+++ b/utilities/CMakeLists.txt
@@ -123,14 +123,11 @@ MYSQL_ADD_EXECUTABLE(perror
   DEPENDENCIES GenError
   LINK_LIBRARIES mysys
   )
-IF(BUILD_BUNDLED_LZ4)
-  MYSQL_ADD_EXECUTABLE(lz4_decompress
-    lz4_decompress.cc
-    COMPONENT Server
-    LINK_LIBRARIES ${LZ4_LIBRARY} mysys
-    )
-ENDIF()
-
+MYSQL_ADD_EXECUTABLE(lz4_decompress
+  lz4_decompress.cc
+  COMPONENT Server
+  LINK_LIBRARIES ${LZ4_LIBRARY} mysys
+  )
 MYSQL_ADD_EXECUTABLE(zlib_decompress
   zlib_decompress.cc
   COMPONENT Server
