--- a/configure.ac	2017-08-30 13:46:34.075359703 +0100
+++ b/configure.ac	2017-09-07 21:28:48.110879291 +0100
@@ -19,19 +19,6 @@
 #but libglusterfs fails to build with contrib (Then are not set up that way?)
 #AM_INIT_AUTOMAKE([subdir-objects])
 
-m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES(yes)])
-
-if make --help 2>&1 | grep -q no-print-directory; then
-   AM_MAKEFLAGS="$AM_MAKEFLAGS --no-print-directory";
-fi
-
-if make --help 2>&1 | grep -q quiet; then
-   AM_MAKEFLAGS="$AM_MAKEFLAGS --quiet"
-fi
-
-if libtool --help 2>&1 | grep -q quiet; then
-   AM_LIBTOOLFLAGS="--quiet";
-fi
 
 AC_CONFIG_HEADERS([config.h site.h])
 
