--- configure.in.orig	2009-03-11 08:15:14.000000000 -0500
+++ configure.in	2009-03-11 08:25:36.000000000 -0500
@@ -63,9 +63,14 @@
 AC_SUBST(AVAHI_LIBS)
 
 AC_ARG_ENABLE(oggvorbis,[  --enable-oggvorbis      Enable Ogg/Vorbis support],
+			[ case "${enableval}" in
+			  yes)
                            use_oggvorbis=true;
 #                           LDFLAGS="${LDFLAGS} -logg -lvorbis";
-                           CPPFLAGS="${CPPFLAGS} -DOGGVORBIS")
+                           CPPFLAGS="${CPPFLAGS} -DOGGVORBIS"
+			   esac
+			 ]
+			   )
 
 AM_CONDITIONAL(COND_REND_HOWL, test x$rend_howl = xtrue)
 AM_CONDITIONAL(COND_REND_AVAHI, test x$rend_avahi = xtrue)
