--- gkrellm-2.2.4/src/gui.c.font	2005-05-17 15:37:16.000000000 +0200
+++ gkrellm-2.2.4/src/gui.c	2005-05-17 15:37:46.000000000 +0200
@@ -1575,11 +1575,11 @@
 	if (!_GK.theme_path || !g_file_test(_GK.theme_path, G_FILE_TEST_IS_DIR))
 		gkrellm_dup_string(&_GK.theme_path, "");
 	if (!large_font.string)
-		gkrellm_dup_string(&large_font.string, "Serif 11");
+		gkrellm_dup_string(&large_font.string, "Sans 11");
 	if (!normal_font.string)
-		gkrellm_dup_string(&normal_font.string, "Serif 9");
+		gkrellm_dup_string(&normal_font.string, "Sans 9");
 	if (!small_font.string)
-		gkrellm_dup_string(&small_font.string, "Serif 8");
+		gkrellm_dup_string(&small_font.string, "Sans 8");
 	}
 
 
