diff -urpN expect-5.43.0.orig/work/expect-5.43/exp_glob.c expect-5.43.0/work/expect-5.43/exp_glob.c
--- expect-5.43.0.orig/work/expect-5.43/exp_glob.c	2009-06-07 19:36:31.000000000 +0200
+++ expect-5.43.0/work/expect-5.43/exp_glob.c	2009-06-07 19:39:04.000000000 +0200
@@ -11,6 +11,7 @@ would appreciate credit if this program 
 
 */
 
+#include <string.h>
 #include "expect_cf.h"
 #include "tcl.h"
 #include "exp_int.h"
diff -urpN expect-5.43.0.orig/work/expect-5.43/exp_main_exp.c expect-5.43.0/work/expect-5.43/exp_main_exp.c
--- expect-5.43.0.orig/work/expect-5.43/exp_main_exp.c	2009-06-07 19:36:31.000000000 +0200
+++ expect-5.43.0/work/expect-5.43/exp_main_exp.c	2009-06-07 19:37:39.000000000 +0200
@@ -9,6 +9,7 @@ would appreciate credit if this program 
 
 #include "expect_cf.h"
 #include <stdio.h>
+#include <stdlib.h>
 #include "tcl.h"
 #include "expect_tcl.h"
 
diff -urpN expect-5.43.0.orig/work/expect-5.43/exp_trap.c expect-5.43.0/work/expect-5.43/exp_trap.c
--- expect-5.43.0.orig/work/expect-5.43/exp_trap.c	2009-06-07 19:36:31.000000000 +0200
+++ expect-5.43.0/work/expect-5.43/exp_trap.c	2009-06-07 19:38:51.000000000 +0200
@@ -11,6 +11,7 @@ would appreciate credit if this program 
 #include "expect_cf.h"
 
 #include <stdio.h>
+#include <string.h>
 #include <signal.h>
 #include <sys/types.h>
 
diff -urpN expect-5.43.0.orig/work/expect-5.43/pty_termios.c expect-5.43.0/work/expect-5.43/pty_termios.c
--- expect-5.43.0.orig/work/expect-5.43/pty_termios.c	2009-06-07 19:36:31.000000000 +0200
+++ expect-5.43.0/work/expect-5.43/pty_termios.c	2009-06-07 19:38:19.000000000 +0200
@@ -8,6 +8,7 @@ would appreciate credit if you use this 
 */
 
 #include <stdio.h>
+#include <string.h>
 #include <signal.h>
 
 #if defined(SIGCLD) && !defined(SIGCHLD)
