diff -Naur src/javax/jmi/reflect/RefClass.java src2/javax/jmi/reflect/RefClass.java
--- src/javax/jmi/reflect/RefClass.java	2006-12-09 00:45:59.000000000 +0100
+++ src/javax/jmi/reflect/RefClass.java	2002-05-02 13:44:42.000000000 +0200
@@ -8,6 +8,6 @@
     public Collection refAllOfClass();
     public RefStruct refCreateStruct(RefObject struct, List params);
     public RefStruct refCreateStruct(String structName, List params);
-    public RefEnum refGetEnum(RefObject enum, String name);
+    public RefEnum refGetEnum(RefObject enumObj, String name);
     public RefEnum refGetEnum(String enumName, String name);
 }
\ No newline at end of file
diff -Naur src/javax/jmi/reflect/RefPackage.java src2/javax/jmi/reflect/RefPackage.java
--- src/javax/jmi/reflect/RefPackage.java	2006-12-09 00:46:08.000000000 +0100
+++ src/javax/jmi/reflect/RefPackage.java	2002-05-02 13:44:16.000000000 +0200
@@ -14,7 +14,7 @@
     public Collection refAllAssociations();
     public RefStruct refCreateStruct(RefObject struct, List params);
     public RefStruct refCreateStruct(String structName, List params);
-    public RefEnum refGetEnum(RefObject enum, String name);
+    public RefEnum refGetEnum(RefObject enumObj, String name);
     public RefEnum refGetEnum(String enumName, String name);
     public void refDelete();
 } 
