From aca6355eb291171d652f09a530040d18f29dbd2e Mon Sep 17 00:00:00 2001
From: OBattler <oubattler@gmail.com>
Date: Sat, 27 Jul 2024 16:58:36 +0200
Subject: [PATCH] Rewrote a line in unix.c, fixes #4658.

---
 src/unix/unix.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/unix/unix.c b/src/unix/unix.c
index 1fe477a231..d4d73817f9 100644
--- a/src/unix/unix.c
+++ b/src/unix/unix.c
@@ -640,8 +640,12 @@ ui_msgbox_header(int flags, void *header, void *message)
     SDL_MessageBoxData       msgdata;
     SDL_MessageBoxButtonData msgbtn;
 
-    if (!header)
-        header = (void *) ((flags & MBX_ANSI) ? "86Box" : L"86Box");
+    if (!header) {
+        if (flags & MBX_ANSI)
+            header = (void *) "86Box";
+        else
+            header = (void *) L"86Box";
+    }
 
     msgbtn.buttonid = 1;
     msgbtn.text     = "OK";
