diff --git a/PySide6/CMakeLists.txt b/PySide6/CMakeLists.txt
index e24fdb682..37b7a6c9c 100644
--- a/PySide6/CMakeLists.txt
+++ b/PySide6/CMakeLists.txt
@@ -111,4 +111,4 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/pyside6_global.h
         DESTINATION include/${BINDING_NAME}${pyside6_SUFFIX})
 
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/QtAsyncio"
-        DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
+        DESTINATION "${PYTHON_SITE_PACKAGES}/${BINDING_NAME}${pyside6_SUFFIX}")
