--- src/Makefile.orig	2013-02-26 01:16:14.367046358 +0400
+++ src/Makefile	2013-02-26 01:16:26.054046729 +0400
@@ -28,7 +28,7 @@
 	python ../scripts/get_symbol_versions.py
 	rm -f linking-helper
 	perl -w ../scripts/genconf.pl
-	$(CC) $(CFLAGS) $(SRC) -nostartfiles -shared -fPIC -Wl,-soname,libtrash.so.${MAJOR} \
+	$(CC) $(CFLAGS) $(SRC) -nostartfiles -shared -fPIC $(LDFLAGS) -Wl,-soname,libtrash.so.${MAJOR} \
         -o libtrash.so.${VERSION} -ldl
 
 install-libtrash: 
