--- scripts/cs_helpers.pyin.orig	2004-11-28 14:35:23.000000000 +0100
+++ scripts/cs_helpers.pyin	2005-06-19 20:55:49.000000000 +0200
@@ -10,6 +10,7 @@
 #  (at your option) any later version.
 
 import os
+import time
 
 # the name of the config file read by the scripts; see there for options and
 # descriptions
@@ -170,6 +171,7 @@
 	msg['Subject']=mail_subject
 	msg['From']=mail_from
 	msg['To']=mail_to
+	msg['Date']=time.strftime('%a, %d %b %Y %H:%M:%S %z')
 
 	msg.preamble = 'This is a Multipart-MIME-message. Please use a capable mailer.\n'
 	msg.epilogue = '' # To guarantee the message ends with a newline
@@ -267,6 +269,7 @@
 	msg['Subject'] = mail_subject
 	msg['From'] = mail_from
 	msg['To'] = mail_to
+	msg['Date'] = time.strftime('%a, %d %b %Y %H:%M:%S %z')
 
 	try:
 		server = smtplib.SMTP('localhost')
