diff -ur capisuite-0.4.5.orig/scripts/cs_helpers.pyin capisuite-0.4.5/scripts/cs_helpers.pyin
--- capisuite-0.4.5.orig/scripts/cs_helpers.pyin	2004-11-28 15:35:23.000000000 +0200
+++ capisuite-0.4.5/scripts/cs_helpers.pyin	2009-05-12 20:58:34.000000000 +0300
@@ -227,7 +227,7 @@
 		elif (mail_type=="la"): # voice file
 			# la -> wav
 			# don't use stdout as sox needs a file to be able to seek in it otherwise the header will be incomplete
-			ret = os.spawnlp(os.P_WAIT,"sox","sox",attachment,"-w",basename+"wav")
+			ret = os.spawnlp(os.P_WAIT,"sox","sox",attachment,"-2",basename+"wav")
 			if (ret or not os.access(basename+"wav",os.R_OK)):
 				raise "conv-error","Error while calling sox. Not installed?"
 			filepart = email.MIMEAudio.MIMEAudio(open(basename+"wav").read(),"x-wav",email.Encoders.encode_base64,name=os.path.basename(basename)+"wav")
