--- a/telnetd/Makefile	2004-09-29 22:16:50.175839840 -0500
+++ b/telnetd/Makefile	2004-09-29 22:09:30.439689896 -0500
@@ -10,7 +10,7 @@
 
 CFLAGS += '-DISSUE_FILE="/etc/issue.net"' -DPARANOID_TTYS \
 	   -DNO_REVOKE -DKLUDGELINEMODE -DDIAGNOSTICS \
-	   -DLOGIN_WRAPPER=\"/usr/lib/telnetlogin\"
+	   -DLOGIN_WRAPPER=\"/usr/sbin/telnetlogin\" -D_GNU_SOURCE
 # LIBS += $(LIBTERMCAP)
 
 OBJS = telnetd.o state.o termstat.o slc.o sys_term.o utility.o \
--- a/telnet/Makefile    2004-11-01 15:23:29 -0500
+++ b/telnet/Makefile     2004-11-01 16:05:05 -0500
@@ -6,7 +6,7 @@
 #CXXFLAGS:=$(patsubst -O2, -g, $(CXXFLAGS))
 
 # -DAUTHENTICATE
-CXXFLAGS += -DUSE_TERMIO -DKLUDGELINEMODE
+CXXFLAGS += -DUSE_TERMIO -DKLUDGELINEMODE -D_GNU_SOURCE
 LIBS = $(LIBTERMCAP)
 
 SRCS = commands.cc main.cc network.cc ring.cc sys_bsd.cc telnet.cc \

