--- gliv-1.9.7/Makefile.am.orig
+++ gliv-1.9.7/Makefile.am
@@ -31,9 +31,9 @@
 	@echo
 	@if desktop-file-install --help > /dev/null 2>&1; then              \
 	 echo "Installing the GLiv desktop file with desktop-file-install"; \
-	 mkdir -p $(datadir)/{pixmaps,applications} || true;                \
-	 cp gliv.png $(datadir)/pixmaps/gliv.png || true;                   \
-	 desktop-file-install --vendor=gnome --dir=$(datadir)/applications  \
+	 mkdir -p $(DESTDIR)$(datadir)/{pixmaps,applications} || true;                \
+	 cp gliv.png $(DESTDIR)$(datadir)/pixmaps/gliv.png || true;                   \
+	 desktop-file-install --vendor=gnome --dir=$(DESTDIR)$(datadir)/applications  \
                                              gliv.desktop || true;          \
 	else                                                                \
 	 echo "No desktop-file-install(1), skipping gliv.desktop";          \
