Index: liborigin.pro
===================================================================
--- liborigin.pro	(revision 1751)
+++ liborigin.pro	(working copy)
@@ -17,10 +17,20 @@
 HEADERS += OriginFile.h
 HEADERS += OriginParser.h
 HEADERS += OriginDefaultParser.h
-HEADERS += Origin750Parser.h
+SOURCES += Origin600Parser.h
+SOURCES += Origin610Parser.h
+SOURCES += Origin700Parser.h
+SOURCES += Origin750Parser.h
+SOURCES += Origin800Parser.h
+SOURCES += Origin810Parser.h
 
 SOURCES += OriginFile.cpp
 SOURCES += OriginParser.cpp
 SOURCES += OriginDefaultParser.cpp
+SOURCES += Origin600Parser.cpp
+SOURCES += Origin610Parser.cpp
+SOURCES += Origin700Parser.cpp
 SOURCES += Origin750Parser.cpp
+SOURCES += Origin800Parser.cpp
+SOURCES += Origin810Parser.cpp
 
