From: Bernd Waibel <waebbl-gentoo@posteo.net>
Date: Tue, 22 Jun 2021 11:46:30 +0200
Subject: [PATCH 1/2] install imathnumpy.so

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>
---
 PyImathNumpy/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/PyImathNumpy/CMakeLists.txt b/PyImathNumpy/CMakeLists.txt
index 7a70f06..2822127 100644
--- a/PyImathNumpy/CMakeLists.txt
+++ b/PyImathNumpy/CMakeLists.txt
@@ -49,4 +49,5 @@ if(TARGET Python3::Python AND
     LIBRARY_OUTPUT_NAME "imathnumpy"
     DEBUG_POSTFIX ""
   )
+  install(TARGETS imathnumpy_python3 DESTINATION ${PyIlmBase_Python3_SITEARCH_REL})
 endif()
-- 
2.32.0

