--- src/formisc.c.orig	2006-01-22 00:35:30.000000000 -0800
+++ src/formisc.c	2006-01-22 00:35:55.000000000 -0800
@@ -183,7 +183,7 @@
 		 retval=excode;
 	    }					       /* reap some children */
 	while(childlimit&&children>=childlimit||(child=fork())==-1&&children)
-	   for(--children;(excode=waitfor((pid_t)0))!=NO_PROCESS;)
+	   for(;(excode=waitfor((pid_t)0))!=NO_PROCESS;)
 	    { if(excode!=EXIT_SUCCESS)
 		 retval=excode;
 	      if(--children<=maxchild)
