diff -NrU5 procmail-3.22.orig/src/fields.c procmail-3.22/src/fields.c
--- procmail-3.22.orig/src/fields.c	2009-05-21 18:21:25.000000000 +0200
+++ procmail-3.22/src/fields.c	2009-05-21 18:22:14.000000000 +0200
@@ -108,20 +108,20 @@
 	extractfield(p);
 }
 		    /* try and append one valid field to rdheader from stdin */
 int readhead P((void))
 { int idlen;
-  getline();
+  procmail_getline();
   if((idlen=breakfield(buf,buffilled))<=0) /* not the start of a valid field */
      return 0;
   if(idlen==STRLEN(FROM)&&eqFrom_(buf))			/* it's a From_ line */
    { if(rdheader)
 	return 0;			       /* the From_ line was a fake! */
-     for(;buflast=='>';getline());	    /* gather continued >From_ lines */
+     for(;buflast=='>';procmail_getline());	    /* gather continued >From_ lines */
    }
   else
-     for(;;getline())		      /* get the rest of the continued field */
+     for(;;procmail_getline())		      /* get the rest of the continued field */
       { switch(buflast)			     /* will this line be continued? */
 	 { case ' ':case '\t':				  /* yep, it sure is */
 	      continue;
 	 }
 	break;
diff -NrU5 procmail-3.22.orig/src/formail.c procmail-3.22/src/formail.c
--- procmail-3.22.orig/src/formail.c	2009-05-21 18:21:25.000000000 +0200
+++ procmail-3.22/src/formail.c	2009-05-21 18:22:14.000000000 +0200
@@ -817,11 +817,11 @@
 #ifdef MAILBOX_SEPARATOR
      if(!strncmp(emboxsep,buf,STRLEN(emboxsep)))	     /* end of mail? */
       { if(split)		       /* gobble up the next start separator */
 	 { buffilled=0;
 #ifdef sMAILBOX_SEPARATOR
-	   getline();buffilled=0;		 /* but only if it's defined */
+	   procmail_getline();buffilled=0;		 /* but only if it's defined */
 #endif
 	   if(buflast!=EOF)					   /* if any */
 	      goto splitit;
 	   break;
 	 }
diff -NrU5 procmail-3.22.orig/src/formisc.c procmail-3.22/src/formisc.c
--- procmail-3.22.orig/src/formisc.c	2009-05-21 18:21:25.000000000 +0200
+++ procmail-3.22/src/formisc.c	2009-05-21 18:22:14.000000000 +0200
@@ -113,11 +113,11 @@
 { if(buffilled==buflen)
      buf=realloc(buf,buflen+=Bsize);
   buf[buffilled++]=c;
 }
 
-int getline P((void))			   /* read a newline-terminated line */
+int procmail_getline P((void))			   /* read a newline-terminated line */
 { if(buflast==EOF)			 /* at the end of our Latin already? */
    { loadchar('\n');					  /* fake empty line */
      return EOF;					  /* spread the word */
    }
   loadchar(buflast);			    /* load leftover into the buffer */
diff -NrU5 procmail-3.22.orig/src/formisc.h procmail-3.22/src/formisc.h
--- procmail-3.22.orig/src/formisc.h	2009-05-21 18:21:25.000000000 +0200
+++ procmail-3.22/src/formisc.h	2009-05-21 18:22:14.000000000 +0200
@@ -15,6 +15,6 @@
  closemine P((void)),
  opensink P((void));
 char*
  skipwords P((char*start));
 int
- getline P((void));
+ procmail_getline P((void));
