--- a/configure.ac	2015-12-17 11:09:56.807893315 -0500
+++ b/configure.ac	2015-12-17 11:11:06.697570856 -0500
@@ -234,10 +234,14 @@
 
 # Unless the user specify --without-curses, check for curses.
 if test "x$with_curses" != "xno"; then
+  PKG_CHECK_MODULES([NCURSES],[ncurses],[
+    AC_DEFINE(HAVE_LIBCURSES, 1, [Define if you have a curses library])
+    GRUB_LIBS="$GRUB_LIBS $NCURSES_LIBS"
+  ],[
   AC_CHECK_LIB(ncurses, wgetch, [GRUB_LIBS="$GRUB_LIBS -lncurses"
   AC_DEFINE(HAVE_LIBCURSES, 1, [Define if you have a curses library])],
     [AC_CHECK_LIB(curses, wgetch, [GRUB_LIBS="$GRUB_LIBS -lcurses"
-       AC_DEFINE(HAVE_LIBCURSES, 1, [Define if you have a curses library])])])
+       AC_DEFINE(HAVE_LIBCURSES, 1, [Define if you have a curses library])])])])
 fi
 
 AC_SUBST(GRUB_LIBS)
