diff --git a/agent/utils/scim-bridge-agent-panel-client.cpp b/agent/utils/scim-bridge-agent-panel-client.cpp
index 07cb14b..62fd3b3 100644
--- a/agent/utils/scim-bridge-agent-panel-client.cpp
+++ b/agent/utils/scim-bridge-agent-panel-client.cpp
@@ -24,6 +24,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <unistd.h>
 
 #define Uses_SCIM_CONFIG_PATH
 
diff --git a/agent/scim-bridge-agent-signal-listener.cpp b/agent/scim-bridge-agent-signal-listener.cpp
index 433dc15..c9eff47 100644
--- a/agent/scim-bridge-agent-signal-listener.cpp
+++ b/agent/scim-bridge-agent-signal-listener.cpp
@@ -22,6 +22,7 @@
 #include <signal.h>
 #include <stdio.h>
 #include <string.h>
+#include <unistd.h>
 
 #include <sys/socket.h>
 #include <sys/types.h>
