diff -ruN dhex_0.63.orig/Makefile dhex_0.63/Makefile
--- dhex_0.63.orig/Makefile	2011-01-29 00:19:37.000000000 +0100
+++ dhex_0.63/Makefile	2011-01-30 17:13:51.000000000 +0100
@@ -1,11 +1,4 @@
-CC=		gcc
-LDFLAGS=	-L/usr/lib	-L/usr/local/lib  	-L/usr/lib/ncurses	-L/usr/local/lib/ncurses
-CPPFLAGS=	-I/usr/include	-I/usr/local/include	-I/usr/include/ncurses	-I/usr/local/include/ncurses
-CFLAGS=		-O3 -Wall -std=c99 
-#CFLAGS+= -ffunction-sections -fdata-sections
-#LDFLAGS+= --gc-sections 
 LIBS=		-lncurses
-DESTDIR=	/usr/local/
 
 OFILES=buffers.o input.o output.o machine_type.o main.o menu.o ui.o hexcalc.o search.o gpl.o configfile.o markers.o
 
