--- a/setup.py.old	2020-06-06 21:50:25.432740405 +0200
+++ b/setup.py	2020-06-06 21:50:43.109010442 +0200
@@ -35,7 +35,7 @@
                    'Topic :: Scientific/Engineering :: Chemistry',
                    'Topic :: Software Development :: Libraries :: Python Modules',
                    ],
-      packages=find_packages(exclude=['tests','scripts','extras','doc/examples']),
+      packages=find_packages(exclude=['tests','tests.*','scripts','extras','doc/examples']),
       scripts = [
                  'scripts/gw-join_parts.py',
                  'scripts/gw-merge_topologies.py',
