diff --git a/setup.py b/setup.py
index 92d059ff..cbc5b881 100644
--- a/setup.py
+++ b/setup.py
@@ -67,7 +67,6 @@ else:
 
 
 needs_pytest = {"pytest", "test", "ptr"}.intersection(sys.argv)
-pytest_runner = ["pytest-runner"] if needs_pytest else []
 
 
 def ensure_scripts(linux_scripts):
@@ -151,7 +150,6 @@ def install(**kwargs):
         extras_require=extras_require,
         test_suite="test",
         python_requires=">=3.5.*",
-        setup_requires=pytest_runner,
         tests_require=["pytest"],
         **kwargs
     )
