 CMakeLists.txt       | 12 ++----------
 tests/CMakeLists.txt |  5 +----
 2 files changed, 3 insertions(+), 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a76947..a58510b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,13 +7,9 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release")
 endif()
 
-set(ANTLR_DIR libs/antlr-2.7.7)
 set(QHEXEDIT_DIR libs/qhexedit)
-set(QCUSTOMPLOT_DIR libs/qcustomplot-source)
 
-add_subdirectory(${ANTLR_DIR})
 add_subdirectory(${QHEXEDIT_DIR})
-add_subdirectory(${QCUSTOMPLOT_DIR})
 
 if(USE_QT5)
 	find_package(Qt5Widgets REQUIRED)
@@ -191,9 +187,7 @@ endif(EXTRAPATH)
 
 include_directories(
 		"${CMAKE_CURRENT_BINARY_DIR}"
-		${ANTLR_DIR}
 		${QHEXEDIT_DIR}
-		${QCUSTOMPLOT_DIR}
 		${ADDITIONAL_INCLUDE_PATHS}
 		src)
 
@@ -210,12 +204,10 @@ if(USE_QT5)
     qt5_use_modules(${PROJECT_NAME} Gui Widgets Network Test PrintSupport)
     set(QT_LIBRARIES "")
 endif()
-add_dependencies(${PROJECT_NAME} antlr qhexedit qcustomplot)
+add_dependencies(${PROJECT_NAME} qhexedit)
 
 link_directories(
-		"${CMAKE_CURRENT_BINARY_DIR}/${ANTLR_DIR}"
-		"${CMAKE_CURRENT_BINARY_DIR}/${QHEXEDIT_DIR}"
-		"${CMAKE_CURRENT_BINARY_DIR}/${QCUSTOMPLOT_DIR}")
+		"${CMAKE_CURRENT_BINARY_DIR}/${QHEXEDIT_DIR}")
 
 target_link_libraries(${PROJECT_NAME}
 		antlr
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 75009dc..6fd7f48 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -7,9 +7,6 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release")
 endif()
 
-set(ANTLR_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../libs/antlr-2.7.7")
-add_subdirectory("${ANTLR_DIR}" "${CMAKE_CURRENT_BINARY_DIR}/antlr")
-
 if(USE_QT5)
     find_package(Qt5Widgets REQUIRED)
     set(CMAKE_AUTOMOC ON)
@@ -56,7 +53,7 @@ if(NOT USE_QT5)
     QT4_WRAP_CPP(SQLB_MOC ${SQLB_MOC_HDR})
 endif()
 
-include_directories("${ANTLR_DIR}" ../src)
+include_directories(../src)
 
 add_executable(${PROJECT_NAME} ${SQLB_MOC} ${SQLB_HDR} ${SQLB_SRC})
 
