From 4c703e395b55e939f3ac1e4a4bf5cd36aa5b0abc Mon Sep 17 00:00:00 2001
From: Weng Xuetian <wengxt@gmail.com>
Date: Sat, 26 Jan 2013 14:40:00 -0500
Subject: [PATCH] [fcitx] fix compatible with gcc 4.6

---
 src/lib/fcitx/module.h |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/lib/fcitx/module.h b/src/lib/fcitx/module.h
index 28de64f..bae6601 100644
--- a/src/lib/fcitx/module.h
+++ b/src/lib/fcitx/module.h
@@ -109,7 +109,7 @@
     void* FcitxModuleInvokeFunction(FcitxAddon* addon, int functionId, FcitxModuleFunctionArg args);
 #define FcitxModuleInvokeVaArgs(addon, functionId, ARGV...)             \
     (FcitxModuleInvokeFunction(addon, functionId,                       \
-                               (FcitxModuleFunctionArg){ .args = {ARGV} }))
+                               (FcitxModuleFunctionArg){ {ARGV} }))
 
     /**
      * invoke inter module function with addon name, returns NULL when fails (the function itself can also return NULL)
@@ -133,7 +133,7 @@
 #define InvokeVaArgs(INST, MODULE, FUNC, ARGV...)                       \
     ((MODULE##_##FUNC##_RETURNTYPE)FcitxModuleInvokeFunctionByName(     \
         INST, MODULE##_NAME, MODULE##_##FUNC,                           \
-        (FcitxModuleFunctionArg){ .args = {ARGV} }))
+        (FcitxModuleFunctionArg){ {ARGV} }))
 
 /** add a function to a addon */
 #define AddFunction(ADDON, Realname)                                    \
@@ -215,7 +215,7 @@
     }
 
 #define FCITX_DEF_MODULE_ARGS(var, ARGV...)             \
-    FcitxModuleFunctionArg var = { .args = {ARGV} }
+    FcitxModuleFunctionArg var = { {ARGV} }
     /* void *__##var##_array[] = {ARGV};                                   \ */
     /* size_t __##var##_length = sizeof(__##var##_array) / sizeof(void*);  \ */
     /* FcitxModuleFunctionArg var[] = { { .n = __##var##_length,           \ */
-- 
1.7.10

