--- a/configure.ac
+++ b/configure.ac
@@ -641,6 +641,8 @@
 
 AC_CONFIG_HEADERS(include/wvautoconf.h)
 
+AC_CONFIG_SUBDIRS(argp)
+
 AC_OUTPUT
 
 # Now convert PACKAGE_* macros into WVPACKAGE_* in include/wvautoconf.h
--- a/argp/configure.ac
+++ b/argp/configure.ac
@@ -4,7 +4,7 @@
 AC_PREREQ(2.54)
 AC_INIT(argp-ba.c)
 AM_INIT_AUTOMAKE(argp, standalone-1.3)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 # GNU libc defaults to supplying the ISO C library functions only. The
 # _GNU_SOURCE define enables these extensions, in particular we want
@@ -18,7 +18,7 @@
 AC_PROG_RANLIB
 AC_PROGRAM_CHECK(AR,ar,ar,:)
 AC_PROGRAM_CHECK(AR,gar,gar,:)
-AM_PROG_CC_STDC
+AC_PROG_CC
 
 AC_SUBST(AR)
 
