--- ccp4-6.1.1/x-windows/Makefile.in	2008-08-11 19:20:01.000000000 +0200
+++ ccp4-6.1.1/x-windows/Makefile.in.new	2009-02-27 14:32:25.000000000 +0100
@@ -35,7 +35,7 @@
 #
 # all
 #
-all : xdlview libjwc_c libjwc_f rotgen_ hklview xjiffy rasmol2 ipmosflm
+all : rotgen_ hklview xjiffy
 #
 # xdl_view
 #
@@ -157,11 +157,10 @@
 #
 install_obj = hklview rotgen ipmosflm
 install: all
+	mkdir -p $(bindir)
 	$(INSTALL_PROGRAM) $(srcdir)/hklview $(bindir)
 	cd $(srcdir)/XCCPJIFFY ; $(MAKE) bindir=$(bindir) libdir=$(libdir) install
-	$(INSTALL_PROGRAM) $(rasmol_dir)/rasmol $(bindir)
 	cd $(rotgen_dir); $(MAKE) install
-	$(INSTALL_PROGRAM) $(mosflm_dir)/bin/ipmosflm $(bindir)
 #
 # clean
 #
