--- src/pisa/Makefile.am	2007-07-10 10:56:05.000000000 +0200
+++ src/pisa/Makefile.am.new	2009-06-11 21:16:51.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = molref pisastore sbase
+SUBDIRS = pisastore sbase
 
 pkgdatadir = $(datadir)/pisa
 
--- src/pisa/Makefile.in	2009-06-12 21:12:08.000000000 +0200
+++ src/pisa/Makefile.in.new	2009-06-12 21:41:52.000000000 +0200
@@ -82,7 +82,7 @@
 am__quote = @am__quote@
 install_sh = @install_sh@
 pisa_LDFLAGS = @pisa_LDFLAGS@
-SUBDIRS = molref pisastore sbase
+SUBDIRS = pisastore sbase
 
 dist_pkgdata_DATA = pisa.cfg
 
