diff --git a/inc/HTTPAuthSession.php b/inc/HTTPAuthSession.php
index e3b208c6..08663286 100644
--- a/inc/HTTPAuthSession.php
+++ b/inc/HTTPAuthSession.php
@@ -60,7 +60,7 @@ class HTTPAuthSession {
   /**
   * The constructor, which just calls the type supplied or configured
   */
-  function HTTPAuthSession() {
+  function __construct() {
     global $c;
 
     if ( ! empty($_SERVER['PHP_AUTH_DIGEST'])) {
diff --git a/inc/PublicSession.php b/inc/PublicSession.php
index 28e06173..e779a9df 100644
--- a/inc/PublicSession.php
+++ b/inc/PublicSession.php
@@ -59,7 +59,7 @@ class PublicSession {
   /**
   * The constructor, which just calls the actual type configured
   */
-  function PublicSession() {
+  function __construct() {
     global $c;
 
     $principal = new Principal('username','unauthenticated');
diff --git a/inc/caldav-client.php b/inc/caldav-client.php
index 4c1bf67f..b32c5c91 100644
--- a/inc/caldav-client.php
+++ b/inc/caldav-client.php
@@ -48,7 +48,7 @@ class CalDAVClient {
   * @param string $pass      The password for that user
   * @param string $calendar  The name of the calendar (not currently used)
   */
-  function CalDAVClient( $base_url, $user, $pass, $calendar = '' ) {
+  function __construct( $base_url, $user, $pass, $calendar = '' ) {
     $this->user = $user;
     $this->pass = $pass;
     $this->calendar = $calendar;
