diff --git a/setup.py b/setup.py
index c00bdc0..8493d09 100755
--- a/setup.py
+++ b/setup.py
@@ -56,6 +56,7 @@ else:
 
 # the application name
 AppName = "LinkChecker"
+AppVersion = "LINKCHECKER_VERSION"
 Description = "check links in web documents or full websites"
 
 RELEASE_DATE_FILE = "_release_date"
@@ -340,10 +341,7 @@ if os.name == "posix":
 
 setup(
     name=AppName,
-    use_scm_version={
-        "local_scheme": "node-and-timestamp",
-        "version_scheme": "post-release",
-    },
+    version=AppVersion,
     description=Description,
     keywords="link,url,site,checking,crawling,verification,validation",
     author=myname,
@@ -383,7 +381,6 @@ setup(
     options={},
     # Requirements, usable with setuptools or the new Python packaging module.
     python_requires=">= 3.6",
-    setup_requires=["setuptools_scm"],
     install_requires=[
         "importlib_metadata;python_version<'3.8'",
         "requests >= 2.4",
