--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -443,7 +443,7 @@
     if ("${BUILD_TYPE}" STREQUAL "release")
         add_definitions(-DNDEBUG)
         # security compiler flags
-        add_definitions(-O2 -D_FORTIFY_SOURCE=2 -Wformat)
+        add_definitions(-Wformat)
     elseif ("${BUILD_TYPE}" STREQUAL "release-internal" OR
             "${BUILD_TYPE}" STREQUAL "releaseinternal")
             add_definitions(-D_RELEASE_INTERNAL)
@@ -467,12 +467,7 @@
         "${BUILD_TYPE}" STREQUAL "release-internal" OR
         "${BUILD_TYPE}" STREQUAL "releaseinternal")
         add_definitions(-Werror=format-security) # treat format string security warnings as errors
-        add_definitions(-Werror)                 # treat warning as errors
         add_definitions(-fvisibility=hidden)     # to disable exporting internal symbols (by default gcc exports all symbols)
-        add_definitions(-flto)                   # enables Link-Time Optimization (LTO)
-        add_definitions(-fPIC)                   # generates position-independent code during the compilation phase
-        add_definitions(-fPIE)                   # generates position-independent executables during the compilation phase
-        add_definitions(-pie)                    # produces position-independent executables during the linking phase
     elseif ("${BUILD_TYPE}" STREQUAL "debug")
         add_definitions(-g)
     endif ()
