diff --git a/plugins/RSS/test.py b/plugins/RSS/test.py
index eab10c61b..a9e3c07c6 100644
--- a/plugins/RSS/test.py
+++ b/plugins/RSS/test.py
@@ -102,6 +102,7 @@ def testCantAddDuplicatedFeed(self):
 
     @mock_urllib
     def testRemoveAliasedFeed(self, mock):
+        mock._data = xkcd_new
         try:
             self.assertNotError('rss announce add http://xkcd.com/rss.xml')
             self.assertNotError('rss add xkcd http://xkcd.com/rss.xml')
