diff --git a/Makefile b/Makefile
index 252e6f77..6955a258 100644
--- a/Makefile
+++ b/Makefile
@@ -49,9 +49,9 @@ CXX_FOR_BUILD?=$(CXX)
 # compiler and linker flags
 DEFINES=-DLOCALEDIR='"$(localedir)"'
 
-WARNFLAGS=-Werror -Wall -Wextra -Wunreachable-code
+WARNFLAGS=-Wall -Wextra -Wunreachable-code
 INCLUDES=-Iinclude -Istfl -Ifilter -I. -Irss -I$(relative_cargo_target_dir)/cxxbridge/
-BARE_CXXFLAGS=-std=c++11 -O2 -ggdb $(INCLUDES)
+BARE_CXXFLAGS=-std=c++11 $(INCLUDES)
 LDFLAGS+=-L.
 
 # Constants
