diff --git a/./dahdi-tools-3.1.0/xpp/hexfile.h.o b/./dahdi-tools-3.1.0/xpp/hexfile.h
index 33ff50d..3d03f98 100644
--- a/xpp/hexfile.h
+++ b/xpp/hexfile.h
@@ -70,7 +70,9 @@ struct hexdata {
 };
 
 
-__BEGIN_DECLS
+#ifdef __cpluplus
+extern "C" {
+#endif
 
 typedef void (*parse_hexfile_report_func_t)(int level, const char *msg, ...)
 #ifdef	__GNUC__
@@ -86,6 +88,9 @@ int dump_hexfile2(struct hexdata *hexdata, const char *outfile, uint8_t maxwidth
 void dump_binary(struct hexdata *hexdata, const char *outfile);
 void gen_hexline(const uint8_t *data, uint16_t addr, size_t len, FILE *output);
 int bsd_checksum(struct hexdata *hexdata);
-__END_DECLS
+
+#ifdef __cpluplus
+}
+#endif
 
 #endif
