diff --git a/doc/conf.py b/doc/conf.py
index 017ce210e..5a397082b 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -170,10 +170,10 @@ source_encoding = "utf-8"
 master_doc = 'contents'
 
 # General substitutions.
-from matplotlib.compat.subprocess import check_output
-SHA = check_output(['git', 'describe', '--dirty']).decode('utf-8').strip()
+#from matplotlib.compat.subprocess import check_output
+#SHA = check_output(['git', 'describe', '--dirty']).decode('utf-8').strip()
 
-html_context = {'sha': SHA}
+#html_context = {'sha': SHA}
 
 project = 'Matplotlib'
 copyright = ('2002 - 2012 John Hunter, Darren Dale, Eric Firing, '
