https://github.com/Tatsh/bpmdetect/pull/4.patch
--- a/src/trackoggvorbis.cpp
+++ b/src/trackoggvorbis.cpp
@@ -223,7 +223,7 @@
     if (tag == NULL) {
         return;
     }
-    tag->removeField("TBPM");
+    tag->removeFields("TBPM");
     f.save();
     //open();
 #endif
--- a/src/trackflac.cpp
+++ b/src/trackflac.cpp
@@ -239,7 +239,7 @@
     TagLib::FLAC::File f ( fname.c_str(), false );
     TagLib::Ogg::XiphComment* xiph = f.xiphComment (true);
     if (xiph != NULL) {
-        xiph->removeField ("TBPM");
+        xiph->removeFields ("TBPM");
     }
 
     TagLib::ID3v2::Tag* tag = f.ID3v2Tag (true);
