Index: test/json.test
==================================================================
--- test/json.test
+++ test/json.test
@@ -107,11 +107,11 @@
     set request [subst {POST $url HTTP/1.0\r
 Host: localhost\r
 User-Agent: Fossil-Test\r
 Cookie: $cookie\r
 Content-Type: application/json
-Content-Length $len
+Content-Length: $len
 \r
 $data}]
   } else {
     set request [subst {GET $url HTTP/1.0\r
 Host: localhost\r

