--- gvm-libs-9.0.3/CMakeLists.txt	2019-02-04 14:50:41.230530261 +0300
+++ gvm-libs-9.0.3/CMakeLists.txt	2019-02-04 14:55:08.649433070 +0300
@@ -162,9 +162,9 @@
   set (DATADIR "${CMAKE_INSTALL_PREFIX}/share")
 endif (NOT DATADIR)
 
-if (NOT OPENVAS_PID_DIR)
-  set (OPENVAS_PID_DIR "${LOCALSTATEDIR}/run")
-endif (NOT OPENVAS_PID_DIR)
+#if (NOT OPENVAS_PID_DIR)
+#  set (OPENVAS_PID_DIR "${LOCALSTATEDIR}/run")
+#endif (NOT OPENVAS_PID_DIR)
 
 #if (NOT SYSCONFDIR)
 #  set (SYSCONFDIR "${CMAKE_INSTALL_PREFIX}/etc")
--- gvm-libs-9.0.3/base/pidfile.c	2018-08-29 19:01:12.000000000 +0300
+++ gvm-libs-9.0.3/base/pidfile.c	2019-02-04 15:08:04.600958560 +0300
@@ -59,7 +59,7 @@
 pidfile_create (gchar * daemon_name)
 {
   gchar *name_pid = g_strconcat (daemon_name, ".pid", NULL);
-  gchar *pidfile_name = g_build_filename (OPENVAS_PID_DIR, name_pid, NULL);
+  gchar *pidfile_name = g_build_filename (name_pid, NULL);
   FILE *pidfile = g_fopen (pidfile_name, "w");
 
   g_free (name_pid);
@@ -88,7 +88,7 @@
 pidfile_remove (gchar * daemon_name)
 {
   gchar *name_pid = g_strconcat (daemon_name, ".pid", NULL);
-  gchar *pidfile_name = g_build_filename (OPENVAS_PID_DIR, name_pid, NULL);
+  gchar *pidfile_name = g_build_filename (name_pid, NULL);
   gchar *pidfile_contents;
 
   g_free (name_pid);
--- gvm-libs-9.0.3/CMakeLists.txt	2019-02-04 17:39:39.776282496 +0300
+++ gvm-libs-9.0.3/CMakeLists.txt	2019-02-04 17:39:23.160269081 +0300
@@ -329,7 +329,6 @@
 
 install (DIRECTORY DESTINATION ${OPENVAS_LOG_DIR})
 install (DIRECTORY DESTINATION ${OPENVAS_DATA_DIR})
-install (DIRECTORY DESTINATION ${OPENVAS_PID_DIR})
 install (DIRECTORY DESTINATION ${OPENVAS_SYSCONF_DIR})
 install (DIRECTORY
   DESTINATION ${OPENVAS_SYSCONF_DIR}/gnupg
