--- gvm-libs-9.0.3/CMakeLists.txt	2019-02-04 21:42:27.367627369 +0300
+++ gvm-libs-9.0.3/CMakeLists.txt	2019-02-04 21:45:48.961135211 +0300
@@ -80,7 +80,7 @@
 # TODO: Check pkg-config (maybe with code like in gsa/CMakeLists.txt).
 
 ## CPack configuration
-
+set (CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
 set (CPACK_CMAKE_GENERATOR "Unix Makefiles")
 set (CPACK_GENERATOR "TGZ")
 set (CPACK_INSTALL_CMAKE_PROJECTS ".;openvas-libraries;ALL;/")
