--- gvm-libs-9.0.3/CMakeLists.txt	2018-08-29 19:01:12.000000000 +0300
+++ gvm-libs-9.0.3/CMakeLists.txt	2019-02-01 20:58:45.946113012 +0300
@@ -238,7 +238,7 @@
     OUTPUT_STRIP_TRAILING_WHITESPACE)
 
   message (STATUS "Looking for netsnmp...")
-  find_library (SNMP snmp)
+  find_library (SNMP netsnmp)
   message (STATUS "Looking for netsnmp... ${SNMP}")
   if (SNMP)
     execute_process (COMMAND net-snmp-config --libs
