diff -r -u collectd-4.10.2.orig/configure.in collectd-4.10.2/configure.in
--- collectd-4.10.2.orig/configure.in	2010-11-28 03:27:47.000000000 +0100
+++ collectd-4.10.2/configure.in	2010-11-28 03:28:23.000000000 +0100
@@ -2446,14 +2446,14 @@
 ])
 if test "x$ORACLE_HOME" != "x"
 then
-	with_oracle_cppflags="-I$ORACLE_HOME/rdbms/public"
+	with_oracle_cppflags="-I$ORACLE_HOME/rdbms/public -locci"
 
 	if test -e "$ORACLE_HOME/lib/ldflags"
 	then
 		with_oracle_libs=`cat "$ORACLE_HOME/lib/ldflags"`
 	fi
 	#with_oracle_libs="-L$ORACLE_HOME/lib $with_oracle_libs -lclntsh"
-	with_oracle_libs="-L$ORACLE_HOME/lib -lclntsh"
+	with_oracle_libs="-L$ORACLE_HOME/lib -lclntsh -locci"
 fi
 if test "x$with_oracle" = "xyes"
 then
