--- retroshare-gui/src/retroshare-gui.pro.orig	2015-04-16 12:35:21.878472205 +0300
+++ retroshare-gui/src/retroshare-gui.pro	2015-04-16 12:35:46.330471776 +0300
@@ -84,21 +84,7 @@
 
 	SQLCIPHER_OK = $$system(pkg-config --exists sqlcipher && echo yes)
 	isEmpty(SQLCIPHER_OK) {
-# We need a explicit path here, to force using the home version of sqlite3 that really encrypts the database.
-
-		exists(../../../lib/sqlcipher/.libs/libsqlcipher.a) {
-
-			LIBS += ../../../lib/sqlcipher/.libs/libsqlcipher.a
-			DEPENDPATH += ../../../lib/sqlcipher/src/
-			INCLUDEPATH += ../../../lib/sqlcipher/src/
-			DEPENDPATH += ../../../lib/sqlcipher/tsrc/
-			INCLUDEPATH += ../../../lib/sqlcipher/tsrc/
-		} else {
-			message(libsqlcipher.a not found. Compilation will not use SQLCIPHER. Database will be unencrypted.)
-			DEFINES *= NO_SQLCIPHER
-			LIBS *= -lsqlite3
-		}
-
+		error(libsqlcipher.a not found)
 	} else {
 		LIBS += -lsqlcipher
 	}
--- retroshare-nogui/src/retroshare-nogui.pro.orig	2015-04-16 12:36:00.940471520 +0300
+++ retroshare-nogui/src/retroshare-nogui.pro	2015-04-16 12:36:32.459470968 +0300
@@ -36,19 +36,7 @@
 	gxs {
 		SQLCIPHER_OK = $$system(pkg-config --exists sqlcipher && echo yes)
 			isEmpty(SQLCIPHER_OK) {
-# We need a explicit path here, to force using the home version of sqlite3 that really encrypts the database.
-
-				exists(../../../lib/sqlcipher/.libs/libsqlcipher.a) {
-
-					LIBS += ../../../lib/sqlcipher/.libs/libsqlcipher.a
-						DEPENDPATH += ../../../lib/sqlcipher/src/
-						INCLUDEPATH += ../../../lib/sqlcipher/src/
-				} else {
-					message(libsqlcipher.a not found. Compilation will not use SQLCIPHER. Database will be unencrypted.)
-						DEFINES *= NO_SQLCIPHER
-						LIBS *= -lsqlite3
-				}
-
+				error(libsqlcipher.a not found)
 			} else {
 				LIBS *= -lsqlcipher
 			}
