commit b686771a38c93339ece773bd5ae4d0db1fa629b7
Author: Gleb Nemshilov <gleb@fastmail.com>
Date:   Sun Aug 6 14:27:26 2017 +0700

    fix C++ comment to avoid compilation error

diff --git a/supportlibs/pegmarkdown/utility_functions.c b/supportlibs/pegmarkdown/utility_functions.c
index 08f910274..9445de420 100644
--- a/supportlibs/pegmarkdown/utility_functions.c
+++ b/supportlibs/pegmarkdown/utility_functions.c
@@ -26,7 +26,7 @@ element *reverse(element *list) {
     element *new = NULL;
     element *next = NULL;
 #warning Phenom (2017-07-21): I don't know if it is a real memLeak for new. If not remove this warning and add a comment how it is deleted.
-    // cppcheck-suppress memleak
+    /* cppcheck-suppress memleak */
     while (list != NULL) {
         next = list->next;
         new = cons(list, new);
