Index: menhir-20151112/src/Makefile
===================================================================
--- menhir-20151112.orig/src/Makefile
+++ menhir-20151112/src/Makefile
@@ -19,7 +19,7 @@ endif
 # ----------------------------------------------------------------------------
 # Ocamlbuild tool and settings.
 
-OCAMLBUILD := ocamlbuild -classic-display -j 0 -cflags "-safe-string -bin-annot"
+OCAMLBUILD := ocamlbuild -no-hygiene -classic-display -j 0 -cflags "-safe-string -bin-annot"
 
 # ----------------------------------------------------------------------------
 # For everyday development.
