diff --git a/Megadeth/Prim.hs b/Megadeth/Prim.hs
index 2f39336..7911bc3 100644
--- a/bundled/megadeth/Megadeth/Prim.hs
+++ b/bundled/megadeth/Megadeth/Prim.hs
@@ -82,2 +82,7 @@ simpleConView tyName c =
   ForallC _ _ innerCon -> simpleConView tyName innerCon
+#if MIN_VERSION_template_haskell(2,11,0)
+  -- handling: GadtC [Codec.Picture.Metadata.Gamma] [] (AppT (ConT Codec.Picture.Metadata.Keys) (ConT GHC.Types.Double))
+  GadtC [n] sts _ -> let ts = map snd sts
+                     in SimpleCon n (count ts) ts
+#endif
   _ -> error $ "simpleConView: failed on " ++ show c
