From: Julian Ospald <hasufell@gentoo.org>
Date: Sun Mar  9 13:03:10 UTC 2014
Subject: fix ImageFile import for pillow

--- freevo-1.9.0/setup.py
+++ freevo-1.9.0/setup.py
@@ -26,7 +26,7 @@
     ('kaa.imlib2', '\"svn co svn://svn.freevo.org/kaa/trunk/ kaa\"' ),
     ('BeautifulSoup', 'http://www.crummy.com/software/BeautifulSoup/' ),
     ('pygame', 'http://www.pygame.org'),
-    ('Image', 'http://www.pythonware.com/products/pil/'),
+    ('PIL.Image', 'http://pypi.python.org/pypi/Pillow'),
     ('twisted', 'http://www.twistedmatrix.com/'),
     ('zope.interface', 'http://www.zope.org/Products/ZopeInterface'),
     ('twisted.web.microdom', 'http://www.twistedmatrix.com/'),
--- freevo-1.9.0/src/plugins/zoneminder.py
+++ freevo-1.9.0/src/plugins/zoneminder.py
@@ -50,7 +50,7 @@
 import pygame
 import MySQLdb
 import threading
-import ImageFile
+from PIL import ImageFile
 
 # Freevo
 import config
