*** xbomb-2.1/Makefile	1996-01-20 16:02:46.000000000 -0500
--- xbomb-2.1.new/Makefile
***************
*** 27,34 ****
  
  OBJ=xbomb.o xwindow.o hiscore.o
  
- INSTDIR=/usr/local
- 
  ########
  
  xbomb : $(OBJ)
--- 27,32 ----
***************
*** 52,60 ****
  
  install :
  	strip xbomb
! 	install -d $(INSTDIR)/bin
! 	install -d $(INSTDIR)/man/man6
! 	install -d $(INSTDIR)/lib/app-defaults
! 	install -m 755 xbomb $(INSTDIR)/bin
! 	install -m 644 xbomb.6 $(INSTDIR)/man/man6
! 	install -m 644 xbomb.ad $(INSTDIR)/lib/app-defaults/XBomb
--- 50,58 ----
  
  install :
  	strip xbomb
! 	install -d $(DESTDIR)/usr/bin
! 	install -d $(DESTDIR)/usr/share/man/man6
! 	install -d $(DESTDIR)/etc/X11/app-defaults
! 	install -m 755 xbomb $(DESTDIR)/usr/bin
! 	install -m 644 xbomb.6 $(DESTDIR)/usr/share/man/man6
! 	install -m 644 xbomb.ad $(DESTDIR)/etc/X11/app-defaults/XBomb
