diff -Naur pinepgp-0.18.0-orig/Makefile.in pinepgp-0.18.0/Makefile.in
--- pinepgp-0.18.0-orig/Makefile.in	2002-05-03 15:30:07.000000000 +0200
+++ pinepgp-0.18.0/Makefile.in	2006-11-16 00:30:27.000000000 +0100
@@ -44,52 +44,52 @@
 all:	$(SOURCES) $(BUILD_FILTERS)
 
 $(SOURCES):	$(SOURCES:%=%.in)
-	$(CAT) $(@).in | $(SED) -e 's%\@SED\@%$(SED)%;\
-		s%\@CAT\@%$(CAT)%;\
-		s%\@RM\@%$(RM)%;\
-		s%\@MKTEMP\@%$(MKTEMP)%;\
-		s%\@BASH\@%$(BASH)%;\
-		s%\@bindir\@%$(bindir)%;\
-		s%\@tmpdir\@%$(tmpdir)%' > $(@)
+	$(CAT) $(@).in | $(SED) -e 's%\@SED\@%$(SED)%'\
+		-e 's%\@CAT\@%$(CAT)%'\
+		-e 's%\@RM\@%$(RM)%'\
+		-e 's%\@MKTEMP\@%$(MKTEMP)%'\
+		-e 's%\@BASH\@%$(BASH)%'\
+		-e 's%\@bindir\@%$(bindir)%'\
+		-e 's%\@tmpdir\@%$(tmpdir)%' > $(@)
 	chmod 755 $(@)
 
 pinegpg:	pinegpgp
-	$(CAT) pinegpgp | $(SED) -e 's%^cename=.*%cename=GnuPG%;\
-		s%^ces=.*%ces="$(GPG) -at -s"%;\
-		s%^cee=.*%cee="$(GPG) -at -e"%;\
-		s%^cese=.*%cese="$(GPG) -at -se"%;\
-		s%^cec=.*%cec="\$(GPG)"%;\
-		s%^cefp=.*%cefp=gpg%' > $(@).c
+	$(CAT) pinegpgp | $(SED) -e 's%^cename=.*%cename=GnuPG%'\
+		-e 's%^ces=.*%ces="$(GPG) -at -s"%'\
+		-e 's%^cee=.*%cee="$(GPG) -at -e"%'\
+		-e 's%^cese=.*%cese="$(GPG) -at -se"%'\
+		-e 's%^cec=.*%cec="\$(GPG)"%'\
+		-e 's%^cefp=.*%cefp=gpg%' > $(@).c
 	$(CPP) -DGPG $(BUILD_FLAGS) $(@).c | $(GREP) -v "^# .* .*" > $(@)
 	chmod 755 $(@)
 
 pinepgp26x:	pinegpgp
-	$(CAT) pinegpgp | $(SED) -e 's%^cename=.*%cename="PGP 2.6.x"%;\
-		s%^ces=.*%ces="\$(PGP) -fat -s"%;\
-		s%^cee=.*%cee="\$(PGP) -fat -e"%;\
-		s%^cese=.*%cese="\$(PGP) -fat -se"%;\
-		s%^cec=.*%cec="\$(PGP) -f"%;\
-		s%^cefp=.*%cefp=pgp26x%' > $(@).c
+	$(CAT) pinegpgp | $(SED) -e 's%^cename=.*%cename="PGP 2.6.x"%'\
+		-e 's%^ces=.*%ces="\$(PGP) -fat -s"%'\
+		-e 's%^cee=.*%cee="\$(PGP) -fat -e"%'\
+		-e 's%^cese=.*%cese="\$(PGP) -fat -se"%'\
+		-e 's%^cec=.*%cec="\$(PGP) -f"%'\
+		-e 's%^cefp=.*%cefp=pgp26x%' > $(@).c
 	$(CPP) -DPGP26X $(BUILD_FLAGS) $(@).c | $(GREP) -v "^# .* .*" > $(@)
 	chmod 755 $(@)
 
 pinepgp5x:	pinegpgp
-	$(CAT) pinegpgp | $(SED) -e 's%^cename=.*%cename="PGP 5.x"%;\
-		s%^ces=.*%ces="$(PGP5S) -fat"%;\
-		s%^cee=.*%cee="$(PGP5E) -fat"%;\
-		s%^cese=.*%cese="$(PGP5E) -fat -s"%;\
-		s%^cec=.*%cec="$(PGP5V) -f"%;\
-		s%^cefp=.*%cefp=pgp26x%' > $(@).c
+	$(CAT) pinegpgp | $(SED) -e 's%^cename=.*%cename="PGP 5.x"%'\
+		-e 's%^ces=.*%ces="$(PGP5S) -fat"%'\
+		-e 's%^cee=.*%cee="$(PGP5E) -fat"%'\
+		-e 's%^cese=.*%cese="$(PGP5E) -fat -s"%'\
+		-e 's%^cec=.*%cec="$(PGP5V) -f"%'\
+		-e 's%^cefp=.*%cefp=pgp26x%' > $(@).c
 	$(CPP) -DPGP5X $(BUILD_FLAGS) $(@).c | $(GREP) -v "^# .* .*" > $(@)
 	chmod 755 $(@)
 
 pinepgp65x:	pinegpgp
-	$(CAT) pinegpgp | $(SED) -e 's%^cename=.*%cename="PGP 6.5.x"%;\
-		s%^ces=.*%ces="\$(PGP6) -fat -s -u $$USER"%;\
-		s%^cee=.*%cee="\$(PGP6) -fat -e -u $$USER"%;\
-		s%^cese=.*%cese="\$(PGP6) -fat -se -u $$USER"%;\
-		s%^cec=.*%cec="\$(PGP6) -f -u $$USER"%;\
-		s%^cefp=.*%cefp=pgp65x%' > $(@).c
+	$(CAT) pinegpgp | $(SED) -e 's%^cename=.*%cename="PGP 6.5.x"%'\
+		-e 's%^ces=.*%ces="\$(PGP6) -fat -s -u $$USER"%'\
+		-e 's%^cee=.*%cee="\$(PGP6) -fat -e -u $$USER"%'\
+		-e 's%^cese=.*%cese="\$(PGP6) -fat -se -u $$USER"%'\
+		-e 's%^cec=.*%cec="\$(PGP6) -f -u $$USER"%'\
+		-e 's%^cefp=.*%cefp=pgp65x%' > $(@).c
 	$(CPP) -DPGP65X $(BUILD_FLAGS) $(@).c | $(GREP) -v "^# .* .*" > $(@)
 	chmod 755 $(@)
 
