From d271652d3124c9e65a97df310a141c607a0b5ee5 Mon Sep 17 00:00:00 2001
From: fredrik <fredrik.johansson@gmail.com>
Date: Wed, 29 Sep 2021 19:55:34 +0200
Subject: [PATCH] add missing include

---
 acb_poly/lgamma_series.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/acb_poly/lgamma_series.c b/acb_poly/lgamma_series.c
index 504b0cb4..d50ffb9f 100644
--- a/acb_poly/lgamma_series.c
+++ b/acb_poly/lgamma_series.c
@@ -10,6 +10,7 @@
 */
 
 #include "acb_poly.h"
+#include "acb_hypgeom.h"
 
 void
 _acb_log_rising_correct_branch(acb_t t,
