--- GraphicsMagick-1.3.20/magick/studio.h
+++ GraphicsMagick-1.3.20/magick/studio.h
@@ -51,12 +51,11 @@
   threads.  Most CPUs have cache lines of 32 or 64 bytes.  IBM Power CPUs have
   cache lines of 128 bytes.
 */
-/* FIXME: C pre-processor does not support comparing strings. */
-/* #if defined(MAGICK_TARGET_CPU) && (MAGICK_TARGET_CPU == powerpc) */
-/* #  define MAGICK_CACHE_LINE_SIZE 128 */
-/* #else */
-#  define MAGICK_CACHE_LINE_SIZE 64
-/* #endif */
+#ifdef __powerpc__
+#define MAGICK_CACHE_LINE_SIZE 128
+#else
+#define MAGICK_CACHE_LINE_SIZE 64
+#endif
 
 
 /*
