diff -urN kmplayer-0.11.3d.old/src/kmplayer_part.desktop kmplayer-0.11.3d/src/kmplayer_part.desktop
--- kmplayer-0.11.3d.old/src/kmplayer_part.desktop	2012-06-10 14:35:46.000000000 +0200
+++ kmplayer-0.11.3d/src/kmplayer_part.desktop	2012-09-11 14:42:04.160426844 +0200
@@ -93,7 +93,7 @@
 Comment[zh_TW]=影像播放器外掛程式（與 QuickTime，XMovie，Windows Media 與 RealPlayer 相容）
 X-KDE-Library=libkmplayerpart
 # Keep in sync with the audio types in kmplayer.desktop
-MimeType=application/ogg;application/smil;application/vnd.ms-asf;application/vnd.rn-realmedia;application/x-kmplayer;application/x-mplayer2;application/x-ogg;application/xspf+xml;audio/basic;audio/m3u;audio/mp4;audio/mpeg;audio/mpegurl;audio/vnd.rn-realaudio;audio/vorbis;audio/x-aiff;audio/x-flac;audio/x-m3u;audio/x-matroska;audio/x-mp2;audio/x-mp3;audio/x-mpegurl;audio/x-ms-asx;audio/x-ms-wma;audio/x-musepack;audio/x-mod;audio/x-oggflac;audio/x-pn-realaudio;audio/x-scpls;audio/x-speex;audio/x-vorbis;audio/x-wav;video/avi;video/mediaplayer;video/mp4;video/mpeg;video/quicktime;video/vnd.rn-realvideo;video/x-avi;video/x-flic;video/x-matroska;video/x-ms-asf;video/x-msvideo;video/x-ms-wmp;video/x-ms-wmv;video/x-ogm;video/x-theora;x-scheme-handler/mms;x-scheme-handler/pnm;x-scheme-handler/rtspt;x-scheme-handler/rtspu;
+MimeType=application/x-shockwave-flash;application/ogg;application/smil;application/vnd.ms-asf;application/vnd.rn-realmedia;application/x-kmplayer;application/x-mplayer2;application/x-ogg;application/xspf+xml;audio/basic;audio/m3u;audio/mp4;audio/mpeg;audio/mpegurl;audio/vnd.rn-realaudio;audio/vorbis;audio/x-aiff;audio/x-flac;audio/x-m3u;audio/x-matroska;audio/x-mp2;audio/x-mp3;audio/x-mpegurl;audio/x-ms-asx;audio/x-ms-wma;audio/x-musepack;audio/x-mod;audio/x-oggflac;audio/x-pn-realaudio;audio/x-scpls;audio/x-speex;audio/x-vorbis;audio/x-wav;video/avi;video/mediaplayer;video/mp4;video/mpeg;video/quicktime;video/vnd.rn-realvideo;video/x-avi;video/x-flic;video/x-matroska;video/x-ms-asf;video/x-msvideo;video/x-ms-wmp;video/x-ms-wmv;video/x-ogm;video/x-theora;x-scheme-handler/mms;x-scheme-handler/pnm;x-scheme-handler/rtspt;x-scheme-handler/rtspu;
 Type=Service
 Icon=kmplayer
 X-KDE-ServiceTypes=KParts/ReadOnlyPart,Browser/View,KMediaPlayer/Player
