--- a/configure.ac
+++ b/configure.ac
@@ -13,9 +13,10 @@
 
 # Checks for libraries.
 AC_CHECK_LIB([ncurses], [initscr], [],
-             AC_MSG_ERROR([Can't find ncurses library. Install it first.]))
+             AC_MSG_ERROR([Cannot find ncurses library. Install it first.]))
+AC_SEARCH_LIBS([define_key], [tinfo])
 AC_CHECK_LIB([z], [gzopen], [],
-             AC_MSG_ERROR([Can't find zlib library. Install it first.]))
+             AC_MSG_ERROR([Cannot find zlib library. Install it first.]))
 
 # Checks for header files.
 AC_HEADER_STDC
