--- a/main/src/addins/MonoDevelop.PackageManagement/MonoDevelop.PackageManagement/ImageLoader.cs
+++ b/main/src/addins/MonoDevelop.PackageManagement/MonoDevelop.PackageManagement/ImageLoader.cs
@@ -91,8 +91,7 @@ namespace MonoDevelop.PackageManagement
 		ImageLoadedEventArgs LoadImage (Uri uri, object state)
 		{
 			try {
-				var httpClient = new HttpClient (uri);
-				Stream stream = httpClient.GetResponse ().GetResponseStream ();
+				Stream stream = GetResponseStream (uri);
 				Image image = Image.FromStream (stream);
 
 				return new ImageLoadedEventArgs (image, uri, state);
@@ -101,6 +100,17 @@ namespace MonoDevelop.PackageManagement
 			}
 		}
 
+		static Stream GetResponseStream (Uri uri)
+		{
+			if (uri.IsFile) {
+				var request = WebRequest.Create (uri);
+				return request.GetResponse ().GetResponseStream ();
+			}
+
+			var httpClient = new HttpClient (uri);
+			return httpClient.GetResponse ().GetResponseStream ();
+		}
+
 		void OnLoaded (ITask<ImageLoadedEventArgs> task, Uri uri, object state)
 		{
 			if (task.IsFaulted) {

