From a1542342335d6346c5772a41c5cf9da6cdbe9e9a Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Tue, 8 Mar 2022 19:19:06 +0100
Subject: [PATCH] Add missing KCoreAddons link

---
 CMakeLists.txt     | 2 +-
 src/CMakeLists.txt | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d177510..72e75ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ include(FeatureSummary)
 find_package (Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} CONFIG REQUIRED Core Widgets Svg)
 
 find_package (KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
-    I18n XmlGui GuiAddons DocTools DBusAddons Crash
+    CoreAddons I18n XmlGui GuiAddons DocTools DBusAddons Crash
 )
 
 find_package (Phonon4Qt${QT_MAJOR_VERSION} REQUIRED)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d6491d9..a322de9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -29,6 +29,7 @@ ecm_add_app_icon(blinken_SRCS ICONS ${ICON_SRCS})
 add_executable(blinken ${blinken_SRCS})
 
 target_link_libraries(blinken
+    KF5::CoreAddons
     KF5::I18n
     KF5::XmlGui
     KF5::GuiAddons
-- 
GitLab

