--- src/mgui/ffviewer.cpp.ORIG	2013-08-01 11:05:12.962407787 +0200
+++ src/mgui/ffviewer.cpp	2013-08-01 11:08:26.907409685 +0200
@@ -62,7 +62,7 @@
 
 typedef struct AVCodecTag {
 #if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(52,39,00)
-    enum CodecID id;
+    enum AVCodecID id;
 #else
     int id;
 #endif
@@ -70,14 +70,14 @@
 } AVCodecTag;
 
 #if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(52,34,00)
-static uint FFCodecID2Tag(CodecID codec_id) 
+static uint FFCodecID2Tag(AVCodecID codec_id) 
 {
     unsigned int ff_codec_get_tag(const AVCodecTag *tags, int id);
     extern const AVCodecTag ff_codec_bmp_tags[];
     return ff_codec_get_tag(ff_codec_bmp_tags, codec_id);
 }
 #else
-static uint FFCodecID2Tag(CodecID codec_id) 
+static uint FFCodecID2Tag(AVCodecID codec_id) 
 {
     unsigned int codec_get_tag(const AVCodecTag *tags, int id);
     extern const AVCodecTag codec_bmp_tags[];
@@ -388,7 +388,7 @@
     return (tag>>bit_begin) & 0xFF;
 }
 
-static std::string CodecID2Str(CodecID codec_id)
+static std::string CodecID2Str(AVCodecID codec_id)
 {
 #ifdef _MSC_VER
     std::string tag_str = boost::format("%1%") % codec_id % bf::stop;
@@ -406,7 +406,7 @@
 
 #else // CALC_FF_TAG
 
-static std::string CodecID2Str(CodecID codec_id)
+static std::string CodecID2Str(AVCodecID codec_id)
 {
     return Int2Str(codec_id);
 }
