From 8e75c8678b874c6813e590634943166daa98124b Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Fri, 10 May 2019 11:01:21 +0200
Subject: [PATCH] Don't make the socket world writable

Signed-off-by: Lars Wendler <polynomial-c@gentoo.org>
---
 radeon-profile-daemon/rpdthread.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/radeon-profile-daemon/rpdthread.cpp b/radeon-profile-daemon/rpdthread.cpp
index 4cf0ac0..137d5b2 100644
--- a/radeon-profile-daemon/rpdthread.cpp
+++ b/radeon-profile-daemon/rpdthread.cpp
@@ -47,7 +47,7 @@ void rpdThread::createServer()
 
     QLocalServer::removeServer(serverName);
     daemonServer.listen(serverName);
-    QFile::setPermissions("/tmp/" + serverName, QFile("/tmp/" + serverName).permissions() | QFile::WriteOther | QFile::ReadOther);
+    QFile::setPermissions("/tmp/" + serverName, QFile("/tmp/" + serverName).permissions() | QFile::WriteGroup | QFile::ReadGroup);
 }
 
 void rpdThread::closeConnection()
-- 
2.21.0

