--- a/contrib/libuv-cmake/CMakeLists.txt
+++ b/contrib/libuv-cmake/CMakeLists.txt
@@ -1,3 +1,9 @@
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(UV REQUIRED IMPORTED_TARGET libuv)
+add_library(_uv INTERFACE)
+target_link_libraries(_uv INTERFACE PkgConfig::UV)
+add_library(ch_contrib::uv ALIAS _uv)
+return()
 # This file is a modified version of contrib/libuv/CMakeLists.txt
 
 set (SOURCE_DIR "${PROJECT_SOURCE_DIR}/contrib/libuv")
