--- a/contrib/libcpuid-cmake/CMakeLists.txt
+++ b/contrib/libcpuid-cmake/CMakeLists.txt
@@ -1,3 +1,9 @@
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(CPUID REQUIRED IMPORTED_TARGET libcpuid)
+add_library(_cpuid INTERFACE)
+target_link_libraries(_cpuid INTERFACE PkgConfig::CPUID)
+add_library(ch_contrib::cpuid ALIAS _cpuid)
+return()
 if(ARCH_AMD64)
     option (ENABLE_CPUID "Enable libcpuid library (only internal)" ${ENABLE_LIBRARIES})
 elseif(ENABLE_CPUID)
