From 60dde499ac87c2399ac24ae85c98ed8cce564858 Mon Sep 17 00:00:00 2001
From: Timothy Stack <timothyshanestack@gmail.com>
Date: Thu, 31 Dec 2020 23:23:56 -0800
Subject: [PATCH] [tests] fix test to make it insensitive to the current date

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>
---
 test/test_sql.sh | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/test/test_sql.sh b/test/test_sql.sh
index c7480ecc..bbb98dce 100644
--- a/test/test_sql.sh
+++ b/test/test_sql.sh
@@ -1012,13 +1012,13 @@ EOF
 
 
 cat ${test_dir}/logfile_syslog.0 | run_test ${lnav_test} -n \
-    -c ";select log_time from syslog_log where log_procname = 'automount'"
+    -c ";select log_body from syslog_log where log_procname = 'automount'"
 
 check_output "querying against stdin is not working?" <<EOF
-        log_time
-2020-11-03 09:23:38.000
-2020-11-03 09:23:38.000
-2020-11-03 09:23:38.000
+                log_body
+ lookup(file): lookup for foobar failed
+ attempting to mount entry /auto/opt
+ lookup(file): lookup for opt failed
 EOF
 
 
-- 
2.31.1

