Disable failing tests for bug #630916.

--- lnav-0.8.3/test/Makefile.am
+++ lnav-0.8.3/test/Makefile.am
@@ -374,7 +374,6 @@
 	test_concise \
 	test_line_buffer2 \
 	test_line_buffer.sh \
-	test_listview.sh \
 	test_grep_proc.sh \
 	test_grep_proc2 \
 	test_hist_source \
@@ -388,18 +387,15 @@
 	test_scripts.sh \
 	test_sessions.sh \
 	test_shlexer.sh \
-	test_sql.sh \
 	test_sql_coll_func.sh \
 	test_sql_json_func.sh \
 	test_sql_fs_func.sh \
 	test_sql_str_func.sh \
 	test_sql_time_func.sh \
-	test_view_colors.sh \
 	test_vt52_curses.sh \
 	test_top_status \
 	test_data_parser.sh \
-	test_yajlpp \
-	test_pretty_print.sh
+	test_yajlpp
 
 if HAVE_LIBCURL
 TESTS += \
