From 20fb86991babb1aa5aa278b2cd4dc454c78d5cd4 Mon Sep 17 00:00:00 2001
From: Hleb Valoshka <375gnu@gmail.com>
Date: Sat, 24 Feb 2018 18:36:20 +0300
Subject: [PATCH] fix symlink creation

---
 src/celestia/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/celestia/Makefile.am b/src/celestia/Makefile.am
index 25eb7d64..60609fd3 100644
--- a/src/celestia/Makefile.am
+++ b/src/celestia/Makefile.am
@@ -79,4 +79,4 @@ noinst_DATA = ../../celestia
 CLEANFILES = ../../celestia
 
 ../../celestia: celestia
-	(cd ../..; ln -s src/celestia/celestia)
+	(cd ../..; ln -sf src/celestia/celestia)
