From 80bd68c7ff75f3e87574a4557ccf149fc70752b3 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Tue, 15 May 2018 11:28:08 +0200
Subject: [PATCH 1/2] Fix gtk related undefined references

---
 src/celestia/Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/celestia/Makefile.am b/src/celestia/Makefile.am
index 60609fd3..a08fb0c7 100644
--- a/src/celestia/Makefile.am
+++ b/src/celestia/Makefile.am
@@ -14,7 +14,8 @@ endif
 
 if ENABLE_GTK
 SUBDIRS += gtk
-celestiaGTKLIBS = $(GTK_LIBS) gtk/libgtkgui.a
+celestiaGTKLIBS = gtk/libgtkgui.a
+LIBS += $(GTK_LIBS)
 endif
 
 if ENABLE_QT
-- 
2.17.0

