Index: Integer.h
===================================================================
RCS file: /home/imocvs/cdd+/Integer.h,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 Integer.h
--- Integer.h	2008/12/15 01:19:52	1.1.1.1
+++ Integer.h	2008/12/15 01:24:04
@@ -23,6 +23,7 @@
 #include <gmp_init.h>
 #include <cctype>
 #include <limits>
+#include <climits>
 
 class Integer; class Rational;
 
Index: Rational.cc
===================================================================
RCS file: /home/imocvs/cdd+/Rational.cc,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 Rational.cc
--- Rational.cc	2008/12/15 01:19:52	1.1.1.1
+++ Rational.cc	2008/12/15 01:25:14
@@ -17,6 +17,7 @@
 
 #include <cctype>
 #include "Rational.h"
+#include <cstring>
 
 Rational& Rational::set(const char* s) throw (gmp_error)
 {
Index: Rational.h
===================================================================
RCS file: /home/imocvs/cdd+/Rational.h,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 Rational.h
--- Rational.h	2008/12/15 01:19:52	1.1.1.1
+++ Rational.h	2008/12/15 01:27:08
@@ -17,6 +17,7 @@
 #define _POLYMAKE_GMP_RATIONAL_H "$Project: polymake $$Id: cdd+-077a-headers.patch,v 1.1 2010/11/12 15:04:32 tomka Exp $"
 
 #include <Integer.h>
+#include <cstring>
 
 #if __GNU_MP_VERSION < 4
 #define _tmp_little_Integer(x) \
@@ -1178,7 +1179,7 @@ std::basic_ostream<char, Traits>& operat
    Integer::little_buffer buf(s);
    numerator(a).putstr(os.flags(), buf);
    if (show_den) {
-      char *den_buf=buf+strlen(buf);
+      char *den_buf=buf+std::strlen(buf);
       *den_buf++ = '/';
       denominator(a).putstr(os.flags(), den_buf);
    }
Index: gmp_init.cc
===================================================================
RCS file: /home/imocvs/cdd+/gmp_init.cc,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 gmp_init.cc
--- gmp_init.cc	2008/12/15 01:19:52	1.1.1.1
+++ gmp_init.cc	2008/12/15 01:23:45
@@ -16,6 +16,7 @@
 #ident "$Project: polymake $$Id: cdd+-077a-headers.patch,v 1.1 2010/11/12 15:04:32 tomka Exp $"
 
 #include <memory>
+#include <cstring>
 #include "gmp_init.h"
 
 #if defined(__GNUC__)
