diff -ru arj-3.10.22.orig/environ.c arj-3.10.22/environ.c
--- arj-3.10.22.orig/environ.c	2008-03-30 16:51:40 +0200
+++ arj-3.10.22/environ.c	2008-03-30 16:50:16 +0200
@@ -61,7 +61,7 @@
  #elif defined(__FreeBSD__)||defined(__NetBSD__)||defined(__APPLE__)
   #include <sys/param.h>
   #include <sys/mount.h>
- #elif defined(__QNXNTO__)
+ #elif defined(__QNXNTO__) || defined(__INTERIX)
   #include <sys/statvfs.h>
  #else
   #include <sys/statfs.h>
@@ -2286,7 +2286,7 @@
   else
    return((LONG_MAX/(spclu*bps)<fclu)?LONG_MAX:spclu*bps*fclu);
  #elif TARGET==UNIX
-  #if defined(__QNXNTO__)||defined(__sco__)||defined(SUNOS)
+  #if defined(__QNXNTO__)||defined(__sco__)||defined(SUNOS)||defined(__INTERIX)
    struct statvfs vfs;
 
    if(statvfs(name, &vfs)==-1)
@@ -3802,7 +3802,9 @@
  #elif TARGET==WIN32
   return(0);
  #elif TARGET==UNIX
+ #ifndef __INTERIX
   sync();
+ #endif
   return(0);
  #endif
 }
diff -ru arj-3.10.22.orig/gnu/configure.in arj-3.10.22/gnu/configure.in
--- arj-3.10.22.orig/gnu/configure.in	2008-03-30 16:51:40 +0200
+++ arj-3.10.22/gnu/configure.in	2008-03-30 16:41:37 +0200
@@ -67,8 +67,9 @@
 	DLL_EXT=".dylib"
 	DLL_FLAGS=""
 	;;
-interix3*)
+interix*)
 	# not ELF
+	DYN_LIBS="-ldl"
 	DLL_FLAGS="-shared"
 	DLL_CFLAGS=""
 	;;
