--- sauerbraten_unix.orig	2008-08-15 20:05:44.000000000 +0200
+++ sauerbraten_unix	2008-08-15 22:49:47.000000000 +0200
@@ -3,6 +3,7 @@
 #SAUER_DIR=~/sauerbraten
 #SAUER_DIR=/usr/local/sauerbraten
 SAUER_DIR=.
+SAUER_DATADIR=.
 
 # SAUER_OPTIONS contains any command line options you would like to start Sauerbraten with.
 #SAUER_OPTIONS="-f"
@@ -46,7 +47,7 @@
 
 if [ -x ${SAUER_DIR}/bin_unix/${MACHINE_NAME}${SYSTEM_NAME}client ]
 then
-  cd ${SAUER_DIR}
+  cd ${SAUER_DATADIR}
   exec ${SAUER_DIR}/bin_unix/${MACHINE_NAME}${SYSTEM_NAME}client ${SAUER_OPTIONS} $@
 else
   echo "Your platform does not have a pre-compiled Sauerbraten client."
