diff -NrU5 cgterm-1.6.original/Makefile cgterm-1.6/Makefile
--- cgterm-1.6.original/Makefile	2009-09-07 08:43:52.000000000 -0600
+++ cgterm-1.6/Makefile	2009-09-07 09:06:07.000000000 -0600
@@ -8,11 +8,11 @@
 # for solaris:
 #SOCKETLIBS = -lsocket -lnsl
 
 CC=gcc
 CFLAGS=-O3 -Wall `sdl-config --cflags` -DPREFIX=\"$(PREFIX)\"
-LDFLAGS=`sdl-config --libs` $(SOCKETLIBS)
+LDFLAGS+=`sdl-config --libs` $(SOCKETLIBS)
 
 
 OBJS= \
 	kernal.o \
 	gfx.o \
@@ -62,25 +62,22 @@
 
 testimage: testimage.c diskimage.c dir.c
 	$(CC) -g -O3 -Wall -o testimage $^
 
 install: all installdirs
-	strip cgterm$(EXESUFFIX)
 	cp cgterm$(EXESUFFIX) $(PREFIX)/bin/
 	cp cgchat$(EXESUFFIX) $(PREFIX)/bin/
 	cp *.bmp *.kbd *.wav $(PREFIX)/share/cgterm/
 #	cp cgterm.cfg $(PREFIX)/etc/
 #	cp cgchat.cfg $(PREFIX)/etc/
 
-installdirs: $(PREFIX)/bin $(PREFIX)/share $(PREFIX)/share/cgterm $(PREFIX)/etc
+installdirs: $(PREFIX)/bin $(PREFIX)/share/cgterm $(PREFIX)/etc
 
 $(PREFIX)/bin:
-	mkdir $(PREFIX)/bin > /dev/null 2>&1
-$(PREFIX)/share:
-	mkdir $(PREFIX)/share > /dev/null 2>&1
+	mkdir -p $(PREFIX)/bin > /dev/null 2>&1
 $(PREFIX)/share/cgterm:
-	mkdir $(PREFIX)/share/cgterm > /dev/null 2>&1
+	mkdir -p $(PREFIX)/share/cgterm > /dev/null 2>&1
 $(PREFIX)/etc:
-	mkdir $(PREFIX)/etc > /dev/null 2>&1
+	mkdir -p $(PREFIX)/etc > /dev/null 2>&1
 
 clean:
 	rm -f cgterm$(EXESUFFIX) cgchat$(EXESUFFIX) testkbd$(EXESUFFIX) *.o *~
