--- openjpeg-2.0.0/CMakeLists.txt
+++ openjpeg-2.0.0/CMakeLists.txt
@@ -279,11 +279,4 @@
   DESTINATION ${OPENJPEG_INSTALL_PACKAGE_DIR}
 )
 
-#-----------------------------------------------------------------------------
-# install CHANGES and LICENSE
-if(EXISTS ${OPENJPEG_SOURCE_DIR}/CHANGES)
-  install(FILES CHANGES DESTINATION ${OPENJPEG_INSTALL_DOC_DIR})
-endif()
-install(FILES LICENSE DESTINATION ${OPENJPEG_INSTALL_DOC_DIR})
-
 include (cmake/OpenJPEGCPack.cmake)
