diff -urN a/src/app_sysenv/Makefile.am b/src/app_sysenv/Makefile.am
--- a/src/app_sysenv/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/app_sysenv/Makefile.am	2018-01-28 20:16:44.143484825 +0000
@@ -15,8 +15,8 @@
 
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
-AM_CFLAGS += -Wpedantic
+#AM_CFLAGS += -Werror
+#AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
 AM_CFLAGS += -fdump-rtl-expand
diff -urN a/src/base/Makefile.am b/src/base/Makefile.am
--- a/src/base/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/base/Makefile.am	2018-01-28 20:21:31.124936834 +0000
@@ -4,7 +4,7 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
+#AM_CFLAGS += -Werror
 # AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
diff -urN a/src/cmdline/Makefile.am b/src/cmdline/Makefile.am
--- a/src/cmdline/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/cmdline/Makefile.am	2018-01-28 20:20:32.178848976 +0000
@@ -4,7 +4,7 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS  = -Wall 
-AM_CFLAGS += -Werror
+#AM_CFLAGS += -Werror
 # flags g_option parser structs in cmd_parser_goption.c
 # AM_CFLAGS += -Wpedantic
 
diff -urN a/src/ddc/Makefile.am b/src/ddc/Makefile.am
--- a/src/ddc/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/ddc/Makefile.am	2018-01-28 20:18:09.643626498 +0000
@@ -11,8 +11,8 @@
 
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
-AM_CFLAGS += -Wpedantic
+#AM_CFLAGS += -Werror
+#AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
 AM_CFLAGS += -fdump-rtl-expand
diff -urN a/src/i2c/Makefile.am b/src/i2c/Makefile.am
--- a/src/i2c/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/i2c/Makefile.am	2018-01-28 20:17:30.271562118 +0000
@@ -4,8 +4,8 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
-AM_CFLAGS += -Wpedantic
+#AM_CFLAGS += -Werror
+#AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
 AM_CFLAGS += -fdump-rtl-expand
diff -urN a/src/sample_clients/Makefile.am b/src/sample_clients/Makefile.am
--- a/src/sample_clients/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/sample_clients/Makefile.am	2018-01-28 20:19:57.230795822 +0000
@@ -7,7 +7,7 @@
 -I$(top_srcdir)/src
 
 AM_CFLAGS = -Wall -fPIC
-AM_CFLAGS += -Werror
+#AM_CFLAGS += -Werror
 
 check_PROGRAMS =
 if ENABLE_SHARED_LIB_COND
diff -urN a/src/test/Makefile.am b/src/test/Makefile.am
--- a/src/test/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/test/Makefile.am	2018-01-28 20:19:03.509712410 +0000
@@ -12,7 +12,7 @@
 
 
 
-AM_CFLAGS = -Wall -Werror
+AM_CFLAGS = -Wall
 
 if ENABLE_CALLGRAPH_COND
 AM_CFLAGS += -fdump-rtl-expand
diff -urN a/src/usb/Makefile.am b/src/usb/Makefile.am
--- a/src/usb/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/usb/Makefile.am	2018-01-28 20:18:26.146653076 +0000
@@ -5,7 +5,7 @@
 -I$(top_srcdir)/src/public 
 
 AM_CFLAGS  = -Wall 
-AM_CFLAGS += -Wpedantic
+#AM_CFLAGS += -Wpedantic
 
 if ENABLE_CALLGRAPH_COND
 AM_CFLAGS += -fdump-rtl-expand
diff -urN a/src/util/Makefile.am b/src/util/Makefile.am
--- a/src/util/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/util/Makefile.am	2018-01-28 20:21:07.245901502 +0000
@@ -3,7 +3,7 @@
   $(LIBDRM_CFLAGS)
 
 AM_CFLAGS = -Wall 
-AM_CFLAGS += -Werror
+#AM_CFLAGS += -Werror
 # -pedantic issues warnings re code that doesn't conform to ISO C
 #  Alas, -m modifier on sscanf is a POSIX extension, not ISO C
 #  In combination with -Werror, causes device_util.c to fail compilation
diff -urN a/src/vcp/Makefile.am b/src/vcp/Makefile.am
--- a/src/vcp/Makefile.am	2018-01-20 06:40:55.000000000 +0000
+++ b/src/vcp/Makefile.am	2018-01-28 20:19:28.024750741 +0000
@@ -4,7 +4,7 @@
   -I$(top_srcdir)/src/public
 
 AM_CFLAGS  = -Wall 
-AM_CFLAGS += -Werror
+#AM_CFLAGS += -Werror
 # vcp_feature_codes.c requires extensive changes if -Wpedantic
 # AM_CFLAGS += -Wpedantic
 
