--- shpcreate.c.orig	2009-09-15 20:09:39.641422056 -0700
+++ shpcreate.c	2009-09-15 20:20:31.310673754 -0700
@@ -51,6 +51,7 @@
 static char rcsid[] = 
   "$Id$";
 
+#include <stdlib.h>
 #include "shapefil.h"
 
 int main( int argc, char ** argv )
--- shpdump.c.orig	2009-09-15 20:09:47.615421028 -0700
+++ shpdump.c	2009-09-15 20:20:29.654673421 -0700
@@ -67,6 +67,7 @@
 static char rcsid[] = 
   "$Id$";
 
+#include <stdlib.h>
 #include "shapefil.h"
 
 int main( int argc, char ** argv )
--- shprewind.c.orig	2009-09-15 20:09:58.193422322 -0700
+++ shprewind.c	2009-09-15 20:20:26.954542660 -0700
@@ -43,6 +43,7 @@
  *
  */
 
+#include <stdlib.h>
 #include "shapefil.h"
 
 int main( int argc, char ** argv )
