Allow fallback to a non GL sink for unittests.

--- a/pitivi/utils/pipeline.py	2015-05-25 13:58:33.980916075 +0200
+++ b/pitivi/utils/pipeline.py	2015-05-25 13:57:52.277261138 +0200
@@ -182,6 +182,9 @@
         # Create a cluttersink element used for display. Subclasses must connect
         # it to self._pipeline themselves
         self.video_sink = Gst.ElementFactory.make("glimagesink", None)
+        if self.video_sink is None:
+            self.video_sink = Gst.ElementFactory.make("xvimagesink", None)
+
         if isinstance(pipeline, GES.Pipeline):
             self._pipeline.preview_set_video_sink(self.video_sink)
         else:
