diff -Naur mozc-1.2.809.102.orig/build_mozc.py mozc-1.2.809.102/build_mozc.py
--- mozc-1.2.809.102.orig/build_mozc.py	2011-08-17 10:02:41.000000000 +0900
+++ mozc-1.2.809.102/build_mozc.py	2011-08-18 01:01:25.109704261 +0900
@@ -654,9 +654,7 @@
     if envvar in os.environ:
       os.environ[envvar] = os.getenv(envvar)
 
-  build_args = ['-j%s' % options.jobs,
-                'MAKE_JOBS=%s' % options.jobs,
-                'BUILDTYPE=%s' % options.configuration]
+  build_args = ['BUILDTYPE=%s' % options.configuration]
   build_args.append('builddir_name=%s' % GetBuildBaseName(options))
 
   RunOrDie([make_command] + build_args + target_names)
