diff --git a/src/base/absl.gyp b/src/base/absl.gyp
index a022b31..45749d8 100644
--- a/src/base/absl.gyp
+++ b/src/base/absl.gyp
@@ -275,7 +275,7 @@
           'all_dependent_settings': {
             'link_settings': {
               'libraries': [
-                '-labsl_random_distributions -labsl_random_internal_pool_urbg -labsl_random_internal_randen -labsl_random_internal_randen_hwaes -labsl_random_internal_randen_hwaes_impl -labsl_random_internal_randen_slow -labsl_random_internal_seed_material -labsl_random_seed_gen_exception -labsl_random_seed_sequences',
+                '-labsl_random_distributions -labsl_random_internal_entropy_pool -labsl_random_internal_randen -labsl_random_internal_randen_hwaes -labsl_random_internal_randen_hwaes_impl -labsl_random_internal_randen_slow -labsl_random_internal_seed_material -labsl_random_seed_gen_exception -labsl_random_seed_sequences',
               ],
             },
           },
