diff --git a/setup.py b/setup.py
index fa1ff8b..3f36120 100644
--- a/setup.py
+++ b/setup.py
@@ -52,11 +52,11 @@ if os.getenv('BUILD_WITH_CYTHON') and CYTHON_AVAILABLE:
         Extension(
             'csimdjson',
             [
-                'simdjson/simdjson.cpp',
                 'simdjson/util.cpp',
                 'simdjson/csimdjson.pyx'
             ],
             define_macros=macros,
+            libraries=["simdjson"],
             extra_compile_args=extra_compile_args
         )
     ], compiler_directives=compiler_directives, force=force)
