--- requests-2.12.1/setup.py
+++ requests-2.12.1/setup.py
@@ -36,14 +36,6 @@
 packages = [
     'requests',
     'requests.packages',
-    'requests.packages.chardet',
-    'requests.packages.idna',
-    'requests.packages.urllib3',
-    'requests.packages.urllib3.packages',
-    'requests.packages.urllib3.contrib',
-    'requests.packages.urllib3.util',
-    'requests.packages.urllib3.packages.ssl_match_hostname',
-    'requests.packages.urllib3.packages.backports',
 ]
 
 requires = []
