--- Makefile
+++ Makefile
@@ -75,9 +75,9 @@
 isomaster: $(OBJECTS) lib iniparser
 	@echo 'Linking isomaster'
 ifndef USE_SYSTEM_INIPARSER
-	@$(CC) $(LDFLAGS) `pkg-config --libs gtk+-2.0` $(OBJECTS) bk/bk.a iniparser-2.17/libiniparser.a -o isomaster
+	@$(CC) $(LDFLAGS) $(OBJECTS) -o isomaster bk/bk.a iniparser-2.17/libiniparser.a `pkg-config --libs gtk+-2.0` 
 else
-	@$(CC) $(LDFLAGS) `pkg-config --libs gtk+-2.0` $(OBJECTS) bk/bk.a -liniparser -o isomaster
+	@$(CC) $(LDFLAGS) $(OBJECTS) -o isomaster bk/bk.a -liniparser `pkg-config --libs gtk+-2.0`
 endif
 
 # static pattern rule
