--- pyifp-0.2.2/setup.py	2005-02-21 01:36:05.000000000 -0500
+++ pyifp-0.2.2-fixed/setup.py	2005-10-15 13:29:26.000000000 -0400
@@ -54,7 +54,7 @@
         self.swig_inc = 'SWIG'
         self.swig_shadow = None        
     
-    def swig_sources(self, sources):
+    def swig_sources(self, sources, extension=None):
         """Override the definition of 'swig_sources' in build_ext.  This
         is essentially the same function but with better swig support.
         I will now quote the original docstring:
