Index: stk-4.4.3/src/Makefile.in
===================================================================
--- stk-4.4.3.orig/src/Makefile.in
+++ stk-4.4.3/src/Makefile.in
@@ -74,7 +74,7 @@ $(SHAREDLIB) : $(OBJECTS)
 	$(RM) -f $(@) $(SHAREDLIB).$(MAJOR) $(SHAREDLIB)
 	$(CC) $(LDFLAGS) -fPIC @libflags@ $(OBJECT_PATH)/*.o $(LIBS)
 	$(LN) -s @sharedname@ $(SHAREDLIB)
-#	$(LN) -s @sharedname@ $(SHAREDLIB).$(MAJOR)
+	$(LN) -s @sharedname@ $(SHAREDLIB).$(MAJOR)
 
 $(OBJECTS) : Stk.h
 
