Index: stk-4.4.4/configure.ac
===================================================================
--- stk-4.4.4.orig/configure.ac
+++ stk-4.4.4/configure.ac
@@ -80,8 +80,8 @@ fi
 CPPFLAGS="$CPPFLAGS $cppflag"
 
 # For debugging and optimization ... overwrite default because it has both -g and -O2
-echo "$CXXFLAGS"
-CXXFLAGS="$cxxflag"
+#echo "$CXXFLAGS"
+#CXXFLAGS="$cxxflag"
 
 # Check compiler and use -Wall if gnu.
 if [test $GXX = "yes" ;] then
