--- doc/prelink.conf.orig	2008-08-29 08:18:59.000000000 -0400
+++ doc/prelink.conf	2008-08-29 08:21:48.000000000 -0400
@@ -16,8 +16,11 @@
 -b *.a
 -b *.js
 -b /lib/modules
--b /usr/lib/locale
--b /usr/X11R6/lib{,64}/X11/xfig
+-b /usr/lib{,32,64}/locale
+-b /usr/lib{,32,64}/wine
+-b /usr/lib{,32,64}/valgrind
+-b /usr/X11R6/lib{,32,64}/X11/xfig
+-b /usr/src
 -l /bin
 -l /usr/bin
 -l /sbin
@@ -27,9 +30,9 @@
 -l /usr/games
 -l /usr/libexec
 -l /var/ftp/bin
--l /lib{,64}
--l /usr/lib{,64}
--l /usr/X11R6/lib{,64}
--l /usr/kerberos/lib{,64}
+-l /lib{,32,64}
+-l /usr/lib{,32,64}
+-l /usr/X11R6/lib{,32,64}
+-l /usr/kerberos/lib{,32,64}
 -l /usr/X11R6/LessTif
--l /var/ftp/lib{,64}
+-l /var/ftp/lib{,32,64}
