--- a/Makefile.orig	2004-12-11 01:36:11.000000000 -0800
+++ b/Makefile	2004-12-11 01:36:14.000000000 -0800
@@ -13,9 +13,7 @@
 MIXER=-lSDL_mixer
 SDL_LIB=$(shell sdl-config --static-libs) $(MIXER)
 
-CFLAGS=-Wall -pedantic -std=c99 -O2 \
-	$(shell sdl-config --cflags) \
-	-DDATAPATH=\"$(DATAPATH)\"
+CFLAGS= $(ECFLAGS) $(shell sdl-config --cflags) -DDATAPATH=\"$(DATAPATH)\"
 
 all:	komi
 
@@ -24,15 +22,13 @@
 	-rm *.o
 
 install:
-	install -d $(DATAPATH)
-	cp -R komidata/* $(DATAPATH)
-	chmod -R 644 $(DATAPATH)
-	chmod 755 $(DATAPATH)
-	install -d $(BINPATH)
-	cp komi $(BINPATH)
-	chmod 755 $(BINPATH)komi
-	install -d $(MANPATH)
-	cp komi.6 $(MANPATH)komi.6
+	install -d $(DESTDIR)$(DATAPATH)
+	cp -R komidata/* $(DESTDIR)$(DATAPATH)
+	chmod -R 644 $(DESTDIR)$(DATAPATH)
+	chmod 755 $(DESTDIR)$(DATAPATH)
+	install -d $(DESTDIR)$(BINPATH)
+	cp komi $(DESTDIR)$(BINPATH)
+	chmod 755 $(DESTDIR)$(BINPATH)/komi
 
 komi:	komi.o
 	$(CC) $(CFLAGS) komi.o -o komi $(SDL_LIB)
