diff -Naur apbs-1.2-source/tools/python/Makefile.am apbs-1.2-source.new/tools/python/Makefile.am
--- apbs-1.2-source/tools/python/Makefile.am	2009-10-14 13:20:57.000000000 -0400
+++ apbs-1.2-source.new/tools/python/Makefile.am	2009-10-29 16:08:18.000000000 -0400
@@ -15,5 +15,5 @@
 
 noinst_PROGRAMS       = _apbslib.so
 _apbslib_so_SOURCES   = apbslib.c
-_apbslib_so_LINK      = $(PY_CC) $(PY_LINKFLAGS) -o $@ $(PY_SHARED)
-_apbslib_so_LDADD     = ${top_builddir}/bin/.libs/libapbsmainroutines.a ${top_builddir}/src/aaa_lib/.libs/libapbs.a ${top_builddir}/contrib/lib/libmaloc.a ${top_builddir}/contrib/blas/.libs/libapbsblas.a $(PY_LDFLAGS)
+_apbslib_so_LINK      = $(PY_CC) $(PY_LINKFLAGS) $(LDFLAGS) -o $@ $(PY_SHARED)
+_apbslib_so_LDADD     = ${top_builddir}/bin/.libs/libapbsmainroutines.a ${top_builddir}/src/aaa_lib/.libs/libapbs.a $(PY_LDFLAGS)
diff -Naur apbs-1.2-source/tools/python/vgrid/Makefile.am apbs-1.2-source.new/tools/python/vgrid/Makefile.am
--- apbs-1.2-source/tools/python/vgrid/Makefile.am	2009-10-14 13:20:57.000000000 -0400
+++ apbs-1.2-source.new/tools/python/vgrid/Makefile.am	2009-10-29 16:08:29.000000000 -0400
@@ -15,6 +15,6 @@
 if PY_PRESENT
    noinst_PROGRAMS     = _vgrid.so
    _vgrid_so_SOURCES   = vgridlib.c
-   _vgrid_so_LINK      = $(PY_CC) $(PY_LINKFLAGS) -o $@ $(PY_SHARED)
-   _vgrid_so_LDADD     = ${top_builddir}/bin/.libs/libapbsmainroutines.a ${top_builddir}/src/aaa_lib/.libs/libapbs.a $(PY_LDFLAGS) ${top_builddir}/contrib/lib/libmaloc.a @lib_list@
+   _vgrid_so_LINK      = $(PY_CC) $(PY_LINKFLAGS) $(LDFLAGS) -o $@ $(PY_SHARED)
+   _vgrid_so_LDADD     = ${top_builddir}/bin/.libs/libapbsmainroutines.a ${top_builddir}/src/aaa_lib/.libs/libapbs.a @lib_list@
 endif
