commit ae943198bf74d563adcb1f3d36ee4ba1b7b274a9
Author: Aleix Pol <aleixpol@kde.org>
Date:   Fri Jul 28 13:21:38 2017 +0200

    Prefer using https for kns providers
    
    CCBUG: 382820

diff --git a/components/shellprivate/widgetexplorer/plasmoids.knsrc b/components/shellprivate/widgetexplorer/plasmoids.knsrc
index c683a257..03c7de0e 100644
--- a/components/shellprivate/widgetexplorer/plasmoids.knsrc
+++ b/components/shellprivate/widgetexplorer/plasmoids.knsrc
@@ -36,7 +36,7 @@ Name[x-test]=xxPlasma Widgetsxx
 Name[zh_CN]=Plasma 部件
 Name[zh_TW]=Plasma 元件
 
-ProvidersUrl=http://download.kde.org/ocs/providers.xml
+ProvidersUrl=https://download.kde.org/ocs/providers.xml
 Categories=Plasma 5 Plasmoid
 StandardResource=tmp
 InstallationCommand=kpackagetool5 --install %f --type Plasma/Applet
diff --git a/wallpapers/image/wallpaper.knsrc b/wallpapers/image/wallpaper.knsrc
index 4a0cf2d6..2decc7e7 100644
--- a/wallpapers/image/wallpaper.knsrc
+++ b/wallpapers/image/wallpaper.knsrc
@@ -36,7 +36,7 @@ Name[x-test]=xxWallpapersxx
 Name[zh_CN]=壁纸
 Name[zh_TW]=桌布
 
-ProvidersUrl=http://download.kde.org/ocs/providers.xml
+ProvidersUrl=https://download.kde.org/ocs/providers.xml
 Categories=KDE Wallpaper 800x600,KDE Wallpaper 1024x768,KDE Wallpaper 1280x1024,KDE Wallpaper 1440x900,KDE Wallpaper 1600x1200,KDE Wallpaper (other)
 StandardResource=wallpaper
 Uncompress=archive
