From bf569560bf195ac4e79d65d4103a7161a6a2f2ac Mon Sep 17 00:00:00 2001
From: Elias Probst <mail@eliasprobst.eu>
Date: Sat, 4 Jul 2015 11:33:01 +0200
Subject: [PATCH] [startkde] Gentoo FHS script support

---
 startkde/startkde.cmake | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/startkde/startkde.cmake b/startkde/startkde.cmake
index 2585600..9350158 100644
--- a/startkde/startkde.cmake
+++ b/startkde/startkde.cmake
@@ -210,6 +210,11 @@ for prefix in `echo $scriptpath`; do
   done
 done
 
+# Gentoo part for FHS installs
+for file in "@GENTOO_PORTAGE_EPREFIX@/etc/plasma/startup/"*.sh; do
+  test -r "${file}" && . "${file}"
+done
+
 # Activate the kde font directories.
 #
 # There are 4 directories that may be used for supplying fonts for KDE.
@@ -417,6 +422,11 @@ for prefix in `echo "$scriptpath"`; do
   done
 done
 
+# Gentoo part for FHS installs
+for file in "@GENTOO_PORTAGE_EPREFIX@/etc/plasma/shutdown/"*.sh; do
+  test -r "${file}" && . "${file}"
+done
+
 unset KDE_FULL_SESSION
 xprop -root -remove KDE_FULL_SESSION
 unset KDE_SESSION_VERSION
-- 
2.3.6

