From 16bdcf9f0ab24b7d57b7e065534b87810beff63d Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Tue, 10 Apr 2018 10:21:29 +0200
Subject: [PATCH] Fix musicbrainz linking

---
 coquillo.pro | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/coquillo.pro b/coquillo.pro
index 63b1cd1..5bbe6a2 100644
--- a/coquillo.pro
+++ b/coquillo.pro
@@ -16,13 +16,7 @@ RCC_DIR = build
 unix {
   LIBS += -ltag
 
-  exists(/etc/arch-release) {
-    LIBS += -lmusicbrainz5
-  }
-
-  exists(/etc/debian_version) {
-    LIBS += -lmusicbrainz5cc
-  }
+  LIBS += -lmusicbrainz5
 }
 
 
-- 
2.17.0

