From 49aa0675d5d84e3286e9879ceb2333cb83f1d824 Mon Sep 17 00:00:00 2001
From: Laurent Montel <montel@kde.org>
Date: Thu, 24 Feb 2022 17:33:56 +0100
Subject: [PATCH] Fix include

---
 .../webengine/articleviewerwebenginewidgetng.cpp                | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/articleviewer-ng/webengine/articleviewerwebenginewidgetng.cpp b/src/articleviewer-ng/webengine/articleviewerwebenginewidgetng.cpp
index 3784894e..20e33120 100644
--- a/src/articleviewer-ng/webengine/articleviewerwebenginewidgetng.cpp
+++ b/src/articleviewer-ng/webengine/articleviewerwebenginewidgetng.cpp
@@ -13,7 +13,7 @@
 #include <MessageViewer/ViewerPluginToolManager>
 #include <QVBoxLayout>
 #if KPIMTEXTEDIT_TEXT_TO_SPEECH
-#include <KPIMTextEdit/kpimtextedit/texttospeechwidget.h>
+#include <kpimtextedit/texttospeechwidget.h>
 #endif
 #include <KPIMTextEdit/SlideContainer>
 
-- 
GitLab

