From 4b4b94f0bafcb49cefb1e453edd04a46d6ceacce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=96mer=20Fad=C4=B1l=20Usta?= <omerusta@gmail.com>
Date: Thu, 12 May 2022 05:01:48 +0000
Subject: [PATCH] Fix build with GCC 12 (standard attributes in middle of
 decl-specifiers)

---
 src/common/KReportDataSource.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/KReportDataSource.h b/src/common/KReportDataSource.h
index 89490b47..0e45b644 100644
--- a/src/common/KReportDataSource.h
+++ b/src/common/KReportDataSource.h
@@ -124,7 +124,7 @@ public:
     //! Creates a new instance with data source. Default implementation returns @c nullptr.
     //! @a source is implementation-specific identifier.
     //! Owner of the returned pointer is the caller.
-    virtual Q_REQUIRED_RESULT KReportDataSource* create(const QString &source) const;
+    Q_REQUIRED_RESULT virtual KReportDataSource* create(const QString &source) const;
 
 private:
     Q_DISABLE_COPY(KReportDataSource)
-- 
GitLab

