diff --git a/Makefile.in b/Makefile.in
index 7a5f1fd..16d7fd8 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -119,7 +119,8 @@ install: $(LIBNEWT) install-sh whiptail
 	install -m 755 whiptail $(DESTDIR)/$(bindir)
 	install -m 644 whiptail.1 $(DESTDIR)/$(man1dir)
 	$(MAKE) -C po datadir=$(DESTDIR)/$(datadir) install
-	install -m 644 -D libnewt.pc $(DESTDIR)/$(pkgconfigdir)/libnewt.pc
+	install -d $(DESTDIR)/$(pkgconfigdir)
+	install -m 644 libnewt.pc $(DESTDIR)/$(pkgconfigdir)/libnewt.pc
 
 install-sh: sharedlib $(WHIPTCLSO) _snackmodule.so
 	[ -d $(DESTDIR)/$(libdir) ] || install -m 755 -d $(DESTDIR)/$(libdir)
