Index: gazebo-11.10.1/gazebo/common/VideoEncoder.cc
===================================================================
--- gazebo-11.10.1.orig/gazebo/common/VideoEncoder.cc
+++ gazebo-11.10.1/gazebo/common/VideoEncoder.cc
@@ -224,7 +224,10 @@ bool VideoEncoder::Start(const std::stri
 
   // The remainder of this function handles FFMPEG initialization of a video
   // stream
-  const AVOutputFormat *outputFormat = nullptr;
+#if LIBAVFORMAT_VERSION_MAJOR >= 59
+  const
+#endif
+  AVOutputFormat *outputFormat = nullptr;
 
   // This 'if' and 'free' are just for safety. We chech the value of formatCtx
   // below.
