From: Felipe Sateler <fsateler@debian.org>
Date: Thu, 5 Apr 2018 15:44:26 -0300
Subject: Use the pyqt5 dbus mainloop integration

Forwarded: https://bugs.freedesktop.org/show_bug.cgi?id=102572
---
 src/utils/qpaeq | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/utils/qpaeq b/src/utils/qpaeq
index 508b233..a319dad 100755
--- a/src/utils/qpaeq
+++ b/src/utils/qpaeq
@@ -20,7 +20,7 @@ import os,math,sys
 try:
     import PyQt5,sip
     from PyQt5 import QtWidgets,QtCore
-    import dbus.mainloop.qt
+    import dbus.mainloop.pyqt5
     import dbus
 except ImportError as e:
     sys.stderr.write('There was an error importing needed libraries\n'
@@ -565,7 +565,7 @@ def subdivide(xs, t_points):
     return left+right
 
 def main():
-    dbus.mainloop.qt.DBusQtMainLoop(set_as_default=True)
+    dbus.mainloop.pyqt5.DBusQtMainLoop(set_as_default=True)
     app=QtWidgets.QApplication(sys.argv)
     qpaeq_main=QPaeq()
     qpaeq_main.show()
