--- libnss-mysql-1.5/acinclude.m4
+++ libnss-mysql-1.5.sal/acinclude.m4
@@ -80,6 +80,18 @@
     fi
 done
 
+dnl Test also $libdir path
+if test -f "$libdir/libmysqlclient.so"
+then
+    MYSQL_LIB_DIR="$libdir"
+    break
+fi
+if test -f "$libdir/mysql/libmysqlclient.so"
+then
+    MYSQL_LIB_DIR="$libdir/mysql"
+    break
+fi
+
 if test -n "$MYSQL_LIB_DIR"
 then
     AC_MSG_RESULT([$MYSQL_LIB_DIR])
