https://bugs.gentoo.org/show_bug.cgi?id=266591

--- OS/Makefile-Base
+++ OS/Makefile-Base
@@ -114,8 +114,8 @@
 
 # Targets for special-purpose configuration header builders
 buildconfig: buildconfig.c
-	@echo "$(CC) buildconfig.c"
-	$(FE)$(CC) $(CFLAGS) $(INCLUDE) -o buildconfig buildconfig.c $(LIBS)
+	@echo "$(HOSTCC) buildconfig.c"
+	$(FE)$(HOSTCC) $(HOSTCFLAGS) $(INCLUDE) -o buildconfig buildconfig.c $(LIBS)
 
 
 # Target for the exicyclog utility script
