diff -Nru bfilter-1.1.4.orig/configure.in bfilter-1.1.4/configure.in
--- bfilter-1.1.4.orig/configure.in	2007-11-11 23:14:49.000000000 +0100
+++ bfilter-1.1.4/configure.in	2009-10-18 18:26:32.000000000 +0200
@@ -260,11 +260,6 @@
 Makefile
 binreloc/Makefile
 foundation/Makefile
-boost/Makefile
-boost/libs/Makefile
-boost/libs/regex/Makefile
-boost/libs/program_options/Makefile
-boost/libs/test/Makefile
 reactor/Makefile
 mkskel/Makefile
 lexgen/Makefile
diff -Nru bfilter-1.1.4.orig/main/daemon/Makefile.am bfilter-1.1.4/main/daemon/Makefile.am
--- bfilter-1.1.4.orig/main/daemon/Makefile.am	2007-01-19 20:21:02.000000000 +0100
+++ bfilter-1.1.4/main/daemon/Makefile.am	2009-10-18 18:25:52.000000000 +0200
@@ -6,7 +6,7 @@
 $(top_builddir)/boost/libs/program_options/libprogram_options.la
 else
 BOOST_CPPFLAGS =
-BOOST_LIBS     =
+BOOST_LIBS     = -lboost_regex-mt -lboost_program_options-mt
 endif
 
 AM_CPPFLAGS = -DSYSCONFDIR=\"$(sysconfdir)\" \
diff -Nru bfilter-1.1.4.orig/main/gui/gtk/Makefile.am bfilter-1.1.4/main/gui/gtk/Makefile.am
--- bfilter-1.1.4.orig/main/gui/gtk/Makefile.am	2008-06-02 17:25:27.000000000 +0200
+++ bfilter-1.1.4/main/gui/gtk/Makefile.am	2009-10-18 18:25:52.000000000 +0200
@@ -50,7 +50,7 @@
 BOOST_LIBS     = $(top_builddir)/boost/libs/regex/libregex.la
 else
 BOOST_CPPFLAGS =
-BOOST_LIBS     =
+BOOST_LIBS     = -lboost_regex-mt
 endif
 
 if WITH_GUI
diff -Nru bfilter-1.1.4.orig/Makefile.am bfilter-1.1.4/Makefile.am
--- bfilter-1.1.4.orig/Makefile.am	2007-11-11 23:16:04.000000000 +0100
+++ bfilter-1.1.4/Makefile.am	2009-10-18 18:26:46.000000000 +0200
@@ -1,7 +1,7 @@
 EXTRA_DIST = doc cmake bootstrap m4 VERSION CMakeLists.txt types.h pstdint.h \
 config.h.cm bfilter.desktop bfilter.png Doxyfile
 DISTCLEANFILES = itypes.h
-SUBDIRS = binreloc foundation boost reactor mkskel lexgen libjs main conf \
+SUBDIRS = binreloc foundation reactor mkskel lexgen libjs main conf \
 packaging tests
 
 dist_man_MANS = bfilter.8
diff -Nru bfilter-1.1.4.orig/tests/Makefile.am bfilter-1.1.4/tests/Makefile.am
--- bfilter-1.1.4.orig/tests/Makefile.am	2007-05-07 16:19:40.000000000 +0200
+++ bfilter-1.1.4/tests/Makefile.am	2009-10-18 18:25:52.000000000 +0200
@@ -6,7 +6,7 @@
                  $(top_builddir)/boost/libs/regex/libregex.la
 else
 BOOST_CPPFLAGS =
-BOOST_LIBS     =
+BOOST_LIBS     = -lboost_regex-mt -lboost_unit_test_framework-mt
 endif
 
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_builddir) \
