--- madplay-0.15.2b/audio_carbon.c.bak	2005-04-04 02:03:58.000000000 -0700
+++ madplay-0.15.2b/audio_carbon.c	2005-04-04 02:04:39.000000000 -0700
@@ -94,7 +94,7 @@
 }
 
 static
-int wait(struct buffer *buffer)
+int carbon_wait(struct buffer *buffer)
 {
   if (MPWaitOnSemaphore(buffer->semaphore, kDurationForever) != noErr) {
     audio_error = _("MPWaitOnSemaphore() failed");
@@ -263,7 +263,7 @@
   /* wait for block to finish playing */
 
   if (buffer->pcm_nsamples == 0) {
-    if (wait(buffer) == -1)
+    if (carbon_wait(buffer) == -1)
       return -1;
 
     buffer->pcm_length = 0;
