diff -Naur x2goserver-4.0.1.18.orig/x2goserver/bin/x2golistdesktops x2goserver-4.0.1.18/x2goserver/bin/x2golistdesktops
--- x2goserver-4.0.1.18.orig/x2goserver/bin/x2golistdesktops	2014-10-06 23:23:27.000000000 +0200
+++ x2goserver-4.0.1.18/x2goserver/bin/x2golistdesktops	2015-01-14 10:52:55.044479499 +0100
@@ -60,7 +60,15 @@
 $rdisp="I${rdisp}I";
 
 my $uname=$ENV{'USER'};
+
+# on Debian the ss command is in /usr/bin, on Fedora, it is in /usr/sbin...
+# -> so we need to tweak the PATH env var a little here...
+my $old_PATH=$ENV{'PATH'};
+$ENV{'PATH'}="$ENV{'PATH'}:/usr/sbin:/sbin";
 my $lines=`ss -lxu`;
+$ENV{'PATH'}=$old_PATH;
+undef $old_PATH;
+
 my @lines=split("\n", "$lines");
 
 my @outp = grep ( /(@| )\/tmp\/.X11-unix\/X.*/, @lines);
