--- a/build.gradle
+++ b/build.gradle
@@ -1309,6 +1309,7 @@ project(":graphics") {
         antlr3 group: "org.antlr", name: "antlr", version: "3.1.3"
         antlr3 group: "org.antlr", name: "antlr-runtime",  version: "3.1.3"
         antlr3 group: "org.antlr", name: "stringtemplate", version: "3.2"
+        antlr3 group: "antlr", name: "antlr", version: "2.7.7"
     }
 
     // Create a single "native" task which will depend on all the individual native tasks for graphics
--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -82,10 +82,13 @@ getConfigurations().create("antlr3");
 
 dependencies {
     compile     group: "org.antlr", name: "antlr",          version: "3.1.3"
+    compile     group: "org.antlr", name: "antlr-runtime",  version: "3.1.3"
+    compile     group: "org.antlr", name: "stringtemplate", version: "3.2"
     testCompile group: "junit",     name: "junit",          version: "4.8.2"
     antlr3      group: "org.antlr", name: "antlr-runtime",  version: "3.1.3"
     antlr3      group: "org.antlr", name: "stringtemplate", version: "3.2"
     antlr3      group: "org.antlr", name: "antlr",          version: "3.1.3"
+    antlr3      group: "antlr",     name: "antlr",          version: "2.7.7"
 }
 
 // At the moment the ASM library shipped with Gradle that is used to
