--- a/lib/libwebsockets.c	2017-02-15 18:03:12.000000000 +0800
+++ b/lib/libwebsockets.c	2017-03-17 10:03:56.635474181 +0800
@@ -642,7 +642,7 @@
 #ifdef LWS_USE_IPV6
 	if (LWS_IPV6_ENABLED(vh)) {
 		if (!lws_plat_inet_ntop(AF_INET6, &((struct sockaddr_in6 *)ads)->sin6_addr, rip, rip_len)) {
-			lwsl_err("inet_ntop", strerror(LWS_ERRNO));
+			lwsl_err("inet_ntop: %s", strerror(LWS_ERRNO));
 			return -1;
 		}
 
