--- a/htb.init
+++ b/htb.init
@@ -473,8 +473,8 @@
 	### List of classes in $HTB_PATH
 	htb_class_list () {
 		for dev in `htb_device_list`; do
-			find $HTB_PATH \( -type f -or -type l \) \
-			-name "$dev-*" -not -name '*~' -maxdepth 1 \
+			find $HTB_PATH -maxdepth 1 \( -type f -or -type l \) \
+			-name "$dev-*" -not -name '*~' \
 			-printf "%f\n"| sort
 		done
 	} # htb_class_list
@@ -488,8 +488,8 @@
 	### Validate cache against config files
 	htb_valid_cache () {
 		for dev in `htb_device_list`; do
-			[ `find $HTB_PATH \( -type f -or -type l \) \
-			  -name "$dev*" -maxdepth 1 -newer $HTB_CACHE| \
+			[ `find $HTB_PATH -maxdepth 1 \( -type f -or -type l \) \
+			  -name "$dev*" -newer $HTB_CACHE| \
 			  wc -l` -gt 0 ] && VALID=0
 			[ $VALID -ne 1 ] && break
 		done
@@ -498,7 +498,7 @@
 
 	### Find class config for device $1, which is newer than cache
 	htb_cache_older () {
-		[ `find $HTB_PATH -type f -name "$1*" -maxdepth 1 \
+		[ `find $HTB_PATH -maxdepth 1 -type f -name "$1*" \
 		   -newer $HTB_CACHE| wc -l` -gt 0 ] && return 0
 		return 1
 	} # htb_cache_older
