From d8cb9d52a7e7e7e20cd371349583794e17fd0778 Mon Sep 17 00:00:00 2001
From: Alon Bar-Lev <alon.barlev@gmail.com>
Date: Thu, 1 Dec 2016 20:50:35 +0200
Subject: [PATCH] utils: safeint.h: use botan's to_string

Signed-off-by: Alon Bar-Lev <alon.barlev@gmail.com>
---
 src/utils/safeint.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/safeint.h b/src/utils/safeint.h
index e0bd662..0a03e4e 100644
--- a/src/utils/safeint.h
+++ b/src/utils/safeint.h
@@ -17,7 +17,7 @@ class Integer_Overflow_Detected : public Exception
    {
    public:
       Integer_Overflow_Detected(const std::string& file, int line) :
-         Exception("Integer overflow detected at " + file + ":" + std::to_string(line))
+         Exception("Integer overflow detected at " + file + ":" + to_string(line))
          {}
    };
 
-- 
2.7.3

