From 5bb0b47e608e083dda5e39132174b840f3b091cf Mon Sep 17 00:00:00 2001
From: Alon Bar-Lev <alon.barlev@gmail.com>
Date: Tue, 15 Sep 2015 00:18:19 +0300
Subject: [PATCH] build: support multiple python versions

Signed-off-by: Alon Bar-Lev <alon.barlev@gmail.com>
---
 src/scripts/install.py | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/scripts/install.py b/src/scripts/install.py
index 2c69213..d1d62b6 100755
--- a/src/scripts/install.py
+++ b/src/scripts/install.py
@@ -181,11 +181,12 @@ def main(args = None):
                   os.path.join(pkgconfig_dir, os.path.basename(cfg['botan_pkgconfig'])))
 
     if 'ffi' in cfg['mod_list'].split('\n'):
-        py_lib_path = os.path.join(lib_dir, 'python%s' % (cfg['python_version']), 'site-packages')
-        logging.debug('Installing python module to %s' % (py_lib_path))
-        makedirs(py_lib_path)
-        for py in ['botan.py']:
-            copy_file(os.path.join(cfg['python_dir'], py), os.path.join(py_lib_path, py))
+        for ver in cfg['python_version'].split(','):
+            py_lib_path = os.path.join(lib_dir, 'python%s' % (ver), 'site-packages')
+            logging.debug('Installing python module to %s' % (py_lib_path))
+            makedirs(py_lib_path)
+            for py in ['botan.py']:
+                copy_file(os.path.join(cfg['python_dir'], py), os.path.join(py_lib_path, py))
 
     shutil.rmtree(target_doc_dir, True)
     shutil.copytree(cfg['doc_output_dir'], target_doc_dir)
-- 
2.4.6

