--- a/mailx-8.1.2-0.20050715cvs.orig/EXT/vis.h
+++ b/mailx-8.1.2-0.20050715cvs.orig/EXT/vis.h
@@ -70,9 +70,6 @@
  */
 #define	UNVIS_END	1	/* no more characters */
 
-#include <sys/cdefs.h>
-
-__BEGIN_DECLS
 char	*vis(char *, int, int, int);
 int	strvis(char *, const char *, int);
 int	strnvis(char *, const char *, size_t, int)
@@ -84,6 +81,4 @@ int	unvis(char *, char, int *, int);
 ssize_t strnunvis(char *, const char *, size_t)
 /*		__attribute__ ((__bounded__(__string__,1,3)))*/;
 
-__END_DECLS
-
 #endif /* !_VIS_H_ */
