--- a/configure.ac	2016-08-31 12:49:35.000000000 +0000
+++ b/configure.ac	2017-03-04 00:58:02.354372332 +0000
@@ -29,12 +29,12 @@
 
 adolclib=adolc
 
-if test "x$CFLAGS" == x ; then
+if test "x$CFLAGS" = x ; then
     ac_shell_cflags="not_set"
 else
     ac_shell_cflags="$CFLAGS"
 fi
-if test "x$CXXFLAGS" == x ; then
+if test "x$CXXFLAGS" = x ; then
     ac_shell_cxxflags="not_set"
 else
     ac_shell_cxxflags="$CXXFLAGS"
@@ -231,12 +231,12 @@
    AC_MSG_RESULT(no)
    ])
 
-if test x$adolc_harddebug == xyes ; then
+if test x$adolc_harddebug = xyes ; then
 adolc_debug=yes
 AC_DEFINE(ADOLC_HARDDEBUG,1,[ADOL-C hard debug mode])
 fi
 
-if test x$adolc_debug == xyes ; then
+if test x$adolc_debug = xyes ; then
 AC_DEFINE(ADOLC_DEBUG,1,[ADOL-C debug mode])
 fi
 
@@ -254,7 +254,7 @@
   AC_MSG_RESULT(yes)
   ])
 
-if test x$adolc_stdczero == xyes; then
+if test x$adolc_stdczero = xyes; then
 AC_DEFINE(ADOLC_ADOUBLE_STDCZERO,1,[ADOL-C adouble zeroing mode])
 fi
 
@@ -271,7 +271,7 @@
   AC_MSG_RESULT(no)
   ])
 
-if test x$adolc_late_init == xyes; then
+if test x$adolc_late_init = xyes; then
 AC_DEFINE(ADOLC_ADOUBLE_LATEINIT,1,[ADOL-C adouble late initialization mode])
 fi
 
@@ -288,7 +288,7 @@
   AC_MSG_RESULT(no)
   ])
 
-if test x$adolc_tserrno == xyes ; then
+if test x$adolc_tserrno = xyes ; then
 AC_DEFINE(ADOLC_THREADSAVE_ERRNO,1,[ADOL-C thread save errno mode])
 fi
 
@@ -376,8 +376,8 @@
             [AS_HELP_STRING([--with-cflags=FLAGS],
                             [use CFLAGS=FLAGS (default: -O2)])],
             [ac_adolc_cflags="$withval"
-             if test x${adolc_debug} == xyes ||
-                test x${adolc_harddebug} == xyes ; then
+             if test "x${adolc_debug}" = xyes ||
+                test "x${adolc_harddebug}" = xyes ; then
                 ac_adolc_cflags="$ac_adolc_cflags -g -O0"
              fi
              ac_adolc_cflags="$ac_adolc_cflags $ac_adolc_openmpflag"
@@ -385,8 +385,8 @@
              AC_MSG_RESULT($ac_adolc_cflags)],
             [if test "$ac_shell_cflags" = "not_set"; then
                 ac_adolc_cflags="-O2"
-                if test x${adolc_debug} == xyes ||
-                   test x${adolc_harddebug} == xyes ; then
+                if test "x${adolc_debug}" = xyes ||
+                   test "x${adolc_harddebug}" = xyes ; then
                    ac_adolc_cflags="-g -O0 -Wall -ansi"
                 fi
              else
@@ -402,8 +402,8 @@
             [AS_HELP_STRING([--with-cxxflags=FLAGS],
                             [use CXXFLAGS=FLAGS (default: -O2)])],
             [ac_adolc_cxxflags="$withval"
-             if test x${adolc_debug} == xyes ||
-                test x${adolc_harddebug} == xyes ; then
+             if test x${adolc_debug} = xyes ||
+                test x${adolc_harddebug} = xyes ; then
                 ac_adolc_cxxflags="$ac_adolc_cxxflags -g -O0"
              fi
              ac_adolc_cxxflags="$ac_adolc_cxxflags $ac_adolc_openmpflag"
@@ -411,8 +411,8 @@
              AC_MSG_RESULT($ac_adolc_cxxflags)],
             [if test "$ac_shell_cxxflags" = "not_set"; then
                 ac_adolc_cxxflags="-O2"
-                if test x${adolc_debug} == xyes ||
-                   test x${adolc_harddebug} == xyes ; then
+                if test x${adolc_debug} = xyes ||
+                   test x${adolc_harddebug} = xyes ; then
                    ac_adolc_cxxflags="-g -O0 -Wall"
                 fi
              else
@@ -445,7 +445,7 @@
   ])
 AC_MSG_RESULT($tapedoc_values)
 
-if test x$tapedoc_values == xyes; then
+if test x$tapedoc_values = xyes; then
 AC_DEFINE(ADOLC_TAPE_DOC_VALUES,1,[ADOL-C tape_doc routine computes values])
 fi
 
@@ -614,7 +614,7 @@
         break
     fi
 done
-if test "x${pdir}" == "x/" ; then
+if test "x${pdir}" = "x/" ; then
     writable=false
 fi
 if ! ${writable} ; then
@@ -626,15 +626,6 @@
     warn=true
 fi
 
-case ${prefix} in 
-   ${HOME}|${HOME}/*)
-if $warn ; then
-    for (( i=5; i>0; i-- )); do
-        echo -ne "\rPlease read the above messages carefully, continuing in ${i} seconds"
-        sleep 1
-    done
-    echo -e "\rPlease read the above messages carefully                                 "
-fi
 echo \
 "
 *****************************************************************************
@@ -660,8 +651,6 @@
 
 *****************************************************************************
 "
-;;
-esac
 if test "x$enable_static" != "xno" ; then
 echo \
 " Linking a user program with a static ADOL-C library will always result
