diff --git a/bin/cask b/bin/cask
index 55d4d86..37a5c9c 100755
--- a/bin/cask
+++ b/bin/cask
@@ -39,17 +39,7 @@ case $subcommand in
 	*)
 		WHICH=${WHICH:-which}
 		READLINK=$($WHICH readlink || true)
-		if [ "$(uname || true)" == "Darwin" ] ; then
-			READLINK=
-		fi
-		READLINK=${READLINK:-$($WHICH greadlink || true)}
-		if [ ! -z "$READLINK" ] ; then
-			SRCDIR__=$($READLINK -f "$CASK")
-		else
-			SRCDIR__=$(python -c "import os, sys; print(os.path.realpath(sys.argv[1]))" "$CASK")
-		fi
-		SRCDIR_=$(dirname "$SRCDIR__")
-		SRCDIR=$(dirname "$SRCDIR_")
-		"$EMACS" -Q --script "$SRCDIR/cask-cli.el" -- $subcommand "$@"
+		SRCDIR="@SITELISP@"
+		"$EMACS" -q --script "$SRCDIR/cask-cli.el" -- $subcommand "$@"
 		;;
 esac
