--- a/Makefile.in
+++ b/Makefile.in
@@ -69,9 +69,8 @@
 SYSCONFIG_DIR = @SYSCONFIG_DIR@
 
 # Compiler flags
-CFLAGS = @CFLAGS@ -Wall -Wstrict-prototypes -pipe -O2
+CFLAGS = @CFLAGS@ -Wall -Wstrict-prototypes 
 DFLAGS = @DFLAGS@
-LFLAGS = @STRIP@
 DFLAGS += -DSA_DIR=\"$(SA_DIR)\" -DSADC_PATH=\"$(SADC_PATH)\"
 DFLAGS += $(DFSENSORS)
 
@@ -154,7 +153,7 @@
 	$(CC) -o $@ -c $(CFLAGS) $(DFLAGS) $<
 
 % : %.o
-	$(CC) -o $@ $(CFLAGS) $^ $(LFLAGS)
+	$(CC) -o $@ $(CFLAGS) $(LFLAGS) $^ $(LIBS)
 
 all: sadc sar sadf iostat mpstat pidstat nfsiostat cifsiostat locales
 
@@ -196,7 +195,7 @@
 
 sadc.o: sadc.c sa.h version.h common.h ioconf.h sysconfig.h rd_stats.h rd_sensors.h
 
-sadc: LFLAGS += $(LFSENSORS)
+sadc: LIBS = $(LFSENSORS)
 sadc: sadc.o act_sadc.o sa_wrap.o sa_common.o librdstats.a librdsensors.a libsyscom.a
 
 sar.o: sar.c sa.h version.h common.h ioconf.h pr_stats.h sysconfig.h
