--- src/graphics/C4SurfaceLoaders.cpp.old	2015-01-02 18:01:35.768676874 +0100
+++ src/graphics/C4SurfaceLoaders.cpp	2015-01-02 18:02:17.705896683 +0100
@@ -303,7 +303,7 @@
 	// The doc says to give fake end-of-inputs if there is no more data
 	cinfo->src->next_input_byte = &end_of_input;
 	cinfo->src->bytes_in_buffer = 1;
-	return true;
+	return (boolean)true;
 }
 static void skip_input_data (j_decompress_ptr cinfo, long num_bytes)
 {
@@ -354,7 +354,7 @@
 	blub.term_source = jpeg_noop;
 
 	// a missing image is an error
-	jpeg_read_header(&cinfo, true);
+	jpeg_read_header(&cinfo, (boolean)true);
 
 	// Let libjpeg convert for us
 	cinfo.out_color_space = JCS_RGB;
