From 24c7ca17eaf69211b75f2882c11aeff01a3cad01 Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <leio@gentoo.org>
Date: Sat, 8 Aug 2020 18:59:25 +0300
Subject: [PATCH] I#158 - Workaround build error with gtk-doc-1.32

---
 src/calendar/libecal/e-cal-time-util.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/calendar/libecal/e-cal-time-util.h b/src/calendar/libecal/e-cal-time-util.h
index 8acb13061..11c9d3369 100644
--- a/src/calendar/libecal/e-cal-time-util.h
+++ b/src/calendar/libecal/e-cal-time-util.h
@@ -150,6 +150,7 @@ void		time_to_gdate_with_zone	(GDate *date,
  * struct tm manipulation
  **************************************************************************/
 
+#ifndef __GTK_DOC_IGNORE__
 struct tm	e_cal_util_icaltime_to_tm
 					(const ICalTime *itt);
 struct tm	e_cal_util_icaltime_to_tm_with_zone
@@ -159,6 +160,7 @@ struct tm	e_cal_util_icaltime_to_tm_with_zone
 ICalTime *	e_cal_util_tm_to_icaltime
 					(struct tm *tm,
 					 gboolean is_date);
+#endif
 
 G_END_DECLS
 
-- 
2.20.1

