diff --git a/CMakeLists.txt b/CMakeLists.txt
index bea0af828..f72304f05 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,55 +149,8 @@ IF (ENABLE_LUAJIT MATCHES "ON")
 ELSE (ENABLE_LUAJIT MATCHES "ON")
 
     ProcessPackage(LIBLUA LIBRARY "lua"
-            "lua-5.3"
-            LIB_SUFFIXES "lua5.3"
-            INCLUDE lua.h INCLUDE_SUFFIXES
-            "include/lua-5.3"
-            "include/lua5.3"
-            "include/lua53"
-            "include/lua"
             ROOT ${LUA_ROOT}
-            MODULES lua53
-            OPTIONAL)
-
-    IF (NOT WITH_LIBLUA)
-        ProcessPackage(LIBLUA LIBRARY "lua"
-                "lua-5.4"
-                LIB_SUFFIXES "lua5.4"
-                INCLUDE lua.h INCLUDE_SUFFIXES
-                "include/lua-5.4"
-                "include/lua5.4"
-                "include/lua54"
-                "include/lua"
-                ROOT ${LUA_ROOT}
-                MODULES lua54
-                OPTIONAL)
-        IF (NOT WITH_LIBLUA)
-            ProcessPackage(LIBLUA LIBRARY "lua"
-                    "lua-5.2"
-                    LIB_SUFFIXES "lua5.2"
-                    INCLUDE lua.h INCLUDE_SUFFIXES
-                    "include/lua-5.2"
-                    "include/lua5.2"
-                    "include/lua52"
-                    "include/lua"
-                    ROOT ${LUA_ROOT}
-                    MODULES lua52
-                    OPTIONAL)
-
-            IF (NOT WITH_LIBLUA)
-                ProcessPackage(LIBLUA LIBRARY "lua"
-                        "lua-5.1"
-                        INCLUDE lua.h INCLUDE_SUFFIXES
-                        "include/lua-5.1"
-                        "include/lua5.1"
-                        "include/lua51"
-                        "include/lua"
-                        ROOT ${LUA_ROOT}
-                        MODULES lua51)
-            ENDIF ()
-        ENDIF ()
-    ENDIF ()
+            MODULES lua)
 ENDIF (ENABLE_LUAJIT MATCHES "ON")
 
 IF (ENABLE_JEMALLOC MATCHES "ON" AND NOT SANITIZE)
-- 
2.41.0

