--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -139,40 +139,8 @@
 ELSE(ENABLE_LUAJIT MATCHES "ON")
 
 	ProcessPackage(LIBLUA LIBRARY	"lua"
-			"lua-5.3"
-			INCLUDE lua.h INCLUDE_SUFFIXES
-			"include/lua-5.3"
-			"include/lua5.3"
-			"include/lua53"
-			"include/lua"
 			ROOT ${LUA_ROOT}
-			MODULES lua53
-			OPTIONAL)
-
-	IF (NOT WITH_LIBLUA)
-		ProcessPackage(LIBLUA LIBRARY	"lua"
-				"lua-5.2"
-				INCLUDE lua.h INCLUDE_SUFFIXES
-				"include/lua-5.2"
-				"include/lua5.2"
-				"include/lua52"
-				"include/lua"
-				ROOT ${LUA_ROOT}
-				MODULES lua52
-				OPTIONAL)
-
-		IF (NOT WITH_LIBLUA)
-			ProcessPackage(LIBLUA LIBRARY	"lua"
-					"lua-5.1"
-					INCLUDE lua.h INCLUDE_SUFFIXES
-					"include/lua-5.1"
-					"include/lua5.1"
-					"include/lua51"
-					"include/lua"
-					ROOT ${LUA_ROOT}
-					MODULES lua51)
-		ENDIF()
-	ENDIF()
+			MODULES lua)
 ENDIF(ENABLE_LUAJIT MATCHES "ON")
 
 IF(ENABLE_JEMALLOC MATCHES "ON" AND NOT SANITIZE)
