diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 502f097..35b8660 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -6,6 +6,7 @@ set( HEADERS
   "qwt_color_map.h"
   "qwt_clipper.h"
   "qwt_double_interval.h"
+  "qwt_double_rect.h"
   "qwt_dyngrid_layout.h"
   "qwt_global.h"
   "qwt_layout_metrics.h"
@@ -34,6 +35,7 @@ set( SRCS
   "qwt_color_map.cpp"
   "qwt_clipper.cpp"
   "qwt_double_interval.cpp"
+  "qwt_double_rect.cpp"
   "qwt_dyngrid_layout.cpp"
   "qwt_layout_metrics.cpp"
   "qwt_math.cpp"
@@ -156,6 +158,7 @@ if( QWT_WIDGETS )
 endif()
 
 add_library( qwt5-qt5 ${SRCS} ${HEADERS} )
+set_target_properties( qwt5-qt5 PROPERTIES PUBLIC_HEADER "${HEADERS}" )
 
 target_link_libraries( qwt5-qt5 Qt5::Core Qt5::Gui Qt5::Widgets Qt5::PrintSupport )
 
