--- configure.ac.orig	2011-05-15 21:40:14.000000000 +0300
+++ configure.ac	2011-05-15 21:46:11.000000000 +0300
@@ -138,6 +138,9 @@
 ac_gsm_codec=false
 ac_spx_codec=false
 
+AC_ARG_ENABLE(gsm, [ --disable-gsm          disable gsm support])
+if test "$enable_gsm" != "no" ; then
+
 AC_CHECK_HEADER(gsm/gsm.h, [
     ac_gsm_codec=true
     AC_DEFINE(HAVE_GSM_GSM_H, [1], [gsm header found])
@@ -145,9 +148,11 @@
 ],[
     AC_CHECK_HEADER(gsm.h, [
         AUDIO_LIBS="$AUDIO_LIBS -lgsm"
-        AC_DEFINE(HAVE_GSM_H, [1], [gsm default header])
+	AC_DEFINE(HAVE_GSM_H, [1], [gsm default header])
+
     ])
 ])
+fi
 
 AC_ARG_ENABLE(speex, [ --disable-speex          disable speex support])
 if test "$enable_speex" != "no" ; then
