diff -Nru src/modules.orig/Makefile.in src/modules/Makefile.in
--- src/modules.orig/Makefile.in	2006-05-20 01:28:11.000000000 +0300
+++ src/modules/Makefile.in	2006-05-20 01:29:11.827604500 +0300
@@ -110,11 +110,11 @@
 
 passwd_pgsql.so: passwd_sql.c ../modules.h ../oops.h
 	${CC} ${CFLAGS} ${CDEFS} -D__PGSQL -c passwd_sql.c -o passwd_pgsql.o
-	${LD} $(SOFLAGS) $(LDFLAGS) $(LIBS) -o passwd_pgsql.so passwd_pgsql.o $(PGSQL_LIB)
+	${LD} $(SOFLAGS) $(LDFLAGS) -o passwd_pgsql.so passwd_pgsql.o $(LIBS) $(PGSQL_LIB)
 
 passwd_mysql.so: passwd_sql.c ../modules.h ../oops.h
 	${CC} ${CFLAGS} ${CDEFS} -D__MYSQL -c passwd_sql.c -o passwd_mysql.o
-	${LD} $(SOFLAGS) $(LDFLAGS) $(LIBS) -o passwd_mysql.so passwd_mysql.o $(MYSQL_LIB)
+	${LD} $(SOFLAGS) $(LDFLAGS) -o passwd_mysql.so passwd_mysql.o $(LIBS) $(MYSQL_LIB)
 
 passwd_pgsql.o: passwd_sql.c ../modules.h ../oops.h
 	${CC} ${CFLAGS} ${CDEFS} -D__PGSQL -c passwd_sql.c -o passwd_pgsql.o
