diff -ru opensmtpd-5.4.5p2/configure opensmtpd-5.4.5p2-fixed/configure
--- opensmtpd-5.4.5p2/configure	2015-04-29 22:53:03.000000000 +0200
+++ opensmtpd-5.4.5p2-fixed/configure	2015-05-27 14:20:48.977763792 +0200
@@ -19667,7 +19667,7 @@
 	if(fd == NULL)
 		exit(1);
 
-	if ((rc = fprintf(fd ,"%s\n", _EVENT_VERSION)) <0)
+	if ((rc = fprintf(fd ,"%s\n", LIBEVENT_VERSION)) <0)
 		exit(1);
 
 	exit(0);
@@ -19794,7 +19794,7 @@
 main ()
 {
 
-	exit(strcmp(event_get_version(), _EVENT_VERSION) == 0 ? 0 : 1);
+	exit(strcmp(event_get_version(), LIBEVENT_VERSION) == 0 ? 0 : 1);
 
   ;
   return 0;
diff -ru opensmtpd-5.4.5p2/configure.ac opensmtpd-5.4.5p2-fixed/configure.ac
--- opensmtpd-5.4.5p2/configure.ac	2015-04-29 22:52:18.000000000 +0200
+++ opensmtpd-5.4.5p2-fixed/configure.ac	2015-05-27 14:21:00.868764207 +0200
@@ -2697,7 +2697,7 @@
 	if(fd == NULL)
 		exit(1);
 
-	if ((rc = fprintf(fd ,"%s\n", _EVENT_VERSION)) <0)
+	if ((rc = fprintf(fd ,"%s\n", LIBEVENT_VERSION)) <0)
 		exit(1);
 
 	exit(0);
@@ -2773,7 +2773,7 @@
 #include <stdlib.h>
 #include <string.h>
 	]], [[
-	exit(strcmp(event_get_version(), _EVENT_VERSION) == 0 ? 0 : 1);
+	exit(strcmp(event_get_version(), LIBEVENT_VERSION) == 0 ? 0 : 1);
 	]])],
 	[
 		AC_MSG_RESULT([yes])
