diff -ru opensmtpd-5.7.3p2/openbsd-compat/arc4random.c opensmtpd-5.7.3p2-fixed/openbsd-compat/arc4random.c
--- opensmtpd-5.7.3p2/openbsd-compat/arc4random.c	2016-02-02 08:40:06.000000000 +0100
+++ opensmtpd-5.7.3p2-fixed/openbsd-compat/arc4random.c	2016-03-02 17:34:24.535030362 +0100
@@ -188,6 +188,7 @@
 	_ARC4_UNLOCK();
 }
 
+#if 0
 u_int32_t
 arc4random(void)
 {
@@ -198,12 +199,13 @@
 	_ARC4_UNLOCK();
 	return val;
 }
+#endif
 
 /*
  * If we are providing arc4random, then we can provide a more efficient 
  * arc4random_buf().
  */
-# ifndef HAVE_ARC4RANDOM_BUF
+#if 0
 void
 arc4random_buf(void *buf, size_t n)
 {
@@ -215,7 +217,7 @@
 #endif /* !HAVE_ARC4RANDOM */
 
 /* arc4random_buf() that uses platform arc4random() */
-#if !defined(HAVE_ARC4RANDOM_BUF) && defined(HAVE_ARC4RANDOM)
+#if 0
 void
 arc4random_buf(void *_buf, size_t n)
 {
