From: Bernd Waibel <waebbl-gentoo@posteo.net>
Date: Fri, 30 Jul 2021 17:33:47 +0200
Subject: [PATCH] fix to build with gcc-11

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>
---
 lib/dpx/dpx.hh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/dpx/dpx.hh b/lib/dpx/dpx.hh
index 02ae833..5639aaa 100644
--- a/lib/dpx/dpx.hh
+++ b/lib/dpx/dpx.hh
@@ -297,7 +297,7 @@ struct dpx {
 		compliance_e current_compliance;
 		endian_mode_e current_endian_mode;
 
-		std::istream::streampos header_start;
+		std::streampos header_start;
 
 	public:
 		virtual ~dpx();
-- 
2.32.0

