From 57f951d1e32739ddfd0d924df81da010e47d35b9 Mon Sep 17 00:00:00 2001
From: David Seifert <soap@gentoo.org>
Date: Sun, 23 Jan 2022 19:09:25 +0100
Subject: [PATCH] Add missing #include for musl

Bug: https://bugs.gentoo.org/719920

Signed-off-by: David Seifert <soap@gentoo.org>
---
 src/gp_socket.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gp_socket.c b/src/gp_socket.c
index 9070928..493399a 100644
--- a/src/gp_socket.c
+++ b/src/gp_socket.c
@@ -10,6 +10,7 @@
 #include <fcntl.h>
 #include <netinet/in.h>
 #include <stdlib.h>
+#include <string.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <sys/types.h>
