diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8085e0a..eaae8ce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,10 +23,10 @@ SET (CMAKE_VERBOSE_MAKEFILE ON)
 
 #############################################################
 # Enable warnings and treat them as errors
-SET (PEDANTIC TRUE CACHE BOOL "Should we compile with -Wall -Werror.")
+SET (PEDANTIC TRUE CACHE BOOL "Should we compile with -Wall.")
 
 IF (PEDANTIC)
-  ADD_DEFINITIONS (-Wall -Werror)
+  ADD_DEFINITIONS (-Wall)
 ENDIF (PEDANTIC)
 
 #############################################################
@@ -51,9 +51,9 @@ ENDIF (WIN32)
 IF (UNIX AND NOT APPLE)
   SET (MASSXPERT_BIN_DIR ${CMAKE_INSTALL_PREFIX}/bin)
   SET (MASSXPERT_DATA_DIR ${CMAKE_INSTALL_PREFIX}/share/massxpert)
-  SET (MASSXPERT_PLUGIN_DIR ${CMAKE_INSTALL_PREFIX}/lib/massxpert/plugins)
+  SET (MASSXPERT_PLUGIN_DIR ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/massxpert/plugins)
   SET (MASSXPERT_LOCALE_DIR ${CMAKE_INSTALL_PREFIX}/share/massxpert/locales)
-  SET (MASSXPERT_DOC_DIR ${CMAKE_INSTALL_PREFIX}/share/doc/massxpert)
+  SET (MASSXPERT_DOC_DIR ${CMAKE_INSTALL_PREFIX}/share/doc/massxpert-${VERSION})
   SET (MASSXPERT_USERMAN_DIR ${MASSXPERT_DOC_DIR}/usermanual)
 ENDIF (UNIX AND NOT APPLE)
 
@@ -122,10 +122,6 @@ IF (${BUILD_PROGRAM})
   ###############
   # install stuff
 
-  # The license file
-  INSTALL (FILES COPYING
-    DESTINATION ${MASSXPERT_DOC_DIR})
-
   # The desktop file
   IF (UNIX AND NOT APPLE)
     INSTALL (FILES massxpert.desktop
@@ -165,10 +161,6 @@ IF (${BUILD_DATA})
   ###############
   # install stuff
   
-  # The license file
-  INSTALL (FILES COPYING
-    DESTINATION ${MASSXPERT_DOC_DIR})
-  
   # The manual pages (data)
   IF (UNIX AND NOT APPLE)
     INSTALL (FILES massxpert-data.7
@@ -201,10 +193,6 @@ IF (${BUILD_USERMANUAL})
   ###############
   # install stuff
   
-  # The license file
-  INSTALL (FILES COPYING
-    DESTINATION ${MASSXPERT_DOC_DIR})
-  
   # The manual pages (user manual)
   IF (UNIX AND NOT APPLE)
     INSTALL (FILES massxpert-doc.7
diff --git a/gui/configurationSettingsDlg.cpp b/gui/configurationSettingsDlg.cpp
index bf595d5..82cad21 100644
--- a/gui/configurationSettingsDlg.cpp
+++ b/gui/configurationSettingsDlg.cpp
@@ -309,13 +309,11 @@ namespace massXpert
   bool 
   ConfigurationSettingsDlg::checkLocalizationDir(const QDir &dir)
   {
-    // At the moment there is the french translation: massxpert_fr.qm
+    // Gentoo provides optional installing of translations.
+    // If you decide not to have translations, then this check
+    // would fail.
 
-    QString filePath(dir.absolutePath() +
-		      QDir::separator() +
-		      "massxpert_fr.qm");
-    
-    return QFile::exists(filePath);
+    return true;
   }
   
 
