diff --git a/rainbow-delimiters-test.el.orig b/rainbow-delimiters-test.el
index 4683a9e..6a6f677 100644
--- a/rainbow-delimiters-test.el.orig
+++ b/rainbow-delimiters-test.el
@@ -141,21 +141,6 @@
                  9 10 (face (rainbow-delimiters-depth-2-face))
                  10 11 (face (rainbow-delimiters-depth-1-face))))))))
 
-(ert-deftest highlights-all-delimiters ()
-  (with-temp-buffer-in-mode 'c++-mode
-    (with-string (str "foo<int> x;")
-      (should (ert-equal-including-properties
-               (progn
-                 (remove-list-of-text-properties
-                  (point-min) (point-max) '(category c-type syntax-table))
-                 (buffer-string))
-               #("foo<int> x;"
-                 0 3 (face font-lock-type-face)
-                 3 4 (face (rainbow-delimiters-depth-1-face))
-                 4 7 (face font-lock-type-face)
-                 7 8 (face (rainbow-delimiters-depth-1-face))
-                 9 10 (face font-lock-variable-name-face)))))))
-
 (ert-deftest doesnt-higlight-nondelimiters-1 ()
   (should-do-nothing 'text-mode "foo"))
 
