Debundle zlib, fixes parallel build, respect user FLAGS

--- src/readSet.c
+++ src/readSet.c
@@ -23,18 +23,13 @@
 #include <string.h>
 #include <math.h>
 #include <time.h>
+#include <zlib.h>
 
 #include "globals.h"
 #include "tightString.h"
 #include "readSet.h"
 #include "utility.h"
 
-#if defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
-#include "../third-party/zlib-1.2.3/Win32/include/zlib.h"
-#else
-#include "../third-party/zlib-1.2.3/zlib.h"
-#endif
-
 #if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(__CYGWIN__)
 #  include <fcntl.h>
 #  include <io.h>

--- Makefile
+++ Makefile
@@ -1,77 +1,62 @@
-CC = gcc
-CFLAGS = -Wall
-DEBUG = -g
-LDFLAGS = -lm
-OPT = -O3
+LIBS = -lm -lz
 MAXKMERLENGTH=31
 CATEGORIES=2
 DEF = -D MAXKMERLENGTH=$(MAXKMERLENGTH) -D CATEGORIES=$(CATEGORIES)
 
-Z_LIB_DIR=third-party/zlib-1.2.3
-Z_LIB_FILES=$(Z_LIB_DIR)/*.o
-
-# Mac OS users: uncomment the following lines
-# Z_LIB_FILES=
-# LDFLAGS = -lm -lz
-# CFLAGS = -Wall -m64
-
-# Sparc/Solaris users: uncomment the following line
-# CFLAGS = -Wall -m64
-
 OBJ = obj/tightString.o obj/run.o obj/splay.o obj/splayTable.o obj/graph.o obj/run2.o obj/fibHeap.o obj/fib.o obj/concatenatedGraph.o obj/passageMarker.o obj/graphStats.o obj/correctedGraph.o obj/dfib.o obj/dfibHeap.o obj/recycleBin.o obj/readSet.o obj/shortReadPairs.o obj/locallyCorrectedGraph.o obj/graphReConstruction.o obj/roadMap.o obj/preGraph.o obj/preGraphConstruction.o obj/concatenatedPreGraph.o obj/readCoherentGraph.o obj/crc.o obj/utility.o obj/kmer.o obj/scaffold.o
 OBJDBG = $(subst obj,obj/dbg,$(OBJ))
+OBJDE  = $(subst obj,objde,$(OBJ))
 
-default : cleanobj zlib obj velveth velvetg doc
+default : cleanobj  obj velveth velvetg 
 
 clean :
-	-rm obj/*.o obj/dbg/*.o ./velvet* 
-	cd $(Z_LIB_DIR) && make clean
+	-rm -f obj/*.o obj/dbg/*.o ./velvet* objde/*.o
 	-rm -f doc/manual_src/Manual.toc doc/manual_src/Manual.aux doc/manual_src/Manual.out doc/manual_src/Manual.log
 
 cleanobj: 
-	-rm obj/*.o obj/dbg/*.o 
-
-zlib : 
-	cd $(Z_LIB_DIR); ./configure; make; rm minigzip.o; rm example.o
+	-rm -f obj/*.o obj/dbg/*.o objde/*.o
 
 velveth : obj 
-	$(CC) $(CFLAGS) $(OPT) $(LDFLAGS) -o velveth obj/tightString.o obj/run.o obj/recycleBin.o obj/splay.o obj/splayTable.o obj/readSet.o obj/crc.o obj/utility.o obj/kmer.o $(Z_LIB_FILES)
+	$(CC) $(CFLAGS) $(OPT) $(LDFLAGS) -o velveth obj/tightString.o obj/run.o obj/recycleBin.o obj/splay.o obj/splayTable.o obj/readSet.o obj/crc.o obj/utility.o obj/kmer.o $(LIBS)
 
 
 velvetg : obj
-	$(CC) $(CFLAGS) $(OPT) $(LDFLAGS) -o velvetg obj/tightString.o obj/graph.o obj/run2.o obj/fibHeap.o obj/fib.o obj/concatenatedGraph.o obj/passageMarker.o obj/graphStats.o obj/correctedGraph.o obj/dfib.o obj/dfibHeap.o obj/recycleBin.o obj/readSet.o obj/shortReadPairs.o obj/scaffold.o obj/locallyCorrectedGraph.o obj/graphReConstruction.o obj/roadMap.o obj/preGraph.o obj/preGraphConstruction.o obj/concatenatedPreGraph.o obj/readCoherentGraph.o obj/utility.o obj/kmer.o $(Z_LIB_FILES)
+	$(CC) $(CFLAGS) $(OPT) $(LDFLAGS) -o velvetg obj/tightString.o obj/graph.o obj/run2.o obj/fibHeap.o obj/fib.o obj/concatenatedGraph.o obj/passageMarker.o obj/graphStats.o obj/correctedGraph.o obj/dfib.o obj/dfibHeap.o obj/recycleBin.o obj/readSet.o obj/shortReadPairs.o obj/scaffold.o obj/locallyCorrectedGraph.o obj/graphReConstruction.o obj/roadMap.o obj/preGraph.o obj/preGraphConstruction.o obj/concatenatedPreGraph.o obj/readCoherentGraph.o obj/utility.o obj/kmer.o $(LIBS)
 
 debug : cleanobj obj/dbg
-	$(CC) $(CFLAGS) $(LDFLAGS) $(DEBUG) -o velveth obj/dbg/tightString.o obj/dbg/run.o obj/dbg/recycleBin.o obj/dbg/splay.o obj/dbg/splayTable.o obj/dbg/readSet.o obj/dbg/crc.o obj/dbg/utility.o obj/dbg/kmer.o $(Z_LIB_FILES)
-	$(CC) $(CFLAGS) $(LDFLAGS) $(DEBUG) -o velvetg obj/dbg/tightString.o obj/dbg/graph.o obj/dbg/run2.o obj/dbg/fibHeap.o obj/dbg/fib.o obj/dbg/concatenatedGraph.o obj/dbg/passageMarker.o obj/dbg/graphStats.o obj/dbg/correctedGraph.o obj/dbg/dfib.o obj/dbg/dfibHeap.o obj/dbg/recycleBin.o obj/dbg/readSet.o obj/dbg/shortReadPairs.o obj/dbg/scaffold.o obj/dbg/locallyCorrectedGraph.o obj/dbg/graphReConstruction.o obj/dbg/roadMap.o obj/dbg/preGraph.o obj/dbg/preGraphConstruction.o obj/dbg/concatenatedPreGraph.o obj/dbg/readCoherentGraph.o obj/dbg/utility.o obj/dbg/kmer.o $(Z_LIB_FILES)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(DEBUG) -o velveth obj/dbg/tightString.o obj/dbg/run.o obj/dbg/recycleBin.o obj/dbg/splay.o obj/dbg/splayTable.o obj/dbg/readSet.o obj/dbg/crc.o obj/dbg/utility.o obj/dbg/kmer.o $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(DEBUG) -o velvetg obj/dbg/tightString.o obj/dbg/graph.o obj/dbg/run2.o obj/dbg/fibHeap.o obj/dbg/fib.o obj/dbg/concatenatedGraph.o obj/dbg/passageMarker.o obj/dbg/graphStats.o obj/dbg/correctedGraph.o obj/dbg/dfib.o obj/dbg/dfibHeap.o obj/dbg/recycleBin.o obj/dbg/readSet.o obj/dbg/shortReadPairs.o obj/dbg/scaffold.o obj/dbg/locallyCorrectedGraph.o obj/dbg/graphReConstruction.o obj/dbg/roadMap.o obj/dbg/preGraph.o obj/dbg/preGraphConstruction.o obj/dbg/concatenatedPreGraph.o obj/dbg/readCoherentGraph.o obj/dbg/utility.o obj/dbg/kmer.o $(LIBS)
 
 color : cleanobj obj_de
-	$(CC) $(CFLAGS) $(OPT) $(LDFLAGS) -o velveth_de obj/tightString.o obj/run.o obj/recycleBin.o obj/splay.o obj/splayTable.o obj/readSet.o obj/crc.o obj/utility.o obj/kmer.o $(Z_LIB_FILES)
-	$(CC) $(CFLAGS) $(OPT) $(LDFLAGS) -o velvetg_de obj/tightString.o obj/graph.o obj/run2.o obj/fibHeap.o obj/fib.o obj/concatenatedGraph.o obj/passageMarker.o obj/graphStats.o obj/correctedGraph.o obj/dfib.o obj/dfibHeap.o obj/recycleBin.o obj/readSet.o obj/shortReadPairs.o obj/scaffold.o obj/locallyCorrectedGraph.o obj/graphReConstruction.o obj/roadMap.o obj/preGraph.o obj/preGraphConstruction.o obj/concatenatedPreGraph.o obj/readCoherentGraph.o obj/utility.o obj/kmer.o $(Z_LIB_FILES)
+	$(CC) $(CFLAGS) $(OPT) $(LDFLAGS) -o velveth_de objde/tightString.o objde/run.o objde/recycleBin.o objde/splay.o objde/splayTable.o objde/readSet.o objde/crc.o objde/utility.o objde/kmer.o $(LIBS)
+	$(CC) $(CFLAGS) $(OPT) $(LDFLAGS) -o velvetg_de objde/tightString.o objde/graph.o objde/run2.o objde/fibHeap.o objde/fib.o objde/concatenatedGraph.o objde/passageMarker.o objde/graphStats.o objde/correctedGraph.o objde/dfib.o objde/dfibHeap.o objde/recycleBin.o objde/readSet.o objde/shortReadPairs.o objde/scaffold.o objde/locallyCorrectedGraph.o objde/graphReConstruction.o objde/roadMap.o objde/preGraph.o objde/preGraphConstruction.o objde/concatenatedPreGraph.o objde/readCoherentGraph.o objde/utility.o objde/kmer.o $(LIBS)
 
 colordebug : cleanobj obj/dbg_de
-	$(CC) $(CFLAGS) $(LDFLAGS) $(DEBUG) -o velveth_de obj/dbg/tightString.o obj/dbg/run.o obj/dbg/recycleBin.o obj/dbg/splay.o obj/dbg/splayTable.o obj/dbg/readSet.o obj/dbg/crc.o obj/dbg/utility.o obj/dbg/kmer.o $(Z_LIB_FILES)
-	$(CC) $(CFLAGS) $(LDFLAGS) $(DEBUG) -o velvetg_de obj/dbg/tightString.o obj/dbg/graph.o obj/dbg/run2.o obj/dbg/fibHeap.o obj/dbg/fib.o obj/dbg/concatenatedGraph.o obj/dbg/passageMarker.o obj/dbg/graphStats.o obj/dbg/correctedGraph.o obj/dbg/dfib.o obj/dbg/dfibHeap.o obj/dbg/recycleBin.o obj/dbg/readSet.o obj/dbg/shortReadPairs.o obj/dbg/scaffold.o obj/dbg/locallyCorrectedGraph.o obj/dbg/graphReConstruction.o obj/dbg/roadMap.o obj/dbg/preGraph.o obj/dbg/preGraphConstruction.o obj/dbg/concatenatedPreGraph.o obj/dbg/readCoherentGraph.o obj/dbg/utility.o obj/dbg/kmer.o $(Z_LIB_FILES)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(DEBUG) -o velveth_de objde/dbg/tightString.o objde/dbg/run.o objde/dbg/recycleBin.o objde/dbg/splay.o objde/dbg/splayTable.o objde/dbg/readSet.o objde/dbg/crc.o objde/dbg/utility.o objde/dbg/kmer.o $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(DEBUG) -o velvetg_de objde/dbg/tightString.o objde/dbg/graph.o objde/dbg/run2.o objde/dbg/fibHeap.o objde/dbg/fib.o objde/dbg/concatenatedGraph.o objde/dbg/passageMarker.o objde/dbg/graphStats.o objde/dbg/correctedGraph.o objde/dbg/dfib.o objde/dbg/dfibHeap.o objde/dbg/recycleBin.o objde/dbg/readSet.o objde/dbg/shortReadPairs.o objde/dbg/scaffold.o objde/dbg/locallyCorrectedGraph.o objde/dbg/graphReConstruction.o objde/dbg/roadMap.o objde/dbg/preGraph.o objde/dbg/preGraphConstruction.o objde/dbg/concatenatedPreGraph.o objde/dbg/readCoherentGraph.o objde/dbg/utility.o objde/dbg/kmer.o $(LIBS)
 
 objdir:
 	mkdir -p obj
+	mkdir -p objde
 
-obj: zlib objdir $(OBJ)
+obj:  objdir $(OBJ)
 
-obj_de: override DEF := $(DEF) -D COLOR
-obj_de: zlib cleanobj objdir $(OBJ)
+obj_de:  cleanobj objdir $(OBJDE)
 
 obj/dbgdir: 
 	mkdir -p obj/dbg
 
-obj/dbg: zlib obj/dbgdir $(OBJDBG)
+obj/dbg: obj/dbgdir $(OBJDBG)
 
 obj/dbg_de: override DEF := $(DEF) -D COLOR
-obj/dbg_de: zlib cleanobj obj/dbgdir $(OBJDBG)
+obj/dbg_de: cleanobj obj/dbgdir $(OBJDBG)
 
 obj/%.o: src/%.c
 	$(CC) $(CFLAGS) $(OPT) $(DEF) -c $? -o $@ 
 
+objde/%.o: src/%.c
+	$(CC) $(CFLAGS) $(OPT) $(DEF) -D COLOR -c $? -o $@ 
+
 obj/dbg/%.o: src/%.c
 	$(CC) $(CFLAGS) $(DEBUG) $(DEF) -c $? -o $@ 
 
