diff --git a/Makefile b/Makefile
index 39b4872..4c04865 100644
--- a/Makefile
+++ b/Makefile
@@ -154,8 +154,8 @@ mkdir -p $1$(PREFIX)/bin
 mkdir -p $1$(PREFIX)/share/applications
 mkdir -p $1$(PREFIX)/share/icons/hicolor/128x128/apps
 mkdir -p $1$(PREFIX)/share/$(PACKAGE_NAME)/$(PACKAGE_VERSION)
-cp $(GSL_SHELL_GUI) $(GSL_SHELL) $1$(PREFIX)/bin
-strip $1$(PREFIX)/bin/$(GSL_SHELL) $1$(PREFIX)/bin/$(GSL_SHELL_GUI)
+test -f $(GSL_SHELL_GUI) && cp $(GSL_SHELL_GUI) $1$(PREFIX)/bin  || :
+test -f $(GSL_SHELL) && cp $(GSL_SHELL) $1$(PREFIX)/bin  || :
 test -f $(LUAJIT_SO) && \
   mkdir -p $1$(PREFIX)/lib && \
   cp $(LUAJIT_SO) $1$(PREFIX)/lib && \
@@ -168,7 +168,7 @@ $(CP_REL) $(LUA_BASE_FILES) $1$(PREFIX)/share/$(PACKAGE_NAME)/$(PACKAGE_VERSION)
 $(CP_REL) $(EXAMPLES_FILES) $1$(PREFIX)/bin
 endef
 
-install: $(GSL_SHELL) $(GSL_SHELL_GUI)
+install:
 	$(call install-to-dir,$(DESTDIR))
 
 debian: $(DEBIAN_PACKAGE)
