diff -urN latex2html-2002-2-1.orig/config/config.pl latex2html-2002-2-1/config/config.pl
--- latex2html-2002-2-1.orig/config/config.pl	2006-06-02 20:41:18.000000000 +0200
+++ latex2html-2002-2-1/config/config.pl	2006-06-02 20:42:07.000000000 +0200
@@ -1275,7 +1275,7 @@
     ($stat,$msg,$err) = &get_out_err("$pnmcrop -version");
     my $vers = '';
     $msg = $msg || $err;
-    if ($msg =~ /(^|\s*)Version.*\s([\d\.]+)\s*([\n\r]|$)/is) { $vers = $2; }
+    if ($msg =~ /(^|\s*)Version.*\s(\d+)\..*\s*([\n\r]|$)/is) { $vers = $2; }
     if ($vers =~ /^199/) {
 	# try left crop
 	&checking('if pnmcrop can crop from one direction');
