--- projects/rscriptd/Makefile.org	2013-01-18 16:22:46.000000000 +0200
+++ projects/rscriptd/Makefile	2013-01-18 16:25:08.000000000 +0200
@@ -68,6 +68,13 @@
 	# Install etc
 	mkdir -m $(DIR_MODE) -p $(PREFIX)/etc/rscriptd
 	install -m $(DATA_MODE) -o $(OWNER) ./rscriptd.conf $(PREFIX)/etc/rscriptd/rscriptd.conf
+ifeq ($(OS), linux)
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnConnect $(PREFIX)/etc/rscriptd/
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnDisconnect $(PREFIX)/etc/rscriptd/
+else
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/freebsd/etc/stargazer/OnConnect $(PREFIX)/etc/rscriptd/
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/freebsd/etc/stargazer/OnDisconnect $(PREFIX)/etc/rscriptd/
+endif
 
 uninstall: uninstall-bin uninstall-data
 
@@ -77,7 +84,8 @@
 uninstall-data:
 	# Uninstall etc
 	rm -f $(PREFIX)/etc/rscriptd/rscriptd.conf
-
+	rm -f $(PREFIX)/etc/rscriptd/OnConnect
+	rm -f $(PREFIX)/etc/rscriptd/OnDisconnect
 
 ifneq ($(MAKECMDGOALS),distclean)
 ifneq ($(MAKECMDGOALS),clean)
