--- projects/sgauth/Makefile.org	2013-01-05 23:47:38.000000000 +0200
+++ projects/sgauth/Makefile	2013-01-05 23:48:34.000000000 +0200
@@ -67,8 +67,8 @@
 
 install-data:
 	# Install etc
-	mkdir -m $(DIR_MODE) -p $(PREFIX)/etc
-	install -m $(DATA_MODE) -o $(OWNER) ./sgauth.conf $(PREFIX)/etc/sgauth.conf
+	mkdir -m $(DIR_MODE) -p $(PREFIX)/etc/stargazer
+	install -m $(DATA_MODE) -o $(OWNER) ./sgauth.conf $(PREFIX)/etc/stargazer/sgauth.conf
 
 uninstall: uninstall-bin uninstall-data
 
--- projects/stargazer/Makefile.org	2013-01-05 22:53:26.000000000 +0200
+++ projects/stargazer/Makefile	2013-01-05 22:54:52.000000000 +0200
@@ -92,30 +92,8 @@
 	mkdir -m $(DIR_MODE) -p $(PREFIX)/etc/stargazer/conf-available.d
 	mkdir -m $(DIR_MODE) -p $(PREFIX)/etc/stargazer/conf-enabled.d
 	install -m $(DATA_MODE) -o $(OWNER) $(ETC_DIR)/stargazer.conf $(PREFIX)/etc/stargazer/stargazer.conf
-	install -m $(DATA_MODE) -o $(OWNER) $(ETC_DIR)/conf-available.d/*.conf $(PREFIX)/etc/stargazer/conf-available.d
-	ln -sf ../conf-available.d/mod_ao.conf $(PREFIX)/etc/stargazer/conf-enabled.d/mod_ao.conf
-	ln -sf ../conf-available.d/mod_ia.conf $(PREFIX)/etc/stargazer/conf-enabled.d/mod_ia.conf
-	ln -sf ../conf-available.d/mod_ping.conf $(PREFIX)/etc/stargazer/conf-enabled.d/mod_ping.conf
-	ln -sf ../conf-available.d/mod_sg.conf $(PREFIX)/etc/stargazer/conf-enabled.d/mod_sg.conf
-	ln -sf ../conf-available.d/store_files.conf $(PREFIX)/etc/stargazer/conf-enabled.d/store_files.conf
-
-ifeq ($(OS),linux)
-	ln -sf ../conf-available.d/mod_cap_ether.conf $(PREFIX)/etc/stargazer/conf-enabled.d/mod_cap_ether.conf
-else
-	ln -sf ../conf-available.d/mod_cap_bpf.conf $(PREFIX)/etc/stargazer/conf-enabled.d/mod_cap_bpf.conf
-endif
-
 	install -m $(DATA_MODE) -o $(OWNER) $(ETC_DIR)/rules $(PREFIX)/etc/stargazer/rules
 	install -m $(BIN_MODE) -o $(OWNER) $(ETC_DIR)/On* $(PREFIX)/etc/stargazer/
-	
-	# Install file db
-	mkdir -m $(DIR_MODE) -p $(PREFIX)/var/stargazer/admins
-	mkdir -m $(DIR_MODE) -p $(PREFIX)/var/stargazer/tariffs
-	mkdir -m $(DIR_MODE) -p $(PREFIX)/var/stargazer/users/test
-	install -m $(DATA_MODE) -o $(OWNER) $(VAR_DIR)/admins/admin.adm $(PREFIX)/var/stargazer/admins/admin.adm
-	install -m $(DATA_MODE) -o $(OWNER) $(VAR_DIR)/tariffs/tariff.tf $(PREFIX)/var/stargazer/tariffs/tariff.tf
-	install -m $(DATA_MODE) -o $(OWNER) $(VAR_DIR)/users/test/conf $(PREFIX)/var/stargazer/users/test/conf
-	install -m $(DATA_MODE) -o $(OWNER) $(VAR_DIR)/users/test/stat $(PREFIX)/var/stargazer/users/test/stat
 
 uninstall: uninstall-bin uninstall-data
 
--- projects/rscriptd/Makefile.org	2013-01-19 17:03:42.000000000 +0200
+++ projects/rscriptd/Makefile	2013-01-19 17:04:25.000000000 +0200
@@ -65,15 +65,13 @@
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 install-data:
-	# Install etc
-	mkdir -m $(DIR_MODE) -p $(PREFIX)/etc/rscriptd
-	install -m $(DATA_MODE) -o $(OWNER) ./rscriptd.conf $(PREFIX)/etc/rscriptd/rscriptd.conf
+	install -D -m $(DATA_MODE) -o $(OWNER) ./rscriptd.conf $(PREFIX)/etc/stargazer/rscriptd.conf
 ifeq ($(OS), linux)
-	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnConnect $(PREFIX)/etc/rscriptd/
-	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnDisconnect $(PREFIX)/etc/rscriptd/
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnConnect $(PREFIX)/etc/stargazer/
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnDisconnect $(PREFIX)/etc/stargazer/
 else
-	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/freebsd/etc/stargazer/OnConnect $(PREFIX)/etc/rscriptd/
-	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/freebsd/etc/stargazer/OnDisconnect $(PREFIX)/etc/rscriptd/
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/freebsd/etc/stargazer/OnConnect $(PREFIX)/etc/stargazer/
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/freebsd/etc/stargazer/OnDisconnect $(PREFIX)/etc/stargazer/
 endif
 
 uninstall: uninstall-bin uninstall-data
--- projects/rlm_stg/Makefile.org	2013-02-03 14:02:58.000000000 +0200
+++ projects/rlm_stg/Makefile	2013-02-03 14:05:43.000000000 +0200
@@ -56,19 +56,11 @@
 install: install-bin
 
 install-bin:
-ifeq ($(DEBUG), yes)
 ifeq ($(OS), linux)
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/lib/freeradius/$(PROG)
 else
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/lib/$(PROG)
 endif
-else
-ifeq ($(OS), linux)
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/lib/freeradius/$(PROG)
-else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/lib/$(PROG)
-endif
-endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 uninstall: uninstall-bin
--- projects/rscriptd/Makefile.org	2013-02-03 14:17:01.000000000 +0200
+++ projects/rscriptd/Makefile	2013-02-03 14:17:32.000000000 +0200
@@ -57,11 +57,7 @@
 install: install-bin install-data
 
 install-bin:
-ifeq ($(DEBUG), yes)
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/sbin/$(PROG)
-else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
-endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 install-data:
--- projects/sgauth/Makefile.org	2013-02-03 14:18:37.000000000 +0200
+++ projects/sgauth/Makefile	2013-02-03 14:18:48.000000000 +0200
@@ -58,11 +58,7 @@
 install: install-bin install-data
 
 install-bin:
-ifeq ($(DEBUG), yes)
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/sbin/$(PROG)
-else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
-endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 install-data:
--- projects/sgconf/Makefile.org	2013-02-03 14:19:18.000000000 +0200
+++ projects/sgconf/Makefile	2013-02-03 14:19:28.000000000 +0200
@@ -66,11 +66,7 @@
 install: install-bin
 
 install-bin:
-ifeq ($(DEBUG), yes)
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/bin/$(PROG)
-else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/bin/$(PROG)
-endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 uninstall: uninstall-bin
--- projects/sgconf_xml/Makefile.org	2013-02-03 14:19:53.000000000 +0200
+++ projects/sgconf_xml/Makefile	2013-02-03 14:19:59.000000000 +0200
@@ -66,11 +66,7 @@
 install: install-bin install-data
 
 install-bin:
-ifeq ($(DEBUG), yes)
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/bin/$(PROG)
-else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/bin/$(PROG)
-endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 uninstall: uninstall-bin uninstall-data
--- projects/sgconv/Makefile.org	2013-02-03 14:20:24.000000000 +0200
+++ projects/sgconv/Makefile	2013-02-03 14:20:33.000000000 +0200
@@ -61,11 +61,7 @@
 install: install-bin
 
 install-bin:
-ifeq ($(DEBUG), yes)
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/bin/$(PROG)
-else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/bin/$(PROG)
-endif
 	$(MAKE) -C $(DIR_PLUGINS) install
 
 uninstall: uninstall-bin
--- projects/stargazer/Makefile.org	2013-02-03 14:20:56.000000000 +0200
+++ projects/stargazer/Makefile	2013-02-03 14:21:30.000000000 +0200
@@ -77,11 +77,7 @@
 install: install-bin install-data
 
 install-bin: $(PROG)
-ifeq ($(DEBUG), yes)
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/sbin/$(PROG)
-else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
-endif
 	$(MAKE) -C $(DIR_INCLUDE) install
 	$(MAKE) -C $(DIR_LIBSRC) install
 	$(MAKE) -C $(DIR_PLUGINS) install
--- stglibs/Makefile.in.org	2013-02-03 14:53:02.000000000 +0200
+++ stglibs/Makefile.in	2013-02-03 14:53:11.000000000 +0200
@@ -35,11 +35,7 @@
 	rm -f deps $(PROG) *.o *.a *.so tags *.*~ 
 
 install: $(PROG)
-ifeq ($(DEBUG), yes)
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/lib/stg/$(PROG)
-else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/lib/stg/$(PROG)
-endif
 	mkdir -m $(DIR_MODE) -p $(PREFIX)/usr/include/stg
 	install -m $(DATA_MODE) -o $(OWNER) $(addprefix include/stg/,$(INCS)) $(PREFIX)/usr/include/stg/
 
--- projects/stargazer/plugins/Makefile.in.org	2013-02-03 15:38:41.000000000 +0200
+++ projects/stargazer/plugins/Makefile.in	2013-02-03 15:38:51.000000000 +0200
@@ -28,11 +28,7 @@
 
 install: $(PROG)
 	mkdir -m $(DIR_MODE) -p $(PREFIX)/usr/lib/stg
-ifeq ($(DEBUG), yes)
 	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/lib/stg/$(PROG)
-else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/lib/stg/$(PROG)
-endif
 
 uninstall:
 	rm -f $(PREFIX)/usr/lib/stg/$(PROG)
