diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,10 +75,6 @@ find_package( GLX       REQUIRED )
 find_package( XRender   REQUIRED )
 find_package( XRandr    REQUIRED )
 find_package( GLEW      REQUIRED )
-# This library is needed only for Ubuntu it seems, some platforms don't even
-# ship with it. I couldn't find a way to do a test compile to check if librt
-# was needed, so instead I just didn't mark it as REQUIRED.
-find_package( RT )
 
 set( CMAKE_CXX_FLAGS
      "${CMAKE_CXX_FLAGS} ${CMAKE_IMLIB2_CXX_FLAGS}" )
@@ -96,10 +92,6 @@ include_directories( "${IMLIB2_INCLUDE_DIR}"
                      ${XRENDER_INCLUDE_DIRS} )
 endif()
 
-if ( RT_INCLUDE_DIR )
-    include_directories( ${RT_INCLUDE_DIR} )
-endif()
-
 # Executable
 add_executable( "${BIN_TARGET}" ${source} )
 
@@ -119,11 +111,6 @@ if ( CMAKE_OPENGL_SUPPORT )
 endif()
 
 
-if ( RT_LIBRARY )
-    target_link_libraries( "${BIN_TARGET}"
-                           "${RT_LIBRARY}" )
-endif()
-
 install( TARGETS ${BIN_TARGET}
          DESTINATION "${CMAKE_INSTALL_PREFIX}/bin" )
 
