Don't clobber toolchain defaults.

https://bugs.gentoo.org/892928
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ MAKE_DIR     ?= install -d
 INSTALL_DATA ?= install
 
 CC?=gcc
-CFLAGS?=$(CFLAGS_EXTRA) -D_FORTIFY_SOURCE=2 -fPIC
+CFLAGS?=$(CFLAGS_EXTRA) -fPIC
 LDFLAGS?=$(LDFLAGS_EXTRA) -Wl,-soname,libscrypt.so.0 -Wl,--version-script=libscrypt.version
 CFLAGS_EXTRA?=-Wl,-rpath=. -O2 -Wall -g -fstack-protector
 LDFLAGS_EXTRA?=-Wl,-z,relro
