--- a/Po4aBuilder.pm
+++ b/Po4aBuilder.pm
@@ -29,7 +29,12 @@
     my $self = shift;
     $self->depends_on('binpo');
     $self->make_files_writable("po/pod");
-    system("./share/po4a-build -f po4a-build.conf") && die;
+    { # Fix for massive slowdown/Memory consumption
+       open( my $fh, '-|', qw( ./share/po4a-build -f po4a-build.conf ))  or die;
+       while( defined ( my $line = <$fh> ) ){
+          print $line;
+       }
+    }
     File::Path::mkpath( File::Spec->catdir( 'blib', 'manl10n' ), 0, oct(777) );
     system ("cp -R _build/po4a/man/* blib/manl10n") && die;
 }
