diff --git a/sources/webkit-sharp.dll.config.in b/sources/webkit-sharp.dll.config.in
index 75788f8..4220311 100644
--- a/sources/webkit-sharp.dll.config.in
+++ b/sources/webkit-sharp.dll.config.in
@@ -1,3 +1,3 @@
 <configuration>
-  <dllmap dll="webkit-1.0" target="webkit-1.0@LIB_PREFIX@.2@LIB_SUFFIX@"/>
+  <dllmap dll="webkit-1.0" target="libwebkitgtk-1.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
 </configuration>
