--- src/Mod/CMakeLists.txt	2013-02-02 18:09:17.000000000 +0100
+++ src/Mod/CMakeLists.txt.new	2014-01-19 18:00:27.215825143 +0100
@@ -18,7 +18,7 @@
     MESSAGE("Due to the missing Eigen3 library the Sketcher module won't be built")
     MESSAGE("Due to the missing Eigen3 library the Robot module won't be built")
 ENDIF(EIGEN3_FOUND)
-add_subdirectory(Machining_Distortion)
+#add_subdirectory(Machining_Distortion)
 
 add_subdirectory(ReverseEngineering)
 add_subdirectory(MeshPart)
