From d7ea5a0f20888a2ddae3eb48e508e9067662cdc7 Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl-gentoo@posteo.net>
Date: Wed, 28 Apr 2021 08:57:53 +0200
Subject: [PATCH] fix test when built without GUI

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>
---
 src/Mod/Arch/ArchRoof.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Mod/Arch/ArchRoof.py b/src/Mod/Arch/ArchRoof.py
index 99130d0..e2c2955 100644
--- a/src/Mod/Arch/ArchRoof.py
+++ b/src/Mod/Arch/ArchRoof.py
@@ -22,7 +22,6 @@
 import math
 
 import ArchComponent
-import Arch_rc
 import Draft
 import DraftGeomUtils
 import DraftVecUtils
@@ -32,6 +31,7 @@ import Part
 from FreeCAD import Vector
 
 if FreeCAD.GuiUp:
+    import Arch_rc
     import FreeCADGui
     from PySide import QtCore, QtGui
     from DraftTools import translate
-- 
2.31.1

