From acc8a26b73a87ae024bce30e9f1531610b6e0e5f Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl-gentoo@posteo.net>
Date: Sat, 13 Feb 2021 14:54:28 +0100
Subject: [PATCH] [Gentoo specific] don't check vcs

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>
---
 src/Tools/SubWCRev.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Tools/SubWCRev.py b/src/Tools/SubWCRev.py
index 9795450..59938bd 100644
--- a/src/Tools/SubWCRev.py
+++ b/src/Tools/SubWCRev.py
@@ -455,7 +455,7 @@ def main():
         if o in ("-b", "--bindir"):
             bindir = a
 
-    vcs=[GitControl(), DebianGitHub(), BazaarControl(), Subversion(), MercurialControl(), DebianChangelog(), UnknownControl()]
+    vcs=[UnknownControl()]
     for i in vcs:
         if i.extractInfo(srcdir, bindir):
             # Open the template file and the version file
-- 
2.30.1

