--- a/tests/src/Base/CMakeLists.txt
+++ b/tests/src/Base/CMakeLists.txt
@@ -11,4 +11,6 @@ target_sources(
             ${CMAKE_CURRENT_SOURCE_DIR}/tst_Tools.cpp
 )
 
-setup_qt_test(InventorBuilder)
+if(BUILD_GUI)
+    setup_qt_test(InventorBuilder)
+endif()
--- a/tests/src/Gui/CMakeLists.txt
+++ b/tests/src/Gui/CMakeLists.txt
@@ -6,4 +6,6 @@ target_sources(
 )
 
 # Qt tests
-setup_qt_test(QuantitySpinBox)
+if(BUILD_GUI)
+    setup_qt_test(QuantitySpinBox)
+endif()
