--- a/dialog.py.orig	2016-05-22 09:20:46.385000000 +0000
+++ b/dialog.py	2016-05-22 09:20:53.170000000 +0000
@@ -62,7 +62,7 @@
     def __repr__(self):
         # Unicode strings are not supported as the result of __repr__()
         # in Python 2.x (cf. <http://bugs.python.org/issue5876>).
-        return b"{0}.{1}".format(__name__, _VersionInfo.__repr__(self))
+        return "{0}.{1}".format(__name__, _VersionInfo.__repr__(self))
 
 #: Version of pythondialog as a :class:`VersionInfo` instance.
 #:
