diff -ru GtkAda-2.4.0-orig/src/gtkada-mdi.adb GtkAda-2.4.0/src/gtkada-mdi.adb
--- GtkAda-2.4.0-orig/src/gtkada-mdi.adb	2004-11-09 09:35:57.000000000 +0100
+++ GtkAda-2.4.0/src/gtkada-mdi.adb	2006-05-18 23:15:13.917488750 +0200
@@ -4778,6 +4778,7 @@
             when Top | Bottom =>
                Widths (Side)  := 0;
                Heights (Side) := Get_Allocation_Height (MDI.Docks (Side));
+            when None => Null;
          end case;
 
          --  If not allocated yet
