From: Julian Ospald <hasufell@gentoo.org>
Date: Sun Sep  8 11:05:26 UTC 2013
Subject: fix DATADIR detection for minimal setups

--- a/src/engine/shared/storage.cpp
+++ b/src/engine/shared/storage.cpp
@@ -208,7 +208,7 @@
 			for (i = 0; i < DirsCount; i++)
 			{
 				char aBuf[128];
-				str_format(aBuf, sizeof(aBuf), "%s/mapres", aDirs[i]);
+				str_format(aBuf, sizeof(aBuf), "%s/maps", aDirs[i]);
 				if(fs_is_dir(aBuf))
 				{
 					str_copy(m_aDatadir, aDirs[i], sizeof(m_aDatadir));
