From de281151995a2f83ead0c272c6a7c35db2d824ca Mon Sep 17 00:00:00 2001
From: Kang-min Liu <gugod@gugod.org>
Date: Sat, 10 Jun 2017 16:12:44 +0200
Subject: [PATCH] deal with the removal of "." in @INC.

---
 Makefile.PL | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile.PL b/Makefile.PL
index fce95a70..279c3c09 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -1,3 +1,7 @@
+
+use FindBin '$Bin';
+BEGIN { push @INC, $Bin; }
+
 use inc::Module::Install;
 
 license 'mit';
