diff -purN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2017-08-17 08:40:34.524350644 +0100
+++ b/CMakeLists.txt	2017-08-17 08:56:34.200174313 +0100
@@ -22,9 +22,13 @@ set(CPACK_SOURCE_PACKAGE_FILE_NAME
   )
 set(CPACK_SOURCE_IGNORE_FILES
   "/.git*;/.cvs*;${CPACK_SOURCE_IGNORE_FILES}")
+set(CMAKE_THREAD_PREFER_PTHREAD ON)
+
 INCLUDE ( CPack )
 INCLUDE ( GNUInstallDirs )
 
+find_package( PkgConfig QUIET )
+find_package( Threads )
 
 # Allow the developer to select if Dynamic or Static libraries are built
 OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" ON)
@@ -174,21 +174,7 @@ INSTALL ( FILES
 
 )
 
-FILE ( WRITE ${CMAKE_BINARY_DIR}/IlmBase.pc "prefix=${CMAKE_INSTALL_PREFIX}\n" )
-FILE ( APPEND ${CMAKE_BINARY_DIR}/IlmBase.pc "exec_prefix=\${prefix}
-libdir=\${exec_prefix}/lib
-includedir=\${prefix}/include
-OpenEXR_includedir=\${prefix}/include/OpenEXR
-
-Name: IlmBase
-Description: Base math and exception libraries
-Version: ${ILMBASE_VERSION}
-Libs: -L\${libdir} -lImath${ILMBASE_LIBSUFFIX} -lIexMath${ILMBASE_LIBSUFFIX} -lHalf -lIex${ILMBASE_LIBSUFFIX} -lIlmThread${ILMBASE_LIBSUFFIX} -pthread
-Cflags: -pthread -I\${OpenEXR_includedir}
-")
-
-INSTALL ( FILES
-  ${CMAKE_BINARY_DIR}/IlmBase.pc
-  DESTINATION
-  lib/pkgconfig
-)
+if ( PKG_CONFIG_FOUND )
+	configure_file(IlmBase.pc.cmake "${PROJECT_BINARY_DIR}/IlmBase.pc" @ONLY)
+	install( FILES "${PROJECT_BINARY_DIR}/IlmBase.pc" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig COMPONENT dev )
+endif()
diff -purN a/IlmBase.pc.cmake b/IlmBase.pc.cmake
--- a/IlmBase.pc.cmake	1970-01-01 01:00:00.000000000 +0100
+++ b/IlmBase.pc.cmake	2017-08-17 08:48:29.114556759 +0100
@@ -0,0 +1,9 @@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+OpenEXR_includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@/OpenEXR
+
+Name: IlmBase
+Description: Base math and exception libraries
+Version: @ILMBASE_VERSION@
+Libs: -L${libdir} -lHalf -lIex@ILMBASE_LIBSUFFIX@ -lIexMath@ILMBASE_LIBSUFFIX@ -lIlmThread@ILMBASE_LIBSUFFIX@ -lImath@ILMBASE_LIBSUFFIX@ @CMAKE_THREAD_LIBS_INIT@
+Cflags: @CMAKE_THREAD_LIBS_INIT@ -I${OpenEXR_includedir}
