--- IlmThread/Makefile.in
+++ IlmThread/Makefile.in
@@ -209,7 +209,7 @@
 			  IlmThreadMutexPosix.cpp
 
 libIlmThread_la_LDFLAGS = -version-info @LIBTOOL_VERSION@ -no-undefined
-libIlmThread_la_LIBADD = ../Iex/libIex.la
+libIlmThread_la_LIBADD = ../Iex/libIex.la -lpthread
 libIlmThreadincludedir = $(includedir)/OpenEXR
 libIlmThreadinclude_HEADERS = IlmThreadPool.h IlmThread.h \
 			      IlmThreadSemaphore.h IlmThreadMutex.h
