diff -ur memtest86+-5.31b/Makefile memtest86+-5.31b.new/Makefile
--- memtest86+-5.31b/Makefile	2021-09-16 09:27:24.401570971 -0500
+++ memtest86+-5.31b.new/Makefile	2021-09-16 09:30:24.332704245 -0500
@@ -64,6 +64,9 @@
 memsize.o: memsize.c
 	$(CC) -Wall -Werror -march=i486 -m32 -O0 -fomit-frame-pointer -fno-builtin -ffreestanding -fPIC $(SMP_FL) -fno-stack-protector   -c -o memsize.o memsize.c
 
+test.o: test.c
+test.o: CFLAGS += -fno-strict-overflow
+
 random.o: random.c
 random.o: CFLAGS += -O3 -fno-PIC