https://bugs.gentoo.org/552280

--- a/Makefile
+++ b/Makefile
@@ -20,6 +20,9 @@
 BUILD_USER = \"$(shell whoami)\"
 
 CFLAGS := -Wall -Os -m32 -nostdlib
+BUILD_CFLAGS := -Wall -O2
+BUILD_CC ?= $(CC)
+BUILD_LD ?= $(BUILD_CC)
 
 ASFLAGS := $(CFLAGS)
 ASFLAGS += -DBUILD_DATE="$(BUILD_DATE)"
@@ -55,7 +56,7 @@
 	$(LD) $(LDFLAGS) $(OBJS) -o $@
 
 csum8: csum8.c
-	$(CC) -Wall -O2 -o $@ $<
+	$(BUILD_CC) $(BUILD_CFLAGS) $(BUILD_CPPFLAGS) -o $@ $<
 
 sgabios.o: buildinfo
 
