--- lib/Makefile.old	2015-10-10 15:49:25.377235002 +0300
+++ lib/Makefile	2015-10-10 15:49:50.021464544 +0300
@@ -76,17 +76,13 @@
 .depend: $(filter-out $(GENERATED),$(SOURCES))
 -include .depend
 
-install-lockdir:
-	$(Q) $(INSTALL) -d $(DESTDIR)$(LOCKDIR)
-.PHONY: install-lockdir
-
 install-pc: $(PC)
 	$(E) "  INSTALL " $(PC)
 	$(Q) $(INSTALL) -d $(DESTDIR)$(PCDIR)
 	$(Q) $(INSTALL) $(PC) $(DESTDIR)$(PCDIR)
 .PHONY: install-pc
 
-install: all install-lockdir install-pc
+install: all  install-pc
 	$(Q) $(INSTALL) -d $(DESTDIR)$(LIBDIR)
 	$(E) "  INSTALL " $(LIBPLOOP)
 	$(Q) $(INSTALL) -m 644 $(LIBPLOOP) $(DESTDIR)$(LIBDIR)
