diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7a1098f..f83ed7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,7 @@ project(qjackctl
   LANGUAGES C CXX)
 
 set (CONFIG_VERSION ${PROJECT_VERSION})
+#[[
 execute_process (
   COMMAND git describe --tags --dirty --abbrev=6
   OUTPUT_VARIABLE GIT_DESCRIBE_OUTPUT
@@ -27,6 +28,8 @@ if (GIT_DESCRIBE_RESULT EQUAL 0)
 else ()
   set (VERSION "${PROJECT_VERSION}")
 endif ()
+]]
+set (VERSION "${PROJECT_VERSION}")
 
 set (PACKAGE_NAME "QjackCtl")
 set (PACKAGE_VERSION "${VERSION}")
