--- a/configure.in
+++ b/configure.in
@@ -1633,16 +1633,6 @@
 			#
 			AC_MSG_ERROR([Due to freedesktop.org bug 74029, D-Bus capture support is not available on OS X])
 			;;
-		esac
-	else		
-		case "$host_os" in
-
-		darwin*)
-			#
-			# https://bugs.freedesktop.org/show_bug.cgi?id=74029
-			#
-			;;
-
 		*)
 			AC_CHECK_PROG([PKGCONFIG], [pkg-config], [pkg-config], [no])
 			if test "x$PKGCONFIG" != "xno"; then
