--- core/cmake/BareosCheckSymbols.cmake.orig	2018-12-05 23:39:37.307627387 +0100
+++ core/cmake/BareosCheckSymbols.cmake	2018-12-05 23:40:00.999464771 +0100
@@ -21,7 +21,7 @@
 
 
 CHECK_SYMBOL_EXISTS(__stub_lchmod features.h LCHMOD_IS_A_STUB1)
-CHECK_SYMBOL_EXISTS(__stub___lchmod  features_h LCHMOD_IS_A_STUB2)
+#CHECK_SYMBOL_EXISTS(__stub___lchmod  features_h LCHMOD_IS_A_STUB2)
 
 
 
@@ -43,7 +43,9 @@
 
 cmake_push_check_state()
 SET(CMAKE_REQUIRED_LIBRARIES ${RADOS_LIBRARIES})
+if ("${HAVE_RADOS}")
 CHECK_SYMBOL_EXISTS(rados_ioctx_set_namespace rados/librados.h  HAVE_RADOS_NAMESPACES)
 CHECK_SYMBOL_EXISTS(rados_nobjects_list_open rados/librados.h HAVE_RADOS_NOBJECTS_LIST)
+endif()
 cmake_pop_check_state()
 
