--- docs/CommandGuide/Makefile.orig	2009-10-26 16:54:33.000000000 +0100
+++ docs/CommandGuide/Makefile	2009-10-26 17:32:03.000000000 +0100
@@ -19,7 +19,7 @@
 DST_PS_DIR=ps/
 
 # If we are in BUILD_FOR_WEBSITE mode, default to the all target.
-all:: html man ps
+all:: html man
 
 clean:
 	rm -f pod2htm*.*~~ $(HTML) $(MAN) $(PS)
@@ -81,7 +81,7 @@
 MAN_DIR  := $(PROJ_mandir)/man1
 PS_DIR   := $(PROJ_docsdir)/ps
 
-install-local:: $(HTML) $(INSTALL_MANS) $(PS)
+install-local:: $(HTML) $(INSTALL_MANS)
 	$(Echo) Installing HTML CommandGuide Documentation
 	$(Verb) $(MKDIR) $(HTML_DIR)
 	$(Verb) $(DataInstall) $(HTML) $(HTML_DIR)
@@ -90,9 +90,6 @@
 	$(Echo) Installing MAN CommandGuide Documentation
 	$(Verb) $(MKDIR) $(MAN_DIR)
 	$(Verb) $(DataInstall) $(INSTALL_MANS) $(MAN_DIR)
-	$(Echo) Installing PS CommandGuide Documentation
-	$(Verb) $(MKDIR) $(PS_DIR)
-	$(Verb) $(DataInstall) $(PS) $(PS_DIR)
 
 uninstall-local::
 	$(Echo) Uninstalling CommandGuide Documentation
