From c6040f618d8f2706a7b46d1cdf37d1a587f9701f Mon Sep 17 00:00:00 2001
From: Andrew Stromnov <stromnov@gmail.com>
Date: Thu, 28 Nov 2013 16:58:43 +0400
Subject: [PATCH] fix compiling with FreeType 2.5.1

---
 _imagingft.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/_imagingft.c b/_imagingft.c
index 47d50bd..f19555b 100644
--- a/_imagingft.c
+++ b/_imagingft.c
@@ -59,7 +59,11 @@ struct {
     const char* message;
 } ft_errors[] =
 
+#if defined(USE_FREETYPE_2_1)
+#include FT_ERRORS_H
+#else
 #include <freetype/fterrors.h>
+#endif
 
 /* -------------------------------------------------------------------- */
 /* font objects */
-- 
1.8.5.1

