diff --git a/PIL/ImageQt.py b/PIL/ImageQt.py
index 36b4e1eb..ddde2cd9 100644
--- a/PIL/ImageQt.py
+++ b/PIL/ImageQt.py
@@ -20,24 +20,7 @@ from . import Image
 from ._util import isPath
 from io import BytesIO
 
-qt_is_installed = True
-qt_version = None
-try:
-    from PyQt5.QtGui import QImage, qRgba, QPixmap
-    from PyQt5.QtCore import QBuffer, QIODevice
-    qt_version = '5'
-except (ImportError, RuntimeError):
-    try:
-        from PyQt4.QtGui import QImage, qRgba, QPixmap
-        from PyQt4.QtCore import QBuffer, QIODevice
-        qt_version = '4'
-    except (ImportError, RuntimeError):
-        try:
-            from PySide.QtGui import QImage, qRgba, QPixmap
-            from PySide.QtCore import QBuffer, QIODevice
-            qt_version = 'side'
-        except ImportError:
-            qt_is_installed = False
+qt_is_installed = False
 
 
 def rgb(r, g, b, a=255):
