diff --git a/api/src/java/org/apache/avalon/framework/.Enum.java.swp b/api/src/java/org/apache/avalon/framework/.Enum.java.swp
deleted file mode 100644
index 1eb30e7..0000000
Binary files a/api/src/java/org/apache/avalon/framework/.Enum.java.swp and /dev/null differ
diff --git a/api/src/java/org/apache/avalon/framework/Enum.java b/api/src/java/org/apache/avalon/framework/Enum.java
index 6ceba68..37c2aec 100644
--- a/api/src/java/org/apache/avalon/framework/Enum.java
+++ b/api/src/java/org/apache/avalon/framework/Enum.java
@@ -126,11 +126,11 @@ public abstract class Enum
         if( !(o instanceof Enum) )
             return false;
 
-        final Enum enum = (Enum)o;
+        final Enum enum_ = (Enum)o;
 
-        if( !getClass().equals( enum.getClass() ) )
+        if( !getClass().equals( enum_.getClass() ) )
             return false;
-        if( m_name != null ? !m_name.equals( enum.m_name ) : enum.m_name != null )
+        if( m_name != null ? !m_name.equals( enum_.m_name ) : enum_.m_name != null )
             return false;
 
         return true;
