Index: rednotebook-1.6.5/rednotebook/gui/editor.py
===================================================================
--- rednotebook-1.6.5.orig/rednotebook/gui/editor.py
+++ rednotebook-1.6.5/rednotebook/gui/editor.py
@@ -25,12 +25,7 @@ import gtk
 import gobject
 import pango
 
-# Try to import gtkspell
 gtkspell = None
-try:
-    import gtkspell
-except ImportError:
-    pass
 
 from rednotebook.gui import t2t_highlight
 from rednotebook import undo
