diff -Naur pwavecat-0.4.5.orig/Makefile pwavecat-0.4.5/Makefile
--- pwavecat-0.4.5.orig/Makefile	2005-10-06 15:52:56.000000000 +0200
+++ pwavecat-0.4.5/Makefile	2007-10-06 16:27:25.000000000 +0200
@@ -15,9 +15,9 @@
 parts.o
 
 all:
-	gcc $(CFLAGS) -c $(SRC1)
+	$(CC) $(CFLAGS) -c $(SRC1)
 
-	gcc -o pwavecat   $(OBJECT1) $(LIBRARY)
+	$(CC) $(LDFLAGS) -o pwavecat   $(OBJECT1) $(LIBRARY)
 
 # DEPENDENCIES
 pwavecat.o : pwavecat.c pwavecat.h pwavecat_proto.h parts.h wave_header.h
diff -Naur pwavecat-0.4.5.orig/pwavecat.h pwavecat-0.4.5/pwavecat.h
--- pwavecat-0.4.5.orig/pwavecat.h	2005-10-06 15:52:56.000000000 +0200
+++ pwavecat-0.4.5/pwavecat.h	2007-10-06 16:27:50.000000000 +0200
@@ -27,7 +27,7 @@
 
 char *input_filename[MAX_FILES];
 
-#define VERSION "0.4.4"
+#define VERSION "0.4.5"
 
 #endif  /* _PWAVECAT_H_ */
 
