--- gmorgan-0.23/src/GMORGAN.cxx.old	2006-11-07 14:01:15.000000000 +0100
+++ gmorgan-0.23/src/GMORGAN.cxx	2006-11-07 18:14:56.000000000 +0100
@@ -29997,7 +29997,7 @@
 }
 
 inline void GMORGAN::a_ver_i(Fl_Slider* o, void*) {
-  int kd = (int) o->user_data();
+  long kd = (long) o->user_data();
 int kv = (int) o->value();
 veloplus[kd] = kv;
 }
@@ -30540,7 +30540,7 @@
 }
 
 inline void GMORGAN::song_call_i(Fl_Input* o, void*) {
-  int kd = (int) o->user_data();
+  long kd = (long) o->user_data();
 
 if (kd <= 128) strcpy(S[kd].ch1,(char *) o->value());
 if ((kd > 128) && (kd <=256)) strcpy(S[kd-128].ch2,(char *) o->value());
@@ -30555,7 +30555,7 @@
 }
 
 inline void GMORGAN::song_call1_i(Fl_Value_Input* o, void*) {
-  int kd = (int) o->user_data();
+  long kd = (long) o->user_data();
 if ((kd > 896) && (kd <=1024)) S[kd-896].times = (int) o->value();
 if ((kd > 1024) && (kd <=1152)) S[kd-1024].pattern = (int) o->value();
 if ((kd > 1792) && (kd <=1920)) S[kd-1792].atimes = (int) o->value();
@@ -30567,7 +30567,7 @@
 
 inline void GMORGAN::song_call2_i(Fl_Button* o, void*) {
   int i;
-int kd = (int) o->user_data();
+long kd = (long) o->user_data();
 if ((kd > 256) && (kd <=384))
 {
  for(i=1; i<=128; i++)
@@ -30868,7 +30868,7 @@
 inline void GMORGAN::dr_call1_i(Fl_Value_Input* o, void*) {
   int i,j;
 
-int kd = (int) o->user_data();
+long kd = (long) o->user_data();
 int kv = (int) o->value();
 i = kd / 130;
 j = kd % 130;
@@ -30881,7 +30881,7 @@
 
 inline void GMORGAN::dr_call2_i(Fl_Output* o, void*) {
   lastcasdr = sde + 10000;
-int kd = (int) o->user_data();
+long kd = (long) o->user_data();
 if (kd != lastcasdr) cambiacasdr = 1;
 o->cursor_color(FL_WHITE);
 o->color(FL_WHITE);
@@ -31873,7 +31873,7 @@
 inline void GMORGAN::ctrl_call1_i(Fl_Value_Input* o, void*) {
   int i,j;
 
-int kd = (int) o->user_data();
+long kd = (long) o->user_data();
 int kv = (int) o->value();
 i = kd / 130;
 j = kd % 130;
@@ -31886,7 +31886,7 @@
 
 inline void GMORGAN::ctrl_call2_i(Fl_Output* o, void*) {
   lastcascr = sdec + 10000;
-int kd = (int) o->user_data();
+long kd = (long) o->user_data();
 if (kd != lastcascr) cambiacascr=1;
 sdec = kd-10000;
 o->color(FL_WHITE);
@@ -32956,11 +32956,11 @@
 
 void GMORGAN::PonColores(int usdata) {
   int i;
-int ladata;
+long ladata;
 for(i=0; i<61; i++)
 
 {
-ladata = (int) Scroll1->child(i)->user_data();
+ladata = (long) Scroll1->child(i)->user_data();
 if ( ladata == usdata) 
 {
 Scroll1->child(i)->color(59);
@@ -32972,11 +32972,11 @@
 
 void GMORGAN::PonColores2(int usdata) {
   int i;
-int ladata;
+long ladata;
 for(i=0; i<32; i++)
 
 {
-ladata = (int) CScroll1->child(i)->user_data();
+ladata = (long) CScroll1->child(i)->user_data();
 if ( ladata == usdata) 
 {
 CScroll1->child(i)->color(59);
