From f8beee19f0cd43b5267a57e2bde89f6f43e17276 Mon Sep 17 00:00:00 2001
From: David Rosca <nowrep@gmail.com>
Date: Sun, 13 Aug 2017 13:18:52 +0200
Subject: Autotests: Fix finding fakebluez binary

BUG: 383453
---
 autotests/autotests.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/autotests/autotests.cpp b/autotests/autotests.cpp
index 28a7e1f..46730ea 100644
--- a/autotests/autotests.cpp
+++ b/autotests/autotests.cpp
@@ -62,7 +62,7 @@ private:
 
 StartJob::StartJob()
     : QObject(nullptr)
-    , m_fakebluezPath(QCoreApplication::applicationDirPath() + QStringLiteral("/fakebluez"))
+    , m_fakebluezPath(QCoreApplication::applicationDirPath() + QStringLiteral("/fakebluez/fakebluez"))
 {
     // Fallback for old layout, can be removed after August 2017
     if (!QFile::exists(m_fakebluezPath))
-- 
cgit v0.11.2


