Fix build sys-libs/ncurses[tinfo]

Bug: https://bugs.gentoo.org/690348

--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,7 @@
 #CLOCK	    = -DCLOCK_SUPPORT # Experimental!
 READLINE    = -DREADLINE_SUPPORT
 CFLAGS      += -D_GNU_SOURCE $(COLOR) $(CLOCK) $(READLINE) $(ADD_CFLAGS)
-LDFLAGS     += -lncurses -lreadline
+LDFLAGS     += `pkg-config --libs ncurses` -lreadline
 
 ###########################
 # LINUX with Wide character support (testet with ncurses 5.4)
