From ae11f5cb975749bd8d6ffda48301eb4aadfee35d Mon Sep 17 00:00:00 2001
From: hasufell <hasufell@gentoo.org>
Date: Sun, 6 Oct 2013 21:02:12 +0200
Subject: [PATCH] fix parallel make

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index b2285a0..4cf1753 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -112,7 +112,7 @@ XPaint.ad: app-defaults
 XPaint.ad.h: substads XPaint.ad
 	./substads -ad2c XPaint.ad $@
 
-xpaint.1: xpaint.1.in version.h
+xpaint.1: substads xpaint.1.in version.h
 	./substads -single xpaint.1.in xpaint.1 XPAINT_VERSION $(PACKAGE_VERSION)
 
 install-exec-hook:
-- 
1.8.3.2

