From 69a76e5f4a72d83635cd413b7460df4776b2605f Mon Sep 17 00:00:00 2001
From: hasufell <hasufell@gentoo.org>
Date: Sun, 16 Aug 2015 16:51:26 +0200
Subject: [PATCH] Fix build with clang

---
 rw/readWriteTGA.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rw/readWriteTGA.c b/rw/readWriteTGA.c
index 306a751..9046ee3 100644
--- a/rw/readWriteTGA.c
+++ b/rw/readWriteTGA.c
@@ -607,7 +607,7 @@ int WriteTGA(char *file_name, Image * image)
 	int tga_defdir = 1;
         int transparency = -1;
 
-        if (!image) return;
+        if (!image) return -1;
         w = image->width;
         h = image->height;
         bpp = image->scale;
-- 
2.5.0

