diff --git a/racket/src/bc/Makefile.in b/racket/src/bc/Makefile.in
index 3393e0c55e..6f0014881a 100644
--- a/bc/Makefile.in
+++ b/bc/Makefile.in
@@ -524,7 +524,7 @@ unix-install-3m:
 	$(RESTORE_SIGNATURE) "$(DESTDIR)$(bindir)/racket@MMM_INSTALLED@"
 
 unix-install-libs-3m:
-	cd ..; $(ICP_LIB) bc/libracket3m.@LIBSFX@ "$(DESTDIR)$(libdir)/libracket3m.@LIBSFX@"
+	cd ..; libtool --mode=install install bc/libracket3m.@LIBSFX@ "$(DESTDIR)$(libdir)/libracket3m.@LIBSFX@"
 	cd ..; $(ICP_STATIC_LIB) bc/rktio/librktio.@LIBSFX@ "$(DESTDIR)$(libdir)/librktio.@LIBSFX@"
 	cd ..; $(STRIP_LIB_DEBUG) "$(DESTDIR)$(libdir)/libracket3m.@LIBSFX@"
 
