diff --git a/tests/meson.build b/tests/meson.build
index 377d48a..8a6ebe0 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -87,13 +87,11 @@ pytest_files = [
   'test_clipboard.py',
   'test_documents.py',
   'test_document_fuse.py',
-  'test_dynamiclauncher.py',
   'test_email.py',
   'test_filechooser.py',
   'test_globalshortcuts.py',
   'test_inhibit.py',
   'test_inputcapture.py',
-  'test_location.py',
   'test_openuri.py',
   'test_permission_store.py',
   'test_print.py',
@@ -191,4 +189,4 @@ if enable_installed_tests
       install_dir: installed_tests_data_dir,
     )
   endforeach
-endif
\ No newline at end of file
+endif
