diff -u nss_db-2.2.orig/configure.in nss_db-2.2/configure.in
--- nss_db-2.2.orig/configure.in	2003-06-27 11:41:13.000000000 +0200
+++ nss_db-2.2/configure.in	2003-06-27 11:44:30.000000000 +0200
@@ -43,22 +43,28 @@
 AC_CHECK_HEADER(db.h,, AC_MSG_ERROR([
 *** Could not find Berkeley DB headers.]))
 
-AC_CHECK_LIB(db, db_version,, AC_MSG_ERROR([
+
+AC_CHECK_LIB(db-3, db_version,, 
+  AC_CHECK_LIB(db, db_version,,AC_MSG_ERROR([
 *** Could not find Berkeley DB library.]))
+)
 
 AC_CACHE_CHECK([Berkeley DB version], nss_db_cv_db_version,
   [AC_TRY_CPP(
     [
-#    include <db.h>
-#    ifndef DB_VERSION_STRING
-#     error "Unknown"
-#    endif
-#    if DB_VERSION_MAJOR < 2
-#     error "Too old"
-#    endif
-#    if DB_VERSION_MAJOR == 2 && DB_VERSION_MINOR < 4
-#     error "Too old"
-#    endif
+    include <db.h>
+    ifndef DB_VERSION_STRING
+     error "Unknown"
+    endif
+    if DB_VERSION_MAJOR < 2
+     error "Too old"
+    endif
+    if DB_VERSION_MAJOR == 2 && DB_VERSION_MINOR < 4
+     error "Too old"
+    endif
+    if DB_VERSION_MAJOR >= 4
+     error "Too new"
+    endif
     ],
     nss_db_cv_db_version="ok",
     nss_db_cv_db_version="not ok")
