Index: camlp5-rel712/Makefile
===================================================================
--- camlp5-rel712.orig/Makefile
+++ camlp5-rel712/Makefile
@@ -57,7 +57,7 @@ install:
 	for i in $(DIRS) compile; do \
 	  (cd $$i; $(MAKE) install DESTDIR=$(DESTDIR); cd ..); \
 	done
-	cp etc/topfind.camlp5 `ocamlc -where`/. || true
+	cp etc/topfind.camlp5 "$(DESTDIR)`ocamlc -where`/." || true
 
 uninstall:
 	@if test -z "$(LIBDIR)"; then \
