From 413cc230998c8b6544614c9d23b8e607f0e18a3a Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentnl@gentoo.org>
Date: Thu, 13 Aug 2020 23:17:17 +1200
Subject: Include '.' in @INC for perl 5.26+

---
 Makefile.PL | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.PL b/Makefile.PL
index 9e6cf51..eb9cf4f 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -5,6 +5,7 @@
 #
 use 5.007002;
 use strict;
+use lib q[.];
 use inc::Module::Install;
 use Config;
 
-- 
2.28.0

