--- src/icalendar/prop.py.orig	2010-03-29 21:05:47.000000000 -0400
+++ src/icalendar/prop.py	2010-03-29 21:01:47.000000000 -0400
@@ -585,7 +585,7 @@
                 wrong_type_used = 0
         if wrong_type_used:
             raise ValueError ('You must use datetime, date or timedelta')
-        if isinstance(dt, date):
+        if type(dt) is date:
             self.params = Parameters(dict(value='DATE'))
 
         self.dt = dt
--- src/icalendar/cal.py.orig	2009-12-14 08:43:50.000000000 -0500
+++ src/icalendar/cal.py	2010-04-25 09:11:44.000000000 -0400
@@ -520,7 +520,7 @@
     >>> event.set('dtstart', datetime(2005,4,4,8,0,0))
     >>> cal.add_component(event)
     >>> cal.subcomponents[0].as_string()
-    'BEGIN:VEVENT\\r\\nDTSTART;VALUE=DATE:20050404T080000\\r\\nSUMMARY:Python meeting about calendaring\\r\\nUID:42\\r\\nEND:VEVENT\\r\\n'
+    'BEGIN:VEVENT\\r\\nDTSTART:20050404T080000\\r\\nSUMMARY:Python meeting about calendaring\\r\\nUID:42\\r\\nEND:VEVENT\\r\\n'
 
     Write to disc
     >>> import tempfile, os
