diff --git a/startx.cpp b/startx.cpp
index dfbebe1..082f780 100644
--- a/startx.cpp
+++ b/startx.cpp
@@ -202,17 +202,6 @@ XCOMM process server arguments
 if [ x"$server" = x ]; then
     server=$defaultserver
 
-#ifdef __linux__
-    XCOMM When starting the defaultserver start X on the current tty to avoid
-    XCOMM the startx session being seen as inactive:
-    XCOMM "https://bugzilla.redhat.com/show_bug.cgi?id=806491"
-    tty=$(tty)
-    if expr "$tty" : '/dev/tty[0-9][0-9]*$' > /dev/null; then
-        tty_num=${tty#/dev/tty}
-        vtarg="vt$tty_num -keeptty"
-    fi
-#endif
-
     XCOMM For compatibility reasons, only use xserverrc if there were no server command line arguments
     if [ x"$serverargs" = x -a x"$display" = x ]; then
 	if [ -f "$userserverrc" ]; then
-- 
2.37.4

