diff -ur a/startx.cpp b/startx.cpp
--- a/startx.cpp	2014-09-11 19:31:42.000000000 +0200
+++ b/startx.cpp	2015-02-08 19:09:56.416428623 +0100
@@ -188,17 +188,6 @@
 if [ x"$server" = x ]; then
     server=$defaultserver
 
-#ifdef __linux__
-    XCOMM When starting the defaultserver start X on the current tty to avoid
-    XCOMM the startx session being seen as inactive:
-    XCOMM "https://bugzilla.redhat.com/show_bug.cgi?id=806491"
-    tty=$(tty)
-    if expr match "$tty" '^/dev/tty[0-9]\+$' > /dev/null; then
-        tty_num=$(echo "$tty" | grep -oE '[0-9]+$')
-        vtarg="vt$tty_num"
-    fi
-#endif
-
     XCOMM For compatibility reasons, only use xserverrc if there were no server command line arguments
     if [ x"$serverargs" = x -a x"$display" = x ]; then
 	if [ -f "$userserverrc" ]; then
