From https://bugzilla.xfce.org/show_bug.cgi?id=11107#c3

diff --git a/xfsettingsd/displays.c b/xfsettingsd/displays.c
index 3497804..449500b 100644
--- a/xfsettingsd/displays.c
+++ b/xfsettingsd/displays.c
@@ -368,7 +368,7 @@ xfce_displays_helper_reload (XfceDisplaysHelper *helper)
     xfsettings_dbg (XFSD_DEBUG_DISPLAYS, "Refreshing RandR cache.");
 
     /* Free the caches */
-    g_ptr_array_unref (helper->outputs);
+    g_ptr_array_free (helper->outputs, TRUE);
     g_ptr_array_unref (helper->crtcs);
 
     gdk_error_trap_push ();
