diff -ru sndpeek-1.3~/src/marsyas/LPC.h sndpeek-1.3/src/marsyas/LPC.h
--- sndpeek-1.3~/src/marsyas/LPC.h	2008-06-28 12:37:05.000000000 +0200
+++ sndpeek-1.3/src/marsyas/LPC.h	2008-06-28 12:37:15.000000000 +0200
@@ -35,7 +35,7 @@
 #include "MagFFT.h"
 #include "fvec.h"
 #include "fmatrix.h"
-#include <math.h>
+#include <cstdlib>
 #include "AutoCorrelation.h"
 #include "MarSignal.h"
 
diff -ru sndpeek-1.3~/src/sndpeek/RtAudio.h sndpeek-1.3/src/sndpeek/RtAudio.h
--- sndpeek-1.3~/src/sndpeek/RtAudio.h	2008-06-28 12:37:05.000000000 +0200
+++ sndpeek-1.3/src/sndpeek/RtAudio.h	2008-06-28 12:37:15.000000000 +0200
@@ -43,8 +43,9 @@
 #define __RTAUDIO_H
 
 #include "RtError.h"
-#include <string>
+#include <cstring>
 #include <vector>
+#include <cstdlib>
 using namespace std;
 
 // Operating system dependent thread functionality.
