--- CMakeLists.txt.orig	2015-06-13 01:47:50.493401323 -0400
+++ CMakeLists.txt	2015-06-13 01:48:53.691255054 -0400
@@ -128,6 +128,9 @@
 find_package(OpenGL REQUIRED)
 include_directories(${OPENGL_INCLUDE_DIR})
 
+find_package(X11 REQUIRED)
+include_directories(${X11_INCLUDE_DIR})
+
 if(UNIX AND NOT APPLE)
     find_library(IRRLICHT_XF86VM_LIBRARY Xxf86vm)
     mark_as_advanced(IRRLICHT_XF86VM_LIBRARY)
@@ -239,6 +242,7 @@
     ${IRRLICHT_XF86VM_LIBRARY}
     ${OPENAL_LIBRARY}
     ${OPENGL_LIBRARIES}
+    ${X11_LIBRARIES}
     png
     jpeg
     bz2
