--- abcmidi/Makefile.in.orig
+++ abcmidi/Makefile.in
@@ -140,17 +140,10 @@
 	rm *.o ${binaries}
 
 install: abc2midi midi2abc abc2abc mftext midicopy yaps abcmatch
-	test -d $(DESTDIR)${prefix}/${bindir} || mkdir -p $(DESTDIR)${prefix}/${bindir}
-	$(INSTALL) -m 755 ${binaries} $(DESTDIR)${prefix}/${bindir}
-
-	# install documentation
-	test -d $(DESTDIR)${PREFIX}/share/doc/abcmidi || mkdir -p $(DESTDIR)${prefix}/${docdir}
-	$(INSTALL) -m 644 doc/*.txt $(DESTDIR)${prefix}/${docdir}
-	$(INSTALL) -m 644 doc/AUTHORS $(DESTDIR)${prefix}/${docdir}
-	$(INSTALL) -m 644 doc/CHANGES $(DESTDIR)${prefix}/${docdir}
-	$(INSTALL) -m 644 VERSION $(DESTDIR)${prefix}/${docdir}
+	test -d $(DESTDIR)${bindir} || mkdir -p $(DESTDIR)${bindir}
+	$(INSTALL) -m 755 ${binaries} $(DESTDIR)${bindir}
 
 	# install manpages
-	test -d $(DESTDIR)${prefix}/${mandir} || mkdir -p $(DESTDIR)${prefix}/${mandir};
-	$(INSTALL) -m 644 doc/*.1 $(DESTDIR)${prefix}/${mandir}
+	test -d $(DESTDIR)${mandir} || mkdir -p $(DESTDIR)${mandir}
+	$(INSTALL) -m 644 doc/*.1 $(DESTDIR)${mandir}
 
