diff --git a/configure.in b/configure.in
index d6aa366..20aafb8 100644
--- a/configure.in
+++ b/configure.in
@@ -165,11 +165,11 @@ XSLT_LOCALE_XLOCALE=0
 XSLT_LOCALE_WINAPI=0
 
 AC_CHECK_HEADERS([locale.h xlocale.h])
-if test $ac_cv_header_xlocale_h = yes; then
+if test $ac_cv_header_locale_h = yes; then
 dnl
 dnl Check for generic locale_t declaration
 dnl
-AC_MSG_CHECKING([if xlocale program link])
+AC_MSG_CHECKING([if locale program link])
 AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
diff --git a/libxslt/xsltlocale.h b/libxslt/xsltlocale.h
index 8a9ca15..2f64424 100644
--- a/libxslt/xsltlocale.h
+++ b/libxslt/xsltlocale.h
@@ -17,7 +17,9 @@
 #ifdef XSLT_LOCALE_XLOCALE
 
 #include <locale.h>
+#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ < 26
 #include <xlocale.h>
+#endif
 
 #ifdef __GLIBC__
 /*locale_t is defined only if _GNU_SOURCE is defined*/
