From aa30e1075973b1ca3e27ad466674aeff3c5de461 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Sat, 20 Aug 2022 22:15:51 +0200
Subject: [PATCH] Drop unused KF5DBusAddons

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 apps/lib/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/apps/lib/CMakeLists.txt b/apps/lib/CMakeLists.txt
index a883cf5c5..031a142ea 100644
--- a/apps/lib/CMakeLists.txt
+++ b/apps/lib/CMakeLists.txt
@@ -32,7 +32,6 @@ find_package(
   COMPONENTS
     TextEditor
     WindowSystem
-    DBusAddons
     Crash
   OPTIONAL_COMPONENTS
     Activities
@@ -44,7 +43,6 @@ target_link_libraries(
     KF5::I18n
     KF5::TextEditor
     KF5::WindowSystem
-    KF5::DBusAddons
     KF5::Crash
     KF5::CoreAddons
 )
-- 
2.37.2

