--- a/lib/dbdriver.cpp
+++ b/lib/dbdriver.cpp
@@ -257,7 +257,11 @@
 	}
 	
 	if ((n == 1) &&
+#ifdef CLIENT_LONG_PASSWORD
 			(o >= CLIENT_LONG_PASSWORD) &&
+#else
+			(o >= CLIENT_MYSQL) &&
+#endif
 #if MYSQL_VERSION_ID > 40000	// highest flag value varies by version
 			(o <= CLIENT_MULTI_RESULTS)
 #else
--- a/lib/common.h
+++ b/lib/common.h
@@ -130,12 +130,6 @@
 	#define MYSQLPP_PATH_SEPARATOR '/'
 #endif
 
-#if defined(MYSQLPP_MYSQL_HEADERS_BURIED)
-#	include <mysql/mysql_version.h>
-#else
-#	include <mysql_version.h>
-#endif
-
 namespace mysqlpp {
 
 /// \brief Alias for 'true', to make code requesting exceptions more
