diff --git a/spec/applications/browser_spec.rb b/spec/applications/browser_spec.rb
index 8cfa8e7..3886ce3 100644
--- a/spec/applications/browser_spec.rb
+++ b/spec/applications/browser_spec.rb
@@ -12,20 +12,7 @@ describe Launchy::Application::Browser do
     ENV.delete( 'KDE_FULL_SESSION' )
     ENV.delete( 'BROWSER' )
   end
 
-  { 'windows' => 'start "launchy" /b' ,
-    'darwin'  => '/usr/bin/open',
-    'cygwin'  => 'cmd /C start "launchy" /b',
-
-    # when running these tests on a linux box, this test will fail
-    'linux'   => nil                 }.each  do |host_os, cmdline|
-    it "when host_os is '#{host_os}' the appropriate 'app_list' method is called" do
-      Launchy.host_os = host_os
-      browser = Launchy::Application::Browser.new
-      browser.app_list.first.must_equal cmdline
-    end
-  end
-
   %w[ linux windows darwin cygwin ].each do |host_os|
     it "the BROWSER environment variable overrides any host defaults on '#{host_os}'" do
       ENV['BROWSER'] = "my_special_browser --new-tab '%s'"
