--- configure.ac.old	2015-11-21 13:17:01.399591500 +0100
+++ configure.ac	2015-11-21 13:27:24.407089263 +0100
@@ -906,17 +906,12 @@
 dnl *** soup ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_SOUP, true)
 AG_GST_CHECK_FEATURE(SOUP, [soup http client plugin (2.4)], souphttpsrc, [
-  PKG_CHECK_MODULES(SOUP, libsoup-gnome-2.4 >= 2.3.2, [
-    HAVE_SOUP="yes"
-    AC_DEFINE(HAVE_LIBSOUP_GNOME, 1, [soup gnome integration])
-  ],[
     PKG_CHECK_MODULES(SOUP, libsoup-2.4 >= 2.26, HAVE_SOUP="yes", [
-      HAVE_SOUP="no"
-    ])
-  ])
+    HAVE_SOUP="no"])
+  ]
   AC_SUBST(SOUP_CFLAGS)
   AC_SUBST(SOUP_LIBS)
-])
+)
 
 dnl *** speex >= 1.0.4 or >= 1.1.5 ***
 dnl   1.1.4 and earlier were not API/ABI compatible with 1.0
