https://bugs.gentoo.org/905453

--- a/src/modules/FFmpeg/CMakeLists.txt
+++ b/src/modules/FFmpeg/CMakeLists.txt
@@ -67,7 +67,7 @@ if(USE_FFMPEG_VAAPI OR USE_FFMPEG_VDPAU OR USE_FFMPEG_DXVA2 OR USE_FFMPEG_D3D11V
             list(APPEND FFmpeg_HDR VAAPIVulkan.hpp)
             list(APPEND FFmpeg_SRC VAAPIVulkan.cpp)
         endif()
-        pkg_check_modules(LBVA REQUIRED libva libva-glx libva-drm egl)
+        pkg_check_modules(LBVA REQUIRED libva libva-drm egl)
         include_directories(${LBVA_INCLUDE_DIRS})
         link_directories(${LBVA_LIBRARY_DIRS})
         list(APPEND FFmpeg_LIBS ${LBVA_LIBRARIES})
