--- a/focuswriter.pro
+++ b/focuswriter.pro
@@ -7,6 +7,7 @@
 
 TEMPLATE = app
 QT += network widgets printsupport multimedia concurrent
+CONFIG += qtsingleapplication
 macx {
 	QT += macextras
 }
@@ -83,7 +84,7 @@
 		src/spelling/dictionary_provider_voikko.cpp
 }
 
-INCLUDEPATH += src src/fileformats src/qtsingleapplication src/qtzip src/spelling
+INCLUDEPATH += src src/fileformats src/qtzip src/spelling
 
 # Specify program sources
 HEADERS += src/action_manager.h \
@@ -142,8 +143,6 @@
 	src/fileformats/rtf_tokenizer.h \
 	src/fileformats/rtf_writer.h \
 	src/fileformats/txt_reader.h \
-	src/qtsingleapplication/qtsingleapplication.h \
-	src/qtsingleapplication/qtlocalpeer.h \
 	src/qtzip/qtzipreader.h \
 	src/qtzip/qtzipwriter.h \
 	src/spelling/abstract_dictionary.h \
@@ -207,8 +206,6 @@
 	src/fileformats/rtf_tokenizer.cpp \
 	src/fileformats/rtf_writer.cpp \
 	src/fileformats/txt_reader.cpp \
-	src/qtsingleapplication/qtsingleapplication.cpp \
-	src/qtsingleapplication/qtlocalpeer.cpp \
 	src/qtzip/qtzip.cpp \
 	src/spelling/dictionary_dialog.cpp \
 	src/spelling/dictionary_manager.cpp \
