From 69c66475d2887e50150ceb4481934749ad46feb6 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Thu, 21 May 2020 01:38:02 +0200
Subject: [PATCH] Support the AR variable

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 2e2fcc3..897282e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -30,6 +30,7 @@ fi
 # Checks for programs.
 AC_PROG_INSTALL
 AC_PROG_CC
+AM_PROG_AR
 AC_PROG_RANLIB
 if test "$USE_MAN" = "yes"; then
 	RSVN_CHECK_MAN_PROGS
-- 
2.26.2

