From 1e0b3f4556f88013a2268bb2ef0c8d4bfaa40f90 Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org.ua>
Date: Fri, 14 Apr 2017 07:48:35 +0000
Subject: Fix a typo

---
diff --git a/src/gdbm.h.in b/src/gdbm.h.in
index e9b7b1b..de0d2b9 100644
--- a/src/gdbm.h.in
+++ b/src/gdbm.h.in
@@ -224,7 +224,7 @@ extern int gdbm_copy_meta (GDBM_FILE dst, GDBM_FILE src);
 # define _GDBM_MAX_ERRNO	GDBM_DIR_OVERFLOW
 
 /* This one was never used and will be removed in the future */
-# define GDBM_UNKNOWN_UPDATE GDBM_ERR_UNKNOWN_ERROR
+# define GDBM_UNKNOWN_UPDATE GDBM_UNKNOWN_ERROR
   
 typedef int gdbm_error;
 extern gdbm_error gdbm_errno;
--
cgit v0.9.0.3
