diff --git a/CorsixTH/CMakeLists.txt b/CorsixTH/CMakeLists.txt
index dfdb59e..9b97b3c 100755
--- a/CorsixTH/CMakeLists.txt
+++ b/CorsixTH/CMakeLists.txt
@@ -191,5 +191,5 @@ ELSE()
   install(DIRECTORY Bitmap DESTINATION CorsixTH
         FILES_MATCHING REGEX ".*\\.(tab|pal|dat|png)$"
         PATTERN "*.svn" EXCLUDE)
-  install(FILES CorsixTH.lua LICENSE.txt CorsixTH.ico DESTINATION CorsixTH )
+  install(FILES CorsixTH.lua CorsixTH.ico DESTINATION CorsixTH )
 ENDIF()
