diff --git a/src/GPSInterface_Serial.cpp b/src/GPSInterface_Serial.cpp
index c3e9435..fcae209 100644
--- a/src/GPSInterface_Serial.cpp
+++ b/src/GPSInterface_Serial.cpp
@@ -243,7 +243,7 @@ IGPSInterface::EGPSStatus GPSInterface_Serial::GetData(wxGPSEvent * pGPSEvent)
 				
 				// Echo DeLorme Tripmate's and Earthmate's hello message
 				LibRoadnavDebug1(wxT("SerialIO"), wxT("Echoing hello message: %s"), strSentence.c_str());
-				sprintf(szOut, "%s\r\n", strSentence.mb_str(*wxConvCurrent));
+				sprintf(szOut, "%s\r\n", static_cast<const char *>(strSentence.mb_str(*wxConvCurrent)));
 				WriteSerial(szOut);
 			}
 			else if (!VerifyGPSChecksum(strSentence)) // check NMEA validity
diff --git a/src/TTS_Festival.cpp b/src/TTS_Festival.cpp
index 8623cb3..2950490 100644
--- a/src/TTS_Festival.cpp
+++ b/src/TTS_Festival.cpp
@@ -88,7 +88,7 @@ ITTS::ETTSStatus TTS_Festival::Speak(const wxString & strText)
 			ppszArgs[2], 
 			64 + strText.Length(),
 			wxT("echo \\(SayText \\\"%ss\\\"\\) | festival"), 
-			strText.mb_str(*wxConvCurrent)
+			static_cast<const char *>(strText.mb_str(*wxConvCurrent))
 			);
 
 	wxExecute(ppszArgs, wxEXEC_SYNC);
