diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1491,11 +1491,12 @@
 ])
 
 # check for gtk-doc
-m4_ifdef([GTK_DOC_CHECK], [
-    GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
-],[
-    AM_CONDITIONAL([ENABLE_GTK_DOC], false)
-])
+#m4_ifdef([GTK_DOC_CHECK], [
+#    GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
+#],[
+#    AM_CONDITIONAL([ENABLE_GTK_DOC], false)
+#])
+GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
 
 dnl Library versioning.
 MAX_PROC_NR=`cat $srcdir/src/MAX_PROC_NR`
