 configure.ac |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0a3d7c1..6f8356d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -150,11 +150,10 @@ if test x"$use_schemas" = x"yes"; then
   fi
   xsltdir=`$PKG_CONFIG --variable=xsltdir compiz-gconf`
   AC_SUBST(xsltdir)
+else
+  AM_CONDITIONAL(GCONF_SCHEMAS_INSTALL, test "x$use_gconf" = "xyes")
 fi
 
-AM_GCONF_SOURCE_2
-
-
 AM_CONDITIONAL(USE_SCHEMAS, test "x$use_schemas" = "xyes")
 if test "$use_schemas" = yes; then
   AC_DEFINE(USE_SCHEMAS, 1, [Build gconf schemas])
