--- a/src/probes/unix/Process58Probe.cpp
+++ b/src/probes/unix/Process58Probe.cpp
@@ -743,26 +743,8 @@
 }
 
 bool Process58Probe::RetrieveSelinuxDomainLabel(pid_t pid, string *label, string *err) {
-	security_context_t sctx;
-	int ec = getpidcon(pid, &sctx);
-	if (ec == -1) {
-		// getpidcon man page doesn't say errno is set... so we can't get a
-		// reason for the error.
-		*err = "getpidcon() failed";
-		return false;
-	}
-
-	SecurityContextGuard scg(sctx);
-	ContextGuard cg(sctx);
-
-	const char *tmp = context_type_get(cg);
-	if (!tmp) {
-		*err = string("context_get_type(")+sctx+"): "+strerror(errno);
-		return false;
-	}
-
-	*label = tmp;
-	return true;
+	*err = string("context_get_type(NotImplmented)");
+	return false;
 }
 
 #elif defined SUNOS
