--- a/build/Makefile.linux
+++ b/build/Makefile.linux
@@ -24,13 +24,13 @@ EXECS=server/maradns
 # end the Solaris section
 # Non-Solaris version of "M"
 V="VERSION=$(VERSION)"
-Q="DEFINES=-DSELECT_PROBLEM"
+Q="$(LDFLAGS) DEFINES=-DSELECT_PROBLEM"
 
 # Debug
 
 FLAGS = -O2 -Wall -DSELECT_PROBLEM
-M="CC=$(CC) $(FLAGS)"
-D="CC=$(CC) $(FLAGS) -DDEBUG -DTHREADS"
+M="CC=$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS)"
+D="CC=$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -DDEBUG -DTHREADS"
 #FLAGS = -g
 
 all: 
@@ -39,7 +39,7 @@ all:
 	cd ../qual ; make $(M) ; cd ../server ; \
 	make $(M) $(V) COMPILED=\"$(COMPILED)\" ; \
 	cd ../tools ; make $(M) ; \
-	cd ../deadwood-*/src/ ; make FLAGS=-O2 ; \
+	cd ../deadwood-*/src/ ; $(MAKE)  $(M) ; \
 	cd ../../tcp ; make $(M) $(V) ; cat ../00README.FIRST
 
 debug: 
