Respect externally set PYTHON_MODULE_INSTALL_DIR.

--- csound-6.09.1/CMakeLists.txt
+++ csound-6.09.1/CMakeLists.txt
@@ -218,11 +218,13 @@
     set(PLUGIN_INSTALL_DIR "${LIBRARY_INSTALL_DIR}/csound/plugins-${APIVERSION}")
 endif()
 
-execute_process (
-     COMMAND python -c
-     "import site, sys; sys.stdout.write(site.getusersitepackages())"
-     OUTPUT_VARIABLE PYTHON_MODULE_INSTALL_DIR
-   )
+if(NOT DEFINED PYTHON_MODULE_INSTALL_DIR)
+    execute_process (
+        COMMAND python -c
+        "import site, sys; sys.stdout.write(site.getusersitepackages())"
+        OUTPUT_VARIABLE PYTHON_MODULE_INSTALL_DIR
+    )
+endif()
 
 set(JAVA_MODULE_INSTALL_DIR ${LIBRARY_INSTALL_DIR})
 set(LUA_MODULE_INSTALL_DIR ${LIBRARY_INSTALL_DIR})
