http://bugs.gentoo.org/335511

--- cdrwtool/cdrwtool.c
+++ cdrwtool/cdrwtool.c
@@ -31,6 +31,7 @@
 #include <sys/ioctl.h>
 #include <asm/param.h>
 
+#include <limits.h> /* INT_MAX */
 #include <linux/cdrom.h>
 
 #include "cdrwtool.h"
--- pktsetup/pktsetup.c
+++ pktsetup/pktsetup.c
@@ -26,6 +26,8 @@
 #include <bits/types.h>
 #include <sys/types.h>
 
+#include <limits.h> /* INT_MAX */
+#include <stdlib.h> /* system */
 #include <linux/cdrom.h>
 
 /*
