diff -Naurp FusionSound-1.1.1-original/configure.in FusionSound-1.1.1/configure.in
--- FusionSound-1.1.1-original/configure.in	2008-02-11 05:31:45.000000000 -0600
+++ FusionSound-1.1.1/configure.in	2009-04-10 15:18:02.000000000 -0600
@@ -513,11 +513,11 @@ AC_ARG_WITH(ffmpeg,
     
 if test "x$with_ffmpeg" = "xyes"; then
   AC_MSG_CHECKING([for libavcodec/libavformat])
-  if $PKG_CONFIG libavcodec libavformat; then
+  if $PKG_CONFIG libavcodec libavformat libavutil; then
     AC_MSG_RESULT(yes)
     ffmpeg="yes"
-    FFMPEG_CFLAGS=`$PKG_CONFIG --cflags libavcodec libavformat`
-    FFMPEG_LIBS=`$PKG_CONFIG --libs libavcodec libavformat`
+    FFMPEG_CFLAGS=`$PKG_CONFIG --cflags libavcodec libavformat libavutil`
+    FFMPEG_LIBS=`$PKG_CONFIG --libs libavcodec libavformat libavutil`
   else
     AC_MSG_RESULT(no)
     AC_MSG_WARN([
diff -Naurp FusionSound-1.1.1-original/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c FusionSound-1.1.1/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c
--- FusionSound-1.1.1-original/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c	2008-02-11 05:31:45.000000000 -0600
+++ FusionSound-1.1.1/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c	2009-04-10 15:18:21.000000000 -0600
@@ -41,8 +41,8 @@
 
 #include <misc/sound_util.h>
 
-#include <avcodec.h>
-#include <avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 
 static DFBResult
 Probe( IFusionSoundMusicProvider_ProbeContext *ctx );
@@ -123,8 +123,8 @@ av_read_callback( void *opaque, uint8_t 
      return len;
 }
 
-static offset_t
-av_seek_callback( void *opaque, offset_t offset, int whence )
+static int64_t
+av_seek_callback( void *opaque, int64_t offset, int whence )
 {
      IFusionSoundMusicProvider_FFmpeg_data *data = opaque;
      unsigned int                           pos  = 0;
