Fixes a segfault on probe.

Index: FusionSound-1.1.1/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c
===================================================================
--- FusionSound-1.1.1.orig/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c
+++ FusionSound-1.1.1/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c
@@ -1138,7 +1138,7 @@ IFusionSoundMusicProvider_FFmpeg_WaitSta
 static DFBResult
 Probe( IFusionSoundMusicProvider_ProbeContext *ctx )
 {
-     AVProbeData    pd;
+     AVProbeData    pd = { 0 };
      AVInputFormat *format;
           
      av_register_all();
@@ -1165,7 +1165,7 @@ Construct( IFusionSoundMusicProvider *th
            const char                *filename,
            DirectStream              *stream )
 {
-     AVProbeData    pd;
+     AVProbeData    pd = { 0 };
      AVInputFormat *fmt;
      AVCodec       *c;
      unsigned char  buf[64];
