# Work around Gentoo's crippled dash

--- pingus-0.7.6/Makefile
+++ pingus-0.7.6/Makefile
@@ -39,7 +39,7 @@
 	install -d "$(DESTDIR)$(BINDIR)"
 
 	install -D build/pingus "$(DESTDIR)$(BINDIR)/pingus.bin"
-	echo -e "#!/bin/sh\nexec \"$(BINDIR)/pingus.bin\" --datadir \"$(DATADIR)\"" > "$(DESTDIR)$(BINDIR)/pingus"
+	printf '%s\n%s' "#!/bin/sh" "exec \"$(BINDIR)/pingus.bin\" --datadir \"$(DATADIR)\"" > "$(DESTDIR)$(BINDIR)/pingus"
 	chmod 755 "$(DESTDIR)$(BINDIR)/pingus"
 
 install-data:
