diff -ur grass-6.4.1.orig/gui/wxpython/Makefile grass-6.4.1/gui/wxpython/Makefile
--- grass-6.4.1.orig/gui/wxpython/Makefile
+++ grass-6.4.1/gui/wxpython/Makefile
@@ -11,7 +11,7 @@
 ETCDIR = $(ETC)/wxpython
 
 SRCFILES := $(wildcard compat/* gui_modules/* icons/*.* icons/silk/* images/* xml/*) gis_set.py wxgui.py README
-DSTFILES := $(patsubst %,$(ETCDIR)/%,$(SRCFILES)) $(patsubst %.py,$(ETCDIR)/%.pyc,$(filter %.py,$(SRCFILES)))
+DSTFILES := $(patsubst %,$(ETCDIR)/%,$(SRCFILES))
 
 default: install_scripts
 	$(MAKE) parsubdirs
