diff -ruN hydrogen-0.9.5.orig/libs/hydrogen/src/local_file_mgr.cpp hydrogen-0.9.5/libs/hydrogen/src/local_file_mgr.cpp
--- hydrogen-0.9.5.orig/libs/hydrogen/src/local_file_mgr.cpp	2011-01-29 00:33:23.000000000 +0100
+++ hydrogen-0.9.5/libs/hydrogen/src/local_file_mgr.cpp	2017-09-26 20:07:31.128723252 +0200
@@ -980,7 +980,7 @@
 	
 	std::string playlistInfoFile = patternname;
 	std::ifstream verify( playlistInfoFile.c_str() , std::ios::in | std::ios::binary );
-	if ( verify == NULL ) {
+	if ( !verify ) {
 		//ERRORLOG( "Load Playlist: Data file " + playlistInfoFile + " not found." );
 		return NULL;
 	}
