diff --git a/lutris/gui/dialogs/webconnect_dialog.py b/lutris/gui/dialogs/webconnect_dialog.py
index 12fbbdc..a6cd838 100644
--- a/lutris/gui/dialogs/webconnect_dialog.py
+++ b/lutris/gui/dialogs/webconnect_dialog.py
@@ -3,7 +3,11 @@ import os
 from gettext import gettext as _
 
 import gi
-gi.require_version("WebKit2", "4.0")
+try:
+    gi.require_version("WebKit2", "4.1")
+except (ImportError, ValueError):
+    gi.require_version("WebKit2", "4.0")
+
 from gi.repository import WebKit2
 
 from lutris.gui.dialogs import ModalDialog
