From 99e6e5819118c3ebf06bccc879a15175b9221148 Mon Sep 17 00:00:00 2001
From: Carl Eastlund <ceastlund@janestreet.com>
Date: Thu, 19 Jan 2023 12:25:16 -0500
Subject: [PATCH] Stop using [Ppxlib.File_path].

Signed-off-by: Carl Eastlund <ceastlund@janestreet.com>
---
 src/ppx_inline_test.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ppx_inline_test.ml b/src/ppx_inline_test.ml
index 7f2e854..37c9791 100644
--- a/src/ppx_inline_test.ml
+++ b/src/ppx_inline_test.ml
@@ -82,7 +82,7 @@ let rec short_desc_of_expr ~max_len e =
 ;;
 
 let descr ~(loc:Location.t) ?(inner_loc=loc) e_opt id_opt =
-  let filename  = File_path.get_default_path loc                 in
+  let filename  = loc.loc_start.pos_fname                        in
   let line      = loc.loc_start.pos_lnum                         in
   let start_pos = loc.loc_start.pos_cnum - loc.loc_start.pos_bol in
   let end_pos   = inner_loc.Location.loc_end.pos_cnum - loc.loc_start.pos_bol in
