diff --git a/Makefile.in b/Makefile.in
index b388455..52d3bc3 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -58,8 +58,8 @@ CFLAGS = @MYCFLAGS@
 LDFLAGS = @MYLDFLAGS@
 CMDLDFLAGS = @MYCMDLDFLAGS@
 LIBS = @LIBS@
-LDENV = LD_RUN_PATH=/lib:/usr/lib:$(LIBDIR):$(HOME)/lib:/usr/local/lib:@MYRUNPATH@:.
-RUNENV = @MYLDLIBPATHENV@=.:/lib:/usr/lib:$(LIBDIR):$(HOME)/lib:/usr/local/lib:@MYRUNPATH@
+LDENV = 
+RUNENV = @MYLDLIBPATHENV@=.:$(LIBDIR)
 POSTCMD = @MYPOSTCMD@
 
 
@@ -120,10 +120,6 @@ install :
 	cp -Rf $(LIBRARYFILES) $(DESTDIR)$(LIBDIR)
 	mkdir -p $(DESTDIR)$(BINDIR)
 	cp -Rf $(COMMANDFILES) $(DESTDIR)$(BINDIR)
-	mkdir -p $(DESTDIR)$(SBINDIR)
-	cp -Rf $(INITFILES) $(DESTDIR)$(SBINDIR)
-	mkdir -p $(DESTDIR)$(DATADIR)
-	cp -Rf $(SCREXTFILES) $(DOCUMENTFILES) $(DESTDIR)$(DATADIR)
 	mkdir -p $(DESTDIR)$(MAN1DIR)
 	cd man && cp -Rf $(MAN1FILES) $(DESTDIR)$(MAN1DIR)
 	mkdir -p $(DESTDIR)$(MAN3DIR)
@@ -132,7 +128,6 @@ install :
 	cd man && cp -Rf $(MAN8FILES) $(DESTDIR)$(MAN8DIR)
 	mkdir -p $(DESTDIR)$(PCDIR)
 	cp -Rf $(PCFILES) $(DESTDIR)$(PCDIR)
-	[ "$$UID" == 0 ] && ldconfig || true
 	@printf '\n'
 	@printf '#================================================================\n'
 	@printf '# Thanks for using Tokyo Tyrant.\n'
diff --git a/configure.in b/configure.in
index fbe0c71..809a839 100644
--- a/configure.in
+++ b/configure.in
@@ -28,33 +28,15 @@ MYDOCUMENTFILES="COPYING ChangeLog THANKS doc tokyotyrant.idl"
 MYPCFILES="tokyotyrant.pc"
 
 # Building flags
-MYCFLAGS="-std=c99 -Wall -fPIC -fsigned-char -O2"
+MYCFLAGS="-std=c99 -Wall -fPIC -fsigned-char"
 MYCPPFLAGS="-I. -I\$(INCLUDEDIR) -I$HOME/include -I/usr/local/include"
 MYCPPFLAGS="$MYCPPFLAGS -DNDEBUG -D_GNU_SOURCE=1 -D_REENTRANT -D__EXTENSIONS__"
-MYLDFLAGS="-L. -L\$(LIBDIR) -L$HOME/lib -L/usr/local/lib"
+MYLDFLAGS="-L. -L\$(LIBDIR)"
 MYCMDLDFLAGS=""
 MYRUNPATH="\$(LIBDIR)"
 MYLDLIBPATHENV="LD_LIBRARY_PATH"
 MYPOSTCMD="true"
 
-# Building paths
-PATH="$PATH:$HOME/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin"
-PATH="$PATH:/opt/SUNWspro/bin:/usr/ccs/bin:/usr/xpg4/bin:/usr/xpg6/bin:/usr/ucb"
-CPATH="$HOME/include:/usr/local/include:$CPATH"
-LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LIBRARY_PATH"
-LD_LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LD_LIBRARY_PATH"
-PKG_CONFIG_PATH="$HOME/lib/pkgconfig:/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH"
-export PATH CPATH LIBRARY_PATH LD_LIBRARY_PATH PKG_CONFIG_PATH
-if type pkg-config > /dev/null 2>&1
-then
-  PATH="$PATH:`pkg-config --variable=bindir tokyocabinet`"
-  CPATH="$CPATH:`pkg-config --variable=includedir tokyocabinet`"
-  LIBRARY_PATH="$LIBRARY_PATH:`pkg-config --variable=libdir tokyocabinet`"
-  LD_LIBRARY_PATH="$LD_LIBRARY_PATH:`pkg-config --variable=libdir tokyocabinet`"
-fi
-
-
-
 #================================================================
 # Options
 #================================================================
diff --git a/tokyotyrant.pc.in b/tokyotyrant.pc.in
index efb3749..efec1f6 100644
--- a/tokyotyrant.pc.in
+++ b/tokyotyrant.pc.in
@@ -10,5 +10,5 @@ datadir=@datadir@
 Name: Tokyo Tyrant
 Description: network interface of Tokyo Cabinet
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ltokyotyrant @LIBS@
+Libs: -L${libdir} -ltokyotyrant
 Cflags: -I${includedir}
