From 9dd1e8d9ad0396a8c9092c2e9f17d498c58e0208 Mon Sep 17 00:00:00 2001
From: elifoster <elifosterwy@gmail.com>
Date: Tue, 5 Dec 2017 14:30:13 -0800
Subject: [PATCH] Fix uninitialized constant DateTime Close #80

---
 lib/safe_yaml/parse/date.rb | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/safe_yaml/parse/date.rb b/lib/safe_yaml/parse/date.rb
index cd3c62a..3a30a8b 100644
--- a/lib/safe_yaml/parse/date.rb
+++ b/lib/safe_yaml/parse/date.rb
@@ -1,3 +1,5 @@
+require 'time'
+
 module SafeYAML
   class Parse
     class Date
