--- elinks-0.10.4.conf	2005-04-08 15:29:49.000000000 -0700
+++ elinks-0.10.4-syscharset.conf	2006-11-04 00:26:26.000000000 -0800
@@ -12,8 +12,6 @@
   # obsolete config.saving_style. You shouldn't touch it.
   set config.saving_style_w = 1
 
-    ## terminal.linux.charset <codepage>
-    set terminal.linux.charset = "us-ascii"
     ## terminal.linux.transparency [0|1]
     set terminal.linux.transparency = 1
     ## terminal.linux.colors [0|1]
@@ -29,8 +27,6 @@
     ## terminal.linux.type <num>
     set terminal.linux.type = 2
 
-    ## terminal.vt100.charset <codepage>
-    set terminal.vt100.charset = "us-ascii"
     ## terminal.vt100.transparency [0|1]
     set terminal.vt100.transparency = 1
     ## terminal.vt100.colors [0|1]
@@ -46,8 +42,6 @@
     ## terminal.vt100.type <num>
     set terminal.vt100.type = 1
 
-    ## terminal.vt110.charset <codepage>
-    set terminal.vt110.charset = "us-ascii"
     ## terminal.vt110.transparency [0|1]
     set terminal.vt110.transparency = 1
     ## terminal.vt110.colors [0|1]
@@ -63,8 +57,6 @@
     ## terminal.vt110.type <num>
     set terminal.vt110.type = 1
 
-    ## terminal.xterm.charset <codepage>
-    set terminal.xterm.charset = "us-ascii"
     ## terminal.xterm.transparency [0|1]
     set terminal.xterm.transparency = 1
     ## terminal.xterm.colors [0|1]
@@ -80,8 +72,6 @@
     ## terminal.xterm.type <num>
     set terminal.xterm.type = 1
 
-    ## terminal.xterm-color.charset <codepage>
-    set terminal.xterm-color.charset = "us-ascii"
     ## terminal.xterm-color.transparency [0|1]
     set terminal.xterm-color.transparency = 1
     ## terminal.xterm-color.colors [0|1]
@@ -96,8 +86,7 @@
     set terminal.xterm-color.m11_hack = 0
     ## terminal.xterm-color.type <num>
     set terminal.xterm-color.type = 1
-    ## terminal.screen.charset <codepage>
-    set terminal.screen.charset = "us-ascii"
+
     ## terminal.screen.transparency [0|1]
     set terminal.screen.transparency = 1
     ## terminal.screen.colors [0|1]
@@ -270,10 +259,6 @@
 
 
 
-    ## document.codepage.assume <codepage>
-    # Default document codepage.
-    set document.codepage.assume = "ISO-8859-1"
-
     ## document.codepage.force_assumed [0|1]
     # Ignore charset info sent by server.
     set document.codepage.force_assumed = 0
@@ -343,10 +328,6 @@
 
 
 
-    ## document.dump.codepage <codepage>
-    # Codepage used in dump output.
-    set document.dump.codepage = "us-ascii"
-
     ## document.dump.width <num>
     # Width of screen in characters when dumping a HTML document.
     set document.dump.width = 80
