Disable check for newer versions on start.

--- a/gui/preferences.ui	2017-01-02 06:05:19.000000000 +0100
+++ b/gui/preferences.ui	2017-01-21 00:38:44.929972188 +0100
@@ -40,13 +40,6 @@
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
-         <widget class="QCheckBox" name="startupCheck">
-          <property name="text">
-           <string>Check for newer version on start.</string>
-          </property>
-         </widget>
-        </item>
-        <item>
          <widget class="QCheckBox" name="reportStatisticsCheck">
           <property name="text">
            <string>Anonymously report usage data.</string>
--- a/gui/preferences.cc	2017-01-02 06:05:19.000000000 +0100
+++ b/gui/preferences.cc	2017-01-21 00:39:20.648418701 +0100
@@ -39,7 +39,6 @@
 {
   ui_.setupUi(this);
 
-  ui_.startupCheck->setChecked(babelData_.startupVersionCheck_);
   ui_.reportStatisticsCheck->setChecked(babelData_.reportStatistics_);
   ui_.ignoreVersionMismatchCheck->setChecked(babelData_.ignoreVersionMismatch_);
   // Because of an unfortunate bug in 1.4.0, we turn this off in 1.4.1.
@@ -82,7 +81,6 @@
     formatList_[i].setHidden(item->checkState() == Qt::Unchecked);
   }
 
-  babelData_.startupVersionCheck_ = ui_.startupCheck->isChecked();
   babelData_.reportStatistics_ = ui_.reportStatisticsCheck->isChecked();
   babelData_.ignoreVersionMismatch_ = ui_.ignoreVersionMismatchCheck->isChecked();
   accept();
--- a/gui/babeldata.h	2017-01-21 00:33:30.858046004 +0100
+++ b/gui/babeldata.h	2017-01-21 00:40:03.888959247 +0100
@@ -63,7 +63,7 @@
     upgradeErrors_(0),
     upgradeOffers_(0),
     runCount_(0),
-    startupVersionCheck_(true),
+    startupVersionCheck_(false),
     reportStatistics_(false),
     allowBetaUpgrades_(false),
     ignoreVersionMismatch_(false),
--- a/gui/mainwinui.ui	2017-01-02 06:05:19.000000000 +0100
+++ b/gui/mainwinui.ui	2017-01-21 00:40:40.453416335 +0100
@@ -637,7 +637,6 @@
     <addaction name="actionHelp"/>
     <addaction name="separator"/>
     <addaction name="actionAbout"/>
-    <addaction name="actionUpgradeCheck"/>
     <addaction name="separator"/>
     <addaction name="actionVisit_Website"/>
     <addaction name="actionMake_a_Donation"/>
@@ -666,11 +665,6 @@
     <string>Preferences...</string>
    </property>
   </action>
-  <action name="actionUpgradeCheck">
-   <property name="text">
-    <string>Check for Upgrade</string>
-   </property>
-  </action>
   <action name="actionVisit_Website">
    <property name="text">
     <string>Visit Website...</string>
--- a/gui/mainwindow.cc	2017-01-02 06:05:19.000000000 +0100
+++ b/gui/mainwindow.cc	2017-01-21 00:42:02.939447485 +0100
@@ -147,7 +147,6 @@
   connect(ui_.actionAbout, SIGNAL(triggered()), this, SLOT(aboutActionX()));
   connect(ui_.actionVisit_Website, SIGNAL(triggered()), this, SLOT(visitWebsiteActionX()));
   connect(ui_.actionMake_a_Donation, SIGNAL(triggered()), this, SLOT(donateActionX()));
-  connect(ui_.actionUpgradeCheck, SIGNAL(triggered()), this, SLOT(upgradeCheckActionX()));
   connect(ui_.actionPreferences, SIGNAL(triggered()), this, SLOT(preferencesActionX()));
 
   connect(ui_.inputFormatCombo,  SIGNAL(currentIndexChanged(int)),
@@ -208,17 +208,6 @@
   restoreSettings();
 
   upgrade = new UpgradeCheck(parent, formatList_, babelData_);
-  if (babelData_.startupVersionCheck_) {
-    upgrade->checkForUpgrade(babelVersion_, babelData_.upgradeCheckTime_,
-                             allowBetaUpgrades());
-  }
-
-  if (!babelData_.ignoreVersionMismatch_ && babelVersion_ != VERSION) {
-    VersionMismatch vm(0, babelVersion_, QString(VERSION));
-
-    vm.exec();
-    babelData_.ignoreVersionMismatch_ = vm.neverAgain();
-  }
 }
 
 //------------------------------------------------------------------------
@@ -1021,10 +1010,6 @@
 //------------------------------------------------------------------------
 void MainWindow::closeActionX()
 {
-  QDateTime wt= upgrade->getUpgradeWarningTime();
-  if (wt.isValid()) {
-    babelData_.upgradeCheckTime_ = wt;
-  }
   babelData_.runCount_++;
 
   QDateTime now = QDateTime::currentDateTime();
