diff -urN gpsbabel-1.4.2.old/configure.in gpsbabel-1.4.2/configure.in
--- gpsbabel-1.4.2.old/configure.in	2011-06-10 10:30:54.888410566 +0200
+++ gpsbabel-1.4.2/configure.in	2011-06-10 10:42:08.800985499 +0200
@@ -421,6 +421,6 @@
 else
        AC_MSG_RESULT(no)
 fi
-AC_CONFIG_FILES([Makefile gbversion.h gui/setup.iss gui/makelinuxdist.sh xmldoc/makedoc tools/mkcapabilities win32/gpsbabel.rc jeeps/Makefile shapelib/Makefile zlib/empty])
+AC_CONFIG_FILES([Makefile gbversion.h gui/setup.iss gui/makelinuxdist.sh xmldoc/makedoc tools/mkcapabilities win32/gpsbabel.rc jeeps/Makefile zlib/empty])
 AC_OUTPUT
 
diff -urN gpsbabel-1.4.2.old/Makefile.in gpsbabel-1.4.2/Makefile.in
--- gpsbabel-1.4.2.old/Makefile.in	2011-06-10 10:30:54.899410428 +0200
+++ gpsbabel-1.4.2/Makefile.in	2011-06-10 10:43:13.973170735 +0200
@@ -87,7 +87,7 @@
 
 @PALM_DB_CMT@PALM_DB=pdbfile.o
 
-SHAPE=shapelib/shpopen.o shapelib/dbfopen.o
+SHAPE=-lshp
 
 ZLIB=zlib/adler32.o zlib/compress.o zlib/crc32.o zlib/deflate.o zlib/inffast.o \
 	zlib/inflate.o zlib/infback.o zlib/inftrees.o zlib/trees.o \
@@ -98,7 +98,7 @@
           uuid.o formspec.o xmltag.o cet.o cet_util.o fatal.o rgbcolors.o \
 	  inifile.o garmin_fs.o gbsleep.o units.o @GBSER@ gbser.o \
 	  gbfile.o parse.o avltree.o session.o \
-	$(PALM_DB) $(GARMIN) $(JEEPS) $(SHAPE) @ZLIB@ $(FMTS) $(FILTERS)
+	$(PALM_DB) $(GARMIN) $(JEEPS) @ZLIB@ $(FMTS) $(FILTERS)
 OBJS = main.o globals.o $(LIBOBJS) @FILEINFO@
 
 .c.o:
@@ -113,10 +113,10 @@
 all: gpsbabel$(EXEEXT)
 
 gpsbabel$(EXEEXT): configure Makefile $(OBJS) @GPSBABEL_DEBUG@ 
-	$(CC)  $(CFLAGS) $(LDFLAGS) $(OBJS) @LIBS@ @EXPAT_LIB@ @USB_LIBS@ $(OUTPUT_SWITCH)$@
+	$(CC)  $(CFLAGS) $(LDFLAGS) $(OBJS) $(SHAPE) @LIBS@ @EXPAT_LIB@ @USB_LIBS@ $(OUTPUT_SWITCH)$@
 
 gpsbabel-debug: $(OBJS)
-	$(CC)  $(CFLAGS) $(LDFLAGS) $(OBJS) @LIBS@ @EFENCE_LIB@ @EXPAT_LIB@ @USB_LIBS@ $(OUTPUT_SWITCH)$@
+	$(CC)  $(CFLAGS) $(LDFLAGS) $(OBJS) $(SHAPE) @LIBS@ @EFENCE_LIB@ @EXPAT_LIB@ @USB_LIBS@ $(OUTPUT_SWITCH)$@
 
 Makefile gbversion.h: Makefile.in config.status xmldoc/makedoc.in \
 	  gbversion.h.in gui/setup.iss.in
@@ -818,8 +818,7 @@
 session.o: session.c defs.h config.h queue.h gbtypes.h zlib/zlib.h \
   zlib/zconf.h gbfile.h cet.h cet_util.h inifile.h session.h
 shape.o: shape.c defs.h config.h queue.h gbtypes.h zlib/zlib.h \
-  zlib/zconf.h gbfile.h cet.h cet_util.h inifile.h session.h \
-  shapelib/shapefil.h
+  zlib/zconf.h gbfile.h cet.h cet_util.h inifile.h session.h
 skyforce.o: skyforce.c defs.h config.h queue.h gbtypes.h zlib/zlib.h \
   zlib/zconf.h gbfile.h cet.h cet_util.h inifile.h session.h strptime.h
 skytraq.o: skytraq.c defs.h config.h queue.h gbtypes.h zlib/zlib.h \
@@ -1092,8 +1091,6 @@
   jeeps/gpsread.h jeeps/gpsutil.h jeeps/gpsapp.h jeeps/gpsprot.h \
   jeeps/gpscom.h jeeps/gpsfmt.h jeeps/gpsmath.h jeeps/gpsmem.h \
   jeeps/gpsrqst.h jeeps/gpsinput.h jeeps/gpsproj.h
-shapelib/dbfopen.o: shapelib/dbfopen.c shapelib/shapefil.h config.h
-shapelib/shpopen.o: shapelib/shpopen.c shapelib/shapefil.h config.h
 zlib/adler32.o: zlib/adler32.c zlib/zlib.h zlib/zconf.h
 zlib/compress.o: zlib/compress.c zlib/zlib.h zlib/zconf.h
 zlib/crc32.o: zlib/crc32.c zlib/zutil.h zlib/zlib.h zlib/zconf.h \
diff -urN gpsbabel-1.4.2.old/shape.c gpsbabel-1.4.2/shape.c
--- gpsbabel-1.4.2.old/shape.c	2011-06-10 10:30:54.932410016 +0200
+++ gpsbabel-1.4.2/shape.c	2011-06-10 10:42:08.802985475 +0200
@@ -20,7 +20,7 @@
 
  */
 #include "defs.h"
-#include "shapelib/shapefil.h"
+#include <libshp/shapefil.h>
 
 #if SHAPELIB_ENABLED
 static SHPHandle ihandle;
