--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -205,7 +205,7 @@
     if (BUNDLER_PATH)
         message(STATUS "Bundler found, installing dependencies for Ruby tests...")
         execute_process(COMMAND ${BUNDLER_PATH} install --retry 2 --path=vendor/bundle WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/lib")
-        add_test(NAME "libfacter\\ specs" COMMAND ${BUNDLER_PATH} exec rspec WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/lib")
+        add_test(NAME "libfacter\\ specs" COMMAND ${BUNDLER_PATH} exec rspec-2 WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/lib")
     endif()
 endif()
