--- configure.pl.orig	2010-07-31 20:52:58.000000000 +0300
+++ configure.pl	2010-09-03 22:09:46.437788435 +0300
@@ -158,7 +158,7 @@
 # config file to get all the defaults.
 #
 my $ConfigPath = "";
-my $ConfigFileOK = 1;
+my $ConfigFileOK = 0;
 while ( 1 ) {
     if ( $ConfigFileOK && -f "/etc/BackupPC/config.pl" ) {
         $ConfigPath = "/etc/BackupPC/config.pl";
@@ -213,7 +213,8 @@
         $bpc->{LogDir} = $Conf{LogDir}  = "$Conf{TopDir}/log"
                     if ( $Conf{LogDir} eq '' );
     }
-    $bpc->{ConfDir} = $Conf{ConfDir} = $confDir;
+    # Disable this as it's not really neccessary for this ebuild
+    # $bpc->{ConfDir} = $Conf{ConfDir} = $confDir;
     my $err = $bpc->ServerConnect($Conf{ServerHost}, $Conf{ServerPort}, 1);
     if ( $err eq "" ) {
         print <<EOF;
@@ -726,7 +727,7 @@
     if ( $Conf{CgiDir} =~ m{cgi-bin(/.*)} ) {
 	$Conf{CgiURL} = "'http://$Conf{ServerHost}/cgi-bin$1/BackupPC_Admin'";
     } else {
-	$Conf{CgiURL} = "'http://$Conf{ServerHost}/cgi-bin/BackupPC_Admin'";
+	$Conf{CgiURL} = "'http://$Conf{ServerHost}/BackupPC_Admin'";
     }
 }
 
