diff -durr ViennaRNA-1.8.3.orig/Cluster/AS_main.c ViennaRNA-1.8.3/Cluster/AS_main.c
--- ViennaRNA-1.8.3.orig/Cluster/AS_main.c	2009-09-10 11:56:01.636674988 +0000
+++ ViennaRNA-1.8.3/Cluster/AS_main.c	2009-09-10 11:58:32.915952366 +0000
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include "distance_matrix.h"
 #include "statgeom.h"
 #include "split.h"
diff -durr ViennaRNA-1.8.3.orig/Cluster/statgeom.c ViennaRNA-1.8.3/Cluster/statgeom.c
--- ViennaRNA-1.8.3.orig/Cluster/statgeom.c	2009-09-10 11:56:01.636674988 +0000
+++ ViennaRNA-1.8.3/Cluster/statgeom.c	2009-09-10 11:58:53.192452141 +0000
@@ -7,6 +7,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <strings.h>
+#include <stdlib.h>
 #include <ctype.h>
 #include "utils.h"
 #include "PS3D.h"
diff -durr ViennaRNA-1.8.3.orig/Cluster/treeplot.c ViennaRNA-1.8.3/Cluster/treeplot.c
--- ViennaRNA-1.8.3.orig/Cluster/treeplot.c	2009-09-10 11:56:01.636674988 +0000
+++ ViennaRNA-1.8.3/Cluster/treeplot.c	2009-09-10 11:59:22.457509165 +0000
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include <math.h>
 #include "distance_matrix.h"
 #include "utils.h"
diff -durr ViennaRNA-1.8.3.orig/RNAforester/src/main.cpp ViennaRNA-1.8.3/RNAforester/src/main.cpp
--- ViennaRNA-1.8.3.orig/RNAforester/src/main.cpp	2009-09-10 11:56:01.624743558 +0000
+++ ViennaRNA-1.8.3/RNAforester/src/main.cpp	2009-09-10 12:00:37.693252890 +0000
@@ -19,6 +19,8 @@
 #include <list>
 #include <sstream>
 #include <string>
+#include <cstring>
+#include <climits>
 #include <map>
 
 //#include <sys/timeb.h>
diff -durr ViennaRNA-1.8.3.orig/RNAforester/src/ppforestbase.h ViennaRNA-1.8.3/RNAforester/src/ppforestbase.h
--- ViennaRNA-1.8.3.orig/RNAforester/src/ppforestbase.h	2009-09-10 11:56:01.624743558 +0000
+++ ViennaRNA-1.8.3/RNAforester/src/ppforestbase.h	2009-09-10 12:01:41.631932654 +0000
@@ -16,6 +16,7 @@
 
 #include <algorithm>
 #include <cassert>
+#include <cstring>
 
 #include "misc.h"
 #include "types.h"
diff -durr ViennaRNA-1.8.3.orig/RNAforester/src/rna_algebra.h ViennaRNA-1.8.3/RNAforester/src/rna_algebra.h
--- ViennaRNA-1.8.3.orig/RNAforester/src/rna_algebra.h	2009-09-10 11:56:01.620703294 +0000
+++ ViennaRNA-1.8.3/RNAforester/src/rna_algebra.h	2009-09-10 12:02:29.687952726 +0000
@@ -4,6 +4,7 @@
 #include <assert.h>
 #include <algorithm>
 #include <climits>
+#include <cstring>
 
 #include "algebra.h"
 #include "debug.h"
diff -durr ViennaRNA-1.8.3.orig/Readseq/ureadseq.c ViennaRNA-1.8.3/Readseq/ureadseq.c
--- ViennaRNA-1.8.3.orig/Readseq/ureadseq.c	2009-09-10 11:56:01.632743452 +0000
+++ ViennaRNA-1.8.3/Readseq/ureadseq.c	2009-09-10 12:03:15.343952776 +0000
@@ -20,6 +20,7 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <string.h>
+#include <stdlib.h>
 
 #define UREADSEQ_G
 #include "ureadseq.h"
