diff -Naur ViennaRNA-1.7.2/Perl/Makefile.am ViennaRNA-1.7.2.new/Perl/Makefile.am
--- ViennaRNA-1.7.2/Perl/Makefile.am	2007-12-19 05:33:14.000000000 -0500
+++ ViennaRNA-1.7.2.new/Perl/Makefile.am	2008-10-06 14:20:54.000000000 -0400
@@ -4,7 +4,7 @@
 	     RNA.i tmaps.i
 
 Makefile.perl: Makefile.PL Makefile
-	@PerlCmd@ Makefile.PL
+	@PerlCmd@ Makefile.PL LDDLFLAGS="${LDFLAGS} -shared"
 
 # the perl command repairs some errors in the SWIG CVS version
 RNA_wrap.c: RNA.i tmaps.i
diff -Naur ViennaRNA-1.7.2/Readseq/Makefile ViennaRNA-1.7.2.new/Readseq/Makefile
--- ViennaRNA-1.7.2/Readseq/Makefile	2008-01-23 05:32:00.000000000 -0500
+++ ViennaRNA-1.7.2.new/Readseq/Makefile	2008-10-06 14:21:13.000000000 -0400
@@ -40,7 +40,7 @@
 
 build: $(SOURCES)
 	@echo "Compiling readseq..."
-	$(CC) $(CFLAGS) -o readseq readseq.c ureadseq.c
+	$(CC) $(LDFLAGS) $(CFLAGS) -o readseq readseq.c ureadseq.c
 
 # if using NCBI, uncomment these lines in place of build: above
 #build: $(SOURCES)
