--- ser2net-2.7.orig/devcfg.c	2009-10-23 14:23:25.000000000 +0200
+++ ser2net-2.7/devcfg.c	2012-07-17 21:03:49.708603108 +0200
@@ -113,6 +113,9 @@
 	} else if (strcmp(pos, "115200") == 0) {
 	    cfsetospeed(termctl, B115200);
 	    cfsetispeed(termctl, B115200);
+	} else if (strcmp(pos, "230400") == 0) {
+	    cfsetospeed(termctl, B230400);
+	    cfsetispeed(termctl, B230400);
 	} else if (strcmp(pos, "1STOPBIT") == 0) {
 	    termctl->c_cflag &= ~(CSTOPB);
 	} else if (strcmp(pos, "2STOPBITS") == 0) {
@@ -190,6 +193,7 @@
     case B38400: str = "38400"; break;
     case B57600: str = "57600"; break;
     case B115200: str = "115200"; break;
+    case B230400: str = "230400"; break;
     default: str = "unknown speed";
     }
     return str;
