--- irda-utils-0.9.18/smcinit/smcinit.c.orig	2006-07-11 08:16:06.000000000 +0200
+++ irda-utils-0.9.18/smcinit/smcinit.c	2008-03-22 05:49:38.000000000 +0100
@@ -292,9 +292,9 @@
 	}
 
 	twobyte = pci_read_word(dev, DID);
-	if ((twobyte != 0x24cc) & (twobyte != 0x248c)) {
+	if ((twobyte != 0x244c) & (twobyte != 0x24cc) & (twobyte != 0x248c)) {
 		fprintf(stderr,
-			"%s IO hub device %x not 82801CAM (0x248c or 0x24cc)\n",
+			"%s IO hub device %x not 82801BAM/82801CAM (0x24cc or 0x244c or 0x248c)\n",
 			PROGNAME, twobyte);
 		return 1;
 	}
