commit 156f8deaae2a50d8b9614b99488f34b284382eaa
Author: Drup <drupyog@zoho.com>
Date:   Thu Dec 11 16:52:33 2014 +0100

    Remove dependency to Lwt_util.

diff --git a/src/server/ocsigen_server.ml b/src/server/ocsigen_server.ml
index c5563a6..e548f5e 100644
--- a/src/server/ocsigen_server.ml
+++ b/src/server/ocsigen_server.ml
@@ -1040,7 +1040,7 @@ let rec wait_connection use_ssl port socket =
          >>= decr_connected
        in
 
-       Lwt_util.iter handle_one l >>= fun () ->
+       Lwt_list.iter_p handle_one l >>= fun () ->
        match e with
        | Some e -> handle_exn e
        | None -> Lwt.return ())
