commit 8edb1db4016fc90a8406e4e0c5d1a04378f72e5f
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Tue Dec 1 23:47:36 2015 +0100

    Initialise KCrash for DrKonqui integration.

diff --git a/src/main.cpp b/src/main.cpp
index 2b5d8cc..665a92d 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -22,6 +22,7 @@
 #include <QCommandLineParser>
 
 #include <KAboutData>
+#include <KCrash>
 #include <KLocalizedString>
 
 
@@ -44,6 +45,9 @@ int main (int argc, char *argv[])
                                )
                         );
 
+    // TODO: for >=KF5.15 use KCrash::initialize();
+    KCrash::setCrashHandler(KCrash::defaultCrashHandler);
+
     aboutData.addAuthor(i18n("Stefan Böhmann"), i18n("Current maintainer"), QStringLiteral("kde@hilefoks.org"), QStringLiteral("http://www.hilefoks.org"), QStringLiteral("hilefoks"));
     aboutData.addAuthor(i18n("Matthias Hoelzer-Kluepfel"), QString(), QStringLiteral("hoelzer@kde.org"));
     aboutData.addAuthor(i18n("Martin Willers"), QString(), QStringLiteral("willers@xm-arts.de"));
