From 231123e1b275f88000e336a2d80d3d96c470ac5e Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Fri, 13 Oct 2017 15:10:08 +0200
Subject: [PATCH] We need LDLIBS instead of LDFLAGS

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 039795e..a6bff4b 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ CFLAGS ?= -O2 -ggdb
 CFLAGS += -Wall
 CFLAGS += -fPIC
 CPPFLAGS += -D_GNU_SOURCE
-LDFLAGS += -lm
+LDLIBS += -lm
 
 # -- Actions section --
 
-- 
2.14.2

