From b577afbc012078da92c04b0e52b38c09a2f6fb51 Mon Sep 17 00:00:00 2001
From: Tom Stellard <tstellar@redhat.com>
Date: Fri, 19 Feb 2021 19:49:41 -0800
Subject: [PATCH 1/2] [PATCH][flang] Disable use of sphinx_markdown_tables

---
 flang/docs/conf.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docs/conf.py b/docs/conf.py
index 7ad2915..da72db5 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -54,7 +54,7 @@ else:
   else:
     source_parsers = {'.md': CustomCommonMarkParser}
   source_suffix['.md'] = 'markdown'
-  extensions.append('sphinx_markdown_tables')
+  #extensions.append('sphinx_markdown_tables')
 
   # Setup AutoStructify for inline .rst toctrees in index.md
   from recommonmark.transform import AutoStructify
-- 
1.8.3.1

