diff -ru deutex-4.4.0.orig/Makefile deutex-4.4.0/Makefile
--- deutex-4.4.0.orig/Makefile	1999-12-24 07:50:16.000000000 -0500
+++ deutex-4.4.0/Makefile	2010-03-05 14:53:23.763604022 -0500
@@ -16,11 +16,6 @@
 
 PREFIX=/usr/local
 
-# Compiled by users
-CFLAGS   = -O2
-CC       = cc
-LDFLAGS  =
-
 # Compiled by developers
 DCFLAGS  = -g -Wall -Wpointer-arith -Wstrict-prototypes
 DCC      = gcc
@@ -182,11 +177,11 @@
 	(sleep 1; mkdir -p tmp; touch tmp/_deusf) &
 
 tmp/_deutex:
-	-mkdir tmp
+	@mkdir tmp 2> /dev/null || true
 	touch $@
 
 tmp/_deusf:
-	-mkdir tmp
+	@mkdir tmp 2> /dev/null || true
 	touch $@
 
 install:
