--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
+cmake_minimum_required (VERSION 2.6...4.1)
+
 project (reiserfs-defrag)
 
-cmake_minimum_required (VERSION 2.6)
-
 add_executable (reiserfs-defrag
 	defrag.cpp
--- a/misc/CMakeLists.txt
+++ b/misc/CMakeLists.txt
@@ -1,6 +1,6 @@
+cmake_minimum_required (VERSION 2.6...4.1)
+
 project (reiserfs-toys)
 
-cmake_minimum_required (VERSION 2.6)
-
 add_library (mrfsu STATIC
 	../reiserfs.cpp
