diff -NrU5 postal-0.69.orig/bhm.cpp postal-0.69/bhm.cpp
--- postal-0.69.orig/bhm.cpp	2008-04-23 01:47:25.000000000 +0200
+++ postal-0.69/bhm.cpp	2008-04-23 02:10:17.000000000 +0200
@@ -67,11 +67,11 @@
   Logit *debug;
 } thread_data;
 
 enum { eFree = 0, eUsed, eFinished };
 
-int check_sender(CPCCHAR addr)
+int check_sender(CPCCHAR)
 {
   return 0;
 }
 
 int readCommand(base_tcp &t, char *buf, int bufSize, bool stripCR, int timeout = 60);

