Index: trunk/gdal/frmts/nitf/nitffile.c
===================================================================
--- a/frmts/nitf/nitffile.c	(revision 38085)
+++ b/frmts/nitf/nitffile.c	(revision 38234)
@@ -350,5 +350,6 @@
     if (nOffset != -1)
         nOffset = NITFCollectSegmentInfo( psFile, nHeaderLen, nOffset, "RE", 4, 7, &nNextData);
-    else
+
+    if( nOffset < 0 )
     {
         NITFClose(psFile);
