https://github.com/apitrace/apitrace/pull/941.patch
diff --git a/frametrim/ft_dependecyobject.hpp b/frametrim/ft_dependecyobject.hpp
index 2cd0b76879..0f17b601a3 100644
--- a/frametrim/ft_dependecyobject.hpp
+++ b/frametrim/ft_dependecyobject.hpp
@@ -28,6 +28,7 @@
 #pragma once
 
 #include "ft_tracecall.hpp"
+#include <stdint.h>
 
 namespace frametrim {
 
diff --git a/retrace/metric_writer.cpp b/retrace/metric_writer.cpp
index e923ed1d3b..4e4ce74833 100644
--- a/retrace/metric_writer.cpp
+++ b/retrace/metric_writer.cpp
@@ -24,6 +24,7 @@
  **************************************************************************/
 
 #include <iostream>
+#include <stdint.h>
 
 #include "metric_writer.hpp"
 
