diff --git a/CMakeLists.txt b/CMakeLists.txt
index 34123b3..9d4aefb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -182,7 +182,7 @@ if (ENABLE_GUI)
         cmake_policy (SET CMP0020 NEW)
     endif()
     if (ENABLE_QT6)
-        find_package (Qt6 COMPONENTS Widgets Network ${REQUIRE_GUI})
+        find_package (Qt6 COMPONENTS Widgets ${REQUIRE_GUI})
     else ()
         find_package (Qt5 5.15 COMPONENTS Widgets Network ${REQUIRE_GUI})
     endif ()
diff --git a/gui/CMakeLists.txt b/gui/CMakeLists.txt
index 99ae918..afb740c 100644
--- a/gui/CMakeLists.txt
+++ b/gui/CMakeLists.txt
@@ -99,7 +99,6 @@ add_executable (qapitrace ${qapitrace_SRCS} ${qapitrace_UIS_H})
 
 target_link_libraries (qapitrace
     Qt::Widgets
-    Qt::Network
     qubjson
     image
     common
