--- a/meson.build
+++ b/meson.build
@@ -81,21 +81,6 @@ if cc.has_argument('-mbranch-protection=standard')
 	add_global_arguments([ '-mbranch-protection=standard' ], language: 'c')
 endif
 
-# MINGW64 cannot handle lto properly at the moment
-if (cc.has_argument('-flto') and
-    host_machine.system() != 'windows' and
-    get_option('efi').disabled())
-	add_global_arguments([ '-flto=auto' ], language: 'c')
-	add_project_link_arguments('-flto=auto', language : 'c')
-endif
-
-if (cc.has_argument('-ffat-lto-objects') and
-    host_machine.system() != 'windows' and
-    host_machine.system() != 'darwin' and
-    get_option('efi').disabled())
-	add_global_arguments([ '-ffat-lto-objects' ], language: 'c')
-endif
-
 if get_option('optimization') == '0'
 	add_global_arguments([ '-DDEBUG' ], language: 'c')
 elif get_option('efi').disabled()
