--- umockdev-0.9.4/src/umockdev-record.c
+++ umockdev-0.9.4/src/umockdev-record.c
@@ -30,6 +30,7 @@
 #include <stdio.h>
 #include <stdarg.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <errno.h>
 #include <sys/types.h>
 #include <glib/gstdio.h>
--- umockdev-0.9.4/tests/test-umockdev-vala.c
+++ umockdev-0.9.4/tests/test-umockdev-vala.c
@@ -36,6 +36,7 @@
 #include <unistd.h>
 #include <glib/gstdio.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <gobject/gvaluecollector.h>
 
 #define _g_error_free0(var) ((var == NULL) ? NULL : (var = (g_error_free (var), NULL)))
