--- lynx2-8-6/WWW/Library/Implementation/www_tcp.h.old	2009-03-18 06:45:10.000000000 +0000
+++ lynx2-8-6/WWW/Library/Implementation/www_tcp.h	2009-03-18 06:45:22.000000000 +0000
@@ -94,7 +94,7 @@
 # endif
 #endif /* HAVE_DIRENT_H */
 
-#if !(defined(DOSPATH) || defined(__EMX__) || defined(__CYGWIN__))
+#if !(defined(DOSPATH) || defined(__EMX__) || defined(__CYGWIN__) || defined(__MINT__))
 #define STRUCT_DIRENT__D_INO 1
 #endif
 
