--- a/src/tse3/file/XML.h
+++ b/src/tse3/file/XML.h
@@ -154,6 +154,7 @@
                 void element(const std::string &name, const char        *value);
                 void element(const std::string &name, int                value);
                 void element(const std::string &name, unsigned int       value);
+                void element(const std::string &name, unsigned long      value);
                 void element(const std::string &name, bool               value);
 
                 void comment(const std::string &comment);
--- a/src/tse3/file/XML.cpp
+++ b/src/tse3/file/XML.cpp
@@ -116,6 +116,12 @@
     out << "<" << name << " value=\"" << value << "\"/>\n";
 }
 
+void TSE3::File::XmlFileWriter::element(const std::string &name, unsigned long value)
+{
+    indent(out);
+    out << "<" << name << " value=\"" << value << "\"/>\n";
+}
+
 
 void TSE3::File::XmlFileWriter::element(const std::string &name, bool value)
 {
