 setup.py | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/setup.py b/setup.py
index 0177c44..58d13b0 100644
--- a/setup.py
+++ b/setup.py
@@ -300,14 +300,14 @@ class MyBuild(build.build):
     """
 
     def run(self):
-        build_tvtk_classes_zip()
+#        build_tvtk_classes_zip()
         build.build.run(self)
-        self.run_command('gen_docs')
-        try:
-            self.run_command('build_docs')
-        except:
-            log.warn("Couldn't build documentation:\n%s" %
-                     traceback.format_exception(*sys.exc_info()))
+#        self.run_command('gen_docs')
+#        try:
+#            self.run_command('build_docs')
+#        except:
+#            log.warn("Couldn't build documentation:\n%s" %
+#                     traceback.format_exception(*sys.exc_info()))
 
 
 
@@ -494,6 +494,6 @@ numpy.distutils.core.setup(
     license = "BSD",
     long_description = '\n'.join(DOCLINES[3:]),
     platforms = ["Windows", "Linux", "Mac OS-X", "Unix", "Solaris"],
-    zip_safe = False,
+    zip_safe = True,
     **config
 )
