--- Mouse-0.64/inc/Module/Install/XSUtil.pm
+++ Mouse-0.64/inc/Module/Install/XSUtil.pm
@@ -116,24 +116,6 @@
 
     my $filename = 'ppport.h';
 
-    $dppp_version ||= 3.19; # the more, the better
-    $self->configure_requires('Devel::PPPort' => $dppp_version);
-    $self->build_requires('Devel::PPPort' => $dppp_version);
-
-    print "Writing $filename\n";
-
-    my $e = do{
-        local $@;
-        eval qq{
-            use Devel::PPPort;
-            Devel::PPPort::WriteFile(q{$filename});
-        };
-        $@;
-    };
-    if($e){
-         print "Cannot create $filename because: $@\n";
-    }
-
     if(-e $filename){
         $self->clean_files($filename);
         $self->cc_define('-DUSE_PPPORT');
