diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d94776..b04fa78 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -293,10 +293,12 @@ if(NOT MSVC)
 	target_link_libraries(flint PUBLIC m)
 endif()
 
+include(GNUInstallDirs)
+
 install(TARGETS flint
-            RUNTIME DESTINATION bin
-            ARCHIVE DESTINATION lib
-            LIBRARY DESTINATION lib
+            RUNTIME DESTINATION "${CMAKE_INSTALL_FULL_BINDIR}"
+            ARCHIVE DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}"
+            LIBRARY DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}"
         )
 
 install(FILES ${HEADERS} DESTINATION include/flint)
