Index: Tensile-rocm-4.3.0/Tensile/Tensile.py
===================================================================
--- Tensile-rocm-4.3.0.orig/Tensile/Tensile.py
+++ Tensile-rocm-4.3.0/Tensile/Tensile.py
@@ -97,7 +97,6 @@ def addCommonArguments(argParser):
     """
 
     (key, value) = par.split("=")
-    value = eval(value)
     return (key, value)
 
   argParser.add_argument("-d", "--device", dest="device", type=int, \
