--- a/src/main.c
+++ b/src/main.c
@@ -44,7 +44,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifndef __UCLIBC__
 #include <mcheck.h>
+#endif
 #include <signal.h> /* sig_atomic_t */
 
 #include <dirent.h> /* PATH_MAX */
--- a/src/tasks/cdw_create_image.c
+++ b/src/tasks/cdw_create_image.c
@@ -21,6 +21,7 @@
 
 #include <inttypes.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 #include "cdw_file_manager.h"
 #include "gettext.h"
--- a/src/utilities/cdw_utils.c
+++ b/src/utilities/cdw_utils.c
@@ -28,6 +28,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <unistd.h>
 
 #include "cdw_logging.h"
 #include "gettext.h"
