--- crda-3.18/reglib.c	2018-10-26 12:39:19.128083735 +1100
+++ crda-3.18.a/reglib.c	2018-10-26 12:42:39.737916626 +1100
@@ -111,7 +111,7 @@ int reglib_verify_db_signature(uint8_t *
 			goto out;
 		}
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)
 		rsa->e = rsa_e;
 		rsa->n = rsa_n;
 #else
