diff -dupr yadex-1.7.901-patched/src/pic2img.cc yadex-1.7.901/src/pic2img.cc
--- yadex-1.7.901-patched/src/pic2img.cc	2006-11-06 17:35:03.000000000 +0100
+++ yadex-1.7.901/src/pic2img.cc	2015-11-24 09:54:25.578048649 +0100
@@ -92,9 +92,9 @@ uint8_t      *buf;	/* This variable is s
 		   should be pasted. It can be off the image buffer! */
 
 // Locate the lump where the picture is
+   MasterDirectory dirbuf;
 if (picloc.wad != 0)
    {
-   MasterDirectory dirbuf;
    dirbuf.wadfile   = picloc.wad;
    dirbuf.dir.start = picloc.ofs;
    dirbuf.dir.size  = picloc.len;
