diff -Naur yadex-1.7.0/configure yadex-1.7.0.new/configure
--- yadex-1.7.0/configure	2006-03-16 18:49:20.000000000 -0800
+++ yadex-1.7.0.new/configure	2006-03-16 16:36:21.000000000 -0800
@@ -408,16 +408,16 @@
   ETCDIR=/etc/$APPNAME/%v
   ETCDIRNV=/etc/$APPNAME
   MANDIR=/usr/share/man
-  SHAREDIR=/usr/share/games/$APPNAME/%v
-  SHAREDIRNV=/usr/share/games/$APPNAME
+  SHAREDIR=/usr/share/$APPNAME/%v
+  SHAREDIRNV=/usr/share/$APPNAME
 elif expr "$PREFIX" : '//*usr//*local/*$' >/dev/null
 then
   BINDIR=/usr/local/bin		# FHS-ly correct is /usr/local/games
   ETCDIR=/etc/$APPNAME/%v
   ETCDIRNV=/etc/$APPNAME
   MANDIR=/usr/local/man
-  SHAREDIR=/usr/local/share/games/$APPNAME/%v
-  SHAREDIRNV=/usr/local/share/games/$APPNAME
+  SHAREDIR=/usr/local/share/$APPNAME/%v
+  SHAREDIRNV=/usr/local/share/$APPNAME
 elif expr "$PREFIX" : '//*opt/*$' >/dev/null
 then
   echo '/opt ? Surely you mean /opt/something, Mr. Feynman !' 1>&2
diff -Naur yadex-1.7.0/yadex.cfg yadex-1.7.0.new/yadex.cfg
--- yadex-1.7.0/yadex.cfg	2006-03-16 18:49:20.000000000 -0800
+++ yadex-1.7.0.new/yadex.cfg	2006-03-16 18:49:43.000000000 -0800
@@ -20,16 +20,17 @@
 #  9 doompr   Doom press release pre-beta (/pub/idgames/historic/doomprbt.zip)
 # 10 strife10 Strife 1.0 (demo or commercial)
 
-  iwad1  = /usr/local/share/games/doom/doom.wad
-  iwad2  = /usr/local/share/games/doom2/doom2.wad
-  iwad3  = /usr/local/share/games/heretic/heretic.wad
-  iwad4  = /usr/local/share/games/hexen/hexen.wad
-  iwad5  = /usr/local/share/games/strife/strife1.wad
-  iwad6  = /usr/local/share/games/doom02/doom.wad
-  iwad7  = /usr/local/share/games/doom04/doom.wad
-  iwad8  = /usr/local/share/games/doom05/doom.wad
-  iwad9  = /usr/local/share/games/doompr/doompres.wad
-  iwad10 = /usr/local/share/games/strife10/strife1.wad
+  iwad1  = /usr/share/doom/doom.wad
+#  iwad2  = /usr/share/doom/doom2.wad
+  iwad2  = /usr/share/doom/freedoom.wad
+  iwad3  = /usr/share/doom/heretic.wad
+  iwad4  = /usr/share/doom/hexen.wad
+  iwad5  = /usr/share/doom/strife1.wad
+  iwad6  = /usr/share/doom/doom.wad
+  iwad7  = /usr/share/doom/doom.wad
+  iwad8  = /usr/share/doom/doom.wad
+  iwad9  = /usr/share/doom/doompres.wad
+  iwad10 = /usr/share/doom/strife1.wad
 
 # Name of the patch wads you want to load. None by default.
 
