--- yadex-1.7.0/src/wadlist.cc.old	2003-01-29 14:45:19.000000000 -0700
+++ yadex-1.7.0/src/wadlist.cc	2003-01-29 14:45:56.000000000 -0700
@@ -175,7 +175,7 @@
   priv->iter = priv->list.erase (i);
   if (priv->iter == priv->list.begin ())
   {
-    priv->iter = 0;			// Catch bugs
+    priv->iter = (std::_List_iterator<boost::shared_ptr<Wad_file> >)NULL;			// Catch bugs
     priv->rewound = true;
   }
 }
