diff -Naur orion-1.6.1.orig/distfiles/Orion.desktop orion-1.6.1/distfiles/Orion.desktop
--- orion-1.6.1.orig/distfiles/Orion.desktop	2017-07-06 22:26:29.000000000 +0200
+++ orion-1.6.1/distfiles/Orion.desktop	2017-09-19 13:36:49.754743707 +0200
@@ -2,9 +2,9 @@
 Version=1.0
 Type=Application
 Name=Orion
-Icon=/usr/local/share/icons/orion.svg
+Icon=orion
 Exec=/usr/bin/orion
 Comment=Seek and watch streams on Twitch
-Categories=Games
+Categories=Game
 Terminal=false
 StartupWMClass=orion
