# backported from xastir-2.1.4 (see bug #688444)
--- a/src/map_geo.c	2019-07-26 15:54:19.534488550 +0200
+++ b/src/map_geo.c	2019-07-26 15:56:47.779713950 +0200
@@ -125,20 +125,7 @@
 #undef PACKAGE_TARNAME
 #define XASTIR_PACKAGE_VERSION PACKAGE_VERSION
 #undef PACKAGE_VERSION
-#ifdef HAVE_GRAPHICSMAGICK
-/*#include <GraphicsMagick/magick/api.h>*/
-/* Define MAGICK_IMPLEMENTATION to access private interfaces
- * such as DestroyImagePixels(). This may not be a good thing,
- * but DestroyImagePixels() has been in this code for a long
- * time. Defining MAGIC_IMPLEMENTATION eliminates the warning that is
- * now (9/28/2010) being seen on some distros (Ubuntu 10.04 and
- * OpenSuSE-11.3)
- */
-#define MAGICK_IMPLEMENTATION
 #include <magick/api.h>
-#else   // HAVE_GRAPHICSMAGICK
-#include <magick/api.h>
-#endif  // HAVE_GRAPHICSMAGICK
 #undef PACKAGE_BUGREPORT
 #define PACKAGE_BUGREPORT XASTIR_PACKAGE_BUGREPORT
 #undef XASTIR_PACKAGE_BUGREPORT
@@ -2030,7 +2017,6 @@
             if (!SyncImagePixels(image))
                 fprintf(stderr, "SyncImagePixels Failed....\n");
 	}
-        DestroyImagePixels(image);
     }
 
     // If were are drawing to a low bpp display (typically < 8bpp)
