Bug: https://bugs.gentoo.org/895898
--- a/src/gui/render_templates_sai.h
+++ b/src/gui/render_templates_sai.h
@@ -43,7 +43,7 @@ inline void conc2d(Super2xSaI,SBPP)(PTYPE * line0, PTYPE * line1, const PTYPE *
 	} else if (C4 == C8 && C7 != C5) {
 		line1[1] = line0[1] = C4;
 	} else if (C4 == C8 && C7 == C5) {
-		register int r = 0;
+		int r = 0;
 		r += conc2d(GetResult,SBPP)(C5,C4,C6,D1);
 		r += conc2d(GetResult,SBPP)(C5,C4,C3,C1);
 		r += conc2d(GetResult,SBPP)(C5,C4,D2,D5);
@@ -127,7 +127,7 @@ inline void conc2d(SuperEagle,SBPP)(PTYPE * line0, PTYPE * line1, const PTYPE *
 				line1[0] = interp_w2(C7,C8,1U,1U);
 			}
 		} else {
-			register int r = 0;
+			int r = 0;
 			r += conc2d(GetResult,SBPP)(C5,C4,C6,D1);
 			r += conc2d(GetResult,SBPP)(C5,C4,C3,C1);
 			r += conc2d(GetResult,SBPP)(C5,C4,D2,D5);
@@ -185,7 +185,7 @@ inline void conc2d(_2xSaI,SBPP)(PTYPE * line0, PTYPE * line1, const PTYPE * fc)
 			line1[0] = C4;
 			line1[1] = C4;
 		} else {
-			register int r = 0;
+			int r = 0;
 			r += conc2d(GetResult,SBPP)(C4,C5,C3,C1);
 			r -= conc2d(GetResult,SBPP)(C5,C4,D4,C2);
 			r -= conc2d(GetResult,SBPP)(C5,C4,C6,D1);
