diff --git a/src/plugins/vala-pack/meson.build b/src/plugins/vala-pack/meson.build
index dd36d351d..a8c53c6e5 100644
--- a/src/plugins/vala-pack/meson.build
+++ b/src/plugins/vala-pack/meson.build
@@ -6,8 +6,8 @@ endif
 
 add_languages('vala')
 
-valac = find_program ('valac')
-libvala_version = run_command(valac, '--api-version').stdout().strip()
+valac = meson.get_compiler('vala')
+libvala_version = run_command(valac.cmd_array()[0], '--api-version').stdout().strip()
 libvala = dependency('libvala-@0@'.format(libvala_version))
 
 vala_pack_sources = [
