diff --git a/meson_options.txt b/meson_options.txt
index 36fcfc969..cef10804d 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -13,6 +13,7 @@ option('channel',
 
 option('help', type: 'boolean', value: false)
 option('docs', type: 'boolean', value: false)
+option('vapi', type: 'boolean')
 
 option('network_tests', type: 'boolean', value: true, description: 'Allow networking in unit-tests')
 
diff --git a/src/meson.build b/src/meson.build
index bc7f91841..53d678297 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -124,6 +124,7 @@ configure_file(
            copy: true,
 )
 
+if get_option('vapi')
 libide_vapi = gnome.generate_vapi('libide-' + libide_api_version,
         sources: libide_gir[0],
         install: true,
@@ -141,3 +142,4 @@ libide_vapi = gnome.generate_vapi('libide-' + libide_api_version,
 
 # Must be after vapi generation
 subdir('plugins/vala-pack')
+endif
