 axel.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/axel.h b/axel.h
index e90d30d..f6ebfe8 100644
--- a/axel.h
+++ b/axel.h
@@ -66,7 +66,7 @@
 /* Compiled-in settings							*/
 #define MAX_STRING		1024
 #define MAX_ADD_HEADERS	10
-#define MAX_REDIR		5
+#define MAX_REDIR		25
 #define AXEL_VERSION_STRING	"2.4"
 #define DEFAULT_USER_AGENT	"Axel " AXEL_VERSION_STRING " (" ARCH ")"
 
