Index: source/texk/web2c/luatexdir/image/pdftoepdf.w
===================================================================
--- source.orig/texk/web2c/luatexdir/image/pdftoepdf.w
+++ source/texk/web2c/luatexdir/image/pdftoepdf.w
@@ -26,6 +26,8 @@ static const char _svn_version[] =
 // define DEBUG
 #define __STDC_FORMAT_MACROS /* for PRId64 etc.  */
 
+#include <stdint.h>
+#include <inttypes.h>
 #include "image/epdf.h"
 
 // This file is mostly C and not very much C++; it's just used to interface
Index: source/texk/web2c/luatexdir/image/writejp2.w
===================================================================
--- source.orig/texk/web2c/luatexdir/image/writejp2.w
+++ source/texk/web2c/luatexdir/image/writejp2.w
@@ -30,6 +30,7 @@ ISO/IEC 15444-1, Second edition, 2004-09
 @c
 #include <math.h>
 #include <assert.h>
+#include <stdint.h>
 #include "ptexlib.h"
 #include "image/image.h"
 #include "image/writejp2.h"
