Index: gwyddion/modules/file/ambprofile.c
===================================================================
--- gwyddion/modules/file/ambprofile.c	(revision 22650)
+++ gwyddion/modules/file/ambprofile.c	(revision 22651)
@@ -32,8 +32,8 @@
  *   <comment>Ambios XML profile data</comment>
  *   <magic priority="80">
  *     <match type="string" offset="0" value="&lt;?xml">
- *       <match type="string" offset="40:120" value="<ProfilometerData>">
- *          <match type="string" offset="60:140" value="<Header>"/>
+ *       <match type="string" offset="40:120" value="&lt;ProfilometerData&gt;">
+ *          <match type="string" offset="60:140" value="&lt;Header&gt;"/>
  *       </match>
  *     </match>
  *   </magic>
