diff -Nru gateway-1.4.1.orig/configure.in gateway-1.4.1/configure.in
--- gateway-1.4.1.orig/configure.in	2008-06-24 02:00:07.000000000 +0200
+++ gateway-1.4.1/configure.in	2008-06-24 02:06:05.000000000 +0200
@@ -108,7 +108,6 @@
 AC_PROG_INSTALL
 AC_PROG_RANLIB
 AC_PROG_YACC
-AC_PROG_LEX
 AC_PATH_PROG(CONVERT, convert)
 AC_PATH_PROG(PERL, perl)
 
diff -Nru gateway-1.4.1.orig/Makefile.in gateway-1.4.1/Makefile.in
--- gateway-1.4.1.orig/Makefile.in	2006-05-09 07:46:56.000000000 +0200
+++ gateway-1.4.1/Makefile.in	2008-06-24 02:06:21.000000000 +0200
@@ -60,7 +60,6 @@
 SHELL = @SHELL@
 VERSION = @VERSION@
 SUFFIX = @SUFFIX@
-LEX = @LEX@
 PERL = @PERL@
 YACC = @YACC@
 
