--- src/src.pro	2011-01-11 09:27:21.530166986 +0100
+++ src/src.pro	2011-01-11 09:30:55.743128934 +0100
@@ -7,10 +7,7 @@
 APR_INCLUDE = /usr/include/apr-1
 exists(local-config.pri):include(local-config.pri)
 
-VERSION = $$system(git --no-pager show --pretty=oneline --no-notes | head -1 | cut -b-40)
-    !isEmpty(VERSION){
-      VERSION = $${VERSION}
-    }
+VERSION = "1.0.5"
 
 VERSTR = '\\"$${VERSION}\\"'  # place quotes around the version string
 DEFINES += VER=\"$${VERSTR}\" # create a VER macro containing the version string
