From f7d821a6837b4cc9defa02f271256796394fd8c3 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Wed, 21 Sep 2011 14:37:31 +0200
Subject: [PATCH] Allow compilation with zlib-1.2.5.1-r1

---
 quazip/ioapi.h |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/quazip/ioapi.h b/quazip/ioapi.h
index 716dd4b..a4ac8d7 100644
--- a/quazip/ioapi.h
+++ b/quazip/ioapi.h
@@ -37,6 +37,10 @@
 extern "C" {
 #endif
 
+#ifndef OF
+# define OF(x) x
+#endif
+
 typedef voidpf (ZCALLBACK *open_file_func) OF((voidpf opaque, voidpf file, int mode));
 typedef uLong  (ZCALLBACK *read_file_func) OF((voidpf opaque, voidpf stream, void* buf, uLong size));
 typedef uLong  (ZCALLBACK *write_file_func) OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
-- 
1.7.6.1

