--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -384,7 +384,9 @@
 
 # manual pages
 IF( UNIX )
-    INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/debian/vbam.1 DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man1)
+    IF( ENABLE_SDL )
+        INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/debian/vbam.1 DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man1)
+    ENDIF( ENABLE_SDL )
     IF( ENABLE_GTK )
         INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/debian/gvbam.1 DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man1)
     ENDIF( ENABLE_GTK )
