--- branches/cfv-1_x-branch/cfv/cfv	2006/03/28 07:29:01	340
+++ branches/cfv-1_x-branch/cfv/cfv	2007/03/04 01:57:56	401
@@ -1360,12 +1360,15 @@
 #---------- .torrent ----------
 _btimporterror = None
 try:
-	from BitTorrent import bencode, btformats
+	from BTL import bencode, btformats
 except ImportError, e1:
 	try:
-		from BitTornado import bencode; from BitTornado.BT1 import btformats
+		from BitTorrent import bencode, btformats
 	except ImportError, e2:
-		_btimporterror = '%s and %s'%(e1,e2)
+		try:
+			from BitTornado import bencode; from BitTornado.BT1 import btformats
+		except ImportError, e3:
+			_btimporterror = '%s and %s and %s'%(e1,e2,e3)
 
 class Torrent(ChksumType):
 	description = 'BitTorrent metainfo'
