diff -NrU5 libdca-0.0.5.orig/test/Makefile.am libdca-0.0.5/test/Makefile.am
--- libdca-0.0.5.orig/test/Makefile.am	2008-06-13 23:54:27.000000000 +0200
+++ libdca-0.0.5/test/Makefile.am	2008-06-13 23:54:46.000000000 +0200
@@ -1,10 +1,10 @@
 AM_CFLAGS = $(DCADEC_CFLAGS)
 
-noinst_PROGRAMS = compare
+check_PROGRAMS = compare
 compare_SOURCES = compare.c
 compare_LDADD = $(COMPARE_LIBS)
 
 dist_check_SCRIPTS = regression compile globals
 EXTRA_DIST = tests
 TESTS = regression compile
 #broken test: globals
