diff -Naur conky-1.10.7_old/src/linux.cc conky-1.10.7/src/linux.cc
--- conky-1.10.7_old/src/linux.cc	2018-01-19 16:59:14.158830825 +0100
+++ conky-1.10.7/src/linux.cc	2018-01-19 16:59:44.796144615 +0100
@@ -923,6 +923,9 @@
 			} else {
 				idx = 0;
 			}
+			if (idx > info.cpu_count) {
+				continue;
+			}
 			sscanf(buf, stat_template, &(cpu[idx].cpu_user),
 				&(cpu[idx].cpu_nice), &(cpu[idx].cpu_system),
 				&(cpu[idx].cpu_idle), &(cpu[idx].cpu_iowait),
