diff --git a/python/uv_build/__init__.py b/python/uv_build/__init__.py
index c8ce9c5..835ba0f 100644
--- a/python/uv_build/__init__.py
+++ b/python/uv_build/__init__.py
@@ -39,11 +39,11 @@ def call(
 
     warn_config_settings(config_settings)
     # Unlike `find_uv_bin`, this mechanism must work according to PEP 517
-    uv_bin = shutil.which("uv-build")
+    uv_bin = shutil.which("uv")
     if uv_bin is None:
         raise RuntimeError("uv was not properly installed")
     # Forward stderr, capture stdout for the filename
-    result = subprocess.run([uv_bin, *args], stdout=subprocess.PIPE)
+    result = subprocess.run([uv_bin, "build-backend", *args], stdout=subprocess.PIPE)
     if result.returncode != 0:
         sys.exit(result.returncode)
     # If there was extra stdout, forward it (there should not be extra stdout)
