diff -Nuar MiniSat_v1.14/Makefile MiniSat_v1.14.new/Makefile
--- MiniSat_v1.14/Makefile	2006-04-02 01:33:46.000000000 -0800
+++ MiniSat_v1.14.new/Makefile	2006-04-02 01:31:39.000000000 -0800
@@ -26,10 +26,11 @@
 RANLIB = ranlib
 AR = ar
 
-.PHONY : ls s p d r build clean depend
+.PHONY : lr ls s p d r build clean depend
 
 s:	WAY=standard
 ls:	WAY=standard
+lr: WAY=release
 p:	WAY=profile
 d:	WAY=debug
 r:	WAY=release
@@ -38,8 +39,7 @@
 s:	CFLAGS+=$(COPTIMIZE) -ggdb -D DEBUG
 p:	CFLAGS+=$(COPTIMIZE) -pg -ggdb -D DEBUG
 d:	CFLAGS+=-O0 -ggdb -D DEBUG
-r:	CFLAGS+=$(COPTIMIZE) -D NDEBUG
-rs:	CFLAGS+=$(COPTIMIZE) -D NDEBUG
+r rs ls lr:	CFLAGS+=$(COPTIMIZE) -D NDEBUG
 
 s:	build $(EXEC)
 p:	build $(EXEC)_profile
@@ -48,7 +48,7 @@
 rs:	build $(EXEC)_static
 
 s:	CFLAGS+=$(COPTIMIZE) -ggdb -D DEBUG
-ls:	lbuild $(LIB)_s 
+ls lr:	lbuild $(LIB)_s 
 
 build:
 	@echo Building $(EXEC) "("$(WAY)")"
@@ -63,7 +63,7 @@
 ## Build rule
 %.o %.op %.od %.or:	%.C
 	@echo Compiling: $<
-	@$(CXX) $(CFLAGS) -c -o $@ $<
+	$(CXX) $(CFLAGS) -c -o $@ $<
 
 ## Linking rules (standard/profile/debug/release)
 $(EXEC): $(COBJS)
