--- Makefile.inc.in.orig	2012-06-20 09:08:48.000000000 +0200
+++ Makefile.inc.in	2012-06-20 09:10:23.000000000 +0200
@@ -42,12 +42,12 @@
 INSTALL		= @INSTALL@
 INSTALL_PROGRAM	= @INSTALL_PROGRAM@
 INSTALL_DATA	= @INSTALL_DATA@
-prefix		= @prefix@
+prefix		= $(DESTDIR)@prefix@
 exec_prefix	= @exec_prefix@
 includedir	= @includedir@
 bindir		= @bindir@
-mandir		= @mandir@
-libdir		= @libdir@
+mandir		= $(DESTDIR)@mandir@
+libdir		= $(DESTDIR)@libdir@
 
 # http://www.gnu.org/s/hello/manual/autoconf/Changed-Directory-Variables.html
 datarootdir = @datarootdir@
