diff -Naur torsocks-2.0.0-rc2.orig/doc/Makefile.am torsocks-2.0.0-rc2/doc/Makefile.am
--- torsocks-2.0.0-rc2.orig/doc/Makefile.am	2013-09-02 14:06:44.000000000 -0400
+++ torsocks-2.0.0-rc2/doc/Makefile.am	2013-09-17 14:19:37.769272561 -0400
@@ -6,5 +6,5 @@
 dist_man5_MANS = torsocks.conf.5
 dist_man8_MANS = torsocks.8
 
-dist_data_DATA = socks/SOCKS5 socks/socks-extensions.txt notes/DEBUG
+#dist_data_DATA = socks/SOCKS5 socks/socks-extensions.txt notes/DEBUG
 
diff -Naur torsocks-2.0.0-rc2.orig/Makefile.am torsocks-2.0.0-rc2/Makefile.am
--- torsocks-2.0.0-rc2.orig/Makefile.am	2013-09-02 14:06:44.000000000 -0400
+++ torsocks-2.0.0-rc2/Makefile.am	2013-09-17 14:20:07.945273935 -0400
@@ -2,6 +2,6 @@
 
 SUBDIRS = src doc tests
 
-dist_doc_DATA = ChangeLog
+#dist_doc_DATA = ChangeLog
 
 EXTRA_DIST = gpl-2.0.txt
