=== modified file 'Makefile.PL'
--- Makefile.PL	2010-09-16 06:16:42 +0000
+++ Makefile.PL	2010-09-16 20:24:41 +0000
@@ -64,6 +64,12 @@
 $test_install_extra .
 '
 
+speedy/speedy: subdirs
+	$(NOECHO) cd speedy && $(MAKE) $(USEMAKEFILE) $(FIRST_MAKEFILE) all $(PASTHRU)
+
+speedy_backend/speedy_backend: subdirs
+	$(NOECHO) cd speedy_backend && $(MAKE) $(USEMAKEFILE) $(FIRST_MAKEFILE) all $(PASTHRU)
+
 mydist: README README.html lib/CGI/SpeedyCGI.pm readable tardist
 
 readable:

