--- Gauche-0.9.4.orig/m4/gauche.m4
+++ Gauche-0.9.4/m4/gauche.m4
@@ -103,15 +103,11 @@
   i686-*) I686OPT="-DUSE_I686_PREFETCH";;
 esac
 case "$CC" in
-  gcc*)  # some systems may have gcc-2.95, gcc-3, etc.
+  *gcc*)  # some systems may have gcc-2.95, gcc-3, etc.
     case "$target" in
       *mingw*) ;;
       *)     GCCOPT="-fomit-frame-pointer";;
     esac
-    case "$target" in
-     i586-*) GCCOPT="$GCCOPT -march=i586";;
-     i686-*) GCCOPT="$GCCOPT -march=i686";;
-    esac
     ;;
 esac
 OPTFLAGS="$GCCOPT $I686OPT"
