diff -urNad ogdi-dfsg~/vpflib/include/machine.h ogdi-dfsg/vpflib/include/machine.h
--- ogdi-dfsg~/vpflib/include/machine.h	2007-03-27 17:35:24.000000000 +0200
+++ ogdi-dfsg/vpflib/include/machine.h	2007-03-27 17:41:16.000000000 +0200
@@ -15,10 +15,11 @@
     long            output;
 } xBYTE_ORDER;
 
-#if SYS_BIG_ENDIAN == 1
-#define MACHINE_BYTE_ORDER MOST_SIGNIFICANT
-#else
+#include <endian.h>
+#if __BYTE_ORDER == __LITTLE_ENDIAN
 #define MACHINE_BYTE_ORDER LEAST_SIGNIFICANT
+#elif __BYTE_ORDER == __BIG_ENDIAN
+#define MACHINE_BYTE_ORDER MOST_SIGNIFICANT
 #endif
 
 #ifdef _WINDOWS
