diff -ruN digikam-3.0.0.orig/extra/kipi-plugins/CMakeLists.txt digikam-3.0.0/extra/kipi-plugins/CMakeLists.txt
--- digikam-3.0.0.orig/extra/kipi-plugins/CMakeLists.txt	2013-02-07 00:52:03.000000000 +0100
+++ digikam-3.0.0/extra/kipi-plugins/CMakeLists.txt	2013-02-09 22:33:55.284100771 +0100
@@ -424,7 +424,11 @@
     ADD_SUBDIRECTORY(picasawebexport)
     ADD_SUBDIRECTORY(smug)
     ADD_SUBDIRECTORY(printimages)
-    ADD_SUBDIRECTORY(expoblending)
+
+    IF(ENABLE_expoblending)
+        ADD_SUBDIRECTORY(expoblending)
+    ENDIF(ENABLE_expoblending)
+
     ADD_SUBDIRECTORY(kopete)
     ADD_SUBDIRECTORY(batchprocessimages)
     ADD_SUBDIRECTORY(rajceexport)
@@ -492,9 +496,9 @@
         ADD_SUBDIRECTORY(mediawiki)
     ENDIF(MEDIAWIKI_FOUND)
 
-    IF(FLEX_FOUND AND BISON_FOUND)
+    IF(FLEX_FOUND AND BISON_FOUND AND ENABLE_panorama)
         ADD_SUBDIRECTORY(panorama)
-    ENDIF(FLEX_FOUND AND BISON_FOUND)
+    ENDIF(FLEX_FOUND AND BISON_FOUND AND ENABLE_panorama)
 
     IF(ImageMagick_FOUND AND QTGSTREAMER_FOUND)
         ADD_SUBDIRECTORY(videoslideshow)
