From 5f8fb9cd211f0e12ef40a0fc34a42c31f2a24475 Mon Sep 17 00:00:00 2001
From: Johnny Jazeix <jazeix@gmail.com>
Date: Tue, 7 Dec 2021 19:03:19 +0100
Subject: [PATCH] fix end of paragraph tag in html for the rksetupwizard

---
 rkward/dialogs/rksetupwizard.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/rkward/dialogs/rksetupwizard.cpp b/rkward/dialogs/rksetupwizard.cpp
index b14e06ec..e4787793 100644
--- a/rkward/dialogs/rksetupwizard.cpp
+++ b/rkward/dialogs/rksetupwizard.cpp
@@ -130,12 +130,12 @@ RKSetupWizard::RKSetupWizard(QWidget* parent, InvokationReason reason, const QLi
 	auto l = new QVBoxLayout(firstpage);
 	QString intro = i18n("<p>This dialog will guide you through a quick check of the basic setup of the required (or recommended) components.</p>");
 	if (reason == NewVersionRKWard) {
-		intro += i18n("<p>The setup assistant has been invoked, automatically, because a new version of RKWard has been detected.</p");
+		intro += i18n("<p>The setup assistant has been invoked, automatically, because a new version of RKWard has been detected.</p>");
 	} else if (reason == NewVersionR) {
 		// TODO: invoke this!
-		intro += i18n("<p>The setup assistant has been invoked, automatically, because a new version of R has been detected.</p");
+		intro += i18n("<p>The setup assistant has been invoked, automatically, because a new version of R has been detected.</p>");
 	} else if (reason == ProblemsDetected) {
-		intro += i18n("<p>The setup assistant has been invoked, automatically, because a problem has been detected in your setup.</p");
+		intro += i18n("<p>The setup assistant has been invoked, automatically, because a problem has been detected in your setup.</p>");
 	}
 	l->addWidget(RKCommonFunctions::wordWrappedLabel(intro));
 	auto waiting_to_start_label = RKCommonFunctions::wordWrappedLabel(i18n("<b>Waiting for R backend...</b>") + "<p>&nbsp;</p><p>&nbsp;</p>");
-- 
GitLab

