diff --git a/ext/cool.io/cool.io.h b/ext/cool.io/cool.io.h
index 364d4ae..05ce5bc 100644
--- a/ext/cool.io/cool.io.h
+++ b/ext/cool.io/cool.io.h
@@ -8,7 +8,11 @@
 #define COOLIO_H
 
 #include "ruby.h"
+#ifdef HAVE_RUBY_IO_H
+#include "ruby/io.h"
+#else
 #include "rubyio.h"
+#endif
 
 #ifdef GetReadFile
 #define FPTR_TO_FD(fptr) (fileno(GetReadFile(fptr)))
diff --git a/ext/cool.io/iowatcher.c b/ext/cool.io/iowatcher.c
index 09ff87b..ccf6a3f 100644
--- a/ext/cool.io/iowatcher.c
+++ b/ext/cool.io/iowatcher.c
@@ -5,7 +5,11 @@
  */
 
 #include "ruby.h"
+#ifdef HAVE_RUBY_IO_H
+#include "ruby/io.h"
+#else
 #include "rubyio.h"
+#endif
 
 #include "ev_wrap.h"
 
