From 88924722c0e25566953276c52df7ef5040714cc5 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Tue, 5 Dec 2017 01:19:53 +0100
Subject: [PATCH] Fix Qt5 moc detection, throw away Qt4

---
 configure.ac | 24 ++++++------------------
 1 file changed, 6 insertions(+), 18 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6580963..eb4ccec 100644
--- a/configure.ac
+++ b/configure.ac
@@ -325,24 +325,12 @@ if eval "test $enable_sherlock265 = yes" && eval "test $have_videogfx != yes" &&
 fi
 
 if eval "test $enable_sherlock265 = yes" ; then
-  PKG_CHECK_MODULES([QT], [Qt5Core Qt5Gui Qt5Widgets], [found_qt=5], [found_qt=no])
-  if eval "test $found_qt = no" ; then
-    PKG_CHECK_MODULES([QT], [QtCore QtGui], [found_qt=4])
-  fi
-  AC_PATH_PROGS([QTCHOOSER],[qtchooser])
-  if eval "test x$QTCHOOSER = x" ; then
-    AC_PATH_PROGS([QTMOC],[moc-qt$found_qt moc])
-    if eval "test x$QTMOC = x" ; then
-      AC_MSG_ERROR([Need the "moc" commandline tool which is required to generate the Qt files required for sherlock265.])
-    fi
-  else
-    QTMOC="$QTCHOOSER -run-tool=moc -qt=$found_qt"
-  fi
-  AC_MSG_CHECKING([for version of $QTMOC])
-  QTMOC_VERSION=`$QTMOC -v 2>&1 | $GREP -o '[[0-9]]\+.[[0-9]]\+.[[0-9]]\+'`
-  AC_MSG_RESULT([$QTMOC_VERSION])
-  AX_COMPARE_VERSION([$QTMOC_VERSION], [lt], [$found_qt],
-      [AC_MSG_ERROR([Please install "moc" for Qt$found_qt (found $QTMOC_VERSION).])])
+  PKG_CHECK_MODULES([QT], [Qt5Core >= 5.1.0 Qt5Gui Qt5Widgets], [
+    QT_PATH="$( eval $PKG_CONFIG --variable=host_bins Qt5Core )"
+    AC_PATH_PROGS([QTMOC], [moc-qt5 moc], [moc], [$QT_PATH])
+  ], [
+    AC_MSG_ERROR([Qt5 moc not found, which is required to generate the Qt files for sherlock265.])
+  ])
   AC_SUBST(QTMOC)
 fi
 
-- 
2.15.1

