diff -ur ogmtools-1.5.orig/ogminfo.c ogmtools-1.5/ogminfo.c
--- ogmtools-1.5.orig/ogminfo.c	2003-11-21 00:21:36.000000000 +0200
+++ ogmtools-1.5/ogminfo.c	2009-05-12 12:41:58.000000000 +0300
@@ -513,7 +513,7 @@
               stream->serial, stream->size,
               (stream->size * 8.0 / 1000.0) / now,
               (stream->size / 1024.0) / now,
-              stream->num_packets, now * 1000.0);
+              stream->num_packets, now);
       stream = stream->next;
     }
   }
