--- a/lib/epson-escpage-mono.c
+++ b/lib/epson-escpage-mono.c
@@ -885,7 +885,7 @@
 
 	EPS_LOG_FUNCIN;
 
-	memset(pBand, 0, sizeof(pBand));
+	memset(pBand, 0, sizeof(EPS_PAGE_BAND_M));
 
     pBand->WidthInPixels = widthInPixels;
     pBand->WidthInBytes = (pBand->WidthInPixels+7)/8;
--- a/lib/epson-escpr-api.h
+++ b/lib/epson-escpr-api.h
@@ -98,6 +98,9 @@
     /*** -------------------------------------------------------------------------------*/
 extern EPS_ERR_CODE epsMakeMainteCmd     (EPS_INT32, EPS_UINT8*, EPS_UINT32*            );
 
+extern EPS_ERR_CODE SetupJobAttrib          (const EPS_JOB_ATTRIB*                      );
+extern EPS_ERR_CODE SendStartJob            (EPS_BOOL                                   );
+extern EPS_ERR_CODE PrintBand               (const EPS_UINT8*, EPS_UINT32, EPS_UINT32*  );
 #ifdef __cplusplus
 }
 #endif
--- a/src/filter.c
+++ b/src/filter.c
@@ -38,6 +38,7 @@
 #include "epson-escpr-api.h"
 #include "epson-escpr-services.h"
 #include "epson-escpr-mem.h"
+#include "epson-escpage.h"
 
 #include "err.h"
 #include "mem.h"
--- a/src/mem.c
+++ b/src/mem.c
@@ -23,6 +23,7 @@
 #endif
 
 #include <stdlib.h>
+#include "err.h"
 #include "mem.h"
 
 void *
