diff -uNr opentyrian.ORIG//Makefile opentyrian/Makefile
--- opentyrian.ORIG//Makefile	2011-05-17 11:26:17.030004857 +0100
+++ opentyrian/Makefile	2011-05-17 11:26:52.595004876 +0100
@@ -39,6 +39,10 @@
 ALL_LDFLAGS += $(LDFLAGS)
 LDLIBS += $(EXTRA_LDLIBS) $(SDL_LDLIBS)
 
+ifneq ($(DATA_PATH), )
+	CFLAGS += -DDATA_PATH=\"$(DATA_PATH)\"
+endif
+
 # RULES ####################################################
 
 .PHONY : all release clean
diff -uNr opentyrian.ORIG//src/file.c opentyrian/src/file.c
--- opentyrian.ORIG//src/file.c	2011-05-17 11:26:17.020004860 +0100
+++ opentyrian/src/file.c	2011-05-17 11:26:52.595004876 +0100
@@ -34,7 +34,7 @@
 #ifdef TARGET_MACOSX
 		tyrian_game_folder(),
 #endif
-		"/usr/share/opentyrian/data"
+		DATA_PATH
 	};
 	
 	static const char *dir = NULL;
