diff -Nur mpb-1.4.2.orig/configure.ac mpb-1.4.2/configure.ac
--- mpb-1.4.2.orig/configure.ac	2003-03-04 02:38:41.000000000 +0000
+++ mpb-1.4.2/configure.ac	2008-11-25 17:31:12.000000000 +0000
@@ -6,15 +6,8 @@
 MPB_VERSION=1.4.2
 AC_SUBST(MPB_VERSION)
 
-AC_MSG_CHECKING([for vendor f77 to be used instead of g77])
-AC_MSG_RESULT()
-AC_CHECK_PROGS(F77, f77 xlf77 cf77 fort77 f90 xlf90)
 AC_PROG_F77
-
-# Get C compiler.
-AC_MSG_CHECKING([for vendor cc to be used instead of gcc])
-AC_MSG_RESULT()
-AC_CHECK_PROGS(CC, cc xlc) # prefer vendor cc, to stay in sync with Fortran
+AC_F77_WRAPPERS
 AC_PROG_CC
 
 # Checks for programs.
@@ -79,12 +72,6 @@
 	echo "*********************** HERMITIAN EPS. ***********************"
 fi
 
-# Add lots of compiler warnings to check for if we are using gcc:
-# (The variable $GCC is set to "yes" by AC_PROG_CC if we are using gcc.)
-if test "$GCC" = "yes"; then
-	CFLAGS="$CFLAGS -Wall -W -Wbad-function-cast -Wcast-qual -Wpointer-arith -Wcast-align -pedantic"
-fi
-
 ##############################################################################
 #                       Checks for libraries:
 ##############################################################################
@@ -203,6 +190,10 @@
 	                     # how to run mpb-mpi in general
 fi
 
+# Checks for header files.
+AC_HEADER_STDC
+AC_CHECK_HEADERS(unistd.h getopt.h)
+
 ##############################################################################
 # Check for HDF5 library
 AC_ARG_WITH(hdf5, [AC_HELP_STRING([--without-hdf5],[do not use HDF5 output])], ok=$withval, ok=yes)
@@ -349,9 +340,7 @@
 ]], [AC_MSG_RESULT(ok)], [AC_MSG_ERROR([libctl version ${LIBCTL_MAJOR}.${LIBCTL_MINOR}.${LIBCTL_BUGFIX} or later is required])])
 
 ##############################################################################
-# Checks for header files.
-AC_HEADER_STDC
-AC_CHECK_HEADERS(unistd.h getopt.h)
+
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
diff -Nur mpb-1.4.2.orig/Makefile.in mpb-1.4.2/Makefile.in
--- mpb-1.4.2.orig/Makefile.in	2002-01-18 22:52:41.000000000 +0000
+++ mpb-1.4.2/Makefile.in	2008-11-25 17:31:12.000000000 +0000
@@ -2,6 +2,9 @@
 
 SHELL = @SHELL@
 
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
 CC = @CC@
 CFLAGS = @CFLAGS@
 CPPFLAGS = @CPPFLAGS@
diff -Nur mpb-1.4.2.orig/mpb-ctl/Makefile.in mpb-1.4.2/mpb-ctl/Makefile.in
--- mpb-1.4.2.orig/mpb-ctl/Makefile.in	2002-09-14 23:20:55.000000000 +0100
+++ mpb-1.4.2/mpb-ctl/Makefile.in	2008-11-25 18:23:20.000000000 +0000
@@ -13,6 +13,12 @@
 
 SHELL = @SHELL@
 
+mandir = @mandir@
+
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
+
 ##############################################################################
 
 # Makefile.in for programs using libctl (assumes that autoconf is
@@ -37,8 +43,8 @@
           ../src/util/util.a
 
 MY_LDFLAGS =         # extra -L flags go here
-MY_CPPFLAGS = -I. -I../src/util -I../src/matrices -I../src/matrixio \
-              -I../src/maxwell
+MY_CPPFLAGS = -I$(srcdir) -I$(top_srcdir)/src/util -I$(top_srcdir)/src/matrices -I$(top_srcdir)/src/matrixio \
+              -I$(top_srcdir)/src/maxwell
 MY_DEFS = -DHAVE_CTL_HOOKS=1 -DHAVE_CTL_EXPORT_HOOK=1
 
 # The following variables should be detected and set by autoconf:
@@ -134,11 +140,11 @@
 
 install-mpb: $(PROGRAM_NAME)
 	$(INSTALL) -d $(prefix)/bin
-	$(INSTALL) -m 0755 -s .$(PROGRAM_NAME) $(prefix)/bin/`echo $(PROGRAM_NAME)|sed '$(transform)'`
+	$(INSTALL) -m 0755 .$(PROGRAM_NAME) $(prefix)/bin/`echo $(PROGRAM_NAME)|sed '$(transform)'`
 	$(INSTALL) -d $(prefix)/share/libctl/specs
 	$(INSTALL) -m 0644 $(SPECIFICATION_FILE) $(prefix)/share/libctl/specs
 	$(INSTALL) -d $(mandir)/man1
-	$(INSTALL) -m 0644 $(PROGRAM_NAME).1 $(mandir)/man1
+	$(INSTALL) -m 0644 $(srcdir)/$(PROGRAM_NAME).1 $(mandir)/man1
 
 mpb-split-tmp:
 	rm -f mpb-split-tmp
@@ -149,7 +155,7 @@
 	mpb_name=`echo mpb | sed '$(transform)'`; \
 	$(INSTALL) -m 0755 mpb-split-tmp $(prefix)/bin/$${mpb_name}-split
 	$(INSTALL) -d $(mandir)/man1
-	$(INSTALL) -m 0644 mpb-split.1 $(mandir)/man1
+	$(INSTALL) -m 0644 $(srcdir)/mpb-split.1 $(mandir)/man1
 
 clean:
 	rm -f $(ALL_OBJECTS) ctl-io.c ctl-io.h main.c geom.c \
diff -Nur mpb-1.4.2.orig/src/matrices/Makefile.in mpb-1.4.2/src/matrices/Makefile.in
--- mpb-1.4.2.orig/src/matrices/Makefile.in	2002-09-14 23:20:55.000000000 +0100
+++ mpb-1.4.2/src/matrices/Makefile.in	2008-11-25 17:31:12.000000000 +0000
@@ -2,9 +2,11 @@
 
 SHELL = @SHELL@
 
-CC = @CC@
-CFLAGS = @CFLAGS@
-CPPFLAGS = -I../util -I. @CPPFLAGS@
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
+
+CPPFLAGS = -I$(top_srcdir)/src/util -I$(srcdir) -I. @CPPFLAGS@
 DEFS = @DEFS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
diff -Nur mpb-1.4.2.orig/src/matrixio/Makefile.in mpb-1.4.2/src/matrixio/Makefile.in
--- mpb-1.4.2.orig/src/matrixio/Makefile.in	2002-09-14 23:20:55.000000000 +0100
+++ mpb-1.4.2/src/matrixio/Makefile.in	2008-11-25 17:31:12.000000000 +0000
@@ -2,9 +2,11 @@
 
 SHELL = @SHELL@
 
-CC = @CC@
-CFLAGS = @CFLAGS@
-CPPFLAGS = -I../util -I../matrices -I. @CPPFLAGS@
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
+
+CPPFLAGS = -I$(top_srcdir)/src/util -I$(top_srcdir)/src/matrices -I$(srcdir) -I. @CPPFLAGS@
 DEFS = @DEFS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
diff -Nur mpb-1.4.2.orig/src/maxwell/Makefile.in mpb-1.4.2/src/maxwell/Makefile.in
--- mpb-1.4.2.orig/src/maxwell/Makefile.in	2002-09-14 23:20:55.000000000 +0100
+++ mpb-1.4.2/src/maxwell/Makefile.in	2008-11-25 17:31:12.000000000 +0000
@@ -2,9 +2,11 @@
 
 SHELL = @SHELL@
 
-CC = @CC@
-CFLAGS = @CFLAGS@
-CPPFLAGS = -I../util -I../matrices @CPPFLAGS@
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
+
+CPPFLAGS = -I$(top_srcdir)/src/util -I$(top_srcdir)/src/matrices -I. @CPPFLAGS@
 DEFS = @DEFS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
diff -Nur mpb-1.4.2.orig/src/util/Makefile.in mpb-1.4.2/src/util/Makefile.in
--- mpb-1.4.2.orig/src/util/Makefile.in	2002-09-14 23:20:55.000000000 +0100
+++ mpb-1.4.2/src/util/Makefile.in	2008-11-25 17:31:12.000000000 +0000
@@ -2,9 +2,11 @@
 
 SHELL = @SHELL@
 
-CC = @CC@
-CFLAGS = @CFLAGS@
-CPPFLAGS = -I. @CPPFLAGS@
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
+
+CPPFLAGS = -I$(top_srcdir) -I. @CPPFLAGS@
 DEFS = @DEFS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
diff -Nur mpb-1.4.2.orig/tests/Makefile.in mpb-1.4.2/tests/Makefile.in
--- mpb-1.4.2.orig/tests/Makefile.in	2002-09-14 23:20:55.000000000 +0100
+++ mpb-1.4.2/tests/Makefile.in	2008-11-25 17:31:58.000000000 +0000
@@ -2,12 +2,14 @@
 
 SHELL = @SHELL@
 
-CC = @CC@
-CFLAGS = @CFLAGS@
-CPPFLAGS = -I../src/util -I../src/matrices -I../src/maxwell @CPPFLAGS@
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
+
+CPPFLAGS = -I$(top_srcdir)/src/util -I$(top_srcdir)/src/matrices -I$(top_srcdir)/src/maxwell -I. @CPPFLAGS@
 DEFS = @DEFS@
-LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
+LDFLAGS = @LDFLAGS@
 RANLIB = @RANLIB@
 @SET_MAKE@
 
@@ -15,11 +17,11 @@
          ../src/matrices/matrices.a \
          ../src/util/util.a 
 
-all: blastest eigs_test maxwell_test malloctest normal_vectors
+all: 
 
 libs:
 
-check: all
+check: blastest eigs_test maxwell_test malloctest normal_vectors
 	rm -f blastest.out.test
 	./blastest | sed 's/\-0\.000\([ ,)]\)/ 0.000\1/g' | sed 's/\-0\.000$$/ 0.000/g' > blastest.out.test
 	diff blastest.@SCALAR_TYPE@.out blastest.out.test
diff -Nur mpb-1.4.2.orig/utils/Makefile.in mpb-1.4.2/utils/Makefile.in
--- mpb-1.4.2.orig/utils/Makefile.in	2002-09-14 23:20:55.000000000 +0100
+++ mpb-1.4.2/utils/Makefile.in	2008-11-25 18:23:46.000000000 +0000
@@ -1,13 +1,16 @@
 # Variables substituted by the autoconf configure script:
 
 SHELL = @SHELL@
+mandir = @mandir@
+
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
 
-CC = @CC@
-CFLAGS = @CFLAGS@
-CPPFLAGS = -I../src/util -I../src/matrices -I../src/matrixio @CPPFLAGS@
+CPPFLAGS = -I$(top_srcdir)/src/util -I$(top_srcdir)/src/matrices -I$(top_srcdir)/src/matrixio -I. @CPPFLAGS@
 DEFS = @DEFS@ -DMPB_VERSION='"'"@MPB_VERSION@"'"'
-LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
+LDFLAGS=@LDFLAGS@
 RANLIB = @RANLIB@
 @SET_MAKE@
 
@@ -36,9 +39,9 @@
 
 install-mpb-data: mpb-data
 	$(INSTALL) -d $(prefix)/bin
-	$(INSTALL) -m 0755 -s mpb-data $(prefix)/bin/`echo mpb-data|sed '$(transform)'`
+	$(INSTALL) -m 0755 mpb-data $(prefix)/bin/`echo mpb-data|sed '$(transform)'`
 	$(INSTALL) -d $(mandir)/man1
-	$(INSTALL) -m 0644 mpb-data.1 $(mandir)/man1
+	$(INSTALL) -m 0644 $(srcdir)/mpb-data.1 $(mandir)/man1
 
 clean: 
 	rm -f mpb-data mpb-data.o core
