--- makefile.old	2010-10-08 10:04:46.000000000 +0200
+++ makefile	2010-10-08 10:05:25.000000000 +0200
@@ -995,7 +995,7 @@
 else
 	@echo Linking Raine...
 endif
-	$(LDV) $(LFLAGS) -g -Wall -Wno-write-strings -o $(RAINE_EXE) $(OBJS) $(LIBS) -lstdc++
+	$(LDV) $(LDFLAGS) $(LFLAGS) -g -Wall -Wno-write-strings -o $(RAINE_EXE) $(OBJS) $(LIBS) -lstdc++
 
 converter: source/bonus/converter.c
 	$(CCV) $(CFLAGS) -c $< -o $(OBJDIR)/converter.o
