Bug: https://bugs.gentoo.org/724256
Patch taken from https://github.com/cgdb/cgdb/pull/142
--- a/configure.ac
+++ b/configure.ac
@@ -13,6 +13,13 @@ AC_PROG_INSTALL
 AC_PROG_RANLIB
 AM_PROG_LEX
 AC_PROG_YACC
+m4_ifndef([AC_PROG_AR],[dnl
+  AN_MAKEVAR([AR], [AC_PROG_AR])
+  AN_PROGRAM([ar], [AC_PROG_AR])
+  AC_DEFUN([AC_PROG_AR],
+  [AC_CHECK_TOOL(AR, ar, :)])
+])
+AC_PROG_AR
 
 dnl determine if we are running in linux, cygwin, solaris, or freebsd
 AC_CANONICAL_HOST
