--- config/get_dlld.orig	2011-07-28 14:27:13.341072367 +1200
+++ config/get_dlld	2011-07-28 14:27:46.225178023 +1200
@@ -94,7 +94,7 @@
       *) extra=;;
     esac
     case "$osname" in
-      darwin) shared=-dynamiclib;;
+      darwin) shared="-dynamiclib -install_name \$(PURELIBDIR)/\$(LIBPARI_DYN)";;
       *) shared=-shared;;
     esac
     DLLDFLAGS="$shared $extra \$(CFLAGS) \$(DLCFLAGS) $DLLDFLAGS"
--- config/Makefile.SH.orig	2011-07-28 14:29:06.753681364 +1200
+++ config/Makefile.SH	2011-07-28 14:29:10.589276933 +1200
@@ -234,6 +234,7 @@
 # DESTDIR is used to install to a false hierachy (to build a Debian package)
 INCLUDEDIR= "\$(DESTDIR)$includedir"
 LIBDIR    = "\$(DESTDIR)$libdir"
+PURELIBDIR= $libdir
 BINDIR    = "\$(DESTDIR)$bindir"
 MANDIR    = "\$(DESTDIR)$mandir"
 DATADIR   = "\$(DESTDIR)$datadir"
