diff --git a/configure.ac b/configure.ac
index 633803d..279fad6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -348,7 +348,7 @@ else
     AC_CHECK_LIB(blas,dscal_,
         [blas_use="yes";blas_lib="${blas_lib_path}-lblas"],
         [blas_use="";blas_lib=""],
-        [${blas_lib_path}-lblas -lvf2c])
+        [${blas_lib_path}-lblas])
     if test -z "${blas_use}"; then
         AC_MSG_RESULT([** BLAS library missing ...will BUILD **])
         blas="blas";
@@ -410,7 +410,7 @@ else
     AC_CHECK_LIB(lapack,dsysv_,
         [lapack_use="yes";lapack_lib="${lapack_lib_path}-llapack"],
         [lapack_use="";lapack_lib=""],
-        [${lapack_lib_path}-llapack ${blas_lib} -lvf2c])
+        [${lapack_lib_path}-llapack ${blas_lib}])
     if test -z "${lapack_use}"; then
         AC_MSG_RESULT([** LAPACK library missing ...will BUILD **])
         lapack="lapack";
@@ -472,7 +472,7 @@ else
     AC_CHECK_LIB(superlu,Destroy_SuperMatrix_Store,
         [superlu_use="yes";superlu_lib="${superlu_lib_path}-lsuperlu"],
         [superlu_use="";superlu_lib=""],
-        [${superlu_lib_path}-lsuperlu ${lapack_lib} ${blas_lib} -lvf2c])
+        [${superlu_lib_path}-lsuperlu ${lapack_lib} ${blas_lib}])
     if test -z "${superlu_use}"; then
         AC_MSG_RESULT([** SUPERLU library missing ...will BUILD **])
         superlu="superlu";
@@ -534,7 +534,7 @@ else
     AC_CHECK_LIB(arpack,dsgets_,
         [arpack_use="yes";arpack_lib="${arpack_lib_path}-larpack"],
         [arpack_use="";arpack_lib=""],
-        [${arpack_lib_path}-larpack ${blas_lib} -lvf2c])
+        [${arpack_lib_path}-larpack ${blas_lib}])
     if test -z "${arpack_use}"; then
         AC_MSG_RESULT([** ARPACK library missing ...will BUILD **])
         arpack="arpack";
@@ -658,7 +658,7 @@ else
     AC_CHECK_LIB(pmg,buildg_,
         [pmg_use="yes";pmg_lib="${pmg_lib_path}-lpmg"],
         [pmg_use="";pmg_lib=""],
-        [${pmg_lib_path}-lpmg ${blas_lib} -lvf2c])
+        [${pmg_lib_path}-lpmg ${blas_lib}])
     if test -z "${pmg_use}"; then
         AC_MSG_RESULT([** PMG library missing ...will BUILD **])
         pmg="pmg";
