--- numpy/distutils/system_info.py      2014-09-07 08:57:47.000000000 +0000
+++ numpy/distutils/system_info.py      2014-10-19 17:57:12.099448481 +0000
@@ -296,20 +296,7 @@
       1 - display warning message
       2 - raise error
     """
-    cl = {'atlas': atlas_info,  # use lapack_opt or blas_opt instead
-          'atlas_threads': atlas_threads_info,                # ditto
-          'atlas_blas': atlas_blas_info,
-          'atlas_blas_threads': atlas_blas_threads_info,
-          'lapack_atlas': lapack_atlas_info,  # use lapack_opt instead
-          'lapack_atlas_threads': lapack_atlas_threads_info,  # ditto
-          'mkl': mkl_info,
-          # openblas which may or may not have embedded lapack
-          'openblas': openblas_info,          # use blas_opt instead
-          # openblas with embedded lapack
-          'openblas_lapack': openblas_lapack_info, # use blas_opt instead
-          'lapack_mkl': lapack_mkl_info,      # use lapack_opt instead
-          'blas_mkl': blas_mkl_info,          # use blas_opt instead
-          'x11': x11_info,
+    cl = {'x11': x11_info,
           'fft_opt': fft_opt_info,
           'fftw': fftw_info,
           'fftw2': fftw2_info,

