From 3dd1e0a8ea7db95b76a7b38e27a9b54d6fe75c76 Mon Sep 17 00:00:00 2001
From: Alexander Tsoy <alexander@tsoy.me>
Date: Fri, 5 Jun 2020 17:21:10 +0300
Subject: [PATCH] Do not strip lv2 plugins

---
 trunk/wscript | 1 -
 1 file changed, 1 deletion(-)

diff --git a/wscript b/wscript
index 2699325f..f2d8f86d 100644
--- a/wscript
+++ b/wscript
@@ -375,7 +375,6 @@ def configure(conf):
     conf.env['DESKAPPS_DIR'] = os.path.normpath(os.path.join(conf.env['SHAREDIR'], 'applications'))
     conf.env['BIN_NAME'] = APPNAME
     if opt.lv2:
-        conf.gxload('strip')
         conf.gxload('lv2')
     conf.env['LADSPA'] = opt.ladspa
     conf.env['NEW_LADSPA'] = opt.new_ladspa
-- 
2.26.2

