diff --git a/libexec/atf/Makefile.inc b/libexec/atf/Makefile.inc
index a28f546..9f55937 100644
--- a/libexec/atf/Makefile.inc
+++ b/libexec/atf/Makefile.inc
@@ -27,6 +27,6 @@
 
 ATF = ${.CURDIR}/../../../contrib/atf
 
-CFLAGS+=	-DHAVE_CONFIG_H
+CXXFLAGS+=	-DHAVE_CONFIG_H
 
 WARNS?=		3
diff --git a/libexec/atf/atf-check/Makefile b/libexec/atf/atf-check/Makefile
index fafb1e4..3710403 100644
--- a/libexec/atf/atf-check/Makefile
+++ b/libexec/atf/atf-check/Makefile
@@ -34,7 +34,7 @@ PROG_CXX=	atf-check
 SRCS=		atf-check.cpp
 MAN=		atf-check.1
 
-CFLAGS+=	-I${ATF}
+CXXFLAGS+=	-I${ATF}
 
 DPADD+=		${LIBATF_CXX} ${LIBATF_C}
 
