--- a/doc/examples/arr_script
+++ b/doc/examples/arr_script
@@ -12,8 +12,7 @@
   TOLERANCE=15
   MAX_RETRIES=3
 
-  NETPERF="/usr/local/netperf/netperf"
-  NETPERF="./netperf"
+  NETPERF="/usr/bin/netperf"
 
   NPROC_LIST=""
   RR_SIZES=""
--- a/doc/examples/packet_byte_script
+++ b/doc/examples/packet_byte_script
@@ -25,7 +25,7 @@
 fi
 
 # where is netperf
-NETPERF_DIR=${NETPERF_DIR:=/opt/netperf2/bin}
+NETPERF_DIR=${NETPERF_DIR:=/usr/bin}
 
 
 # at what port will netserver be waiting? If you decide to run
--- a/doc/examples/snapshot_script
+++ b/doc/examples/snapshot_script
@@ -37,7 +37,7 @@
 #
 # where is netperf installed, there are a few possible places:
 
-NETPERF_CMD=${NETPERF_CMD:=/opt/netperf/netperf}
+NETPERF_CMD=${NETPERF_CMD:=/usr/bin/netperf}
 
 
 # there should be no more than two parms passed
