diff --git a/ioflo/aio/udp/test/test_udping.py b/ioflo/aio/udp/test/test_udping.py
index 9aefdf0..ec44721 100644
--- a/ioflo/aio/udp/test/test_udping.py
+++ b/ioflo/aio/udp/test/test_udping.py
@@ -122,6 +122,7 @@ class BasicTestCase(unittest.TestCase):
         shutil.rmtree(tempDirpath)
         console.reinit(verbosity=console.Wordage.concise)
 
+    @unittest.skip("needs network access")
     def testBroadcast(self):
         """
         Test Class SocketUdpNb
diff --git a/ioflo/base/test/test_logging.py b/ioflo/base/test/test_logging.py
index 0b8ce13..b64d469 100644
--- a/ioflo/base/test/test_logging.py
+++ b/ioflo/base/test/test_logging.py
@@ -33,6 +33,7 @@ def tearDownModule():
     console.reinit(verbosity=console.Wordage.concise)
 
 
+@unittest.skip("needs network access")
 class LoggerTestCase(testing.LoggerIofloTestCase):
     """
     Example TestCase
@@ -927,6 +928,7 @@ class LoggerTestCase(testing.LoggerIofloTestCase):
             pass
 
 
+@unittest.skip("needs network access")
 class HouseTestCase(testing.HouseIofloTestCase):
     """
     Example TestCase
