From 47f61ae6b971e71c55788b790146ffe63d64c834 Mon Sep 17 00:00:00 2001
From: Florian Schmaus <flo@geekplace.eu>
Date: Tue, 1 Jun 2021 11:05:05 +0200
Subject: [PATCH] Do not install man page

---
 CMakeLists.txt | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ab8168df336..901de539097e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,15 +28,3 @@ target_link_libraries(filebrowser
 )
 
 install(TARGETS filebrowser DESTINATION ${ROFI_PLUGINS_DIR})
-
-
-
-# Manpage
-
-add_custom_command(OUTPUT "doc/rofi-file-browser-extended.1.gz"
-                   COMMAND gzip -k "doc/rofi-file-browser-extended.1"
-                   COMMENT "Packing manpage")
-
-add_custom_target(manpage ALL DEPENDS "doc/rofi-file-browser-extended.1.gz")
-
-install(FILES "doc/rofi-file-browser-extended.1.gz" DESTINATION "/usr/share/man/man1")
-- 
2.31.1

