diff -ur extb.orig/bin/extb.sh extb/bin/extb.sh
--- extb.orig/bin/extb.sh	2004-05-05 17:38:29.000000000 +0200
+++ extb/bin/extb.sh	2004-07-19 22:19:49.459865904 +0200
@@ -10,17 +10,11 @@
 trap 'rm /tmp/tmp{1,2,3}.$$; exit 0' 2 15
 
 #Defaults
-SCRIPT_PATH="/opt/extb/bin/"
-LINPIC_PATH="/opt/extb/bin/"
-HEX_FILES_PATH="/opt/extb/src/PIC/firm/"
-LIRCD_CMD="lircd"
-IREXEC_CMD="irexec"
+SCRIPT_PATH="/usr/bin/"
+LINPIC_PATH="/usr/bin/"
+HEX_FILES_PATH=/usr/share/extb/"
 SETSERIAL_CMD="setserial"
-LIRC_CONF="/etc/lircd.conf"
-LIRC_RC="/etc/lircrc"
 COM_PORT="/dev/ttyS1"
-VDR_PATH="/VDR/1.2.5/VDR/"
-VDR_CMD="./runvdr"
 
 # -----------------------------------------------------------------------------
 
@@ -329,7 +323,7 @@
 ;;
 
 1)
- cp /tmp/PICflags PICflags.conf
+ cp /tmp/PICflags /etc/extb/PICflags.conf
  _MAIN
 ;;
 
@@ -477,7 +471,8 @@
 0)
  tx.sh M2 # sending M2 to Extension Board to disable video/audio
  clear
- killall irexec; killall lircd
+ /etc/init.d/irexec stop
+ /etc/init.d/lircd stop
  sleep 1
  $SETSERIAL_CMD $COM_PORT uart 16550
  sleep 1
@@ -487,14 +482,15 @@
  echo ""
  $SETSERIAL_CMD $COM_PORT uart none
  sleep 1
- $LIRCD_CMD $LIRC_CONF; $IREXEC_CMD $LIRC_RC &
+ /etc/init.d/lircd start
+ /etc/init.d/irexc start
  echo ""
  echo "Stopping VDR ..."
- killall vdr
+ /etc/init.d/vdr stop
  sleep 1
  echo ""
  echo "Restarting VDR ..."
- cd $VDR_PATH; $VDR_CMD
+ rcvdr start
  cd $SCRIPT_PATH
  echo ""
  # just in case the Extension Board plugin is not installed ...
diff -ur extb.orig/bin/tx.sh extb/bin/tx.sh
--- extb.orig/bin/tx.sh	2004-04-12 18:23:16.000000000 +0200
+++ extb/bin/tx.sh	2004-07-19 22:17:08.373354776 +0200
@@ -1,2 +1,2 @@
 #!/bin/bash
-rc send_once EXTB_TX $1
\ No newline at end of file
+irsend send_once EXTB_TX $1
\ No newline at end of file
