--- mssilk.orig/sdk/Makefile.am	2012-02-16 13:22:51.000000000 +0400
+++ mssilk/sdk/Makefile.am	2012-05-15 20:35:37.374526548 +0400
@@ -2,28 +2,14 @@
 silk_extracted_directory = SILK_SDK_SRC_v$(silk_version)
 silk_src_dir = $(silk_extracted_directory)/SILK_SDK_SRC_$(SILK_FLAVOUR)_v$(silk_version)
 
-silk_zip = $(silk_extracted_directory).zip
-silk_url = http://developer.skype.com/silk/$(silk_zip)
-
-downloaded_files = $(silk_zip)
-checkmarks = $(downloaded_files:%=%.check)
-BUILT_SOURCES = $(downloaded_files) $(checkmarks) $(silk_extracted_directory) $(silk_src_dir) $(silk_extracted_directory)
+BUILT_SOURCES = $(silk_extracted_directory) $(silk_src_dir) $(silk_extracted_directory)
 
 checksum_dir = $(srcdir)/known_checksums
 checksum_files = $(addprefix $(checksum_dir)/, $(downloaded_files:%=%.md5))
 EXTRA_DIST = $(checksum_files)
 
-$(srcdir)/$(silk_zip):
-	$(WGET) $(silk_url) -O $(srcdir)/${silk_zip}
-
-
-# for check, go to $(srcdir) or ./, depending on where file is
-$(checkmarks): %.check: %
-	cd $(<D); $(MD5SUM) -c $(checksum_dir)/$(<F).md5
-	touch $@
-
-$(silk_extracted_directory): $(checkmarks)
-	$(UNZIP) $(srcdir)/$(silk_zip) $(silk_src_dir)/*
+$(silk_extracted_directory):
+	ln -s ../../$(silk_extracted_directory)
 
 $(silk_src_dir): $(silk_extracted_directory)
 	cp $(srcdir)/patch_pic.diff $(silk_src_dir)
@@ -41,6 +27,6 @@
 	cd $(silk_src_dir) && $(MAKE) $(AM_MAKEFLAGS) clean
 
 install-data-local: $(silk_src_dir)
-	cp $(silk_src_dir)/libSKP_SILK_SDK.a $(prefix)/lib
+	echo "stub: nothing to install"
 
 DISTCLEANFILES = $(BUILT_SOURCES)
