--- pkcs11-helper-1.27.0/include/pkcs11-helper-1.0/pkcs11.h.orig	2020-11-17 19:38:56.000000000 +0100
+++ pkcs11-helper-1.27.0/include/pkcs11-helper-1.0/pkcs11.h	2021-06-12 20:53:07.127234688 +0200
@@ -1262,6 +1262,11 @@
 
 #define NULL_PTR NULL
 
+typedef CK_RV (*CK_NSS_GetFIPSStatus)(CK_SESSION_HANDLE hSession,
+                                      CK_OBJECT_HANDLE hObject,
+                                      CK_ULONG ulOperationType,
+                                      CK_ULONG *pulFIPSStatus);
+
 /* Delete the helper macros defined at the top of the file.  */
 #undef ck_flags_t
 #undef ck_version
