--- a/src/dumb/dumb_input.c
+++ b/src/dumb/dumb_input.c
@@ -79,7 +79,7 @@
 /* Read one line, including the newline, into s.  Safely avoids buffer
  * overruns (but that's kind of pointless because there are several
  * other places where I'm not so careful).  */
-static void getline(char *s)
+static void get_line(char *s)
 {
   int c;
   char *p = s;
@@ -203,7 +203,7 @@
       fputs(prompt, stdout);
     else
       dumb_show_prompt(show_cursor, (timeout ? "tTD" : ")>}")[type]);
-    getline(s);
+    get_line(s);
     if ((s[0] != '\\') || ((s[1] != '\0') && !islower(s[1]))) {
       /* Is not a command line.  */
       translate_special_chars(s);
@@ -265,7 +265,7 @@
 	  if (!*current_page)
 	    break;
 	  printf("HELP: Type <return> for more, or q <return> to stop: ");
-	  getline(s);
+	  get_line(s);
 	  if (!strcmp(s, "q\n"))
 	    break;
 	}
