diff -Nur antlr-2.7.7/examples/cpp/inherit.tinyc/Makefile.in antlr-2.7.7_patched/examples/cpp/inherit.tinyc/Makefile.in
--- antlr-2.7.7/examples/cpp/inherit.tinyc/Makefile.in	2006-11-01 23:37:17.000000000 +0200
+++ antlr-2.7.7_patched/examples/cpp/inherit.tinyc/Makefile.in	2009-03-30 02:09:28.000000000 +0300
@@ -143,7 +143,7 @@
 @abs_this_builddir@/examples/cpp/tinyc/TinyCLexer.cpp \
 @abs_this_builddir@/examples/cpp/tinyc/TinyCLexer.hpp \
 	:
-	@ @MAKE@ -C @abs_this_builddir@/examples/cpp/tinyc
+	@ $(MAKE) -C @abs_this_builddir@/examples/cpp/tinyc
 
 $(objdir)/.subc.g : $(srcdir)/subc.g $(objdir)/TinyCParserTokenTypes.txt @ANTLR_JAR@
 	@ANTLRFLAGS="-glib `@CYGPATH_W@ $(_srcdir)/../tinyc/tinyc.g` -o `@CYGPATH_W@ $(thisdir)`" @ANTLR_COMPILE_CMD@ $(srcdir)/subc.g
diff -Nur antlr-2.7.7/examples/java/inherit.tinyc/Makefile.in antlr-2.7.7_patched/examples/java/inherit.tinyc/Makefile.in
--- antlr-2.7.7/examples/java/inherit.tinyc/Makefile.in	2006-11-01 23:37:17.000000000 +0200
+++ antlr-2.7.7_patched/examples/java/inherit.tinyc/Makefile.in	2009-03-30 02:07:25.000000000 +0300
@@ -77,7 +77,7 @@
 	@ @CP@ $< $@
 
 @abs_this_builddir@/examples/java/tinyc/TinyCParserTokenTypes.txt	:
-	@ @MAKE@ -C @abs_this_builddir@/examples/java/tinyc test
+	@ $(MAKE) -C @abs_this_builddir@/examples/java/tinyc test
 
 
 ## How to generated *.java ..
diff -Nur antlr-2.7.7/examples/python/inherit.tinyc/Makefile.in antlr-2.7.7_patched/examples/python/inherit.tinyc/Makefile.in
--- antlr-2.7.7/examples/python/inherit.tinyc/Makefile.in	2006-11-01 23:37:18.000000000 +0200
+++ antlr-2.7.7_patched/examples/python/inherit.tinyc/Makefile.in	2009-03-30 02:09:07.000000000 +0300
@@ -61,11 +61,11 @@
 ## file resides. Therefore we are going to "make" the base
 ## grammar first and then copy required vocabulary over ..
 TinyCTokenTypes.txt : 
-	@MAKE@ -C ../tinyc
+	$(MAKE) -C ../tinyc
 	@CP@ ../tinyc/$@ $@
 
 @abs_this_builddir@/examples/python/tinyc/%.py :
-	@MAKE@ -C ../tinyc
+	$(MAKE) -C ../tinyc
 
 test :: test1
 
diff -Nur antlr-2.7.7/Makefile.in antlr-2.7.7_patched/Makefile.in
--- antlr-2.7.7/Makefile.in	2006-11-01 23:37:18.000000000 +0200
+++ antlr-2.7.7_patched/Makefile.in	2009-03-30 02:05:41.000000000 +0300
@@ -31,7 +31,7 @@
 
 test clean distclean install ::
 	@ if test -f examples/Makefile ; then \
-		@MAKE@ -C examples $@ ; \
+		$(MAKE) -C examples $@ ; \
 	fi
 
 # Rule  to  remove  all objects, cores, ANTLR generated,
diff -Nur antlr-2.7.7/scripts/config.make.in antlr-2.7.7_patched/scripts/config.make.in
--- antlr-2.7.7/scripts/config.make.in	2006-11-01 23:37:18.000000000 +0200
+++ antlr-2.7.7_patched/scripts/config.make.in	2009-03-30 02:04:54.000000000 +0300
@@ -47,9 +47,9 @@
 		*) \
 		if test -f "$${d}/Makefile" ; then \
 		  echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>" ; \
-		  echo ">> @MAKE@ -C $(subdir)/$${d} $@                          " ; \
+		  echo ">> $(MAKE) -C $(subdir)/$${d} $@                          " ; \
 			echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>" ; \
-			@MAKE@ -C "$$d" $@ || exit 1 ;\
+			$(MAKE) -C "$$d" $@ || exit 1 ;\
 		fi ; \
 		;; \
 		esac ; \
