diff --git a/Makefile b/Makefile
index c77d1f9..b970d08 100644
--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,8 @@ endif
 CC := $(CROSS_COMPILE)$(CC)
 LD := $(CROSS_COMPILE)$(LD)
 
-CFLAGS += -Wall -Wextra -g -O2 -I. -Iinclude/ -Wimplicit -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -D__linux__
+CFLAGS ?= -g -O2 -D_FORTIFY_SOURCE=2
+CFLAGS += -Wall -Wextra -I. -Iinclude/ -Wimplicit -D_GNU_SOURCE -D__linux__
 
 CFLAGS += $(shell if $(CC) -std=gnu11 -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-std=gnu11"; else echo "-std=gnu99"; fi)
 
