--- a/src/system.hh.in	2024-05-21 22:50:15.026661018 -0500
+++ b/src/system.hh.in	2024-05-21 22:50:43.466824215 -0500
@@ -144,7 +144,6 @@
 #include <boost/date_time/posix_time/posix_time_io.hpp>
 #include <boost/date_time/gregorian/gregorian_io.hpp>
 
-#include <boost/filesystem/convenience.hpp>
 #include <boost/filesystem/exception.hpp>
 #include <boost/filesystem/fstream.hpp>
 #include <boost/filesystem/operations.hpp>
