--- tnftp-20230507/tnftp.h
+++ tnftp-20230507/tnftp.h
@@ -487,28 +487,7 @@
 #define	TM_YEAR_BASE	1900
 
 #if defined(USE_SOCKS)		/* (Dante) SOCKS5 */
-#define connect		Rconnect
-#define bind		Rbind
-#define getsockname	Rgetsockname
-#define getpeername	Rgetpeername
-#define accept		Raccept
-#define rresvport	Rrresvport
-#define bindresvport	Rbindresvport
-#define gethostbyname	Rgethostbyname
-#define gethostbyname2	Rgethostbyname2
-#define sendto		Rsendto
-#define recvfrom	Rrecvfrom
-#define recvfrom	Rrecvfrom
-#define write		Rwrite
-#define writev		Rwritev
-#define send		Rsend
-#define sendmsg		Rsendmsg
-#define read		Rread
-#define readv		Rreadv
-#define recv		Rrecv
-#define recvmsg		Rrecvmsg
-#define getaddrinfo	Rgetaddrinfo
-#define getipnodebyname	Rgetipnodebyname
+#include <socks.h>
 #endif /* defined(USE_SOCKS) */
 
 
