From c26e9b3882d68fbef471bd67c7ea0eeb5a26c481 Mon Sep 17 00:00:00 2001
From: andriish <andriish@users.noreply.github.com>
Date: Thu, 4 May 2023 19:08:37 +0200
Subject: [PATCH] Update fcn.hpp

---
 src/fcn.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/fcn.hpp b/src/fcn.hpp
index 3e2ff80c..0c53e1be 100644
--- a/src/fcn.hpp
+++ b/src/fcn.hpp
@@ -1,6 +1,7 @@
 #include <Minuit2/FCNGradientBase.h>
 #include <pybind11/pytypes.h>
 #include <vector>
+#include <cstdint>
 
 namespace py = pybind11;
 
