--- a/hydra-irc.c
+++ b/hydra-irc.c
@@ -7,7 +7,6 @@ RFC 1459: Internet Relay Chat Protocol
 */
 
 extern char *HYDRA_EXIT;
-char *buf;
 char buffer[300] = "";
 int32_t myport = PORT_IRC, mysslport = PORT_IRC_SSL;
 
--- a/hydra-rexec.c
+++ b/hydra-rexec.c
@@ -5,7 +5,6 @@
 #define COMMAND "/bin/ls /"
 
 extern char *HYDRA_EXIT;
-char *buf;
 
 int32_t start_rexec(int32_t s, char *ip, int32_t port, unsigned char options, char *miscptr, FILE * fp) {
   char *empty = "";
--- a/hydra-rlogin.c
+++ b/hydra-rlogin.c
@@ -12,7 +12,6 @@ no memleaks found on 110425
 #define TERM "vt100/9600"
 
 extern char *HYDRA_EXIT;
-char *buf;
 
 int32_t start_rlogin(int32_t s, char *ip, int32_t port, unsigned char options, char *miscptr, FILE * fp) {
   char *empty = "";
--- a/hydra-rsh.c
+++ b/hydra-rsh.c
@@ -11,7 +11,6 @@ no memleaks found on 110425
 */
 
 extern char *HYDRA_EXIT;
-char *buf;
 
 int32_t start_rsh(int32_t s, char *ip, int32_t port, unsigned char options, char *miscptr, FILE * fp) {
   char *empty = "";
--- a/hydra-rtsp.c
+++ b/hydra-rtsp.c
@@ -12,7 +12,6 @@
 #include "sasl.h"
 
 extern char *HYDRA_EXIT;
-char *buf;
 char packet[500];
 char packet2[500];
 
--- a/hydra-teamspeak.c
+++ b/hydra-teamspeak.c
@@ -36,7 +36,6 @@ struct team_speak {
 extern int32_t hydra_data_ready_timed(int32_t socket, long sec, long usec);
 
 extern char *HYDRA_EXIT;
-char *buf;
 
 int32_t start_teamspeak(int32_t s, char *ip, int32_t port, unsigned char options, char *miscptr, FILE * fp) {
   char *empty = "";
