diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 41ed77d900..9a0c7d5128 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -40,8 +40,7 @@ if(GINKGO_BUILD_EXTLIB_EXAMPLE)
     list(APPEND EXAMPLES_LIST external-lib-interfacing)
 endif()
 
-find_package(OpenCV QUIET)
-if(OpenCV_FOUND)
+if(FALSE)
     list(APPEND EXAMPLES_LIST heat-equation schroedinger-splitting)
 else()
     message(STATUS "No OpenCV found, disabling examples with video output")
@@ -55,8 +54,7 @@ if(GINKGO_BUILD_MPI)
     list(APPEND EXAMPLES_LIST distributed-solver)
 endif()
 
-find_package(Kokkos QUIET)
-if(Kokkos_FOUND)
+if(FALSE)
     if(GINKGO_WITH_CCACHE)
         message(WARNING "The CMAKE_CXX_COMPILER_LAUNCHER is set due to "
             "GINKGO_WITH_CCACHE=ON which is known to casue issues with CUDA enabled "
