--- a/bindings/python/Makefile.am	2017-07-17 12:03:05.653000000 +0200
+++ b/bindings/python/Makefile.am	2017-09-06 18:51:57.179778549 +0200
@@ -9,35 +9,6 @@
 all-am: python-build
 
 python-build: _prelude.cxx
-if HAVE_PYTHON2
-	CC="$(CXX)" $(PYTHON2) setup.py build
-endif
-if HAVE_PYTHON3
-	CC="$(CXX)" $(PYTHON3) setup.py build
-endif
-install-exec-hook:
-if HAVE_PYTHON2
-	$(PYTHON2) setup.py install `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` --prefix @prefix@
-endif
-if HAVE_PYTHON3
-	$(PYTHON3) setup.py install `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` --prefix @prefix@
-endif
-
-uninstall-hook:
-if HAVE_PYTHON2
-	$(PYTHON2) setup.py uninstall `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` --prefix @prefix@
-endif
-if HAVE_PYTHON3
-	$(PYTHON3) setup.py uninstall `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` --prefix @prefix@
-endif
-
-clean-local:
-if HAVE_PYTHON2
-	$(PYTHON2) setup.py clean -a
-endif
-if HAVE_PYTHON3
-	$(PYTHON3) setup.py clean -a
-endif
 
 _prelude.cxx prelude.py: $(top_srcdir)/bindings/c++/include/*.hxx $(top_srcdir)/bindings/libpreludecpp.i libpreludecpp-python.i 
 if HAVE_SWIG
