diff --git a/device/common/misc.cpp b/device/common/misc.cpp
index 7b3a7fb..612aacd 100644
--- a/device/common/misc.cpp
+++ b/device/common/misc.cpp
@@ -198,7 +198,7 @@ const char *Device::cidrToNetmask(const char *cidr)
 unsigned int Device::ipAddressToNum(const char *address)
 {
 	// Variables...
-	char *addressPointer = 0;
+	const char *addressPointer = 0;
 	unsigned int returnAddress = 0;
 	int tempInt = 0; 
 
