--- viewmol-2.4.1.orig/source/getmachine	2004-08-29 08:04:09.000000000 -0700
+++ viewmol-2.4.1/source/getmachine	2005-12-02 11:19:36.000000000 -0800
@@ -156,7 +158,7 @@
   echo "PNGINCLUDE = $pnginclude" >> .config.$os
 
   # Python
-  list=`find /usr -name Python.h -print 2> /dev/null`
+  list=`find /usr/include -name Python.h -print 2> /dev/null`
   version=`(for i in $list
             do
               dir=\`dirname $i\`
@@ -179,7 +181,7 @@
   echo "PYTHONINCLUDE = $pythoninclude" >> .config.$os
   if [ "$version" != "" ]
   then
-    pythonlib=`dirname \`find /usr -name "lib${version}*.a" -print 2> /dev/null | grep "$version/"\``
+    pythonlib=`dirname \`find /usr/lib -name "lib${version}*.a" -print 2> /dev/null | grep "$version/"\``
   else
     givehint
     echo -n "Please enter the name of the directory where the Python library can be found: "
