diff -Naur curl-7.30.0.orig/curl-config.in curl-7.30.0/curl-config.in
--- curl-7.30.0.orig/curl-config.in	2013-02-06 09:44:37.000000000 -0500
+++ curl-7.30.0/curl-config.in	2013-04-17 18:43:56.000000000 -0400
@@ -134,7 +134,7 @@
         else
           CPPFLAG_CURL_STATICLIB=""
         fi
-        if test "X@includedir@" = "X/usr/include"; then
+        if test "X@includedir@" = "X@GENTOO_PORTAGE_EPREFIX@/usr/include"; then
           echo "$CPPFLAG_CURL_STATICLIB"
         else
           echo "${CPPFLAG_CURL_STATICLIB}-I@includedir@"
@@ -142,7 +142,7 @@
         ;;
 
     --libs)
-        if test "X@libdir@" != "X/usr/lib" -a "X@libdir@" != "X/usr/lib64"; then
+        if test "X@libdir@" != "X@GENTOO_PORTAGE_EPREFIX@/usr/lib" -a "X@libdir@" != "X@GENTOO_PORTAGE_EPREFIX@/usr/lib64"; then
            CURLLIBDIR="-L@libdir@ "
         else
            CURLLIBDIR=""
