diff --git a/cmake/DownloadNlohmannJson.cmake b/cmake/DownloadNlohmannJson.cmake
index 1db953bed..4bbbe35fa 100644
--- a/cmake/DownloadNlohmannJson.cmake
+++ b/cmake/DownloadNlohmannJson.cmake
@@ -37,6 +37,10 @@ function (_download_json_hpp)
             execute_process(COMMAND "${CMAKE_COMMAND}" -E sleep
                                     "${sleep_seconds}")
         endif ()
+        if (EXISTS "${DEST}/json.hpp")
+            set(download_error_code 0)
+            break()
+        endif ()
         file(DOWNLOAD "${JSON_URL}" "${DEST}/json.hpp" STATUS download_status)
         list(GET download_status 0 download_error_code)
         if (download_error_code EQUAL 0)
