diff --git a/CMakeLists.txt b/CMakeLists.txt
index d84f002..a35b662 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,12 +79,12 @@ OPTION(ALEMBIC_LIB_USES_TR1
 
 # Set static/dynamic build options
 SET(LIB_TYPE STATIC)
-SET(RUNTIME_INSTALL_DIR lib)
-SET(LIBRARY_INSTALL_DIR lib)
-SET(ARCHIVE_INSTALL_DIR lib)
+SET(RUNTIME_INSTALL_DIR lib64)
+SET(LIBRARY_INSTALL_DIR lib64)
+SET(ARCHIVE_INSTALL_DIR lib64)
 IF (ALEMBIC_SHARED_LIBS)
     SET(LIB_TYPE SHARED)
-    SET(ARCHIVE_INSTALL_DIR lib)
+    SET(ARCHIVE_INSTALL_DIR lib64)
     IF (WIN32)
         ADD_DEFINITIONS(-DALEMBIC_DLL)
     ENDIF()
diff --git a/bin/AbcConvert/CMakeLists.txt b/bin/AbcConvert/CMakeLists.txt
index 3725ed7..edea767 100644
--- a/bin/AbcConvert/CMakeLists.txt
+++ b/bin/AbcConvert/CMakeLists.txt
@@ -38,6 +38,6 @@ TARGET_LINK_LIBRARIES(abcconvert Alembic)
 
 set_target_properties(abcconvert PROPERTIES
     INSTALL_RPATH_USE_LINK_PATH TRUE
-    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib64)
 
 INSTALL(TARGETS abcconvert DESTINATION bin)
diff --git a/bin/AbcDiff/CMakeLists.txt b/bin/AbcDiff/CMakeLists.txt
index e2fd355..2234650 100644
--- a/bin/AbcDiff/CMakeLists.txt
+++ b/bin/AbcDiff/CMakeLists.txt
@@ -38,6 +38,6 @@ TARGET_LINK_LIBRARIES(abcdiff Alembic)
 
 set_target_properties(abcdiff PROPERTIES
     INSTALL_RPATH_USE_LINK_PATH TRUE
-    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib64)
 
 INSTALL(TARGETS abcdiff DESTINATION bin)
diff --git a/bin/AbcEcho/CMakeLists.txt b/bin/AbcEcho/CMakeLists.txt
index 396226e..116c122 100644
--- a/bin/AbcEcho/CMakeLists.txt
+++ b/bin/AbcEcho/CMakeLists.txt
@@ -41,6 +41,6 @@ TARGET_LINK_LIBRARIES(abcechobounds Alembic)
 
 set_target_properties(abcecho abcechobounds PROPERTIES
     INSTALL_RPATH_USE_LINK_PATH TRUE
-    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib64)
 
 INSTALL(TARGETS abcecho abcechobounds DESTINATION bin)
diff --git a/bin/AbcLs/CMakeLists.txt b/bin/AbcLs/CMakeLists.txt
index 1dbeb9d..7b721d0 100644
--- a/bin/AbcLs/CMakeLists.txt
+++ b/bin/AbcLs/CMakeLists.txt
@@ -39,6 +39,6 @@ TARGET_LINK_LIBRARIES(abcls Alembic)
 
 set_target_properties(abcls PROPERTIES
     INSTALL_RPATH_USE_LINK_PATH TRUE
-    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib64)
 
 INSTALL(TARGETS abcls DESTINATION bin)
diff --git a/bin/AbcStitcher/CMakeLists.txt b/bin/AbcStitcher/CMakeLists.txt
index 3106bf2..0c8a205 100644
--- a/bin/AbcStitcher/CMakeLists.txt
+++ b/bin/AbcStitcher/CMakeLists.txt
@@ -38,6 +38,6 @@ TARGET_LINK_LIBRARIES(abcstitcher Alembic)
 
 set_target_properties(abcstitcher PROPERTIES
     INSTALL_RPATH_USE_LINK_PATH TRUE
-    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib64)
 
 INSTALL(TARGETS abcstitcher DESTINATION bin)
diff --git a/bin/AbcTree/CMakeLists.txt b/bin/AbcTree/CMakeLists.txt
index 3c41004..de7c7fc 100644
--- a/bin/AbcTree/CMakeLists.txt
+++ b/bin/AbcTree/CMakeLists.txt
@@ -38,6 +38,6 @@ TARGET_LINK_LIBRARIES(abctree Alembic)
 
 set_target_properties(abctree PROPERTIES
     INSTALL_RPATH_USE_LINK_PATH TRUE
-    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+    INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib64)
 
 INSTALL(TARGETS abctree DESTINATION bin)
diff --git a/lib/Alembic/CMakeLists.txt b/lib/Alembic/CMakeLists.txt
index 3e16518..b07cb7d 100644
--- a/lib/Alembic/CMakeLists.txt
+++ b/lib/Alembic/CMakeLists.txt
@@ -86,9 +86,9 @@ ENDIF()
 
 INSTALL(TARGETS Alembic
         EXPORT AlembicTargets
-        LIBRARY DESTINATION lib
-        ARCHIVE DESTINATION lib
-        RUNTIME DESTINATION lib)
+        LIBRARY DESTINATION lib64
+        ARCHIVE DESTINATION lib64
+        RUNTIME DESTINATION lib64)
 
 #-******************************************************************************
 # PACKAGE EXPORTS
diff --git a/python/PyAlembic/CMakeLists.txt b/python/PyAlembic/CMakeLists.txt
index 06daf30..254ca2b 100644
--- a/python/PyAlembic/CMakeLists.txt
+++ b/python/PyAlembic/CMakeLists.txt
@@ -128,7 +128,7 @@ IF (Boost_PYTHON_LIBRARY AND ALEMBIC_PYTHON_LIBRARY)
     )
 
     INSTALL (TARGETS alembic
-        DESTINATION lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages
+        DESTINATION lib64/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages
     )
 
     IF (USE_TESTS)
@@ -137,7 +137,7 @@ IF (Boost_PYTHON_LIBRARY AND ALEMBIC_PYTHON_LIBRARY)
 
     set_target_properties(alembic PROPERTIES
         INSTALL_RPATH_USE_LINK_PATH TRUE
-        INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+        INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib64)
 ELSE()
     MESSAGE(FATAL_ERROR "Unable to find Python libs")
 ENDIF()
