--- a/tools/common.jam
+++ b/tools/common.jam
@@ -763,7 +763,15 @@
         switch $(f:G)
         {
             case <base> :
-                result += $(name:B) ;
+		local matched = [ MATCH "^(boost.*python)-.*" : $(name) ] ;
+		if $(matched) = boost_python || $(matched) = boost_mpi_python
+		{
+			result += $(name) ;
+		}
+		else
+		{
+			result += $(name:B) ;
+		}
 
             case <toolset> :
                 result += [ join-tag $(f:G=) : [ toolset-tag $(name) : $(type) :
