--- a/src/rxvtlib.h
+++ b/src/rxvtlib.h
@@ -790,7 +790,7 @@ typedef struct {
 } action_t;
 
 /* Values for macro_t.type. Must sync this with macroNames from macros.c */
-enum {
+enum macroFnNames {
     MacroFnDummy=0,
     MacroFnEsc,
     MacroFnStr,
@@ -820,7 +820,7 @@ enum {
     MacroFnSaveConfig,
     MacroFnToggleMacros,
     NMACRO_FUNCS
-} macroFnNames;
+};
 
 typedef unsigned char macro_priority_t;
 typedef struct {
