--- ./qtiplot/src/analysis/FFT.cpp.orig	2011-07-06 11:51:08.000000000 +0000
+++ ./qtiplot/src/analysis/FFT.cpp	2013-10-23 23:07:45.674925476 +0000
@@ -353,7 +353,7 @@
 	ml->arrangeLayers(false, false);
 }
 
-bool FFT::setDataFromTable(Table *t, const QString& realColName, const QString& imagColName, int from, int to)
+bool FFT::setDataFromTable(Table *t, const QString& realColName, const QString& imagColName, int from, int to, bool unused)
 {
 	d_init_err = true;
 
--- ./qtiplot/src/analysis/FFT.h.orig	2011-07-05 22:05:03.000000000 +0000
+++ ./qtiplot/src/analysis/FFT.h	2013-10-23 23:11:34.277924367 +0000
@@ -52,6 +52,8 @@
 	Matrix *realOutputMatrix(){return d_re_out_matrix;}
 	Matrix *imaginaryOutputMatrix(){return d_im_out_matrix;}
 
+    bool setDataFromTable(Table *t, const QString& realColName, const QString& imagColName = QString(), int from = 0, int to = -1, bool unused = false);
+
 private:
     void init();
     void output();
@@ -60,8 +62,6 @@
 	void fftTable();
 	void fftMatrix();
 
-    bool setDataFromTable(Table *t, const QString& realColName, const QString& imagColName = QString(), int from = 0, int to = -1);
-
     double d_sampling;
     //! Flag telling if an inverse FFT must be performed.
     bool d_inverse;
