diff --git a/3rdparty/qwt/qwtconfig.pri b/3rdparty/qwt/qwtconfig.pri
index b7d0d0e..e36f131 100755
--- a/3rdparty/qwt/qwtconfig.pri
+++ b/3rdparty/qwt/qwtconfig.pri
@@ -7,17 +7,6 @@ VER_MIN      = 2
 VER_PAT      = 0
 VERSION      = $${VER_MAJ}.$${VER_MIN}.$${VER_PAT}
 
-unix {
-    INSTALLBASE    = /usr/local/qwt-5.2.0
-}
-
-win32 {
-    INSTALLBASE    = C:/Qwt-5.2.0
-}
-
-target.path    = $$INSTALLBASE/lib
-headers.path   = $$INSTALLBASE/include
-doc.path       = $$INSTALLBASE/doc
 
 ######################################################################
 # qmake internal options
