unchanged:
--- a/Rakefile
+++ b/Rakefile
@@ -86,7 +86,7 @@
   s.rdoc_options      = %w(--title EventMachine --main README --line-numbers -x lib/em/version -x lib/emva -x lib/evma/ -x lib/pr_eventmachine -x lib/jeventmachine)
   s.extra_rdoc_files  = Dir['README,docs/*']
 
-  s.files             = `git ls-files`.split("\n")
+  s.files             = `git ls-files`.split("\n") rescue Errno::ENOENT
 
   s.require_path      = 'lib'
 
@@ -301,7 +301,7 @@ rescue LoadError
   require 'rake/rdoctask'
   Rake::RDocTask
 end
-df = begin; require 'rdoc/generator/darkfish'; true; rescue LoadError; end
+df = begin; require 'rdoc/rdoc'; require 'rdoc/generator/darkfish'; true; rescue LoadError; end
 rdtask = rdoc_task_type.new do |rd|
   rd.title = Spec.name
   rd.rdoc_dir = 'rdoc'
unchanged:
--- a/tests/test_defer.rb
+++ b/tests/test_defer.rb
@@ -28,6 +28,7 @@ $:.unshift "../lib"
 require 'eventmachine'
 require 'test/unit'
 
+unless RUBY_VERSION >= '1.9.0'
 class TestDeferUsage < Test::Unit::TestCase
 
   def test_defers
@@ -41,7 +42,8 @@ class TestDeferUsage < Test::Unit::TestCase
       }
     }
     assert_equal( n, n_times )
-  end unless RUBY_VERSION >= '1.9.0'
+  end
 
 end
+end
 
only in patch2:
unchanged:
--- a/tests/test_process_watch.rb
+++ b/tests/test_process_watch.rb
@@ -2,6 +2,7 @@ $:.unshift "../lib"
 require 'eventmachine'
 require 'test/unit'
 
+if EM.kqueue?
 class TestProcessWatch < Test::Unit::TestCase
   module ParentProcessWatcher
     def process_forked
@@ -45,4 +46,5 @@ class TestProcessWatch < Test::Unit::TestCase
     assert($exited)
     assert($unbind)
   end
-end
\ No newline at end of file
+end
+end
