--- src/editorsave.cpp.old
+++ src/editorsave.cpp
@@ -642,7 +642,8 @@
     file << "<?xml version=\"1.0\" encoding=\"ISO-8859-1\"?>\n";
     file << "<pack>\n<name>"<< stripXML(editorboard::filetitle) <<"</name><author>"<< stripXML(editorboard::author) <<"</author>\n<description>"<< stripXML(editorboard::description) <<"</description>\n";
     
-    for (int i=first; i<editorboard::CountLevels(); i++) {
+    int i;
+    for (i=first; i<editorboard::CountLevels(); i++) {
         editorboard::LoadLevelNumber(board, i);
         file << "\n<level>\n";
         file << "<title>"<< stripXML(board->title) <<"</title>\n";
--- src/gen.cpp.old
+++ src/gen.cpp
@@ -21,6 +21,7 @@
 #include<cstdio>
 #include<iostream>
 #include<fstream>
+#include <unistd.h>
 
 const float F_RAND_MAX = (float)(RAND_MAX);
 
