--- doxygen-1.4.6.orig/debian/patches/qstring-gcc40.dpatch
+++ doxygen-1.4.6/debian/patches/qstring-gcc40.dpatch
@@ -0,0 +1,41 @@
+#! /bin/sh -e
+
+# DP: Remove unused code that does not build with gcc 4.0 (#338099).
+
+dir=
+if [ $# -eq 3 -a "$2" = '-d' ]; then
+    pdir="-d $3"
+    dir="$3/"
+elif [ $# -ne 1 ]; then
+    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
+    exit 1
+fi
+case "$1" in
+    -patch)
+        patch $pdir -f --no-backup-if-mismatch -p0 < $0
+        ;;
+    -unpatch)
+        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
+        ;;
+    *)
+	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
+        exit 1
+esac
+exit 0
+
+--- qtools/qstring.h.orig	2004-12-01 19:26:16.000000000 +0000
++++ qtools/qstring.h	2005-12-04 20:09:01.000000000 +0000
+@@ -163,8 +163,13 @@
+     bool isLetterOrNumber() const;
+     bool isDigit() const;
+ 
++    // the non-const ::cell and ::row cannot be compiled with gcc 4.0 on ARM,
++    // so they have been removed; the methods are not needed for doxygen
++#if 0
+     uchar& cell() { return cl; }
+     uchar& row() { return rw; }
++#endif
++
+     uchar cell() const { return cl; }
+     uchar row() const { return rw; }
+ 
