Index: gogoc-1_2-RELEASE/gogoc-messaging/src/clientmsgsender.cc
===================================================================
--- gogoc-1_2-RELEASE.orig/gogoc-messaging/src/clientmsgsender.cc
+++ gogoc-1_2-RELEASE/gogoc-messaging/src/clientmsgsender.cc
@@ -15,7 +15,7 @@
 // **************************************************************************
 #include <gogocmessaging/clientmsgsender.h>
 #include <assert.h>
-
+#include <cstddef>
 
 namespace gogocmessaging
 {
Index: gogoc-1_2-RELEASE/gogoc-messaging/src/servermsgsender.cc
===================================================================
--- gogoc-1_2-RELEASE.orig/gogoc-messaging/src/servermsgsender.cc
+++ gogoc-1_2-RELEASE/gogoc-messaging/src/servermsgsender.cc
@@ -15,7 +15,7 @@
 // **************************************************************************
 #include <gogocmessaging/servermsgsender.h>
 #include <assert.h>
-
+#include <cstddef>
 
 namespace gogocmessaging
 {
