diff --git a/yesod-core/Yesod/Core/Types.hs b/yesod-core/Yesod/Core/Types.hs
index d8a2b2e..cec2c11 100644
--- a/Yesod/Core/Types.hs
+++ b/Yesod/Core/Types.hs
@@ -49,7 +49,7 @@ import qualified Network.Wai                        as W
 import qualified Network.Wai.Parse                  as NWP
 import           System.Log.FastLogger              (LogStr, LoggerSet, toLogStr, pushLogStr)
 import qualified System.Random.MWC                  as MWC
-import           Network.Wai.Logger                 (DateCacheGetter)
+import           Network.Wai.Logger
 import           Text.Blaze.Html                    (Html, toHtml)
 import           Text.Hamlet                        (HtmlUrl)
 import           Text.Julius                        (JavascriptUrl)
@@ -553,7 +553,7 @@ instance ParseRoute WaiSubsite where
 
 data Logger = Logger
     { loggerSet :: !LoggerSet
-    , loggerDate :: !DateCacheGetter
+    , loggerDate :: !(IO ZonedDate)
     }
 
 loggerPutStr :: Logger -> LogStr -> IO ()
