diff -Naurp libapreq2-2.08.old/build/doxygen.conf libapreq2-2.08/build/doxygen.conf
--- libapreq2-2.08.old/build/doxygen.conf	2006-08-09 00:30:19.000000000 -0400
+++ libapreq2-2.08/build/doxygen.conf	2007-05-05 07:54:21.000000000 -0400
@@ -27,7 +27,7 @@ WARN_IF_UNDOCUMENTED   = YES
 WARN_IF_DOC_ERROR      = YES
 WARN_FORMAT            = "$file:$line: $text"
 
-INPUT                  = . include module/apache module/apache2
+INPUT                  = . include module/apache2
 FILE_PATTERNS          = *.h *.dox CHANGES STATUS
 RECURSIVE              = NO
 EXTRACT_STATIC         = YES
diff -Naurp libapreq2-2.08.old/build/doxygen.conf.in libapreq2-2.08/build/doxygen.conf.in
--- libapreq2-2.08.old/build/doxygen.conf.in	2006-08-09 00:26:44.000000000 -0400
+++ libapreq2-2.08/build/doxygen.conf.in	2007-05-05 07:57:18.000000000 -0400
@@ -27,7 +27,7 @@ WARN_IF_UNDOCUMENTED   = YES
 WARN_IF_DOC_ERROR      = YES
 WARN_FORMAT            = "$file:$line: $text"
 
-INPUT                  = . include module/apache module/apache2
+INPUT                  = . include module/apache2
 FILE_PATTERNS          = *.h *.dox CHANGES STATUS
 RECURSIVE              = NO
 EXTRACT_STATIC         = YES
