--- gif2apng-1.9-src/Makefile
+++ gif2apng-1.9-src/Makefile
@@ -1,11 +1,9 @@
 PACKAGE    = gif2apng
-CC         = gcc
-SRC_DIRS   = . 7z zopfli
-CFLAGS     = -Wall -pedantic
-CFLAGS_OPT = -O2
+CC         ?= gcc
+SRC_DIRS   = . 7z
+CFLAGS     += -Wall -pedantic
 CFLAGS_7Z  = -Wno-sign-compare -Wno-reorder -Wno-maybe-uninitialized -Wno-parentheses
-LDFLAGS    = -s
-LIBS       = -lstdc++ -lm -lz
+LIBS       = -lstdc++ -lm -lz -lzopfli
 
 INCUDE_DIRS := $(addprefix -I./, $(SRC_DIRS))
 OBJ_DIRS := $(addprefix obj/, $(SRC_DIRS))
