diff -NrU5 chillispot-1.1.0.original/src/tun.c chillispot-1.1.0/src/tun.c
--- chillispot-1.1.0.original/src/tun.c	2009-07-18 20:50:38.000000000 +0000
+++ chillispot-1.1.0/src/tun.c	2009-07-18 21:06:53.000000000 +0000
@@ -364,11 +364,11 @@
   tun_sifflags(this, IFF_UP | IFF_RUNNING); /* TODO */
   close(fd);
   this->addrs++;
   return 0;
 
-#elif defined (__FreeBSD__) defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
+#elif defined (__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
 
   int fd;
   struct ifaliasreq      areq;
 
   /* TODO: Is this needed on FreeBSD? */
@@ -443,11 +443,11 @@
   ifr.ifr_dstaddr.sa_family = AF_INET;
 
 #if defined(__linux__)
   ifr.ifr_netmask.sa_family = AF_INET;
 
-#elif defined(__FreeBSD__) defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
+#elif defined(__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
   ((struct sockaddr_in *) &ifr.ifr_addr)->sin_len = 
     sizeof (struct sockaddr_in);
   ((struct sockaddr_in *) &ifr.ifr_dstaddr)->sin_len = 
     sizeof (struct sockaddr_in);
 #endif
@@ -495,11 +495,11 @@
     this->netmask.s_addr = netmask->s_addr;
 #if defined(__linux__)
     ((struct sockaddr_in *) &ifr.ifr_netmask)->sin_addr.s_addr = 
       netmask->s_addr;
 
-#elif defined(__FreeBSD__) defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
+#elif defined(__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
     ((struct sockaddr_in *) &ifr.ifr_addr)->sin_addr.s_addr = 
       netmask->s_addr;
 
 #elif defined(__sun__)
     ((struct sockaddr_in *) &ifr.ifr_addr)->sin_addr.s_addr = 
@@ -583,11 +583,11 @@
     }
   }
   close(fd);
   return 0;
   
-#elif defined(__FreeBSD__) defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
+#elif defined(__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
 
 struct {
   struct rt_msghdr rt;
   struct sockaddr_in dst;
   struct sockaddr_in gate;
@@ -672,11 +672,11 @@
 {
 
 #if defined(__linux__)
   struct ifreq ifr;
 
-#elif defined(__FreeBSD__) defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
+#elif defined(__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
   char devname[IFNAMSIZ+5]; /* "/dev/" + ifname */
   int devnum;
   struct ifaliasreq areq;
   int fd;
 
@@ -720,11 +720,11 @@
   (*tun)->devname[IFNAMSIZ] = 0;
   
   ioctl((*tun)->fd, TUNSETNOCSUM, 1); /* Disable checksums */
   return 0;
   
-#elif defined(__FreeBSD__) defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
+#elif defined(__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__) || defined (__APPLE__)
 
   /* Find suitable device */
   for (devnum = 0; devnum < 255; devnum++) { /* TODO 255 */ 
     snprintf(devname, sizeof(devname), "/dev/tun%d", devnum);
     devname[sizeof(devname)] = 0;
