diff --git a/Lib/test/test___all__.py b/Lib/test/test___all__.py
index c077881511..0728bef6c4 100644
--- a/Lib/test/test___all__.py
+++ b/Lib/test/test___all__.py
@@ -71,6 +71,8 @@ class AllTest(unittest.TestCase):
         blacklist = set([
             # Will raise a SyntaxError when compiling the exec statement
             '__future__',
+            # imports numpy which causes warnings
+            'test.test_pickletools',
         ])
 
         if not sys.platform.startswith('java'):
