https://sourceforge.net/p/ddrescue/tickets/3/

--- a/Makefile
+++ b/Makefile
@@ -500,11 +500,11 @@ check: $(TARGETS) find_nonzero md5 sha1 sha256 sha512 fmt_no
 	# Extra xattrs (should be preserved)
 	#make check_xattr_copy
 	# Tests with HMAC
-	echo -n "what do ya want for nothing?" > TEST
-	echo "750c783e6ab0b503eaa86e310a5db738 *TEST" > HMACS.md5
-	$(VG) ./dd_rescue -L ./libddr_hash.so=md5:hmacpwd=Jefe:chknm= TEST /dev/null
-	rm -f /tmp/dd_rescue CHECKSUMS.sha512 TEST HMACS.md5
-	if ./calchmac.py sha1 pass dd_rescue; then $(MAKE) check_hmac; else echo "Sorry, no more HMAC test due to missing python-hashlib support"; true; fi
+	#echo -n "what do ya want for nothing?" > TEST
+	#echo "750c783e6ab0b503eaa86e310a5db738 *TEST" > HMACS.md5
+	#$(VG) ./dd_rescue -L ./libddr_hash.so=md5:hmacpwd=Jefe:chknm= TEST /dev/null
+	#rm -f /tmp/dd_rescue CHECKSUMS.sha512 TEST HMACS.md5
+	#if ./calchmac.py sha1 pass dd_rescue; then $(MAKE) check_hmac; else echo "Sorry, no more HMAC test due to missing python-hashlib support"; true; fi
 	$(MAKE) check_fault
 	#$(MAKE) check_aes
 	$(MAKE) check_crypt
