--- a/configure.ac
+++ b/configure.ac
@@ -38,6 +38,7 @@
 AC_PATH_PROGS(SED, sed gsed)
 AX_PATH_QMAKE4
 AX_PATH_QMAKE5
+PKG_PROG_PKG_CONFIG
 
 AM_MAINTAINER_MODE
 
@@ -1618,7 +1619,7 @@
   fi
   AX_LANG_WNOERROR  # end AC_LANG_WERROR
 
-  QT_CONFIG_OPTS="debug"
+  QT_CONFIG_OPTS="debug nostrip"
 else
   CFLAGS="$CFLAGS -DNDEBUG"
   CXXFLAGS="$CXXFLAGS -DNDEBUG"
@@ -1629,7 +1630,7 @@
   AX_CFLAGS_GCC_OPTION([-Wno-unused-function], [UIM_SCM_CFLAGS])
   AX_LANG_WNOERROR  # end AC_LANG_WERROR
 
-  QT_CONFIG_OPTS="release"
+  QT_CONFIG_OPTS="release nostrip"
 fi
 
 AC_ARG_ENABLE(warnings-into-error,
