From 503fb876bd5664b455d234ba21294735a8972c91 Mon Sep 17 00:00:00 2001
From: Ivan Brezina <ibrezina@ibrezina.net>
Date: Wed, 5 Jul 2017 10:07:14 +0200
Subject: [PATCH] linux fix

---
 src/editor/tosqltext.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/editor/tosqltext.cpp b/src/editor/tosqltext.cpp
index c329daf0..60935c12 100644
--- a/src/editor/tosqltext.cpp
+++ b/src/editor/tosqltext.cpp
@@ -50,6 +50,7 @@
 #include <QToolTip>
 #endif
 
+#include <QMenu>
 #include <QListWidget>
 #include <QVBoxLayout>
 #include <QApplication>
