diff --git a/src/shared/Makefile.in b/src/shared/Makefile.in
index 5d93e82..db14156 100644
--- a/src/shared/Makefile.in
+++ b/src/shared/Makefile.in
@@ -18,7 +18,7 @@ SHLIB_LD_FLAGS = @SHLIB_LD_FLAGS@ @LD_RUN_PATH@
 SHLIB_TCL_ONLY_LIB_SPECS = @SHLIB_TCL_ONLY_LIB_SPECS@ 
 SHLIB_LIB_SPECS = @SHLIB_LIB_SPECS@ 
 SHLIB_LD =	@SHLIB_LD@
-LDFLAGS =	@LDFLAGS@ @LD_RUN_PATH@
+LDFLAGS +=	@LDFLAGS@
 
 prefix =	@prefix@
 exec_prefix =	@exec_prefix@
@@ -144,14 +144,14 @@ $(lib_so): $(OBJS)
 	$(CC) -c $(CC_SWITCHES) -DBLT_LIBRARY=\"$(scriptdir)\" \
 		$(srcdir)/bltInit.c
 	$(RM) $@
-	$(SHLIB_LD) $(SHLIB_LD_FLAGS) -o $@ bltInit.o $(OBJS) \
+	$(SHLIB_LD) $(SHLIB_LD_FLAGS) $(LDFLAGS) -o $@ bltInit.o $(OBJS) \
 		$(SHLIB_LIB_SPECS)
 
 $(tcl_only_lib_so): $(TCL_ONLY_OBJS)
 	$(CC) -c $(CC_SWITCHES) -DTCL_ONLY -DBLT_LIBRARY=\"$(scriptdir)\" \
 		$(srcdir)/bltInit.c
 	$(RM) $@
-	$(SHLIB_LD) $(SHLIB_LD_FLAGS) -o $@ bltInit.o $(TCL_ONLY_OBJS) \
+	$(SHLIB_LD) $(SHLIB_LD_FLAGS) $(LDFLAGS) -o $@ bltInit.o $(TCL_ONLY_OBJS) \
 		$(SHLIB_TCL_ONLY_LIB_SPECS)
 
 install: mkdirs install-lib install-demo
