Index: source/texk/web2c/lib/setupvar.c
===================================================================
--- source/texk/web2c/lib/setupvar.c	(revision 21623)
+++ source/texk/web2c/lib/setupvar.c	(revision 21624)
@@ -23,7 +23,7 @@
     if (conf_val < 0 || (conf_val == 0 && dflt > 0)) {
       fprintf (stderr,
                "%s: Bad value (%ld) in texmf.cnf for %s, keeping %ld.\n",
-               program_invocation_name,
+               kpse_invocation_name,
                (long) conf_val, var_name, (long) dflt);
     } else {
       *var = conf_val; /* We'll make further checks later.  */
