diff --git a/geopy/geocoders/placefinder.py b/geopy/geocoders/placefinder.py
index 97950fe..814dbaf 100644
--- a/geopy/geocoders/placefinder.py
+++ b/geopy/geocoders/placefinder.py
@@ -71,7 +71,6 @@ def __init__(
             client_secret=self.consumer_secret,
             signature_method=u"HMAC-SHA1",
             signature_type=u"AUTH_HEADER",
-            encoding=None # already UTF-8
         )
         self.api = "https://yboss.yahooapis.com/geo/placefinder"
 
