--- a/sysrqd.c
+++ b/sysrqd.c
@@ -40,8 +40,8 @@
 #define BIND_MAX_LEN 16
 #define PROMPT "sysrq> "
 #define SYSRQ_TRIGGER_PATH "/proc/sysrq-trigger"
-#define AUTH_FILE "/etc/sysrqd.secret"
-#define BINDIP_FILE "/etc/sysrqd.bind"
+#define AUTH_FILE "/etc/sysrqd/sysrqd.secret"
+#define BINDIP_FILE "/etc/sysrqd/sysrqd.bind"
 #define PID_FILE "/var/run/sysrqd.pid"
 #define SYSRQD_PRIO -19
 #define SYSRQD_LISTEN_PORT 4094
