--- Sources/Makefile.old	2012-11-09 08:04:43.029680861 +0100
+++ Sources/Makefile	2012-11-09 08:16:02.650491943 +0100
@@ -35,10 +35,6 @@
 export
 VERSION=0_7_2
 
-CC=g++
-LINK=g++
-MAKE=make
-
 SRC= KrystalDrop/main.cpp                              \
      KrystalDrop/Names.cpp                             \
      KrystalDrop/Controller/BackgroundController.cpp   \
@@ -159,7 +155,7 @@
   CCFLAGS:=$(CCFLAGS) -ggdb -DDEBUG -DDEBUG_SANITY_CHECK
 endif
 ifeq ($(DEBUG_MODE),no)
-  CCFLAGS:=$(CCFLAGS) -O2 -DNDEBUG
+  CCFLAGS:=$(CCFLAGS) -DNDEBUG
 endif
 ifeq ($(DISPLAY_FPS),yes)
   CCFLAGS:=$(CCFLAGS) -DDISPLAY_FPS
@@ -186,11 +182,11 @@
 	@$(MAKE) -s --no-print-directory -f Makefile.dep $@
 
 kdrop: $(OBJ)
-	$(LINK) $(LDFLAGS) $(LCFLAGS) -o $@ $(OBJ) $(LIBS)
+	$(CXX) $(LDFLAGS) $(LCFLAGS) -o $@ $(OBJ) $(LIBS)
 
 $(OBJ): %.o: dep/%.d
 $(OBJ): %.o: %.cpp
-	$(CC) $(CCFLAGS) -o $@ -c $<
+	$(CXX) $(CCFLAGS) $(CXXFLAGS) -o $@ -c $<
 
 clean:
 	find -name "*.o" -exec rm {} ";"
