diff -Naur a/tutorials/far/tutorial_0/CMakeLists.txt b/tutorials/far/tutorial_0/CMakeLists.txt
--- a/tutorials/far/tutorial_0/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/far/tutorial_0/CMakeLists.txt	2020-05-13 16:06:40.857855388 +1000
@@ -33,5 +33,5 @@
     $<TARGET_OBJECTS:far_obj>
 )
 
-install(TARGETS far_tutorial_0 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS far_tutorial_0 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/far/tutorial_1/CMakeLists.txt b/tutorials/far/tutorial_1/CMakeLists.txt
--- a/tutorials/far/tutorial_1/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/far/tutorial_1/CMakeLists.txt	2020-05-13 16:06:53.137874901 +1000
@@ -33,5 +33,5 @@
     $<TARGET_OBJECTS:far_obj>
 )
 
-install(TARGETS far_tutorial_1 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS far_tutorial_1 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/far/tutorial_2/CMakeLists.txt b/tutorials/far/tutorial_2/CMakeLists.txt
--- a/tutorials/far/tutorial_2/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/far/tutorial_2/CMakeLists.txt	2020-05-13 16:07:04.297892363 +1000
@@ -33,5 +33,5 @@
     $<TARGET_OBJECTS:far_obj>
 )
 
-install(TARGETS far_tutorial_2 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS far_tutorial_2 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/far/tutorial_3/CMakeLists.txt b/tutorials/far/tutorial_3/CMakeLists.txt
--- a/tutorials/far/tutorial_3/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/far/tutorial_3/CMakeLists.txt	2020-05-13 16:07:20.377917078 +1000
@@ -33,5 +33,5 @@
     $<TARGET_OBJECTS:far_obj>
 )
 
-install(TARGETS far_tutorial_3 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS far_tutorial_3 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/far/tutorial_4/CMakeLists.txt b/tutorials/far/tutorial_4/CMakeLists.txt
--- a/tutorials/far/tutorial_4/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/far/tutorial_4/CMakeLists.txt	2020-05-13 16:07:36.727941691 +1000
@@ -33,5 +33,5 @@
     $<TARGET_OBJECTS:far_obj>
 )
 
-install(TARGETS far_tutorial_4 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS far_tutorial_4 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/far/tutorial_5/CMakeLists.txt b/tutorials/far/tutorial_5/CMakeLists.txt
--- a/tutorials/far/tutorial_5/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/far/tutorial_5/CMakeLists.txt	2020-05-13 16:07:44.797953646 +1000
@@ -33,5 +33,5 @@
     $<TARGET_OBJECTS:far_obj>
 )
 
-install(TARGETS far_tutorial_5 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS far_tutorial_5 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/far/tutorial_6/CMakeLists.txt b/tutorials/far/tutorial_6/CMakeLists.txt
--- a/tutorials/far/tutorial_6/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/far/tutorial_6/CMakeLists.txt	2020-05-13 16:07:53.847966914 +1000
@@ -33,5 +33,5 @@
     $<TARGET_OBJECTS:far_obj>
 )
 
-install(TARGETS far_tutorial_6 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS far_tutorial_6 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/far/tutorial_7/CMakeLists.txt b/tutorials/far/tutorial_7/CMakeLists.txt
--- a/tutorials/far/tutorial_7/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/far/tutorial_7/CMakeLists.txt	2020-05-13 16:08:02.047978804 +1000
@@ -33,5 +33,5 @@
     $<TARGET_OBJECTS:far_obj>
 )
 
-install(TARGETS far_tutorial_7 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS far_tutorial_7 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/far/tutorial_8/CMakeLists.txt b/tutorials/far/tutorial_8/CMakeLists.txt
--- a/tutorials/far/tutorial_8/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/far/tutorial_8/CMakeLists.txt	2020-05-13 16:08:16.287999162 +1000
@@ -33,5 +33,5 @@
     $<TARGET_OBJECTS:far_obj>
 )
 
-install(TARGETS far_tutorial_8 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS far_tutorial_8 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/hbr/tutorial_0/CMakeLists.txt b/tutorials/hbr/tutorial_0/CMakeLists.txt
--- a/tutorials/hbr/tutorial_0/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/hbr/tutorial_0/CMakeLists.txt	2020-05-13 16:08:37.278028518 +1000
@@ -30,5 +30,5 @@
     ${SOURCE_FILES}
 )
 
-install(TARGETS hbr_tutorial_0 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS hbr_tutorial_0 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/hbr/tutorial_1/CMakeLists.txt b/tutorials/hbr/tutorial_1/CMakeLists.txt
--- a/tutorials/hbr/tutorial_1/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/hbr/tutorial_1/CMakeLists.txt	2020-05-13 16:08:45.028039168 +1000
@@ -30,5 +30,5 @@
     ${SOURCE_FILES}
 )
 
-install(TARGETS hbr_tutorial_1 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS hbr_tutorial_1 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/hbr/tutorial_2/CMakeLists.txt b/tutorials/hbr/tutorial_2/CMakeLists.txt
--- a/tutorials/hbr/tutorial_2/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/hbr/tutorial_2/CMakeLists.txt	2020-05-13 16:08:54.138051562 +1000
@@ -30,5 +30,5 @@
     ${SOURCE_FILES}
 )
 
-install(TARGETS hbr_tutorial_2 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS hbr_tutorial_2 DESTINATION "${CMAKE_BINDIR_BASE}")
 
diff -Naur a/tutorials/osd/tutorial_0/CMakeLists.txt b/tutorials/osd/tutorial_0/CMakeLists.txt
--- a/tutorials/osd/tutorial_0/CMakeLists.txt	2020-05-13 16:06:02.537792402 +1000
+++ b/tutorials/osd/tutorial_0/CMakeLists.txt	2020-05-13 16:09:10.748073802 +1000
@@ -35,5 +35,5 @@
     osd_static_cpu
 )
 
-install(TARGETS osd_tutorial_0 DESTINATION "${CMAKE_BINDIR_BASE}/tutorials")
+install(TARGETS osd_tutorial_0 DESTINATION "${CMAKE_BINDIR_BASE}")
 
