--- a/libconfig-1.4.9/tests/Makefile.am
+++ b/libconfig-1.4.9/tests/Makefile.am
@@ -3,10 +3,10 @@
 
 libconfig_tests_SOURCES = tests.c
 
-libconfig_tests_CPPFLAGS = -I../tinytest -I../lib
+libconfig_tests_CPPFLAGS = -I$(top_srcdir)/tinytest -I$(top_srcdir)/lib
 
-libconfig_tests_LDADD = -L../tinytest -ltinytest \
-	-L../lib/.libs -lconfig
+libconfig_tests_LDADD = -L$(top_builddir)/tinytest -ltinytest \
+	-L$(top_builddir)/lib/.libs -lconfig
 
 
 EXTRA_DIST = \
