Fix build with >=jpeg-9
Thanks-to: Moran Z. <o542018138@gmail.com>

--- a/dcraw.cc	2014-09-02 07:50:38.000000000 +0300
+++ b/dcraw.cc	2015-02-21 04:54:13.957561352 +0200
@@ -2330,7 +2330,7 @@ 
 #endif
   cinfo->src->next_input_byte = jpeg_buffer;
   cinfo->src->bytes_in_buffer = nbytes;
-  return TRUE;
+  return boolean(TRUE);
 }
 
 void CLASS kodak_jpeg_load_raw()
@@ -2346,7 +2346,7 @@ 
   jpeg_create_decompress (&cinfo);
   jpeg_stdio_src (&cinfo, ifp);
   cinfo.src->fill_input_buffer = fill_input_buffer;
-  jpeg_read_header (&cinfo, TRUE);
+  jpeg_read_header (&cinfo, boolean(TRUE));
   jpeg_start_decompress (&cinfo);
   if ((cinfo.output_width      != width  ) ||
       (cinfo.output_height*2   != height ) ||
@@ -2419,7 +2419,7 @@ 
     if (tile_length < INT_MAX)
       fseek (ifp, get4(), SEEK_SET);
     jpeg_stdio_src (&cinfo, ifp);
-    jpeg_read_header (&cinfo, TRUE);
+    jpeg_read_header (&cinfo, boolean(TRUE));
     jpeg_start_decompress (&cinfo);
     buf = (*cinfo.mem->alloc_sarray)
 	((j_common_ptr) &cinfo, JPOOL_IMAGE, cinfo.output_width*3, 1);
