--- ganeti-2.16.0~rc1.orig/configure.ac	2017-01-05 10:09:56.480211588 -0800
+++ ganeti-2.16.0~rc1/configure.ac	2017-01-05 10:15:34.462684390 -0800
@@ -380,9 +380,9 @@
    group_luxid="${withval}luxid";
    group_masterd="${withval}masterd";
    group_metad="${withval}metad";
-   group_noded="$group_default";
+   group_noded="root";
    group_daemons="${withval}daemons";
-   group_mond="$group_default"],
+   group_mond="root"],
   [group_rapi="$group_default";
    group_admin="$group_default";
    group_confd="$group_default";
@@ -391,9 +391,9 @@
    group_luxid="$group_default";
    group_masterd="$group_default";
    group_metad="$group_default";
-   group_noded="$group_default";
+   group_noded="root";
    group_daemons="$group_default";
-   group_mond="$group_default"])
+   group_mond="root"])
 AC_SUBST(RAPI_GROUP, $group_rapi)
 AC_SUBST(ADMIN_GROUP, $group_admin)
 AC_SUBST(CONFD_GROUP, $group_confd)
