diff -Nuar ganeti-3.0.2.orig/lib/cmdlib/instance_storage.py ganeti-3.0.2/lib/cmdlib/instance_storage.py
--- ganeti-3.0.2.orig/lib/cmdlib/instance_storage.py	2022-02-28 12:37:07.000000000 -0800
+++ ganeti-3.0.2/lib/cmdlib/instance_storage.py	2022-06-25 15:35:20.742784824 -0700
@@ -58,6 +58,7 @@
 import ganeti.masterd.instance
 
 
+_SLEEP_MIN_TIME = 10
 _DISK_TEMPLATE_NAME_PREFIX = {
   constants.DT_PLAIN: "",
   constants.DT_RBD: ".rbd",
@@ -1501,7 +1502,7 @@
     if done or oneshot:
       break
 
-    time.sleep(min(60, max_time))
+    time.sleep(min(_SLEEP_MIN_TIME, max_time))
 
   if done:
     lu.LogInfo("Instance %s's disks are in sync", instance.name)
