From a1f1231a73c221155d539a281181ef37f874869d Mon Sep 17 00:00:00 2001
From: William Langford <wlangfor@gmail.com>
Date: Tue, 20 Nov 2018 09:58:25 -0500
Subject: [PATCH] Add missing jv_copy when printing with -ar

---
 src/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index b154689e..61ae43f9 100644
--- a/src/main.c
+++ b/src/main.c
@@ -168,7 +168,7 @@ static int process(jq_state *jq, jv value, int flags, int dumpopts) {
   while (jv_is_valid(result = jq_next(jq))) {
     if ((options & RAW_OUTPUT) && jv_get_kind(result) == JV_KIND_STRING) {
       if (options & ASCII_OUTPUT) {
-        jv_dumpf(result, stdout, JV_PRINT_ASCII);
+        jv_dumpf(jv_copy(result), stdout, JV_PRINT_ASCII);
       } else {
         fwrite(jv_string_value(result), 1, jv_string_length_bytes(jv_copy(result)), stdout);
       }
