diff -Naur squirrelsh-1.2.6.orig//configure squirrelsh-1.2.6/configure
--- squirrelsh-1.2.6.orig//configure	2012-03-12 22:02:40.000000000 -0400
+++ squirrelsh-1.2.6/configure	2012-03-12 22:05:42.000000000 -0400
@@ -1172,15 +1172,10 @@
 # ... CFLAGS, CXXFLAGS, DEFINES, LDFLAGS (machine-dependent)
 case "$target" in
 x86)
-	in_CFLAGS="-m32 $in_CFLAGS"
-	in_CXXFLAGS="-m32 $in_CXXFLAGS"
-	in_LDFLAGS="-m32 $in_LDFLAGS";;
+	:;;
 
 x86_64)
-	in_CFLAGS="-m64 $in_CFLAGS"
-	in_CXXFLAGS="-m64 $in_CXXFLAGS"
-	in_DEFINES="$in_DEFINES -D_SQ64"
-	in_LDFLAGS="-m64 $in_LDFLAGS";;
+	in_DEFINES="$in_DEFINES -D_SQ64";;
 
 ia64)
 	in_DEFINES="$in_DEFINES -D_SQ64";;
