https://github.com/nullobsi/cantata/pull/51.patch
diff --git a/cmake/FindCDParanoia.cmake b/cmake/FindCDParanoia.cmake
index c5eaf4a7..d145d78c 100644
--- a/cmake/FindCDParanoia.cmake
+++ b/cmake/FindCDParanoia.cmake
@@ -43,6 +43,8 @@ The following cache variables may also be set:
   The directory containing ``cdda_interface.h``.
 #]===]
 
+include(CheckCSourceCompiles)
+
 # First use PKG-Config as a starting point.
 find_package(PkgConfig)
 if(PKG_CONFIG_FOUND)
@@ -96,6 +98,7 @@ if(CDParanoia_FOUND)
     )
   endif ()
   if(NOT TARGET CDDA::Interface)
+    add_library(CDDA::Interface UNKNOWN IMPORTED)
     set_target_properties(CDDA::Interface PROPERTIES
             IMPORTED_LOCATION "${CDParanoia_INTERFACE_LIBRARY}"
             INTERFACE_COMPILE_OPTIONS "${PC_CDParanoia_CFLAGS_OTHER}"
@@ -118,4 +121,4 @@ mark_as_advanced(
         CDParanoia_INTERFACE_INCLUDE_DIR
         CDParanoia_PARANOIA_LIBRARY
         CDParanoia_INTERFACE_LIBRARY
-)
\ No newline at end of file
+)
