From 4da7a9128f2c5eaf23ae2a5006d300dc4f21fc6a Mon Sep 17 00:00:00 2001
From: Craig Drummond <craig.p.drummond@gmail.com>
Date: Tue, 28 Nov 2017 22:37:01 +0000
Subject: [PATCH] Create QSqlQuery, then call prepare Issue #1130

---
 db/librarydb.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/db/librarydb.cpp b/db/librarydb.cpp
index 603d8abc1..4f75dc28f 100644
--- a/db/librarydb.cpp
+++ b/db/librarydb.cpp
@@ -453,7 +453,8 @@ class SqlQuery
         if (limit>0) {
             sql+=" LIMIT "+QString::number(limit);
         }
-        query=QSqlQuery(sql, db);
+        query = QSqlQuery(db);
+        query.prepare(sql);
 
         for (const QVariant &value: boundValues) {
             query.addBindValue(value);
