--- doc/example-config/conf.d/10-ssl.conf	2014-12-08 07:58:21.000000000 +0000
+++ doc/example-config/conf.d/10-ssl.conf	2014-12-08 08:02:19.000000000 +0000
@@ -49,6 +49,12 @@
 #ssl_protocols = !SSLv2
 
 # SSL ciphers to use
+# ###############
+# Added by Gentoo
+# You are encouraged to change the cipher list to
+#ssl_cipher_list = DEFAULT:!EXPORT:!LOW:!MEDIUM:!MD5
+# if you are not required to support legacy mail clients.
+# ###############
 #ssl_cipher_list = ALL:!LOW:!SSLv2:!EXP:!aNULL
 
 # Prefer the server's order of ciphers over client's.
