--- a/src/frontends/qt4/GuiDocument.cpp	2018-02-25 01:11:18.000000000 +0100
+++ b/src/frontends/qt4/GuiDocument.cpp	2018-03-30 19:45:47.283928297 +0200
@@ -68,11 +68,12 @@
 #include "frontends/alert.h"
 
 #include <QAbstractItemModel>
-#include <QHeaderView>
+#include <QButtonGroup>
 #include <QColor>
 #include <QColorDialog>
 #include <QCloseEvent>
 #include <QFontDatabase>
+#include <QHeaderView>
 #include <QScrollBar>
 #include <QTextBoundaryFinder>
 #include <QTextCursor>
