diff --git a/extra/openssl_compat.c b/extra/openssl_compat.c
index 399df90..4cee4cd 100644
--- a/extra/openssl_compat.c
+++ b/extra/openssl_compat.c
@@ -26,8 +26,6 @@
 
 #include "gnutls_int.h"
 
-#include <gnutls_global.h>
-#include <gnutls_errors.h>
 #include <string.h>		/* memset */
 #include <x509/x509_int.h>
 #include <libtasn1.h>
diff --git a/tests/windows/cng-windows.c b/tests/windows/cng-windows.c
index f435afb..e92f7b0 100644
--- a/tests/windows/cng-windows.c
+++ b/tests/windows/cng-windows.c
@@ -31,6 +31,8 @@
 
 #ifndef _WIN32
 
+#include <stdlib.h>
+
 void doit()
 {
 	exit(77);
