diff --git a/autowaf.py b/autowaf.py
index 4797ea3..f32f994 100644
--- a/autowaf.py
+++ b/autowaf.py
@@ -375,8 +375,5 @@ def build_version_files(header_path, source_path, domain, major, minor, micro):
 	return None
 
 def shutdown():
-	# This isn't really correct (for packaging), but people asking is annoying
-	if Options.commands['install']:
-		try: os.popen("/sbin/ldconfig")
-		except: pass
+	return
 
