--- dnssec-tools-1.13/tools/scripts/dtinitconf
+++ dnssec-tools-1.13/tools/scripts/dtinitconf
@@ -59,7 +59,9 @@
 my @COMMON_BINDLOCS =
 (
 	"/usr/local/sbin",
+	"/usr/local/bin",
 	"/usr/sbin",
+	"/usr/bin",
 );
 
 #
@@ -421,8 +423,8 @@
 
 	$admin	   = $opts{'admin'}	|| dnssec_tools_default("admin");
 
-	$binddir   = $opts{'binddir'};
-	$dtdir	   = $opts{'dtdir'};
+	$binddir   = $opts{'binddir'} || 'path';
+	$dtdir	   = $opts{'dtdir'} || 'path';
 #	$viewimg   = $opts{'viewimage'}	|| dnssec_tools_default("viewimage");
 
 	$outfile   = $opts{'outfile'}	|| getconffile();
