diff -Naupr bbkeys-0.9.0.orig/src/actions.cc bbkeys-0.9.0/src/actions.cc
--- bbkeys-0.9.0.orig/src/actions.cc	2005-02-20 22:03:43.000000000 +0100
+++ bbkeys-0.9.0/src/actions.cc	2008-02-24 09:26:30.000000000 +0100
@@ -25,7 +25,8 @@
 #include "actions.hh"
 
 #include <iostream>
-#include <string>
+#include <cstring>
+#include <cstdlib>
 #include <sstream>
 
 Action::Action(enum ActionType type, Display * display, KeyCode keycode, 
diff -Naupr bbkeys-0.9.0.orig/src/Config.cpp bbkeys-0.9.0/src/Config.cpp
--- bbkeys-0.9.0.orig/src/Config.cpp	2004-09-03 03:08:00.000000000 +0200
+++ bbkeys-0.9.0/src/Config.cpp	2008-02-24 09:29:55.000000000 +0100
@@ -24,7 +24,7 @@
 
 
 #include "Config.h"
-#include <string>
+#include <cstring>
 
 #include <iostream>
 using std::cout;
diff -Naupr bbkeys-0.9.0.orig/src/KeyClient.cpp bbkeys-0.9.0/src/KeyClient.cpp
--- bbkeys-0.9.0.orig/src/KeyClient.cpp	2004-09-26 04:27:46.000000000 +0200
+++ bbkeys-0.9.0/src/KeyClient.cpp	2008-02-24 09:26:14.000000000 +0100
@@ -58,6 +58,7 @@ extern "C" {
 #include <iostream>
 #include <algorithm>
 #include <vector>
+#include <cstring>
 
 //--------------------------------------------------------
 // Constructor/Destructor
diff -Naupr bbkeys-0.9.0.orig/src/main.cpp bbkeys-0.9.0/src/main.cpp
--- bbkeys-0.9.0.orig/src/main.cpp	2004-09-16 02:53:51.000000000 +0200
+++ bbkeys-0.9.0/src/main.cpp	2008-02-24 09:31:38.000000000 +0100
@@ -27,6 +27,8 @@
 
 #include "main.h"
 
+#include <cstring>
+
 //--------------------------------------------------------
 // parseOptions
 //--------------------------------------------------------
diff -Naupr bbkeys-0.9.0.orig/src/Netclient.cpp bbkeys-0.9.0/src/Netclient.cpp
--- bbkeys-0.9.0.orig/src/Netclient.cpp	2004-09-26 04:27:46.000000000 +0200
+++ bbkeys-0.9.0/src/Netclient.cpp	2008-02-24 09:30:41.000000000 +0100
@@ -25,6 +25,7 @@
 // Methods, ideas, implementations taken from Openbox's XAtom class *sigh*
 
 #include "Netclient.h"
+#include <cstring>
 
 Netclient::Netclient (const bt::Display &display)
   : bt::EWMH(display), _display(display)
