diff -Nru 3.0.1.vanilla/scite/src/SciTEProps.cxx 3.0.1/scite/src/SciTEProps.cxx
--- 3.0.1.vanilla/scite/src/SciTEProps.cxx	2011-11-30 19:39:41.059801342 +0100
+++ 3.0.1/scite/src/SciTEProps.cxx	2011-11-30 19:39:53.211835004 +0100
@@ -1702,10 +1702,12 @@
 
 // return the int value of the command name passed in.
 int SciTEBase::GetMenuCommandAsInt(SString commandName) {
+#ifndef NO_LUA
 	int i = IFaceTable::FindConstant(commandName.c_str());
 	if (i != -1) {
 		return IFaceTable::constants[i].value;
 	}
+#endif
 	// Otherwise we might have entered a number as command to access a "SCI_" command
 	return commandName.value();
 }
