--- SuperTux-v0.6.0-Source/CMakeLists.txt
+++ SuperTux-v0.6.0-Source/CMakeLists.txt
@@ -846,7 +846,7 @@
     install(FILES ${DLLS} DESTINATION ${INSTALL_SUBDIR_BIN})
   endif()
 
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/images/engine/icons/supertux.png ${CMAKE_CURRENT_SOURCE_DIR}/data/images/engine/icons/supertux.ico DESTINATION ".")
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/images/engine/icons/supertux.png DESTINATION ".")
 
   option(PACKAGE_VCREDIST "Package the VCREDIST libraries with the program" OFF)
 
