Add missing libs to relevant targets

--- a/configure.ac	2017-09-03 17:10:57.552216900 +0200
+++ b/configure.ac	2017-09-03 17:16:39.674036975 +0200
@@ -4,7 +4,7 @@
 AC_PREREQ(2.57)
 
 AC_INIT(libgksu, 2.0.12, kov@debian.org)
-AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
+AM_INIT_AUTOMAKE([foreign])
 AC_CONFIG_SRCDIR(libgksu/libgksu.c)
 AC_CONFIG_HEADERS(config.h)
 
@@ -44,7 +44,7 @@
 PKG_CHECK_MODULES(GKSU_PROPERTIES, [gtk+-2.0 >= 2.12, gconf-2.0])
 
 PKG_CHECK_MODULES(GLIB, [glib-2.0 gthread-2.0])
-PKG_CHECK_MODULES(GTK, [gtk+-2.0 gconf-2.0])
+PKG_CHECK_MODULES(GTK, [gtk+-2.0 gconf-2.0 x11])
 
 # Checks for library functions.
 ALL_LINGUAS="ca cs da de es eu fr hu it ko lt pl pt_BR ro ru sk sv nb nl th zh_CN"
