From 9349b722b23382204d88c4dfb8da654e43e9b3d5 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Sat, 19 Apr 2025 22:19:39 +0200
Subject: [PATCH] Disable git usage, disable build host info

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d1ba9e0..9a679a2c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,7 @@ set(AKO_PREFIX "KPim6")
 
 message("Akonadi Prefix is ${AKO_PREFIX}")
 
+if(0)
 find_package(Git QUIET)
 if (Git_FOUND AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
     include(GetGitRevisionDescription)
@@ -66,6 +67,12 @@ if(${CMAKE_VERSION} VERSION_GREATER "3.22.0")
 else ()
     set(BUILD_HOST_DISTRI "unknown distribution")
 endif()
+else()
+    message("GIT_SHA1: ${GIT_SHA1}")
+    message("GIT_BRANCH: ${GIT_BRANCH}")
+    set(BUILD_HOST_NAME "denied")
+    set(BUILD_HOST_DISTRI "Gentoo Linux")
+endif()
 message("Build host name: ${BUILD_HOST_NAME}")
 message("Build host distribution: ${BUILD_HOST_DISTRI}")
 
-- 
2.52.0

