--- duplicity-0.8.11.1596/setup.py	2020-03-05 09:18:42.532941503 -0500
+++ duplicity-0.8.11.1596/setup.py	2020-03-05 09:20:10.612061492 -0500
@@ -88,18 +88,6 @@
                 u'bin/rdiffdir.1'
                 ]
             ),
-            (u'share/doc/duplicity-%s' % version_string,
-                [
-                u'AUTHORS',
-                u'CHANGELOG',
-                u'Changelog.GNU',
-                u'COPYING',
-                u'README',
-                u'README-LOG',
-                u'README-REPO',
-                u'README-TESTING',
-                ],
-            ),
         ]
 
     if not os.environ.get(u'READTHEDOCS') == u'True':
