--- a/setup.py	2024-12-22 08:12:58.285362768 -0500
+++ b/setup.py	2024-12-22 08:13:31.138806077 -0500
@@ -113,18 +113,6 @@
                 "man/duplicity.1",
             ],
         ),
-        (
-            f"share/doc/duplicity-{Version}",
-            [
-                "CHANGELOG.md",
-                "AUTHORS.md",
-                "COPYING",
-                "README.md",
-                "README-LOG.md",
-                "README-REPO.md",
-                "README-TESTING.md",
-            ],
-        ),
     ]
 
     # short circuit fot READTHEDOCS
