--- duplicity-1.2.2/setup.py	2023-05-01 09:21:01.020192022 -0400
+++ duplicity-1.2.2/setup.py	2023-05-01 09:21:51.268882266 -0400
@@ -95,17 +95,6 @@
                 u'bin/rdiffdir.1'
                 ]
             ),
-            (u'share/doc/duplicity-%s' % Version,
-                [
-                u'CHANGELOG.md',
-                u'CONTRIBUTING.md',
-                u'COPYING',
-                u'README.md',
-                u'README-LOG.md',
-                u'README-REPO.md',
-                u'README-TESTING.md',
-                ],
-            ),
         ]
 
     # short circuit fot READTHEDOCS
