--- src/dird/dird.c.orig	2021-01-19 15:25:16.233573546 -0500
+++ src/dird/dird.c		2021-01-19 15:26:23.884070798 -0500
@@ -1263,11 +1263,11 @@
          }
          if (catalog->db_driver) {
            /* To copy dbdriver field into "CAT" catalog resource class (local)
             * from dbdriver in "BDB" catalog DB Interface class (global)
             */
-            bstrncpy(catalog->db_driver, BDB_db_driver, db_driver_len);
+            bstrncpy(catalog->db_driver, BDB_db_driver, db_driver_len+1);
          }
       }
 
       if (!db || !db_open_database(NULL, db)) {
          Pmsg2(000, _("Could not open Catalog \"%s\", database \"%s\".\n"),
