diff --git a/meson.build b/meson.build
index 3b0d7b7..6e077cf 100644
--- a/meson.build
+++ b/meson.build
@@ -20,9 +20,8 @@ girara = dependency('girara-gtk3')
 glib = dependency('glib-2.0')
 cairo = dependency('cairo')
 mupdf = cc.find_library('mupdf')
-mupdfthird = cc.find_library('mupdfthird')
 
-build_dependencies = [zathura, girara, glib, cairo, mupdf, mupdfthird]
+build_dependencies = [zathura, girara, glib, cairo, mupdf]
 
 # defines
 defines = [
