https://github.com/pkgcore/pkgcheck/commit/b4c34a95f4c66db9c658f6dd75d9193c6e55dab4

--- a/setup.py
+++ b/setup.py
@@ -36,12 +36,12 @@ class build_treesitter(Command, SubCommand):
         pass

     def get_source_files(self):
-        cwd = Path(__file__).parent / "tree-sitter-bash/src"
+        src = "tree-sitter-bash/src/"
         return [
-            str(cwd / "GNUmakefile"),
-            str(cwd / "tree_sitter/parser.h"),
-            str(cwd / "parser.c"),
-            str(cwd / "scanner.cc"),
+            src + "GNUmakefile",
+            src + "tree_sitter/parser.h",
+            src + "parser.c",
+            src + "scanner.cc",
         ]

     library_path = Path(__file__).parent / "src/pkgcheck/bash/lang.so"
