From 34df1dd98a7ce3a17c22081fadb427827b431d7f Mon Sep 17 00:00:00 2001
From: Carlos <nzlosh@yahoo.com>
Date: Fri, 22 Apr 2022 23:45:44 +0200
Subject: [PATCH] Fixes #1566 remove unsupport argument reconnection_interval
 from irc backend.

diff --git a/errbot/backends/irc.py b/errbot/backends/irc.py
index 95a8757d4..9f7e75b29 100644
--- a/errbot/backends/irc.py
+++ b/errbot/backends/irc.py
@@ -377,7 +377,6 @@ def __init__(
             [(server, port, password)],
             nickname,
             username,
-            reconnection_interval=reconnect_on_disconnect,
         )
 
     def connect(self, *args, **kwargs) -> None:
