--- libeventdb-0.90/event.h.orig	2008-03-13 23:05:10.000000000 -0700
+++ libeventdb-0.90/event.h	2008-03-13 23:05:40.000000000 -0700
@@ -130,7 +130,7 @@ extern GSList *event_list (EventSource *
 			   GError **error)
      __attribute__ ((visibility ("hidden")));
 
-#define LIVE(ev) (g_assert (! EVENT (ev)->dead))
+#define LIVE(ev) G_STMT_START { g_assert (! EVENT (ev)->dead); } G_STMT_END
 
 /* Marks the event as well as the calendar in which lives as
    modified.  */
