From dfd304e93bb9010ef8226ee4d026d45c9bf9d2f5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sat, 28 Jul 2012 15:54:57 +0200
Subject: [PATCH 3/3] Import cstdint for int64_t.

---
 src/net/download.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/net/download.h b/src/net/download.h
index c0bdf20..b9ad17e 100644
--- a/src/net/download.h
+++ b/src/net/download.h
@@ -18,6 +18,7 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <cstdint>
 #include <cstdio>
 #include <string>
 
-- 
1.7.11.1

