From 3f4b0661b63efaca78791612ce9be84271d784a5 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 5 Nov 2012 02:08:27 -0500
Subject: [PATCH] fix build build system

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 utils/Makefile | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/utils/Makefile b/utils/Makefile
index 710ebbf..26aea6b 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -16,14 +16,13 @@
 
 #KERNELDIR = /usr/src/kernel-headers-2.4.18
 
-CFLAGS =    -O2 -Wall -DCONFIG_YAFFS_UTIL
+CFLAGS+=    -Wall -DCONFIG_YAFFS_UTIL -DCONFIG_YAFFS_DEFINES_TYPES
 CFLAGS+=   -Wshadow -Wpointer-arith -Wwrite-strings -Wstrict-prototypes -Wmissing-declarations
 CFLAGS+=   -Wmissing-prototypes -Wredundant-decls -Wnested-externs -Winline
 
 ## Change if you are using a cross-compiler
 MAKETOOLS = 
 
-CC=$(MAKETOOLS)gcc
 
 COMMON_BASE_C_LINKS = yaffs_ecc.c
 COMMON_BASE_LINKS = $(COMMON_BASE_C_LINKS) yaffs_ecc.h yaffs_guts.h yaffs_packedtags2.h yaffs_trace.h
@@ -57,10 +56,10 @@ $(COMMONOBJS) $(MKYAFFSIMAGEOBJS) $(MKYAFFS2IMAGEOBJS) : %.o: %.c
 	$(CC) -c $(CFLAGS) $< -o $@
 
 mkyaffsimage: $(MKYAFFSIMAGEOBJS) $(COMMONOBJS)
-	$(CC) -o $@  $^
+	$(CC) $(CFLAGS) -o $@  $^ $(LDFLAGS)
 
 mkyaffs2image: $(MKYAFFS2IMAGEOBJS) $(COMMONOBJS)
-	$(CC) -o $@ $^
+	$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS)
 
 
 clean:
-- 
1.7.12.4

