--- a/CMakeLists.txt	2020-12-07 17:13:38.000000000 +0100
+++ b/CMakeLists.txt	2021-02-20 20:01:41.706789439 +0100
@@ -18,25 +18,6 @@
 endif()
 
 ###############################################################################################
-# Do distribution dependent stuff
-###############################################################################################
-string(TIMESTAMP YEAR "%Y")
-set(EXPIRE_YEAR "2021")
-set(DEBIAN_RELEASE_INFO "/etc/debian_version")
-
-if(((EXISTS ${DEBIAN_RELEASE_INFO}) AND (${YEAR} LESS ${EXPIRE_YEAR})) OR APPLE)
-    message( STATUS "Apply patches to fix build system on Debian")
-    execute_process(COMMAND bash "-c" "patch --quiet -p1 < FindPROJ4.patch"
-                    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-                    OUTPUT_QUIET)
-    execute_process(COMMAND bash "-c" "patch --quiet -p1 < FindQuaZip5.patch"
-                    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-                    OUTPUT_QUIET)
-endif(((EXISTS ${DEBIAN_RELEASE_INFO}) AND (${YEAR} LESS ${EXPIRE_YEAR})) OR APPLE)
-
-
-
-###############################################################################################
 # Define project name and version tags
 ###############################################################################################
 # this is not only the QMS version it will be the package version, too.
