diff --git a/setup.py b/setup.py
index 882fe92..f0d28ab 100755
--- a/setup.py
+++ b/setup.py
@@ -36,10 +36,6 @@ setup(
     install_requires=install_requirements,
     entry_points={
         'console_scripts': ['mycli = mycli.main:cli'],
-        'distutils.commands': [
-            'lint = tasks:lint',
-            'test = tasks:test',
-        ],
     },
     classifiers=[
         'Intended Audience :: Developers',
