diff --git a/docs/conf.py b/docs/conf.py
index 094ed23..a58cb85 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -41,7 +41,6 @@ import skbuild
 # Add any Sphinx extension module names here, as strings. They can be
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
 extensions = [
-    'cmake',
     'sphinx.ext.autodoc',
     'sphinx.ext.viewcode',
     'sphinx_issues'
@@ -291,4 +290,4 @@ on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
 if not on_rtd:  # only import and set the theme if we're building docs locally
     import sphinx_rtd_theme
     html_theme = 'sphinx_rtd_theme'
-    html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
\ No newline at end of file
+    html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
