From a4970ce40ec43c0a1db418aa05ee2853497a0c14 Mon Sep 17 00:00:00 2001
From: Hannah von Reth <vonreth@kde.org>
Date: Wed, 31 Jan 2024 13:24:09 +0000
Subject: [PATCH] Fix cmake config if lib was not build

---
 KF6BreezeIconsConfig.cmake.in | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/KF6BreezeIconsConfig.cmake.in b/KF6BreezeIconsConfig.cmake.in
index be4ae7310..2d03f0c6e 100644
--- a/KF6BreezeIconsConfig.cmake.in
+++ b/KF6BreezeIconsConfig.cmake.in
@@ -3,6 +3,5 @@
 if(@ICONS_LIBRARY@)
     include(CMakeFindDependencyMacro)
     find_dependency(Qt6Gui "@REQUIRED_QT_VERSION@")
+    include("${CMAKE_CURRENT_LIST_DIR}/KF6BreezeIconsTargets.cmake")
 endif()
-
-include("${CMAKE_CURRENT_LIST_DIR}/KF6BreezeIconsTargets.cmake")
-- 
GitLab

