--- a/CMakeLists.txt	2015-11-10 00:21:27.222697427 +0100
+++ b/CMakeLists.txt	2015-11-10 00:26:44.481731143 +0100
@@ -7,9 +7,6 @@
 include(WriteBasicConfigVersionFile)
 include(FeatureSummary)
 
-if(USE_KDE4)
-  add_subdirectory(kstyle)
-else()
   find_package(ECM 0.0.9 REQUIRED NO_MODULE)
   set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 
@@ -19,24 +16,7 @@
   include(KDECompilerSettings)
   include(GenerateExportHeader)
 
-  add_subdirectory(cursors)
   add_subdirectory(icons)
   add_subdirectory(icons-dark)
-  add_subdirectory(kdecoration)
-  add_subdirectory(kstyle)
-  add_subdirectory(misc)
-  add_subdirectory(qtquickcontrols)
-  add_subdirectory(wallpapers)
-
-  INSTALL(FILES colors/Breeze.colors DESTINATION ${DATA_INSTALL_DIR}/color-schemes/)
-  INSTALL(FILES colors/BreezeDark.colors DESTINATION ${DATA_INSTALL_DIR}/color-schemes/)
-  INSTALL(FILES colors/BreezeHighContrast.colors DESTINATION ${DATA_INSTALL_DIR}/color-schemes/)
-  if(EXISTS ${CMAKE_SOURCE_DIR}/po AND IS_DIRECTORY ${CMAKE_SOURCE_DIR}/po)
-    find_package(KF5I18n CONFIG REQUIRED)
-    ki18n_install(po)
-  endif()
-endif()
 
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
-message(" * Orion theme for GTK 2 and 3 will be set by gtkbreeze kconf_update binary,"
-        " please install or add a runtime dependency to packages. https://github.com/shimmerproject/Orion")
