--- a/SConstruct
+++ b/SConstruct
@@ -284,10 +284,10 @@
     ('LINK_PRIORITY','Priority list in which to sort library and include paths (default order is internal, other, frameworks, user, then system - see source of `sort_paths` function for more detail)',','.join(DEFAULT_LINK_PRIORITY)),
 
     # Install Variables
-    ('PREFIX', 'The install path "prefix"', '/usr/local'),
+    ('PREFIX', 'The install path "prefix"', '/usr'),
     ('LIBDIR_SCHEMA', 'The library sub-directory appended to the "prefix", sometimes lib64 on 64bit linux systems', LIBDIR_SCHEMA_DEFAULT),
     ('PYTHON_PREFIX','Custom install path "prefix" for python bindings (default of no prefix)',''),
-    ('DESTDIR', 'The root directory to install into. Useful mainly for binary package building', '/'),
+    ('DESTDIR', 'The root directory to install into. Useful mainly for binary package building', '${D}'),
     ('PATH', 'A custom path (or multiple paths divided by ":") to append to the $PATH env to prioritize usage of command line programs (if multiple are present on the system)', ''),
     ('PATH_REMOVE', 'A path prefix to exclude from all known command and compile paths', ''),
     ('PATH_REPLACE', 'Two path prefixes (divided with a :) to search/replace from all known command and compile paths', ''),
