--- src/com/kenai/jaffl/provider/Library.java.orig	2011-09-07 12:13:06.942731817 +0200
+++ src/com/kenai/jaffl/provider/Library.java	2011-09-07 12:13:08.983731771 +0200
@@ -23,6 +23,7 @@
             userLibraryPath.addAll(getPropertyPaths("jaffl.library.path"));
             // Add JNA paths for compatibility
             userLibraryPath.addAll(getPropertyPaths("jna.library.path"));
+            userLibraryPath.addAll(getPropertyPaths("java.library.path"));
         }
     };
     
