Index: python-dsv-1.4.1/DSV/DSV.py
===================================================================
--- python-dsv-1.4.1.orig/DSV/DSV.py	2003-08-09 06:37:32.000000000 -0400
+++ python-dsv-1.4.1/DSV/DSV.py	2010-01-26 20:38:02.000000000 -0500
@@ -127,11 +127,11 @@
 import copy
 import exceptions
 import string
-# RedHat 8.0 (or rather GTK2?) sets LANG = en_us.UTF-8 and apparently some
-# older apps (including wxGTK) can't handle this.  The fix is to set LANG=C
-# before running the app.  Thanks to Ahmad Baitalmal for supplying this info.
-import os
-os.putenv('LANG', 'C')
+## RedHat 8.0 (or rather GTK2?) sets LANG = en_us.UTF-8 and apparently some
+## older apps (including wxGTK) can't handle this.  The fix is to set LANG=C
+## before running the app.  Thanks to Ahmad Baitalmal for supplying this info.
+# import os
+# os.putenv('LANG', 'C')
 
 try:
     from wxPython import wx, grid
