--- Makefile
+++ Makefile
@@ -12,7 +12,7 @@
 #
 #CFLAGS = -Wall -O -D_GNU_SOURCE -g -DDEBUG
 #CFLAGS = -Wall -g -Os -D_GNU_SOURCE
-CFLAGS = -Wall -g -O3 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 \
+CFLAGS += -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 \
	 -I/usr/include/ffmpeg \
 	 -I/usr/include/libavutil -I/usr/include/libavcodec -I/usr/include/libavformat \
	 -I/usr/include/ffmpeg/libavutil -I/usr/include/ffmpeg/libavcodec -I/usr/include/ffmpeg/libavformat
