diff -Naur fuse-1.5.7/ui/widget/widget.c fuse-1.5.7-patched/ui/widget/widget.c
--- fuse-1.5.7/ui/widget/widget.c	2018-08-06 15:36:04.000000000 +0200
+++ fuse-1.5.7-patched/ui/widget/widget.c	2020-09-20 15:27:42.045382588 +0200
@@ -90,9 +90,6 @@
 
 static widget_recurse_t widget_return[10]; /* The stack to recurse on */
 
-/* The settings used whilst playing with an options dialog box */
-settings_info widget_options_settings;
-
 static int widget_read_font( const char *filename )
 {
   utils_file file;
