--- perl/Makefile.in	2011-06-21 22:50:39.000000000 +0300
+++ perl/Makefile.in	2011-06-22 12:42:04.000000000 +0300
@@ -74,6 +74,7 @@
 cyrus_prefix = @cyrus_prefix@
 
 CC = @CC@
+LDFLAGS = @LDFLAGS@
 PLMAKE = imap/Makefile
 
 all::
@@ -83,6 +84,7 @@
 			if [ -f Makefile.PL -a ! -f Makefile ]; then \
 			   LIB_RT="$(LIB_RT)" \
 			   ZLIB="$(ZLIB)" \
+			   LDDLFLAGS="-shared $(LDFLAGS)" \
 			   BDB_LIB="$(BDB_LIB)" BDB_INC="$(BDB_INC)" \
 			   OPENSSL_LIB="$(OPENSSL_LIB)" OPENSSL_INC="$(OPENSSL_INC)" \
 			   SASL_LIB="$(SASL_LIB)" SASL_INC="$(SASL_INC)" CC="$(CC)" \
--- perl/imap/Makefile.PL	2011-06-21 22:50:39.000000000 +0300
+++ perl/imap/Makefile.PL	2011-06-22 12:42:22.000000000 +0300
@@ -53,6 +53,8 @@
 my $LIB_RT = $ENV{LIB_RT};
 my $ZLIB = $ENV{ZLIB};
 
+my $LDDLFLAGS = $ENV{LDDLFLAGS};
+
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
 $libs = "-lcyrus";
@@ -75,6 +77,7 @@
     'OBJECT'    => 'IMAP.o',
     'MYEXTLIB'  => '../../lib/libcyrus.a ../../lib/libcyrus_min.a',
     'LIBS'	=> ["$BDB_LIB $SASL_LIB $OPENSSL_LIB $LIB_RT $ZLIB -lssl -lcrypto"],
+    'LDDLFLAGS' => "$LDDLFLAGS",
     'DEFINE'	=> '-DPERL_POLLUTE',    # e.g., '-DHAVE_SOMETHING' 
     'INC'	=> "-I../../lib -I../.. -I../../com_err/et $SASL_INC $OPENSSL_INC", 
     'EXE_FILES' => [cyradm],
--- perl/sieve/managesieve/Makefile.PL	2011-06-21 22:50:39.000000000 +0300
+++ perl/sieve/managesieve/Makefile.PL	2011-06-22 13:13:04.000000000 +0300
@@ -53,6 +53,8 @@
 my $LIB_RT = $ENV{LIB_RT};
 my $ZLIB = $ENV{ZLIB};
 
+my $LDDLFLAGS = $ENV{LDFLAGS};
+
 $libs = "-lcyrus";
 
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
@@ -63,6 +65,7 @@
     'VERSION_FROM' => 'managesieve.pm', # finds $VERSION
     'MYEXTLIB'  => '../lib/isieve.o ../lib/lex.o ../lib/mystring.o ../lib/request.o ../../../lib/libcyrus.a ../../../lib/libcyrus_min.a',
     'LIBS'	=> ["$BDB_LIB $SASL_LIB $OPENSSL_LIB $LIB_RT $ZLIB -lssl -lcrypto"], 
+    'LDDLFLAGS' => "-shared $LDDLFLAGS",
     'DEFINE'	=> '-DPERL_POLLUTE',     # e.g., '-DHAVE_SOMETHING' 
     'INC'	=> "-I../lib/ -I../../../lib/ $SASL_INC $OPENSSL_INC",
 );
