diff --git a/CMakeLists.txt b/CMakeLists.txt
index f7a0dfe..22d1f4f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -271,7 +271,6 @@ if (ENABLE_MONGOC)
       VERSION_CURRENT
       VERSION_RELEASED
       # This sub-directory is added later, so manually include here
-      generate_uninstall/CMakeLists.txt
    )
 
    set_local_dist (top_DIST ${top_DIST_local})
@@ -341,14 +340,8 @@ if (WIN32)
 else ()
    set (UNINSTALL_PROG "uninstall.sh")
 endif ()
-set (UNINSTALL_PROG_DIR "${CMAKE_INSTALL_FULL_DATADIR}/mongo-c-driver")
 
 # Create uninstall program and associated uninstall target
 #
 # This needs to be last (after all other add_subdirectory calls) to ensure that
 # the generated uninstall program is complete and correct
-add_subdirectory (generate_uninstall)
-
-add_custom_target (uninstall
-   COMMAND "${UNINSTALL_PROG_DIR}/${UNINSTALL_PROG}"
-)
