diff --git a/CMakeLists.txt b/CMakeLists.txt
index 22d1f4f..8ab33c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,7 +118,6 @@ if (ENABLE_BSON STREQUAL SYSTEM)
       message ("--   libbson include path \"${BSON_INCLUDE_DIRS}\"")
       message ("--   libbson libraries \"${BSON_LIBRARIES}\"")
       message ("--   disabling test-libmongoc since using system libbson")
-      SET (ENABLE_TESTS OFF)
 
       if (ENABLE_STATIC MATCHES "ON|AUTO")
          find_package (libbson-static-1.0
diff --git a/src/libbson/tests/test-bson.c b/src/libbson/tests/test-bson.c
index 6ac94f3..cec430f 100644
--- a/src/libbson/tests/test-bson.c
+++ b/src/libbson/tests/test-bson.c
@@ -18,7 +18,7 @@
 #include <bson/bson.h>
 #include <bson/bcon.h>
 #define BSON_INSIDE
-#include <bson/bson-private.h>
+#include "bson/bson-private.h"
 #undef BSON_INSIDE
 #include <fcntl.h>
 #include <time.h>
