 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 60033eb..13ddbd0 100755
--- a/Makefile.in
+++ b/Makefile.in
@@ -53,7 +53,7 @@ clipper: FORCE fftw
 
 diffractionImg: 
 	if test -n "$(diffimdir)"; then cd $(top_srcdir)/$(diffimdir); \
-	$(MAKE) -i $(MFLAGS) $(MVARS); else true; fi
+	$(MAKE) $(MFLAGS) $(MVARS); fi
 
 ssm: 
 	if test -n "$(ssmdir)"; then cd $(top_srcdir)/$(ssmdir); \
