diff -U 3 -dHrN Normaliz2.12/source/libnormaliz/general.h Normaliz2.12.patched/source/libnormaliz/general.h
--- Normaliz2.12/source/libnormaliz/general.h	2014-10-23 15:01:56.000000000 +0200
+++ Normaliz2.12.patched/source/libnormaliz/general.h	2015-03-01 13:20:01.057017555 +0100
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <assert.h>
+#include <cstddef>
 
 #ifdef _WIN32 //for 32 and 64 bit windows
     #define NMZ_MPIR //always use MPIR
