diff --git a/components/editor/gtkhtml-spell-checker.c b/components/editor/gtkhtml-spell-checker.c
index 0927f57..89e9873 100644
--- a/components/editor/gtkhtml-spell-checker.c
+++ b/components/editor/gtkhtml-spell-checker.c
@@ -375,7 +375,7 @@ gtkhtml_spell_checker_add_word (GtkhtmlSpellChecker *checker,
 	if ((dict = spell_checker_request_dict (checker)) == NULL)
 		return;
 
-	enchant_dict_add_to_pwl (dict, word, length);
+	enchant_dict_add (dict, word, length);
 	g_signal_emit (G_OBJECT (checker), signals[ADDED], 0, word, length);
 }
 
diff --git a/configure.ac b/configure.ac
index e4946aa..e5771e3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,7 +10,7 @@ PACKAGE=gtkhtml
 
 dnl Required Package Versions
 m4_define([gtk_minimum_version], [3.2.0])
-m4_define([enchant_minimum_version], [1.1.7])
+m4_define([enchant_minimum_version], [2.0.0])
 m4_define([cairo_minimum_version], [1.10.0])
 
 GTKHTML_API_VERSION=4.0
@@ -140,7 +140,7 @@ AC_SUBST(GAIL_MODULES)
 dnl **********************************
 dnl GTKHTML modules
 dnl **********************************
-GTKHTML_MODULES="gtk+-3.0 >= gtk_minimum_version enchant >= enchant_minimum_version gsettings-desktop-schemas iso-codes cairo >= cairo_minimum_version"
+GTKHTML_MODULES="gtk+-3.0 >= gtk_minimum_version enchant-2 >= enchant_minimum_version gsettings-desktop-schemas iso-codes cairo >= cairo_minimum_version"
 PKG_CHECK_MODULES(GTKHTML, $GTKHTML_MODULES)
 AC_SUBST(GTKHTML_CFLAGS)
 AC_SUBST(GTKHTML_LIBS)
