From e8d26df2a888f307e61584367f5dc14894d8be82 Mon Sep 17 00:00:00 2001
From: Jonathan Scruggs <j.scruggs@gmail.com>
Date: Sat, 23 Sep 2017 13:39:34 +0100
Subject: [PATCH] OpenEXR_Viewers: Remove nVidia automagic

---
 OpenEXR_Viewers/configure.ac | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 94c88cd..1de94f6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -194,7 +194,10 @@ AM_PATH_FLTK(have_fltk=yes, have_fltk=no)
 dnl
 dnl check for Cg
 dnl
-if test "${have_fltk}" = yes; then
+AC_ARG_ENABLE(cg, AC_HELP_STRING([--enable-cg],
+  [enable nvidia cg support [[default=check]]]), [want_cg="${enableval}"], [want_cg=yes])
+
+if test "${have_fltk}" = yes -a "${want_cg}" != "no"; then
   AM_PATH_CG(have_cg=yes, have_cg=no)
   if test "${have_cg}" = yes; then
     AC_DEFINE(HAVE_FRAGMENT_SHADERS)
-- 
2.14.1

