--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ if(NOT CMAKE_CONFIGURATION_TYPES)
     set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
     # Set the possible values of build type for cmake-gui
     set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "RelWithDebInfo")
-  elseif(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo)$")
+  elseif(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|Gentoo)$")
     message(FATAL_ERROR "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} is not supported, use one of Debug, Release or RelWithDebInfo.")
   endif()
 endif()
@@ -1320,23 +1320,6 @@ install(
     "${MIXXX_INSTALL_LICENSEDIR}"
 )
 
-# Documentation
-install(
-  FILES
-    "${CMAKE_CURRENT_SOURCE_DIR}/README.md"
-    "${CMAKE_CURRENT_SOURCE_DIR}/res/Mixxx-Keyboard-Shortcuts.pdf"
-  DESTINATION
-    "${MIXXX_INSTALL_DOCDIR}"
-)
-if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/res/Mixxx-Manual.pdf")
-  install(
-    FILES
-      "${CMAKE_CURRENT_SOURCE_DIR}/res/Mixxx-Manual.pdf"
-    DESTINATION
-      "${MIXXX_INSTALL_DOCDIR}"
-  )
-endif()
-
 # Additional Linux-only files
 if(UNIX AND NOT APPLE)
   # .desktop file for KDE/GNOME menu
