--- mixxx-release-2.2.4/src/SConscript
+++ mixxx-release-2.2.4/src/SConscript
@@ -405,7 +405,7 @@
                 translations = env.Install(os.path.join(unix_share_path, 'mixxx', 'translations'), translation_files)
                 keyboardmappings = env.Install(os.path.join(unix_share_path, 'mixxx', 'keyboard'), keyboardmappings_files)
                 dotdesktop = env.Install(os.path.join(unix_share_path, 'applications'), dotdesktop_files)
-                dotappstream = env.Install(os.path.join(unix_share_path, 'appdata'), dotappstream_files)
+                dotappstream = env.Install(os.path.join(unix_share_path, 'metadata'), dotappstream_files)
                 docs = env.Install(os.path.join(unix_share_path, 'doc', 'mixxx'), docs_files)
                 icon = env.Install(os.path.join(unix_share_path, 'pixmaps'), icon_files)
 
