--- setup.py	2012-11-29 11:01:22.000000000 -0500
+++ setup.py.wanted	2012-12-27 23:55:56.679344322 -0500
@@ -39,17 +39,6 @@
     return True
 
 
-test_requirements = [
-    'pytest',
-]
-"dependencies for running tests"
-
-if sys.version_info < (2, 7) or (
-        sys.version_info >= (3, 0) and sys.version_info < (3, 1)):
-    # Require unittest2 for Python which doesn't contain the new unittest
-    # module (appears in Python 2.7 and Python 3.1)
-    test_requirements.append('unittest2')
-
 setup_params = dict(
     name = 'keyring',
     version = "1.0",
@@ -74,11 +63,6 @@
     platforms = ["Many"],
     packages = ['keyring', 'keyring.tests', 'keyring.util',
                 'keyring.backends'],
-    extras_require = {'test': test_requirements},
-    tests_require = test_requirements,
-    setup_requires = [
-        'pytest-runner',
-    ],
 )
 
 
