diff --git a/Makefile b/Makefile
index 97a6e1f..6779293 100644
--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 # Use at your own risk. See README file for more details.
 
-JPEG_DIR ?= /opt/libjpeg-turbo
+JPEG_DIR ?= /usr
 JPEG_INCLUDE ?= $(JPEG_DIR)/include
 JPEG_LIB ?= $(JPEG_DIR)/lib`getconf LONG_BIT`
 
@@ -15,14 +15,14 @@ CC    = -std=c++11 -x c++ -Wall -fPIC -no-pie
 GTK   = `pkg-config --libs --cflags gtk+-3.0` `pkg-config --libs x11`
 LIBAV = `pkg-config --libs --cflags libswscale libavutil`
 LIBS  =  -lspeex -lasound -lpthread -lm
-JPEG  = -I$(JPEG_INCLUDE) $(JPEG_LIB)/libturbojpeg.a
+JPEG  = -I$(JPEG_INCLUDE) $(JPEG_LIB)/libturbojpeg.so
 SRC      = src/connection.c src/settings.c src/decoder*.c src/av.c src/usb.c
 
 all: droidcam-cli droidcam
 
 ifeq "$(RELEASE)" "1"
-LIBAV = /usr/lib/x86_64-linux-gnu/libswscale.a /usr/lib/x86_64-linux-gnu/libavutil.a
-SRC  += /usr/lib/x86_64-linux-gnu/libusbmuxd.a /usr/lib/x86_64-linux-gnu/libxml2.a src/libplist-2.0.a
+LIBAV = /usr/lib/x86_64-linux-gnu/libswscale.so /usr/lib/x86_64-linux-gnu/libavutil.so
+SRC  += /usr/lib/x86_64-linux-gnu/libusbmuxd.so /usr/lib/x86_64-linux-gnu/libxml2.so src/libplist-2.0.so
 package: clean all
 	zip -x *.png src/ src/* Makefile -r droidcam_`date +%s`.zip ./*
 
@@ -34,10 +34,10 @@ gresource: .gresource.xml icon2.png
 	glib-compile-resources .gresource.xml --generate-source --target=src/resources.c
 
 droidcam-cli: src/droidcam-cli.c $(SRC)
-	$(GXX) $(CC) $^ $(JPEG) $(LIBAV) $(LIBS) -o droidcam-cli
+	$(GXX) $(CC) $(LDFLAGS) $(CFLAGS) $^ $(JPEG) $(LIBAV) $(LIBS) -o droidcam-cli
 
 droidcam: src/droidcam.c src/resources.c $(SRC)
-	$(GXX) $(CC) $^ $(GTK) $(JPEG) $(LIBAV) $(LIBS) -o droidcam
+	$(GXX) $(CC) $(LDFLAGS) $(CFLAGS) $^ $(GTK) $(JPEG) $(LIBAV) $(LIBS) -o droidcam
 
 clean:
 	rm droidcam || true
