commit 107338c381b6a489c24c2dc9874de028a2cd08a3
Author: Drup <drupyog@zoho.com>
Date:   Fri Jan 16 16:42:33 2015 +0100

    Remove dependency to Lwt_util.

diff --git a/src/lib/eliom_state.server.ml b/src/lib/eliom_state.server.ml
index f7ea0ca..3523fc3 100644
--- a/src/lib/eliom_state.server.ml
+++ b/src/lib/eliom_state.server.ml
@@ -561,7 +561,7 @@ let set_persistent_data_session_group ?set_max
     ?set_max
     (fst sitedata.Eliom_common.max_persistent_data_sessions_per_group)
     c.Eliom_common.pc_value !grp n in
-  lwt () = Lwt_util.iter
+  lwt () = Lwt_list.iter_p
     (Eliommod_persess.close_persistent_state2
        ~scope:(scope:>Eliom_common.user_scope) sitedata None) l in
   grp := n;
diff --git a/src/lib/server/eliommod_gc.ml b/src/lib/server/eliommod_gc.ml
index 5655ddf..6ab298d 100644
--- a/src/lib/server/eliommod_gc.ml
+++ b/src/lib/server/eliommod_gc.ml
@@ -135,7 +135,7 @@ let gc_timeouted_services now tables =
               Lwt.return ()
         end
   in
-  Lwt_util.iter_serial
+  Lwt_list.iter_s
     (fun (_gen, _prio, t) -> empty_one t) tables.Eliom_common.table_services
   >>= fun () ->
   tables.Eliom_common.table_services <-
diff --git a/src/lib/server/eliommod_pagegen.ml b/src/lib/server/eliommod_pagegen.ml
index 1e2d0b4..e0a96a6 100644
--- a/src/lib/server/eliommod_pagegen.ml
+++ b/src/lib/server/eliommod_pagegen.ml
@@ -99,7 +99,6 @@ let update_cookie_table ?now sitedata (ci, sci) =
 
 
   (* Update persistent expiration date, user timeout and value *)
-  (* Lwt_util.iter *)
     Eliom_common.Full_state_name_table.fold
 
       (fun name v thr ->
diff --git a/src/lib/server/eliommod_persess.ml b/src/lib/server/eliommod_persess.ml
index 18c0f5b..9e25906 100644
--- a/src/lib/server/eliommod_persess.ml
+++ b/src/lib/server/eliommod_persess.ml
@@ -152,7 +152,7 @@ let rec find_or_create_persistent_cookie_
       ?set_max:set_max_in_group
       (fst sitedata.Eliom_common.max_persistent_data_sessions_per_group)
       c fullsessgrp >>= fun l ->
-    Lwt_util.iter (close_persistent_state2
+    Lwt_list.iter_p (close_persistent_state2
                      ~scope:(cookie_scope :> Eliom_common.user_scope)
                      sitedata None) l
     >>= fun () ->
diff --git a/src/lib/server/eliommod_sessiongroups.ml b/src/lib/server/eliommod_sessiongroups.ml
index e69a62a..b7eb8f7 100644
--- a/src/lib/server/eliommod_sessiongroups.ml
+++ b/src/lib/server/eliommod_sessiongroups.ml
@@ -475,7 +475,7 @@ module Pers = struct
         (* First we close all sessions in the group *)
 
         find sess_grp >>= fun cl ->
-        Lwt_util.iter
+        Lwt_list.iter_p
           (close_persistent_session2
              ~cookie_level:(match cookie_level with
                | `Client_process _ -> `Client_process | `Session -> `Session)
