--- gmic-1.7.9/CMakeLists.txt
+++ gmic-1.7.9/CMakeLists.txt
@@ -334,11 +334,10 @@
 if(BUILD_MAN)
   file(MAKE_DIRECTORY ${CMAKE_SOURCE_DIR}/man)
   add_custom_command(
-    OUTPUT ${CMAKE_SOURCE_DIR}/man/gmic.1.gz
+    OUTPUT ${CMAKE_SOURCE_DIR}/man/gmic.1
     DEPENDS gmic
     COMMAND ${CMAKE_BINARY_DIR}/gmic -v - ${CMAKE_SOURCE_DIR}/src/gmic_stdlib.gmic raw:${CMAKE_SOURCE_DIR}/src/gmic_stdlib.gmic,uchar -__help man 2> ${CMAKE_SOURCE_DIR}/man/gmic.1
-    COMMAND gzip -f ${CMAKE_SOURCE_DIR}/man/gmic.1
   )
-  add_custom_target(man ALL DEPENDS ${CMAKE_SOURCE_DIR}/man/gmic.1.gz)
-  INSTALL(FILES ${CMAKE_SOURCE_DIR}/man/gmic.1.gz DESTINATION ${CMAKE_INSTALL_MANDIR}/man1/)
+  add_custom_target(man ALL DEPENDS ${CMAKE_SOURCE_DIR}/man/gmic.1)
+  INSTALL(FILES ${CMAKE_SOURCE_DIR}/man/gmic.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1/)
 endif(BUILD_MAN)
