--- a/python/mozbuild/mozbuild/backend/recursivemake.py	2016-04-14 13:55:21.000000000 -0400
+++ b/python/mozbuild/mozbuild/backend/recursivemake.py	2017-02-28 14:43:09.431448016 -0500
@@ -1301,17 +1301,17 @@
 
         for path, files in files.walk():
             target_var = (mozpath.join(target, path)
                           if path else target).replace('/', '_')
             have_objdir_files = False
             for f in files:
                 if not isinstance(f, ObjDirPath):
                     dest = mozpath.join(reltarget, path, mozpath.basename(f))
-                    install_manifest.add_symlink(f.full_path, dest)
+                    install_manifest.add_copy(f.full_path, dest)
                 else:
                     backend_file.write('%s_FILES += %s\n' % (
                         target_var, self._pretty_path(f, backend_file)))
                     have_objdir_files = True
             if have_objdir_files:
                 backend_file.write('%s_DEST := $(DEPTH)/%s\n'
                                    % (target_var,
                                       mozpath.join(target, path)))
