--- a/js/src/js-config.in	2013-03-25 16:34:20.000000000 -0400
+++ b/js/src/js-config.in	2013-08-09 22:15:29.000901763 -0400
@@ -2,7 +2,7 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#filter substitution
+%filter substitution
 
 prefix='@prefix@'
 mozilla_version='@MOZILLA_VERSION@'
--- a/js/src/Makefile.in	2013-03-25 16:34:20.000000000 -0400
+++ b/js/src/Makefile.in	2013-08-09 22:17:20.211903793 -0400
@@ -778,7 +778,7 @@
 
 $(JS_CONFIG_NAME): js-config.in Makefile $(DEPTH)/config/autoconf.mk $(topsrcdir)/config/config.mk $(topsrcdir)/config/rules.mk
 	$(RM) $@.tmp
-	$(PYTHON) $(topsrcdir)/config/Preprocessor.py $(JS_CONFIG_SUBSTITUTIONS) $< > $@.tmp \
+	$(PYTHON) $(topsrcdir)/config/Preprocessor.py --marker="%" $(JS_CONFIG_SUBSTITUTIONS) $< > $@.tmp \
 	&& mv $@.tmp $@ && chmod +x $@
 
 SCRIPTS = $(JS_CONFIG_NAME)
