diff -r 22f2684b1b82 configure.ac
--- a/configure.ac	Mon Sep 23 16:24:03 2024 +0100
+++ b/configure.ac	Sat Sep 28 14:09:56 2024 +0200
@@ -2518,16 +2518,16 @@
 dnl We need a recent version of makeinfo
 AC_MSG_CHECKING([for makeinfo >= 4.12])
 MAKEINFO=
-for prog in `which -a makeinfo`; do
+if prog=`command -v makeinfo`
+then
   mi_verstr=[`$prog --version | sed -n '1s/^.* \([0-9][0-9.]*\)$/\1/p'`]
   mi_major=`echo $mi_verstr | cut -d. -f1`
   mi_minor=`echo $mi_verstr | cut -d. -f2`
   if test "$mi_major" -gt 4 || ( test "$mi_major" -eq 4 && test "$mi_minor" -gt 11 );
   then
     MAKEINFO=$prog
-    break
   fi
-done
+fi
 if test -z $MAKEINFO; then
   AC_MSG_RESULT([no])
   AC_MSG_WARN([Found Makeinfo $mi_verstr.  4.12 or later required.])
