https://bugs.gentoo.org/show_bug.cgi?id=264273

Index: noweb-2.11b/Makefile
===================================================================
--- noweb-2.11b.orig/Makefile
+++ noweb-2.11b/Makefile
@@ -9,14 +9,14 @@ SHELL=/bin/sh
 CINAME=-Nv`echo $(VERSION) | tr . _`
 CIMSG=-f -m'standard checkin preparing to export version $(VERSION)'
 
-source: ;	for i in src; do (cd $$i; make source); done
-www: ;		for i in src/xdoc examples; do (cd $$i; make www); done
+source: ;	for i in src; do (cd $$i; $(MAKE) source); done
+www: ;		for i in src/xdoc examples; do (cd $$i; $(MAKE) www); done
 
 clean:
-	for i in src examples contrib; do (cd $$i; make clean); done
+	for i in src examples contrib; do (cd $$i; $(MAKE) clean); done
 	rm -f nwsrcfilter *~ */*~
 clobber: clean
-	for i in src examples contrib; do (cd $$i; make clobber); done
+	for i in src examples contrib; do (cd $$i; $(MAKE) clobber); done
 
 DATE:
 	(./echo -n "Version $(VERSION) of "; date) > DATE
@@ -51,6 +51,6 @@ emacscheck:
 	diff src/elisp/noweb-mode.el $(HOME)/emacs/noweb-mode.el
 
 checkin:
-	(cd src; make "CINAME=$(CINAME)" "CIMSG=$(CIMSG)" checkin)
+	(cd src; $(MAKE) "CINAME=$(CINAME)" "CIMSG=$(CIMSG)" checkin)
 
 
Index: noweb-2.11b/contrib/norman/Makefile
===================================================================
--- noweb-2.11b.orig/contrib/norman/Makefile
+++ noweb-2.11b/contrib/norman/Makefile
@@ -1,10 +1,10 @@
 LIB=/dev/null	# to be overridden
 DIRS=numarkup
 
-all: ;	for i in $(DIRS); do (cd $$i; make ICONC=$(ICONC) ICONT=$(ICONT) all); done
-install: ;	for i in $(DIRS); do (cd $$i; make LIB=$(LIB) BIN=$(BIN) install); done
-source: ;	for i in $(DIRS); do (cd $$i; make source); done
-clean: ;	for i in $(DIRS); do (cd $$i; make clean); done
+all: ;	for i in $(DIRS); do (cd $$i; $(MAKE) ICONC=$(ICONC) ICONT=$(ICONT) all); done
+install: ;	for i in $(DIRS); do (cd $$i; $(MAKE) LIB=$(LIB) BIN=$(BIN) install); done
+source: ;	for i in $(DIRS); do (cd $$i; $(MAKE) source); done
+clean: ;	for i in $(DIRS); do (cd $$i; $(MAKE) clean); done
 iconlib:	# cheap hack for slackmake
 	true
 
Index: noweb-2.11b/contrib/norman/numarkup/Makefile
===================================================================
--- noweb-2.11b.orig/contrib/norman/numarkup/Makefile
+++ noweb-2.11b/contrib/norman/numarkup/Makefile
@@ -11,11 +11,11 @@ OBJS = main.o pass1.o latex.o input.o sc
 
 all: 
 	noweb -t numarkup.nw
-	make $(TARGET)
+	$(MAKE) $(TARGET)
 
 install:
 	noweb -t numarkup.nw
-	make $(TARGET)
+	$(MAKE) $(TARGET)
 	strip $(TARGET)
 	cp $(TARGET) $(LIB)
 
Index: noweb-2.11b/src/Makefile
===================================================================
--- noweb-2.11b.orig/src/Makefile
+++ noweb-2.11b/src/Makefile
@@ -41,9 +41,9 @@ LIBNAME=$(LIB)
 TEXNAME=$(TEXINPUTS)
 
 all: 
-	cd c; make "CC=$(CC)" "CFLAGS=$(CFLAGS)" all 
-	for i in shell lib xdoc tex; do (cd $$i; make all); done
-	cd $(LIBSRC); make "ICONT=$(ICONT)" "ICONC=$(ICONC)" all
+	cd c; $(MAKE) "CC=$(CC)" "CFLAGS=$(CFLAGS)" all 
+	for i in shell lib xdoc tex; do (cd $$i; $(MAKE) all); done
+	cd $(LIBSRC); $(MAKE) "ICONT=$(ICONT)" "ICONC=$(ICONC)" all
 
 FAQ: FAQ.html
 	sleep 1; html2ascii FAQ.html > FAQ
@@ -101,13 +101,13 @@ install-code: install-shell
 	-mkdir $(BIN) $(LIB) 2>/dev/null
 	true c/nt c/markup c/mnt c/finduses
 	cp c/nt c/markup c/mnt c/finduses $(LIB)
-	cd $(LIBSRC); make ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) install
-	cd lib; make LIB=$(LIB) install
+	cd $(LIBSRC); $(MAKE) ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) install
+	cd lib; $(MAKE) LIB=$(LIB) install
 
 uninstall-code: uninstall-shell
 	rm $(LIB)/nt $(LIB)/markup $(LIB)/mnt $(LIB)/finduses 
-	cd $(LIBSRC); make ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) uninstall
-	cd lib; make LIB=$(LIB) uninstall
+	cd $(LIBSRC); $(MAKE) ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) uninstall
+	cd lib; $(MAKE) LIB=$(LIB) uninstall
 install-man:
 	-mkdir $(MAN) $(MANDIR) $(MAN7DIR) 2>/dev/null
 	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/cpif.1 > $(MANDIR)/cpif.$(MANEXT)
@@ -193,24 +193,24 @@ install-elisp:
 uninstall-elisp:
 	rm -f $(ELISP)/noweb-mode.el
 checkin:
-	for i in lib tex xdoc; do (cd $$i; make "CINAME=$(CINAME)" "CIMSG=$(CIMSG)" checkin); done
+	for i in lib tex xdoc; do (cd $$i; $(MAKE) "CINAME=$(CINAME)" "CIMSG=$(CIMSG)" checkin); done
 	for i in c icon awk; do (cd $$i; ci -l $(CINAME) $(CIMSG) *.nw Makefile); done
 	(cd elisp; ci -l $(CINAME) $(CIMSG) *.el)
 	ci -l $(CINAME) $(CIMSG) Makefile.nw INSTALL INSTALL.DOS README FAQ COPYRIGHT nwmake *.nw
 
 source: FAQ
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make CPIF=">" source); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) CPIF=">" source); done
 	sleep 1
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make touch); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) touch); done
 touch:
 	touch FAQ
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make touch); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) touch); done
 clean:
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make clean); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) clean); done
 	rm -f nwsrcfilter *~ */*~
 
 clobber: clean
-	for i in c lib icon awk xdoc; do (cd $$i; make clobber); done
+	for i in c lib icon awk xdoc; do (cd $$i; $(MAKE) clobber); done
 
 Makefile: Makefile.nw
 	chmod +w Makefile
