--- nemesis-1.4.orig/configure	2006-12-15 13:56:46.000000000 -0700
+++ nemesis-1.4/configure	2006-12-15 14:17:33.000000000 -0700
@@ -4150 +4150 @@
-CPPFLAGS="${CPPFLAGS} `libnet-config --defines`"
+CPPFLAGS="${CPPFLAGS} `libnet-1.0-config --defines`"
@@ -4153 +4153 @@
-if test `libnet-config --cflags | wc -c` = "1"; then
+if test `libnet-1.0-config --cflags | wc -c` = "1"; then
@@ -4157 +4157 @@
-    CPPFLAGS="${CPPFLAGS} `libnet-config --cflags`"
+    CPPFLAGS="${CPPFLAGS} `libnet-1.0-config --cflags`"
@@ -4160,3 +4160,3 @@
-if test `libnet-config --libs | wc -c` = "1"; then
-    { echo "$as_me:$LINENO: WARNING: libnet-config --libs is broken on your system.  If you" >&5
-echo "$as_me: WARNING: libnet-config --libs is broken on your system.  If you" >&2;}
+if test `libnet-1.0-config --libs | wc -c` = "1"; then
+    { echo "$as_me:$LINENO: WARNING: libnet-1.0-config --libs is broken on your system.  If you" >&5
+echo "$as_me: WARNING: libnet-1.0-config --libs is broken on your system.  If you" >&2;}
@@ -4167 +4167 @@
-    LIBS="${LIBS} `libnet-config --libs | sed s/\-lnet//`"
+    LIBS="${LIBS} `libnet-1.0-config --libs | sed s/\-lnet-1.0//`"
@@ -4712,2 +4712,2 @@
-echo "$as_me:$LINENO: checking for libnet_build_ip in -lnet" >&5
-echo $ECHO_N "checking for libnet_build_ip in -lnet... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for libnet_build_ip in -lnet-1.0" >&5
+echo $ECHO_N "checking for libnet_build_ip in -lnet-1.0... $ECHO_C" >&6
@@ -4718 +4718 @@
-LIBS="-lnet  $LIBS"
+LIBS="-lnet-1.0  $LIBS"
@@ -4781 +4781 @@
-  LIBS="-lnet $LIBS"
+  LIBS="-lnet-1.0 $LIBS"
@@ -5263 +5263 @@
-for ac_header in libnet.h
+for ac_header in libnet-1.0.h
@@ -5433 +5433 @@
-    libnet_dir=`libnet-config --cflags | cut -dI -f2`
+    libnet_dir=`libnet-1.0-config --cflags | cut -dI -f2`
@@ -5438 +5438 @@
-    if test -r $i/libnet.h; then
+    if test -r $i/libnet-1.0.h; then
@@ -5444 +5444 @@
-    if eval "grep LIBNET_VERSION $LIBNET_INC_DIR/libnet.h | grep -v 1.0.2a >/dev/null"; then
+    if eval "grep LIBNET_VERSION $LIBNET_INC_DIR/libnet-1.0.h | grep -v 1.0.2a >/dev/null"; then
@@ -5455 +5455 @@
-   echo "  ERROR: unable to find" "libnet 1.0.2a (libnet.h)"
+   echo "  ERROR: unable to find" "libnet 1.0.2a (libnet-1.0.h)"
@@ -5474 +5474 @@
-   echo "  ERROR: unable to find" "libnet 1.0.2a (libnet.h)"
+   echo "  ERROR: unable to find" "libnet 1.0.2a (libnet-1.0.h)"
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-arp.h nemesis-1.4beta3/src/nemesis-arp.h
--- nemesis-1.4beta3.orig/src/nemesis-arp.h	2003-08-19 15:45:45.675664688 -0400
+++ nemesis-1.4beta3/src/nemesis-arp.h	2003-08-19 15:47:07.631205552 -0400
@@ -30 +30 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-dns.h nemesis-1.4beta3/src/nemesis-dns.h
--- nemesis-1.4beta3.orig/src/nemesis-dns.h	2003-08-19 15:45:45.675664688 -0400
+++ nemesis-1.4beta3/src/nemesis-dns.h	2003-08-19 15:47:07.638204488 -0400
@@ -30 +30 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-ethernet.h nemesis-1.4beta3/src/nemesis-ethernet.h
--- nemesis-1.4beta3.orig/src/nemesis-ethernet.h	2003-08-19 15:45:45.676664536 -0400
+++ nemesis-1.4beta3/src/nemesis-ethernet.h	2003-08-19 15:47:07.641204032 -0400
@@ -29 +29 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-functions.c nemesis-1.4beta3/src/nemesis-functions.c
--- nemesis-1.4beta3.orig/src/nemesis-functions.c	2003-08-19 15:45:45.676664536 -0400
+++ nemesis-1.4beta3/src/nemesis-functions.c	2003-08-19 15:47:07.656201752 -0400
@@ -37 +37 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-icmp.h nemesis-1.4beta3/src/nemesis-icmp.h
--- nemesis-1.4beta3.orig/src/nemesis-icmp.h	2003-08-19 15:45:45.676664536 -0400
+++ nemesis-1.4beta3/src/nemesis-icmp.h	2003-08-19 15:47:07.665200384 -0400
@@ -31 +31 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-igmp.h nemesis-1.4beta3/src/nemesis-igmp.h
--- nemesis-1.4beta3.orig/src/nemesis-igmp.h	2003-08-19 15:45:45.676664536 -0400
+++ nemesis-1.4beta3/src/nemesis-igmp.h	2003-08-19 15:47:07.670199624 -0400
@@ -30 +30 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-ip.h nemesis-1.4beta3/src/nemesis-ip.h
--- nemesis-1.4beta3.orig/src/nemesis-ip.h	2003-08-19 15:45:45.676664536 -0400
+++ nemesis-1.4beta3/src/nemesis-ip.h	2003-08-19 15:47:07.677198560 -0400
@@ -26 +26 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-ospf.h nemesis-1.4beta3/src/nemesis-ospf.h
--- nemesis-1.4beta3.orig/src/nemesis-ospf.h	2003-08-19 15:45:45.677664384 -0400
+++ nemesis-1.4beta3/src/nemesis-ospf.h	2003-08-19 15:47:07.684197496 -0400
@@ -33 +33 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-rip.h nemesis-1.4beta3/src/nemesis-rip.h
--- nemesis-1.4beta3.orig/src/nemesis-rip.h	2003-08-19 15:45:45.678664232 -0400
+++ nemesis-1.4beta3/src/nemesis-rip.h	2003-08-19 15:47:07.717192480 -0400
@@ -33 +33 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-tcp.h nemesis-1.4beta3/src/nemesis-tcp.h
--- nemesis-1.4beta3.orig/src/nemesis-tcp.h	2003-08-19 15:45:45.678664232 -0400
+++ nemesis-1.4beta3/src/nemesis-tcp.h	2003-08-19 15:47:07.723191568 -0400
@@ -30 +30 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis-udp.h nemesis-1.4beta3/src/nemesis-udp.h
--- nemesis-1.4beta3.orig/src/nemesis-udp.h	2003-08-19 15:45:45.678664232 -0400
+++ nemesis-1.4beta3/src/nemesis-udp.h	2003-08-19 15:47:07.728190808 -0400
@@ -30 +30 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
diff -r -U 0 nemesis-1.4beta3.orig/src/nemesis.h nemesis-1.4beta3/src/nemesis.h
--- nemesis-1.4beta3.orig/src/nemesis.h	2003-08-19 15:45:45.678664232 -0400
+++ nemesis-1.4beta3/src/nemesis.h	2003-08-19 15:47:07.731190352 -0400
@@ -15 +15 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
--- nemesis-1.4.orig/src/nemesis-printout.c	2006-12-15 13:56:46.000000000 -0700
+++ nemesis-1.4/src/nemesis-printout.c	2006-12-15 14:47:14.000000000 -0700
@@ -34 +34 @@
-#include <libnet.h>
+#include <libnet-1.0.h>
