diff -NrU5 ezbounce-1.04c.original/lib/general.cpp ezbounce-1.04c/lib/general.cpp
--- ezbounce-1.04c.original/lib/general.cpp	2009-08-08 18:50:44.000000000 -0600
+++ ezbounce-1.04c/lib/general.cpp	2009-08-08 18:55:30.000000000 -0600
@@ -366,11 +366,11 @@
     while (*p == sep)
         p++;
     do {
         if (numfound == which && *p)
         {
-            char * next = strchr(p, sep);
+            const char * next = strchr(p, sep);
             if ((get_rest && buffer) || (!next /* && *(p+1) */) )
             {
                 if (buffer)
                     safe_strcpy(buffer, p, maxsize);
                 return 1;
diff -NrU5 ezbounce-1.04c.original/mdidentd/identd.cpp ezbounce-1.04c/mdidentd/identd.cpp
--- ezbounce-1.04c.original/mdidentd/identd.cpp	2009-08-08 18:50:44.000000000 -0600
+++ ezbounce-1.04c/mdidentd/identd.cpp	2009-08-08 19:00:08.000000000 -0600
@@ -739,11 +739,11 @@
     while (*p == sep)
         p++;
     do {
         if (numfound == which && *p)
         {
-            char * next = strchr(p, sep);
+            const char * next = strchr(p, sep);
             if ((get_rest && buffer) || (!next /* && *(p+1) */) )
             {
                 if (buffer)
                     safe_strcpy(buffer, p, maxsize);
                 return 1;
