diff -urN a2ps-4.13.ORIG/config.h.in a2ps-4.13/config.h.in
--- a2ps-4.13.ORIG/config.h.in	2003-09-27 19:33:38.000000000 +0900
+++ a2ps-4.13/config.h.in	2003-09-27 21:25:46.000000000 +0900
@@ -181,6 +181,9 @@
 /* Define to 1 if you have the `rename' function. */
 #undef HAVE_RENAME
 
+/* Define to 1 if you have the <locale.h> header file. */
+#undef HAVE_LOCALE_H
+
 /* Define to 1 if you have the `setlocale' function. */
 #undef HAVE_SETLOCALE
 
diff -urN a2ps-4.13.ORIG/configure.in a2ps-4.13/configure.in
--- a2ps-4.13.ORIG/configure.in	2003-09-27 19:33:38.000000000 +0900
+++ a2ps-4.13/configure.in	2003-09-27 21:26:05.000000000 +0900
@@ -71,7 +71,7 @@
 AC_HEADER_DIRENT
 AC_CHECK_HEADERS(sys/time.h sys/stat.h sys/types.h sys/ioctl.h  \
  math.h string.h strings.h stdlib.h unistd.h stdarg.h pwd.h malloc.h \
- memory.h errno.h fcntl.h stdbool.h fnmatch.h)
+ memory.h errno.h fcntl.h stdbool.h fnmatch.h locale.h)
 
 ## --------------------- ##
 ## Checks for typedefs.  ##
