=== modified file 'lib/xstrrpl.c'
--- lib/xstrrpl.c	2008-04-14 18:04:50 +0000
+++ lib/xstrrpl.c	2008-04-14 18:20:20 +0000
@@ -20,12 +20,10 @@
 
 #include "system.h"
 #include <assert.h>
 #include "xstrrpl.h"
 
-extern char * stpcpy();
-
 /* Perform subsitutions in string.  Result is malloc'd
    E.g., result = xstrrrpl ("1234", subst) gives result = "112333"
    where subst = { {"1", "11"}, {"3", "333"}, { "4", ""}}
    */
 char *

