--- a2ps-4.13/contrib/psset.in.orig	2006-03-16 12:30:48.000000000 +0100
+++ a2ps-4.13/contrib/psset.in		2006-03-16 12:28:14.000000000 +0100
@@ -241,7 +241,7 @@
 # spaces with a `\' too...
 pspagedevicelen=`echo "$pspagedevice" | wc -l`
 pspagedevice=`echo "$pspagedevice" | \
- sed -e "$pspagedevicelen!s/\$/\\\\\\/;s/^/\\\\\\/"`
+ sed -e "$pspagedevicelen!s/\$/\\\\\\/;s/^ /\\\\\\/"`

 case $at in
   0) # Insert last in the Setup, so that we win over other requests.
