fix chmod error
https://bugs.gentoo.org/167670

--- a2ps-4.13.orig/man/Makefile.maint	2002-03-04 20:46:26.000000000 +0200
+++ a2ps-4.13/man/Makefile.maint	2007-02-19 21:50:29.000000000 +0200
@@ -20,7 +20,7 @@
 	    --include=$(basename $@).x \
 	    --include=common.x \
 	    $(executable) > $@-t || exit 1; \
-	  chmod -w $@-t || exit 1; \
+	  chmod a-w $@-t || exit 1; \
 	  rm -f $@ || exit 1; \
 	  mv $@-t $@ || exit 1; \
 	else \
