--- geda-gaf-1.9.2/m4/geda-guile.m4
+++ geda-gaf-1.9.2/m4/geda-guile.m4
@@ -35,8 +35,13 @@
   GUILE_MIN_TEENY=`echo ${GUILE_MIN_VER} | sed -e 's;.*\.;;'`
 
   _found_pkg_config_guile=yes
+  PKG_CHECK_MODULES(GUILE, [guile-2.2 >= $GUILE_MIN_VER],
+                           [GUILE_PKG_NAME=guile-2.2], [_found_pkg_config_guile=no])
+
+  if test "${_found_pkg_config_guile}" = "no" ; then
   PKG_CHECK_MODULES(GUILE, [guile-2.0 >= $GUILE_MIN_VER],
                            [GUILE_PKG_NAME=guile-2.0], [_found_pkg_config_guile=no])
+  fi
 
   if test "${_found_pkg_config_guile}" = "no" ; then
    PKG_CHECK_MODULES(GUILE, [guile-1.8 >= $GUILE_MIN_VER],
