https://github.com/robbat2/readahead-list/pull/1

From 8cfc35923827b6f6e3aff9641aa4cb7c4ac7949c Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Sun, 25 Dec 2022 20:33:44 +0000
Subject: [PATCH 1/6] Modernise autotools

Signed-off-by: Sam James <sam@gentoo.org>
--- a/configure.ac
+++ b/configure.ac
@@ -1,14 +1,14 @@
 # Copyright (C) 2004-2005 Robin Hugh Johnson <robbat2@orbis-terrarum.net>
-#  
+#
 # This file is free software; as a special exception the author gives
-# unlimited permission to copy and/or distribute it, with or without 
+# unlimited permission to copy and/or distribute it, with or without
 # modifications, as long as this notice is preserved.
-# 
+#
 # This program is distributed in the hope that it will be useful, but
 # WITHOUT ANY WARRANTY, to the extent permitted by law; without even the
 # implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
-AC_PREREQ(2.59)
+AC_PREREQ([2.71])
 # version stamp is 0.YYYYMMDD.HHMM in UTC!
 # date +"1.%Y%m%d.%H%M" -u
 AC_INIT([readahead-list],[1.20060421.1016],[Robin H. Johnson robbat2@orbis-terrarum.net],[readahead-list])
@@ -38,11 +38,9 @@ AC_PATH_PROG(SED, sed, sed)
 AC_PATH_PROG(CMP, cmp, cmp)
 
 AC_CANONICAL_HOST
-LF_SET_WARNINGS
 AC_PROG_RANLIB
 
 # Checks for header files.
-AC_HEADER_STDC
 AC_CHECK_HEADERS([ errno.h fcntl.h getopt.h linux/fs.h sched.h stdio.h stdlib.h string.h sys/ioctl.h sys/mman.h sys/syscall.h sys/types.h unistd.h ])
 
 # Checks for typedefs, structures, and compiler characteristics.
-- 
2.39.0

