--- a/libkkc/Makefile.am
+++ b/libkkc/Makefile.am
@@ -49,7 +49,11 @@
 	-DLOCALEDIR=\"$(datadir)/locale\"	\
 	-I $(top_srcdir)			\
 	$(NULL)
-libkkc_la_LIBADD = $(LIBKKC_LIBS) $(top_builddir)/marisa-glib/libmarisa-glib.la
+libkkc_la_LIBADD =					\
+	$(LIBKKC_LIBS)					\
+	$(top_builddir)/marisa-glib/libmarisa-glib.la	\
+	$(MARISA_GLIB_STATIC_DEPENDENCIES)		\
+	$(NULL)
 libkkc_la_LDFLAGS =							\
 	$(CODE_COVERAGE_LDFLAGS)					\
 	-version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"	\
@@ -66,7 +70,6 @@
 libkkc_internals_la_LIBADD =			\
 	$(libkkc_la_LIBADD)			\
 	$(libkkc_la_OBJECTS)			\
-	$(MARISA_GLIB_STATIC_DEPENDENCIES)	\
 	$(NULL)
 libkkc_internals_la_LDFLAGS =			\
 	$(CODE_COVERAGE_LDFLAGS)		\
