--- ntp-4.2.8p12.orig/include/libssl_compat.h	2018-12-08 01:14:58.434733497 +0000
+++ ntp-4.2.8p12/include/libssl_compat.h	2018-12-08 01:42:04.479474757 +0000
@@ -107,10 +107,13 @@
 
 #define OpenSSL_version_num	SSLeay
 #define OpenSSL_version		SSLeay_version
+
+# if !defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER < 0x20800000L
 #define X509_get0_notBefore	X509_get_notBefore
 #define X509_getm_notBefore	X509_get_notBefore
 #define X509_get0_notAfter	X509_get_notAfter
 #define X509_getm_notAfter	X509_get_notAfter
+#endif /* !defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER < 0x20800000L */
 
 /* ----------------------------------------------------------------- */
 #endif /* OPENSSL_VERSION_NUMBER < v1.1.0 */
