From 45fb2a39195b5b31f61f376323ff78ab40940760 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sat, 8 Dec 2018 16:19:35 +0100
Subject: [PATCH] Fix unwanted automagic activation of aalib support

---
 configure.ac | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index a6068c9..c19fed9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -306,8 +306,9 @@ aa_msg="no (http://www.sourceforge.net/projects/aa-project)"
 try_aa=:
 have_aa=false
 AC_ARG_WITH([aalib],AS_HELP_STRING([--with-aalib],[Do use aalib for preview ASCII rendering]),[
-        if test x$withval = xyes; then
-                try_aa=:
+        if test x$withval = xno; then
+                try_aa=false
+                aa_msg="no (not requested)"
         fi
 ])
 if $try_aa; then
@@ -323,8 +324,6 @@ if $try_aa; then
                         AA_LIBS="-laa"
                 fi
 	])
-else
-        aa_msg="no (not requested)"
 fi
 AM_CONDITIONAL([HAVE_AA], [$have_aa])
 AC_SUBST([AA_LIBS])
-- 
2.19.2

