--- backintime-1.0.24-r2/common/backintime
+++ backintime-1.0.24-r2/common/backintime
@@ -17,15 +17,14 @@
 #    with this program; if not, write to the Free Software Foundation, Inc.,
 #    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-if [ -f backintime.py ]; then
-	APP_PATH="."
-else
-	APP_PATH="/usr/share/backintime/common"
-fi
+APP_PATH="/usr/share/backintime/common"
 
 #starting a new ssh-agent all the time is just a workaround for
 #https://bugs.launchpad.net/ubuntu/+source/openssh/+bug/841672
 #normally this should only be necessary if run as cronjob
 #and the user is not logged in
-ssh-agent python $APP_PATH/backintime.py "$@"
+if [ "x$SSH_AUTH_SOCK" = "x" ]; then
+	eval "$(ssh-agent)"
+fi
 
+python2 $APP_PATH/backintime.py "$@"
--- backintime-1.0.24-r2/gnome/backintime-gnome
+++ backintime-1.0.24-r2/gnome/backintime-gnome
@@ -17,14 +17,13 @@
 #    with this program; if not, write to the Free Software Foundation, Inc.,
 #    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-if [ -f app.py ]; then
-	APP_PATH="."
-else
-	APP_PATH="/usr/share/backintime/gnome"
-fi
+APP_PATH="/usr/share/backintime/gnome"
 
 #starting a new ssh-agent all the time is just a workaround for
 #https://bugs.launchpad.net/ubuntu/+source/openssh/+bug/841672
 #normally we don't need to start ssh-agent for backintime-gnome
-ssh-agent python ${APP_PATH}/app.py "$@"
+if [ "x$SSH_AUTH_SOCK" = "x" ]; then
+	eval "$(ssh-agent)"
+fi
 
+python2 ${APP_PATH}/app.py "$@"
--- backintime-1.0.24-r2/kde4/backintime-kde4
+++ backintime-1.0.24-r2/kde4/backintime-kde4
@@ -17,15 +17,10 @@
 #    with this program; if not, write to the Free Software Foundation, Inc.,
 #    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-if [ -f app.py ]; then
-	APP_PATH="."
-else
-	APP_PATH="/usr/share/backintime/kde4"
-fi
+APP_PATH="/usr/share/backintime/kde4"
 
 if [ "x$SSH_AUTH_SOCK" = "x" ]; then
         eval "$(ssh-agent)"
 fi
 
-python ${APP_PATH}/app.py "$@"
-
+python2 ${APP_PATH}/app.py "$@"
