From 0005710eccdf702915ec776830c79102a6b66f1b Mon Sep 17 00:00:00 2001
From: Jonathan Riddell <jr@jriddell.org>
Date: Tue, 25 Feb 2025 16:45:29 +0000
Subject: [PATCH] update version for new release

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6ab65f06bf8..61932a493c1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.16)
 
-set(PROJECT_VERSION "6.3.2") # Handled by release scripts
+set(PROJECT_VERSION "6.3.3") # Handled by release scripts
 project(KWin VERSION ${PROJECT_VERSION})
 
 set(CMAKE_C_STANDARD 99)
-- 
GitLab

