Disable the effect video button if gstreamer is missing.

diff --git a/kcmkwin/kwincompositing/qml/Effect.qml b/kcmkwin/kwincompositing/qml/Effect.qml
index fcd5e8a..2331da2 100644
--- a/kcmkwin/kwincompositing/qml/Effect.qml
+++ b/kcmkwin/kwincompositing/qml/Effect.qml
@@ -144,7 +144,7 @@ Rectangle {
 
         Button {
             id: videoButton
-            visible: model.VideoRole.toString() !== ""
+            visible: false
             iconName: "video"
             onClicked: videoItem.showHide()
         }
