diff -uprN src.orig/test/net/sf/cglib/reflect/TestFastClass.java src/test/net/sf/cglib/reflect/TestFastClass.java
--- src.orig/test/net/sf/cglib/reflect/TestFastClass.java	2005-01-05 19:33:15.945618576 +0100
+++ src/test/net/sf/cglib/reflect/TestFastClass.java	2005-01-05 19:34:21.957583232 +0100
@@ -26,7 +26,7 @@ import net.sf.cglib.core.DefaultGenerato
 import net.sf.cglib.transform.ClassTransformerTee;
 import net.sf.cglib.transform.NullClassVisitor;
 import net.sf.cglib.transform.TransformingClassGenerator;
-import org.objectweb.asm.util.DumpClassVisitor;
+import org.objectweb.asm.util.ASMifierClassVisitor;
 
 public class TestFastClass extends net.sf.cglib.CodeGenTestCase {
     public static class Simple {
