From 44015c26ba215f955b1e653f9c8f3c894a532707 Mon Sep 17 00:00:00 2001
From: Jeffrey Walton <noloader@gmail.com>
Date: Tue, 9 Jun 2015 09:49:02 -0400
Subject: [PATCH] Cleared ill-formed conversion in C++11

---
 wake.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wake.cpp b/wake.cpp
index c34165b..e0f194e 100644
--- a/wake.cpp
+++ b/wake.cpp
@@ -23,7 +23,7 @@ void WAKE_Base::GenKey(word32 k0, word32 k1, word32 k2, word32 k3)
 	signed int x, z, p;	
 	// x and z were declared as "long" in Wheeler's paper, which is a signed type. I don't know if that was intentional, but it's too late to change it now. -- Wei 7/4/2010
 	CRYPTOPP_COMPILE_ASSERT(sizeof(x) == 4);
-	static int tt[10]= {
+	static unsigned int tt[10]= {
 		0x726a8f3b,								 // table
 		0xe69a3b5c,
 		0xd3c71fe5,
