diff -Nur statcvs-0.4.0/src/net/sf/statcvs/output/RepoMapPageMaker.java statcvs-0.4.0_patched/src/net/sf/statcvs/output/RepoMapPageMaker.java
--- statcvs-0.4.0/src/net/sf/statcvs/output/RepoMapPageMaker.java	2008-04-14 21:35:00.000000000 +0300
+++ statcvs-0.4.0_patched/src/net/sf/statcvs/output/RepoMapPageMaker.java	2008-09-20 20:46:05.000000000 +0300
@@ -20,7 +20,8 @@
 package net.sf.statcvs.output;
 
 import java.io.BufferedWriter;
-import java.io.File;
+import java.io.File;
+import java.io.FileInputStream;
 import java.io.FileWriter;
 import java.io.IOException;
 import java.io.InputStream;
@@ -92,7 +93,7 @@
     }
 
     private String getApplet() {
-        return "<applet archive=\"./" + Messages.getString("JTREEMAP_JAR") + "\" code=\"net.sf.jtreemap.swing.example.JTreeMapAppletExample\""
+        return "<applet archive=\"./jtreemap.jar\" code=\"net.sf.jtreemap.swing.example.JTreeMapAppletExample\""
                 + " width=\"940\" height=\"600\"><param name=\"dataFile\" value=\"" + REPO_FILE + "\"/>" + "<param name=\"viewTree\" value=\"true\"/>"
                 + "<param name=\"showWeight\" value=\"true\"/>" + "<param name=\"valuePrefix\" value=\"Change:\"/>"
                 + "<param name=\"weightPrefix\" value=\"LOC:\"/>" + "<param name=\"dataFileType\" value=\"xml\"/>"
@@ -102,7 +103,7 @@
     private void buildXmlForJTreeMap() {
         BufferedWriter out = null;
         try {
-            copyJar(Messages.getString("JTREEMAP_JAR"));
+            copyJar("jtreemap.jar");
             out = new BufferedWriter(new FileWriter(ConfigurationOptions.getOutputDir() + REPO_FILE));
             out.write("<?xml version='1.0' encoding='ISO-8859-1'?>\n");
             // out.append("<!DOCTYPE root SYSTEM \"TreeMap.dtd\" >\n");
@@ -129,7 +130,7 @@
     private void copyJar(final String jtreemapJar) throws IOException {
         InputStream stream = null;
         try {
-            stream = RepoMapPageMaker.class.getResourceAsStream(WEB_FILE_PATH + jtreemapJar);
+            stream = new FileInputStream("/usr/share/jtreemap/lib/" + jtreemapJar);
             if (stream != null) {
                 FileUtils.copyFile(stream, new File(ConfigurationOptions.getOutputDir() + jtreemapJar));
             } else {
