diff -Naur a/src/games/strategy/net/UniversalPlugAndPlayHelper.java b/src/games/strategy/net/UniversalPlugAndPlayHelper.java
--- a/src/games/strategy/net/UniversalPlugAndPlayHelper.java	2015-08-19 16:35:06.000000000 +0100
+++ b/src/games/strategy/net/UniversalPlugAndPlayHelper.java	2016-10-11 22:10:37.287834187 +0100
@@ -166,8 +166,8 @@
     System.out.println("To " + m_local.getHostAddress() + ":" + internalPort);
     boolean mapped = false;
     try {
-      mapped = m_device.addPortMapping("TripleA Game Hosting", "TCP", null, externalPort, m_local.getHostAddress(),
-          internalPort, 0);
+      mapped = m_device.addPortMapping("TripleA Game Hosting", null, internalPort, externalPort,
+          m_local.getHostAddress(), 0, "TCP");
     } catch (final IOException e) {
       System.out.println("Port Mapping Failed! Please try to Forward Ports manually! \r\n " + e.getMessage());
       return "Port Mapping Failed! Please try to Forward Ports manually! \r\n " + e.getMessage();
