From 42008129f1659bc34c454ce14632fd279679f556 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Mon, 22 Oct 2018 21:44:33 +0200
Subject: [PATCH] xembed-sni-proxy: Allow standalone build

---
 CMakeLists.txt | 35 +++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c2f67ac6..30b2c1b5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,3 +1,34 @@
+project(xembed-sni-proxy)
+
+if(${CMAKE_SOURCE_DIR} STREQUAL ${xembed-sni-proxy_SOURCE_DIR})
+    set(PROJECT_VERSION "GENTOO_PV")
+    set(PROJECT_VERSION_MAJOR 5)
+
+    cmake_minimum_required(VERSION 3.0)
+
+    set(QT_MIN_VERSION "5.11.0")
+    set(KF5_MIN_VERSION "5.54.0")
+    find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS Core DBus X11Extras)
+    find_package(ECM ${KF5_MIN_VERSION} REQUIRED NO_MODULE)
+    set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
+
+    include(KDEInstallDirs)
+    include(KDECMakeSettings)
+    include(KDECompilerSettings NO_POLICY_SCOPE)
+    include(FeatureSummary)
+    include(ECMQtDeclareLoggingCategory)
+
+    find_package(KF5WindowSystem ${KF5_MIN_VERSION} REQUIRED)
+    find_package(X11 REQUIRED)
+    find_package(XCB MODULE REQUIRED COMPONENTS XCB RANDR)
+    set_package_properties(XCB PROPERTIES TYPE REQUIRED)
+
+    configure_file(../config-X11.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-X11.h)
+else()
+    set(PROJECT_VERSION "5.14.90")
+    set(PROJECT_VERSION_MAJOR 5)
+endif()
+
 add_definitions(-DQT_NO_CAST_TO_ASCII
 -DQT_NO_CAST_FROM_ASCII
 -DQT_NO_URL_CAST_FROM_STRING
@@ -64,3 +95,7 @@ target_link_libraries(xembedsniproxy
 install(TARGETS xembedsniproxy ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES xembedsniproxy.desktop DESTINATION ${KDE_INSTALL_AUTOSTARTDIR})
 
+if ("${CMAKE_BINARY_DIR}" STREQUAL "${CMAKE_CURRENT_BINARY_DIR}")
+    feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
+endif()
+
-- 
2.19.1

