--- xpra-0.7.1/setup.py
+++ xpra-0.7.1/setup.py
@@ -135,7 +135,7 @@
     #then expects to find the cython modules!? ie:
     #python2.7 setup.py build -b build-2.7 install --no-compile --root=/var/tmp/portage/x11-wm/xpra-0.7.0/temp/images/2.7
     if "--no-compile" in sys.argv and not ("build" in sys.argv and "install" in sys.argv):
-        return
+        pass
     global ext_modules, cmdclass
     cython_version_check(min_version)
     from Cython.Distutils import build_ext
