diff -ru bloboats-1.0.2.orig/src/menu.cpp bloboats-1.0.2/src/menu.cpp
--- bloboats-1.0.2.orig/src/menu.cpp	2010-08-25 17:55:26.000000000 -0400
+++ bloboats-1.0.2/src/menu.cpp	2010-11-28 01:19:57.664406227 -0500
@@ -118,7 +118,7 @@
 			bool used=false;
 
 			if (fgets(nick, NICK_MAXLEN+2, fp)) {
-				if (nick && strlen(nick)>1 ) {
+				if (strlen(nick)>1 ) {
 					nick[strlen(nick)-1]='\0';
 					used=true;
 					Player[i].setnick(nick);
diff -ru bloboats-1.0.2.orig/src/path.cpp bloboats-1.0.2/src/path.cpp
--- bloboats-1.0.2.orig/src/path.cpp	2006-08-02 18:05:36.000000000 -0400
+++ bloboats-1.0.2/src/path.cpp	2010-11-28 01:19:19.385415683 -0500
@@ -111,7 +111,7 @@
 			}
 	
 			if (type>0) {
-				if (d2) {
+				if (*d2) {
 					char *ps = new char[p.size()+1];
 					strncpy (ps, p.c_str(), p.size()+1);
 		
