--- scorched.old/src/common/porting/windows.h
+++ scorched/src/common/porting/windows.h
@@ -9,6 +9,7 @@
 #include <ctype.h>
 #include <common/Defines.h>
 
+#ifdef _WIN32
 typedef char * 		LPSTR;
 typedef const char *	LPCTSTR;
 typedef unsigned int 	DWORD;
@@ -27,6 +28,7 @@
 typedef int 	LRESULT;
 typedef void *	HDC;
 typedef void *  HGLRC;
+#endif  /* _WIN32 */
 
 #define _strnicmp(a,b,c)	strncasecmp(a,b,c)
 #define stricmp(a,b)		strcasecmp(a,b)
