Index: gz-common-ignition-common3_3.17.0/profiler/include/gz/common/Profiler.hh
===================================================================
--- gz-common-ignition-common3_3.17.0.orig/profiler/include/gz/common/Profiler.hh
+++ gz-common-ignition-common3_3.17.0/profiler/include/gz/common/Profiler.hh
@@ -20,6 +20,7 @@
 
 #include <memory>
 #include <string>
+#include <cstdint> // uint32
 
 #include <gz/common/profiler/Export.hh>
 #include <gz/common/SingletonT.hh>
