--- lxd/rsync.go.orig	2015-09-17 21:00:29.290772099 -0500
+++ lxd/rsync.go	2015-09-17 21:00:54.737770359 -0500
@@ -73,7 +73,7 @@
 	 * command (i.e. the command to run on --server). However, we're
 	 * hardcoding that at the other end, so we can just ignore it.
 	 */
-	rsyncCmd := fmt.Sprintf("sh -c \"nc -U %s\"", f.Name())
+	rsyncCmd := fmt.Sprintf("sh -c \"nc.openbsd -U %s\"", f.Name())
 	cmd := exec.Command("rsync", "-arvP", "--devices", "--partial", path, "localhost:/tmp/foo", "-e", rsyncCmd)
 	if err := cmd.Start(); err != nil {
 		return nil, nil, err
