diff -aur ap-utils-1.4-orig/lib/wlan.c ap-utils-1.4/lib/wlan.c
--- ap-utils-1.4-orig/lib/wlan.c	2003-10-31 12:15:00.000000000 -0500
+++ ap-utils-1.4/lib/wlan.c	2004-02-24 23:00:25.214209480 -0500
@@ -140,6 +140,7 @@
     _("Short"), _("Long")};
     short __rates[4] = { 2, 4, 11, 22 };
     int i, c = 0;
+    char ch;
     unsigned int ap_regdomain;
 
     if (ap_type == ATMEL12350) {
@@ -508,10 +509,10 @@
 		i++;
 	    }
 	    if (m_essid) {
-		c = strlen(domain);
+		c = ch = strlen(domain);
 		varbinds[i].oid = operESSIDLength;
 		varbinds[i].len_oid = sizeof(operESSIDLength);
-		varbinds[i].value = (char *)&c;
+		varbinds[i].value = (char *)&ch;
 		varbinds[i].len_val = 1;
 		varbinds[i].type = INT_VALUE;
 		i++;
