--- meme_3.5.4/website/cgi-bin/process_request.pl	2006-09-21 19:46:29.000000000 +0000
+++ trunk/website/cgi-bin/process_request.pl	2007-05-30 01:35:02.000000000 +0000
@@ -1,6 +1,6 @@
 #!@WHICHPERL@
 ##
-## $Id: process_request.pl 1339 2006-09-21 19:46:28Z tbailey $
+## $Id: process_request.pl 1807 2007-05-30 01:34:31Z tbailey $
 ##
 ## $Log: process_request.pl,v $
 ## Revision 1.6.6.1  2006/02/16 23:22:35  nadya
@@ -55,7 +55,8 @@
 $blocks_url = "http://blocks.fhcrc.org/blocks-bin/process_blocks.pl";
 #
 # You can change this if you wish to use a different JASPAR server
-$jaspar_root = "http://mordor.cgb.ki.se";
+#$jaspar_root = "http://mordor.cgb.ki.se";
+$jaspar_root = "http://asp.ii.uib.no:8090";
 $jaspar_url = "$jaspar_root/cgi-bin/jaspar2005/jaspar_db.pl";
 #
 # You can change this if you wish to use a different Meta-MEME server
@@ -221,7 +222,7 @@
 
   $fasta = "";					# return value
   @lines = split(/\n/, $block); 		# split block into lines
-  for ($i = 1; $i<$#lines; $i++) {
+  for ($i = 2; $i<$#lines; $i++) {
     last if $lines[$i] =~ /^\/\//;
     @words = split(/\s+/, $lines[$i]);		# split line into words
     # get sequence line
@@ -239,7 +240,7 @@
 
   $fasta = "";					# return value
   @lines = split(/\n/, $block); 		# split block into lines
-  for ($i = 1; $i<$#lines; $i++) {
+  for ($i = 2; $i<$#lines; $i++) {
     last if $lines[$i] =~ /^\/\//;
     @words = split(/\s+/, $lines[$i]);		# split line into words
     # get id line and sequence line
@@ -311,7 +312,8 @@
   $content = $request->content;
 
   # fix bug in JASPAR output; add database field to view buttons
-  $content =~ s/rm=present/rm=present&db=$sub_db/g;
+  # remove fix: JASPAR fixed the bug
+  # $content =~ s/rm=present/rm=present&db=$sub_db/g;
 
   # display the page
   print $content;
--- meme_3.5.4/src/ureadseq.c	2006-09-21 19:46:28.000000000 +0000
+++ trunk/src/ureadseq.c	2007-05-18 08:18:05.000000000 +0000
@@ -1,5 +1,5 @@
 /*
- * $Id: ureadseq.c 1339 2006-09-21 19:46:28Z tbailey $
+ * $Id: ureadseq.c 1787 2007-05-18 08:17:28Z tbailey $
  * 
  * $Log$
  * Revision 1.2  2006/03/08 20:50:11  nadya
@@ -206,10 +206,9 @@
 
 Local void addinfo(char *s, struct ReadSeqVars *V)
 {
-  char s2[256], *si;
+  char *si = (char *) malloc((strlen(s) + 40) * sizeof(char));
   boolean saveadd;
 
-  si = s2;
   while (*s == ' ') s++;
   sprintf(si, " %d)  %s\n", V->nseq, s);
 
@@ -217,6 +216,7 @@
   V->addit = true;
   V->isseqchar = isAnyChar;
   addseq( si, V);
+  free(si);
   V->addit = saveadd;
   V->isseqchar = isSeqChar;
 }
@@ -966,7 +966,6 @@
     } while ((l == 0) && !feof(V->f));
 
     if (feof(V->f)) V->err = eNoData;
-
     else switch (format_) {
       case kPlain : readPlain(V); break;
       case kIG    : readIG(V); break;
@@ -1181,7 +1180,7 @@
   int       nlines= 0, k=0, splen= 0, otherlines= 0, aminolines= 0, dnalines= 0;
   char      sp[MAXLINE];
   long      linestart=0;
-  int     maxlines2check=500;
+  int     maxlines2check=5000;
 
 #define ReadOneLine(sp)   \
   { done |= (feof(fseq)); \
--- meme_3.5.4/src/include/ureadseq.h	2006-09-21 19:46:28.000000000 +0000
+++ trunk/src/ureadseq.h	2007-05-18 08:18:05.000000000 +0000
@@ -1,5 +1,5 @@
 /*
- * $Id: ureadseq.h 1339 2006-09-21 19:46:28Z tbailey $
+ * $Id: ureadseq.h 1048 2006-07-06 20:07:44Z cegrant $
  * 
  * $Log$
  * Revision 1.1  2005/07/29 19:12:07  nadya
@@ -15,7 +15,7 @@
 #include "config.h"
 #include "macros.h"
 
-#define MAXLINE 1024
+#define MAXLINE 10240
 
 typedef char  boolean;
 #define NEWLINE         '\n'
--- meme_3.5.4/src/read_seq_file.c	2006-09-21 19:46:28.000000000 +0000
+++ ./read_seq_file.c	2007-05-18 06:51:06.000000000 +0000
@@ -433,6 +433,7 @@
       name[i++] = c;			/* non-blank: add to name */
     }
   }
+  Resize(name, i+1, char);
   name[i] = '\0';
 
   /* read in description */
--- meme_3.5.4/website/html/meme-install.html	2006-09-21 19:46:29.000000000 +0000
+++ fred/meme-install.html	2007-05-30 02:00:56.000000000 +0000
@@ -208,42 +208,39 @@
 <h2><a name="patch">Getting and installing the patches</a></h2>
 <p>The distribution may have patches associated with it. They are available 
 from <a href="http://meme.nbcr.net/downloads/">http://meme.nbcr.net/downloads/</a>.
-The patch file name is <span class="command">filename.VERSION.patch</span>. In addition, a
-patched file is distributed as well and can be used as a drop-in substitute
-for the original file. The drop-in file is <span class="command">filename.VERSION</span>.
-It is necessary to download only one of the two files depending on the method used for
-patching. All patches for a specific version should be installed. The list
-below provides instructions for installation of availble patches for specific
-version.
+Patch files are located in a directory named
+<span class="command">VERSION.patches</span>,
+for example, <span class="command">meme_3.5.4.patches</span>.
+Patch file have names like:
+<span class="command">VERSION.patch_SERIAL_NO</span>, for example, <span class="command">meme_3.5.4.patch_3</span>.
+</p> <p> To install a patch, download the patch file from the URL given above.
+Then perform the following commands to install it:
+        <br><span class="command">$ cp PATCH_FILE VERSION </span>         <br><span class="command">$ cd VERSION </span>
+        <br><span class="command">$ patch -p1 < PATCH_FILE </span>
+        <br><span class="command">$ make install </span>
+        <br><span class="command">$ make test </span>
+</p>
+<p>
+For example, to install the first patch to version meme_3.5.4, you would perform the following commands:
+        <br><span class="command">$ cp meme_3.5.4.patch_1 meme_3.5.4 </span>
+        <br><span class="command">$ cd meme_3.5.4 </span>
+        <br><span class="command">$ patch -p1 < meme_3.5.4.patch_1 </span>
+        <br><span class="command">$ make install </span>
+        <br><span class="command">$ make test </span>
+</p>
+<p>
+You must install all of the patches for a specific version in serial
+number order.  For example, if you wish to install patch number 3,
+you must first have installed patches number 1 and 2 for that version.
+This is easy to do.  Just download all the patches for your current
+version, copy them to your current versions's directory, and then
+install them in order by repeating the <span class="command">patch</span>,
+command above, with each patch file.  You only need to run the
+<span class="command">install</span> and
+<span class="command">make test</span> commands once, after
+the last <span class="command">patch</patch> command.
 </p>
 
-<center>
-<table class="large" border=1  cellspacing=0>
-<tr>
-  <th width=7% class="head">Version</th>
-  <th width=25% class="head">Patch list</th>
-  <th class="head">Installation</th>
-</tr>
-<tr>
-  <td align=center>3.5.0</td>
-  <td align=center>mast-client.txt</td>
-  <td>
-  <ol>
-  <li>If downloaded a patch file <code>mast-client.txt.3.5.0.patch</code>:
-  <br><span class="command"># cp mast-client.txt.3.5.0 meme_3.5.0/scripts/</span>
-  <br><span class="command"># cd meme_3.5.0/scripts/</span>
-  <br><span class="command"># patch -p0 &lt; mast-client.txt.3.5.0.patch </span>
-  <br>
-  </li>
-  <li>If downloaded a patched file <code>mast-client.txt.3.5.0</code>:
-  <br><span class="command"># cp mast-client.txt.3.5.0 meme_3.5.0/scripts/mast-client.txt</span>
-  </li>
-  </ol>
-  </td>
-</tr>
-</table>
-</center>
-
 <p>
 <p>[<a href="#top"> Top </a>]</p>
 
