From 2b809c68d40a3d0af2da668bad3f4d9ced86f56e Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Thu, 10 Mar 2016 20:26:57 +0100
Subject: KF5DocTools is optional, so add doc subdir conditionally

REVIEW: 127336
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d88174a..da7458f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,7 +60,10 @@ file (GLOB SymbolEditor_UI ${CMAKE_CURRENT_SOURCE_DIR}/ui/*.ui)
 ki18n_wrap_ui (SymbolEditor_SRCS ${SymbolEditor_UI})
 
 add_subdirectory (icons)
-add_subdirectory (doc)
+
+if (KF5DocTools_FOUND)
+    add_subdirectory (doc)
+endif (KF5DocTools_FOUND)
 
 kconfig_add_kcfg_files (SymbolEditor_SRCS SymbolEditor.kcfgc)
 
-- 
cgit v1.1
