diff -Naur babel-1.6/Makefile babel-1.6.new/Makefile
--- babel-1.6/Makefile	1997-01-21 10:53:05.000000000 -0500
+++ babel-1.6.new/Makefile	2008-10-19 10:41:42.000000000 -0400
@@ -1,12 +1,16 @@
-CFLAGS        = -O 
+#Commented out to pull CFLAGS from make.conf
+#CFLAGS        = -O 
 
 LIBFLAGS      = -O -DLIBRARY 
 
-CC	      = cc
+#Probably not necessary to specify gcc
+#CC	      = gcc
 
 PROGRAM	      = babel
 
-LDFLAGS	      =
+DEST          = ${DESTDIR}/$(PROGRAM)
+
+#LDFLAGS	      =
 
 LIBS	      =
 
@@ -334,7 +338,8 @@
 index:;		ctags -wx $(HDRS) $(SRCS)
 
 install:	$(PROGRAM)
-		install -s $(PROGRAM) $(DEST)
+		install -d ${DESTDIR}
+		install $(PROGRAM) $(DEST)
 
 print:;		$(PRINT) $(HDRS) $(SRCS)
 
