diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-01-08 22:16:00.493079361 +0800
+++ b/CMakeLists.txt	2020-01-08 22:16:44.790285583 +0800
@@ -252,6 +252,8 @@
 
 enable_testing()
 
+INCLUDE ( GNUInstallDirs )
+
 # Add our cmake modules
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
diff -Naur a/openvdb/CMakeLists.txt b/openvdb/CMakeLists.txt
--- a/openvdb/CMakeLists.txt	2020-01-08 22:16:07.362646127 +0800
+++ b/openvdb/CMakeLists.txt	2020-01-08 22:19:54.638308684 +0800
@@ -492,7 +492,7 @@
   install(TARGETS
     openvdb_static
     DESTINATION
-    lib
+    ${CMAKE_INSTALL_LIBDIR}
   )
 endif()
 
@@ -500,14 +500,14 @@
   if(WIN32)
     install(TARGETS
       openvdb_shared
-      RUNTIME DESTINATION bin COMPONENT Runtime
-      ARCHIVE DESTINATION lib COMPONENT Runtime
+      RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT Runtime
+      ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Runtime
     )
   else()
     install(TARGETS
       openvdb_shared
       DESTINATION
-      lib
+      ${CMAKE_INSTALL_LIBDIR}
     )
   endif()
 endif()
diff -Naur a/openvdb/python/CMakeLists.txt b/openvdb/python/CMakeLists.txt
--- a/openvdb/python/CMakeLists.txt	2020-01-08 22:16:14.452199013 +0800
+++ b/openvdb/python/CMakeLists.txt	2020-01-08 22:21:45.311324663 +0800
@@ -232,7 +232,7 @@
   ${PYOPENVDB_INSTALL_DIRECTORY}
 )
 
-install(FILES ${PYTHON_PUBLIC_INCLUDE_NAMES} DESTINATION include/openvdb/python)
+install(FILES ${PYTHON_PUBLIC_INCLUDE_NAMES} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/openvdb/python)
 
 # pytest
 
