--- a/filters/recoll-we-move-files.py
+++ b/filters/recoll-we-move-files.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python2
 # Copyright (C) 2017 J.F.Dockes
 #   This program is free software; you can redistribute it and/or modify
 #   it under the terms of the GNU General Public License as published by
@@ -42,6 +42,8 @@ try:
 except:
     import rclconfig
 
+from __future__ import print_function
+
 verbosity = 0
 def logdeb(s):
     if verbosity >= 4:
