diff -ur tls1.5_org/pkgIndex.tcl.in tls1.5/pkgIndex.tcl.in
--- tls1.5_org/pkgIndex.tcl.in	2007-04-05 23:09:11.000000000 +0200
+++ tls1.5/pkgIndex.tcl.in	2007-04-05 23:09:40.000000000 +0200
@@ -5,5 +5,5 @@
 #    replace the original which didn't include the commands from "tls.tcl".
 #
 
-package ifneeded tls 1.5 "[list load [file join $dir @RELPATH@ @tls_LIB_FILE@] ] ; [list source [file join $dir tls.tcl] ]"
+package ifneeded tls 1.50 "[list load [file join $dir @RELPATH@ @tls_LIB_FILE@] ] ; [list source [file join $dir tls.tcl] ]"
 
