diff -ur sbin.orig/ping/Makefile sbin.setXid/ping/Makefile
--- sbin.orig/ping/Makefile	Thu Apr  7 20:27:57 2005
+++ sbin.setXid/ping/Makefile	Thu Apr  7 20:42:02 2005
@@ -18,4 +18,6 @@
 LDADD+=	-lipsec
 .endif
 
+LDFLAGS+= -Wl,-z,now
+
 .include <bsd.prog.mk>
diff -ur sbin.orig/shutdown/Makefile sbin.setXid/shutdown/Makefile
--- sbin.orig/shutdown/Makefile	Thu Apr  7 20:27:57 2005
+++ sbin.setXid/shutdown/Makefile	Thu Apr  7 20:42:17 2005
@@ -7,4 +7,6 @@
 BINGRP=	operator
 BINMODE=4550
 
+LDFLAGS+= -Wl,-z,now
+
 .include <bsd.prog.mk>
