 tclxslt-libxslt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tclxslt-libxslt.c b/tclxslt-libxslt.c
index d943bf7..d18f645 100644
--- a/tclxslt-libxslt.c
+++ b/tclxslt-libxslt.c
@@ -1506,7 +1506,7 @@ TclXSLTExtFunction(xpathCtxt, nargs)
     obj = TclXSLT_ConvertTclObjToXPathObj(extinfo->interp, resultPtr);
     valuePush(xpathCtxt, obj);
   } else {
-    xmlGenericError(xmlGenericErrorContext,
+    xmlGenericError(xmlGenericErrorContext, "%s",
 		    Tcl_GetStringFromObj(resultPtr, NULL));
     /* Need to define a new error code - this is the closest in meaning */
     xpathCtxt->error = XPATH_UNKNOWN_FUNC_ERROR;
