diff -Naur tclxml-3.1.orig/Makefile.in tclxml-3.1/Makefile.in
--- tclxml-3.1.orig/Makefile.in	2005-05-20 21:02:01.000000000 +0900
+++ tclxml-3.1/Makefile.in	2007-12-24 13:28:55.000000000 +0900
@@ -267,7 +267,7 @@
 #		-testdir $(srcdir)/tea.tests
 
 test: binaries libraries
-	(cd $(srcdir)/tests; $(bindir)/tclsh8.4 all.tcl)
+	(cd $(srcdir)/tests; $(bindir)/tclsh all.tcl)
 
 shell: binaries libraries
 	@$(TCLSH) $(SCRIPT)
diff -Naur tclxml-3.1.orig/examples/README tclxml-3.1/examples/README
--- tclxml-3.1.orig/examples/README	2000-12-28 15:07:12.000000000 +0900
+++ tclxml-3.1/examples/README	2007-12-24 13:29:21.000000000 +0900
@@ -31,7 +31,7 @@
 REC-xml-20001006.xml 
         The W3C XML spec in XML format. A handy file to run xmlwc over
         to test your build. You should get this output with the command
-             tclsh8.3 xmlwc REC-xml-20001006.xml 
+             tclsh xmlwc REC-xml-20001006.xml 
         :
  
            2929   14978  116827 REC-xml-20001006.xml 
diff -Naur tclxml-3.1.orig/examples/flatten.tcl tclxml-3.1/examples/flatten.tcl
--- tclxml-3.1.orig/examples/flatten.tcl	2000-05-20 08:56:20.000000000 +0900
+++ tclxml-3.1/examples/flatten.tcl	2007-12-24 13:29:29.000000000 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 # -*- tcl -*- \
-exec tclsh8.3 "$0" "$@"
+exec tclsh "$0" "$@"
 
 # flatten.tcl --
 #
diff -Naur tclxml-3.1.orig/examples/xmlwc tclxml-3.1/examples/xmlwc
--- tclxml-3.1.orig/examples/xmlwc	2000-12-28 14:53:38.000000000 +0900
+++ tclxml-3.1/examples/xmlwc	2007-12-24 13:29:09.000000000 +0900
@@ -1,6 +1,6 @@
 #!/bin/sh
 # \
-exec tclsh8.3 "$0" "$@"
+exec tclsh "$0" "$@"
 
 package require xml
 
