--- a/prosody.cfg.lua.dist	2019-03-19 15:11:16.355743969 +0100
+++ b/prosody.cfg.lua.dist	2019-03-19 15:14:42.810995725 +0100
@@ -32,6 +32,13 @@
 -- will look for modules first. For community modules, see https://modules.prosody.im/
 --plugin_paths = {}
 
+-- Gentoo uses the jabber:jabber uid/gid for all Jabber-related services
+prosody_user = "jabber"
+prosody_group = "jabber"
+
+-- Make sure prosodyctl checks in the right spot for the PID file
+pidfile = "/run/jabber/prosody.pid"
+
 -- This is the list of modules Prosody will load on startup.
 -- It looks for mod_modulename.lua in the plugins folder, so make sure that exists too.
 -- Documentation for bundled modules can be found at: https://prosody.im/doc/modules
