diff --git a/ext/test/http/curl_http_test.cc b/ext/test/http/curl_http_test.cc
index 0dc6e5d5..2d50d872 100644
--- a/ext/test/http/curl_http_test.cc
+++ b/ext/test/http/curl_http_test.cc
@@ -232,7 +232,7 @@ TEST_F(BasicCurlHttpTests, HttpResponse)
   ASSERT_EQ(count, 4);
 }
 
-TEST_F(BasicCurlHttpTests, SendGetRequest)
+TEST_F(BasicCurlHttpTests, DISABLED_SendGetRequest)
 {
   received_requests_.clear();
   auto session_manager = http_client::HttpClientFactory::Create();
@@ -249,7 +249,7 @@ TEST_F(BasicCurlHttpTests, SendGetRequest)
   ASSERT_TRUE(handler->got_response_);
 }
 
-TEST_F(BasicCurlHttpTests, SendPostRequest)
+TEST_F(BasicCurlHttpTests, DISABLED_SendPostRequest)
 {
   received_requests_.clear();
   auto session_manager = http_client::HttpClientFactory::Create();
@@ -319,7 +319,7 @@ TEST_F(BasicCurlHttpTests, CurlHttpOperations)
   delete handler;
 }
 
-TEST_F(BasicCurlHttpTests, SendGetRequestSync)
+TEST_F(BasicCurlHttpTests, DISABLED_SendGetRequestSync)
 {
   received_requests_.clear();
   curl::HttpClientSync http_client;
@@ -344,7 +344,7 @@ TEST_F(BasicCurlHttpTests, SendGetRequestSyncTimeout)
               result.GetSessionState() == http_client::SessionState::SendFailed);
 }
 
-TEST_F(BasicCurlHttpTests, SendPostRequestSync)
+TEST_F(BasicCurlHttpTests, DISABLED_SendPostRequestSync)
 {
   received_requests_.clear();
   curl::HttpClientSync http_client;
@@ -372,7 +372,7 @@ TEST_F(BasicCurlHttpTests, GetBaseUri)
             "http://127.0.0.1:31339/");
 }
 
-TEST_F(BasicCurlHttpTests, SendGetRequestAsync)
+TEST_F(BasicCurlHttpTests, DISABLED_SendGetRequestAsync)
 {
   curl::HttpClient http_client;
 
@@ -446,7 +446,7 @@ TEST_F(BasicCurlHttpTests, SendGetRequestAsyncTimeout)
   }
 }
 
-TEST_F(BasicCurlHttpTests, SendPostRequestAsync)
+TEST_F(BasicCurlHttpTests, DISABLED_SendPostRequestAsync)
 {
   curl::HttpClient http_client;
 
@@ -485,7 +485,7 @@ TEST_F(BasicCurlHttpTests, SendPostRequestAsync)
   }
 }
 
-TEST_F(BasicCurlHttpTests, FinishInAsyncCallback)
+TEST_F(BasicCurlHttpTests, DISABLED_FinishInAsyncCallback)
 {
   curl::HttpClient http_client;
 
