diff --git a/ext/src/http/client/nosend/CMakeLists.txt b/ext/src/http/client/nosend/CMakeLists.txt
index 497daeb3..949bfd36 100644
--- a/ext/src/http/client/nosend/CMakeLists.txt
+++ b/ext/src/http/client/nosend/CMakeLists.txt
@@ -27,10 +27,4 @@ if(${BUILD_TESTING})
   target_link_libraries(http_client_nosend ${GTEST_BOTH_LIBRARIES} ${GMOCK_LIB}
                         opentelemetry_ext)
 
-  install(
-    TARGETS http_client_nosend
-    EXPORT "${PROJECT_NAME}-target"
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
