--- c/tkImageLoad.c	2009-08-14 21:10:19.000000000 +0200
+++ c/tkImageLoad.c.new	2009-08-14 21:10:55.000000000 +0200
@@ -19,6 +19,7 @@
 //#define BLOCKED 1
 //#define UNBLOCKED 1
 #define IN_BLOCK_FLIP 1
+#define USE_COMPOSITELESS_PHOTO_PUT_BLOCK
 
 /* *********************************************************************/
 
--- c/tkImageLoad.c	2009-08-14 21:13:29.000000000 +0200
+++ c/tkImageLoad.c.new	2009-08-14 21:14:59.000000000 +0200
@@ -272,9 +272,11 @@
   }
   /* put the photo block into the image */
 #ifdef __alpha
-  Tk_PhotoPutBlock(dp, &db, 0, 0, width, height);
+  Tk_PhotoPutBlock(dp, &db, 0, 0, width, height,
+  TK_PHOTO_COMPOSITE_OVERLAY);
 #else
-  Tk_PhotoPutBlock(dp, &db, 0, 0, width, height, TK_PHOTO_COMPOSITE_SET);
+  Tk_PhotoPutBlock(dp, &db, 0, 0, width, height, TK_PHOTO_COMPOSITE_SET,
+  TK_PHOTO_COMPOSITE_OVERLAY);
 #endif
 
   /* Free the photo block's memory now it is finished with */
@@ -417,9 +419,11 @@
 
   /* put the photo block back into the image */
 #ifdef __alpha
- Tk_PhotoPutBlock(photo, &block, 0, 0, block.width, block.height);
+ Tk_PhotoPutBlock(photo, &block, 0, 0, block.width, block.height,
+ TK_PHOTO_COMPOSITE_OVERLAY);
 #else
- Tk_PhotoPutBlock(photo, &block, 0, 0, block.width, block.height,TK_PHOTO_COMPOSITE_SET);
+ Tk_PhotoPutBlock(photo, &block, 0, 0, block.width, block.height,TK_PHOTO_COMPOSITE_SET,
+ TK_PHOTO_COMPOSITE_OVERLAY);
 #endif
   /* Free the photo block's memory now it is finished with */
   if (block.pixelPtr) {
--- c/tkImageLoad.c	2009-08-14 21:15:06.000000000 +0200
+++ c/tkImageLoad.c.new	2009-08-14 21:16:53.000000000 +0200
@@ -9,6 +9,7 @@
 \************************************************************************/
 
 #include <stdio.h>
+#include <string.h>
 #include <tcl.h>
 #include <tk.h>
 #include <assert.h>
