--- grip-3.3.1/src/status_window.c
+++ grip-3.3.1/src/status_window.c
@@ -23,6 +23,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <unistd.h>
+#include <string.h>
 #include <fcntl.h>
 #include <vte/vte.h>
 #include "status_window.h"
--- grip-3.3.1/src/tray.c
+++ grip-3.3.1/src/tray.c
@@ -20,6 +20,9 @@
  * USA
  */
 
+#include "cdplay.h"
+#include "common.h"
+#include "rip.h"
 #include "tray.h"
 #include "../pixmaps/rip1.xpm"
 #include "../pixmaps/menuplay.xpm"
--- grip-3.3.1/src/cdplay.c
+++ grip-3.3.1/src/cdplay.c
@@ -28,6 +28,7 @@
 #include "cddev.h"
 #include "discedit.h"
 #include "dialog.h"
+#include "tray.h"
 #include "rip.h"
 #include "grip_id3.h"
 
