--- a/ssl/ssldecode.c
+++ b/ssl/ssldecode.c
@@ -51,6 +51,7 @@
 #include <openssl/ssl.h>
 #include <openssl/hmac.h>
 #include <openssl/evp.h>
+#include <openssl/md5.h>
 #include <openssl/x509v3.h>
 #endif
 #include "ssldecode.h"
@@ -131,6 +132,7 @@
     ssl_decode_ctx *d=0;
     int r,_status;
     
+    SSL_library_init();
     SSLeay_add_all_algorithms();
     if(!(d=(ssl_decode_ctx *)malloc(sizeof(ssl_decode_ctx))))
       ABORT(R_NO_MEMORY);
