--- a/CMakeLists.txt	2024-05-21 20:11:22.098008343 +0200
+++ b/CMakeLists.txt	2024-05-21 20:11:35.538853965 +0200
@@ -31,7 +31,7 @@
 # Actual configuration
 # ------------------------------------------------------------------------
 
-add_compile_options (-std=c++11 -Wall -Wextra -pedantic -Werror)
+add_compile_options (-std=c++11 -Wall -Wextra -pedantic)
 
 set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/out)
 set (CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
