From 2c00334135aa004213758a0121a23a872660aa79 Mon Sep 17 00:00:00 2001
From: Jussi Lind <jussi.lind@iki.fi>
Date: Sat, 28 Jan 2023 12:55:15 +0200
Subject: [PATCH] Set CMake policy CMP0100 to NEW

---
 src/game/MiniCore/CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/game/MiniCore/CMakeLists.txt b/src/game/MiniCore/CMakeLists.txt
index 27073103..07430201 100644
--- a/src/game/MiniCore/CMakeLists.txt
+++ b/src/game/MiniCore/CMakeLists.txt
@@ -19,6 +19,11 @@ if(POLICY CMP0054)
     cmake_policy(SET CMP0054 NEW)
 endif()
 
+# Automoc for .hh
+if(POLICY CMP0100)
+    cmake_policy(SET CMP0100 NEW)
+endif()
+
 option(GLES "Build for OpenGL ES 2.0" OFF)
 
 option(NO_GLEW "Don't use GLEW to resolve OpenGL extensions if enabled." ON)
