--- docbook-utils-0.6.14/doc/HTML/Makefile.am~	2007-10-08 04:13:04 +0200
+++ docbook-utils-0.6.14/doc/HTML/Makefile.am	2007-10-08 04:13:34 +0200
@@ -23,7 +23,7 @@
 		$(top_srcdir)/doc/refentry/docbook2texi-spec.pl.sgml \
 		$(top_srcdir)/doc/refentry/frontend-spec.sgml \
 		$(top_srcdir)/doc/refentry/sgmldiff.sgml
-	SGML_CATALOG_FILES=/etc/sgml/catalog \
+	SGML_CATALOG_FILES=@GENTOO_PORTAGE_EPREFIX@/etc/sgml/catalog \
 	SGML_SEARCH_PATH=$(top_srcdir):$(top_srcdir)/doc:.. \
 		jade -t sgml -i html -d $(top_srcdir)/docbook-utils.dsl\#html \
 			-V '%use-id-as-filename%' $<
--- docbook-utils-0.6.14/doc/man/Makefile.am~	2007-10-08 04:14:16 +0200
+++ docbook-utils-0.6.14/doc/man/Makefile.am	2007-10-08 04:14:29 +0200
@@ -10,7 +10,7 @@
 
 $(man1_MANS) $(man7_MANS): $(top_srcdir)/doc/docbook-utils.sgml \
 		$(top_srcdir)/helpers/docbook2man-spec.pl
-	SGML_CATALOG_FILES=/etc/sgml/catalog \
+	SGML_CATALOG_FILES=@GENTOO_PORTAGE_EPREFIX@/etc/sgml/catalog \
 	SGML_SEARCH_PATH=$(top_srcdir)/doc:.. \
 		nsgmls $< | \
 		sgmlspl $(top_srcdir)/helpers/docbook2man-spec.pl
--- docbook-utils-0.6.14/bin/jw.in~	2007-10-08 04:27:18 +0200
+++ docbook-utils-0.6.14/bin/jw.in	2007-10-08 04:28:40 +0200
@@ -63,7 +63,7 @@
 then
   SGML_CONF=`sgmlwhich`
 else
-  SGML_CONF="/etc/sgml/sgml.conf"
+  SGML_CONF="@GENTOO_PORTAGE_EPREFIX@/etc/sgml/sgml.conf"
 fi
 
 # Set version message
@@ -77,7 +77,7 @@
 
 # Set SGML base directory and centralized catalogs directory
 SGML_BASE_DIR="@prefix@/share/sgml"
-SGML_CATALOGS_DIR="/etc/sgml"
+SGML_CATALOGS_DIR="@GENTOO_PORTAGE_EPREFIX@/etc/sgml"
 if [ -f "$SGML_CONF" ]
 then
   RE='^[:space:]*SGML_BASE_DIR[:space:]*=[:space:]*'
--- docbook-utils-0.6.14/backends/txt~	2007-10-08 04:59:59 +0200
+++ docbook-utils-0.6.14/backends/txt	2007-10-08 05:00:52 +0200
@@ -2,21 +2,21 @@
 # Send any comments to Eric Bischoff <eric@caldera.de>
 # This program is under GPL license. See LICENSE file for details.
 
-if [ -x /usr/bin/lynx ]
+if [ -x "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/lynx ]
 then
-  CONVERT=/usr/bin/lynx
+  CONVERT="@GENTOO_PORTAGE_EPREFIX@"/usr/bin/lynx
   ARGS="-force_html -dump -nolist -width=72"
-elif [ -x /usr/bin/links ]
+elif [ -x "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/links ]
 then
-  CONVERT=/usr/bin/links
+  CONVERT="@GENTOO_PORTAGE_EPREFIX@"/usr/bin/links
   ARGS="-dump"
-elif [ -x /usr/bin/elinks ]
+elif [ -x "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/elinks ]
 then
-  CONVERT=/usr/bin/elinks
+  CONVERT="@GENTOO_PORTAGE_EPREFIX@"/usr/bin/elinks
   ARGS="-dump"
-elif [ -x /usr/bin/w3m ]
+elif [ -x "@GENTOO_PORTAGE_EPREFIX@"/usr/bin/w3m ]
 then
-  CONVERT=/usr/bin/w3m
+  CONVERT="@GENTOO_PORTAGE_EPREFIX@"/usr/bin/w3m
   ARGS="-dump"
 else
   echo >&2 "No way to convert HTML to text found."
--- docbook-utils-0.6.14/configure.in~	2004-02-11 15:14:15 +0100
+++ docbook-utils-0.6.14/configure.in	2007-10-14 10:40:51 +0200
@@ -20,11 +20,11 @@
 AC_SUBST(docdir)
 
 dnl Checks for programs.
-jade_bindirs="/usr/bin /usr/local/bin"
+jade_bindirs="@GENTOO_PORTAGE_EPREFIX@/usr/bin"
 AC_FIND_PROGRAM(nsgmls, $jade_bindirs, jade_bindir)
 AC_SUBST(jade_bindir)
 
-perl_bindirs="/usr/bin /usr/local/bin"
+perl_bindirs="@GENTOO_PORTAGE_EPREFIX@/usr/bin"
 AC_FIND_PROGRAM(perl, $perl_bindirs, perl_bindir)
 AC_SUBST(perl_bindir)
 
