diff -ur 2p-deploy.orig/src/alice/tuprolog/ChoicePointStore.java 2p-deploy/src/alice/tuprolog/ChoicePointStore.java
--- 2p-deploy.orig/src/alice/tuprolog/ChoicePointStore.java	2010-01-21 21:29:24.000000000 +1300
+++ 2p-deploy/src/alice/tuprolog/ChoicePointStore.java	2012-01-15 09:20:48.000000000 +1300
@@ -35,7 +35,6 @@
 	
 	/**
 	 * Return the actual choice-point store
-	 * @return
 	 */
 	public ChoicePointContext getPointer() {
 		return pointer;
@@ -44,7 +43,6 @@
 	/**
 	 * Check if a choice point exists in the store.
 	 * As a side effect, removes choice points which have been already used and are now empty. 
-	 * @return
 	 */
 	protected boolean existChoicePoint() {
 		if (pointer == null) return false;
diff -ur 2p-deploy.orig/src/alice/tuprolog/ClauseStore.java 2p-deploy/src/alice/tuprolog/ClauseStore.java
--- 2p-deploy.orig/src/alice/tuprolog/ClauseStore.java	2010-11-04 11:55:42.000000000 +1300
+++ 2p-deploy/src/alice/tuprolog/ClauseStore.java	2012-01-15 09:20:48.000000000 +1300
@@ -62,8 +62,6 @@
 	
 	/**
 	 * Verify if there is a term in compatibleGoals compatible with goal. 
-	 * @param goal
-	 * @param compGoals
 	 * @return true if compatible or false otherwise.
 	 */
 	protected boolean existCompatibleClause() {
diff -ur 2p-deploy.orig/src/alice/tuprolog/EngineManager.java 2p-deploy/src/alice/tuprolog/EngineManager.java
--- 2p-deploy.orig/src/alice/tuprolog/EngineManager.java	2010-01-21 21:29:24.000000000 +1300
+++ 2p-deploy/src/alice/tuprolog/EngineManager.java	2012-01-15 09:20:48.000000000 +1300
@@ -87,7 +87,7 @@
 	/**
 	 *  Solves a query
 	 *
-	 * @param g the term representing the goal to be demonstrated
+	 * @param query the term representing the goal to be demonstrated
 	 * @return the result of the demonstration
 	 * @see SolveInfo
 	 **/
diff -ur 2p-deploy.orig/src/alice/tuprolog/PrimitiveManager.java 2p-deploy/src/alice/tuprolog/PrimitiveManager.java
--- 2p-deploy.orig/src/alice/tuprolog/PrimitiveManager.java	2010-01-21 21:29:24.000000000 +1300
+++ 2p-deploy/src/alice/tuprolog/PrimitiveManager.java	2012-01-15 09:20:48.000000000 +1300
@@ -87,7 +87,7 @@
 	 * This involves identifying structs representing builtin
 	 * predicates and functors, and setting up related structures and links
 	 *
-	 * @parm term the term to be identified
+	 * @param term the term to be identified
 	 * @return term with the identified built-in directive
 	 */
 	public Term identifyDirective(Term term) {
diff -ur 2p-deploy.orig/src/alice/tuprolog/lib/BasicLibrary.java 2p-deploy/src/alice/tuprolog/lib/BasicLibrary.java
--- 2p-deploy.orig/src/alice/tuprolog/lib/BasicLibrary.java	2011-01-28 16:31:44.000000000 +1300
+++ 2p-deploy/src/alice/tuprolog/lib/BasicLibrary.java	2012-01-15 09:20:48.000000000 +1300
@@ -109,7 +109,7 @@
 	/**
 	 * Loads a library constructed from a theory.
 	 * 
-	 * @param theory
+	 * @param th
 	 *            theory text
 	 * @param libName
 	 *            name of the library
diff -ur 2p-deploy.orig/src/alice/tuprolog/lib/IOLibrary.java 2p-deploy/src/alice/tuprolog/lib/IOLibrary.java
--- 2p-deploy.orig/src/alice/tuprolog/lib/IOLibrary.java	2011-01-19 16:34:24.000000000 +1300
+++ 2p-deploy/src/alice/tuprolog/lib/IOLibrary.java	2012-01-15 09:20:48.000000000 +1300
@@ -372,7 +372,7 @@
 	/**
 	 * Sets an arbitrary seed for the Random object.
 	 * 
-	 * @param seed Seed to use
+	 * @param t Seed to use
 	 * @return true if seed Term has a valid long value, false otherwise
 	 */
 	public boolean set_seed_1(Term t) throws PrologError {
diff -ur 2p-deploy.orig/src/alice/tuprolog/lib/JavaLibrary.java 2p-deploy/src/alice/tuprolog/lib/JavaLibrary.java
--- 2p-deploy.orig/src/alice/tuprolog/lib/JavaLibrary.java	2010-01-21 21:29:24.000000000 +1300
+++ 2p-deploy/src/alice/tuprolog/lib/JavaLibrary.java	2012-01-15 09:20:48.000000000 +1300
@@ -1390,8 +1390,6 @@
 
 	/**
 	 * Generates a fresh numeric identifier
-	 * 
-	 * @return
 	 */
 	protected Struct generateFreshId() {
 		return new Struct("$obj_" + id++);
diff -ur 2p-deploy.orig/src/alice/tuprologx/ide/LibraryDialogFrame.java 2p-deploy/src/alice/tuprologx/ide/LibraryDialogFrame.java
--- 2p-deploy.orig/src/alice/tuprologx/ide/LibraryDialogFrame.java	2010-01-21 21:29:22.000000000 +1300
+++ 2p-deploy/src/alice/tuprologx/ide/LibraryDialogFrame.java	2012-01-15 09:20:48.000000000 +1300
@@ -317,7 +317,6 @@
         displayLibraryManagerStatus();
     }
 
-    /** @see alice.tuprologx.ide.SwingFrame#onClose()*/
     public void closeLibraryDialog()
     {
     	onClose();
diff -ur 2p-deploy.orig/src/alice/tuprologx/ide/ToolBar.java 2p-deploy/src/alice/tuprologx/ide/ToolBar.java
--- 2p-deploy.orig/src/alice/tuprologx/ide/ToolBar.java	2010-01-21 21:29:22.000000000 +1300
+++ 2p-deploy/src/alice/tuprologx/ide/ToolBar.java	2012-01-15 09:20:48.000000000 +1300
@@ -232,8 +232,6 @@
 
     /**
      * Set the tuProlog config frame
-     *
-     * @param libraryManager a libraryManager to associate to the libraryDialog 
      */
 	public void setPrologConfig(PrologConfigFrame configFrame)
 	{
diff -ur 2p-deploy.orig/src/alice/tuprologx/pj/lib/PJLibrary.java 2p-deploy/src/alice/tuprologx/pj/lib/PJLibrary.java
--- 2p-deploy.orig/src/alice/tuprologx/pj/lib/PJLibrary.java	2010-01-21 21:29:48.000000000 +1300
+++ 2p-deploy/src/alice/tuprologx/pj/lib/PJLibrary.java	2012-01-15 09:20:48.000000000 +1300
@@ -1252,7 +1252,6 @@
 	
 	/**
 	 * Generates a fresh numeric identifier
-	 * @return
 	 */
 	protected Struct generateFreshId() {
 		return new Struct("$obj_" + id++);        
diff -ur 2p-deploy.orig/src/alice/util/ReadOnlyLinkedList.java 2p-deploy/src/alice/util/ReadOnlyLinkedList.java
--- 2p-deploy.orig/src/alice/util/ReadOnlyLinkedList.java	2010-03-17 18:14:10.000000000 +1300
+++ 2p-deploy/src/alice/util/ReadOnlyLinkedList.java	2012-01-15 09:32:03.000000000 +1300
@@ -12,10 +12,10 @@
 import java.util.ListIterator;
 
 /**
- * ReadOnlyLinkedList<E> encapsulate a {@link LinkedList<E>}
+ * ReadOnlyLinkedList<E> encapsulate a {@link LinkedList}
  * and ensures that the given list is navigated only.
  *
- * Even if ReadOnlyLinkedList<E> implements {@link List<E>} it doesn't
+ * Even if ReadOnlyLinkedList<E> implements {@link List} it doesn't
  * support modifiers methods, and throws {@link UnsupportedOperationException}
  * if invoked.
  *
diff -ur 2p-deploy.orig/src/alice/util/jedit/InputHandler.java 2p-deploy/src/alice/util/jedit/InputHandler.java
--- 2p-deploy.orig/src/alice/util/jedit/InputHandler.java	2010-01-21 21:29:24.000000000 +1300
+++ 2p-deploy/src/alice/util/jedit/InputHandler.java	2012-01-15 09:20:48.000000000 +1300
@@ -24,7 +24,7 @@
  *
  * @author Slava Pestov
  * @version $Id: InputHandler.java,v 1.14 1999/12/13 03:40:30 sp Exp $
- * @see org.gjt.sp.jedit.textarea.DefaultInputHandler
+ * @see alice.util.jedit.DefaultInputHandler
  */
 public abstract class InputHandler extends KeyAdapter
 {
@@ -187,7 +187,7 @@
 	/**
 	 * Grabs the next key typed event and invokes the specified
 	 * action with the key as a the action command.
-	 * @param action The action
+	 * @param listener The action
 	 */
 	public void grabNextKeyStroke(ActionListener listener)
 	{
diff -ur 2p-deploy.orig/src/alice/util/jedit/JEditTextArea.java 2p-deploy/src/alice/util/jedit/JEditTextArea.java
--- 2p-deploy.orig/src/alice/util/jedit/JEditTextArea.java	2010-01-21 21:29:24.000000000 +1300
+++ 2p-deploy/src/alice/util/jedit/JEditTextArea.java	2012-01-15 09:20:48.000000000 +1300
@@ -1389,7 +1389,7 @@
 
 	/**
 	 * Sets if the selection should be rectangular.
-	 * @param overwrite True if the selection should be rectangular,
+	 * @param rectSelect True if the selection should be rectangular,
 	 * false otherwise.
 	 */
 	public final void setSelectionRectangular(boolean rectSelect)
diff -ur 2p-deploy.orig/src/alice/util/jedit/KeywordMap.java 2p-deploy/src/alice/util/jedit/KeywordMap.java
--- 2p-deploy.orig/src/alice/util/jedit/KeywordMap.java	2010-01-21 21:29:22.000000000 +1300
+++ 2p-deploy/src/alice/util/jedit/KeywordMap.java	2012-01-15 09:20:48.000000000 +1300
@@ -75,7 +75,7 @@
 	/**
 	 * Adds a key-value mapping.
 	 * @param keyword The key
-	 * @Param id The value
+	 * @param id The value
 	 */
 	public void add(String keyword, byte id)
 	{
diff -ur 2p-deploy.orig/src/alice/util/jedit/TextAreaPainter.java 2p-deploy/src/alice/util/jedit/TextAreaPainter.java
--- 2p-deploy.orig/src/alice/util/jedit/TextAreaPainter.java	2010-01-21 21:29:24.000000000 +1300
+++ 2p-deploy/src/alice/util/jedit/TextAreaPainter.java	2012-01-15 09:20:48.000000000 +1300
@@ -75,7 +75,7 @@
 	/**
 	 * Returns the syntax styles used to paint colorized text. Entry <i>n</i>
 	 * will be used to paint tokens with id = <i>n</i>.
-	 * @see org.gjt.sp.jedit.syntax.Token
+	 * @see alice.util.jedit.Token
 	 */
 	public final SyntaxStyle[] getStyles()
 	{
@@ -86,7 +86,7 @@
 	 * Sets the syntax styles used to paint colorized text. Entry <i>n</i>
 	 * will be used to paint tokens with id = <i>n</i>.
 	 * @param styles The syntax styles
-	 * @see org.gjt.sp.jedit.syntax.Token
+	 * @see alice.util.jedit.Token
 	 */
 	public final void setStyles(SyntaxStyle[] styles)
 	{
@@ -356,7 +356,7 @@
 
 	/**
 	 * Repaints the text.
-	 * @param g The graphics context
+	 * @param gfx The graphics context
 	 */
 	public void paint(Graphics gfx)
 	{
diff -ur 2p-deploy.orig/src/alice/util/jedit/TokenMarker.java 2p-deploy/src/alice/util/jedit/TokenMarker.java
--- 2p-deploy.orig/src/alice/util/jedit/TokenMarker.java	2010-01-21 21:29:22.000000000 +1300
+++ 2p-deploy/src/alice/util/jedit/TokenMarker.java	2012-01-15 09:20:48.000000000 +1300
@@ -24,7 +24,7 @@
  * @author Slava Pestov
  * @version $Id: TokenMarker.java,v 1.32 1999/12/13 03:40:30 sp Exp $
  *
- * @see org.gjt.sp.jedit.syntax.Token
+ * @see alice.util.jedit.Token
  */
 public abstract class TokenMarker
 {
