From 166ec045945eff08ca27129b38d2f5395710e8fb Mon Sep 17 00:00:00 2001
From: Mikle Kolyada <zlogene@gentoo.org>
Date: Sat, 23 Feb 2019 21:14:08 +0300
Subject: [PATCH] fix wgetpaste services

Signed-off-by: Mikle Kolyada <zlogene@gentoo.org>
---
 wgetpaste | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wgetpaste b/wgetpaste
index 2c198e0..def51de 100755
--- a/wgetpaste
+++ b/wgetpaste
@@ -14,7 +14,7 @@ E=$'\e'
 N=$'\n'
 
 ### services
-SERVICES="codepad bpaste dpaste gists poundpython"
+SERVICES="codepad bpaste dpaste gists"
 # bpaste
 ENGINE_bpaste=pinnwand
 URL_bpaste="https://bpaste.net/"
@@ -722,7 +722,7 @@ load_configs() {
 load_configs
 [[ $SERVICESET ]] && verifyservice "$SERVICESET" && SERVICE=$(escape "$SERVICESET")
 DEFAULT_NICK=${DEFAULT_NICK:-$(whoami)} || die "whoami failed"
-DEFAULT_SERVICE=${DEFAULT_SERVICE:-poundpython}
+DEFAULT_SERVICE=${DEFAULT_SERVICE:-dpaste}
 DEFAULT_LANGUAGE=${DEFAULT_LANGUAGE:-Plain Text}
 DEFAULT_EXPIRATION=${DEFAULT_EXPIRATION:-1 month}
 SERVICE=${SERVICE:-${DEFAULT_SERVICE}}
-- 
2.19.2

