--- a/bsd_discid.c
+++ b/bsd_discid.c
@@ -29,7 +29,7 @@
 #include "config.h"
 
 const char *cdrom_device=DEV_CDROM;
-int cdrom_fd;
+extern int cdrom_fd;
 
 int Discid_Init(void) {
 /*	if ((cdrom_device = getenv("CD_DEVICE")) == NULL) {
--- a/darwin_discid.c
+++ b/darwin_discid.c
@@ -29,7 +29,7 @@
 #include "config.h"
 
 const char *cdrom_device=DEV_CDROM;
-int cdrom_fd;
+extern int cdrom_fd;
 
 int Discid_Init(void) {
 /*	if ((cdrom_device = getenv("CD_DEVICE")) == NULL) {
--- a/linux_discid.c
+++ b/linux_discid.c
@@ -31,7 +31,7 @@
 
 char *cdrom_device = NULL;
 
-int cdrom_fd;
+extern int cdrom_fd;
 
 int Discid_Init(void) {
 /*	if ((cdrom_device = getenv("CD_DEVICE")) == NULL) {
--- a/sun_discid.c
+++ b/sun_discid.c
@@ -30,7 +30,7 @@
 #include "config.h"
 
 const char *cdrom_device=DEV_CDROM;
-int cdrom_fd;
+extern int cdrom_fd;
 
 int Discid_Init(void) {
 /*	if ((cdrom_device = getenv("CD_DEVICE")) == NULL) {
