Index: catdoc-0.94.2/configure.in
===================================================================
--- catdoc-0.94.2.orig/configure.in
+++ catdoc-0.94.2/configure.in
@@ -119,6 +119,6 @@ AC_SUBST(charsetcheck)
 AC_SUBST(installroot)
 AC_SUBST(catdoc_version)
 AC_SUBST(CFLAGS)
-AC_SUBST(WORDS_BIGENDIAN DEFS)
+AC_SUBST(DEFS)
 AC_CONFIG_HEADER(src/config.h)
 AC_OUTPUT(doc/Makefile charsets/Makefile src/Makefile Makefile doc/catdoc.1 doc/xls2csv.1 doc/wordview.1 doc/catppt.1)
