https://bugs.gentoo.org/946388
--- a/libgnucash/app-utils/gnc-quotes.cpp       2024-12-13 17:18:10.294664737 -0800
+++ b/libgnucash/app-utils/gnc-quotes.cpp       2024-12-13 17:18:31.410866021 -0800
@@ -212,7 +212,7 @@ GncFQQuoteSource::run_cmd (const StrVec&
     try
     {
         std::future<std::vector<char> > out_buf, err_buf;
-        boost::asio::io_service svc;
+        boost::asio::io_context svc;
 
         auto input_buf = bp::buffer (json_string);
        bp::child process;
