From 017a1059067ab58a9286b3049505cb3e77696de5 Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl-gentoo@posteo.net>
Date: Mon, 14 Jun 2021 17:48:45 +0200
Subject: [PATCH 2/2] use proper AR

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>
---
 intl/dummy/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/intl/dummy/Makefile b/intl/dummy/Makefile
index c32b0a8..d47e185 100644
--- a/intl/dummy/Makefile
+++ b/intl/dummy/Makefile
@@ -2,4 +2,4 @@ RHIDE_GCC=gcc
 CC=$(RHIDE_GCC)
 
 libtvfintl.a: intldummy.o
-	ar -rv libtvfintl.a intldummy.o
+	$(AR) -rv libtvfintl.a intldummy.o
-- 
2.32.0

