diff -ur a/mk/com32.mk b/mk/com32.mk
--- a/mk/com32.mk	2011-12-09 19:28:17.000000000 +0100
+++ b/mk/com32.mk	2011-12-18 18:22:11.032342645 +0100
@@ -24,6 +24,8 @@
 GCCOPT += $(call gcc_ok,-freg-struct-return,)
 GCCOPT += -mregparm=3 -DREGPARM=3 -march=i386 -Os
 GCCOPT += $(call gcc_ok,-fPIE,-fPIC)
+GCCOPT += $(call gcc_ok,-nopie,)
+GCCOPT += $(call gcc_ok,-fno-pie,)
 GCCOPT += $(call gcc_ok,-fno-exceptions,)
 GCCOPT += $(call gcc_ok,-fno-asynchronous-unwind-tables,)
 GCCOPT += $(call gcc_ok,-fno-strict-aliasing,)
