diff -Naur MIME-Lite-3.030/Makefile.PL MIME-Lite-3.030.new/Makefile.PL
--- MIME-Lite-3.030/Makefile.PL	2013-11-04 02:04:48.000000000 +0000
+++ MIME-Lite-3.030.new/Makefile.PL	2016-02-22 10:15:49.107771292 +0000
@@ -5,39 +5,12 @@
 #------------------------------------------------------------
 # Makefile:
 #------------------------------------------------------------
-my %extra_prereq = (
-  'Mail::Address'     => 1.62,
-  'MIME::Types'       => 1.28,
-  'MIME::Base64'      => undef,
-  'MIME::QuotedPrint' => undef,
-);
-
-for my $module (sort keys %extra_prereq) {
-  local $@;
-  my $ok = eval "require $module; \$extra_prereq{\$module} && \$module->VERSION(\$extra_prereq{\$module}); 1";
-  delete $extra_prereq{ $module } if $ok;
-}
-
 my %prereq = (
   'File::Spec'          => 0,
   'File::Basename'      => 0,
   'Email::Date::Format' => '1.000', # first release
 );
 
-if (%extra_prereq) {
-  print "MIME::Lite is designed to take advantage of a variety of external modules\n",
-        "if they are not present then MIME::Lite will attempt to do its best\n",
-        "but its strongly recommend that you install them.\n\n",
-        "These modules are:\n",
-        (map {; "\t$_\n" } sort keys %extra_prereq),
-        "I can add these modules to the prereq list which will cause the files to be\n",
-        "automatically installed if they aren't already present\n";
-
-  if (prompt("Add prereqs?",'Yes')=~/y/i) {
-    %prereq = (%prereq, %extra_prereq);
-  }
-}
-
 # Write the Makefile:
 WriteMakefile(
   NAME         => 'MIME::Lite',
