diff -ruN Argyll_V1.4.0.orig/Jamtop Argyll_V1.4.0/Jamtop
--- Argyll_V1.4.0.orig/Jamtop	2012-04-19 09:50:31.000000000 +0200
+++ Argyll_V1.4.0/Jamtop	2012-05-07 02:18:37.000000000 +0200
@@ -113,13 +113,13 @@
 # See if we have a system JPEG library.
 if ! $(BUILTIN_JPEG) && $(UNIX) {
 	if [ GLOB /usr/include : jpeglib.h ]  {
-		if  ! $(HOST64) && ( [ GLOB /usr/lib : libjpg.so ] || [ GLOB /usr/lib : libjpg.a ] )
-		 ||   $(HOST64) && ( [ GLOB /usr/lib64 : libjpg.so ] || [ GLOB /usr/lib64 : libjpg.a ] )
-		 ||   $(HOST64) && ( [ GLOB /usr/lib/x86_64-linux-gnu : libjpg.so ] || [ GLOB /usr/lib/x86_64-linux-gnu : libjpg.a ] ) {
+		if  ! $(HOST64) && ( [ GLOB /usr/lib : libjpeg.so ] || [ GLOB /usr/lib : libjpeg.a ] )
+		 ||   $(HOST64) && ( [ GLOB /usr/lib64 : libjpeg.so ] || [ GLOB /usr/lib64 : libjpeg.a ] )
+		 ||   $(HOST64) && ( [ GLOB /usr/lib/x86_64-linux-gnu : libjpeg.so ] || [ GLOB /usr/lib/x86_64-linux-gnu : libjpeg.a ] ) {
 			echo "Using system JPEG library" ;
 			JPEGLIB = ;
 			JPEGINC = ;
-			LINKFLAGS += $(LINKFLAG)tiff ;
+			LINKFLAGS += $(LINKFLAG)jpeg ;
 			HAVE_JPEG = true ;
 		}
 	}
