diff --git a/src/Makefile.am b/src/Makefile.am
index 914237c..0bf982a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -45,7 +45,7 @@ src_libsc_la_SOURCES = \
 src_libsc_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBSC_CPPFLAGS)
 ## This is the official API versioning scheme of libtool.  Please see:
 ## Read https://www.gnu.org/software/libtool/manual/libtool.html#Versioning
-src_libsc_la_LDFLAGS = -version-info 1:0:0
+src_libsc_la_LDFLAGS = -release $(VERSION)
 LDADD += src/libsc.la
 
 nodist_include_HEADERS += $(libsc_generated_headers)
