Bug: https://bugs.gentoo.org/600082

--- a/Build/Laptop/BobbyRGuns.cc
+++ b/Build/Laptop/BobbyRGuns.cc
@@ -700,7 +700,7 @@
 	//center picture in frame
 	ETRLEObject const& pTrav   = uiImage->SubregionProperties(0);
 	UINT32      const  usWidth = pTrav.usWidth;
-	INT16       const  sCenX   = PosX + abs(BOBBYR_GRID_PIC_WIDTH - usWidth) / 2 - pTrav.sOffsetX;
+	INT16       const  sCenX   = PosX + (BOBBYR_GRID_PIC_WIDTH - usWidth) / 2 - pTrav.sOffsetX;
 	INT16       const  sCenY   = PosY + 8;
 
 	//blt the shadow of the item
