Index: rtabmap-0.19.7/corelib/src/camera/CameraOpenni.cpp
===================================================================
--- rtabmap-0.19.7.orig/corelib/src/camera/CameraOpenni.cpp
+++ rtabmap-0.19.7/corelib/src/camera/CameraOpenni.cpp
@@ -126,7 +126,7 @@ bool CameraOpenni::init(const std::strin
 		boost::function<void (
 				const boost::shared_ptr<openni_wrapper::Image>&,
 				const boost::shared_ptr<openni_wrapper::DepthImage>&,
-				float)> f = boost::bind (&CameraOpenni::image_cb, this, _1, _2, _3);
+				float)> f = boost::bind (&CameraOpenni::image_cb, this, boost::placeholders::_1, boost::placeholders::_2, boost::placeholders::_3);
 		connection_ = interface_->registerCallback (f);
 
 		interface_->start ();
