Add a missing library for gdwgc-8
--- a/configure.ac
+++ b/configure.ac
@@ -226,6 +226,8 @@ AC_ARG_ENABLE(gc, AC_HELP_STRING([--enable-gc],
 if test "$gc" = yes; then
   PKG_CHECK_MODULES([BDW_GC], [bdw-gc])
   CXXFLAGS="$BDW_GC_CFLAGS $CXXFLAGS"
+  # a workaround for https://bugs.gentoo.org/679834
+  BDW_GC_LIBS="$BDW_GC_LIBS -lgccpp"
   AC_DEFINE(HAVE_BOEHMGC, 1, [Whether to use the Boehm garbage collector.])
 fi
 
