diff --git a/configure.ac b/configure.ac
index 741f81f..e2a55de 100644
--- a/configure.ac
+++ b/configure.ac
@@ -42,7 +42,6 @@ TORRENT_ENABLE_ALIGNED
 TORRENT_ENABLE_INTERRUPT_SOCKET
 
 TORRENT_ENABLE_ARCH
-TORRENT_WITH_SYSROOT
 
 dnl TORRENT_WITH_XFS
 TORRENT_WITHOUT_KQUEUE
@@ -75,9 +74,8 @@ AC_ARG_ENABLE(execinfo,
     AX_EXECINFO
   ])
 
-AX_CHECK_ZLIB
 AX_PTHREAD
-
+PKG_CHECK_MODULES([ZLIB], [zlib])
 PKG_CHECK_MODULES([CPPUNIT], [cppunit],, [no_cppunit="yes"])
 
 CFLAGS="$PTHREAD_CFLAGS $CPPUNIT_CFLAGS $CFLAGS"
diff --git a/src/torrent/Makefile.am b/src/torrent/Makefile.am
index 5de7e8a..8754a9c 100644
--- a/src/torrent/Makefile.am
+++ b/src/torrent/Makefile.am
@@ -137,7 +137,8 @@ libtorrent_torrent_la_SOURCES = \
 	tracker_list.cc \
 	tracker_list.h
 
-AM_CPPFLAGS = -I$(srcdir) -I$(srcdir)/.. -I$(top_srcdir)
+libtorrent_torrent_la_LIBADD = $(ZLIB_LIBS)
+AM_CPPFLAGS = -I$(srcdir) -I$(srcdir)/.. -I$(top_srcdir) $(ZLIB_CFLAGS)
 
 libtorrent_torrent_data_includedir = $(includedir)/torrent/data
 libtorrent_torrent_data_include_HEADERS = \
