--- a/lib/Gscan2pdf/Dialog/Save.pm
+++ b/lib/Gscan2pdf/Dialog/Save.pm
@@ -570,10 +570,6 @@ sub add_image_type {
     $label = Gtk3::Label->new( __('Postscript backend') );
     $hboxps->pack_start( $label, FALSE, FALSE, 0 );
     my @backends = (
-        [
-            'libtiff', __('LibTIFF'),
-            __('Use LibTIFF (tiff2ps) to create Postscript files from TIFF.')
-        ],
         [
             'pdf2ps',
             __('Ghostscript'),
--- a/lib/Gscan2pdf/Document.pm
+++ b/lib/Gscan2pdf/Document.pm
@@ -4471,22 +4471,7 @@ sub _thread_save_tiff {
             'Save file', sprintf __('Error compressing image: %s'), $error );
         return;
     }
-    if ( defined $options{options}{ps} ) {
-        $self->{message} = __('Converting to PS');
-        @cmd = ( 'tiff2ps', '-3', $options{path}, '-O', $options{options}{ps} );
-        ( $status, undef, $error ) = exec_command( \@cmd, $options{pidfile} );
-        if ( $status or $error ) {
-            $logger->info($error);
-            _thread_throw_error( $self, $options{uuid}, $options{page}{uuid},
-                'Save file',
-                sprintf __('Error converting TIFF to PS: %s'), $error );
-            return;
-        }
-        _post_save_hook( $options{options}{ps}, %{ $options{options} } );
-    }
-    else {
-        _post_save_hook( $options{path}, %{ $options{options} } );
-    }
+    _post_save_hook( $options{path}, %{ $options{options} } );
 
     $self->{return}->enqueue(
         {
