diff -Naur devedeng-4.17.0.orig/setup.py devedeng-4.17.0/setup.py
--- devedeng-4.17.0.orig/setup.py	2022-03-02 16:03:24.000000000 +0100
+++ devedeng-4.17.0/setup.py	2023-07-09 11:07:43.855950815 +0200
@@ -33,10 +33,7 @@
     for lang_name in [f for f in os.listdir('locale')]:
         mofile = os.path.join('locale', lang_name,
                               'LC_MESSAGES', 'devedeng.mo')
-        # translations must be always in /usr/share because Gtk.builder only
-        # search there. If someone knows how to fix this...
-        # share/locale/fr/LC_MESSAGES/
-        target = os.path.join('/usr', 'share', 'locale',
+        target = os.path.join('share', 'locale',
                               lang_name, 'LC_MESSAGES')
         data_files.append((target, [mofile]))
 
