diff -ur pyftpdlib-1.0.1.orig/test/test_ftpd.py pyftpdlib-1.0.1/test/test_ftpd.py
--- test/test_ftpd.py	2013-02-23 03:07:13.000000000 +0800
+++ test/test_ftpd.py	2013-03-13 23:05:48.392965468 +0800
@@ -33,7 +33,7 @@
 import threading
 import unittest
 import socket
-import os
+import os, sys
 import shutil
 import time
 import re
@@ -1358,6 +1358,7 @@
                 self.assertEqual(getmode(), '0555')
 
 
+@unittest.skipIf(hasattr(sys, 'pypy_version_info'), "temp. disabling of pypy")
 class TestFtpStoreData(TestCase):
     """Test STOR, STOU, APPE, REST, TYPE."""
     server_class = FTPd
