--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -467,7 +467,6 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/process/process_metrics.cc',
       'base/profiler/scoped_profile.cc',
       'base/profiler/scoped_tracker.cc',
-      'base/profiler/tracked_time.cc',
       'base/rand_util.cc',
       'base/run_loop.cc',
       'base/sequence_token.cc',
@@ -567,6 +566,7 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/tracked_objects.cc',
       'base/tracking_info.cc',
       'base/unguessable_token.cc',
+      'base/value_iterators.cc',
       'base/values.cc',
       'base/vlog.cc',
   ])
