From 9abbd7da1ddd8003106cc3420170561d475231a9 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Mon, 29 Jun 2020 13:01:27 +0000
Subject: [PATCH] IWYU: add missing include for std::vector

---
 ui/ozone/public/platform_screen.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ui/ozone/public/platform_screen.h b/ui/ozone/public/platform_screen.h
index 9ce293c..ba20597 100644
--- a/ui/ozone/public/platform_screen.h
+++ b/ui/ozone/public/platform_screen.h
@@ -7,6 +7,7 @@
 
 #include <set>
 #include <string>
+#include <vector>
 
 #include "base/component_export.h"
 #include "base/macros.h"
-- 
2.26.2

