Index: tools/gn/bootstrap/bootstrap.py
diff --git a/tools/gn/bootstrap/bootstrap.py b/tools/gn/bootstrap/bootstrap.py
index c3642e8f487cab9ee9d22ef6a4e5c20871593802..12806cb3a9033b4d532f97e87061d8eecf3bd202 100755
--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -154,6 +154,13 @@ def write_buildflag_header_manually(root_gen_dir, header, flags):
 
   os.remove(temp_path)
 
+def write_build_date_header(root_gen_dir):
+  check_call([
+       os.path.join(SRC_ROOT, 'build', 'write_build_date_header.py'),
+       os.path.join(root_gen_dir, 'base/generated_build_date.h'),
+       'default',
+  ])
+
 def build_gn_with_ninja_manually(tempdir, options):
   root_gen_dir = os.path.join(tempdir, 'gen')
   mkdir_p(root_gen_dir)
@@ -167,6 +174,8 @@ def build_gn_with_ninja_manually(tempdir, options):
           'ENABLE_MEMORY_TASK_PROFILER': 'false'
       })
 
+  write_build_date_header(root_gen_dir)
+
   if is_mac:
     # //base/build_time.cc needs base/generated_build_date.h,
     # and this file is only included for Mac builds.
@@ -374,6 +383,7 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/at_exit.cc',
       'base/base_paths.cc',
       'base/base_switches.cc',
+      'base/build_time.cc',
       'base/callback_internal.cc',
       'base/command_line.cc',
       'base/debug/activity_tracker.cc',
@@ -382,6 +392,7 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/debug/stack_trace.cc',
       'base/debug/task_annotator.cc',
       'base/environment.cc',
+      'base/feature_list.cc',
       'base/files/file.cc',
       'base/files/file_descriptor_watcher_posix.cc',
       'base/files/file_enumerator.cc',
@@ -413,6 +424,8 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/message_loop/message_pump.cc',
       'base/message_loop/message_pump_default.cc',
       'base/metrics/bucket_ranges.cc',
+      'base/metrics/field_trial.cc',
+      'base/metrics/field_trial_param_associator.cc',
       'base/metrics/histogram.cc',
       'base/metrics/histogram_base.cc',
       'base/metrics/histogram_samples.cc',
@@ -428,12 +441,14 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/pending_task.cc',
       'base/pickle.cc',
       'base/process/kill.cc',
+      'base/process/memory.cc',
       'base/process/process_handle.cc',
       'base/process/process_iterator.cc',
       'base/process/process_metrics.cc',
       'base/profiler/scoped_profile.cc',
       'base/profiler/scoped_tracker.cc',
       'base/profiler/tracked_time.cc',
+      'base/rand_util.cc',
       'base/run_loop.cc',
       'base/sequence_token.cc',
       'base/sequence_checker_impl.cc',
@@ -533,11 +548,13 @@ def write_gn_ninja(path, root_gen_dir, options):
         'base/files/memory_mapped_file_posix.cc',
         'base/message_loop/message_pump_libevent.cc',
         'base/posix/file_descriptor_shuffle.cc',
+        'base/posix/global_descriptors.cc',
         'base/posix/safe_strerror.cc',
         'base/process/kill_posix.cc',
         'base/process/process_handle_posix.cc',
         'base/process/process_metrics_posix.cc',
         'base/process/process_posix.cc',
+        'base/rand_util_posix.cc',
         'base/strings/string16.cc',
         'base/synchronization/condition_variable_posix.cc',
         'base/synchronization/lock_impl_posix.cc',
@@ -589,6 +606,7 @@ def write_gn_ninja(path, root_gen_dir, options):
         'base/memory/shared_memory_posix.cc',
         'base/nix/xdg_util.cc',
         'base/process/internal_linux.cc',
+        'base/process/memory_linux.cc',
         'base/process/process_handle_linux.cc',
         'base/process/process_iterator_linux.cc',
         'base/process/process_linux.cc',
