--- a/chrome/browser/themes/theme_service.h	2019-06-13 18:56:30.000000000 -0000
+++ b/chrome/browser/themes/theme_service.h	2019-06-17 08:50:57.714289399 -0000
@@ -26,6 +26,7 @@
 #include "extensions/common/extension_id.h"
 #include "ui/base/theme_provider.h"
 #include "ui/native_theme/native_theme_observer.h"
+#include "ui/native_theme/native_theme.h"
 
 class BrowserThemePack;
 class CustomThemeSupplier;
