diff --git a/tools/gn/bootstrap/bootstrap.py b/tools/gn/bootstrap/bootstrap.py
index 9a3740e8..fa81d10 100755
--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -427,6 +427,7 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/profiler/scoped_tracker.cc',
       'base/profiler/tracked_time.cc',
       'base/run_loop.cc',
+      'base/sequence_token.cc',
       'base/sequence_checker_impl.cc',
       'base/sequenced_task_runner.cc',
       'base/sha1.cc',
@@ -439,7 +440,7 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/strings/stringprintf.cc',
       'base/strings/utf_string_conversion_utils.cc',
       'base/strings/utf_string_conversions.cc',
-      'base/synchronization/cancellation_flag.cc',
+      'base/synchronization/atomic_flag.cc',
       'base/synchronization/lock.cc',
       'base/sys_info.cc',
       'base/task_runner.cc',
