For 5.15 kernel with ntfs3 file system

Bug: https://bugs.gentoo.org/830753

--- a/os-probes/mounted/arm64/20microsoft
+++ b/os-probes/mounted/arm64/20microsoft
@@ -15,7 +15,7 @@ fi
 
 # Weed out stuff that doesn't apply to us
 case "$type" in
-	ntfs|ntfs-3g) debug "$1 is a NTFS partition" ;;
+	ntfs|ntfs-3g|ntfs3) debug "$1 is a NTFS partition" ;;
 	vfat) debug "$1 is a FAT32 partition" ;;
 	msdos) debug "$1 is a FAT16 partition" ;;
 	fat) debug "$1 is a FAT partition (mounted by GRUB)" ;;
--- a/os-probes/mounted/x86/20microsoft
+++ b/os-probes/mounted/x86/20microsoft
@@ -15,7 +15,7 @@ fi
 
 # Weed out stuff that doesn't apply to us
 case "$type" in
-	ntfs|ntfs-3g) debug "$1 is a NTFS partition" ;;
+	ntfs|ntfs-3g|ntfs3) debug "$1 is a NTFS partition" ;;
 	vfat) debug "$1 is a FAT32 partition" ;;
 	msdos) debug "$1 is a FAT16 partition" ;;
 	fat) debug "$1 is a FAT partition (mounted by GRUB)" ;;
