From 09fefdb360b69c2de03a2f1c881db87f924d3c76 Mon Sep 17 00:00:00 2001
From: Timo Gurr <timo.gurr@gmail.com>
Date: Mon, 20 Feb 2017 17:33:14 +0100
Subject: [PATCH] Add Exherbo Linux detection

---
 os-probes/mounted/common/90linux-distro | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/os-probes/mounted/common/90linux-distro b/os-probes/mounted/common/90linux-distro
index badfbb1..41a5553 100755
--- a/os-probes/mounted/common/90linux-distro
+++ b/os-probes/mounted/common/90linux-distro
@@ -137,6 +137,9 @@ if (ls "$dir"/lib*/ld*.so* && [ -d "$dir/boot" ] || ls "$dir"/usr/lib*/ld*.so*)
 	elif [ -e "$dir/etc/devuan_version" ]; then
 		short="Devuan"
 		long="$(printf "Devuan GNU/Linux (%s)\n" "$(cat "$dir/etc/devuan_version")")"
+	elif [ -e "$dir/etc/exherbo-release" ]; then
+		short="Exherbo"
+		long="Exherbo Linux"
 	else
 		short="Linux"
 		long="unknown Linux distribution"
-- 
2.11.1

