--- configure.ac~	2012-03-15 13:50:43.000000000 +0000
+++ configure.ac	2012-10-10 12:00:15.225031516 +0000
@@ -328,7 +328,7 @@
 AC_SUBST(PYTHON_LIBS)
 
 AC_MSG_CHECKING(python library directory)
-pythondir=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(1, 0, '\\${prefix}')"`
+pythondir=`$PYTHON -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"`
 AC_MSG_RESULT($pythondir)
 AC_SUBST(pythondir)
 
@@ -661,7 +661,7 @@
 			   [build Python bindings for libmapi]),
 			   enable_pymapi="$enableval",
 			   enable_pymapi=no)
-PYCDIR=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_python_lib(1, prefix='\\$(prefix)')"`
+PYCDIR=`$PYTHON -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"`
 AC_SUBST(PYCDIR)
 
 dnl Enable pyopenchange by default
