http://bugs.gentoo.org/412473

--- src/alsa_backend.c
+++ src/alsa_backend.c
@@ -22,10 +22,8 @@
 //##############################################################################
 
 #include <alsa/asoundlib.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <glib/gstdio.h>
-#include <glib/glist.h>
-#include <glib/giochannel.h>
 
 #include "alsa_backend.h"
 
--- src/oss_backend.c
+++ src/oss_backend.c
@@ -26,10 +26,8 @@
 #include <fcntl.h>
 #include <assert.h>
 #include <stdlib.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <glib/gstdio.h>
-#include <glib/glist.h>
-#include <glib/giochannel.h>
 
 #include "oss_backend.h"
 
