diff -ru radmind-1.7.1~/configure radmind-1.7.1/configure
--- radmind-1.7.1~/configure	2007-02-10 19:20:01.000000000 +0100
+++ radmind-1.7.1/configure	2007-02-10 19:22:04.000000000 +0100
@@ -5265,12 +5265,12 @@
 echo "${ECHO_T}$ac_cv_header_dns_sd_h" >&6
 
 fi
-if test $ac_cv_header_dns_sd_h = yes; then
-  cat >>confdefs.h <<\_ACEOF
-#define HAVE_DNSSD 1
-_ACEOF
-
-fi
+#if test $ac_cv_header_dns_sd_h = yes; then
+#  cat >>confdefs.h <<\_ACEOF
+##define HAVE_DNSSD 1
+#_ACEOF
+#
+#fi
 
 
 
