diff -uNr libss7-1.0.2.ORIG/Makefile libss7-1.0.2/Makefile
--- libss7-1.0.2.ORIG/Makefile	2012-08-23 10:49:15.341155719 +0100
+++ libss7-1.0.2/Makefile	2012-08-23 10:50:35.403236983 +0100
@@ -54,7 +54,7 @@
 	ranlib $(STATIC_LIBRARY)
 
 $(DYNAMIC_LIBRARY): $(DYNAMIC_OBJS)
-	$(CC) -shared $(SOFLAGS) -o $@ $(DYNAMIC_OBJS)
+	$(CC) -shared $(SOFLAGS) $(LDFLAGS) -o $@ $(DYNAMIC_OBJS)
 	$(LDCONFIG) $(LDCONFIG_FLAGS) .
 	ln -sf libss7.so.1 libss7.so
 	ln -sf libss7.so.1.0 libss7.so.1
