--- a/setup.py
+++ b/setup.py
@@ -80,7 +80,7 @@
 
 # Compile covariance functions
 config.add_extension(name='gp.cov_funs.isotropic_cov_funs',\
-sources=['pymc/gp/cov_funs/isotropic_cov_funs.f','blas/BLAS/dscal.f'],\
+sources=['pymc/gp/cov_funs/isotropic_cov_funs.f'],\
 extra_info=lapack_info)
 
 config.add_extension(name='gp.cov_funs.distances',sources=['pymc/gp/cov_funs/distances.f'], extra_info=lapack_info)
