diff --git a/setup.py b/setup.py
index ce5309ee..02f38c61 100755
--- a/setup.py
+++ b/setup.py
@@ -1028,11 +1028,6 @@ def package(args: Options, bundle_type: str) -> None:
     safe_makedirs(launcher_dir)
     build_launcher(args, launcher_dir, bundle_type)
     os.makedirs(os.path.join(libdir, 'logo'))
-    build_terminfo = runpy.run_path('build-terminfo', run_name='import_build')  # type: ignore
-    for x in (libdir, os.path.join(ddir, 'share')):
-        odir = os.path.join(x, 'terminfo')
-        safe_makedirs(odir)
-        build_terminfo['compile_terminfo'](odir)
     shutil.copy2('__main__.py', libdir)
     shutil.copy2('logo/kitty-128.png', os.path.join(libdir, 'logo'))
     shutil.copy2('logo/kitty.png', os.path.join(libdir, 'logo'))
