--- kitty-9999/setup.py.orig	2019-07-16 18:13:07.072868752 -0500
+++ kitty-9999/setup.py	2019-07-16 18:13:40.082039929 -0500
@@ -705,9 +705,9 @@
         run_tool(['make', 'docs'])
     copy_man_pages(ddir)
     copy_html_docs(ddir)
-    icdir = os.path.join(ddir, 'share', 'icons', 'hicolor', '256x256', 'apps')
+    icdir = os.path.join(ddir, 'share', 'icons', 'hicolor', 'scalable', 'apps')
     safe_makedirs(icdir)
-    shutil.copy2('logo/kitty.png', icdir)
+    shutil.copy2('logo/kitty.svg', icdir)
     deskdir = os.path.join(ddir, 'share', 'applications')
     safe_makedirs(deskdir)
     with open(os.path.join(deskdir, 'kitty.desktop'), 'w') as f:
