--- /Makefile.orig	2009-01-16 17:01:47.000000000 +0100
+++ /Makefile	2009-01-16 17:04:18.000000000 +0100
@@ -2,11 +2,11 @@
 CPPFLAGS = -I/usr/X11R6/include -I/usr/local/include
 LIBS = -lXpm -lX11 -ldockapp  
 LDFLAGS = -L/usr/X11R6/lib -L/usr/local/lib
-CFLAGS = -O2 -Wall
+CFLAGS = -Wall
 DFLAGS = -ggdb3 -Wall
+DESTDIR = /usr/bin/
 
 wmlpq: wmlpq.c wmlpq-master.xpm
-	cc $(CFLAGS) $(LDFLAGS) $(CPPFLAGS) -o wmlpq $(LIBS) wmlpq.c
+	cc $(CFLAGS) $(LDFLAGS) $(CPPFLAGS) wmlpq.c -o wmlpq $(LIBS)
-	strip wmlpq
 
 debug: wmlpq.c wmlpq-master.xpm
@@ -18,10 +19,8 @@
 	rm -f wmlpq 
 
 install: wmlpq
-	cp -f wmlpq /usr/local/bin/
-	chmod 755 /usr/local/bin/wmlpq
-	cp -f wmlpq.1x /usr/local/man/man1/
-	chmod 644 /usr/local/man/man1/wmlpq.1x
+	cp -f wmlpq ${DESTDIR}
+	chmod 755 ${DESTDIR}/wmlpq
 
 install-debian: wmlpq
 	cp -f wmlpq $(prefix)/usr/X11R6/bin/
