--- contrib/slapd-modules/smbk5pwd/Makefile.ORIG	2006-05-17 13:11:57.194660019 +0300
+++ contrib/slapd-modules/smbk5pwd/Makefile	2006-05-17 13:11:14.503082288 +0300
@@ -9,29 +9,39 @@
 # top-level directory of the distribution or, alternatively, at
 # <http://www.OpenLDAP.org/license.html>.
 
+#libexecdir=/usr/lib/openldap
+moduledir=$(libexecdir)/openldap
 LIBTOOL=../../../libtool
-OPT=-g -O2
+#OPT=
 CC=gcc
 
 # Omit DO_KRB5 or DO_SAMBA if you don't want to support it.
-DEFS=-DDO_KRB5 -DDO_SAMBA
+#DEFS=
 
-HEIMDAL_INC=-I/usr/heimdal/include
+#KRB5_INC=
 SSL_INC=
 LDAP_INC=-I../../../include -I../../../servers/slapd
-INCS=$(LDAP_INC) $(HEIMDAL_INC) $(SSL_INC)
+INCS=$(LDAP_INC) $(SSL_INC) $(KRB5_INC)
 
-HEIMDAL_LIB=-L/usr/heimdal/lib -lkrb5 -lkadm5srv
+KRB5_LIB=-lkrb5 -lkadm5srv
 SSL_LIB=-lcrypto
-LDAP_LIB=-lldap_r -llber
-LIBS=$(LDAP_LIB) $(HEIMDAL_LIB) $(SSL_LIB)
-
+LDAP_LIB=-L../../../libraries/libldap_r -lldap_r -llber
+ifneq (DDO_KRB5,$(findstring DDO_KRB5,$(DEFS)))
+	LIBS=$(LDAP_LIB) $(SSL_LIB)
+else
+	LIBS=$(LDAP_LIB) $(KRB5_LIB) $(SSL_LIB)
+endif
+	
 all:	smbk5pwd.la
 
 
 smbk5pwd.lo:	smbk5pwd.c
-	$(LIBTOOL) --mode=compile $(CC) $(OPT) $(DEFS) $(INCS) -c $?
+	$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) $(DEFS) $(INCS) -c $?
 
 smbk5pwd.la:	smbk5pwd.lo
-	$(LIBTOOL) --mode=link $(CC) $(OPT) -version-info 0:0:0 \
-	-rpath /usr/local/libexec/openldap -module -o $@ $? $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -version-info 0:0:0 \
+	-rpath $(moduledir) -module -o $@ $? $(LIBS)
+
+install-mod:
+	$(LIBTOOL) --mode=install ../../../build/shtool install -c \
+	-m 755 smbk5pwd.la $(DESTDIR)$(moduledir)
