diff -ur openldap-2.4.17.orig/contrib/ldapc++/src/SaslInteractionHandler.cpp openldap-2.4.17/contrib/ldapc++/src/SaslInteractionHandler.cpp
--- openldap-2.4.17.orig/contrib/ldapc++/src/SaslInteractionHandler.cpp	2008-04-15 02:09:26.000000000 +0300
+++ openldap-2.4.17/contrib/ldapc++/src/SaslInteractionHandler.cpp	2009-08-10 13:21:24.000000000 +0300
@@ -13,6 +13,7 @@
 #include <termios.h>
 #endif
 
+#include <stdio.h>
 #include <string.h>
 #include "SaslInteractionHandler.h"
 #include "SaslInteraction.h"
