--- a/CMakeLists.txt	2020-06-07 13:41:57.205055768 +0200
+++ b/CMakeLists.txt	2020-06-07 16:02:45.704023540 +0200
@@ -10,12 +10,14 @@
 #example command:
 #cmake -DROCclr_DIR=/rocclr -DUSE_COMGR_LIBRARY=yes -DLIBROCclr_STATIC_DIR=/build ..
 
+include (GNUInstallDirs)
+
 set(OPENCL_ICD_LOADER_HEADERS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/khronos/headers/opencl2.2" CACHE PATH "")
 set(BUILD_SHARED_LIBS "Build shared libs" ON)
 
 add_subdirectory(khronos/icd)
 add_subdirectory(amdocl)
-add_subdirectory(tools/clinfo)
+#add_subdirectory(tools/clinfo)
 add_subdirectory(tools/cltrace)
 
 # get date information based on UTC
@@ -45,34 +47,34 @@
 ###--- Packaging ------------------------------------------------------------###
 
 # MAIN package
-install(PROGRAMS $<TARGET_FILE:clinfo>
-        DESTINATION bin
-        COMPONENT MAIN)
-install(PROGRAMS $<TARGET_FILE:cltrace>
-        DESTINATION lib
-        COMPONENT MAIN)
+#install(PROGRAMS $<TARGET_FILE:clinfo>
+#        DESTINATION bin
+#        COMPONENT MAIN)
+#install(PROGRAMS $<TARGET_FILE:cltrace>
+#        DESTINATION lib64
+#        COMPONENT MAIN)
 install(PROGRAMS $<TARGET_FILE:amdocl64>
-        DESTINATION lib
-        COMPONENT MAIN)
-install(PROGRAMS $<TARGET_FILE:OpenCL>
-        DESTINATION lib
-        COMPONENT MAIN)
-install(PROGRAMS $<TARGET_SONAME_FILE:OpenCL>
-        DESTINATION lib
+        DESTINATION lib64
         COMPONENT MAIN)
+#install(PROGRAMS $<TARGET_FILE:OpenCL>
+#        DESTINATION lib
+#        COMPONENT MAIN)
+#install(PROGRAMS $<TARGET_SONAME_FILE:OpenCL>
+#        DESTINATION lib
+#        COMPONENT MAIN)
 
 # DEV package
-install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/khronos/headers/opencl2.2/CL"
-        DESTINATION include
-        COMPONENT DEV
-        USE_SOURCE_PERMISSIONS
-        PATTERN cl_d3d10.h EXCLUDE
-        PATTERN cl_d3d11.h EXCLUDE
-        PATTERN cl_dx9_media_sharing.h EXCLUDE
-        PATTERN cl_egl.h EXCLUDE)
-install(PROGRAMS $<TARGET_LINKER_FILE:OpenCL>
-        DESTINATION lib
-        COMPONENT DEV)
+#install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/khronos/headers/opencl2.2/CL"
+#        DESTINATION include
+#        COMPONENT DEV
+#        USE_SOURCE_PERMISSIONS
+#        PATTERN cl_d3d10.h EXCLUDE
+#        PATTERN cl_d3d11.h EXCLUDE
+#        PATTERN cl_dx9_media_sharing.h EXCLUDE
+#        PATTERN cl_egl.h EXCLUDE)
+#install(PROGRAMS $<TARGET_LINKER_FILE:OpenCL>
+#        DESTINATION lib
+#        COMPONENT DEV)
 
 #############################
 # Packaging steps
