From e4d46b478f78c4464c7b35058154948c928a013b Mon Sep 17 00:00:00 2001
From: Arfrever <arfrever@github.com>
Date: Thu, 29 May 2014 18:07:21 +0200
Subject: [PATCH] Don't install tests (but include them in PyPI package).

Fixes #9
---
 MANIFEST.in | 1 +
 setup.py    | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/MANIFEST.in b/MANIFEST.in
index 1845310..6217360 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -2,3 +2,4 @@ include COPYING
 include README.md
 include ChangeLog.rst
 include AUTHORS.rst
+recursive-include test *.html *.md *.py
diff --git a/setup.py b/setup.py
index 3b7085d..0b2cf08 100644
--- a/setup.py
+++ b/setup.py
@@ -65,7 +65,6 @@ def run(self):
     """,
     license='GNU GPL 3',
     requires=requires_list,
-    packages=find_packages(),
     py_modules=['html2text'],
     include_package_data=True,
     zip_safe=False,
-- 
1.9.3

