diff -Naur ttcut-old/avstream/ttac3audioheader.h ttcut/avstream/ttac3audioheader.h
--- ttcut-old/avstream/ttac3audioheader.h	2006-02-03 21:32:50.000000000 +0100
+++ ttcut/avstream/ttac3audioheader.h	2009-09-20 02:13:04.000000000 +0200
@@ -116,7 +116,7 @@
   2,1,2,3,3,4,4,5
 };
 
-__attribute__ ((unused))static char* AC3Mode[8] =
+__attribute__ ((unused))static const char* AC3Mode[8] =
 {
   "1+1", "1/0", "2/0", "3/0",
   "2/1", "3/1", "2/2", "3/2"
diff -Naur ttcut-old/avstream/ttac3audiostream.cpp ttcut/avstream/ttac3audiostream.cpp
--- ttcut-old/avstream/ttac3audiostream.cpp	2007-04-30 11:06:19.000000000 +0200
+++ ttcut/avstream/ttac3audiostream.cpp	2009-09-20 02:13:04.000000000 +0200
@@ -108,7 +108,7 @@
 
   audio_header->setHeaderOffset( stream_buffer->currentOffset() - 8 ); // +Syncwort
 
-  audio_header->crc1            = daten[0]<<8+daten[1];
+  audio_header->crc1            = daten[0]<<(8+daten[1]);
   audio_header->fscod           = (uint8_t)((daten[2]&0xc0)>>6);
   audio_header->frmsizecod      = (uint8_t)(daten[2]&0x3f);
   audio_header->syncframe_words = AC3FrameLength[audio_header->fscod][audio_header->frmsizecod];
diff -Naur ttcut-old/avstream/ttaviwriter.cpp ttcut/avstream/ttaviwriter.cpp
--- ttcut-old/avstream/ttaviwriter.cpp	2007-04-27 10:32:20.000000000 +0200
+++ ttcut/avstream/ttaviwriter.cpp	2009-09-20 02:13:04.000000000 +0200
@@ -126,7 +126,7 @@
   //qDebug( "%sAVI frame info: width: %d x height: %d",c_name,frameInfo->width,frameInfo->height );
 
   //TODO: avoid setting hard coded frame rate!
-  AVI_set_video(avi_file, frameInfo->width, frameInfo->height, 25.0, "YV12");
+  AVI_set_video(avi_file, frameInfo->width, frameInfo->height, 25.0, (char*)"YV12");
   
   ref_data = new uint8_t [frameInfo->size+2*frameInfo->chroma_size];
 
diff -Naur ttcut-old/avstream/ttdtsaudioheader.h ttcut/avstream/ttdtsaudioheader.h
--- ttcut-old/avstream/ttdtsaudioheader.h	2006-02-02 20:43:48.000000000 +0100
+++ ttcut/avstream/ttdtsaudioheader.h	2009-09-20 02:13:04.000000000 +0200
@@ -78,7 +78,7 @@
   2048000, 3072000, 3840000, 4096000, 0, 0 
 }; 
 
-__attribute__ ((unused))static char* dts_acmod[64] =
+__attribute__ ((unused))static const char* dts_acmod[64] =
 { 
   "1","DM","2/0","2/0", 
   "2/0","3/0","2.1/0","3.1/0", 
diff -Naur ttcut-old/avstream/ttfilebuffer.cpp ttcut/avstream/ttfilebuffer.cpp
--- ttcut-old/avstream/ttfilebuffer.cpp	2007-09-02 13:15:09.000000000 +0200
+++ ttcut/avstream/ttfilebuffer.cpp	2009-09-20 02:13:04.000000000 +0200
@@ -773,8 +773,8 @@
   /* new position outside buffer*/
   else
   {
-    if ( new_pos > -1             &&
-        (file_mode  == fm_open_read && new_pos < stream_length) ||
+    if ( (new_pos > -1             &&
+        (file_mode  == fm_open_read && new_pos < stream_length)) ||
         ((file_mode == fm_open_write || file_mode == fm_create) &&
          new_pos < stream_length + 1) )
     {
diff -Naur ttcut-old/gui/ttcutmainwindow.cpp ttcut/gui/ttcutmainwindow.cpp
--- ttcut-old/gui/ttcutmainwindow.cpp	2007-09-02 13:15:09.000000000 +0200
+++ ttcut/gui/ttcutmainwindow.cpp	2009-09-20 02:13:04.000000000 +0200
@@ -1030,8 +1030,8 @@
     int num_header = current_audio_stream->createHeaderList();
 
     // error reading audio stream or user abort during operation
-    if ( num_header == 0 && audio_type->avStreamType() != TTAVTypes::pcm_audio ||
-        num_header == 1 && audio_type->avStreamType() == TTAVTypes::pcm_audio    ) {
+    if ( (num_header == 0 && audio_type->avStreamType() != TTAVTypes::pcm_audio) ||
+        (num_header == 1 && audio_type->avStreamType() == TTAVTypes::pcm_audio)    ) {
 
       log->errorMsg( oName, "error reading audio stream; no header list (!)" );
       delete current_audio_stream;
diff -Naur ttcut-old/ui/pixmaps/addtolist_18.xpm ttcut/ui/pixmaps/addtolist_18.xpm
--- ttcut-old/ui/pixmaps/addtolist_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/addtolist_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * addtolist_18_xpm[] = {
+static const char * addtolist_18_xpm[] = {
 "18 18 163 2",
 "  	c None",
 ". 	c #3E98E5",
diff -Naur ttcut-old/ui/pixmaps/apply_18.xpm ttcut/ui/pixmaps/apply_18.xpm
--- ttcut-old/ui/pixmaps/apply_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/apply_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * apply_18_xpm[] = {
+static const char * apply_18_xpm[] = {
 "18 18 46 1",
 " 	c None",
 ".	c #707070",
diff -Naur ttcut-old/ui/pixmaps/bframe_18.xpm ttcut/ui/pixmaps/bframe_18.xpm
--- ttcut-old/ui/pixmaps/bframe_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/bframe_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char* const b_frame_18_xpm[] = { 
+static const char * b_frame_18_xpm[] = { 
 "18 18 16 1",
 ". c None",
 "d c #faf515",
diff -Naur ttcut-old/ui/pixmaps/button_ok.xpm ttcut/ui/pixmaps/button_ok.xpm
--- ttcut-old/ui/pixmaps/button_ok.xpm	2007-04-27 10:32:20.000000000 +0200
+++ ttcut/ui/pixmaps/button_ok.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * button_ok_xpm[] = {
+static const char * button_ok_xpm[] = {
 "32 32 132 2",
 "  	c None",
 ". 	c #18AD00",
diff -Naur ttcut-old/ui/pixmaps/chapter_18.xpm ttcut/ui/pixmaps/chapter_18.xpm
--- ttcut-old/ui/pixmaps/chapter_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/chapter_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * chapter_18_xpm[] = {
+static const char * chapter_18_xpm[] = {
 "18 18 157 2",
 "  	c None",
 ". 	c #9EAAD6",
diff -Naur ttcut-old/ui/pixmaps/clock_16.xpm ttcut/ui/pixmaps/clock_16.xpm
--- ttcut-old/ui/pixmaps/clock_16.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/clock_16.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * clock_16_xpm[] = {
+static const char * clock_16_xpm[] = {
 "16 16 168 2",
 "  	c None",
 ". 	c #2A6AC0",
diff -Naur ttcut-old/ui/pixmaps/clock_18.xpm ttcut/ui/pixmaps/clock_18.xpm
--- ttcut-old/ui/pixmaps/clock_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/clock_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * clock_18_xpm[] = {
+static const char * clock_18_xpm[] = {
 "18 18 202 2",
 "  	c None",
 ". 	c #2360BB",
diff -Naur ttcut-old/ui/pixmaps/cutaudio_18.xpm ttcut/ui/pixmaps/cutaudio_18.xpm
--- ttcut-old/ui/pixmaps/cutaudio_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/cutaudio_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * cutaudio_18_xpm[] = {
+static const char * cutaudio_18_xpm[] = {
 "11 18 2 1",
 " 	c None",
 ".	c #000000",
diff -Naur ttcut-old/ui/pixmaps/cutav_18.xpm ttcut/ui/pixmaps/cutav_18.xpm
--- ttcut-old/ui/pixmaps/cutav_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/cutav_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * cutav_18_xpm[] = {
+static const char * cutav_18_xpm[] = {
 "18 18 105 2",
 "  	c None",
 ". 	c #CDCDD9",
diff -Naur ttcut-old/ui/pixmaps/exit_16.xpm ttcut/ui/pixmaps/exit_16.xpm
--- ttcut-old/ui/pixmaps/exit_16.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/exit_16.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * exit_16_xpm[] = {
+static const char * exit_16_xpm[] = {
 "16 16 174 2",
 "  	c None",
 ". 	c #AE3725",
diff -Naur ttcut-old/ui/pixmaps/fileclose_18.xpm ttcut/ui/pixmaps/fileclose_18.xpm
--- ttcut-old/ui/pixmaps/fileclose_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/fileclose_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * fileclose_18_xpm[] = {
+static const char * fileclose_18_xpm[] = {
 "18 18 138 2",
 "  	c None",
 ". 	c #3D3D5D",
diff -Naur ttcut-old/ui/pixmaps/filenew_16.xpm ttcut/ui/pixmaps/filenew_16.xpm
--- ttcut-old/ui/pixmaps/filenew_16.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/filenew_16.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * filenew_16_xpm[] = {
+static const char * filenew_16_xpm[] = {
 "16 16 121 2",
 "  	c None",
 ". 	c #C5CAD6",
diff -Naur ttcut-old/ui/pixmaps/fileopen_16.xpm ttcut/ui/pixmaps/fileopen_16.xpm
--- ttcut-old/ui/pixmaps/fileopen_16.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/fileopen_16.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * fileopen_16_xpm[] = {
+static const char * fileopen_16_xpm[] = {
 "16 16 183 2",
 "  	c None",
 ". 	c #9BA1D9",
diff -Naur ttcut-old/ui/pixmaps/filesave_16.xpm ttcut/ui/pixmaps/filesave_16.xpm
--- ttcut-old/ui/pixmaps/filesave_16.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/filesave_16.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * filesave_16_xpm[] = {
+static const char * filesave_16_xpm[] = {
 "16 16 174 2",
 "  	c None",
 ". 	c #B8B8CC",
diff -Naur ttcut-old/ui/pixmaps/filesaveas_16.xpm ttcut/ui/pixmaps/filesaveas_16.xpm
--- ttcut-old/ui/pixmaps/filesaveas_16.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/filesaveas_16.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * filesaveas_16_xpm[] = {
+static const char * filesaveas_16_xpm[] = {
 "16 16 180 2",
 "  	c None",
 ". 	c #F2C14C",
diff -Naur ttcut-old/ui/pixmaps/goto_18.xpm ttcut/ui/pixmaps/goto_18.xpm
--- ttcut-old/ui/pixmaps/goto_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/goto_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * goto_18_xpm[] = {
+static const char * goto_18_xpm[] = {
 "18 18 82 1",
 " 	c None",
 ".	c #127DFF",
diff -Naur ttcut-old/ui/pixmaps/iframe_18.xpm ttcut/ui/pixmaps/iframe_18.xpm
--- ttcut-old/ui/pixmaps/iframe_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/iframe_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char* const i_frame_18_xpm[] = { 
+static const char * i_frame_18_xpm[] = { 
 "18 18 9 1",
 ". c None",
 "d c #000096",
diff -Naur ttcut-old/ui/pixmaps/next_18.xpm ttcut/ui/pixmaps/next_18.xpm
--- ttcut-old/ui/pixmaps/next_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/next_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * next_18_xpm[] = {
+static const char * next_18_xpm[] = {
 "18 18 59 1",
 " 	c None",
 ".	c #000000",
diff -Naur ttcut-old/ui/pixmaps/note_18.xpm ttcut/ui/pixmaps/note_18.xpm
--- ttcut-old/ui/pixmaps/note_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/note_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * note_18_xpm[] = {
+static const char * note_18_xpm[] = {
 "18 18 93 2",
 "  	c None",
 ". 	c #C5AD09",
diff -Naur ttcut-old/ui/pixmaps/pause_18.xpm ttcut/ui/pixmaps/pause_18.xpm
--- ttcut-old/ui/pixmaps/pause_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/pause_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * pause_18_xpm[] = {
+static const char * pause_18_xpm[] = {
 "18 18 67 1",
 " 	c None",
 ".	c #000000",
diff -Naur ttcut-old/ui/pixmaps/pframe_18.xpm ttcut/ui/pixmaps/pframe_18.xpm
--- ttcut-old/ui/pixmaps/pframe_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/pframe_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char* const p_frame_18_xpm[] = { 
+static const char * const p_frame_18_xpm[] = { 
 "18 18 15 1",
 ". c None",
 "e c #fa0150",
diff -Naur ttcut-old/ui/pixmaps/play_18.xpm ttcut/ui/pixmaps/play_18.xpm
--- ttcut-old/ui/pixmaps/play_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/play_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * play_18_xpm[] = {
+static const char * play_18_xpm[] = {
 "18 18 38 1",
 " 	c None",
 ".	c #000000",
diff -Naur ttcut-old/ui/pixmaps/prev_18.xpm ttcut/ui/pixmaps/prev_18.xpm
--- ttcut-old/ui/pixmaps/prev_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/prev_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * prev_18_xpm[] = {
+static const char * prev_18_xpm[] = {
 "18 18 64 1",
 " 	c None",
 ".	c #0E0E0E",
diff -Naur ttcut-old/ui/pixmaps/preview_18.xpm ttcut/ui/pixmaps/preview_18.xpm
--- ttcut-old/ui/pixmaps/preview_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/preview_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * preview_18_xpm[] = {
+static const char * preview_18_xpm[] = {
 "18 18 134 2",
 "  	c None",
 ". 	c #A9ACB6",
diff -Naur ttcut-old/ui/pixmaps/saveimage_16.xpm ttcut/ui/pixmaps/saveimage_16.xpm
--- ttcut-old/ui/pixmaps/saveimage_16.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/saveimage_16.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * saveimage_16_xpm[] = {
+static const char * saveimage_16_xpm[] = {
 "17 18 80 1",
 " 	c None",
 ".	c #000000",
diff -Naur ttcut-old/ui/pixmaps/search_18.xpm ttcut/ui/pixmaps/search_18.xpm
--- ttcut-old/ui/pixmaps/search_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/search_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * search_18_xpm[] = {
+static const char * search_18_xpm[] = {
 "18 18 126 2",
 "  	c None",
 ". 	c #000000",
diff -Naur ttcut-old/ui/pixmaps/settings_16.xpm ttcut/ui/pixmaps/settings_16.xpm
--- ttcut-old/ui/pixmaps/settings_16.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/settings_16.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * settings_16_xpm[] = {
+static const char * settings_16_xpm[] = {
 "16 16 102 2",
 "  	c None",
 ". 	c #C1C1CE",
diff -Naur ttcut-old/ui/pixmaps/_settings_18.xpm ttcut/ui/pixmaps/_settings_18.xpm
--- ttcut-old/ui/pixmaps/_settings_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/_settings_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * settings_18_xpm[] = {
+static const char * settings_18_xpm[] = {
 "16 16 102 2",
 "  	c None",
 ". 	c #C1C1CE",
diff -Naur ttcut-old/ui/pixmaps/settings_18.xpm ttcut/ui/pixmaps/settings_18.xpm
--- ttcut-old/ui/pixmaps/settings_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/settings_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * settings_18_xpm[] = {
+static const char * settings_18_xpm[] = {
 "16 16 102 2",
 "  	c None",
 ". 	c #C1C1CE",
diff -Naur ttcut-old/ui/pixmaps/stop_18.xpm ttcut/ui/pixmaps/stop_18.xpm
--- ttcut-old/ui/pixmaps/stop_18.xpm	2006-02-27 18:05:47.000000000 +0100
+++ ttcut/ui/pixmaps/stop_18.xpm	2009-09-20 02:13:04.000000000 +0200
@@ -1,5 +1,5 @@
 /* XPM */
-static char * stop_18_xpm[] = {
+static const char * stop_18_xpm[] = {
 "18 18 39 1",
 " 	c None",
 ".	c #1D1D1C",
