diff -ur astropy-0.4.1/astropy/extern/six.py astropy-0.4.1.six/astropy/extern/six.py
--- astropy-0.4.1/astropy/extern/six.py	2014-09-01 21:57:56.663393218 +0200
+++ astropy-0.4.1.six/astropy/extern/six.py	2014-09-01 21:59:02.940933294 +0200
@@ -13,7 +13,7 @@
 # Update this to prevent Astropy from using its bundled copy of six
 # (but only if some other version of at least _SIX_MIN_VERSION can
 # be provided)
-_SIX_SEARCH_PATH = ['astropy.extern.bundled.six', 'six']
+_SIX_SEARCH_PATH = ['six', 'astropy.extern.bundled.six']
 
 
 def _find_module(name, path=None):
