 astropy/tests/helper.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/astropy/tests/helper.py b/astropy/tests/helper.py
index 97d2617..a75db15 100644
--- a/astropy/tests/helper.py
+++ b/astropy/tests/helper.py
@@ -42,7 +42,7 @@ __all__ = ['raises', 'enable_deprecations_as_exceptions', 'remote_data',
            'pickle_protocol', 'generic_recursive_equality_test']
 
 
-if os.environ.get('ASTROPY_USE_SYSTEM_PYTEST') or '_pytest' in sys.modules:
+if True or os.environ.get('ASTROPY_USE_SYSTEM_PYTEST') or '_pytest' in sys.modules:
     import pytest
 
 else:
