--- a/libraries/extos/extos.c	2023-12-15 10:12:53.396506457 +0100
+++ b/libraries/extos/extos.c	2023-12-15 14:32:20.464092779 +0100
@@ -262,7 +262,7 @@
     if (result != 1) goto extos_pfilter_error_fd_remapping;
     do result = dup(pipe_err[1]); while (result < 0 && errno == EINTR);
     if (result != 2) goto extos_pfilter_error_fd_remapping;
-    execvp(filename, args);
+    execvp(filename, (char * const*)args);
     status_buf[0] = errno;
     do result = write(pipe_status[1], status_buf, 1); while (result < 0 && errno == EINTR);
     _exit(0);
@@ -287,7 +287,7 @@
   lua_settop(L, 0);
   lua_newtable(L);  // 1
   while (1) {
-    readdir_r(dir, &entry_buffer, &entry);
+    entry = readdir(dir);
     if (!entry) break;
     // Linux doesn't have d_namlen
     //lua_pushlstring(L, entry->d_name, entry->d_namlen);
