From 9c423cd2d24b6de924afc6e0d715594bb056288d Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <g.dartiguelongue@lexfo.fr>
Date: Wed, 30 Aug 2017 11:27:50 +0200
Subject: [PATCH 2/3] Add support for Gentoo in distribution detection

---
 m4/guestfs_appliance.m4 | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/m4/guestfs_appliance.m4 b/m4/guestfs_appliance.m4
index 890b1999c..9b79e1a41 100644
--- a/m4/guestfs_appliance.m4
+++ b/m4/guestfs_appliance.m4
@@ -114,6 +114,8 @@ elif test -f /etc/frugalware-release; then
     DISTRO=FRUGALWARE
 elif test -f /etc/mageia-release; then
     DISTRO=MAGEIA
+elif test -f /etc/gentoo-release; then
+    DISTRO=GENTOO
 else
 dnl fallback option
     DISTRO=REDHAT
-- 
2.14.1

