Bug: https://bugs.gentoo.org/651012
Bug: https://bugs.gentoo.org/679210

--- a/src/plugins/tts/speechdispatcher/speechdispatcher.pro
+++ b/src/plugins/tts/speechdispatcher/speechdispatcher.pro
@@ -5,6 +5,7 @@
 load(qt_plugin)
 
 QT += core texttospeech
+QT -= gui
 
 CONFIG += link_pkgconfig
 packagesExist(speech-dispatcher): PKGCONFIG = speech-dispatcher
--- a/src/doc/doc.pro
+++ b/src/doc/doc.pro
@@ -2,6 +2,7 @@
 
 CONFIG += force_qt
 QT += core-private
+QT -= doc
 
 QMAKE_DOCS = $$PWD/qtspeech.qdocconf
 
