--- a/CMakeLists.txt	2018-10-31 17:52:18.252760284 -0400
+++ a/CMakeLists.txt	2018-10-31 17:52:54.507461091 -0400
@@ -1272,12 +1272,6 @@
     #COMPONENT runtime
     )
 
-    execute_process(COMMAND ln -s /opt/js8call/bin/js8call ljs8call)
-
-    install(FILES
-        ${CMAKE_BINARY_DIR}/ljs8call DESTINATION /usr/bin/ RENAME js8call
-        #COMPONENT runtime
-    )
 endif (NOT WIN32 AND NOT APPLE)
 
 
