--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -273,9 +273,9 @@ MACRO( create_docs _sources )
 	ENDFOREACH( curFile )
 ENDMACRO( create_docs )
 
-CREATE_DOCS( docs_bz2 AUTHORS CHANGELOG README TODO )
-ADD_CUSTOM_TARGET( docs ALL DEPENDS ${docs_bz2} )
-INSTALL( FILES ${docs_bz2} DESTINATION share/doc/aqemu )
+#CREATE_DOCS( docs_bz2 AUTHORS CHANGELOG README TODO )
+#ADD_CUSTOM_TARGET( docs ALL DEPENDS ${docs_bz2} )
+#INSTALL( FILES ${docs_bz2} DESTINATION share/doc/aqemu )
 
 # Build RCC files (GUI icons)
 MACRO( pack_icons _sources )
