From 8b7648502b003d433a08333b041baf1d4960c9d4 Mon Sep 17 00:00:00 2001
From: Nicolas Cedilnik <nicoco@nicoco.fr>
Date: Mon, 22 Feb 2021 13:07:15 +0100
Subject: [PATCH] Fix deprecation warning

---
 slixmpp/thirdparty/orderedset.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/slixmpp/thirdparty/orderedset.py b/slixmpp/thirdparty/orderedset.py
index f6642db3..43023af4 100644
--- a/slixmpp/thirdparty/orderedset.py
+++ b/slixmpp/thirdparty/orderedset.py
@@ -20,9 +20,9 @@
 #     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 #     OTHER DEALINGS IN THE SOFTWARE.
 
-import collections
+from collections.abc import MutableSet
 
-class OrderedSet(collections.MutableSet):
+class OrderedSet(MutableSet):
 
     def __init__(self, iterable=None):
         self.end = end = []
-- 
GitLab

