--- configure.ac.old    2008-11-21 14:41:15.000000000 +0100
+++ configure.ac        2008-11-21 14:44:19.000000000 +0100
@@ -64,19 +64,11 @@
 CFLAGS="${CFLAGS} -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs"
 
 # Build HTML documentatin?
-AC_PATH_PROG(DOXYGEN, doxygen, no)
+DOXYGEN="no"
 AM_CONDITIONAL(HAVE_DOXYGEN, test "x$DOXYGEN" != "xno")
 
 # Build PDF documentation?
-AC_PATH_PROG(LATEX, pdflatex, no)
-AC_PATH_PROG(KPSEWHICH, kpsewhich, no)
-AC_PATH_PROG(DVIPS, dvips, no)
-AC_PATH_PROG(CONVERT, convert, no)
-AC_PATH_PROG(EPSTOPDF, epstopdf, no)
-if test "${DVIPS}" = "no" -o "${KPSEWHICH}" = "no" -o "${EPSTOPDF}" = "no" \
-     -o "${CONVERT}" = "no"; then
   LATEX="no"
-fi
 if test "x${LATEX}" != "xno"; then
   AC_MSG_CHECKING(for a4.sty and a4wide.sty)
   if "${KPSEWHICH}" a4.sty >/dev/null 2>&1; then
--- Makefile.am.old	2008-11-24 22:35:16.000000000 +0100
+++ Makefile.am	2008-11-24 22:36:32.000000000 +0100
@@ -24,16 +24,6 @@
 	m4 \
 	$(NULL)
 
-# Print out an informative summary.
-all-local:
-	@$(ECHO) "Done."
-	@$(ECHO)
-	@if test "x$(MAKECMDGOALS)" = "xall-am" -o "x$(.TARGETS)" = "xall-am" -o "x$(MAKECMDGOALS)" = "x" -o "x$(.TARGETS)" = "x" ; then \
-	  $(ECHO) "---" ;\
-	  $(ECHO) "Run 'make install' to begin installation into $(prefix)" ;\
-	fi
-	@$(ECHO)
-
 # Upload documentation
 DOC = docs/html docs/latex/ftgl.pdf
 HOST = ftgl.sf.net
