--- a/plugins/Makefile.plugin
+++ b/plugins/Makefile.plugin
@@ -2,7 +2,7 @@
 
 # Every plugin provides a single library to GGobi, define it here.
 # Its name should start with 'lib' and end with '.la'.
-plugindir = $(datadir)/ggobi/plugins/$(PACKAGE_NAME)
+plugindir = $(libdir)/ggobi/plugins/$(PACKAGE_NAME)
 plugin_LTLIBRARIES        = plugin.la
 
 # Most plugins will only require GGobi and its dependencies (gtk, libxml2)
