--- alltray/src/Application.c
+++ alltray/src/Application.c
@@ -16,6 +16,7 @@
 #include <gdk-pixbuf/gdk-pixdata.h>
 #include <config.h>
 
+#include "gtop_glue.h"
 
 #define ALL_TRAY_TYPE_APPLICATION (all_tray_application_get_type ())
 #define ALL_TRAY_APPLICATION(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), ALL_TRAY_TYPE_APPLICATION, AllTrayApplication))
--- alltray/src/AttachHelper.c
+++ alltray/src/AttachHelper.c
@@ -16,6 +16,7 @@
 #include <string.h>
 #include <gtk/gtk.h>
 
+#include "x11_glue.h"
 
 #define ALL_TRAY_TYPE_ATTACH_HELPER (all_tray_attach_helper_get_type ())
 #define ALL_TRAY_ATTACH_HELPER(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), ALL_TRAY_TYPE_ATTACH_HELPER, AllTrayAttachHelper))
--- alltray/src/Process.c
+++ alltray/src/Process.c
@@ -16,6 +16,7 @@
 #include <libwnck/libwnck.h>
 #include <stdio.h>
 
+#include "gtop_glue.h"
 
 #define ALL_TRAY_TYPE_PROCESS (all_tray_process_get_type ())
 #define ALL_TRAY_PROCESS(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), ALL_TRAY_TYPE_PROCESS, AllTrayProcess))
