--- a/src/bwmon.c
+++ b/src/bwmon.c
@@ -36,7 +36,7 @@
 			"by Kimmo Nupponen (kimmoon@users.sourceforge.net)\n"
 			"$Date: 2014/07/10 19:51:24 $\n\n"
 			"usage: %s [-b] [-h] [-a] [-m] [-u seconds]\n"
-			"\t-a Print bandwidth utiliasation in Kbytes rather than Kbits. The default\n"
+			"\t-b Print bandwidth utilisation in KBytes rather than Kbits. The default\n"
 			"\t   is to use Kbits\n"
 			"\t-a Print also average bandwidth since last boot per interface\n"
 			"\t-m Print maximum bandwidth since launch of this utility\n"
