--- a/configure.in	2008-06-23 06:53:17.000000000 +0200
+++ b/configure.in	2008-06-23 20:08:59.000000000 +0200
@@ -174,10 +174,10 @@
 	  [tidyCreate],
 	  [
 	   AC_CHECK_HEADERS(
-	     [tidy.h tidy/tidy.h],
+	     [tidy.h],
 	     [
 	      want_tidy=yes
-	      TIDY_FLAGS="$TIDY_FLAGS -DHAVE_TIDY_H_IN_SUBDIR -DENABLE_TIDY"
+	      TIDY_FLAGS="$TIDY_FLAGS -DENABLE_TIDY"
         TIDY_LIBS="-ltidy"
         #LDFLAGS="$LDFLAGS $want_tidy_path"
 	     ],
