--- makepp-1.40/config.pl
+++ makepp-1.40/config.pl
@@ -34,7 +34,7 @@
 	cd makepp-$(VERSION) && make test    # Make sure it all runs.
 	rm -rf makepp-$(VERSION)
 
-install: all
-	$(PERL) install.pl $(BINDIR) $(DATADIR) $(MANDIR) $(HTMLDIR) $(FINDBIN)
+install:
+	$(PERL) install.pl $(DESTDIR)$(BINDIR) $(DESTDIR)$(DATADIR) $(DESTDIR)$(MANDIR) $(DESTDIR)$(HTMLDIR) $(DESTDIR)$(FINDBIN)
 
 .PHONY: all distribution install test
