Index: src/where.c
===================================================================
--- src/where.c	(revision 622)
+++ src/where.c	(working copy)
@@ -190,7 +190,7 @@
 int dcc_lock_local_cpp(int *cpu_lock_fd)
 {
     int ret;
-    struct dcc_hostdef *chosen;
+    struct dcc_hostdef *chosen = NULL;
     ret = dcc_lock_one(dcc_hostdef_local_cpp, &chosen, cpu_lock_fd);
     dcc_note_state(DCC_PHASE_CPP, NULL, chosen->hostname);
     return ret;
