--- giflib-5.1.8/Makefile
+++ giflib-5.1.8/Makefile
@@ -61,12 +61,11 @@
 LDLIBS=libgif.a -lm
 
 all: libgif.so libgif.a $(UTILS)
-	$(MAKE) -C doc
 
 $(UTILS):: libgif.a
 
 libgif.so: $(OBJECTS) $(HEADERS)
-	$(CC) $(CFLAGS) -shared $(OFLAGS) -Wl,-soname -Wl,libgif.so.$(LIBMAJOR) -o libgif.so $(OBJECTS)
+	$(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,libgif.so.$(LIBMAJOR) -o libgif.so $(OBJECTS)
 
 libgif.a: $(OBJECTS) $(HEADERS)
 	$(AR) rcs libgif.a $(OBJECTS)
