--- a/Makefile
+++ b/Makefile
@@ -5,17 +5,11 @@ VERMIN		= 1
 VERPAT		= 7
 VERSION		= $(VERMAJ).$(VERMIN).$(VERPAT)
 
-# Define SHARED as 1 for Linux shared ELF library
-#SHARED		= 1
 
-ifeq ($(SHARED),1)
 LIBTARGET	= lib$(DIST).so.$(VERSION)
 LIBTARGETSO	= lib$(DIST).so
 LIBTARGETSOMAJ	= $(LIBTARGETSO).$(VERMAJ)
 CCSHRD		= -fPIC
-else
-LIBTARGET	= lib$(DIST).a
-endif
 
 LIBHEAD		= $(DIST).h
 TARGETS		= $(LIBTARGET)
@@ -44,12 +38,7 @@ all: $(TARGETS)
 # that your ar does an implicit ranlib and you do not need to run ranlib
 # separately.  This error is harmless.
 $(LIBTARGET): $(LIBOBJS)
-ifeq ($(SHARED),1)
-	$(CC) -shared -Wl,-soname,$(LIBTARGETSOMAJ) -o $(LIBTARGET) $(LIBOBJS)
-else
-	ar rc $(LIBTARGET) $(LIBOBJS)
-	ranlib $(LIBTARGET) || true
-endif
+	$(CC) ${CFLAGS} ${CPPFLAGS} -shared -Wl,-soname,$(LIBTARGETSOMAJ) -o $(LIBTARGET) $(LIBOBJS)
 
 # Note that you may need GNU's -liberty if your libc lacks strtoul
 example: $(LIBTARGET) example.o
@@ -57,7 +46,7 @@ example: $(LIBTARGET) example.o
 		-L. -I. -L$(INSTLIBDIR) -I$(INSTINCDIR) -lshhopt
 
 .c.o:
-	$(CC) $(CCSHRD) -o $@ -c $(CCOPT) $<
+	$(CC) ${CFLAGS} ${CPPFLAGS} $(CCSHRD) -o $@ -c $(CCOPT) $<
 
 depend dep:
 	$(CC) $(INCDIR) -MM *.c >depend
@@ -66,11 +55,9 @@ install: $(LIBTARGET)
 	$(MKDIRP) $(INSTLIBDIR) $(INSTINCDIR)
 	$(INSTALL) $(LIBTARGET) $(INSTLIBDIR)
 	$(INSTALL) $(LIBHEAD) $(INSTINCDIR)
-ifeq ($(SHARED),1)
 	ln -sf $(LIBTARGET) $(INSTLIBDIR)/$(LIBTARGETSOMAJ)
 	ln -sf $(LIBTARGETSOMAJ) $(INSTLIBDIR)/$(LIBTARGETSO)
 	echo "Now run ldconfig if necessary."
-endif
 
 clean:
 	rm -f *.o core *~ depend
