diff -Nur old/testing/framework/TestSCons.py new/testing/framework/TestSCons.py
--- old/testing/framework/TestSCons.py	2019-03-27 02:15:48.000000000 +0300
+++ new/testing/framework/TestSCons.py	2019-06-07 16:13:48.000000000 +0300
@@ -742,7 +742,8 @@
             version=''
             jni_dirs = ['/System/Library/Frameworks/JavaVM.framework/Headers/jni.h',
-                        '/usr/lib/jvm/default-java/include/jni.h',
-                        '/usr/lib/jvm/java-*-oracle/include/jni.h']
+                        '/usr/lib/jvm/*/include/jni.h',
+                        '/opt/*jdk-bin-*/include/jni.h',
+                        '/usr/lib*/openjdk-*/include/jni.h']
         else:
             jni_dirs = ['/System/Library/Frameworks/JavaVM.framework/Versions/%s*/Headers/jni.h'%version]
         jni_dirs.extend(['/usr/lib/jvm/java-*-sun-%s*/include/jni.h'%version,
