diff --git a/rescue/rescue/Makefile b/rescue/rescue/Makefile
index e747c02..a061436 100644
--- a/rescue/rescue/Makefile
+++ b/rescue/rescue/Makefile
@@ -54,7 +54,7 @@ CRUNCH_PROGS_bin= cat chflags chio chmod cp date dd df echo 	\
 	 ed expr getfacl hostname kenv kill ln ls mkdir mv	\
 	 pkill ps pwd realpath rm rmdir setfacl sh sleep stty	\
 	 sync test
-CRUNCH_LIBS+= -lcrypt -ledit -ljail -lkvm -lelf -ll -ltermcapw -lutil -lxo
+CRUNCH_LIBS+= -lcrypt -ledit -ljail -lkvm -lelf -lfl -lncursesw -lutil -lxo
 CRUNCH_BUILDTOOLS+= bin/sh
 
 # Additional options for specific programs
@@ -128,7 +128,7 @@ CRUNCH_LIBS+= -lavl -lzpool -lzfs_core -lzfs -lnvpair -lpthread -luutil -lumem
 # liblzma needs pthread
 CRUNCH_LIBS+= -lpthread
 .endif
-CRUNCH_LIBS+= -lgeom -lbsdxml -lkiconv
+CRUNCH_LIBS+= -lgeom -lexpat -lkiconv
 .if ${MK_OPENSSL} == "no"
 CRUNCH_LIBS+= -lmd
 .endif
