Fix implicit declarations

Fixed upstream in a larger refactor.

--- a/fontforgeexe/tilepath.c
+++ b/fontforgeexe/tilepath.c
@@ -24,7 +24,9 @@
  * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+#include "cvundoes.h"
 #include "fontforgeui.h"
+#include "fvfonts.h"
 #include "splineutil.h"
 #include "splineutil2.h"
 #include <math.h>
