From 3b51af7de7bfe7b69bbb1a96dbbc184311281704 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Mon, 11 May 2015 23:59:03 +0200
Subject: [PATCH 1/3] Optional gotta be optional

---
 configure.ac | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9713256..bc239de 100644
--- a/configure.ac
+++ b/configure.ac
@@ -508,22 +508,6 @@ else
         AC_MSG_RESULT(no)
 fi

-case $host_os in
-  linux*)
-    if test "$host_cpu" = s390 -o "$host_cpu" = s390x; then
-      # Nothing
-      echo "USB-related panels disabled on S390"
-    else
-      if test "x$have_networkmanager" != "xyes" -o \
-              "x$have_bluetooth" != "xyes" -o \
-              "x$enable_cups" != "xyes" -o \
-              "x$have_wacom" != "xyes"; then
-        AC_MSG_ERROR([The Network, Bluetooth, Printers and Wacom panels are not optional on Linux systems])
-     fi
-    fi
-    ;;
-esac
-
 AC_CONFIG_FILES([
 Makefile
 panels/Makefile
--
2.3.6

