diff -Naur varnish-3.0.5.orig/lib/libvmod_std/vmod.py varnish-3.0.5/lib/libvmod_std/vmod.py
--- varnish-3.0.5.orig/lib/libvmod_std/vmod.py	2014-01-16 13:33:02.983247069 -0500
+++ varnish-3.0.5/lib/libvmod_std/vmod.py	2014-01-16 13:42:36.848254794 -0500
@@ -43,7 +43,7 @@
 if len(sys.argv) == 2:
 	specfile = sys.argv[1]
 else:
-	specfile = "vmod.vcc"
+	specfile = "/etc/varnish/vmod.vcc"
 
 ctypes = {
 	'IP':		"struct sockaddr_storage *",
