diff -Naur varnish-4.0.1.orig/doc/sphinx/Makefile.am varnish-4.0.1/doc/sphinx/Makefile.am
--- varnish-4.0.1.orig/doc/sphinx/Makefile.am	2014-06-23 09:43:06.000000000 -0400
+++ varnish-4.0.1/doc/sphinx/Makefile.am	2014-06-25 08:35:11.887750279 -0400
@@ -144,11 +144,13 @@
 BUILT_SOURCES += include/varnishhist_options.rst \
 	 include/varnishhist_synopsis.rst
 
-reference/vmod_std.generated.rst: $(top_srcdir)/lib/libvmod_std/vmod_std.rst
+reference/vmod_std.generated.rst: $(top_builddir)/lib/libvmod_std/vmod_std.rst
+	mkdir -p reference
 	cp $? $@
 BUILT_SOURCES += reference/vmod_std.generated.rst
 
-reference/vmod_directors.generated.rst: $(top_srcdir)/lib/libvmod_directors/vmod_directors.rst
+reference/vmod_directors.generated.rst: $(top_builddir)/lib/libvmod_directors/vmod_directors.rst
+	mkdir -p reference
 	cp $? $@
 BUILT_SOURCES += reference/vmod_directors.generated.rst
 
