diff -Naur /root/varnish-4.0.1/lib/libvcc/vcc_expr.c varnish-4.0.1/lib/libvcc/vcc_expr.c
--- /root/varnish-4.0.1/lib/libvcc/vcc_expr.c	2014-06-25 07:06:42.550508615 -0400
+++ varnish-4.0.1/lib/libvcc/vcc_expr.c	2014-06-23 09:43:06.000000000 -0400
@@ -670,7 +670,7 @@
 	struct expr *e1, *e2;
 	const char *ip;
 	const struct symbol *sym;
-	double d;
+	double d = 0.0;
 
 	*e = NULL;
 	if (tl->t->tok == '(') {
