diff -ruN Bastille.orig/AutomatedBastille Bastille/AutomatedBastille
--- Bastille.orig/AutomatedBastille	2017-01-02 09:39:20.720443755 +0100
+++ Bastille/AutomatedBastille	2017-01-02 09:40:54.293441835 +0100
@@ -77,7 +77,7 @@
 
 # Use the Curses interface
 use Curses;
-use Curses::Widgets;
+use Curses::Widgets_Bastille;
 
 # Hardcoded List of configurations -- to be replaced by list read from file
 @list = ("WorkstationLax","WorkstationModerate","WorkstationParanoia","ServerLax","ServerModerate","ServerParanoia","Quit");
diff -ruN Bastille.orig/Bastille_Curses.pm Bastille/Bastille_Curses.pm
--- Bastille.orig/Bastille_Curses.pm	2005-04-18 14:32:10.000000000 +0200
+++ Bastille/Bastille_Curses.pm	2017-01-02 09:41:24.470441216 +0100
@@ -34,7 +34,7 @@
 sub do_Bastille {
 
     use Curses;
-    use Curses::Widgets;
+    use Curses::Widgets_Bastille;
 
     # Number_Modules is the number of modules loaded in by Load_Questions
     $Number_Modules=0;
diff -ruN Bastille.orig/Curses/Widgets.pm Bastille/Curses/Widgets.pm
--- Bastille.orig/Curses/Widgets.pm	2005-04-06 01:18:11.000000000 +0200
+++ Bastille/Curses/Widgets.pm	2017-01-02 09:42:31.145439848 +0100
@@ -11,7 +11,7 @@
 #
 ########################################################################
 
-package Curses::Widgets;
+package Curses::Widgets_Bastille;
 
 use strict;
 use vars qw(@ISA @EXPORT @EXPORT_OK %EXPORT_TAGS $VERSION);
diff -ruN Bastille.orig/Install.sh Bastille/Install.sh
--- Bastille.orig/Install.sh	2017-01-02 09:39:20.767443754 +0100
+++ Bastille/Install.sh	2017-01-02 09:41:05.537441604 +0100
@@ -15,7 +15,7 @@
 cp BastilleBackEnd ${DESTDIR}/usr/sbin
 cp Bastille_Curses.pm ${DESTDIR}/${VENDOR_LIB}
 cp Bastille_Tk.pm ${DESTDIR}/${VENDOR_LIB}
-cp Curses/Widgets.pm ${DESTDIR}/${VENDOR_LIB}/Curses
+cp Curses/Widgets.pm ${DESTDIR}/${VENDOR_LIB}/Curses/Widgets_Bastille.pm
 cp InteractiveBastille ${DESTDIR}/usr/sbin
 # Questions.txt has been replaced by Modules.txt and Questions/
 #cp Questions.txt ${DESTDIR}/usr/share/Bastille
