diff -ur vnc_unixsrc.orig/Xvnc/config/cf/Imake.cf vnc_unixsrc/Xvnc/config/cf/Imake.cf
--- vnc_unixsrc.orig/Xvnc/config/cf/Imake.cf	2008-09-24 18:50:40.000000000 +0200
+++ vnc_unixsrc/Xvnc/config/cf/Imake.cf	2008-09-24 18:51:37.000000000 +0200
@@ -711,7 +711,7 @@
 #  undef __alpha__
 # endif /* __alpha__ */
 # ifdef __arm__
-#  define Arm32Architecture
+#  define ArmArchitecture
 #  undef arm
 #  undef __arm
 #  undef __arm__
