diff -cr vnc_unixsrc.orig/Xvnc/config/cf/linux.cf vnc_unixsrc/Xvnc/config/cf/linux.cf
*** vnc_unixsrc.orig/Xvnc/config/cf/linux.cf	Wed Mar 20 02:49:23 2002
--- vnc_unixsrc/Xvnc/config/cf/linux.cf	Sat Nov  4 19:01:06 2006
***************
*** 43,49 ****
  #if defined(AlphaArchitecture) || defined(PowerPCArchitecture) || \
      defined(SparcArchitecture) || defined(ArmArchitecure) || \
      defined(i386Architecture) || defined(s390Architecture) || \
!     defined(s390xArchitecture)
  #define DefaultLinuxCLibMajorVersion	6
  #else
  #define DefaultLinuxCLibMajorVersion	5
--- 43,49 ----
  #if defined(AlphaArchitecture) || defined(PowerPCArchitecture) || \
      defined(SparcArchitecture) || defined(ArmArchitecure) || \
      defined(i386Architecture) || defined(s390Architecture) || \
!     defined(s390xArchitecture) || defined(MipsArchitecture)
  #define DefaultLinuxCLibMajorVersion	6
  #else
  #define DefaultLinuxCLibMajorVersion	5
***************
*** 261,266 ****
--- 261,273 ----
  #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
  #endif /* AlphaArchitecture */
  
+ #ifdef MipsArchitecture
+ #define OptimizedCDebugFlags	-O2
+ #define LinuxMachineDefines	-D__mips__
+ #define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
+ #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+ #endif /* MipsArchitecture */
+ 
  #ifdef Mc68020Architecture
  #define OptimizedCDebugFlags	-O2
  #define LinuxMachineDefines 	-D__mc68000__
diff -cr vnc_unixsrc.orig/Xvnc/programs/Xserver/cfb/Imakefile vnc_unixsrc/Xvnc/programs/Xserver/cfb/Imakefile
*** vnc_unixsrc.orig/Xvnc/programs/Xserver/cfb/Imakefile	Tue Mar 19 21:16:03 2002
--- vnc_unixsrc/Xvnc/programs/Xserver/cfb/Imakefile	Sat Nov  4 19:01:49 2006
***************
*** 10,24 ****
  /* cfb8cppl.c MUST come last to avoid confusing makedepend */
  PSZSRCS = cfb8bit.c cfbteblt8.c cfbglrop8.c cfbpush8.c cfbrctstp8.c cfb8cppl.c
  PSZOBJS = cfb8bit.o cfbteblt8.o cfbglrop8.o cfbpush8.o cfbrctstp8.o cfb8cppl.o
- #ifdef MipsArchitecture
- #ifndef ArcArchitecture
- #ifndef Mips64Architecture
- STIPPLESRC = stipmips.s stipmipste.s
- STIPPLEOBJ = stipmips.o stipmipste.o
- STIPPLEDEF = -DHAS_STIPPLE_CODE
- #endif
- #endif
- #endif
  #ifdef SparcArchitecture
  STIPPLESRC = stipsparc.s stipsparcte.s
  STIPPLEOBJ = stipsparc.o stipsparcte.o
--- 10,15 ----
***************
*** 128,136 ****
  LinkSourceFile(cfbpush8.c,LinkDirectory)
  LinkSourceFile(cfbrctstp8.c,LinkDirectory)
  LinkSourceFile(cfbteblt8.c,LinkDirectory)
- #ifdef MipsArchitecture
- LinkSourceFile(stipmips.s,LinkDirectory)
- #endif
  #ifdef SparcArchitecture
  LinkSourceFile(stipsparc.s,LinkDirectory)
  #endif
--- 119,124 ----
***************
*** 186,202 ****
  #if PixelSize == 8
  ObjectFromSpecialSource(cfbglrop8,cfbglblt8,-DGLYPHROP)
  
- #ifdef MipsArchitecture
- stipmipste.s: stipmips.s
- 	$(RM) $@
- 	$(LN) stipmips.s stipmipste.s
- clean::
- 	$(RM) stipmipste.s
- 
- stipmipste.o: stipmipste.s
- 	$(AS) -o $@ -DTETEXT stipmipste.s
- #endif
- 
  #ifdef SparcArchitecture
  stipsparcte.s: stipsparc.s
  	$(RM) $@
--- 174,179 ----
