--- vncserver.old	2006-08-10 13:01:07.000000000 +0200
+++ vncserver	2006-09-05 19:31:30.000000000 +0200
@@ -36,16 +36,16 @@
 $geometry = "1024x768";
 $depth = 24;
 $desktopName = "X";
-$vncClasses = "/usr/local/vnc/classes";
+$vncClasses = "/usr/share/tightvnc/classes";
 $vncUserDir = "$ENV{HOME}/.vnc";
-$fontPath = "unix/:7100";
+$fontPath = "unix/:-1";
 $authType = "-rfbauth $vncUserDir/passwd";
 
 # Here is another example of setting the font path:
-# $fontPath = "/usr/lib/X11/fonts/misc/,/usr/lib/X11/fonts/75dpi/";
+$fontPath = "/usr/share/fonts/misc/,/usr/share/fonts/Type1/";
 
 # X colors database path is optional, uncomment and edit to use:
-# $colorPath = "/usr/lib/X11/rgb";
+$colorPath = "/usr/share/X11/rgb";
 
 # You might wish to make your vnc directory under /tmp, to make sure
 # passwords are always kept on the local filesystem. To do that, just
