From fb735bb60e3bbfc14a9b38fe57c0cdfef48e15ab Mon Sep 17 00:00:00 2001
From: Florian Schmaus <flo@geekplace.eu>
Date: Mon, 27 Mar 2023 11:26:09 +0200
Subject: [PATCH] Remove bin/echars

--- a/kramdown-rfc2629.gemspec
+++ b/kramdown-rfc2629.gemspec
@@ -7,14 +7,10 @@ spec = Gem::Specification.new do |s|
   s.add_dependency('kramdown', '~> 2.4.0')
   s.add_dependency('kramdown-parser-gfm', '~> 1.1')
   s.add_dependency('json_pure', '~> 2.0')
-  s.add_dependency('unicode-name', '~> 1.0')
-  s.add_dependency('unicode-blocks', '~> 1.0')
-  s.add_dependency('unicode-scripts', '~> 1.0')
   s.add_dependency('net-http-persistent', '~> 4.0')
-  s.add_dependency('differ', '~>0.1')
-  s.files = Dir['lib/**/*.rb'] + %w(README.md LICENSE kramdown-rfc2629.gemspec bin/kdrfc bin/kramdown-rfc bin/kramdown-rfc2629 bin/doilit bin/echars bin/kramdown-rfc-extract-markdown data/kramdown-rfc2629.erb data/encoding-fallbacks.txt data/math.json bin/kramdown-rfc-cache-subseries-bibxml bin/kramdown-rfc-autolink-iref-cleanup bin/de-gfm)
+  s.files = Dir['lib/**/*.rb'] + %w(README.md LICENSE kramdown-rfc2629.gemspec bin/kdrfc bin/kramdown-rfc bin/kramdown-rfc2629 bin/doilit bin/kramdown-rfc-extract-markdown data/kramdown-rfc2629.erb data/encoding-fallbacks.txt data/math.json bin/kramdown-rfc-cache-subseries-bibxml bin/kramdown-rfc-autolink-iref-cleanup bin/de-gfm)
   s.require_path = 'lib'
-  s.executables = ['kramdown-rfc', 'kramdown-rfc2629', 'doilit', 'echars',
+  s.executables = ['kramdown-rfc', 'kramdown-rfc2629', 'doilit',
                    'kramdown-rfc-extract-markdown',
                    'kdrfc', 'kramdown-rfc-cache-i-d-bibxml',
                    'kramdown-rfc-cache-subseries-bibxml',
-- 
2.39.2

