diff --git a/test/TestUtil.hs b/test/TestUtil.hs
index 87f3417..cda262f 100644
--- a/test/TestUtil.hs
+++ b/test/TestUtil.hs
@@ -72,3 +72,5 @@ instance Testable () where
     property = property . (`seq` True)
+#if ! MIN_VERSION_QuickCheck(2,9,0)
     exhaustive _ = True
+#endif
 
@@ -76,3 +78,6 @@ instance Testable a => Testable (IO a) where
     property = property . unsafePerformIO
+#if ! MIN_VERSION_QuickCheck(2,9,0)
     exhaustive = exhaustive . unsafePerformIO
+#endif
+
 
@@ -107,3 +112,5 @@ instance Arbitrary DiffTime where
 
+#if ! MIN_VERSION_QuickCheck(2,9,2)
 instance Arbitrary Version where
     arbitrary = makeVersion . map abs <$> listOf1 arbitrary
+#endif
