From: Julian Ospald <hasufell@gentoo.org>
Date: Sat Oct 27 22:39:31 UTC 2012
Subject: FHS

paths hardcoded until upstream fixed the logic

--- mygui-MyGUI3.2.2.orig/CMake/InstallResources.cmake
+++ mygui-MyGUI3.2.2/CMake/InstallResources.cmake
@@ -24,7 +24,7 @@
 	else ()
 		install(FILES
 			${MYGUI_BINARY_DIR}/bin/${FILENAME}
-			DESTINATION "bin"
+			DESTINATION "/etc/MYGUI"
 		)
 	endif ()
 endfunction(install_file)
@@ -53,7 +53,7 @@
 	if (WIN32)
 		set(MYGUI_MEDIA_DIR "../../Media")
 	elseif (UNIX)
-		set(MYGUI_MEDIA_DIR "../share/MYGUI/Media")
+		set(MYGUI_MEDIA_DIR "${CMAKE_INSTALL_PREFIX}/share/MYGUI/Media")
 	else ()
 		set(MYGUI_MEDIA_DIR "../../Media")
 	endif ()
 
