--- jigdo-0.7.3/Makefile.in.orig	2006-05-19 16:55:41.000000000 +0200
+++ jigdo-0.7.3/Makefile.in	2009-09-04 17:37:31.000000000 +0200
@@ -15,7 +15,7 @@
 
 PACKAGE =	jigdo
 INSTALL =	@INSTALL@
-INSTALL_EXE =	@INSTALL@ -s
+INSTALL_EXE =	@INSTALL@
 INSTALL_DATA =	@INSTALL@ -m 644
 INSTALL_SCRIPT = @INSTALL@ -m 755
 AWK =		@AWK@
--- jigdo-0.7.3/src/Makefile.in.orig	2005-07-06 17:51:28.000000000 +0200
+++ jigdo-0.7.3/src/Makefile.in	2009-09-04 17:38:58.000000000 +0200
@@ -161,8 +161,7 @@
 		    cd .. && sh config.status; \
 		fi
 strip:
-		for p in $(programs); do if test -f "$$p"; then \
-		    strip -R .comment -R .note "$$p"; fi; done
+		true
 
 TEST-DEFAULTOBJS = util/log.o util/string-utf.o util/debug.o
 TEST-LDFLAGS =	@LDFLAGS@ $(GLIBLIBS)
