diff -aur localshell-1.2.old/src/common.cxx localshell-1.2/src/common.cxx
--- localshell-1.2.old/src/common.cxx	2008-12-12 20:03:48.000000000 +0100
+++ localshell-1.2/src/common.cxx	2008-12-12 20:15:31.000000000 +0100
@@ -5,6 +5,7 @@
 #include "structures.hh"
 #include "common.hh"
 #include <fstream>
+#include <cstring>
 using namespace std;
 
 bool match_user(uid_t uid, gid_t gid) {
diff -aur localshell-1.2.old/src/config.cxx localshell-1.2/src/config.cxx
--- localshell-1.2.old/src/config.cxx	2008-12-12 20:03:48.000000000 +0100
+++ localshell-1.2/src/config.cxx	2008-12-12 20:15:07.000000000 +0100
@@ -10,6 +10,8 @@
 #include <fstream>
 #include <map>
 #include <vector>
+#include <cstring>
+#include <cstdlib>
 using namespace std;
 
 int load_config( const char *cfg_filename, configuration &conf) {
diff -aur localshell-1.2.old/src/config.hh localshell-1.2/src/config.hh
--- localshell-1.2.old/src/config.hh	2008-12-12 20:03:48.000000000 +0100
+++ localshell-1.2/src/config.hh	2008-12-12 20:14:00.000000000 +0100
@@ -2,6 +2,7 @@
 #ifndef _CONFIG_HH_
 #define _CONFIG_HH_
 #include "structures.hh"
+#include <fstream>
 
 int load_config( const char *cfg_filename, configuration &conf);
 int parse_config(fstream &fs, configuration &conf);
diff -aur localshell-1.2.old/src/localshell.cxx localshell-1.2/src/localshell.cxx
--- localshell-1.2.old/src/localshell.cxx	2008-12-12 20:03:48.000000000 +0100
+++ localshell-1.2/src/localshell.cxx	2008-12-12 20:04:14.000000000 +0100
@@ -10,6 +10,7 @@
 
 #include <map>
 #include <vector>
+#include <cstring>
 using namespace std;
 
 int main(int argc, char** argv) {
