--- a/src/framework/draw/CMakeLists.txt
+++ b/src/framework/draw/CMakeLists.txt
@@ -123,20 +123,14 @@ else()
     include(${local_path}/harfbuzz.cmake)
 
     # func from ${name}.cmake)
-    cmake_language(CALL harfbuzz_Populate ${remote_url} ${local_path} "source" "" "")
 
-    add_subdirectory(_deps/harfbuzz/harfbuzz)
-    target_no_warning(harfbuzz -Wno-conversion)
-    target_no_warning(harfbuzz -Wno-unused-parameter)
-    target_no_warning(harfbuzz -Wno-unused-variable)
-    target_no_warning(harfbuzz -WMSVC-no-hides-previous)
-    target_no_warning(harfbuzz -WMSVC-no-unreachable)
+    pkg_check_modules(harfbuzz REQUIRED IMPORTED_TARGET harfbuzz)
 
     #add_subdirectory(thirdparty/msdfgen)
 
     set(MODULE_INCLUDE
         ${FREETYPE_INCLUDE_DIRS}
-        ${CMAKE_CURRENT_LIST_DIR}/_deps/harfbuzz/harfbuzz/harfbuzz/src
+	/usr/include/harfbuzz
         #${CMAKE_CURRENT_LIST_DIR}/thirdparty/msdfgen/msdfgen-1.4
     )
 
