From 4c28258db05dd6d33bbcd037a05a7fbbb66f67ab Mon Sep 17 00:00:00 2001
From: Joachim Schmitz <jojo@schmitz-digital.de>
Date: Wed, 8 Aug 2018 09:16:54 +0200
Subject: [PATCH] Fix #52106: fails to compile with
 -DUSE_SYSTEM_QTSINGLEAPPLICATION=ON

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3a30c03d63..be6a50578a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -537,8 +537,13 @@ endif(APPLE)
 ##
 
 if (USE_SYSTEM_QTSINGLEAPPLICATION)
+    if (USE_PATH_WITH_EXPLICIT_QT_VERSION)
+      find_path(QTSINGLEAPPLICATION_INCLUDE_DIRS qtsingleapplication.h PATH_SUFFIXES qt5/QtSolutions)
+      find_library(QTSINGLEAPPLICATION_LIBRARIES Qt5Solutions_SingleApplication-2.6)
+    else (USE_PATH_WITH_EXPLICIT_QT_VERSION)
       find_path(QTSINGLEAPPLICATION_INCLUDE_DIRS qtsingleapplication.h PATH_SUFFIXES QtSolutions)
       find_library(QTSINGLEAPPLICATION_LIBRARIES QtSolutions_SingleApplication-2.6)
+    endif (USE_PATH_WITH_EXPLICIT_QT_VERSION)
 else(USE_SYSTEM_QTSINGLEAPPLICATION)
       subdirs (thirdparty/singleapp)
       set(QTSINGLEAPPLICATION_INCLUDE_DIRS ${PROJECT_SOURCE_DIR}/singleapp/src)
