diff --git a/mscore/scoreview.cpp b/mscore/scoreview.cpp
index f3fe9ce8b..3957f90c6 100644
--- a/mscore/scoreview.cpp
+++ b/mscore/scoreview.cpp
@@ -2649,7 +2649,7 @@ void ScoreView::cmd(const char* s)
                 cv->updateGrips();
             } },
       #ifdef OMR
-        { { "show-omr" }, [](ScoreView* cv, const QByteArray&) {
+        { { "show-omr" }, [&](ScoreView* cv, const QByteArray&) {
                 if (cv->score()->masterScore()->omr()) {
                     showOmr(!_score->masterScore()->showOmr());
                 }
