--- mscore-1.1/mscore/mscore/exportxml.cpp.orig
+++ mscore-1.1/mscore/mscore/exportxml.cpp
@@ -2363,7 +2363,7 @@
                    el->userOff().y());
 */
             if (el->type() == HAIRPIN || el->type() == OTTAVA || el->type() == TEXTLINE) {
-                  SLine* sl = static_cast<const SLine*>(el);
+                  SLine* sl = static_cast<SLine*>(el);
 //                  printf("slin segsz=%d", sl->lineSegments().size());
                   if (sl->lineSegments().size() > 0) {
                         LineSegment* seg = sl->lineSegments().at(0);
