unbundle pugixml
--- a/src/framework/global/CMakeLists.txt
+++ b/src/framework/global/CMakeLists.txt
@@ -150,9 +150,6 @@ set(MODULE_SRC
     ${CMAKE_CURRENT_LIST_DIR}/serialization/xmlstreamwriter.h
     ${CMAKE_CURRENT_LIST_DIR}/serialization/xmldom.cpp
     ${CMAKE_CURRENT_LIST_DIR}/serialization/xmldom.h
-
-    ${CMAKE_CURRENT_LIST_DIR}/thirdparty/pugixml/pugixml.hpp
-    ${CMAKE_CURRENT_LIST_DIR}/thirdparty/pugixml/pugixml.cpp
 )
 
 if (MUSE_THREADS_SUPPORT)
@@ -274,9 +271,9 @@ else ()
     set(Z_LIB z)
 endif ()
 
-list(APPEND MODULE_INCLUDE_PRIVATE
-    ${CMAKE_CURRENT_LIST_DIR}/thirdparty/pugixml
-)
+pkg_check_modules(pugixml REQUIRED IMPORTED_TARGET pugixml)
+list(APPEND MODULE_LINK PkgConfig::pugixml)
+list(APPEND MODULE_INCLUDE_PRIVATE ${pugixml_INCLUDE_DIRS})
 
 list(APPEND MODULE_INCLUDE_PRIVATE ${Z_INCLUDE})
 list(APPEND MODULE_LINK ${Z_LIB})
