diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6437963..2a173b9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -660,7 +660,7 @@ if (NOT MINGW AND NOT MSVC AND NOT APPLE)
     # Compress man pages if gzip is installed (don't on OpenBSD)
     # Note: Compressing man pages is normal on Linux but not OpenBSD
     find_program( GZIP_EXECUTABLE gzip DOC "A tool for compressing manpages (optional)." )
-    if (GZIP_EXECUTABLE AND NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+    if (GZIP_EXECUTABLE AND NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD" AND false)
         message(STATUS "Found 'gzip'. Man pages will be compressed.")
         set(MAN_TARGET ${MAN_BUILD})
         set(MAN_EXTENSION ${MAN_EXTENSION}.gz)
@@ -677,7 +677,7 @@ if (NOT MINGW AND NOT MSVC AND NOT APPLE)
             COMMAND echo "Man pages have been compressed ready for installation."
             VERBATIM
             )
-    else (GZIP_EXECUTABLE AND NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+    else (GZIP_EXECUTABLE AND NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD" AND false)
         if (CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
             message(STATUS "System is OpenBSD: Man pages will not be compressed.")
         else (CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
@@ -687,7 +687,7 @@ if (NOT MINGW AND NOT MSVC AND NOT APPLE)
             COMMAND echo "Man pages will be installed uncompressed."
             VERBATIM
             )
-    endif (GZIP_EXECUTABLE AND NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
+    endif (GZIP_EXECUTABLE AND NOT CMAKE_SYSTEM_NAME MATCHES "OpenBSD" AND false)
     # Install man pages in either compressed or uncompressed form
     install( FILES ${MAN_BUILD} DESTINATION share/man/man1 COMPONENT doc)
     # Create symlink alias for man pages so `man musescore` = `man mscore`
