diff -ur pax-3.4-orig/lib/fts.c pax-3.4/lib/fts.c
--- pax-3.4-orig/lib/fts.c	2005-07-29 09:55:25.000000000 +0200
+++ pax-3.4/lib/fts.c	2014-10-30 21:38:42.404092744 +0100
@@ -35,7 +35,7 @@
 #include <fcntl.h>
 #include <dirent.h>
 #include <errno.h>
-#include <fts.h>
+#include "fts.h"
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
diff -ur pax-3.4-orig/src/ftree.c pax-3.4/src/ftree.c
--- pax-3.4-orig/src/ftree.c	2005-07-29 09:46:06.000000000 +0200
+++ pax-3.4/src/ftree.c	2014-10-30 21:39:14.876093149 +0100
@@ -46,7 +46,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include <stdlib.h>
-#include <fts.h>
+#include "../lib/fts.h"
 #include "pax.h"
 #include "ftree.h"
 #include "extern.h"
