--- a/Makefile.in
+++ b/Makefile.in
@@ -30,6 +30,7 @@
 
 srcdir = @srcdir@
 bindir = @bindir@
+mandir = @mandir@
 
 # The following definition can be set to non-null for special systems
 # like AFS with replication.  It allows the pathnames used for installation
@@ -48,7 +49,9 @@
 BIN_INSTALL_DIR = $(INSTALL_ROOT)$(exec_prefix)/bin
 
 # Top-level directory in which to install manual entries:
-MAN_INSTALL_DIR = $(INSTALL_ROOT)$(prefix)/man/mann
+MAN_INSTALL_DIR = $(INSTALL_ROOT)$(mandir)/mann
+
+DATA_DIR = $(INSTALL_ROOT)$(prefix)/share/iwidgets$(IWIDGETS_VERSION)
 
 # Directory containing Tcl source code (for library used during test):
 TCL_SRC_DIR = @TCL_SRC_DIR@
@@ -135,26 +138,26 @@
 	    done;
 
 install-demos:
-	@$(MKINSTALLDIRS) $(SCRIPT_INSTALL_DIR)/demos/images $(SCRIPT_INSTALL_DIR)/demos/html
-	@echo "Installing demos from $(srcdir)/demos into $(SCRIPT_INSTALL_DIR)/demos"
+	@$(MKINSTALLDIRS) $(DATA_DIR)/demos/images $(DATA_DIR)/demos/html
+	@echo "Installing demos from $(srcdir)/demos into $(DATA_DIR)/demos"
 	@for i in $(srcdir)/demos/*; \
 	    do \
 	    if [ -f $$i ] ; then \
-	        $(INSTALL_SCRIPT) $$i $(SCRIPT_INSTALL_DIR)/demos; \
+	        $(INSTALL_SCRIPT) $$i $(DATA_DIR)/demos; \
 	    fi; \
 	    done;
-	@echo "Installing images from $(srcdir)/demos/images into $(SCRIPT_INSTALL_DIR)/demos/images"
+	@echo "Installing images from $(srcdir)/demos/images into $(DATA_DIR)/demos/images"
 	for i in $(srcdir)/demos/images/*; \
 	    do \
 	    if [ -f $$i ] ; then \
-	        $(INSTALL_DATA) $$i $(SCRIPT_INSTALL_DIR)/demos/images; \
+	        $(INSTALL_DATA) $$i $(DATA_DIR)/demos/images; \
 	    fi; \
 	    done;
-	@echo "Installing html man pages from $(srcdir)/demos/html into $(SCRIPT_INSTALL_DIR)/demos/html"
+	@echo "Installing html man pages from $(srcdir)/demos/html into $(DATA_DIR)/demos/html"
 	for i in $(srcdir)/demos/html/*; \
 	    do \
 	    if [ -f $$i ] ; then \
-	        $(INSTALL_DATA) $$i $(SCRIPT_INSTALL_DIR)/demos/html; \
+	        $(INSTALL_DATA) $$i $(DATA_DIR)/demos/html; \
 	    fi; \
 	    done;
 
@@ -169,7 +172,6 @@
 	        $$i > $(MAN_INSTALL_DIR)/`basename iwidgets_$$i`; \
 	    chmod 444 $(MAN_INSTALL_DIR)/`basename iwidgets_$$i`; \
 	    done;
-	$(INSTALL_DATA) $(srcdir)/license.terms $(SCRIPT_INSTALL_DIR)
 
 Makefile: $(srcdir)/Makefile.in
 	$(SHELL) config.status
