From 71c8754d9978895542cf13eba8958df078a91a6c Mon Sep 17 00:00:00 2001
From: Thaddaeus Tintenfisch <thad.fisch@gmail.com>
Date: Sun, 29 Mar 2015 18:36:47 +0200
Subject: [PATCH] Fix build error (Bug #11778)

---
 src/xfdesktop-icon-view.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/xfdesktop-icon-view.c b/src/xfdesktop-icon-view.c
index adff45c..30914c4 100644
--- a/src/xfdesktop-icon-view.c
+++ b/src/xfdesktop-icon-view.c
@@ -1640,6 +1640,7 @@ xfdesktop_icon_view_drag_drop(GtkWidget *widget,
             
             action = gdk_drag_context_get_selected_action(context);
 
+#ifdef ENABLE_FILE_ICONS
             if(action == GDK_ACTION_ASK) {
                 xfdesktop_dnd_menu(icon_view->priv->manager, icon_on_dest,
                                    context, &action, row, col, time_);
@@ -1649,6 +1650,7 @@ xfdesktop_icon_view_drag_drop(GtkWidget *widget,
                     return ret;
                 }
             }
+#endif
 
             for(l = icon_view->priv->selected_icons; l; l = l->next) {
                 if(xfdesktop_icon_do_drop_dest(icon_on_dest,
-- 
2.1.4
