--- a/setup.py	2019-03-21 01:51:55.751971770 +0300
+++ b/setup.py	2019-03-21 01:54:40.142513567 +0300
@@ -122,12 +122,6 @@
         for f in [ script, manpage, manpage_f ]:
             self.mkpath(os.path.dirname(f))
 
-        # update the interpreter to that of the one the user specified for setup
-        print("Updating staging/ufw to use %s" % (sys.executable))
-        subprocess.call(["sed",
-                         "-i",
-                         "1s%^#.*python.*%#! /usr/bin/env " + sys.executable + "%g",
-                         'staging/ufw'])
         self.copy_file('staging/ufw', script)
         self.copy_file('doc/ufw.8', manpage)
         self.copy_file('doc/ufw-framework.8', manpage_f)
