Do not use C++/C11(?) style local variable declaration.

--- cctools-855/otool/dyld_bind_info.c.orig	2015-01-12 18:55:04.000000000 +0100
+++ cctools-855/otool/dyld_bind_info.c	2015-01-12 18:56:35.000000000 +0100
@@ -353,7 +353,9 @@
 		    skip = read_uleb128(&p, end);
 		    sectName = sectionName(segIndex, segStartAddr + segOffset,
 					   segs, nsegs, segs64, nsegs64);
-		    for (uint32_t i=0; i < count; ++i) {
+		    {
+		    uint32_t i;
+		    for (i=0; i < count; ++i) {
 			if(pass == 2){
 			    (*dbi)[n].segname = segName;
 			    (*dbi)[n].sectname = sectName;
@@ -367,6 +369,7 @@
 			n++;
 			segOffset += skip + sizeof_pointer;
 		    }
+		    }
 		    break;
 		default:
 		    return; /* throwf("bad bind opcode %d", *p); */
