don't install efi's manpages since we don't install efi

--- cctools-698/man/Makefile
+++ cctools-698/man/Makefile
@@ -27,14 +27,21 @@
 MANL3 = libsyminfo.3 redo_prebinding.3
 
 INSTALL_FILES = Makefile $(COMMON_MAN1) $(OTHER_MAN1) $(MAN3) $(MAN5) $(MANL) \
-		$(MANL3) $(DYLD_MAN1) $(DYLD_MAN3) $(DYLD_MANL3) $(EFI1) notes
+		$(MANL3) $(DYLD_MAN1) $(DYLD_MAN3) $(DYLD_MANL3) notes
+
+ifeq "$(EFITOOLS)" "efitools"
+	INSTALL_FILES += $(EFI1)
+endif
 
 MANDIR = /usr/share/man
 LOCMANDIR = /usr/local/man
 EFIMANDIR = /usr/local/efi/share/man
 DSTDIRS = $(DSTROOT)$(MANDIR)/man1 $(DSTROOT)$(MANDIR)/man3 \
 	  $(DSTROOT)$(MANDIR)/man5 $(DSTROOT)$(LOCMANDIR)/man1 \
-	  $(DSTROOT)$(LOCMANDIR)/man3 $(DSTROOT)$(EFIMANDIR)/man1
+	  $(DSTROOT)$(LOCMANDIR)/man3
+ifeq "$(EFITOOLS)" "efitools"
+	DSTDIR += $(DSTROOT)$(EFIMANDIR)/man1
+endif
 
 install: $(DSTDIRS) $(COMMON_MAN1) $(OTHER_MAN1) $(MAN3) $(MAN5) $(MANL) \
 	 $(RC_OS) $(MANL3)
@@ -51,7 +58,9 @@
 	install -c -m 444 $(MAN5) $(DSTROOT)$(MANDIR)/man5
 	install -c -m 444 $(MANL) $(DSTROOT)$(LOCMANDIR)/man1
 	install -c -m 444 $(MANL3) $(DSTROOT)$(LOCMANDIR)/man3
+ifeq "$(EFITOOLS)" "efitools"
 	install -c -m 444 $(EFI1) $(DSTROOT)$(EFIMANDIR)/man1
+endif
 	if [ $(OLD_DYLD_STUFF) ];					\
 	then								\
 	    install -c -m 444 $(DYLD_MANL3) $(DSTROOT)$(LOCMANDIR)/man3;\
