Use correct enum values (backport from 241.9).
 
--- ld64-236.3/src/other/ObjectDump.cpp.orig	2014-11-12 21:17:01.000000000 +0100
+++ ld64-236.3/src/other/ObjectDump.cpp	2014-11-12 21:17:44.000000000 +0100
@@ -995,10 +995,10 @@
 		case ld::Fixup::kindStoreTargetAddressARM64PageOff12:
 			printf("ARM64 store 12-bit page offset of %s", referenceTargetAtomName(ref));
 			break;
-		case ld::Fixup::kindStoreTargetAddressARM64TLVPage21:
+		case ld::Fixup::kindStoreTargetAddressARM64TLVPLoadPage21:
 			printf("ARM64 store 21-bit pcrel ADRP to TLV for %s", referenceTargetAtomName(ref));
 			break;
-		case ld::Fixup::kindStoreTargetAddressARM64TLVPageOff12:
+		case ld::Fixup::kindStoreTargetAddressARM64TLVPLoadPageOff12:
 			printf("ARM64 store 12-bit page offset of TLV of %s", referenceTargetAtomName(ref));
 			break;
 		case ld::Fixup::kindStoreTargetAddressARM64GOTLoadPage21:
