--- a/import/demux_pass.c
+++ b/import/demux_pass.c
@@ -32,8 +32,6 @@
 #include "packets.h"
 
 
-int gop, gop_pts, gop_cnt;
-
 /* if you listen carefully, then you can hear the desesperate
  * whisper of this code calling for a rewrite. Or for a redesign.
  * Or both. --  FR
--- a/import/extract_ac3.c
+++ b/import/extract_ac3.c
@@ -334,7 +334,6 @@
 FILE *fd;
 
 #define MAX_BUF 4096
-char audio[MAX_BUF];
 
 
 /* from ac3scan.c */
@@ -472,6 +471,8 @@
 void extract_ac3(info_t *ipipe)
 {
 
+    char audio[MAX_BUF];
+
     int error=0;
 
     avi_t *avifile;
--- a/import/extract_mp3.c
+++ b/import/extract_mp3.c
@@ -243,7 +243,6 @@
 }
 
 #define MAX_BUF 4096
-char audio[MAX_BUF];
 
 /* ------------------------------------------------------------
  *
@@ -258,6 +257,7 @@
 
 void extract_mp3(info_t *ipipe)
 {
+    char audio[MAX_BUF];
 
     int error=0;
 
