--- external/Makefile	2012-10-06 14:30:40.453248737 -0400
+++ external/Makefile	2012-10-06 14:32:43.283277751 -0400
@@ -25,7 +25,7 @@
 zeromq/Makefile:	zeromq/configure
 	(cd zeromq ; \
 	 ./configure --without-documentation --prefix=${PREFIX} \
-	             --includedir=${PREFIX}/include/mythtv/zeromq)
+	             --includedir=${PREFIX}/include/mythtv/zeromq --libdir=${LIBDIR})
 
 zeromq-install zeromq-uninstall zeromq-clean zeromq-distclean:
 	${MAKE} -C zeromq ${@:zeromq-%=%} DESTDIR=${INSTALL_ROOT}
