From 548d5e83d8bb2fceb1dd2207aeb2a1f70f7625d5 Mon Sep 17 00:00:00 2001
From: Yuri Konotopov <ykonotopov@gmail.com>
Date: Sun, 21 Feb 2016 23:16:24 +0300
Subject: Do not check runtime dependencies at build time


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 375df56..13e2fc2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,31 +15,6 @@ endif(NOT DEFINED(CMAKE_SIZEOF_VOID_P))
 
 include(GNUInstallDirs)
 
-# https://cmake.org/pipermail/cmake/2011-January/041666.html
-function(find_python_module module)
-	string(TOUPPER ${module} module_upper)
-	if(NOT PY_${module_upper})
-		if(ARGC GREATER 1 AND ARGV1 STREQUAL "REQUIRED")
-			set(PY_${module}_FIND_REQUIRED TRUE)
-		endif()
-		# A module's location is usually a directory, but for binary modules
-		# it's a .so file.
-		execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c"
-			"import re, ${module}; print re.compile('/__init__.py.*').sub('',${module}.__file__)"
-			RESULT_VARIABLE _${module}_status
-			OUTPUT_VARIABLE _${module}_location
-			ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-		if(NOT _${module}_status)
-			set(PY_${module_upper} ${_${module}_location} CACHE STRING 
-				"Location of Python module ${module}")
-		endif(NOT _${module}_status)
-	endif(NOT PY_${module_upper})
-
-	include(FindPackageHandleStandardArgs)
-	find_package_handle_standard_args(PY_${module} DEFAULT_MSG PY_${module_upper})
-endfunction(find_python_module)
-
 if(NOT BUILD_EXTENSION AND NOT BUILD_CONNECTOR)
 	message(FATAL_ERROR "No build options selected")
 endif(NOT BUILD_EXTENSION AND NOT BUILD_CONNECTOR)
@@ -56,10 +31,6 @@ if(BUILD_EXTENSION)
 endif(BUILD_EXTENSION)
 
 if(BUILD_CONNECTOR)
-	find_package(PythonInterp 2.7 REQUIRED)
-	find_python_module(gi.repository.GLib REQUIRED)
-	find_python_module(gi.repository.Gio REQUIRED)
-
 	configure_file("${CMAKE_CURRENT_SOURCE_DIR}/connector/io.github.ne0sight.gs_chrome_connector.json"
 			"${CMAKE_BINARY_DIR}/")
 
