--- test/thostnames.C~	2004-06-29 08:47:52.000000000 +0200
+++ test/thostnames.C	2013-09-07 10:30:24.159157101 +0200
@@ -23,7 +23,7 @@
 {
 	public:
 		test () : iosockinet (sockbuf::sock_stream) {
-			cout << "Constructor rdbuf: " << (unsigned int) rdbuf() << endl;
+			cout << "Constructor rdbuf: " << (size_t) rdbuf() << endl;
 		}
 };
 
@@ -39,7 +39,7 @@
   }
 
   sio.rdbuf()->setname ("Testsocket");
-  cout << "rdbuf: " << (unsigned int) sio.rdbuf() << "\n";
+  cout << "rdbuf: " << (size_t) sio.rdbuf() << "\n";
   cout << "socketname: " << sio.rdbuf()->getname() << endl;
 
   try {
