--- a/setup.py	2024-03-08 19:34:42.148677983 +0100
+++ b/setup.py	2024-03-08 19:34:52.402558423 +0100
@@ -1019,10 +1019,6 @@
             extensions.append(
                 Extension(name="caffe2.python.caffe2_pybind11_state_hip", sources=[]),
             )
-    if cmake_cache_vars["BUILD_FUNCTORCH"]:
-        extensions.append(
-            Extension(name="functorch._C", sources=[]),
-        )
 
     cmdclass = {
         "bdist_wheel": wheel_concatenate,
