This header is required for dolphin.

diff --git a/lib/konq/CMakeLists.txt b/lib/konq/CMakeLists.txt
index 8ecbfa9..445afd9 100644
--- a/lib/konq/CMakeLists.txt
+++ b/lib/konq/CMakeLists.txt
@@ -59,7 +59,7 @@ install( FILES
  konq_dndpopupmenuplugin.h
  knewmenu.h
  konq_menuactions.h
- # konq_copytomenu.h - anyone needs it?
+ konq_copytomenu.h
  konq_operations.h
  libkonq_export.h
  konq_events.h
