Source: https://github.com/klayoutmatthias/klayout/issues/127

diff -ruN klayout-0.25.6.orig/src/db/db/dbCellMapping.cc klayout-0.25.6/src/db/db/dbCellMapping.cc
--- klayout-0.25.6.orig/src/db/db/dbCellMapping.cc	2018-11-26 18:53:43.000000000 +0100
+++ klayout-0.25.6/src/db/db/dbCellMapping.cc	2018-12-12 20:05:11.210171097 +0100
@@ -75,6 +75,12 @@
 	return *this;
   }
 
+  SortedCellIndexIterator &operator--()
+  {
+    --m_n;
+    return *this;
+  }
+
   SortedCellIndexIterator &operator+=(size_t n) 
   {
     m_n += n;
