--- a/Makefile
+++ b/Makefile
@@ -92,9 +92,9 @@
 endif
 
 # Compiler options
-CC       = gcc
+CC       := $(CC)
 # compiling flags here
-CFLAGS   = -g -std=c99 -I$(SRCDIR) -I$(LIBDIR)  -Wall -Wextra -fno-common
+CFLAGS   := $(CFLAGS) -g -std=c99 -I$(SRCDIR) -I$(LIBDIR)  -Wall -Wextra -fno-common
 ifndef MAC_OS
 ifndef NODPKG
 	CFLAGS   +=$(shell dpkg-buildflags --get CPPFLAGS)
@@ -106,11 +106,11 @@
 TEST_CFLAGS = $(CFLAGS) -I.
 
 # Linker options
-LINKER   = gcc
+LINKER   := $(CC)
 ifdef MAC_OS
 LFLAGS   = $(LSODIUM) $(LARGP)
 else
-LFLAGS   = $(LSODIUM) $(LSECCOMP) -fno-common
+LFLAGS   := $(LDFLAGS) $(LSODIUM) $(LSECCOMP) -fno-common
 ifndef NODPKG
 LFLAGS +=$(shell dpkg-buildflags --get LDFLAGS)
 endif
@@ -130,12 +130,12 @@
 ifdef MAC_OS
 CLIENT_LFLAGS = -L$(APILIB) $(LARGP) $(LAGENT) $(LSODIUM)
 else
-CLIENT_LFLAGS = -L$(APILIB) $(LAGENT) $(LSODIUM) $(LSECCOMP)
+CLIENT_LFLAGS := $(LDFLAGS) -L$(APILIB) $(LAGENT) $(LSODIUM) $(LSECCOMP)
 ifndef NODPKG
 	CLIENT_LFLAGS += $(shell dpkg-buildflags --get LDFLAGS)
 endif
 endif
-LIB_LFLAGS = -lc $(LSODIUM)
+LIB_LFLAGS := $(LDFLAGS) -lc $(LSODIUM)
 ifndef MAC_OS
 ifndef NODPKG
 	LIB_LFLAGS += $(shell dpkg-buildflags --get LDFLAGS)
