--- a/leave.c
+++ b/leave.c
@@ -42,7 +42,6 @@
 static char sccsid[] = "@(#)leave.c	8.1 (Berkeley) 6/6/93";
 #endif
 #endif /* not lint */
-#include <sys/cdefs.h>
 
 #include <err.h>
 #include <ctype.h>
@@ -52,7 +51,7 @@
 #include <time.h>
 #include <unistd.h>
 
-void doalarm(u_int);
+void doalarm(unsigned);
 static void usage(void);
 
 /*
@@ -65,7 +64,7 @@
 int
 main(int argc, char **argv)
 {
-	u_int secs;
+	unsigned secs;
 	int hours, minutes;
 	char c, *cp = NULL;
 	struct tm *t;
@@ -136,7 +135,7 @@
 }
 
 void
-doalarm(u_int secs)
+doalarm(unsigned secs)
 {
 	int bother;
 	time_t daytime;
@@ -150,7 +149,7 @@
 		printf("Alarm set for %s. (pid %d)\n", tb, pid);
 		exit(0);
 	}
-	sleep((u_int)2);		/* let parent print set message */
+	sleep((unsigned)2);		/* let parent print set message */
 	if (secs >= 2)
 		secs -= 2;
 
@@ -177,7 +176,7 @@
 
 #define	MSG4	"\07\07Time to leave!\n"
 	for (bother = 10; bother--;) {
-		sleep((u_int)ONEMIN);
+		sleep((unsigned)ONEMIN);
 		if (write(STDOUT_FILENO, MSG4, sizeof(MSG4) - 1) != sizeof(MSG4) - 1)
 			exit(0);
 	}
