diff -urN g-wrap-1.9.8.orig/guile/examples/Makefile.am g-wrap-1.9.8/guile/examples/Makefile.am
--- g-wrap-1.9.8.orig/guile/examples/Makefile.am	2007-05-31 15:07:17.000000000 -0700
+++ g-wrap-1.9.8/guile/examples/Makefile.am	2007-06-01 10:07:32.000000000 -0700
@@ -5,7 +5,10 @@
 
 lib_LTLIBRARIES += libgw-guile-miscutils.la
 libgw_guile_miscutils_la_SOURCES = miscutils-guile.c miscutils-guile.h gw-miscutils.c
-libgw_guile_miscutils_la_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/guile
+libgw_guile_miscutils_la_CFLAGS = \
+	-I$(top_srcdir) \
+	-I$(top_srcdir)/guile \
+	$(LIBFFI_CFLAGS)
 libgw_guile_miscutils_la_LDFLAGS = -module
 libgw_guile_miscutils_la_LIBADD = libmiscutils.la \
 				  $(top_builddir)/guile/g-wrap/libgwrap-guile-runtime.la \
