--- configure.ac.old	2007-01-20 10:54:27.000000000 +0100
+++ configure.ac	2007-01-20 10:55:49.000000000 +0100
@@ -95,7 +95,9 @@
 AM_CONDITIONAL(HAVE_SRFI_35, test "$HAVE_SRFI_35" = yes)
 
 dnl Check for optional libraries
-PKG_CHECK_MODULES(GLIB, glib, [have_glib=yes], [have_glib=no])
+if test "x$with_glib" != "xno"; then
+	PKG_CHECK_MODULES([GLIB], [glib-2.0], [have_glib=yes], [have_glib=no])
+fi
 AM_CONDITIONAL(HAVE_GLIB, test "$have_glib" = yes)
 
 dnl Check for compiler flags and warnings. This should be last, since
