diff -u g-wrap-1.3.4-vanilla/doc/g-wrap.info g-wrap-1.3.4/doc/g-wrap.info
--- g-wrap-1.3.4-vanilla/doc/g-wrap.info	2004-04-19 13:16:18.584411000 -0700
+++ g-wrap-1.3.4/doc/g-wrap.info	2004-04-19 13:17:49.384608248 -0700
@@ -1,5 +1,10 @@
 This is g-wrap.info, produced by makeinfo version 4.2 from g-wrap.texi.
 
+INFO-DIR-SECTION The Algorithmic Language Scheme
+START-INFO-DIR-ENTRY
+* G-wrap: (g-wrap).		Export C libraries into Scheme.
+END-INFO-DIR-ENTRY
+
 
 Indirect:
 g-wrap.info-1: 73
diff -u g-wrap-1.3.4-vanilla/doc/g-wrap.info-1 g-wrap-1.3.4/doc/g-wrap.info-1
--- g-wrap-1.3.4-vanilla/doc/g-wrap.info-1	2004-04-19 13:16:28.147958000 -0700
+++ g-wrap-1.3.4/doc/g-wrap.info-1	2004-04-19 13:17:52.254172008 -0700
@@ -1,5 +1,10 @@
 This is g-wrap.info, produced by makeinfo version 4.2 from g-wrap.texi.
 
+INFO-DIR-SECTION The Algorithmic Language Scheme
+START-INFO-DIR-ENTRY
+* G-wrap: (g-wrap).		Export C libraries into Scheme.
+END-INFO-DIR-ENTRY
+
 
 File: g-wrap.info,  Node: Top,  Next: Copying,  Prev: (dir),  Up: (dir)
 
diff -u g-wrap-1.3.4-vanilla/doc/g-wrap.info-2 g-wrap-1.3.4/doc/g-wrap.info-2
--- g-wrap-1.3.4-vanilla/doc/g-wrap.info-2	2004-04-19 13:16:33.402159000 -0700
+++ g-wrap-1.3.4/doc/g-wrap.info-2	2004-04-19 13:17:55.048747168 -0700
@@ -1,5 +1,10 @@
 This is g-wrap.info, produced by makeinfo version 4.2 from g-wrap.texi.
 
+INFO-DIR-SECTION The Algorithmic Language Scheme
+START-INFO-DIR-ENTRY
+* G-wrap: (g-wrap).		Export C libraries into Scheme.
+END-INFO-DIR-ENTRY
+
 
 File: g-wrap.info,  Node: Accessing the wrapped API from the target language,  Next: Coping With Old Guile Versions,  Prev: Generating the glue code,  Up: Generating and using the glue code
 
diff -r -u g-wrap-1.3.4-vanilla/doc/g-wrap.texi g-wrap-1.3.4/doc/g-wrap.texi
--- g-wrap-1.3.4-vanilla/doc/g-wrap.texi	2002-11-07 09:23:43.000000000 -0800
+++ g-wrap-1.3.4/doc/g-wrap.texi	2004-04-19 13:30:50.391877096 -0700
@@ -7,6 +7,11 @@
 @paragraphindent 0
 @c %**end of header
 
+@dircategory The Algorithmic Language Scheme
+@direntry
+* G-wrap: (g-wrap).		Export C libraries into Scheme.
+@end direntry
+
 @iftex
 @finalout
 @c DL: lose the egregious vertical whitespace, esp. around examples
