diff -ruN mod_perl-2.0.10_pre201601.orig/Makefile.PL mod_perl-2.0.10_pre201601/Makefile.PL
--- mod_perl-2.0.10_pre201601.orig/Makefile.PL	2015-05-13 10:01:18.000000000 +0200
+++ mod_perl-2.0.10_pre201601/Makefile.PL	2016-01-09 21:02:00.000000000 +0100
@@ -25,7 +25,7 @@
 
 }
 
-use lib qw(lib Apache-Test/lib);
+use lib qw(lib);
 
 use Config;
 use File::Spec::Functions;
@@ -122,15 +122,6 @@
 
 sub configure {
 
-    # mod_perl test suite relies on having Apache-Test bundled with
-    # the mod_perl source, since any pre-installed version may not do
-    # the right thing
-    unless (-d "Apache-Test") {
-        error "Can't find a sub-directory Apache-Test. " .
-            "Make sure that you are using a complete source distribution";
-        exit 1;
-    }
-
     set_modperl_version();
 
     if ($old_modperl_version) {
@@ -853,14 +844,9 @@
 run_subtests ::
 	cd ModPerl-Registry && $(MAKE) test
 
-run_subtests ::
-	cd Apache-Reload && $(MAKE) test
-
 EOF
 
     $preamble .= <<'EOF' unless $build->mpm_is_threaded();
-run_subtests ::
-	cd Apache-SizeLimit && $(MAKE) test
 
 EOF
 
@@ -871,42 +857,6 @@
     return $preamble;
 }
 
-sub MY::postamble {
-    my $self = shift;
-
-    my $string = $self->ModPerl::BuildMM::MY::postamble;
-
-    if (!WIN32) {
-        $string .= <<'EOF';
-rpm: dist
-	@[ -d $(PWD)/rpm ] || mkdir $(PWD)/rpm
-	rpmbuild -ta --define "_rpmdir $(PWD)/rpm" \
-                 --define "_srcrpmdir $(PWD)/rpm" \
-                 $(DISTVNAME).tar.gz
-	@mv $(PWD)/rpm/*/*.rpm $(PWD)/rpm/
-	@rm -rf $(PWD)/rpm/*/
-    
-EOF
-    }
-
-    $string .= <<'EOF';
-mydist : Apache-Test/META.yml mod_perl.spec manifest tardist
-
-mod_perl.spec: build/make_rpm_spec
-	$(PERL) build/make_rpm_spec
-
-Apache-Test/META.yml:
-	cd Apache-Test && make metafile
-
-tag :
-	svn copy  https://svn.apache.org/repos/asf/perl/modperl/trunk  https://svn.apache.org/repos/asf/perl/modperl/branches/release/$(VERSION_SYM)
-	svn copy  https://svn.apache.org/repos/asf/perl/modperl/branches/release/$(VERSION_SYM)  https://svn.apache.org/repos/asf/perl/modperl/tags/$(VERSION_SYM)
-	svn copy  https://svn.apache.org/repos/asf/perl/modperl/docs/trunk https://svn.apache.org/repos/asf/perl/modperl/docs/tags/$(VERSION_SYM)
-EOF
-
-    return $string;
-}
-
 # this is a workaround so that ModPerl::MM will move MY::constants
 # away, and Apache-Test/Makefile.PL which has its own MY::constants
 # won't get complaints on MY::constants redefined
diff -ruN mod_perl-2.0.10_pre201601.orig/lib/ModPerl/BuildMM.pm mod_perl-2.0.10_pre201601/lib/ModPerl/BuildMM.pm
--- mod_perl-2.0.10_pre201601.orig/lib/ModPerl/BuildMM.pm	2015-04-01 18:19:18.000000000 +0200
+++ mod_perl-2.0.10_pre201601/lib/ModPerl/BuildMM.pm	2016-01-09 21:04:34.000000000 +0100
@@ -39,12 +39,10 @@
 ModPerl::MM::override_eu_mm_mv_all_methods(@methods);
 use strict 'refs';
 
-my $apache_test_dir = catdir Cwd::getcwd(), "Apache-Test", "lib";
-
 #to override MakeMaker MOD_INSTALL macro
 sub mod_install {
     q{$(PERL) -I$(INST_LIB) -I$(PERL_LIB) \\}."\n" .
-    qq{-I$apache_test_dir -MModPerl::BuildMM \\}."\n" .
+    qq{-MModPerl::BuildMM \\}."\n" .
     q{-e "ExtUtils::Install::install({@ARGV},'$(VERBINST)',0,'$(UNINST)');"}."\n";
 }
 
@@ -271,7 +269,7 @@
 
                 push @target,
                     '$(FULLPERL) -I$(INST_LIB) ' .
-                    "-I$apache_test_dir -MModPerl::BuildMM " .
+                    "-MModPerl::BuildMM " .
                     "-e ModPerl::BuildMM::glue_pod $pm $podpath $blib";
 
                 # Win32 doesn't normally install man pages
diff -ruN mod_perl-2.0.10_pre201601.orig/lib/ModPerl/Manifest.pm mod_perl-2.0.10_pre201601/lib/ModPerl/Manifest.pm
--- mod_perl-2.0.10_pre201601.orig/lib/ModPerl/Manifest.pm	2015-05-13 09:23:09.000000000 +0200
+++ mod_perl-2.0.10_pre201601/lib/ModPerl/Manifest.pm	2016-01-09 21:04:50.000000000 +0100
@@ -119,7 +119,6 @@
 t/apr-ext/perlio
 # PAUSE breaks if a dist has more than one META.yml. the top-level
 # META.yml already excludes Apache-Test from indexing
-Apache-Test/META.yml
 # exclude Apache-Test/MANIFEST since it confuses the mp2 build (e.g it
 # wants Apache-Test/META.yml which we don't supply, see above)
 Apache-Test/MANIFEST
