From a9b0c62d34737cc6f5e0806c8f5305c137f9f155 Mon Sep 17 00:00:00 2001
From: Andre Miranda <andreldm@xfce.org>
Date: Fri, 4 Oct 2019 00:41:13 -0300
Subject: [PATCH] Allow compilation with panel 4.15

---
 panel-plugin/cpu.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panel-plugin/cpu.h b/panel-plugin/cpu.h
index 71c90a2..f6d11ee 100644
--- a/panel-plugin/cpu.h
+++ b/panel-plugin/cpu.h
@@ -28,7 +28,7 @@
 #include "config.h"
 #endif
 
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #include "os.h"
 
-- 
GitLab

