--- a/cmake/XiphosDependencies.cmake
+++ b/cmake/XiphosDependencies.cmake
@@ -84,7 +84,7 @@
   "gdk-pixbuf-2.0"
   "gio-2.0"
   "gobject-2.0"
-  "libsoup-2.4"
+  "libsoup-3.0"
   "pango"
   "minizip"
   "zlib"
@@ -144,7 +144,7 @@
     # Gtk+-3.0 + Webkit2 + GtkHtml-editor
      pkg_check_modules(Gtk REQUIRED IMPORTED_TARGET
       "gtk+-3.0"
-      "webkit2gtk-4.0"
+      "webkit2gtk-4.1"
       )
   endif()
 endif (GTK2)
