See icu-fix-tests-depending-on-date.patch

Index: icu/source/test/intltest/dtfmttst.cpp
===================================================================
--- icu.orig/source/test/intltest/dtfmttst.cpp
+++ icu/source/test/intltest/dtfmttst.cpp
@@ -1132,7 +1132,7 @@ DateFormatTest::TestTwoDigitYear()
         return;
     }
     parse2DigitYear(fmt, "5/6/17", date(117, UCAL_JUNE, 5));
-    parse2DigitYear(fmt, "4/6/34", date(34, UCAL_JUNE, 4));
+    parse2DigitYear(fmt, "4/6/34", date(134, UCAL_JUNE, 4));
 }
 
 // -------------------------------------
