diff --git a/lib/tpm2_identity_util.c b/lib/tpm2_identity_util.c
index a3b0e387..9519ca7b 100644
--- a/lib/tpm2_identity_util.c
+++ b/lib/tpm2_identity_util.c
@@ -17,7 +17,7 @@
 
 // Identity-related functionality that the TPM normally does, but using OpenSSL
 
-#if defined(LIBRESSL_VERSION_NUMBER)
+#if 0
 static int RSA_padding_add_PKCS1_OAEP_mgf1(unsigned char *to, int tlen,
         const unsigned char *from, int flen, const unsigned char *param, int plen,
         const EVP_MD *md, const EVP_MD *mgf1md) {
diff --git a/lib/tpm2_openssl.c b/lib/tpm2_openssl.c
index 6d29c0d4..bd375ce3 100644
--- a/lib/tpm2_openssl.c
+++ b/lib/tpm2_openssl.c
@@ -14,7 +14,7 @@
 #include "tpm2_openssl.h"
 
 /* compatibility function for OpenSSL versions < 1.1.0 */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 static int BN_bn2binpad(const BIGNUM *a, unsigned char *to, int tolen) {
     int r;
     int topad;
