--- a/tools/find_unknown/lpi_find_unknown.cc
+++ b/tools/find_unknown/lpi_find_unknown.cc
@@ -219,7 +219,7 @@
 	dump_payload(ident->lpi, 0, pload_out, 500);
 	dump_payload(ident->lpi, 1, pload_in, 500);
         str = (char *)malloc(750);
-        snprintf(str, 1000, "%s %s %u %u %u %.3f %.3f %" PRIu64 " %" PRIu64 " %s %s\n",
+        snprintf(str, 750, "%s %s %u %u %u %.3f %.3f %" PRIu64 " %" PRIu64 " %s %s\n",
 			s_ip, c_ip,
                         f->id.get_server_port(), f->id.get_client_port(),
                         f->id.get_protocol(), ident->start_ts,
--- a/tools/protoident/lpi_protoident.cc
+++ b/tools/protoident/lpi_protoident.cc
@@ -214,7 +214,7 @@
 	dump_payload(ident->lpi, 0, pload_out, 500);
 	dump_payload(ident->lpi, 1, pload_in, 500);
         str = (char *)malloc(750);
-        snprintf(str, 1000, "%s %s %s %u %u %u %.3f %.3f %" PRIu64 " %" PRIu64 " %s %s\n",
+        snprintf(str, 750, "%s %s %s %u %u %u %.3f %.3f %" PRIu64 " %" PRIu64 " %s %s\n",
 			proto->name, s_ip, c_ip,
                         f->id.get_server_port(), f->id.get_client_port(),
                         f->id.get_protocol(), ident->start_ts,
