--- a/src/tools/analyzer/Makefile.am
+++ b/src/tools/analyzer/Makefile.am
@@ -1,5 +1,7 @@
 pkgpythondir = $(python3dir)/sssd
+modulesdir = $(pkgpythondir)/modules
 
+if BUILD_PYTHON_BINDINGS
 dist_pkgpython_SCRIPTS = \
     sss_analyze.py \
     $(NULL)
@@ -10,7 +12,7 @@
     source_reader.py \
     $(NULL)
 
-modulesdir = $(pkgpythondir)/modules
 dist_modules_DATA = \
     modules/request.py \
     $(NULL)
+endif
