diff -ub -purN -x '*~' a/qml/SGVideoNode.cpp b/qml/SGVideoNode.cpp
--- a/qml/SGVideoNode.cpp	2017-06-21 03:47:15.000000000 +0200
+++ b/qml/SGVideoNode.cpp	2020-01-07 05:37:05.304897727 +0100
@@ -25,6 +25,8 @@
 #include "QtAV/VideoFrame.h"
 #include <QtCore/QScopedPointer>
 #include <QtGui/QOpenGLFunctions>
+#include <QtQuick/QSGMaterialType>
+#include <QtQuick/QSGMaterial>
 #include <QtQuick/QSGMaterialShader>
 
 // all in QSGRenderThread
