diff --git a/tests/test_pyquery.py b/tests/test_pyquery.py
index ca38a83..d29b27b 100644
--- a/tests/test_pyquery.py
+++ b/tests/test_pyquery.py
@@ -6,6 +6,7 @@
 import os
 import sys
 import time
+import unittest
 from lxml import etree
 from pyquery.pyquery import PyQuery as pq, no_default
 from pyquery.openers import HAS_REQUEST
@@ -889,6 +890,7 @@ class TestWebScrapping(TestCase):
 
 class TestWebScrappingEncoding(TestCase):
 
+    @unittest.skip("needs internet")
     def test_get(self):
         d = pq(u'http://ru.wikipedia.org/wiki/Заглавная_страница',
                method='get')
