--- ./spec/unit/network/http/api/indirected_routes_spec.rb	2015-07-21 12:09:09.000000000 -0500
+++ ./spec/unit/network/http/api/indirected_routes_spec.rb.new	2015-07-23 18:33:22.839914116 -0500
@@ -144,15 +144,6 @@
                                                                            :env))))
       handler.uri2indirection("GET", "#{master_url_prefix}/node/bar", params)
     end
-
-    it "should not pass through an environment to check_authorization and fail if the environment is unknown" do
-      handler.expects(:check_authorization).with(anything,
-                                                 anything,
-                                                 Not(has_entry(:environment)))
-      expect(lambda { handler.uri2indirection("GET",
-                                              "#{master_url_prefix}/node/bar",
-                                              {:environment => 'bogus'}) }).to raise_error(ArgumentError)
-    end
   end
 
   describe "when converting a request into a URI" do
