diff -Naur countrycodes-1.0.5-orig/src/Makefile countrycodes-1.0.5/src/Makefile
--- countrycodes-1.0.5-orig/src/Makefile	2003-02-01 09:56:08.000000000 -0600
+++ countrycodes-1.0.5/src/Makefile	2007-02-03 20:12:26.000000000 -0600
@@ -70,7 +70,7 @@
 	strip $(PROGRAM)
 
 install:
-	$(INSTALL) -g root -m $(BINMODE) -o root -s ${srcdir}/$(PROGRAM) ${bindir}
+	$(INSTALL) -g root -m $(BINMODE) -o root ${srcdir}/$(PROGRAM) ${bindir}
 	@echo "Installing man page..."
 	@$(SEDBIN) $(SEDCMDS) ${docdir}/iso3166.1.in > ${mandir}/iso3166.1
 	@chown 0.0 ${mandir}/iso3166.1
