--- a/Makefile.PL	2020-05-27 02:16:42.106531746 +0300
+++ b/Makefile.PL	2020-05-27 02:17:30.568531767 +0300
@@ -170,18 +170,15 @@
         INSTALLSCRIPT        => '$(PREFIX)/bin',
         INSTALLSITESCRIPT    => '$(PREFIX)/bin',
         INSTALLVENDORSCRIPT  => '$(PREFIX)/bin',
-        INSTALLLIB           => '$(DATADIR)/lib',
-        INSTALLSITELIB       => '$(DATADIR)/lib',
-        INSTALLVENDORLIB     => '$(DATADIR)/lib',
         INSTALLMAN1DIR       => '$(PREFIX)/share/man/man1',
         INSTALLSITEMAN1DIR   => '$(PREFIX)/share/man/man1',
         INSTALLVENDORMAN1DIR => '$(PREFIX)/share/man/man1',
         INSTALLMAN3DIR       => '$(PREFIX)/share/man/man3',
         INSTALLSITEMAN3DIR   => '$(PREFIX)/share/man/man3',
         INSTALLVENDORMAN3DIR => '$(PREFIX)/share/man/man3',
-        SYSCONFDIR           => '$(PREFIX)/etc/fusioninventory',
+        SYSCONFDIR           => '/etc/fusioninventory',
         DATADIR              => '$(PREFIX)/share/fusioninventory',
-        LOCALSTATEDIR        => '$(PREFIX)/var/fusioninventory',
+        LOCALSTATEDIR        => '/var/lib/fusioninventory',
     );
 
     # allow variables defined on command line to override defaults
--- a/lib/FusionInventory/Agent/Config.pm     2020-05-27 10:06:21.248964808 +0300
+++ b/lib/FusionInventory/Agent/Config.pm     2020-05-27 10:06:57.324964823 +0300
@@ -67,7 +67,7 @@
     my ($class, %params) = @_;
 
     my $self = {
-        '_confdir' => undef, # SYSCONFDIR replaced here from Makefile
+        '_confdir' => '/etc/fusioninventory', # SYSCONFDIR replaced here from Makefile
     };
     bless $self, $class;
     $self->_loadDefaults();

