diff -Nu libjwc_f-1.1/configure.in ccp4-6.1.2/x-windows/libjwc/libjwc_f/configure.in
--- libjwc_f-1.1/configure.in	1999-10-21 16:07:14.000000000 +0200
+++ ccp4-6.1.2/x-windows/libjwc/libjwc_f/configure.in	2004-03-11 17:31:45.000000000 +0100
@@ -8,6 +8,7 @@
 AC_CANONICAL_SYSTEM
 
 AM_INIT_AUTOMAKE(libjwc_f, 1.1)
+AM_MAINTAINER_MODE
 
 case ${with_g77} in
   "yes" ) CC="gcc" GCC=yes F77=g77 ;;
@@ -17,7 +18,7 @@
 AC_PROG_F77
 
 AC_PROG_INSTALL
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 case "$target" in
   *irix6* )
@@ -26,8 +27,9 @@
       case "$FFLAGS" in
       *-64* ) ;;   # assume they meant it
       *-o32* | *-32* ) ;;   # presumably OK
-      * ) FFLAGS="$FFLAGS -o32 -O -g3"
-          CFLAGS=-32
+      *-n32* ) ;;
+      * ) FFLAGS="$FFLAGS -n32 -O -g3"
+          CFLAGS=-n32
           AC_MSG_WARN([
 
 *** Using compiler flag -o32 for probable compatibility with
@@ -47,8 +49,33 @@
 *** See the INSTALL file for more info.
 ])
     fi ;;
+  *linux*  )
+    case $F77 in
+    g77*)
+     case "$FFLAGS" in
+      *no-second-underscore* ) ;; # presumably OK
+      * ) FFLAGS="$FFLAGS -fno-second-underscore -fno-globals" ;;
+      esac
+    esac ;;
+  *darwin* ) 
+    case $F77 in
+    g77*)
+     case "$FFLAGS" in
+      *no-second-underscore* ) ;; # presumably OK
+      * ) FFLAGS="$FFLAGS -fno-second-underscore -fno-globals" ;;
+      esac
+#mac g77 currently goes not handle -fno-common
+    enable_shared=no;
+    esac ;;
   * ) 
 esac
+# efc, version 7.1 at least does not support soname, so use CC
+case "$F77" in
+ *efc ) F77LD=${F77LD:-'${CC}'} ;;
+ * ) F77LD=${F77LD:-'${F77}'}
+esac
+
+AC_SUBST(F77LD)
 
 AM_PROG_LIBTOOL
 
Common subdirectories: libjwc_f-1.1/doc and ccp4-6.1.2/x-windows/libjwc/libjwc_f/doc
diff -Nu libjwc_f-1.1/opn_scr.f ccp4-6.1.2/x-windows/libjwc/libjwc_f/opn_scr.f
--- libjwc_f-1.1/opn_scr.f	1999-09-07 12:50:13.000000000 +0200
+++ ccp4-6.1.2/x-windows/libjwc/libjwc_f/opn_scr.f	2000-10-25 13:03:09.000000000 +0200
@@ -16,6 +16,6 @@
 C
 C====== Open the file
 C
-      OPEN (IUN_SCR, STATUS='SCRATCH')
-      RETURN
+C     OPEN (IUN_SCR, STATUS='SCRATCH')
+      CALL CCPDPN(IUN_SCR,'JWCSCR','SCRATCH','U',0, IFAIL) 
       END
diff -Nu libjwc_f-1.1/rpf_dstar.f ccp4-6.1.2/x-windows/libjwc/libjwc_f/rpf_dstar.f
--- libjwc_f-1.1/rpf_dstar.f	1999-07-09 13:39:38.000000000 +0200
+++ ccp4-6.1.2/x-windows/libjwc/libjwc_f/rpf_dstar.f	2004-06-04 12:10:59.000000000 +0200
@@ -376,7 +376,8 @@
              INR_FLAG = 2
              PHIW = 0.0
           END IF
-        ELSE
+          RETURN
+        ENDIF
 C
 C---- Calculate distance of edge of spot from sphere at end of rotation
 Crotgen.for
@@ -418,7 +419,6 @@
             INR_FLAG = 4
  40         RETURN
           END IF
-        END IF
       END IF
       RETURN
       END
