https://github.com/leethomason/tinyxml2/commit/0838dd11a3d7a122799298d6556cff501b88421a.patch

From 0838dd11a3d7a122799298d6556cff501b88421a Mon Sep 17 00:00:00 2001
From: heitbaum <rudi@heitbaum.com>
Date: Mon, 17 May 2021 15:14:01 +1000
Subject: [PATCH] meson.build: fix typo
--- a/meson.build
+++ b/meson.build
@@ -58,7 +58,7 @@ dep_tinyxml2 = declare_dependency(
 # This is the new way to set dependencies, but let's not break users of older
 # versions of meson
 if meson.version().version_compare('>= 0.54.0')
-  meson.override_dependency('tinyxml2', dep_tinxyml2)
+  meson.override_dependency('tinyxml2', dep_tinyxml2)
 endif
 
 if get_option('tests')
