--- a/swig/ruby/Makefile.am
+++ b/swig/ruby/Makefile.am
@@ -12,7 +12,7 @@ Makefile.ruby: extconf.rb
 	PREFIX=$(prefix) $(RUBY) $< --with-obexftp-include=$(top_srcdir)
 
 obexftp.so: ruby_wrap.c Makefile.ruby
-	$(MAKE) -fMakefile.ruby
+	$(MAKE) -fMakefile.ruby LIBPATH="-Wl,-rpath-link,../../bfb/.libs -Wl,-rpath-link,../../multicobex/.libs -Wl,-rpath-link,../../obexftp/.libs -L../../bfb/.libs -L../../multicobex/.libs -L../../obexftp/.libs"
 
 install-data-local: Makefile.ruby
 	$(MAKE) -j1 -fMakefile.ruby install
