Description: Drop -d which causes an error with GCC 7
Author: Stephen Kitt <skitt@debian.org>

--- a/Makefile
+++ b/Makefile
@@ -61,7 +61,7 @@
 
 
 
-CFLAGS+=-d -Wall  `$(SDL_CONFIG) --cflags` -DPLATFORM_PC -DVERSION=\"$(VERSION)\" $(SMOOTH_SCRL) $(LIGHTNING)\
+CFLAGS+=-Wall  `$(SDL_CONFIG) --cflags` -DPLATFORM_PC -DVERSION=\"$(VERSION)\" $(SMOOTH_SCRL) $(LIGHTNING)\
 	-DPACKAGE_DATA_DIR=\"$(PACKAGE_DATA_DIR)\" $(FONT_USE_PIXMAP) $(MUSIC) $(DESIGNER)
 LDFLAGS?=
 # Use these instead for debugging and/or profiling (mainly intended for GNU Robbo developers)
