diff -urN fdutils-5.4.old/doc/Makefile.in fdutils-5.4/doc/Makefile.in
--- fdutils-5.4.old/doc/Makefile.in	1999-07-02 15:03:25.000000000 -0700
+++ fdutils-5.4/doc/Makefile.in	2003-12-10 00:24:36.000000000 -0800
@@ -22,7 +22,7 @@
                floppycontrol.1 makefloppies.1 xdfcopy.1 fdmount.1 \
                floppymeter.1 setfdprm.1
 
-MANPAGES4   = fd.4
+MANPAGES4   =
 MANPERM     = 644
 UID         = root
 GID         = root
