diff --git a/src/libs/loguru/loguru.cpp b/src/libs/loguru/loguru.cpp
index 9675cc1d..c4b5c0e3 100644
--- a/src/libs/loguru/loguru.cpp
+++ b/src/libs/loguru/loguru.cpp
@@ -83,7 +83,7 @@
 	#ifndef LOGURU_STACKTRACES
 		#define LOGURU_STACKTRACES 0
 	#endif
-#elif defined(__rtems__) || defined(__ANDROID__) || defined(__FreeBSD__)
+#elif defined(__rtems__) || defined(__ANDROID__) || defined(__FreeBSD__) || !defined(__GLIBC__)
 	#define LOGURU_PTHREADS    1
 	#define LOGURU_WINTHREADS  0
 	#ifndef LOGURU_STACKTRACES
-- 
2.34.1

