--- a/lib/fxjpegio.cpp
+++ b/lib/fxjpegio.cpp
@@ -138,7 +138,7 @@
     src->pub.next_input_byte=src->buffer;
     src->pub.bytes_in_buffer=1;
     }
-  return true;
+  return TRUE;
   }
 
 
@@ -216,7 +216,7 @@
   srcinfo.src=&src.pub;
 
   // read the header from the jpg;
-  jpeg_read_header(&srcinfo,true);
+  jpeg_read_header(&srcinfo,TRUE);
 
   // make sure the output is RGB
   srcinfo.out_color_space=JCS_RGB;
@@ -279,7 +279,7 @@
   dest->stream->save(dest->buffer,JPEG_BUFFER_SIZE);
   dest->pub.free_in_buffer=JPEG_BUFFER_SIZE;
   dest->pub.next_output_byte=dest->buffer;
-  return true;
+  return TRUE;
   }
 
 
@@ -336,8 +336,8 @@
   dstinfo.dest=&dst.pub;
 
   jpeg_set_defaults(&dstinfo);
-  jpeg_set_quality(&dstinfo,quality,true);
-  jpeg_start_compress(&dstinfo,true);
+  jpeg_set_quality(&dstinfo,quality,TRUE);
+  jpeg_start_compress(&dstinfo,TRUE);
 
   // Write the jpeg data
   pp=data;
