Index: client/Makefile.am
===================================================================
--- client/Makefile.am	(revision 1257)
+++ client/Makefile.am	(working copy)
@@ -28,20 +28,3 @@
 	electricsheep-preferences.glade electricsheep.h electricsheep-saver \
 	16_electricsheep 
 
-GNOME_SCREENSAVER_DATADIR = $(shell pkg-config --variable themesdir gnome-screensaver)
-GNOME_SCREENSAVER_PRIVEXEDIR = $(shell pkg-config --variable privlibexecdir gnome-screensaver)
-
-install-data-local:
-	-$(INSTALL) electricsheep.desktop $(GNOME_SCREENSAVER_DATADIR)
-	-$(INSTALL) electricsheep-saver $(GNOME_SCREENSAVER_PRIVEXEDIR)
-	-$(INSTALL) -d $(pkgdatadir)
-	-$(INSTALL) 16_electricsheep /usr/share/gconf/defaults
-	-update-gconf-defaults
-
-uninstall-local:
-	-rm /usr/share/gconf/defaults/16_electricsheep
-	-rm -rf $(pkgdatadir)
-	-rm $(GNOME_SCREENSAVER_PRIVEXEDIR)/electricsheep-saver
-	-rm $(GNOME_SCREENSAVER_DATADIR)/electricsheep.desktop
-	-update-gconf-defaults
-
