Index: electricsheep-2.7_beta11/electricsheep.c
===================================================================
--- electricsheep-2.7_beta11.orig/electricsheep.c
+++ electricsheep-2.7_beta11/electricsheep.c
@@ -652,7 +652,7 @@ void copy_out_file(char *fname) {
       return;
     }
 
-    if (0 > av_open_input_file(&ictx, fname, NULL, 0, NULL)) {
+    if (0 > avformat_open_input(&ictx, fname, NULL, NULL)) {
 	perror(fname);
 	exit(1);
     }
@@ -699,8 +699,6 @@ void copy_out_file(char *fname) {
 	output_ctx->oformat = ofmt;
 
 	st = av_new_stream(output_ctx, output_ctx->nb_streams);
-	st->stream_copy = 1;
-	av_set_parameters(output_ctx, NULL);
 
 	icodec = ictx->streams[input_stream_index]->codec;
 	codec = output_ctx->streams[0]->codec;
@@ -718,12 +716,12 @@ void copy_out_file(char *fname) {
 	codec->height = icodec->height;
 	codec->has_b_frames = icodec->has_b_frames;
 
-	if (url_fopen(&output_ctx->pb, pipe, URL_WRONLY) < 0) {
+	if (avio_open(&output_ctx->pb, pipe, AVIO_FLAG_WRITE) < 0) {
 	    fprintf(logout, "Could not open '%s'\n", pipe);
 	    exit(1);
 	}
 
-	av_write_header(output_ctx);
+	avformat_write_header(output_ctx, NULL);
     }
 
     while (1) {
