--- dvipng-1.15-orig/configure.ac
+++ dvipng-1.15/configure.ac
@@ -78,7 +78,8 @@
 AC_CHECK_HEADERS([libgen.h])
 PSFONTS_O=""
 AC_SUBST(PSFONTS_O)
-AC_CHECK_FT2(,[CFLAGS="$FT2_CFLAGS $CFLAGS"
+PKG_CHECK_MODULES([FT2], [freetype2 >= 6.1.0],
+	[CFLAGS="$FT2_CFLAGS $CFLAGS"
 	LIBS="$FT2_LIBS $LIBS"
 	PSFONTS_O="sfd.o ft.o enc.o fontmap.o tfm.o"
 	AC_DEFINE(HAVE_FT2, 1, [Define to 1 if you have freetype2])
