diff --git a/dependencies/CMakeLists.txt b/dependencies/CMakeLists.txt
index 2c41169f..2f5d4436 100644
--- a/dependencies/CMakeLists.txt
+++ b/dependencies/CMakeLists.txt
@@ -129,15 +129,3 @@ endfunction()
 if(SIMDJSON_COMPETITION)
   competition_scope_()
 endif()
-
-cmake_dependent_option(SIMDJSON_CXXOPTS "Download cxxopts (necessary for tools)" ON
-        SIMDJSON_ALLOW_DOWNLOADS OFF)
-
-if(SIMDJSON_CXXOPTS)
-  set_off(CXXOPTS_BUILD_EXAMPLES)
-  set_off(CXXOPTS_BUILD_TESTS)
-  set_off(CXXOPTS_ENABLE_INSTALL)
-
-  import_dependency(cxxopts jarro2783/cxxopts 794c975)
-  add_dependency(cxxopts)
-endif()
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index f1a82be7..b1b7b1a0 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -1,10 +1,8 @@
-if(TARGET cxxopts) # we only build the tools if cxxopts is available
   message(STATUS "We have cxxopts as a dependency and we are buiding the tools (e.g., json2json).")
-  link_libraries(simdjson simdjson-internal-flags simdjson-windows-headers cxxopts)
+  link_libraries(simdjson simdjson-internal-flags simdjson-windows-headers)
   add_executable(json2json json2json.cpp)
   add_executable(jsonstats jsonstats.cpp)
-  add_executable(jsonpointer jsonpointer.cpp)
+  add_executable(simdjsonpointer jsonpointer.cpp)
   add_executable(minify minify.cpp)
-else()
-  message(STATUS "We are missing cxxopts as a dependency so the tools (e.g., json2json) are omitted.")
-endif()
\ No newline at end of file
+  install(TARGETS json2json jsonstats simdjsonpointer minify DESTINATION bin)
+
