--- a/net80211/ieee80211_linux.h	2008-05-09 01:15:01 +0400
+++ b/net80211/ieee80211_linux.h	2008-05-11 02:26:35 +0400
@@ -315,6 +315,9 @@ typedef spinlock_t acl_lock_t;
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,14)
 #define __skb_append(a,b,c)	__skb_append(a, b)
 #endif
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,25)
+#define __skb_append(a,b,c)	__skb_queue_after(c,a,b)
+#endif
 
 /*
  * Per-node power-save queue definitions.  Beware of control
