diff -ru baselayout-1.12.5.orig/sbin/env-update.sh baselayout-1.12.5/sbin/env-update.sh
--- baselayout-1.12.5.orig/sbin/env-update.sh	2010-08-17 12:56:00.968746729 +0200
+++ baselayout-1.12.5/sbin/env-update.sh	2010-08-17 12:56:47.185326760 +0200
@@ -1,4 +1,4 @@
-#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
+#! /usr/bin/env bash
 # Copyright 1999-2004 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
diff -ru baselayout-1.12.5.orig/sbin/runscript.sh baselayout-1.12.5/sbin/runscript.sh
--- baselayout-1.12.5.orig/sbin/runscript.sh	2010-08-17 12:56:00.968746729 +0200
+++ baselayout-1.12.5/sbin/runscript.sh	2010-08-17 12:56:58.362098849 +0200
@@ -1,4 +1,4 @@
-#!@GENTOO_PORTAGE_EPREFIX@/bin/bash
+#! /usr/bin/env bash
 # Copyright 1999-2006 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
