diff -Naur kicad-9.0.4-orig/CMakeLists.txt kicad-9.0.4/CMakeLists.txt
--- kicad-9.0.4-orig/CMakeLists.txt	2025-08-17 12:32:36.000000000 -0700
+++ kicad-9.0.4/CMakeLists.txt	2025-09-15 08:02:58.046597337 -0700
@@ -22,7 +22,7 @@
 #  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 #
 
-cmake_minimum_required( VERSION 3.21 FATAL_ERROR )
+cmake_minimum_required( VERSION 3.5 )
 
 # Generate DEPFILES without transforming relative paths
 cmake_policy( SET CMP0116 OLD )
diff -Naur kicad-9.0.4-orig/thirdparty/pybind11/tests/CMakeLists.txt kicad-9.0.4/thirdparty/pybind11/tests/CMakeLists.txt
--- kicad-9.0.4-orig/thirdparty/pybind11/tests/CMakeLists.txt	2025-08-17 12:32:36.000000000 -0700
+++ kicad-9.0.4/thirdparty/pybind11/tests/CMakeLists.txt	2025-09-15 08:09:03.015215227 -0700
@@ -5,7 +5,7 @@
 # All rights reserved. Use of this source code is governed by a
 # BSD-style license that can be found in the LICENSE file.
 
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff -Naur kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/installed_embed/CMakeLists.txt kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/installed_embed/CMakeLists.txt
--- kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/installed_embed/CMakeLists.txt	2025-08-17 12:32:36.000000000 -0700
+++ kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/installed_embed/CMakeLists.txt	2025-09-15 08:05:47.062205265 -0700
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff -Naur kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/installed_function/CMakeLists.txt kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/installed_function/CMakeLists.txt
--- kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/installed_function/CMakeLists.txt	2025-08-17 12:32:36.000000000 -0700
+++ kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/installed_function/CMakeLists.txt	2025-09-15 08:06:36.082959432 -0700
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 project(test_installed_module CXX)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
diff -Naur kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/installed_target/CMakeLists.txt kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/installed_target/CMakeLists.txt
--- kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/installed_target/CMakeLists.txt	2025-08-17 12:32:36.000000000 -0700
+++ kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/installed_target/CMakeLists.txt	2025-09-15 08:05:56.948357435 -0700
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff -Naur kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/subdirectory_embed/CMakeLists.txt kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/subdirectory_embed/CMakeLists.txt
--- kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/subdirectory_embed/CMakeLists.txt	2025-08-17 12:32:36.000000000 -0700
+++ kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/subdirectory_embed/CMakeLists.txt	2025-09-15 08:07:21.017650001 -0700
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff -Naur kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/subdirectory_function/CMakeLists.txt kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/subdirectory_function/CMakeLists.txt
--- kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/subdirectory_function/CMakeLists.txt	2025-08-17 12:32:36.000000000 -0700
+++ kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/subdirectory_function/CMakeLists.txt	2025-09-15 08:07:23.548688880 -0700
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff -Naur kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/subdirectory_target/CMakeLists.txt kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/subdirectory_target/CMakeLists.txt
--- kicad-9.0.4-orig/thirdparty/pybind11/tests/test_cmake_build/subdirectory_target/CMakeLists.txt	2025-08-17 12:32:36.000000000 -0700
+++ kicad-9.0.4/thirdparty/pybind11/tests/test_cmake_build/subdirectory_target/CMakeLists.txt	2025-09-15 08:07:26.207729720 -0700
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4)
+cmake_minimum_required(VERSION 3.5)
 
 # The `cmake_minimum_required(VERSION 3.4...3.18)` syntax does not work with
 # some versions of VS that have a patched CMake 3.11. This forces us to emulate
diff -Naur kicad-9.0.4-orig/thirdparty/sentry-native/external/crashpad/third_party/zlib/zlib/CMakeLists.txt kicad-9.0.4/thirdparty/sentry-native/external/crashpad/third_party/zlib/zlib/CMakeLists.txt
--- kicad-9.0.4-orig/thirdparty/sentry-native/external/crashpad/third_party/zlib/zlib/CMakeLists.txt	2025-08-17 12:32:36.000000000 -0700
+++ kicad-9.0.4/thirdparty/sentry-native/external/crashpad/third_party/zlib/zlib/CMakeLists.txt	2025-09-15 08:07:32.701829459 -0700
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 set(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS ON)
 
 project(zlib C)
