diff -ur pdv-1.5.1-orig/pdv.c pdv-1.5.1/pdv.c
--- pdv-1.5.1-orig/Makefile.in	2008-12-06 23:17:46.000000000 -0800
+++ pdv-1.5.1/Makefile.in	2008-12-06 23:18:37.000000000 -0800
@@ -536,7 +536,8 @@
 	touch $@
 
 pdv_fixed : pdv_fixed.c pdv
-	$(STRIPBIN) pdv
+	# let portage decide whether to strip binaries
+	#$(STRIPBIN) pdv
 	printf "%-5.5s" $$VERSION >> pdv
 	printf "\000\000\000\000\000\000\000\000" >> pdv
 	sleep 1
