From d22cc5aa53f6c7312fb1da7a2a507155bab8fabc Mon Sep 17 00:00:00 2001
From: Diogo Pereira <sir.suriv@gmail.com>
Date: Wed, 3 Aug 2016 01:15:52 +0100
Subject: [PATCH] Allow disabling libsecret dependency

---
 remmina-plugins-gnome/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/remmina-plugins-gnome/CMakeLists.txt b/remmina-plugins-gnome/CMakeLists.txt
index adf2d03..b392ad6 100644
--- a/remmina-plugins-gnome/CMakeLists.txt
+++ b/remmina-plugins-gnome/CMakeLists.txt
@@ -31,7 +31,7 @@
 # files in the program, then also delete it here.
 
 
-find_package(Libsecret)
+find_suggested_package(Libsecret)
 if(LIBSECRET_FOUND)
 	set(REMMINA_PLUGINS_GNOME_SRCS
 		src/glibsecret_plugin.c
