Index: libexecinfo-1.1/execinfo.c
===================================================================
--- libexecinfo-1.1.orig/execinfo.c
+++ libexecinfo-1.1/execinfo.c
@@ -78,7 +78,7 @@ backtrace_symbols(void *const *buffer, i
     rval = malloc(clen);
     if (rval == NULL)
         return NULL;
-    (char **)cp = &(rval[size]);
+    cp = &(rval[size]);
     for (i = 0; i < size; i++) {
         if (dladdr(buffer[i], &info) != 0) {
             if (info.dli_sname == NULL)
