--- Test-Harness/t/sample-tests/taint
+++ Test-Harness/t/sample-tests/taint
@@ -3,5 +3,5 @@
 use lib qw(t/lib);
 use Test::More tests => 1;
 
-eval { kill 0, $^X };
+eval { `$^X -e1` };
 like( $@, '/^Insecure dependency/', '-T honored' );
--- Test-Harness/t/sample-tests/taint_warn
+++ Test-Harness/t/sample-tests/taint_warn
@@ -6,6 +6,6 @@ use Test::More tests => 1;
 my $warnings = '';
 {
     local $SIG{__WARN__} = sub { $warnings .= join '', @_ };
-    kill 0, $^X;
+    `$^X -e1`;
 }
 like( $warnings, '/^Insecure dependency/', '-t honored' );
