--- configure.ac	2011-06-13 13:46:22.817856139 +0300
+++ configure.ac	2011-06-13 13:47:56.000000000 +0300
@@ -277,7 +277,7 @@
 if test $with_gnutls = yes; then
   AC_CHECK_HEADERS(gnutls/gnutls.h)
   if test $ac_cv_header_gnutls_gnutls_h != no; then
-    AM_PATH_LIBGNUTLS("1.0.0",
+    PKG_CHECK_MODULES(LIBGNUTLS, gnutls >= 1.0.0,
                       [M4_DEFS="$M4_DEFS -DWITH_SSL -DWITH_GNUTLS"
                       AC_DEFINE(USE_GNUTLS, 1,
                                 [Define to 1 if USE_GNUTLS is requested.])],
