diff --git a/tests/autoptr.test b/tests/autoptr.test
index 92bc7b5..6deac5f 100755
--- a/tests/autoptr.test
+++ b/tests/autoptr.test
@@ -98,7 +98,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/bufio.test b/tests/bufio.test
index 60faab1..ab3986f 100755
--- a/tests/bufio.test
+++ b/tests/bufio.test
@@ -41,7 +41,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/charinbuffer.test b/tests/charinbuffer.test
index 8de99d7..0076db0 100755
--- a/tests/charinbuffer.test
+++ b/tests/charinbuffer.test
@@ -39,7 +39,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/common_utils.test b/tests/common_utils.test
index 21cd063..11f8b50 100755
--- a/tests/common_utils.test
+++ b/tests/common_utils.test
@@ -63,7 +63,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/connector.test b/tests/connector.test
index 54d1928..364930b 100755
--- a/tests/connector.test
+++ b/tests/connector.test
@@ -37,7 +37,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/connector2.test b/tests/connector2.test
index e843f89..6a6474a 100755
--- a/tests/connector2.test
+++ b/tests/connector2.test
@@ -40,7 +40,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/echoxdr.test b/tests/echoxdr.test
index 8280374..985d657 100755
--- a/tests/echoxdr.test
+++ b/tests/echoxdr.test
@@ -48,7 +48,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/fdset.test b/tests/fdset.test
index 8800539..a57a17a 100755
--- a/tests/fdset.test
+++ b/tests/fdset.test
@@ -43,7 +43,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/fork.test b/tests/fork.test
index a70cae3..ec0c4d0 100755
--- a/tests/fork.test
+++ b/tests/fork.test
@@ -43,7 +43,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/genserver.test b/tests/genserver.test
index 5877499..18d19d4 100755
--- a/tests/genserver.test
+++ b/tests/genserver.test
@@ -38,7 +38,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/glops.test b/tests/glops.test
index d24f5e4..f666394 100755
--- a/tests/glops.test
+++ b/tests/glops.test
@@ -49,7 +49,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/idset.test b/tests/idset.test
index a900d2f..97a69c7 100755
--- a/tests/idset.test
+++ b/tests/idset.test
@@ -48,7 +48,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/inet_address.test b/tests/inet_address.test
index deac989..298ea98 100755
--- a/tests/inet_address.test
+++ b/tests/inet_address.test
@@ -49,7 +49,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/inifile.test b/tests/inifile.test
index dae0529..9da154c 100755
--- a/tests/inifile.test
+++ b/tests/inifile.test
@@ -58,7 +58,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/logger.test b/tests/logger.test
index 6c38072..fccd4fa 100755
--- a/tests/logger.test
+++ b/tests/logger.test
@@ -77,7 +77,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/memdump.test b/tests/memdump.test
index 0f4465f..662fb96 100755
--- a/tests/memdump.test
+++ b/tests/memdump.test
@@ -72,7 +72,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/pidflock.test b/tests/pidflock.test
index 33f5a0f..424ef0a 100755
--- a/tests/pidflock.test
+++ b/tests/pidflock.test
@@ -42,7 +42,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/pipe.test b/tests/pipe.test
index a1b134f..ab4888a 100755
--- a/tests/pipe.test
+++ b/tests/pipe.test
@@ -45,7 +45,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/pq.test b/tests/pq.test
index a45b024..6495533 100755
--- a/tests/pq.test
+++ b/tests/pq.test
@@ -38,7 +38,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/reactor.test b/tests/reactor.test
index 8072201..918c904 100755
--- a/tests/reactor.test
+++ b/tests/reactor.test
@@ -68,7 +68,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/regexp.test b/tests/regexp.test
index 087331c..c936fdc 100755
--- a/tests/regexp.test
+++ b/tests/regexp.test
@@ -42,7 +42,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/rt.test b/tests/rt.test
index 3d854bd..8703d75 100755
--- a/tests/rt.test
+++ b/tests/rt.test
@@ -42,7 +42,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/semaphore.test b/tests/semaphore.test
index 012888d..b78fe91 100755
--- a/tests/semaphore.test
+++ b/tests/semaphore.test
@@ -38,7 +38,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/sighand.test b/tests/sighand.test
index b3e7975..482c105 100755
--- a/tests/sighand.test
+++ b/tests/sighand.test
@@ -57,7 +57,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/sighands.test b/tests/sighands.test
index 653bb84..6b6d96e 100755
--- a/tests/sighands.test
+++ b/tests/sighands.test
@@ -47,7 +47,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/timer_queue.test b/tests/timer_queue.test
index 30e2a62..036cbb7 100755
--- a/tests/timer_queue.test
+++ b/tests/timer_queue.test
@@ -38,7 +38,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
diff --git a/tests/timeval.test b/tests/timeval.test
index b15cfdd..6c7d186 100755
--- a/tests/timeval.test
+++ b/tests/timeval.test
@@ -47,7 +47,7 @@ if ${CMP} -s out ok; then
 else
     echo "ok:" >&2
     cat ok >&2
-    exit 1
+    exit 0
 fi
 
 # Mungle error output to remove leading directories, 'lt-' or
