injecting random paths breaks random on Prefix platforms

--- configure.ac
+++ configure.ac
@@ -23,14 +23,6 @@
 AC_PROG_INSTALL
 AC_PROG_LN_S
 
-case "$host" in
-	*-*-solaris*)
-		CPPFLAGS="$CPPFLAGS -I/usr/local/include"
-		LDFLAGS="$LDFLAGS -L/usr/local/lib -R/usr/local/lib"
-		;;
-	*)	;;
-esac
-
 search_libs="$search_libs m"
 
 # Checks for libraries.
--- configure
+++ configure
@@ -4071,14 +4071,6 @@
 fi
 
 
-case "$host" in
-	*-*-solaris*)
-		CPPFLAGS="$CPPFLAGS -I/usr/local/include"
-		LDFLAGS="$LDFLAGS -L/usr/local/lib -R/usr/local/lib"
-		;;
-	*)	;;
-esac
-
 search_libs="$search_libs m"
 
 # Checks for libraries.
