Index: y4mdenoise/Makefile.am
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/y4mdenoise/Makefile.am,v
retrieving revision 1.12
retrieving revision 1.13
diff -u -r1.12 -r1.13
--- y4mdenoise/Makefile.am	1 Apr 2012 08:44:49 -0000	1.12
+++ y4mdenoise/Makefile.am	28 Nov 2013 22:39:41 -0000	1.13
@@ -4,8 +4,8 @@
 
 EXTRA_DIST = implementation.html
 
-AM_CFLAGS = -DNDEBUG -finline-functions @PROGRAM_NOPIC@
-AM_CXXFLAGS = -DNDEBUG -finline-functions @PROGRAM_NOPIC@
+AM_CFLAGS = -DNDEBUG -finline-functions
+AM_CXXFLAGS = -DNDEBUG -finline-functions
 
 INCLUDES =  -I$(top_srcdir) -I$(top_srcdir)/utils
 
Index: y4munsharp/Makefile.am
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/y4munsharp/Makefile.am,v
retrieving revision 1.2
retrieving revision 1.3
diff -u -r1.2 -r1.3
--- y4munsharp/Makefile.am	16 Jun 2005 17:09:38 -0000	1.2
+++ y4munsharp/Makefile.am	28 Nov 2013 22:39:41 -0000	1.3
@@ -4,7 +4,7 @@
 
 INCLUDES = -I $(top_srcdir)/utils -I $(top_srcdir)
 
-y4munharp_CFLAGS=@PROGRAM_NOPIC@
+y4munharp_CFLAGS=
 
 bin_PROGRAMS = y4munsharp
 
Index: y4mutils/Makefile.am
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/y4mutils/Makefile.am,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -r1.7 -r1.8
--- y4mutils/Makefile.am	1 Apr 2012 08:44:49 -0000	1.7
+++ y4mutils/Makefile.am	28 Nov 2013 22:39:41 -0000	1.8
@@ -2,7 +2,7 @@
 
 MAINTAINERCLEANFILES = Makefile.in
 
-AM_CFLAGS   = @PROGRAM_NOPIC@
+AM_CFLAGS   =
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/utils
 
 LIBMJPEGUTILS = $(top_builddir)/utils/libmjpegutils.la
Index: yuvscaler/Makefile.am
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/yuvscaler/Makefile.am,v
retrieving revision 1.5
retrieving revision 1.6
diff -u -r1.5 -r1.6
--- yuvscaler/Makefile.am	1 Apr 2012 08:44:49 -0000	1.5
+++ yuvscaler/Makefile.am	28 Nov 2013 22:39:41 -0000	1.6
@@ -18,6 +18,6 @@
 
 EXTRA_DIST = yuvscaler_implementation.txt
 
-yuvscaler_CFLAGS=@PROGRAM_NOPIC@
+yuvscaler_CFLAGS=
 yuvscaler_SOURCES = yuvscaler.c yuvscaler_resample.c yuvscaler_bicubic.c
 yuvscaler_LDADD = $(LIBMJPEGUTILS) $(LIBM_LIBS)
