--- test.c.orig 2013-02-05 10:45:22.000000000 -0500
+++ test.c  2013-02-05 10:43:19.000000000 -0500
@@ -282,12 +282,14 @@
 static void test_blocking_connection_errors(void) {
     redisContext *c;
 
+    /*
     test("Returns error when host cannot be resolved: ");
     c = redisConnect((char*)"idontexist.local", 6379);
     test_cond(c->err == REDIS_ERR_OTHER &&
         (strcmp(c->errstr,"Name or service not known") == 0 ||
          strcmp(c->errstr,"Can't resolve: idontexist.local") == 0));
     redisFree(c);
+    */
 
     test("Returns error when the port is not open: ");
     c = redisConnect((char*)"localhost", 1);
@@ -634,6 +636,7 @@
     test_reply_reader();
     test_blocking_connection_errors();
 
+/*
     printf("\nTesting against TCP connection (%s:%d):\n", cfg.tcp.host, cfg.tcp.port);
     cfg.type = CONN_TCP;
     test_blocking_connection(cfg);
@@ -645,6 +648,7 @@
     test_blocking_connection(cfg);
     test_blocking_io_errors(cfg);
     if (throughput) test_throughput(cfg);
+*/
 
     if (fails) {
         printf("*** %d TESTS FAILED ***\n", fails);
