--- wxdirect-0.90.0.1-orig/wxdirect.cabal	2012-04-19 01:05:11.000000000 +1000
+++ wxdirect-0.90.0.1/wxdirect.cabal	2012-05-19 21:12:02.612085338 +1000
@@ -66,7 +66,7 @@
   if flag(splitBase)
     build-depends:
         base       >= 4     && < 5,
-        containers >= 0.2   && < 0.5
+        containers >= 0.2   && < 0.6
   else
     build-depends:
         base       >= 3     && < 4,
--- wxdirect-0.90.0.1-orig/src/Classes.hs	2012-04-19 01:05:10.000000000 +1000
+++ wxdirect-0.90.0.1/src/Classes.hs	2012-07-14 13:58:48.072492467 +1000
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 -----------------------------------------------------------------------------------------
 {-| Module      :  Classes
     Copyright   :  (c) Daan Leijen 2003
@@ -28,7 +29,9 @@
 import Data.List( sort, sortBy )
 import qualified Data.Set as Set
 import qualified Data.Map as Map
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding ( catch )
+#endif
 import HaskellNames( haskellTypeName, isBuiltin )
 import Types
 
--- wxdirect-0.90.0.1-orig/src/CompileClasses.hs	2012-04-19 01:05:10.000000000 +1000
+++ wxdirect-0.90.0.1/src/CompileClasses.hs	2012-05-19 21:28:27.358052781 +1000
@@ -89,7 +89,7 @@
            (exportsStatic,exportsClassClasses,classCount) = exportDefs decls exportsClass []
 
            methodCount  = length decls
-           ghcoptions   = [ "{-# LANGUAGE ForeignFunctionInterface #-}"]
+           ghcoptions   = [ "{-# LANGUAGE CPP, ForeignFunctionInterface #-}"]
 
            export   = concat  [ ["module " ++ moduleRoot ++ moduleName
                                 , "    ( -- * Global" ]
@@ -103,6 +103,9 @@
                                 , "import System.IO.Unsafe( unsafePerformIO )"
                                 , "import " ++ moduleRoot ++ "WxcTypes"
                                 , "import " ++ moduleRoot ++ moduleClassTypesName
+                                , "#if (__GLASGOW_HASKELL__>=705)"
+                                , "import Foreign.C.Types(CDouble(..), CInt(..), CWchar(..))"
+                                , "#endif"
                                 , ""
                                 ]
                               ]
