--- ParaView-3.98.0-src/VTK/CMake/FindCg.cmake
+++ ParaView-3.98.0-src/VTK/CMake/FindCg.cmake
@@ -82,6 +82,7 @@
     FIND_PROGRAM( CG_COMPILER cgc
       /usr/bin
       /usr/local/bin
+      /opt/nvidia-cg-toolkit/bin
       DOC "The Cg Compiler"
       )
     GET_FILENAME_COMPONENT(CG_COMPILER_DIR "${CG_COMPILER}" PATH)
@@ -89,6 +90,7 @@
     FIND_PATH( CG_INCLUDE_PATH Cg/cg.h
       /usr/include
       /usr/local/include
+      /opt/nvidia-cg-toolkit/include
       ${CG_COMPILER_SUPER_DIR}/include
       DOC "The directory where Cg/cg.h resides"
       )
@@ -98,6 +100,8 @@
       /usr/lib
       /usr/local/lib64
       /usr/local/lib
+      /opt/nvidia-cg-toolkit/lib64
+      /opt/nvidia-cg-toolkit/lib
       ${CG_COMPILER_SUPER_DIR}/lib64
       ${CG_COMPILER_SUPER_DIR}/lib
       DOC "The Cg runtime library"
@@ -108,6 +112,8 @@
       /usr/lib
       /usr/local/lib64
       /usr/local/lib
+      /opt/nvidia-cg-toolkit/lib64
+      /opt/nvidia-cg-toolkit/lib
       ${CG_COMPILER_SUPER_DIR}/lib64
       ${CG_COMPILER_SUPER_DIR}/lib
       DOC "The Cg runtime library"
