--- makefile.in
+++ makefile.in
@@ -65,13 +65,13 @@
 	cd docs; $(MAKE) ship
 
 install: all
-	for d in $(BINDIR) $(LIBLOC) $(LIBLOC)/logolib $(LIBLOC)/helpfiles $(LIBLOC)/csls; do [ -d $$d ] || mkdir -p $$d || exit 1; done
-	cp logo $(BINDIR)/.
-	cp -f logolib/* $(LIBLOC)/logolib/.
-	cp -f helpfiles/* $(LIBLOC)/helpfiles/.
-	cp -f csls/* $(LIBLOC)/csls/.
-	(cd emacs; prefix=$(prefix) LIBLOC=$(LIBLOC) BINDIR=$(BINDIR) $(MAKE) install)
-	(cd docs; prefix=$(prefix) LIBLOC=$(LIBLOC) BINDIR=$(BINDIR) $(MAKE) install)
+	for d in $(BINDIR) $(LIBLOC) $(LIBLOC)/logolib $(LIBLOC)/helpfiles $(LIBLOC)/csls; do [ -d $(DESTDIR)$$d ] || mkdir -p $(DESTDIR)$$d || exit 1; done
+	cp logo $(DESTDIR)$(BINDIR)/.
+	cp -f logolib/* $(DESTDIR)$(LIBLOC)/logolib/.
+	cp -f helpfiles/* $(DESTDIR)$(LIBLOC)/helpfiles/.
+	cp -f csls/* $(DESTDIR)$(LIBLOC)/csls/.
+	(cd emacs; prefix=$(prefix) LIBLOC=$(LIBLOC) BINDIR=$(BINDIR) DESTDIR=$(DESTDIR) $(MAKE) install)
+	(cd docs; prefix=$(prefix) LIBLOC=$(LIBLOC) BINDIR=$(BINDIR) DESTDIR=$(DESTDIR) $(MAKE) install)
 #	prefix=$(prefix); LIBLOC=$(LIBLOC); BINDIR=$(BINDIR); export prefix LIBLOC BINDIR; cd emacs; $(MAKE) install
 
 logo-mode: 
--- docs/makefile
+++ docs/makefile
@@ -30,10 +30,10 @@
 	-rm -f *.{aux,cp,cps,dvi,fn,ky,log,pg,toc,tp,vr}
 
 install: all
-	for d in $(INFODIR) $(DOCSDIR) $(HTMLDIR); do [ -d $$d ] || mkdir -p $$d || exit 1; done
-	-cp -f *.info* $(INFODIR)/.
-	-cp -f html/*.html $(HTMLDIR)/.
-	-cp -f usermanual.ps $(DOCSDIR)/.
-	-cp -f usermanual.pdf $(DOCSDIR)/.
-	-cp -f usermanual.texi $(DOCSDIR)/.
-	-cp -f ../usermanual $(DOCSDIR)/.
+	for d in $(INFODIR) $(DOCSDIR) $(HTMLDIR); do [ -d $(DESTDIR)$$d ] || mkdir -p $(DESTDIR)$$d || exit 1; done
+	-cp -f *.info* $(DESTDIR)$(INFODIR)/.
+	-cp -f html/*.html $(DESTDIR)$(HTMLDIR)/.
+	-cp -f usermanual.ps $(DESTDIR)$(DOCSDIR)/.
+	-cp -f usermanual.pdf $(DESTDIR)$(DOCSDIR)/.
+	-cp -f usermanual.texi $(DESTDIR)$(DOCSDIR)/.
+	-cp -f ../usermanual $(DESTDIR)$(DOCSDIR)/.
--- emacs/makefile
+++ emacs/makefile
@@ -40,16 +40,16 @@
 	-rm -f add.user install-logo-mode dot.emacs dot.loops *.elc
 
 install: all
-	for d in $(EMACSDIR); do [ -d $$d ] || mkdir -p $$d || exit 1; done
-	cp -f logo.* $(EMACSDIR)/.
-	cp -f letrec.* $(EMACSDIR)/.
-	cp -f comint*.* $(EMACSDIR)/.
-	cp -f tutor* $(EMACSDIR)/.
-	cp -f dot.loops $(EMACSDIR)/.
-	cp -f dot.logo $(EMACSDIR)/.logo
-	cp -f dot.LOOPS $(EMACSDIR)/.LOOPS
-	cp -f dot.emacs $(EMACSDIR)/.
-	cp -f README $(EMACSDIR)/.
-	cp -p install-logo-mode $(BINDIR)/.
-	cp -f edfunc.el $(EMACSDIR)/.
-	cp -f check.lg $(EMACSDIR)/.
+	for d in $(EMACSDIR); do [ -d $(DESTDIR)$$d ] || mkdir -p $(DESTDIR)$$d || exit 1; done
+	cp -f logo.* $(DESTDIR)$(EMACSDIR)/.
+	cp -f letrec.* $(DESTDIR)$(EMACSDIR)/.
+	cp -f comint*.* $(DESTDIR)$(EMACSDIR)/.
+	cp -f tutor* $(DESTDIR)$(EMACSDIR)/.
+	cp -f dot.loops $(DESTDIR)$(EMACSDIR)/.
+	cp -f dot.logo $(DESTDIR)$(EMACSDIR)/.logo
+	cp -f dot.LOOPS $(DESTDIR)$(EMACSDIR)/.LOOPS
+	cp -f dot.emacs $(DESTDIR)$(EMACSDIR)/.
+	cp -f README $(DESTDIR)$(EMACSDIR)/.
+	cp -p install-logo-mode $(DESTDIR)$(BINDIR)/.
+	cp -f edfunc.el $(DESTDIR)$(EMACSDIR)/.
+	cp -f check.lg $(DESTDIR)$(EMACSDIR)/.
