--- ixp400_xscale_sw/src/npeMh/IxNpeMhSymbols.c
+++ ixp400_xscale_sw/src/npeMh/IxNpeMhSymbols.c
@@ -63,7 +63,6 @@ EXPORT_SYMBOL(ixNpeMhMessagesReceive);
 EXPORT_SYMBOL(ixNpeMhShow);
 EXPORT_SYMBOL(ixNpeMhShowReset);
 
-extern void ixNpeMhConfigIsr (void *parameter);
 extern BOOL ixNpeMhConfigInFifoIsFull(IxNpeMhNpeId npeId);
 extern BOOL ixNpeMhConfigOutFifoIsEmpty (IxNpeMhNpeId npeId);
 extern void ixNpeMhConfigLockRelease (IxNpeMhNpeId npeId);
@@ -72,7 +71,6 @@ extern void ixNpeMhConfigOutFifoRead (Ix
 extern void ixNpeMhConfigInFifoWrite (IxNpeMhNpeId npeId,IxNpeMhMessage message);
 extern struct ixNpeMhConfigNpeInfo ixNpeMhConfigNpeInfo[IX_NPEMH_NUM_NPES];
 
-EXPORT_SYMBOL(ixNpeMhConfigIsr);
 EXPORT_SYMBOL(ixNpeMhConfigInFifoIsFull);
 EXPORT_SYMBOL(ixNpeMhConfigOutFifoIsEmpty);
 EXPORT_SYMBOL(ixNpeMhConfigLockRelease);
