--- ixp400_xscale_sw/src/oslinux/IxLinuxInit.c
+++ ixp400_xscale_sw/src/oslinux/IxLinuxInit.c
@@ -160,7 +160,7 @@ int ixNpe_dev_open (struct inode *inode,
 	return -EEXIST;
     }
 
-    MOD_INC_USE_COUNT;   /* Increment use count to prevent premature rmmod-ing */
+    /* MOD_INC_USE_COUNT; */  /* Increment use count to prevent premature rmmod-ing */
     ixNpeDlSegmentedListHead = NULL;  /* Reset linked list which keeps track of Microcode fragments */
     ixNpeDlSegmentedListTail = NULL;
     ixNpeDlTotalBytesReadIn = 0;
@@ -225,7 +225,7 @@ int ixNpe_dev_release (struct inode *ino
 	kfree (temp);
     }
 
-    MOD_DEC_USE_COUNT;
+    /* MOD_DEC_USE_COUNT; */
 
     return 0;
 }
