--- ixp400_xscale_sw/Makefile.ixp400_eth
+++ ixp400_xscale_sw/Makefile.ixp400_eth
@@ -0,0 +1,32 @@
+obj-m := ixp400_eth.o
+
+PWD := $(shell pwd)
+
+CFLAGS =  -Wall \
+          -DCONFIG_MODULES -DMODULES -O -I$(LINUX_SRC)/include/ \
+          -I$(IX_XSCALE_SW)/src/include/ \
+          -I$(OSAL_DIR)/ \
+          -I$(OSAL_DIR)/os/linux/include/ \
+          -I$(OSAL_DIR)/os/linux/include/modules/ \
+          -I$(OSAL_DIR)/os/linux/include/modules/ioMem/ \
+          -I$(OSAL_DIR)/os/linux/include/modules/core/ \
+          -I$(OSAL_DIR)/os/linux/include/modules/bufferMgt/ \
+          -I$(OSAL_DIR)/os/linux/include/core/  \
+          -I$(OSAL_DIR)/os/linux/include/platforms/ \
+          -I$(OSAL_DIR)/os/linux/include/platforms/ixp400/ \
+          -I$(OSAL_DIR)/os/linux/include/core/ \
+          -I$(OSAL_DIR)/include/ \
+          -I$(OSAL_DIR)/include/modules/ \
+          -I$(OSAL_DIR)/include/modules/bufferMgt/ \
+          -I$(OSAL_DIR)/include/modules/ioMem/ \
+          -I$(OSAL_DIR)/include/modules/core/ \
+          -I$(OSAL_DIR)/include/platforms/ \
+          -I$(OSAL_DIR)/include/platforms/ixp400/ \
+	  -I$(OSAL_DIR)/os/linux/include/platforms/ixp400/ixp425/
+export CFLAGS
+
+ixp400_eth.ko:
+	$(MAKE) V=1 -C $(LINUX_SRC) SUBDIRS=$(PWD) ARCH=arm modules
+
+clean:
+	rm -f ixp400_eth.ko
