Description: Fix ftbfs
Author: Vincent Legout <vincent@legout.info>
Bug-Debian: http://bugs.debian.org/624924

Index: assogiate/src/type-dialog.cc
===================================================================
--- assogiate.orig/src/type-dialog.cc	2011-05-09 16:51:09.951622099 +0200
+++ assogiate/src/type-dialog.cc	2011-05-09 16:51:53.293459257 +0200
@@ -144,7 +144,7 @@
 	general_upper->pack_start(*Gtk::manage(general_info), true, true);
 
 	if (renameable) {
-		Gtk::ComboBox *cb = new Gtk::ComboBox(CategoriesStore::get_selector());
+		Gtk::ComboBox *cb = new Gtk::ComboBox(CategoriesStore::get_selector(), false);
 		cb->signal_changed().connect
 			(sigc::mem_fun(*this, &TypeDialog::on_category_changed));
 		m_category_widget = cb;
@@ -622,7 +622,7 @@
 	} else {
 		AddItemDialog adder(this, _("Add File Contents Rule"));
 
-		Gtk::ComboBox type(MagicTypesStore::get());
+		Gtk::ComboBox type(MagicTypesStore::get(), false);
 		type.pack_start(MagicTypesColumns::get().name);
 		adder.add_item(_("_Type:"), type, true);
 
