diff -r 7f19e7f2afa2 lib/stddef.in.h
--- a/libgnu/stddef.in.h	Sat Feb 21 06:19:32 2015 -0800
+++ b/libgnu/stddef.in.h	Fri May 01 06:01:06 2015 +0200
@@ -82,6 +82,7 @@
 #endif
 
 /* Some platforms lack max_align_t.  */
+#ifndef _GCC_MAX_ALIGN_T
 #if !@HAVE_MAX_ALIGN_T@
 typedef union
 {
@@ -91,6 +92,7 @@
   long int __i;
 } max_align_t;
 #endif
+#endif
 
 #  endif /* _@GUARD_PREFIX@_STDDEF_H */
 # endif /* _@GUARD_PREFIX@_STDDEF_H */
