diff --git a/tests/ext/aiohttp/test_middleware.py b/tests/ext/aiohttp/test_middleware.py
index c8b2333..4f8cac6 100644
--- a/tests/ext/aiohttp/test_middleware.py
+++ b/tests/ext/aiohttp/test_middleware.py
@@ -279,8 +279,7 @@ async def test_concurrent(test_client, loop, recorder):
 
     await asyncio.wait([get_delay(), get_delay(), get_delay(),
                         get_delay(), get_delay(), get_delay(),
-                        get_delay(), get_delay(), get_delay()],
-                       loop=loop)
+                        get_delay(), get_delay(), get_delay()])
 
     # Ensure all ID's are different
     ids = [item.id for item in recorder.emitter.local]
diff --git a/tests/test_async_local_storage.py b/tests/test_async_local_storage.py
index b43cc0e..4b13ffd 100644
--- a/tests/test_async_local_storage.py
+++ b/tests/test_async_local_storage.py
@@ -19,7 +19,7 @@ def test_localstorage_isolation(loop):
             random_int = random.random()
             local_storage.randint = random_int
 
-            await asyncio.sleep(0.0, loop=loop)
+            await asyncio.sleep(0.0)
 
             current_random_int = local_storage.randint
             assert random_int == current_random_int
@@ -30,7 +30,7 @@ def test_localstorage_isolation(loop):
 
     # Run loads of concurrent tasks
     results = loop.run_until_complete(
-        asyncio.wait([_test() for _ in range(0, 100)], loop=loop)
+        asyncio.wait([_test() for _ in range(0, 100)])
     )
     results = [item.result() for item in results[0]]
 
