diff --git a/wpa_supplicant/dbus/dbus_new_helpers.c b/wpa_supplicant/dbus/dbus_new_helpers.c
index 45623f3..0fc3d08 100644
--- a/wpa_supplicant/dbus/dbus_new_helpers.c
+++ b/wpa_supplicant/dbus/dbus_new_helpers.c
@@ -847,7 +847,7 @@ void wpa_dbus_mark_property_changed(struct wpas_dbus_priv *iface,
 	const struct wpa_dbus_property_desc *dsc;
 	int i = 0;
 
-	if (iface == NULL)
+	if (iface == NULL || path == NULL)
 		return;
 
 	dbus_connection_get_object_path_data(iface->con, path,
