From: Sam James <sam@gentoo.org>
Date: Sun, 6 Oct 2024 09:47:03 +0100
Subject: [PATCH] Forward ported from qemu-7.1.0-capstone-include-path.patch.

Bug: https://bugs.gentoo.org/873157
Signed-off-by: Sam James <sam@gentoo.org>
Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
---
 include/disas/capstone.h | 2 +-
 meson.build              | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/disas/capstone.h b/include/disas/capstone.h
index c43033f7f6..02bc067cd8 100644
--- a/include/disas/capstone.h
+++ b/include/disas/capstone.h
@@ -5,7 +5,7 @@
 
 #define CAPSTONE_AARCH64_COMPAT_HEADER
 #define CAPSTONE_SYSTEMZ_COMPAT_HEADER
-#include <capstone.h>
+#include <capstone/capstone.h>
 
 #else
 
diff --git a/meson.build b/meson.build
index 2c924f8f10..86858a325b 100644
--- a/meson.build
+++ b/meson.build
@@ -1892,7 +1892,7 @@ if not get_option('capstone').auto() or have_system or have_user
   # that reports a wrong -I path, causing the #include to
   # fail later. If the system has such a broken version
   # do not use it.
-  if capstone.found() and not cc.compiles('#include <capstone.h>',
+  if capstone.found() and not cc.compiles('#include <capstone/capstone.h>',
                                           dependencies: [capstone])
     capstone = not_found
     if get_option('capstone').enabled()
-- 
2.45.2

