From: Ole Streicher <debian@liska.ath.cx>
Date: Fri, 5 Aug 2016 16:17:42 +0200
Subject: Fix several bugs that appear in during the mayhem tests.

---
 cphead.c | 2 +-
 getpix.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/cphead.c b/cphead.c
index 0c025dd..2b34f12 100644
--- a/cphead.c
+++ b/cphead.c
@@ -148,7 +148,7 @@ char **av;
 			free (kwd);
 			kwd = kwdnew;
 			}
-		    for (ikwd = nkwd; i < nkwd+nkwd1+32; i++) {
+		    for (ikwd = nkwd; ikwd < nkwd+nkwd1+32; ikwd++) {
 			kwd[ikwd] = (char *) calloc (32, 1);
 			}
 		    strcpy (kwd[nkwd], "RA");
diff --git a/getpix.c b/getpix.c
index ccb38f0..64cc571 100644
--- a/getpix.c
+++ b/getpix.c
@@ -83,6 +83,7 @@ char **av;
     char *crange;       /* Column range string */
     char *rstr;
     char *dstr = NULL;
+    char rastr[32], decstr[32];
     char *cstr;
     int systemp;
     int i;
