From 69660180a59d0d100097dedbc507d4af73267e8d Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Wed, 30 Nov 2011 12:50:53 -0500
Subject: [PATCH] Make staticPath settable from configforced

---
 config.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/config.py b/config.py
index e39e0d3..e028c36 100644
--- a/config.py
+++ b/config.py
@@ -66,7 +66,9 @@ def defPaths():
 
     # Static EVE Data from the staticdata repository, should be in the staticdata
     # directory in our pyfa directory
-    staticPath = os.path.join(pyfaPath, "staticdata")
+    staticPath = getattr(configforced, "staticPath", None)
+    if staticPath is None:
+        staticPath = os.path.join(pyfaPath, "staticdata")
 
     # The database where we store all the fits etc
     saveDB = os.path.join(savePath, "saveddata.db")
-- 
1.8.5.3

