From d5fa475189d708f6e43d2aabac66424a6724bb39 Mon Sep 17 00:00:00 2001
From: Conrad Kostecki <conikost@gentoo.org>
Date: Sat, 19 Mar 2022 19:20:48 +0100
Subject: [PATCH] src/lualdap.c: fix lua5-4 support

Signed-off-by: Conrad Kostecki <conikost@gentoo.org>
---
 src/lualdap.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/lualdap.c b/src/lualdap.c
index 2ad0e6d..1ced88b 100644
--- a/src/lualdap.c
+++ b/src/lualdap.c
@@ -3,6 +3,8 @@
 ** See Copyright Notice in license.md
 */
 
+#define _GNU_SOURCE
+
 #include <string.h>
 
 #ifdef WIN32
