--- a/src/Credits.h
+++ b/src/Credits.h
@@ -47,7 +47,7 @@
 	int   speed;
 	bool  stopped;
  private:
-	static const float yoffset_start = 50;
+	static const float yoffset_start;
 	static const int normal_speed = 40;
 	static const int text_height = 40;
 	
--- a/src/Credits.cpp
+++ b/src/Credits.cpp
@@ -273,3 +273,5 @@
       (*draw_list_iter)->show();
     }
 }
+
+const float Credits::yoffset_start = 50;
