--- a/src/test/meson.build	2018-01-15 20:34:18.000000000 +0100
+++ b/src/test/meson.build	2018-01-31 15:11:33.342989914 +0100
@@ -868,10 +868,10 @@
 #          [],
 #          [threads]],
 #endif // 0
-
-        [['src/libelogind/sd-login/test-login.c'],
-         [],
-         []],
+# Broken in chroot; needs elogind at runtime?
+#        [['src/libelogind/sd-login/test-login.c'],
+#         [],
+#         []],
 ]
 
 #if 0 /// UNNEEDED in elogind
