--- a/meson.build	2018-04-20 17:19:23.912940283 +0200
+++ b/meson.build	2018-04-20 17:20:02.265941434 +0200
@@ -2800,20 +2800,7 @@
 # install_data('modprobe.d/systemd.conf',
 #              install_dir : modprobedir)
 #endif // 0
-install_data('README',
-             'NEWS',
-             'CODING_STYLE',
-#if 0 /// UNNEEDED by elogind
-#              'DISTRO_PORTING',
-#              'ENVIRONMENT.md',
-#endif // 0
-             'LICENSE.GPL2',
-             'LICENSE.LGPL2.1',
-             'src/libelogind/sd-bus/GVARIANT-SERIALIZATION',
-#if 0 /// UNNEEDED by elogind
-#              'TRANSIENT-SETTINGS.md',
-#              'UIDS-GIDS.md',
-#endif // 0
+install_data('src/libelogind/sd-bus/GVARIANT-SERIALIZATION',
              install_dir : docdir)
 
 #if 0 /// UNNEEDED by elogind
