--- a/src/Makefile	2022-01-26 12:50:57.000000000 +0900
+++ b/src/Makefile	2025-02-06 21:35:18.679591028 +0900
@@ -52,21 +52,18 @@
 WINMAKE=mingw32-make
 
 LDLIBS     = -lm
-AR         = ar
-CC         = gcc
-CXX        = g++
-O_OPTS     = -O2
-CC_OPTS    = -g -Wall
-P_DEFINES  =
-
-CFLAGS  = $(CC_OPTS) $(O_OPTS)
-CXXFLAGS = $(CFLAGS) -std=c++11
-LDFLAGS = -g
+O_OPTS     ?= -O2
+CC_OPTS    ?= -g -Wall
+P_DEFINES  ?=
+
+CFLAGS		+= $(CC_OPTS) $(O_OPTS)
+CXXFLAGS	+= $(CC_OPTS) $(O_OPTS) -std=c++11
+LDFLAGS		+=
 # Note, for profiling, use
 # make O_OPTS='-pg -O0' LDFLAGS='-g -pg'
 
 # Target installation directory
-PREFIX     = /usr/local
+# PREFIX     = /usr/local
 
 # ======================================================================
 # If you have trouble with library skew when moving primer3 executables
@@ -114,8 +111,6 @@
     DYNLIBS         = $(LIBPRIMER3_DYN) $(LIBDPAL_DYN) $(LIBTHAL_DYN) $(LIBTHALPARA_DYN) $(LIBAMPTM_DYN) $(LIBOLIGOTM_DYN) $(LIBMASKER_DYN)
 endif
 
-RANLIB          = ranlib
-
 PRIMER_OBJECTS1=primer3_boulder_main.o\
                 format_output.o\
                 read_boulder.o\
@@ -145,57 +140,50 @@
 ifeq ($(TESTOPTS),--windows)
 	cd ../test & $(WINMAKE) clean TESTOPTS=$(TESTOPTS)
 else
-	cd ../test/; make clean
+	cd ../test/; $(MAKE) clean
 endif
 
 $(LIBOLIGOTM): oligotm.o
-	$(AR) rv $@ oligotm.o
-	$(RANLIB) $@
+	$(AR) rcs $@ oligotm.o
 
 $(LIBOLIGOTM_LIB): oligotm.o
-	$(CC) -shared -W1,-soname,liboligotm.so.1 -o $(LIBOLIGOTM_DYN) oligotm.o
+	$(CC) $(LDFLAGS) -shared -W1,-soname,liboligotm.so.1 -o $(LIBOLIGOTM_DYN) oligotm.o
 
 $(LIBAMPTM): amplicontm.o
-	$(AR) rv $@ amplicontm.o
-	$(RANLIB) $@
+	$(AR) rcs $@ amplicontm.o
 
 $(LIBAMPTM_LIB): amplicontm.o
-	$(CC) -shared -W1,-soname,libamplicontm.so.1 -o $(LIBAMPTM_DYN) amplicontm.o
+	$(CC) $(LDFLAGS) -shared -W1,-soname,libamplicontm.so.1 -o $(LIBAMPTM_DYN) amplicontm.o
 
 $(LIBDPAL): dpal_primer.o
-	$(AR) rv $@ dpal_primer.o
-	$(RANLIB) $@
+	$(AR) rcs $@ dpal_primer.o
 
 $(LIBDPAL_DYN): dpal_primer.o
-	$(CC) -shared -W1,-soname,libdpal.so.1 -o $(LIBDPAL_DYN_LIB) dpal_primer.o
+	$(CC) $(LDFLAGS) -shared -W1,-soname,libdpal.so.1 -o $(LIBDPAL_DYN_LIB) dpal_primer.o
 
 $(LIBTHAL): thal_primer.o
-	$(AR) rv $@ thal_primer.o
-	$(RANLIB) $@
+	$(AR) rcs $@ thal_primer.o
 
 $(LIBTHAL_DYN): thal_primer.o
-	$(CC) -shared -W1,-soname,libthal.so.1 -o $(LIBTHAL_DYN_LIB) thal_primer.o
+	$(CC) $(LDFLAGS) -shared -W1,-soname,libthal.so.1 -o $(LIBTHAL_DYN_LIB) thal_primer.o
 
 $(LIBTHALPARA): thal_parameters.o
-	$(AR) rv $@ thal_parameters.o
-	$(RANLIB) $@
+	$(AR) rcs $@ thal_parameters.o
 
 $(LIBTHALPARA_DYN): thal_parameters.o
-	$(CC) -shared -W1,-soname,libthalpara.so.1 -o $(LIBTHALPARA_DYN_LIB) thal_parameters.o
+	$(CC) $(LDFLAGS) -shared -W1,-soname,libthalpara.so.1 -o $(LIBTHALPARA_DYN_LIB) thal_parameters.o
 
 $(LIBPRIMER3): libprimer3.o p3_seq_lib.o
-	$(AR) rv $@ libprimer3.o p3_seq_lib.o
-	$(RANLIB) $@
+	$(AR) rcs $@ libprimer3.o p3_seq_lib.o
 
 $(LIBPRIMER3_DYN): libprimer3.o p3_seq_lib.o
-	$(CC) -shared -W1,-soname,liprimer3.so.1 -o $(LIBPRIMER3_DYN) libprimer3.o p3_seq_lib.o
+	$(CC) $(LDFLAGS) -shared -W1,-soname,liprimer3.so.1 -o $(LIBPRIMER3_DYN) libprimer3.o p3_seq_lib.o
 
 $(LIBMASKER): masker.o
-	$(AR) rv $@ masker.o
-	$(RANLIB) $@
+	$(AR) rcs $@ masker.o
 
 $(LIBMASKER_DYN): masker.o
-	$(CC) -shared -W1,-soname,libmasker.so.1 -o $(LIBMASKER_DYN) masker.o
+	$(CC) $(LDFLAGS) -shared -W1,-soname,libmasker.so.1 -o $(LIBMASKER_DYN) masker.o
 
 $(PRIMER_EXE): $(PRIMER_OBJECTS)
 	$(CXX) $(CXXFLAGS) -lstdc++ $(LDFLAGS) -o $@ $(PRIMER_OBJECTS) $(LIBOPTS) $(LDLIBS)
@@ -274,7 +262,7 @@
 ifeq ($(TESTOPTS),--windows)
 	cd ../test & $(WINMAKE) TESTOPTS=$(TESTOPTS)
 else
-	cd ../test; make test
+	cd ../test; $(MAKE) test
 endif
 
 install: $(EXES) $(LIBRARIES)
--- a/test/Makefile	2022-01-26 12:50:57.000000000 +0900
+++ b/test/Makefile	2025-02-06 21:35:54.048433100 +0900
@@ -116,7 +116,7 @@
 ifeq ($(TESTOPTS),--windows)
 	cd ../src & $(WINMAKE)
 else
-	cd ../src; make
+	cd ../src; $(MAKE)
 endif
 
 clean:
