diff --git a/virtManager/connect.py b/virtManager/connect.py
index f7f268e..c19c224 100644
--- a/virtManager/connect.py
+++ b/virtManager/connect.py
@@ -95,7 +95,10 @@ class vmmConnect(vmmGObjectUI):
             self.avahiserver = Gio.DBusProxy.new_sync(self.dbus, 0, None,
                                     "org.freedesktop.Avahi", "/",
                                     "org.freedesktop.Avahi.Server", None)
+            self.avahiserver.GetVersionString()
+            logging.debug("self.avahiserver=%s", self.avahiserver)
         except Exception, e:
+            self.avahiserver = None
             logging.debug("Couldn't contact avahi: %s", str(e))
 
         self.reset_state()
