--- Looptools/Makefile.am
+++ Looptools/Makefile.am
@@ -1,58 +1,12 @@
-noinst_LTLIBRARIES = libHwLooptoolsXFC.la libHwLooptoolsCFC.la 
-noinst_LTLIBRARIES += libHwLooptools.la
-
-EXTRA_DIST = include/lt.h include/ftypes.h include/ff.h include/ffwarn.h \
-  include/fferr.h include/defs.h include/looptools.h include/ffperm5.h \
-  include/clooptools.h include/cexternals.h include/externals.h \
-  util/solve-LU.F util/solve-LU.h util/solve-Eigen.F
-
-theXFCSOURCES = \
-A/A0.F            A/A00.F           A/ffxa0.F         A/ffca0.F         B/Bget.F \
-B/Bcoeff.F        B/BcoeffC.F       B/Bcoeffa.F       B/Bcoeffb.F       B/ffxb0.F \
-B/ffcb0.F         B/ffxb1.F         B/ffcb1.F         B/ffxb2p.F        B/ffcb2p.F \
-B/ffxdb0.F        B/ffcdb0.F        B/ffxdb1.F        B/ffdel2.F        B/ffcel2.F \
-C/C0.F            C/C0C.F           C/Cget.F          C/ffxc0.F         C/ffcc0.F \
-C/ffxc0i.F        C/ffxc0p.F        C/ffxc0p0.F       C/ffcc0p.F        C/ffdxc0.F \
-C/ffdel3.F        C/ffcel3.F        D/D0.F            D/D0C.F           D/Dget.F \
-D/ffxd0.F         D/ffxd0h.F        D/ffxd0i.F        D/ffxd0p.F        D/ffxd0m0.F \
-D/ffxd0tra.F      D/ffxdbd.F        D/ffdcc0.F        D/ffdel4.F        D/ffd0c.F \
-D/ffTn.F          D/ffT13.F         D/ffS2.F          D/ffS3n.F         D/ffRn.F \
-E/E0.F            E/Eget.F          E/Ecoeffa.F       E/Ecoeffb.F       E/ffxe0.F \
-E/ffdel5.F    util/ini.F    util/auxCD.F   util/solve.F  util/Dump.F \
-util/Li2.F    util/ffinit.F util/ffxli2.F  util/ffcli2.F util/ffxxyz.F \
-util/ffcxyz.F util/ffcrr.F  util/ffcxr.F   util/fftran.F util/ffabcd.F \
-util/ff2dl2.F util/ffcxs3.F util/ffcxs4.F  util/ffdcxs.F util/ffbndc.F
-
-## the following need -DCOMPLEXPARA
-theCFCSOURCES = \
-A/A0.F       A/A00.F      B/Bget.F     B/Bcoeffa.F  C/Cget.F     D/Dget.F     E/E0.F \
-E/Eget.F     E/Ecoeffa.F  E/Ecoeffb.F  util/solve.F    util/Dump.F     util/Li2.F
+noinst_LTLIBRARIES = libHwLooptools.la
 
 commonCPPFLAGS = $(AM_CPPFLAGS) -I$(srcdir)/include
 
-libHwLooptoolsXFC_la_SOURCES = $(theXFCSOURCES)
-libHwLooptoolsXFC_la_CPPFLAGS = $(commonCPPFLAGS)
-libHwLooptoolsXFC_la_FCFLAGS = $(AM_FCFLAGS) -ffixed-line-length-none
-libHwLooptoolsXFC_la_FFLAGS = $(AM_FFLAGS) -ffixed-line-length-none
-
-libHwLooptoolsCFC_la_SOURCES = $(theCFCSOURCES)
-libHwLooptoolsCFC_la_CPPFLAGS = $(commonCPPFLAGS) -DCOMPLEXPARA
-libHwLooptoolsCFC_la_FCFLAGS = $(AM_FCFLAGS) -ffixed-line-length-none
-libHwLooptoolsCFC_la_FFLAGS = $(AM_FFLAGS) -ffixed-line-length-none
-
-## installed already from "include/Makefile.am"
-## noinst_HEADERS = include/clooptools.h
-
-##if NEED_APPLE_FIXES
-##libHwLooptoolsXFC_la_LDFLAGS = -Wl,-single_module
-##libHwLooptoolsCFC_la_LDFLAGS = -Wl,-single_module
-##endif
-
-libHwLooptools_la_SOURCES = util/cache.c clooptools.cc
+libHwLooptools_la_SOURCES = clooptools.cc
 libHwLooptools_la_CXXFLAGS = $(AM_CXXFLAGS) -Wno-strict-aliasing
 libHwLooptools_la_CPPFLAGS = $(commonCPPFLAGS) 
 libHwLooptools_la_CFLAGS = $(AM_CFLAGS) -std=c99 
-libHwLooptools_la_LIBADD = libHwLooptoolsXFC.la libHwLooptoolsCFC.la
+libHwLooptools_la_LIBADD = -looptools
 
 
 ## libHwLooptoolsHelper2_la_SOURCES = clooptools.cc
