From: Julian Ospald <hasufell@gentoo.org>
Date: Thu Mar 27 21:16:02 UTC 2014
Subject: unbundle miniupnpc

--- a/server/CMakeLists.txt
+++ b/server/CMakeLists.txt
@@ -29,8 +29,7 @@
 set(JSONCPP_SOURCE ${JSONCPP_DIR}/jsoncpp.cpp)
 
 # MiniUPnPc
-set(MINIUPNPC_DIR ../libraries/libminiupnpc)
-set(MINIUPNPC_STATIC_LIBRARIES upnpc-static)
+set(MINIUPNPC_DIR /usr/include/miniupnpc)
 
 # Platform definitions
 define_platform()
@@ -54,7 +53,7 @@
   ${COMMON_SOURCES} ${COMMON_HEADERS}
   ${SERVER_SOURCES} ${SERVER_HEADERS}
   ${SERVER_WIN32_HEADERS} ${SERVER_WIN32_RESOURCES})
-target_link_libraries(odasrv ${MINIUPNPC_STATIC_LIBRARIES})
+target_link_libraries(odasrv -lminiupnpc)
 
 if(WIN32)
   target_link_libraries(odasrv winmm wsock32)
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,7 +71,6 @@
 set(UPNPC_BUILD_TESTS OFF CACHE INTERNAL "" FORCE)
 set(UPNPC_INSTALL OFF CACHE INTERNAL "" FORCE)
 mark_as_advanced(FORCE UPNPC_INSTALL)
-add_subdirectory(libraries/libminiupnpc)
 
 # PortMidi
 cmake_dependent_option(USE_INTREE_PORTMIDI "Compile with the version of PortMidi included in the source tree." ON "WIN32" OFF)
