--- a/src/makefile	2016-08-11 10:40:49.466873499 +0200
+++ b/src/makefile	2016-08-11 10:40:59.626762850 +0200
@@ -32,7 +32,7 @@
 # Fallback to gcc when $CC is not in $PATH.
 CC:=$(shell sh -c 'type $(CC) >/dev/null 2>/dev/null && echo $(CC) || echo gcc')
 OPTIMIZATION?=-O2
-WARNINGS=-Wall -Werror -Wstrict-prototypes -fwrapv
+WARNINGS=-Wall -Wstrict-prototypes -fwrapv
 DEBUG?= -g -ggdb
 REAL_CFLAGS=$(OPTIMIZATION) -fPIC -pthread $(CFLAGS) $(WARNINGS) $(DEBUG)
 REAL_LDFLAGS=$(LDFLAGS) -pthread
