--- Makefile.orig	2003-09-04 13:00:48.784055072 -0400
+++ Makefile	2003-09-04 13:04:23.297444080 -0400
@@ -11,11 +11,11 @@
 #OPT	= -pg -g -DPRO -DACPI
 
 CC	= gcc
-CFLAGS	= $(OPT) -Wall -ansi -I/usr/X11R6/include
+CFLAGS	= $(OPT) -Wall -ansi -I/usr/X11R6/include -DAPM
 LDFLAGS = $(OPT) -L/usr/X11R6/lib -lX11 -lXpm -lXext
 
-SRCS	= wmacpi.c libapm.c libacpi.c
-OBJS	= wmacpi.o libapm.o libacpi.o
+SRCS	= wmacpi.c libapm.c
+OBJS	= wmacpi.o libapm.o
 
 all:	wmacpi
 
