From 79025592e95e110a275b92b35b3b6c0ea9996242 Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentnl@gentoo.org>
Date: Wed, 3 Jan 2018 20:02:47 +1300
Subject: Fix for Perl 5.26 without . in INC

---
 Makefile.PL | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.PL b/Makefile.PL
index 00eb501..fd7226d 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -3,6 +3,7 @@
 use strict;
 use warnings;
 
+use lib q[.];
 use inc::Module::Install 0.91;
 
 use Module::Install::AuthorRequires 0.02;
-- 
2.15.1

