From 43fa54df3f7a5d396e3234a7db1099a64356befc Mon Sep 17 00:00:00 2001
From: sin-ack <sin-ack@protonmail.com>
Date: Mon, 23 Mar 2026 01:47:38 +0100
Subject: [PATCH] CMake: Update cmake_minimum_required to 3.10

Bug: https://bugs.gentoo.org/958965
Bug: https://bugs.gentoo.org/965372
---
 wangle/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wangle/CMakeLists.txt b/wangle/CMakeLists.txt
index 337713b..ebc458d 100644
--- a/wangle/CMakeLists.txt
+++ b/wangle/CMakeLists.txt
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.

-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.10)

 if (NOT DEFINED PACKAGE_VERSION)
   set(PACKAGE_VERSION "1.0.0")
--
2.52.0
