diff --git a/Makefile b/Makefile
index 3b6dcb5..8981549 100644
--- a/Makefile
+++ b/Makefile
@@ -20,8 +20,6 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
-GOFILES ?= $(shell find . -type f -name '*.go' -not -path "./vendor/*")
-
 # Set version variables for LDFLAGS
 PROJECT_ID ?= projectsigstore
 RUNTIME_IMAGE ?= gcr.io/distroless/static
@@ -72,18 +70,6 @@ log-%:
 				printf "\033[36m==> %s\033[0m\n", $$2 \
 			}'
 
-.PHONY: checkfmt
-checkfmt: SHELL := /usr/bin/env bash
-checkfmt: ## Check formatting of all go files
-	@ $(MAKE) --no-print-directory log-$@
- 	$(shell test -z "$(shell gofmt -l $(GOFILES) | tee /dev/stderr)")
- 	$(shell test -z "$(shell goimports -l $(GOFILES) | tee /dev/stderr)")
-
-.PHONY: fmt
-fmt: ## Format all go files
-	@ $(MAKE) --no-print-directory log-$@
-	goimports -w $(GOFILES)
-
 cosign: $(SRCS)
 	CGO_ENABLED=0 go build -trimpath -ldflags "$(LDFLAGS)" -o $@ ./cmd/cosign
 
@@ -121,7 +107,6 @@ clean:
 	rm -rf sget
 	rm -rf dist/
 
-
 KOCACHE_PATH=/tmp/ko
 ARTIFACT_HUB_LABELS=--image-label io.artifacthub.package.readme-url="https://raw.githubusercontent.com/sigstore/cosign/main/README.md" \
                     --image-label io.artifacthub.package.logo-url=https://raw.githubusercontent.com/sigstore/cosign/main/images/logo.svg \
