--- plug-ins/cairo/Makefile.am.orig	2014-05-06 14:09:36.080876942 -0400
+++ plug-ins/cairo/Makefile.am	2014-05-06 14:11:21.043041750 -0400
@@ -1,11 +1,14 @@
-
+if WITH_CAIRO
 plugin_sources = \
 	diacairo.c \
 	diacairo.h \
 	diacairo-interactive.c \
 	diacairo-renderer.c \
 	diacairo-print.c \
-	diacairo-print.h 
+	diacairo-print.h
+else
+plugin_sources = 
+endif
 
 pkglib_LTLIBRARIES = libcairo_filter.la
 
