diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 496c076..ce3fab1 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -41,11 +41,11 @@ endif()
 if (CMAKE_SYSTEM_PROCESSOR STREQUAL "64bit")
     message(STATUS "Build 64bit executable binary")
     add_executable(lz4c64 ${LZ4_SRCS})
-    install(TARGETS lz4c64 RUNTIME DESTINATION "./")
+    install(TARGETS lz4c64 RUNTIME DESTINATION "bin/")
     if(NOT BUILD_SHARED_LIBS)
         message(STATUS "Build 32bit executable binary")
         add_executable(lz4c32 ${LZ4_SRCS})
-        install(TARGETS lz4c32 RUNTIME DESTINATION "./")
+        install(TARGETS lz4c32 RUNTIME DESTINATION "bin/")
 
         SET_TARGET_PROPERTIES(lz4c32 PROPERTIES
             COMPILE_FLAGS PROPERTIES COMPILE_FLAGS "-m32" LINK_FLAGS "-m32")
@@ -53,7 +53,7 @@ if (CMAKE_SYSTEM_PROCESSOR STREQUAL "64bit")
 else()
     message(STATUS "Build 32bit executable binary")
     add_executable(lz4c32 ${LZ4_SRCS})
-    install(TARGETS lz4c32 RUNTIME DESTINATION "./")
+    install(TARGETS lz4c32 RUNTIME DESTINATION "bin/")
 endif()
 
 if(BUILD_SHARED_LIBS)
diff --git a/cmake/pack/CMakeLists.txt b/cmake/pack/CMakeLists.txt
index a9b0557..e85416e 100644
--- a/cmake/pack/CMakeLists.txt
+++ b/cmake/pack/CMakeLists.txt
@@ -48,16 +48,16 @@ set(FUZZER_SRCS ${SRC_DIR}lz4.c ${SRC_DIR}lz4hc.c ${SRC_DIR}lz4.h ${SRC_DIR}fuzz
 # EXECUTABLES FOR 32 Bit and 64 versions
 if(CMAKE_SYSTEM_PROCESSOR STREQUAL "64bit")
 	add_executable(lz4c32 ${LZ4_SRCS})
-	install(TARGETS lz4c32 RUNTIME DESTINATION "./")
+	install(TARGETS lz4c32 RUNTIME DESTINATION "bin/")
 SET_TARGET_PROPERTIES(lz4c32 PROPERTIES
 	COMPILE_FLAGS PROPERTIES COMPILE_FLAGS "-m32 -Os" LINK_FLAGS "-m32")
 endif()
 
 add_executable(lz4c ${LZ4_SRCS})
-install(TARGETS lz4c RUNTIME DESTINATION "./")
+install(TARGETS lz4c RUNTIME DESTINATION "bin/")
 
 add_executable(fuzzer ${FUZZER_SRCS})
-install(TARGETS fuzzer RUNTIME DESTINATION "./")
+install(TARGETS fuzzer RUNTIME DESTINATION "bin/")
 
 #target_link_libraries(lz4 ${LZ4_SRCS_LIB})
 ####################### CPACK PACKAGING ###################
