diff --git a/WORKSPACE b/WORKSPACE
index babb14b509..bcdd4f46e2 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -67,7 +67,7 @@ http_archive(
 http_archive(
     name = "bazel_skylib",
     sha256 = "1dde365491125a3db70731e25658dfdd3bc5dbdfd11b840b3e987ecf043c7ca0",
-    urls = ["https://github.com/bazelbuild/bazel-skylib/releases/download/0.9.0/bazel-skylib.0.9.0.tar.gz"],
+    urls = ["https://github.com/bazelbuild/bazel-skylib/releases/download/0.9.0/bazel_skylib-0.9.0.tar.gz"],
 )  # https://github.com/bazelbuild/bazel-skylib/releases
 http_archive(
     name = "com_github_apple_swift_swift_protobuf",
diff --git a/third_party/systemlibs/syslibs_configure.bzl b/third_party/systemlibs/syslibs_configure.bzl
index 8619cddfdd..4b57df0cbe 100644
--- a/third_party/systemlibs/syslibs_configure.bzl
+++ b/third_party/systemlibs/syslibs_configure.bzl
@@ -27,7 +27,7 @@ VALID_LIBS = [
     "grpc",
     "hwloc",
     "icu",
-    "jpeg",
+    "libjpeg_turbo",
     "jsoncpp_git",
     "keras_applications_archive",
     "lmdb",
