From 22275f7dbfa2d483ed4778427b137e83144bac3d Mon Sep 17 00:00:00 2001
From: Jason Zaman <jason@perfinion.com>
Date: Fri, 2 Aug 2019 20:06:33 +0800
Subject: [PATCH 3/3] systemlibs: jsoncpp: update header symlinks for jsoncpp
 1.9

Signed-off-by: Jason Zaman <jason@perfinion.com>
---
 third_party/systemlibs/jsoncpp.BUILD | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/third_party/systemlibs/jsoncpp.BUILD b/third_party/systemlibs/jsoncpp.BUILD
index 526fd0c418..7d54f9289b 100644
--- a/third_party/systemlibs/jsoncpp.BUILD
+++ b/third_party/systemlibs/jsoncpp.BUILD
@@ -6,6 +6,8 @@ filegroup(
 )
 
 HEADERS = [
+    "include/json/allocator.h",
+    "include/json/assertions.h",
     "include/json/autolink.h",
     "include/json/config.h",
     "include/json/features.h",
-- 
2.21.0

