diff --git a/tensorflow/tools/pip_package/setup.py b/tensorflow/tools/pip_package/setup.py
index fec08f66d7..9d5d1f2a3f 100644
--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -62,8 +62,8 @@ REQUIRED_PACKAGES = [
     'opt_einsum >= 2.3.2',
     'six >= 1.10.0',
     'protobuf >= 3.6.1',
-    'tb-nightly >= 1.15.0a20190821, < 1.15.0a20190822',
-    'tf-estimator-nightly >= 1.14.0.dev2019090401, < 1.14.0.dev2019090402',
+    'tensorboard >= 1.15.0',
+    'tensorflow_estimator >= 1.15.0',
     'termcolor >= 1.1.0',
     'wrapt >= 1.11.1',
 ]
