--- Makefile~	2009-11-01 14:53:27.000000000 +0100
+++ Makefile	2010-05-08 16:26:10.000000000 +0200
@@ -131,7 +131,7 @@
 	(cd doc; pdflatex ../dbus-java.tex)
 doc/dbus-java/index.html: dbus-java.tex .doc
 	mkdir -p doc/dbus-java/
-	(cd doc/dbus-java; TEX4HTENV=/etc/tex4ht/tex4ht.env htlatex ../../dbus-java.tex "xhtml,2" "" "-cvalidate")
+	(cd doc/dbus-java; TEX4HTENV=/etc/tex4ht/tex4ht.env mk4ht htlatex ../../dbus-java.tex "xhtml,2" "" "-cvalidate")
 	rm -f doc/dbus-java/*{4ct,4tc,aux,dvi,idv,lg,log,tmp,xref}
 	cp doc/dbus-java/dbus-java.html doc/dbus-java/index.html
 doc/api/index.html: $(SRCDIR)/*.java $(SRCDIR)/dbus/*.java .doc
