Only in xxHash-0.6.3.orig/cmake_unofficial: CMakeLists.txt
diff -ru xxHash-0.6.3.orig/Makefile xxHash-0.6.3/Makefile
--- xxHash-0.6.3.orig/Makefile	2017-12-18 10:47:41.740700013 +0100
+++ xxHash-0.6.3/Makefile	2017-12-18 11:14:05.157379597 +0100
@@ -50,7 +50,7 @@
 default: xxhsum
 
 .PHONY: all
-all: xxhsum xxhsum32 xxhsum_inlinedXXH
+all: xxhsum xxhsum32 xxhsum_inlinedXXH libxxhash.a
 
 xxhsum32: CFLAGS += -m32
 xxhsum xxhsum32: xxhash.c xxhsum.c
@@ -58,6 +58,13 @@
 	ln -sf $@ xxh32sum
 	ln -sf $@ xxh64sum
 
+xxhash.o: CPPFLAGS += -DXXHASH_EXPORT
+xxhash.o: xxhash.c
+	$(CC) $(FLAGS) -c $^
+
+libxxhash.a: xxhash.o
+	$(AR) rcs $@ $^ $@
+
 xxhsum_inlinedXXH: xxhsum.c
 	$(CC) $(FLAGS) -DXXH_PRIVATE_API $^ -o $@$(EXT)
 
