From 4d22c6a53ff1dcb657ab1913f1b100e42db19bca Mon Sep 17 00:00:00 2001
From: David Robillard <d@drobilla.net>
Date: Sat, 18 Mar 2017 12:00:11 +0100
Subject: Use new suil init API to fix Qt5 plugin UIs


diff --git a/src/jalv.c b/src/jalv.c
index 78901ca..20f24e7 100644
--- a/src/jalv.c
+++ b/src/jalv.c
@@ -777,6 +777,7 @@ main(int argc, char** argv)
 	jalv.bpm           = 120.0f;
 	jalv.control_in    = (uint32_t)-1;
 
+	suil_init(&argc, &argv, SUIL_ARG_NONE);
 	if (jalv_init(&argc, &argv, &jalv.opts)) {
 		return EXIT_FAILURE;
 	}
diff --git a/wscript b/wscript
index 19f51bb..e2adc63 100644
--- a/wscript
+++ b/wscript
@@ -59,7 +59,7 @@ def configure(conf):
     autowaf.check_pkg(conf, 'sord-0', uselib_store='SORD',
                       atleast_version='0.12.0', mandatory=True)
     autowaf.check_pkg(conf, 'suil-0', uselib_store='SUIL',
-                      atleast_version='0.6.0', mandatory=True)
+                      atleast_version='0.8.7', mandatory=True)
     autowaf.check_pkg(conf, 'sratom-0', uselib_store='SRATOM',
                       atleast_version='0.6.0', mandatory=True)
     if Options.options.portaudio:
-- 
cgit v0.10.2

