--- dwb-2013.03.30/Makefile
+++ dwb-2013.03.30/Makefile
@@ -14,17 +14,17 @@
 $(TARGET): $(SUBDIRS:%=%.subdir-make)
 
 %.subdir-make: $(SUBDIR_BUILD_FIRST:%=%.subdir-buildfirst)
-	@$(MAKE) $(MFLAGS) -C $*
+	$(MAKE) $(MFLAGS) -C $*
 
 #$(SRCDIR)/%: $(SUBDIR_BUILD_FIRST:%=%.subdir-buildfirst)
 
 %.subdir-buildfirst:
-	@$(MAKE) $(MFLAGS) -C $*
+	$(MAKE) $(MFLAGS) -C $*
 
 clean:  $(SUBDIRS:%=%.subdir-clean) $(SUBDIR_BUILD_FIRST:%=%.subdir-cleanfirst)
 
 %.subdir-clean %.subdir-cleanfirst:
-	@$(MAKE) $(MFLAGS) clean -C $*
+	$(MAKE) $(MFLAGS) clean -C $*
 
 
 install: $(TARGET) install-man install-data
--- dwb-2013.03.30/src/Makefile
+++ dwb-2013.03.30/src/Makefile
@@ -10,30 +10,25 @@
 all: $(TARGET)
 
 $(TARGET): $(OBJ) 
-	@echo "$(CC) $@"
-	@$(CC) $(OBJ) -o $(TARGET) $(LDFLAGS) 
+	$(CC) $(OBJ) -o $(TARGET) $(LDFLAGS) 
 
 -include $(OBJ:.o=.d)
 -include $(DOBJ:.do=.dd)
 
 %.o: %.c %.h config.h dwb.h
-	@echo "${CC} $<"
-	@$(CC) -c -o $@ $< $(CFLAGS) $(CPPFLAGS) 
+	$(CC) -c -o $@ $< $(CFLAGS) $(CPPFLAGS) 
 
 debug: $(DTARGET)
 
 deps.d: %.c %.h
-	@echo "$(CC) -MM $@"
-	@$(CC) $(CFLAGS) -MM $< -o $@
+	$(CC) $(CFLAGS) -MM $< -o $@
 
 
 %.do: %.c %.h config.h
-	@echo "${CC} $<"
-	@$(CC) -c -o $@ $< $(DCFLAGS) 
+	$(CC) -c -o $@ $< $(DCFLAGS) 
 
 $(DTARGET): $(DOBJ) 
-	@echo "$(CC) $@"
-	@$(CC) $(DOBJ) -o $(DTARGET) $(LDFLAGS) 
+	$(CC) $(DOBJ) -o $(DTARGET) $(LDFLAGS) 
 
 dependencies: $(DEPS)
 
--- dwb-2013.03.30/src/util/Makefile
+++ dwb-2013.03.30/src/util/Makefile
@@ -27,16 +27,14 @@
 	@./$(MKTLDS) < $(TLDS_IN) > $@
 
 $(MKTLDS): $(MKTLDS_SRC)
-	@echo "${CC} $<"
-	@$(CC) -o $@ $< $(CFLAGS) $(CPPFLAGS) $(LDFLAGS)
+	$(CC) -o $@ $< $(CFLAGS) $(CPPFLAGS) $(LDFLAGS)
 
 $(HSTS_PRELOAD): $(HSTS) $(TRANSPORT_SECURITY_CERTS) $(TRANSPORT_SECURITY_JSON)
 	@echo gen $(notdir $@)
 	@./$(HSTS) > $@
 
 $(HSTS): $(HSTS).c
-	@echo "${CC} $<"
-	@$(CC) $(CFLAGS) $(CPPFLAGS) -o $(HSTS) $(HSTS).c $(LDFLAGS)
+	$(CC) $(CFLAGS) $(CPPFLAGS) -o $(HSTS) $(HSTS).c $(LDFLAGS)
 
 clean: 
 	$(RM) $(OUTFILES) $(GEN_TOOLS)
--- dwb-2013.03.30/tools/Makefile
+++ dwb-2013.03.30/tools/Makefile
@@ -22,12 +22,10 @@
 all: $(TARGET)
 
 $(TARGET): $(OBJ)
-	@echo $(CC) $@
-	@$(CC) $< -o $@ $(LDFLAGS)
+	$(CC) $< -o $@ $(LDFLAGS)
 
 %.o: %.c
-	@echo $(CC) $<
-	@$(CC) $(CFLAGS) -c $< -o $@ $(CPPFLAGS)
+	$(CC) $(CFLAGS) -c $< -o $@ $(CPPFLAGS)
 
 clean: 
 	rm -f $(TARGET) $(OBJ)
