From 3d3cee8caeec813802a1af54c14805203fb1ddf6 Mon Sep 17 00:00:00 2001
From: "Friedrich W. H. Kossebau" <kossebau@kde.org>
Date: Tue, 7 Sep 2021 23:21:34 +0200
Subject: [PATCH] noteshared: add missing library to link list

GIT_SILENT
---
 src/noteshared/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/noteshared/CMakeLists.txt b/src/noteshared/CMakeLists.txt
index 5de60ab..6aa0b0a 100644
--- a/src/noteshared/CMakeLists.txt
+++ b/src/noteshared/CMakeLists.txt
@@ -16,6 +16,7 @@ target_link_libraries(noteshared
     Qt5::Core
     Qt5::Widgets
     KF5::I18n
+    KF5::AkonadiWidgets
     KF5::AkonadiCore
     KF5::AkonadiNotes
     KF5::XmlGui
-- 
GitLab

