--- a/CMakeLists.txt	2024-01-25 07:39:25.528947981 +0100
+++ b/CMakeLists.txt	2024-01-25 07:39:45.377618695 +0100
@@ -241,7 +241,7 @@
 	endif()
 
 	if(NOT GLM_DISABLE_AUTO_DETECTION)
-		add_compile_options(-Werror -Weverything)
+		add_compile_options(-Weverything)
 	endif()
 #	add_compile_options(-Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-c++11-long-long -Wno-padded -Wno-gnu-anonymous-struct -Wno-nested-anon-types)
 #	add_compile_options(-Wno-undefined-reinterpret-cast -Wno-sign-conversion -Wno-unused-variable -Wno-missing-prototypes -Wno-unreachable-code -Wno-missing-variable-declarations -Wno-sign-compare -Wno-global-constructors -Wno-unused-macros -Wno-format-nonliteral -Wno-float-equal)
@@ -252,7 +252,6 @@
 	endif()
 
 	if(NOT GLM_DISABLE_AUTO_DETECTION)
-		add_compile_options(-Werror)
 #		add_compile_options(-Wpedantic)
 #		add_compile_options(-Wall)
 #		add_compile_options(-Wextra)
