diff -Naruw postgis-3.1.0.orig/configure.ac postgis-3.1.0/configure.ac
--- postgis-3.1.0.orig/configure.ac	2020-12-18 14:39:26.000000000 -0500
+++ postgis-3.1.0/configure.ac	2020-12-23 21:44:09.223273050 -0500
@@ -31,7 +31,7 @@
 dnl
 dnl SQL Preprocessor
 dnl
-AC_PATH_PROG([CPPBIN], [cpp], [])
+AC_PATH_PROGS([CPPBIN], [cpp x86_64-pc-linux-gnu-cpp x86-pc-linux-gnu-cpp], [])
 if test "x$CPPBIN" != "x"; then
   SQLPP="${CPPBIN} -traditional-cpp -w -P"
 else
