--- a/python/setup.py
+++ b/python/setup.py
@@ -8,15 +8,10 @@
     define_macros = []
 
 module = Extension('_sphinx3',
-                   include_dirs = ['../../sphinxbase/include',
-                                   '../include',
-                                   '/usr/local/include/sphinxbase/',
-                                   '/usr/local/include/sphinx3',
-                                   ],
+                   include_dirs = ['../include',
+                                   '@GENTOO_PORTAGE_EPREFIX@/usr/include/sphinxbase'],
 		   define_macros = define_macros,
-		   library_dirs = ['../../sphinxbase/src/libsphinxbase/.libs',
-				'../src/libs3decoder/.libs',
-				'../../sphinxbase/lib/debug',
+		   library_dirs = ['../src/libs3decoder/.libs',
 				'../lib/debug',
 				],
                    libraries = ['sphinxbase', 's3decoder'],
--- a/src/libs3decoder/libsearch/vithist.c
+++ b/src/libs3decoder/libsearch/vithist.c
@@ -647,7 +647,7 @@
 {
     int32 se, fe, filler_done, th;
     vithist_entry_t *ve;
-    heap_t h;
+    heap_t *h;
     s3wid_t *wid;
     int32 i;
 
