diff -r -U1 maxima-5.35.1.orig/configure.ac maxima-5.35.1/configure.ac
--- maxima-5.35.1.orig/configure.ac	2014-12-13 06:44:51.000000000 +0600
+++ maxima-5.35.1/configure.ac	2014-12-20 20:43:04.713305777 +0600
@@ -379,3 +379,3 @@
 dnl n.b. openmcl_default_name is hardcoded in "with" message
-openmcl_default_name=openmcl
+openmcl_default_name=ccl
 AC_ARG_ENABLE(openmcl,
diff -r -U1 maxima-5.35.1.orig/src/Makefile.am maxima-5.35.1/src/Makefile.am
--- maxima-5.35.1.orig/src/Makefile.am	2014-09-24 00:02:16.000000000 +0700
+++ maxima-5.35.1/src/Makefile.am	2014-12-20 20:43:04.713305777 +0600
@@ -359,3 +359,3 @@
 
-EXECUTEOPENMCL = $(OPENMCL_NAME) -e
+EXECUTEOPENMCL = $(OPENMCL_NAME) --no-init -e
 BUILT_FILES += $(OPENMCL_MAXIMA)
Только в maxima-5.35.1/src: Makefile.am.orig
diff -r -U1 maxima-5.35.1.orig/src/maxima.in maxima-5.35.1/src/maxima.in
--- maxima-5.35.1.orig/src/maxima.in	2014-12-03 04:17:05.000000000 +0600
+++ maxima-5.35.1/src/maxima.in	2014-12-20 20:43:04.713305777 +0600
@@ -185,5 +185,5 @@
     if [ -x "$MAXIMA_IMAGESDIR/binary-$MAXIMA_LISP/maxima" ]; then
-      exec "$MAXIMA_IMAGESDIR/binary-$MAXIMA_LISP/maxima" $MAXIMA_LISP_OPTIONS -e '(cl-user::run)' -- "$arg1" "$arg2" "$arg3" "$arg4" "$arg5" "$arg6" "$arg7" "$arg8" "$arg9"
+      exec "$MAXIMA_IMAGESDIR/binary-$MAXIMA_LISP/maxima" $MAXIMA_LISP_OPTIONS --no-init -e '(cl-user::run)' -- "$arg1" "$arg2" "$arg3" "$arg4" "$arg5" "$arg6" "$arg7" "$arg8" "$arg9"
     else
-      exec "@OPENMCL_NAME@" -I "$maxima_image_base.image" $MAXIMA_LISP_OPTIONS -e '(cl-user::run)' -- "$arg1" "$arg2" "$arg3" "$arg4" "$arg5" "$arg6" "$arg7" "$arg8" "$arg9"
+      exec "@OPENMCL_NAME@" -I "$maxima_image_base.image" $MAXIMA_LISP_OPTIONS --no-init -e '(cl-user::run)' -- "$arg1" "$arg2" "$arg3" "$arg4" "$arg5" "$arg6" "$arg7" "$arg8" "$arg9"
    fi
