diff -Naur old/CPP/7zip/UI/FileManager/FM.cpp new/CPP/7zip/UI/FileManager/FM.cpp
--- old/CPP/7zip/UI/FileManager/FM.cpp	2015-01-18 08:59:09.000000000 -1000
+++ new/CPP/7zip/UI/FileManager/FM.cpp	2015-02-09 11:39:55.575712452 -1000
@@ -348,24 +348,7 @@
 
 bool IsLargePageSupported()
 {
-  #ifdef _WIN64
   return true;
-  #else
-  OSVERSIONINFO versionInfo;
-  versionInfo.dwOSVersionInfoSize = sizeof(versionInfo);
-  if (!::GetVersionEx(&versionInfo))
-    return false;
-  if (versionInfo.dwPlatformId != VER_PLATFORM_WIN32_NT || versionInfo.dwMajorVersion < 5)
-    return false;
-  if (versionInfo.dwMajorVersion > 5)
-    return true;
-  if (versionInfo.dwMinorVersion < 1)
-    return false;
-  if (versionInfo.dwMinorVersion > 1)
-    return true;
-  // return IsWow64();
-  return false;
-  #endif
 }
 
 #if 0 //#ifndef UNDER_CE
