--- CMakeLists.txt.orig	2014-05-01 20:43:13.000000000 +0400
+++ CMakeLists.txt	2014-05-12 13:56:55.000000000 +0400
@@ -112,9 +112,9 @@
 endif()
 
 # Get the git revision
-include(GetGitRevisionDescriptionDyND)
-get_git_head_revision("${CMAKE_CURRENT_SOURCE_DIR}" GIT_REFSPEC DYND_GIT_SHA1)
-git_describe("${CMAKE_CURRENT_SOURCE_DIR}" DYND_VERSION_STRING --dirty --always --match "v*")
+SET(DYND_GIT_SHA1 "@DYND_GIT_SHA1@")
+SET(DYND_VERSION_STRING "@DYND_VERSION@")
+
 message(STATUS "DyND version: ${DYND_VERSION_STRING}")
 configure_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/src/dynd/git_version.cpp.in"
