--- CMakeLists.txt.orig	2014-02-10 14:26:48.299381259 -0800
+++ CMakeLists.txt	2014-02-10 14:27:42.511704585 -0800
@@ -481,9 +481,9 @@
 if (DYND_INSTALL_LIB)
     # Install the libdynd binary
     install(TARGETS libdynd
-        RUNTIME DESTINATION lib
-        LIBRARY DESTINATION lib
-        ARCHIVE DESTINATION lib/static)
+        RUNTIME DESTINATION lib${LIB_SUFFIX}
+        LIBRARY DESTINATION lib${LIB_SUFFIX}
+        ARCHIVE DESTINATION lib${LIB_SUFFIX}/static)
     # Install the libdynd headers
     install(DIRECTORY "include/dynd" DESTINATION "${CMAKE_INSTALL_PREFIX}/include")
     # Install the libdynd-config script
