https://github.com/Stellarium/stellarium/issues/4019

--- a/plugins/TelescopeControl/src/INDI/INDIConnection.cpp
+++ b/plugins/TelescopeControl/src/INDI/INDIConnection.cpp
@@ -25,9 +25,9 @@
 #include <limits>
 #include <cmath>
 
-#include <libs/indiclient/baseclient.h>
-#include <libs/indidevice/basedevice.h>
-#include <libs/indibase/inditelescope.h>
+#include <libindi/baseclient.h>
+#include <libindi/basedevice.h>
+#include <libindi/inditelescope.h>
 
 const int INDIConnection::SLEW_STOP = INDI::Telescope::SLEW_GUIDE - 1;
 
--- a/plugins/TelescopeControl/src/INDI/INDIConnection.hpp
+++ b/plugins/TelescopeControl/src/INDI/INDIConnection.hpp
@@ -20,8 +20,8 @@
 #define INDICONNECTION_HPP
 
 #include <QObject>
-#include <libs/indiclient/baseclient.h>
-#include <libs/indidevice/basedevice.h>
+#include <libindi/baseclient.h>
+#include <libindi/basedevice.h>
 
 #include <mutex>
 #include <QStringList>
--- a/plugins/TelescopeControl/src/INDI/TelescopeClientINDI.cpp
+++ b/plugins/TelescopeControl/src/INDI/TelescopeClientINDI.cpp
@@ -25,7 +25,7 @@
 
 #include "StelCore.hpp"
 #include "StelUtils.hpp"
-#include <libs/indibase/inditelescope.h>
+#include <libindi/inditelescope.h>
 #include "INDIControlWidget.hpp"
 
 TelescopeClientINDI::TelescopeClientINDI(const QString &name, const QString &params):
--- a/plugins/TelescopeControl/src/INDI/test/testINDIConnection.cpp
+++ b/plugins/TelescopeControl/src/INDI/test/testINDIConnection.cpp
@@ -23,7 +23,7 @@
 #include <string>
 
 #include "INDIConnection.hpp"
-#include <libs/indidevice/basedevice.h>
+#include <libindi/basedevice.h>
 
 void TestINDIConnection::deafultCoordinates()
 {
