diff -Naur qtwebengine-opensource-src-5.7.0.orig/src/3rdparty/chromium/media/ffmpeg/ffmpeg_common.h qtwebengine-opensource-src-5.7.0/src/3rdparty/chromium/media/ffmpeg/ffmpeg_common.h
--- qtwebengine-opensource-src-5.7.0.orig/src/3rdparty/chromium/media/ffmpeg/ffmpeg_common.h	2016-11-23 13:01:20.929772871 -0800
+++ qtwebengine-opensource-src-5.7.0/src/3rdparty/chromium/media/ffmpeg/ffmpeg_common.h	2016-11-23 13:02:18.879462854 -0800
@@ -24,7 +24,9 @@
 // Disable deprecated features which result in spammy compile warnings.  This
 // list of defines must mirror those in the 'defines' section of BUILD.gn file &
 // ffmpeg.gyp file or the headers below will generate different structures!
+#if !defined(USE_SYSTEM_FFMPEG)
 #define FF_API_CONVERGENCE_DURATION 0
+#endif
 // Upstream libavcodec/utils.c still uses the deprecated
 // av_dup_packet(), causing deprecation warnings.
 // The normal fix for such things is to disable the feature as below,
