--- wmmemload-0.1.6/src/mem_linux.c	       2003-11-03 22:06:39.000000000 +0000
+++ wmmemload-0.1.6/src/mem_linux.c	       2011-11-12 11:05:13.000000000 +0000
@@ -49,8 +49,9 @@
 
    /* new format ? (kernel >= 2.5.1pre?) */
    /* see linux/fs/proc/proc_misc.c */
-   if (version >= 2 && patchlevel >= 5 && sublevel >= 1 || \
-       version >= 2 && patchlevel >= 6 && sublevel >= 0)
+   if (version == 2 && patchlevel >= 5 && sublevel >= 1 || \
+       version == 2 && patchlevel >= 6 && sublevel >= 0 || \
+       version >2 )
       isnewformat = 1;
 }
