--- a/files/usr/lib/cinnamon-settings/modules/cs_backgrounds.py
+++ b/files/usr/lib/cinnamon-settings/modules/cs_backgrounds.py
@@ -334,10 +334,9 @@ class BackgroundWallpaperPane (Gtk.VBox)
     
     def update_icon_view(self):
         pictures_list = []
-        if os.path.exists("/usr/share/cinnamon-background-properties"):
-            for i in os.listdir("/usr/share/cinnamon-background-properties"):
-                if i.endswith(".xml"):
-                    pictures_list += self.parse_xml_backgrounds_list(os.path.join("/usr/share/cinnamon-background-properties", i))
+        for i in os.listdir("/usr/share/gnome-background-properties"):
+            if i.endswith(".xml"):
+                    pictures_list += self.parse_xml_backgrounds_list(os.path.join("/usr/share/gnome-background-properties", i))
         
         path = os.path.join(os.getenv("HOME"), ".cinnamon", "backgrounds")
         if os.path.exists(path):
