
--- a/mk/clang.mk
+++ b/mk/clang.mk
@@ -3,7 +3,8 @@ _INCLUDE_MK_CLANG_=1
 CC?=clang
 RANLIB=ranlib
 ONELIB=0
-CC_AR=ar q ${LIBAR}
+AR?=ar
+CC_AR=$(AR) q ${LIBAR}
 CFLAGS+=-MD
 CFLAGS_INCLUDE=-I
 LDFLAGS_LINK=-l
--- a/mk/gcc.mk
+++ b/mk/gcc.mk
@@ -4,7 +4,8 @@ CC?=gcc
 OBJCOPY=objcopy
 RANLIB=ranlib
 ONELIB=0
-CC_AR=ar q ${LIBAR}
+AR?=ar
+CC_AR=$(AR) q ${LIBAR}
 CFLAGS+=-MD
 CFLAGS_INCLUDE=-I
 LDFLAGS_LINK=-l
