# drop compiler check to enable use of CXX
--- configure.in.old	2013-09-15 09:38:54.000000000 +0200
+++ configure.in	2013-09-15 09:40:31.000000000 +0200
@@ -134,15 +134,8 @@
                 AC_MSG_ERROR("sorry...architecture not supported")
             ;;
         esac
-        case "$CXX_COMP" in
-            c++ | g++)
-                EXTRA_CXXFLAGS="${EXTRA_CXXFLAGS} -m${CPU_ARCH}"
-                EXTRA_ASFLAGS="-m${CPU_ARCH}"
-                ;;
-            *)
-                AC_MSG_ERROR("sorry...compiler not supported")
-            ;;
-        esac
+        EXTRA_CXXFLAGS="${EXTRA_CXXFLAGS} -m${CPU_ARCH}"
+        EXTRA_ASFLAGS="-m${CPU_ARCH}"
         ;;
     *freebsd*)
         case "$target_cpu" in
