--- usr.sbin/adduser/adduser.sh.old	2006-10-01 13:41:16 +0000
+++ usr.sbin/adduser/adduser.sh	2006-10-01 13:41:38 +0000
@@ -809,7 +809,7 @@
 NOHOME="/nonexistent"
 NOLOGIN="nologin"
 NOLOGIN_PATH="/usr/sbin/nologin"
-GREPCMD="/usr/bin/grep"
+GREPCMD="/bin/grep"
 DATECMD="/bin/date"
 
 # Set default values
@@ -823,7 +823,7 @@
 uhome=
 upass=
 ushell=
-udotdir=/usr/share/skel
+udotdir=/etc/skel
 ugroups=
 uexpire=
 upwexpire=
