diff -uNr a/nx-X11/extras/Mesa_6.4.2/src/mesa/main/glheader.h b/nx-X11/extras/Mesa_6.4.2/src/mesa/main/glheader.h
--- a/nx-X11/extras/Mesa_6.4.2/src/mesa/main/glheader.h	2020-11-04 06:43:58.000000000 -0600
+++ b/nx-X11/extras/Mesa_6.4.2/src/mesa/main/glheader.h	2023-03-16 12:58:34.515215736 -0600
@@ -62,9 +62,6 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#if defined(__linux__) && defined(__i386__)
-#include <fpu_control.h>
-#endif
 #endif
 #include <float.h>
 #include <stdarg.h>
diff -uNr a/nx-X11/extras/Mesa_6.4.2/src/mesa/main/imports.c b/nx-X11/extras/Mesa_6.4.2/src/mesa/main/imports.c
--- a/nx-X11/extras/Mesa_6.4.2/src/mesa/main/imports.c	2020-11-04 06:43:58.000000000 -0600
+++ b/nx-X11/extras/Mesa_6.4.2/src/mesa/main/imports.c	2023-03-16 12:59:23.828001633 -0600
@@ -1169,20 +1169,6 @@
    static GLboolean initialized = GL_FALSE;
    if (!initialized) {
       init_sqrt_table();
-
-#if defined(_FPU_GETCW) && defined(_FPU_SETCW)
-      {
-         const char *debug = _mesa_getenv("MESA_DEBUG");
-         if (debug && _mesa_strcmp(debug, "FP")==0) {
-            /* die on FP exceptions */
-            fpu_control_t mask;
-            _FPU_GETCW(mask);
-            mask &= ~(_FPU_MASK_IM | _FPU_MASK_DM | _FPU_MASK_ZM
-                      | _FPU_MASK_OM | _FPU_MASK_UM);
-            _FPU_SETCW(mask);
-         }
-      }
-#endif
       initialized = GL_TRUE;
    }
 
diff -uNr a/nx-X11/programs/Xserver/Xext/xf86bigfont.c b/nx-X11/programs/Xserver/Xext/xf86bigfont.c
--- a/nx-X11/programs/Xserver/Xext/xf86bigfont.c	2020-11-04 06:43:58.000000000 -0600
+++ b/nx-X11/programs/Xserver/Xext/xf86bigfont.c	2023-03-16 13:01:30.256598143 -0600
@@ -40,13 +40,6 @@
 
 #include <sys/types.h>
 #ifdef HAS_SHM
-#if defined(linux) && (!defined(__GNU_LIBRARY__) || __GNU_LIBRARY__ < 2)
-/* libc4 does not define __GNU_LIBRARY__, libc5 defines __GNU_LIBRARY__ as 1 */
-/* Linux libc4 and libc5 only (because glibc doesn't include kernel headers):
-   Linux 2.0.x and 2.2.x define SHMLBA as PAGE_SIZE, but forget to define
-   PAGE_SIZE. It is defined in <asm/page.h>. */
-#include <asm/page.h>
-#endif
 #ifdef SVR4
 #include <sys/sysmacros.h>
 #endif
