--- Makefile.orig	2006-10-24 22:01:58.000000000 -0600
+++ Makefile	2006-10-24 22:02:16.000000000 -0600
@@ -1,4 +1,4 @@
-PREFIX=/usr/local
+PREFIX=/usr
 BINDIR=$(PREFIX)/bin
 LIBDIR=$(PREFIX)/lib/yarssr
 DATADIR=$(PREFIX)/share
@@ -27,7 +27,6 @@
 	@echo Copying share files to $(DESTDIR)/$(DATADIR):
 	@cp -Rp share/* $(DESTDIR)/$(DATADIR)/
 
-	find $(DESTDIR)/$(DATADIR) -type f -exec chmod 644 "{}" \;
 	find $(DESTDIR)/$(LIBDIR) -type f -exec chmod 644 "{}" \;
 
 	install -m 0644 build/locale/en/$(LC_CATEGORY)/yarssr.mo $(DESTDIR)/$(LOCALEDIR)/en/$(LC_CATEGORY)/
