diff -Naur fsviewer-app-0.2.6.orig/configure.ac fsviewer-app-0.2.6/configure.ac
--- fsviewer-app-0.2.6.orig/configure.ac	2012-07-11 17:55:06.448981246 +0200
+++ fsviewer-app-0.2.6/configure.ac	2012-07-11 18:04:15.528940159 +0200
@@ -81,6 +81,8 @@
              AC_MSG_ERROR([WMaker library not found]))
 AC_CHECK_LIB(WINGs, WMAppSetMainMenu,,
              AC_MSG_ERROR([WINGs library not found]))
+AC_CHECK_LIB(WUtil, WMCreatePLString,,
+             AC_MSG_ERROR([WUtil library not found]))
 
 dnl Checks for PropList (not necessary since version 0.2.3b)
 dnl AC_CHECK_HEADERS(proplist.h,, AC_MSG_WARN(Can't find PropList include-file: Please install libPropList (included in WindowMaker)))
diff -Naur fsviewer-app-0.2.6.orig/defs/chdef.c fsviewer-app-0.2.6/defs/chdef.c
--- fsviewer-app-0.2.6.orig/defs/chdef.c	2012-07-11 17:55:06.444981361 +0200
+++ fsviewer-app-0.2.6/defs/chdef.c	2012-07-11 17:55:50.387699640 +0200
@@ -374,7 +374,7 @@
 	SetIntegerForKey(0, "DisplayMCListPixmap");
 	
 	WMWritePropListToFile(filesDB,
-                wdefaultspathfordomain("FSViewer"), True);
+                wdefaultspathfordomain("FSViewer"));
 	result = 0;
     }	    
     else
diff -Naur fsviewer-app-0.2.6.orig/src/extnInspector.c fsviewer-app-0.2.6/src/extnInspector.c
--- fsviewer-app-0.2.6.orig/src/extnInspector.c	2012-07-11 17:55:06.452981130 +0200
+++ fsviewer-app-0.2.6/src/extnInspector.c	2012-07-11 17:56:02.315351669 +0200
@@ -229,7 +229,7 @@
 
     if(numRows > 0)
 	WMWritePropListToFile(filesDB,
-	  wdefaultspathfordomain("FSViewer"), True);
+	  wdefaultspathfordomain("FSViewer"));
 
    if(extn)
 	free(extn);
