--- fsviewer-app-0.2.6.orig/configure.ac	2007-10-13 23:54:47.000000000 +0200
+++ fsviewer-app-0.2.6/configure.ac	2015-09-02 21:10:40.829519545 +0200
@@ -26,15 +26,15 @@
 dnl look for X windows first so further libs check
 AC_PATH_X 
 
-if test "$x_includes" != "NONE"; then
+if test "$x_includes" != "NONE" && test -n "$x_includes"; then
         CFLAGS="$CFLAGS -I$x_includes"
 fi
-if test "$x_libraries" != "NONE"; then
+if test "$x_libraries" != "NONE" && test -n "$x_libraries"; then
         LDFLAGS="$LDFLAGS -L$x_libraries"
 fi
 
 dnl Additional "default" items
-#LIBS="$LIBS -lX11"
+LIBS="$LIBS -lm -lX11 -lXpm"
 #CFLAGS="$CFLAGS -I/usr/X11/include"
 #LDFLAGS="$LDFLAGS -L/usr/X11/lib"
 
