diff -Naur fsviewer-app-0.2.6.orig/src/FSViewer.c fsviewer-app-0.2.6/src/FSViewer.c
--- fsviewer-app-0.2.6.orig/src/FSViewer.c	2007-10-14 17:29:04.000000000 +0200
+++ fsviewer-app-0.2.6/src/FSViewer.c	2020-02-03 00:20:37.979716353 +0100
@@ -25,6 +25,8 @@
 static Bool      focusIn;
 int ModifierFromKey(Display *dpy, char *key);
 
+WMUserDefaults *defaultsDB;
+WMPropList *filesDB;
 void
 wAbort(Bool foo)
 {
diff -Naur fsviewer-app-0.2.6.orig/src/FSViewer.h fsviewer-app-0.2.6/src/FSViewer.h
--- fsviewer-app-0.2.6.orig/src/FSViewer.h	2006-07-23 09:24:15.000000000 +0200
+++ fsviewer-app-0.2.6/src/FSViewer.h	2020-02-03 00:20:39.299718025 +0100
@@ -123,8 +123,8 @@
 } CallbackRec;
 
 /* Application defaults */
-WMUserDefaults *defaultsDB;
-WMPropList *filesDB;
+extern WMUserDefaults *defaultsDB;
+extern WMPropList *filesDB;
 
 /* all Panels must start with the following layout */
 typedef struct PanelRec {
