diff -r -U3 spyder-4.0.0b1.orig/setup.py spyder-4.0.0b1/setup.py
--- spyder-4.0.0b1.orig/setup.py	2018-08-12 04:35:49.000000000 +0700
+++ spyder-4.0.0b1/setup.py	2018-08-14 20:25:32.827148778 +0700
@@ -112,21 +107,6 @@
 
 
 #==============================================================================
-# Make Linux detect Spyder desktop file
-#==============================================================================
-class MyInstallData(install_data):
-    def run(self):
-        install_data.run(self)
-        if sys.platform.startswith('linux'):
-            try:
-                subprocess.call(['update-desktop-database'])
-            except:
-                print("ERROR: unable to update desktop database",
-                      file=sys.stderr)
-CMDCLASS = {'install_data': MyInstallData}
-
-
-#==============================================================================
 # Main scripts
 #==============================================================================
 # NOTE: the '[...]_win_post_install.py' script is installed even on non-Windows
@@ -210,8 +190,7 @@
                  'Intended Audience :: Science/Research',
                  'Intended Audience :: Developers',
                  'Topic :: Scientific/Engineering',
-                 'Topic :: Software Development :: Widget Sets'],
-    cmdclass=CMDCLASS)
+                 'Topic :: Software Development :: Widget Sets'])
 
 
 #==============================================================================
