From b16f4381d0cdc147e4aff0b6f1fdbac3bbf1de7d Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx@gmail.com>
Date: Fri, 25 Apr 2025 00:32:48 +0200
Subject: [PATCH] find hdf5 in global scope

Signed-off-by: Paul Zander <negril.nx@gmail.com>

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 96bb1d4..c5d8e09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -419,6 +419,11 @@ if (VTK_FORBID_DOWNLOADS AND VTK_BUILD_TESTING)
     "copy of the testing data.")
 endif ()
 
+if(VTK_MODULE_ENABLE_VTK_hdf5 STREQUAL "WANT" OR
+      VTK_MODULE_ENABLE_VTK_hdf5 STREQUAL "YES")
+  find_package(HDF5 REQUIRED NO_MODULE COMPONENTS HL)
+endif()
+
 include(vtkModule)
 include(vtkModuleDebugging)
 set(vtk_source_directories
-- 
2.49.0

