diff -up VTK-6.1.0/Rendering/OpenGL/vtkOpenGL.h.glext VTK-6.1.0/Rendering/OpenGL/vtkOpenGL.h
--- VTK-6.1.0/Rendering/OpenGL/vtkOpenGL.h.glext	2014-01-22 08:55:41.000000000 -0700
+++ VTK-6.1.0/Rendering/OpenGL/vtkOpenGL.h	2014-11-19 10:27:12.349345199 -0700
@@ -19,7 +19,8 @@
 #include "vtkConfigure.h"
 
 // To prevent gl.h to include glext.h provided by the system
-#define GL_GLEXT_LEGACY
+// https://bugzilla.redhat.com/show_bug.cgi?id=1138466
+// #define GL_GLEXT_LEGACY
 #if defined(__APPLE__) && (defined(VTK_USE_CARBON) || defined(VTK_USE_COCOA))
 # include <OpenGL/gl.h> // Include OpenGL API.
 #else
