--- a/Accelerators/Vtkm/Core/vtkmlib/vtkmDataArray.hxx
+++ b/Accelerators/Vtkm/Core/vtkmlib/vtkmDataArray.hxx
@@ -334,7 +334,7 @@
 auto TransformForScalarRange(const ArrayHandleType& src,
   const vtkm::cont::ArrayHandle<vtkm::UInt8>& ghost, vtkm::UInt8 ghostValueToSkip, bool finitesOnly)
 {
-  DecoratorForScalarRanage decorator{ ghostValueToSkip, finitesOnly };
+  DecoratorForScalarRanage decorator{ { ghostValueToSkip, finitesOnly } };
   return vtkm::cont::make_ArrayHandleDecorator(src.GetNumberOfValues(), decorator, src, ghost);
 }
 
@@ -342,7 +342,7 @@
 auto TransformForVectorRange(const ArrayHandleType& src,
   const vtkm::cont::ArrayHandle<vtkm::UInt8>& ghost, vtkm::UInt8 ghostValueToSkip, bool finitesOnly)
 {
-  DecoratorForVectorRanage decorator{ ghostValueToSkip, finitesOnly };
+  DecoratorForVectorRanage decorator{ { ghostValueToSkip, finitesOnly } };
   return vtkm::cont::make_ArrayHandleDecorator(src.GetNumberOfValues(), decorator, src, ghost);
 }
 
