--- IO/vtkPNGReader.cxx
+++ IO/vtkPNGReader.cxx
@@ -116,7 +116,7 @@
   // minimum of a byte per pixel
   if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) 
     {
-    png_set_gray_1_2_4_to_8(png_ptr);
+    png_set_expand_gray_1_2_4_to_8(png_ptr);
     }
 
   // add alpha if any alpha found
@@ -225,7 +225,7 @@
   // minimum of a byte per pixel
   if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) 
     {
-    png_set_gray_1_2_4_to_8(png_ptr);
+    png_set_expand_gray_1_2_4_to_8(png_ptr);
     }
 
   // add alpha if any alpha found
