diff --git a/ThirdParty/eigen/vtkeigen/CMakeLists.txt b/ThirdParty/eigen/vtkeigen/CMakeLists.txt
index 8d1e7839..120553e5 100644
--- a/ThirdParty/eigen/vtkeigen/CMakeLists.txt
+++ b/ThirdParty/eigen/vtkeigen/CMakeLists.txt
@@ -22,11 +22,6 @@ if (NOT CMAKE_BUILD_TYPE)
 endif()
 
 string(TOLOWER "${CMAKE_BUILD_TYPE}" cmake_build_type_tolower)
-if(    NOT cmake_build_type_tolower STREQUAL "debug"
-   AND NOT cmake_build_type_tolower STREQUAL "release"
-   AND NOT cmake_build_type_tolower STREQUAL "relwithdebinfo")
-  message(FATAL_ERROR "Unknown build type \"${CMAKE_BUILD_TYPE}\". Allowed values are Debug, Release, RelWithDebInfo (case-insensitive).")
-endif()
 
 
 #############################################################################
