Index: coregrind/m_main.c
===================================================================
--- coregrind/m_main.c	(revision 13287)
+++ coregrind/m_main.c	(revision 13288)
@@ -2636,8 +2636,8 @@
 
    From this derive two requirements:
 
-   1. gcc may emit calls to memcpy and memset to deal with structure
-      assignments etc.  Since we have chosen to ignore all the
+   1. gcc may emit calls to memcpy, memmove and memset to deal with
+      structure assignments etc.  Since we have chosen to ignore all the
       "normal" supporting libraries, we have to provide our own
       implementations of them.  No problem.
 
@@ -2651,6 +2651,10 @@
 void* memcpy(void *dest, const void *src, SizeT n) {
    return VG_(memcpy)(dest,src,n);
 }
+void* memmove(void *dest, const void *src, SizeT n);
+void* memmove(void *dest, const void *src, SizeT n) {
+   return VG_(memmove)(dest,src,n);
+}
 void* memset(void *s, int c, SizeT n);
 void* memset(void *s, int c, SizeT n) {
   return VG_(memset)(s,c,n);
