diff -uNr mono-debugger-2.10.old/sysdeps/server/breakpoints.c mono-debugger-2.10/sysdeps/server/breakpoints.c
--- mono-debugger-2.10.old/sysdeps/server/breakpoints.c	2011-01-13 23:41:31.000000000 +0100
+++ mono-debugger-2.10/sysdeps/server/breakpoints.c	2011-11-12 15:46:57.909192866 +0100
@@ -1,6 +1,6 @@
 #include <server.h>
 #include <breakpoints.h>
-#include <glib/gthread.h>
+#include <glib.h>
 #include <sys/stat.h>
 #include <signal.h>
 #ifdef HAVE_UNISTD_H
