diff -Naur jacl1.4.1.bak/Makefile.in jacl1.4.1/Makefile.in
--- jacl1.4.1.bak/Makefile.in	2008-09-14 14:44:18.000000000 +1200
+++ jacl1.4.1/Makefile.in	2008-09-14 14:46:31.000000000 +1200
@@ -54,7 +54,7 @@
 XTERM			= xterm
 JAVA_FLAGS		= @JAVA_FLAGS@
 JAVA_G_FLAGS		= @JAVA_G_FLAGS@
-JAVAC_FLAGS		= @JAVAC_FLAGS@
+JAVAC_FLAGS		?= @JAVAC_FLAGS@
 JAVAC_D_FLAG		= @JAVAC_D_FLAG@
 JAR_EXTRACT_FLAGS	= @JAR_EXTRACT_FLAGS@
 JAR_COMPRESS_FLAGS	= @JAR_COMPRESS_FLAGS@
@@ -983,8 +983,8 @@
 	$(JAVAC) $(JAVAC_FLAGS) \
 	$(JAVAC_D_FLAG) $(TCLJAVA_BUILD_DIR) \
 		tcl/lang/*.java
-	mkdir $(TCLJAVA_BUILD_DIR)/tcl/lang/library
-	mkdir $(TCLJAVA_BUILD_DIR)/tcl/lang/library/java
+	mkdir -p $(TCLJAVA_BUILD_DIR)/tcl/lang/library
+	mkdir -p $(TCLJAVA_BUILD_DIR)/tcl/lang/library/java
 	cp -fr $(TCLJAVA_SRC_DIR)/tcl/lang/library/java/*.tcl \
 		$(TCLJAVA_BUILD_DIR)/tcl/lang/library/java
 
diff -Naur jacl1.4.1.bak/tcljava.m4 jacl1.4.1/tcljava.m4
--- jacl1.4.1.bak/tcljava.m4	2008-09-14 14:44:18.000000000 +1200
+++ jacl1.4.1/tcljava.m4	2008-09-14 14:44:30.000000000 +1200
@@ -226,9 +226,9 @@
     # Look for a setting for the CLASSPATH, we might need one to run JAVAC
     AC_JAVA_CLASSPATH
 
-    # FIXME : add detection of command line arguments for JAVAC
-
-    JAVAC_FLAGS=-g
+    if test "x${JAVAC_FLAGS}" = "x"; then
+        JAVAC_FLAGS=-g
+    fi
     JAVAC_D_FLAG=-d
 
     dnl Test out the Java compiler with an empty class
