--- griffith-0.13/lib/db/validators.py
+++ griffith-0.13/lib/db/validators.py
@@ -23,7 +23,13 @@
 
 import logging
 
-from sqlalchemy.orm.interfaces import AttributeExtension, InstrumentationManager
+from sqlalchemy.orm.interfaces import AttributeExtension
+try:
+    # sql alchemy 0.8 (and above)
+    from sqlalchemy.ext.instrumentation import InstrumentationManager
+except:
+    # sql alchemy 0.7
+    from sqlalchemy.orm.interfaces import InstrumentationManager
 from sqlalchemy.orm import ColumnProperty
 from sqlalchemy.types import String
 
