--- inkscape-0.48.4/configure.ac	2013-12-31 00:38:18.592732496 +0400
+++ inkscape-0.48.4/configure.ac	2013-12-31 00:39:13.558730872 +0400
@@ -263,7 +263,11 @@
 			# include <gc.h>
 			#endif
 			#include <stdio.h>
-			extern unsigned GC_version;
+			#if (GC_VERSION_MAJOR >= 7 && GC_VERSION_MINOR >= 4)
+				unsigned GC_version = GC_get_version();
+			#else
+				extern unsigned GC_version;
+			#endif
 			int main(void){
 				unsigned min = ((6 << 16) | (4 << 8) | 0);
 				printf("%d.%d.%d ",GC_version >> 16, (GC_version >> 8) & 0xFF, GC_version & 0xFF);
