--- inkscape-0.48.2/configure.ac.orig
+++ inkscape-0.48.2/configure.ac
@@ -713,14 +713,14 @@
 
 with_libwpg=no
 
-PKG_CHECK_MODULES(LIBWPG01, libwpg-0.1 libwpg-stream-0.1, with_libwpg01=yes, with_libwpg01=no)
-if test "x$with_libwpg01" = "xyes"; then
-	AC_DEFINE(WITH_LIBWPG01,1,[Build in libwpg 0.1.x])
-	with_libwpg=yes
-	AC_SUBST(LIBWPG_LIBS, $LIBWPG01_LIBS)
-	AC_SUBST(LIBWPG_CFLAGS, $LIBWPG01_CFLAGS)
-fi
-AM_CONDITIONAL(WITH_LIBWPG01, test "x$with_libwpg01" = "xyes")
+#PKG_CHECK_MODULES(LIBWPG01, libwpg-0.1 libwpg-stream-0.1, with_libwpg01=yes, with_libwpg01=no)
+#if test "x$with_libwpg01" = "xyes"; then
+#    AC_DEFINE(WITH_LIBWPG01,1,[Build in libwpg 0.1.x])
+#    with_libwpg=yes
+#    AC_SUBST(LIBWPG_LIBS, $LIBWPG01_LIBS)
+#    AC_SUBST(LIBWPG_CFLAGS, $LIBWPG01_CFLAGS)
+#fi
+#AM_CONDITIONAL(WITH_LIBWPG01, test "x$with_libwpg01" = "xyes")
 
 PKG_CHECK_MODULES(LIBWPG02, libwpg-0.2 libwpd-0.9 libwpd-stream-0.9, with_libwpg02=yes, with_libwpg02=no)
 if test "x$with_libwpg02" = "xyes"; then
