diff -upNr a/cddl/lib/libzfs/Makefile b/cddl/lib/libzfs/Makefile
--- a/cddl/lib/libzfs/Makefile	2012-03-30 18:58:53.174771000 +0000
+++ b/cddl/lib/libzfs/Makefile	2012-03-30 19:04:39.904906253 +0000
@@ -54,4 +54,8 @@ CFLAGS+= -I${.CURDIR}/../../../cddl/cont
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/lib/libuutil/common
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/lib/libzfs/common
 
+LDFLAGS+= -L${.CURDIR}/../../lib/libnvpair
+LDFLAGS+= -L${.CURDIR}/../../lib/libumem
+LDFLAGS+= -L${.CURDIR}/../../lib/libuutil
+
 .include <bsd.lib.mk>
diff -upNr a/cddl/lib/libzpool/Makefile b/cddl/lib/libzpool/Makefile
--- a/cddl/lib/libzpool/Makefile	2012-03-30 18:58:53.174771000 +0000
+++ b/cddl/lib/libzpool/Makefile	2012-03-30 19:04:55.839233753 +0000
@@ -56,6 +56,10 @@ CFLAGS+=	-I${.CURDIR}/../../../lib/libpt
 CFLAGS+=	-I${.CURDIR}/../../../lib/libpthread/sys
 CFLAGS+=	-I${.CURDIR}/../../../lib/libthr/arch/${MACHINE_CPUARCH}/include
 
+LDFLAGS+= -L${.CURDIR}/../../lib/libavl
+LDFLAGS+= -L${.CURDIR}/../../lib/libnvpair
+LDFLAGS+= -L${.CURDIR}/../../lib/libumem
+
 DPADD=		${LIBMD} ${LIBPTHREAD} ${LIBZ}
 LDADD=		-lmd -lpthread -lz -lumem -lnvpair -lavl
 
diff -upNr a/cddl/sbin/zfs/Makefile b/cddl/sbin/zfs/Makefile
--- a/cddl/sbin/zfs/Makefile	2012-03-30 18:58:53.174771000 +0000
+++ b/cddl/sbin/zfs/Makefile	2012-03-30 19:05:11.094859502 +0000
@@ -21,6 +21,11 @@ CFLAGS+= -I${.CURDIR}/../../../sys/cddl/
 CFLAGS+= -I${.CURDIR}/../../../sys/cddl/contrib/opensolaris/uts/common/sys
 CFLAGS+= -I${.CURDIR}/../../../sys/cddl/contrib/opensolaris/common/zfs
 
+LDFLAGS+= -L${.CURDIR}/../../lib/libnvpair
+LDFLAGS+= -L${.CURDIR}/../../lib/libumem
+LDFLAGS+= -L${.CURDIR}/../../lib/libuutil
+LDFLAGS+= -L${.CURDIR}/../../lib/libzfs
+
 DPADD=	${LIBGEOM} ${LIBM} ${LIBNVPAIR} ${LIBSBUF} ${LIBUMEM} \
 	${LIBUTIL} ${LIBUUTIL} ${LIBZFS}
 LDADD=	-lgeom -lm -lnvpair -lsbuf -lumem -lutil -luutil -lzfs
diff -upNr a/cddl/sbin/zpool/Makefile b/cddl/sbin/zpool/Makefile
--- a/cddl/sbin/zpool/Makefile	2012-03-30 18:58:53.174771000 +0000
+++ b/cddl/sbin/zpool/Makefile	2012-03-30 19:05:24.830721002 +0000
@@ -26,6 +26,12 @@ CFLAGS+= -I${.CURDIR}/../../../sys/cddl/
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/lib/libzpool/common
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/cmd/stat/common
 
+LDFLAGS+= -L${.CURDIR}/../../lib/libavl
+LDFLAGS+= -L${.CURDIR}/../../lib/libnvpair
+LDFLAGS+= -L${.CURDIR}/../../lib/libumem
+LDFLAGS+= -L${.CURDIR}/../../lib/libuutil
+LDFLAGS+= -L${.CURDIR}/../../lib/libzfs
+
 DPADD=	${LIBAVL} ${LIBGEOM} ${LIBM} ${LIBNVPAIR} ${LIBSBUF} \
 	${LIBUMEM} ${LIBUTIL} ${LIBUUTIL} ${LIBZFS}
 LDADD=	-lavl -lgeom -lm -lnvpair -lsbuf -lumem -lutil -luutil -lzfs
diff -upNr a/cddl/usr.bin/ctfconvert/Makefile b/cddl/usr.bin/ctfconvert/Makefile
--- a/cddl/usr.bin/ctfconvert/Makefile	2012-03-30 18:58:53.174771000 +0000
+++ b/cddl/usr.bin/ctfconvert/Makefile	2012-03-30 19:05:49.454897012 +0000
@@ -36,6 +36,8 @@ CFLAGS+=	-I${.CURDIR}/../../../sys/cddl/
 		-I${OPENSOLARIS_USR_DISTDIR}/tools/ctf/cvt \
 		-I${OPENSOLARIS_SYS_DISTDIR}/uts/common
 
+LDFLAGS+=	-L${.CURDIR}/../../lib/libctf
+
 DPADD=		${LIBCTF} ${LIBDWARF} ${LIBELF} ${LIBZ} ${LIBPTHREAD}
 LDADD=		-lctf -ldwarf -lelf -lz -lpthread
 
diff -upNr a/cddl/usr.bin/ctfmerge/Makefile b/cddl/usr.bin/ctfmerge/Makefile
--- a/cddl/usr.bin/ctfmerge/Makefile	2012-03-30 18:58:53.174771000 +0000
+++ b/cddl/usr.bin/ctfmerge/Makefile	2012-03-30 19:06:08.084858412 +0000
@@ -33,6 +33,8 @@ CFLAGS+=	-I${.CURDIR}/../../../sys/cddl/
 		-I${OPENSOLARIS_USR_DISTDIR}/tools/ctf/cvt \
 		-I${OPENSOLARIS_SYS_DISTDIR}/uts/common
 
+LDFLAGS+=	-L${.CURDIR}/../../lib/libctf
+
 DPADD=		${LIBCTF} ${LIBDWARF} ${LIBELF} ${LIBZ} ${LIBPTHREAD}
 LDADD=		-lctf -ldwarf -lelf -lz -lpthread
 
diff -upNr a/cddl/usr.bin/zinject/Makefile b/cddl/usr.bin/zinject/Makefile
--- a/cddl/usr.bin/zinject/Makefile	2012-03-30 18:58:53.174771000 +0000
+++ b/cddl/usr.bin/zinject/Makefile	2012-03-30 19:06:25.084865592 +0000
@@ -19,6 +19,13 @@ CFLAGS+= -I${.CURDIR}/../../../sys/cddl/
 CFLAGS+= -I${.CURDIR}/../../contrib/opensolaris/head
 CFLAGS+= -I${.CURDIR}/../../lib/libumem
 
+LDFLAGS+= -L${.CURDIR}/../../lib/libavl
+LDFLAGS+= -L${.CURDIR}/../../lib/libnvpair
+LDFLAGS+= -L${.CURDIR}/../../lib/libumem
+LDFLAGS+= -L${.CURDIR}/../../lib/libuutil
+LDFLAGS+= -L${.CURDIR}/../../lib/libzfs
+LDFLAGS+= -L${.CURDIR}/../../lib/libzpool
+
 DPADD=	${LIBAVL} ${LIBGEOM} ${LIBM} ${LIBNVPAIR} ${LIBUMEM} ${LIBUUTIL} \
 	${LIBZFS} ${LIBZPOOL}
 LDADD=	-lavl -lgeom -lm -lnvpair -lumem -luutil -lzfs -lzpool
diff -upNr a/cddl/usr.bin/zstreamdump/Makefile b/cddl/usr.bin/zstreamdump/Makefile
--- a/cddl/usr.bin/zstreamdump/Makefile	2012-03-30 18:58:53.174771000 +0000
+++ b/cddl/usr.bin/zstreamdump/Makefile	2012-03-30 19:06:44.544860132 +0000
@@ -18,6 +18,11 @@ CFLAGS+= -I${.CURDIR}/../../../sys/cddl/
 CFLAGS+= -I${.CURDIR}/../../contrib/opensolaris/head
 CFLAGS+= -I${.CURDIR}/../../lib/libumem
 
+LDFLAGS+= -L${.CURDIR}/../../lib/libavl
+LDFLAGS+= -L${.CURDIR}/../../lib/libnvpair
+LDFLAGS+= -L${.CURDIR}/../../lib/libumem
+LDFLAGS+= -L${.CURDIR}/../../lib/libzpool
+
 DPADD=	${LIBM} ${LIBNVPAIR} ${LIBUMEM} ${LIBZPOOL} \
 	${LIBPTHREAD} ${LIBZ} ${LIBAVL}
 LDADD=	-lm -lnvpair -lumem -lzpool -lpthread -lz -lavl
diff -upNr a/cddl/usr.bin/ztest/Makefile b/cddl/usr.bin/ztest/Makefile
--- a/cddl/usr.bin/ztest/Makefile	2012-03-30 18:58:53.174771000 +0000
+++ b/cddl/usr.bin/ztest/Makefile	2012-03-30 19:06:58.344895642 +0000
@@ -17,6 +17,11 @@ CFLAGS+= -I${.CURDIR}/../../../sys/cddl/
 CFLAGS+= -I${.CURDIR}/../../contrib/opensolaris/head
 CFLAGS+= -I${.CURDIR}/../../lib/libumem
 
+LDFLAGS+= -L${.CURDIR}/../../lib/libavl
+LDFLAGS+= -L${.CURDIR}/../../lib/libnvpair
+LDFLAGS+= -L${.CURDIR}/../../lib/libumem
+LDFLAGS+= -L${.CURDIR}/../../lib/libzpool
+
 DPADD=	${LIBM} ${LIBNVPAIR} ${LIBUMEM} ${LIBZPOOL} \
 	${LIBPTHREAD} ${LIBZ} ${LIBAVL}
 LDADD=	-lm -lnvpair -lumem -lzpool -lpthread -lz -lavl
diff -upNr a/cddl/usr.sbin/zdb/Makefile b/cddl/usr.sbin/zdb/Makefile
--- a/cddl/usr.sbin/zdb/Makefile	2012-03-30 18:58:53.164736000 +0000
+++ b/cddl/usr.sbin/zdb/Makefile	2012-03-30 19:07:17.454861002 +0000
@@ -23,6 +23,13 @@ CFLAGS+= -I${.CURDIR}/../../../sys/cddl/
 CFLAGS+= -I${.CURDIR}/../../../cddl/contrib/opensolaris/head
 CFLAGS+= -I${.CURDIR}/../../lib/libumem
 
+LDFLAGS+= -L${.CURDIR}/../../lib/libavl
+LDFLAGS+= -L${.CURDIR}/../../lib/libnvpair
+LDFLAGS+= -L${.CURDIR}/../../lib/libumem
+LDFLAGS+= -L${.CURDIR}/../../lib/libuutil
+LDFLAGS+= -L${.CURDIR}/../../lib/libzfs
+LDFLAGS+= -L${.CURDIR}/../../lib/libzpool
+
 DPADD=	${LIBAVL} ${LIBGEOM} ${LIBM} ${LIBNVPAIR} ${LIBPTHREAD} ${LIBUMEM} \
 	${LIBUUTIL} ${LIBZ} ${LIBZFS} ${LIBZPOOL}
 LDADD=	-lavl -lgeom -lm -lnvpair -lpthread -lumem -luutil -lz -lzfs -lzpool
--- a/cddl/usr.sbin/dtrace/Makefile.old	2011-07-06 00:09:01.000000000 +0000
+++ b/cddl/usr.sbin/dtrace/Makefile	2011-07-06 00:09:48.000000000 +0000
@@ -24,6 +24,8 @@
 #CFLAGS+=	-DNEED_ERRLOC
 #YFLAGS+=	-d
 
+LDFLAGS+= -L${.CURDIR}/../../lib/libdtrace -L${.CURDIR}/../../lib/libctf
+
 DPADD=		${LIBPTHREAD} ${LIBDTRACE} ${LIBY} ${LIBL} ${LIBPROC} \
 		${LIBCTF} ${LIBELF} ${LIBZ} ${LIBUTIL} ${LIBRTLD_DB}
 LDADD=		-lpthread -ldtrace -ly -lfl -lproc -lctf -lelf -lz -lutil \
--- a/cddl/usr.sbin/lockstat/Makefile.old	2011-07-06 00:10:33.000000000 +0000
+++ b/cddl/usr.sbin/lockstat/Makefile	2011-07-06 00:10:57.000000000 +0000
@@ -23,6 +23,8 @@
 
 CFLAGS+=	-DNEED_ERRLOC -g
 
+LDFLAGS+=-L${.CURDIR}/../../lib/libdtrace -L${.CURDIR}/../../lib/libctf
+
 #YFLAGS+=	-d
 
 DPADD=		${LIBPTHREAD} ${LIBDTRACE} ${LIBY} ${LIBL} ${LIBPROC} \
