commit 03ddd0ce0864e629256bc078eaadb0d21d0df6d9
Author: Guido Draheim <guidod@gmx.de>
Date:   Sat Apr 18 02:11:37 2020 +0200

    #58 manvolnum should be in .so (see other) manpages

diff --git a/docs/dbk2man.py b/docs/dbk2man.py
index 49cac4e..e0d93d2 100755
--- a/docs/dbk2man.py
+++ b/docs/dbk2man.py
@@ -387,7 +387,7 @@ def refentry2(man, refentry, subdirectory = ".", title = ""):
                 refentrytitle = manpage
             filename = "%s/man%s/%s.%s" % (subdirectory, manvolnum, manpage, manvolnum)
             if manpage != refentrytitle:
-                manpagetext = ".so %s.%s\n" % (refentrytitle, manvolnum)
+                manpagetext = ".so man%s/%s.%s\n" % (manvolnum, refentrytitle, manvolnum)
                 writefile(filename, manpagetext)
             else:
                 manpagetext = text
