diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index b679f7b..cafa6b4 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -126,9 +126,10 @@ add_custom_command(OUTPUT manpages.tar
     COMMAND ${BASH} -c "test -d man3/man3 && mv man3 man3_; test -d man3_/man3 && mv man3_/man3 .; rm -rf man3_"
     COMMAND ${BASH} -c "chmod 664 man3/*.3"
     COMMAND ${BASH} -c "tar cf manpages.tar man3"
-    COMMAND ${BASH} -c "ls -l `pwd`/manpages.tar `pwd`/htmpages.tar >&2 || true"
+    COMMAND ${BASH} -c "ls -l `pwd`/manpages.tar >&2 || true"
     DEPENDS zziplib.xml zzipmmapped.xml zzipfseeko.xml
     VERBATIM)
+if(ZZIP_HTMPAGES)
 add_custom_command(OUTPUT htmpages.tar
     COMMAND ${BASH} -c "test -d html && rm -rf html; mkdir html"
     COMMAND ${PY} ${srcdir}/dbk2man.py -o html html zziplib.xml
@@ -139,6 +140,7 @@ add_custom_command(OUTPUT htmpages.tar
     COMMAND ${BASH} -c "ls -l `pwd`/htmpages.tar `pwd`/manpages.tar >&2 || true"
     DEPENDS zziplib.xml zzipmmapped.xml zzipfseeko.xml
     VERBATIM)
+endif()
 
 add_custom_target(manpages DEPENDS manpages.tar)
 add_custom_target(htmpages DEPENDS htmpages.tar)
@@ -190,7 +192,7 @@ if(ZZIP_HTMPAGES)
 endif()
 if(ZZIP_MANPAGES)
    install(CODE "message(STATUS \"Installing: man3 pages to ${mandir}/man3\")")
-   install(DIRECTORY man3/ DESTINATION ${mandir}/man3 MESSAGE_NEVER)
+   install(DIRECTORY ${outdir}/man3/ DESTINATION ${mandir}/man3 MESSAGE_NEVER)
 endif()
 add_custom_target(install-docs DEPENDS install-manpages install-htmpages install-site)
 add_custom_target(install-mans DEPENDS install-manpages)
