From b34cf087aaa0b02c4a9c9ab6959a3567ad31acea Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Fri, 17 Jan 2020 02:01:01 +0100
Subject: [PATCH] Disable building docs (requires python2)

---
 Makefile.am | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index c54b2cc..560c9e6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,8 +3,8 @@ ACLOCAL_AMFLAGS = -I m4
 WANT_AUTOMAKE = 1.7
 WANT_AUTOCONF = 2.57
 
-DIST_SUBDIRS = zzip zzipwrap bins test docs  SDL
-SUBDIRS =      zzip zzipwrap bins test docs @SDL@
+DIST_SUBDIRS = zzip zzipwrap bins test  SDL
+SUBDIRS =      zzip zzipwrap bins test @SDL@
 
 rpm: dist-bzip2 $(PACKAGE).spec
 	rpmbuild -ta $(PACKAGE)-$(VERSION).tar.bz2
@@ -62,9 +62,9 @@ osc-distclean:
 indent-check:
 	(cd zzip && $(MAKE) `basename $@`)
 
-doc docs docu clean-doc clean-docs clean-docu   zzip.html zzip.xml zzip.pdf \
+doc docu clean-doc clean-clean-docu   zzip.html zzip.xml zzip.pdf \
 man mans manpages htmpages unpack clean-unpack  changes.htm pdfs \
-omf install-omf install-doc install-docs install-man3 install-mans \
+omf install-omf install-doc install-install-man3 install-mans \
 upload-sourceforge www:
 	(cd docs && $(MAKE) `basename $@`)
 
@@ -104,7 +104,7 @@ msvc6/zziptest.dsp     msvc6/zzip.dsp
 EXTRA_DIST =   zziplib.spec zzipback.sed Makefile.mk \
                $(MSVC8) $(MSVC7) $(MSVC6) $(am__aclocal_m4_deps)
 
-PHONY = auto boottrap rpm doc docs man manpages htmpages sdl testsdl \
+PHONY = auto boottrap rpm doc man manpages htmpages sdl testsdl \
 	comp compats msvc6 msvc7
 # ------------------------------------------------------------------------
 
-- 
2.25.0

