From 31c9ad99ac3dbd34ad8c4285a689c5028490f776 Mon Sep 17 00:00:00 2001
From: Christian Gutschow <chris.g@cern.ch>
Date: Wed, 5 Feb 2025 19:02:38 +0000
Subject: [PATCH] add missing header

---
 src/zstr/zstr.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/zstr/zstr.hpp b/src/zstr/zstr.hpp
index 66c21f847..61c587345 100644
--- a/src/zstr/zstr.hpp
+++ b/src/zstr/zstr.hpp
@@ -9,6 +9,7 @@
 #pragma once
 
 #include <cassert>
+#include <cstdint>
 #include <fstream>
 #include <sstream>
 #include <zlib.h>
-- 
GitLab

