--- configure.old	2009-02-03 14:46:21.000000000 +0100
+++ configure	2009-02-03 14:48:13.000000000 +0100
@@ -638,7 +638,7 @@
 
 print CONFIGMAKE "CXX       = $cxx\n";
 print CONFIGMAKE "CXXFLAGS  = @cxxflags\n";
-print CONFIGMAKE "LDFLAGS   = @ldflags\n";
+print CONFIGMAKE "LDLIBS    = @ldflags\n";
 
 print CONFIGMAKE "prefix    = \${DESTDIR}$prefix\n";
 print CONFIGMAKE "bindir    = \${DESTDIR}$prefix/bin\n";
--- elifekam.old	2009-02-03 14:48:43.000000000 +0100
+++ elifekam	2009-02-03 14:48:58.000000000 +0100
@@ -24,7 +24,7 @@
 all: eboard nls-dicts
 
 eboard: $(OBJS)
-	$(CXX) $(LDFLAGS) -o eboard $(OBJS)
+	$(CXX) $(LDFLAGS) -o eboard $(OBJS) $(LDLIBS)
 
 .cc.o: $< $(HEADERS) $(XPMS)
 	$(CXX) $(CXXFLAGS) -c $< -o $@
