From 32aef89113a1939b20766bf0e09cce7287b56f40 Mon Sep 17 00:00:00 2001
From: Jonathan Esk-Riddell <jr@jriddell.org>
Date: Fri, 21 Oct 2022 12:03:40 +0100
Subject: [PATCH] fix version check which breaks on KF5 5.100

---
 src/newstuff/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/newstuff/CMakeLists.txt b/src/newstuff/CMakeLists.txt
index b51914fd..c0ebfa54 100644
--- a/src/newstuff/CMakeLists.txt
+++ b/src/newstuff/CMakeLists.txt
@@ -19,7 +19,7 @@ TARGET_LINK_LIBRARIES(newstuff
     )
 
 IF( KF5NewStuff_FOUND )
-  if(${ECM_VERSION} STRGREATER "5.57.0")
+  if(${ECM_VERSION} VERSION_GREATER "5.57.0")
     INSTALL(FILES tellico-template.knsrc tellico-script.knsrc DESTINATION ${KDE_INSTALL_KNSRCDIR})
   else()
     INSTALL(FILES tellico-template.knsrc tellico-script.knsrc DESTINATION ${KDE_INSTALL_CONFDIR})
-- 
GitLab

