From 2896800b5e7d0c656710c70fdea57098032f3ccc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=98yvind=20Kol=C3=A5s?= <pippin@gimp.org>
Date: Sun, 6 May 2018 15:09:03 +0200
Subject: ff-save: update another occurance of CODEC_FLAG_GLOBAL_HEADER

---
 operations/external/ff-save.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/operations/external/ff-save.c b/operations/external/ff-save.c
index 1edac31..f30bf10 100644
--- a/operations/external/ff-save.c
+++ b/operations/external/ff-save.c
@@ -638,7 +638,7 @@ add_video_stream (GeglProperties *o, AVFormatContext * oc, int codec_id)
 #endif
 
    if (oc->oformat->flags & AVFMT_GLOBALHEADER)
-     c->flags |= CODEC_FLAG_GLOBAL_HEADER;
+     c->flags |= AV_CODEC_FLAG_GLOBAL_HEADER;
 
   return st;
 }
-- 
cgit v0.12

