Index: python-dateutil/example.py
===================================================================
--- python-dateutil.orig/example.py	2006-09-06 13:21:44.000000000 +0200
+++ python-dateutil/example.py	2006-09-06 13:21:56.000000000 +0200
@@ -5,7 +5,7 @@
 from datetime import *
 import commands
 import os
-now = parse(commands.getoutput("date"))
+now = parse(commands.getoutput("LC_ALL=C date"))
 today = now.date()
 year = rrule(YEARLY,bymonth=8,bymonthday=13,byweekday=FR)[0].year
 rdelta = relativedelta(easter(year), today)
