Index: image_cb_detector/src/rgbd_cb_detector_action.cpp
===================================================================
--- image_cb_detector.orig/src/rgbd_cb_detector_action.cpp
+++ image_cb_detector/src/rgbd_cb_detector_action.cpp
@@ -143,7 +143,7 @@ public:
       for(size_t i = 0; i< features.image_points.size(); i++){
         geometry_msgs::Point pixel = features.image_points[i];
         float depth = *(depth_ptr+width*(unsigned int)pixel.y+(unsigned int)pixel.x);
-        if ( isnan(depth) )
+        if ( std::isnan(depth) )
         {
           continue;
         }
