Index: image_cb_detector/src/image_cb_detector/cb_detector.py
===================================================================
--- image_cb_detector.orig/src/image_cb_detector/cb_detector.py
+++ image_cb_detector/src/image_cb_detector/cb_detector.py
@@ -39,7 +39,7 @@ NAME = 'image_cb_detector_node'
 import roslib; roslib.load_manifest(PKG)
 
 import rospy
-import cv
+import cv2 as cv
 import math
 
 from sensor_msgs.msg import Image
@@ -130,7 +130,7 @@ class ImageCbDetectorNode:
     #we need to convert the ros image to an opencv image
     try:
       image = self.bridge.imgmsg_to_cv(ros_image, "mono8")
-    except CvBridgeError, e:
+    except CvBridgeError as e:
       rospy.logerror("Error importing image %s" % e)
       return
 
