diff -ruN enscript-1.6.5.2.orig/states/hl/enscript.st enscript-1.6.5.2/states/hl/enscript.st
--- enscript-1.6.5.2.orig/states/hl/enscript.st	2010-06-02 00:17:49.000000000 +0200
+++ enscript-1.6.5.2/states/hl/enscript.st	2011-06-02 12:43:13.869855504 +0200
@@ -510,6 +510,7 @@
   /\.tex$/					tex;
   /\.wmlscript$/				wmlscript;
   /\.wmls$/					wmlscript;
+  /\.php$/					php;
   /^.*$/					passthrough;
 }
 
