--- include/unixconf.h.orig	2003-08-07 23:18:03.070299152 -0400
+++ include/unixconf.h	2003-08-07 23:18:39.699730632 -0400
@@ -96,13 +96,13 @@
  */
 
 /* #define NO_FILE_LINKS */	/* if no hard links */
-/* #define LOCKDIR "/usr/games/lib/nethackdir" */	/* where to put locks */
+#define LOCKDIR "GENTOO_STATEDIR"
 
 /*
  * If you want the static parts of your playground on a read-only file
  * system, define VAR_PLAYGROUND to be where the variable parts are kept.
  */
-/* #define VAR_PLAYGROUND "/var/lib/games/nethack" */
+#define VAR_PLAYGROUND "GENTOO_STATEDIR"
 
 
 /*
