diff --git a/aegisub/configure.ac b/aegisub/configure.ac
index d961bd6..17df012 100644
--- a/aegisub/configure.ac
+++ b/aegisub/configure.ac
@@ -369,7 +369,7 @@ AC_ARG_WITH(lua,
             AS_HELP_STRING([--without-lua],
                            [build without lua 5.1 (auto4)]))

-AS_IF([test x$with_lua = xno], [$with_auto4_lua="no"; lua_disabled="(disabled)"], [
+AS_IF([test x$with_lua = xno], [with_auto4_lua="no"; lua_disabled="(disabled)"], [
   # First try pkg-config with several names
   PKG_CHECK_MODULES(LUA, lua5.1  >= 5.1 lua5.1  < 5.2, [with_auto4_lua="yes"], [
   PKG_CHECK_MODULES(LUA, lua-5.1 >= 5.1 lua-5.1 < 5.2, [with_auto4_lua="yes"], [
