--- llvm-3.7.0rc3.src-orig/cmake/modules/AddOCaml.cmake	2015-06-08 05:22:22.000000000 +1000
+++ llvm-3.7.0rc3.src/cmake/modules/AddOCaml.cmake	2015-08-30 19:22:37.216580808 +1000
@@ -131,7 +131,7 @@
   if( APPLE )
     set(ocaml_rpath "@executable_path/../../lib")
   elseif( UNIX )
-    set(ocaml_rpath "\\$ORIGIN/../../lib")
+    set(ocaml_rpath "\\$ORIGIN/../../${CMAKE_INSTALL_LIBDIR}")
   endif()
   list(APPEND ocaml_flags "-ldopt" "-Wl,-rpath,${ocaml_rpath}")
 
@@ -183,12 +183,12 @@
   endforeach()
 
   install(FILES ${install_files}
-          DESTINATION lib/ocaml)
+          DESTINATION ${CMAKE_INSTALL_LIBDIR}/ocaml)
   install(FILES ${install_shlibs}
           PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
                       GROUP_READ GROUP_EXECUTE
                       WORLD_READ WORLD_EXECUTE
-          DESTINATION lib/ocaml)
+          DESTINATION ${CMAKE_INSTALL_LIBDIR}/ocaml)
 
   foreach( install_file ${install_files} ${install_shlibs} )
     get_filename_component(filename "${install_file}" NAME)
--- llvm-3.7.0rc3.src-orig/bindings/ocaml/backends/CMakeLists.txt	2014-12-30 14:24:07.000000000 +1100
+++ llvm-3.7.0rc3.src/bindings/ocaml/backends/CMakeLists.txt	2015-08-30 19:23:49.726583886 +1000
@@ -23,5 +23,5 @@
     "${LLVM_LIBRARY_DIR}/ocaml/META.llvm_${TARGET}")
 
   install(FILES "${LLVM_LIBRARY_DIR}/ocaml/META.llvm_${TARGET}"
-          DESTINATION lib/ocaml)
+          DESTINATION ${CMAKE_INSTALL_LIBDIR}/ocaml)
 endforeach()
--- llvm-3.7.0rc3.src-orig/bindings/ocaml/llvm/CMakeLists.txt	2014-12-30 14:24:07.000000000 +1100
+++ llvm-3.7.0rc3.src/bindings/ocaml/llvm/CMakeLists.txt	2015-08-30 19:23:43.561583624 +1000
@@ -8,4 +8,4 @@
   "${LLVM_LIBRARY_DIR}/ocaml/META.llvm")
 
 install(FILES "${LLVM_LIBRARY_DIR}/ocaml/META.llvm"
-        DESTINATION lib/ocaml)
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/ocaml)
