--- VirtualBox-6.0.0_BETA2/src/VBox/Main/include/ConsoleImpl.h
+++ VirtualBox-6.0.0_BETA2/src/VBox/Main/include/ConsoleImpl.h
@@ -147,9 +147,13 @@
     int i_recordingGetSettings(settings::RecordingSettings &Settings);
     int i_recordingStart(void);
     int i_recordingStop(void);
+# ifdef VBOX_WITH_AUDIO_RECORDING
     AudioVideoRec *i_recordingGetAudioDrv(void) const { return Recording.mAudioRec; }
+# endif
     RecordingContext *i_recordingGetContext(void) const { return Recording.mpCtx; }
+# ifdef VBOX_WITH_AUDIO_RECORDING
     HRESULT i_recordingSendAudio(const void *pvData, size_t cbData, uint64_t uDurationMs);
+# endif
 #endif
 
     const ComPtr<IMachine> &i_machine() const { return mMachine; }
@@ -1038,7 +1042,10 @@
     {
         Recording()
             : mpCtx(NULL)
-            , mAudioRec(NULL) { }
+# ifdef VBOX_WITH_AUDIO_RECORDING
+            , mAudioRec(NULL)
+# endif
+        { }
 
         /** The recording context. */
         RecordingContext       *mpCtx;
--- VirtualBox-6.0.0_BETA2/src/VBox/Main/src-client/Recording.cpp
+++ VirtualBox-6.0.0_BETA2/src/VBox/Main/src-client/Recording.cpp
@@ -587,7 +587,7 @@
 
     return rc;
 #else
-    RT_NOREF(pCtx, pvData, cbData, uTimeStampMs);
+    RT_NOREF(pvData, cbData, uTimeStampMs);
     return VINF_SUCCESS;
 #endif
 }
