--- vector-algorithms-0.5.4.1-orig/vector-algorithms.cabal	2012-08-16 06:39:54.000000000 +1000
+++ vector-algorithms-0.5.4.1/vector-algorithms.cabal	2012-08-17 17:05:37.875993915 +1000
@@ -52,9 +52,15 @@
         FlexibleContexts,
         CPP
 
-    GHC-Options:
-        -Odph
-        -funbox-strict-fields
+    if impl(ghc >= 7.4.0)
+        GHC-Options:
+            -Odph
+            -funbox-strict-fields
+            -fsimpl-tick-factor=153
+    else
+        GHC-Options:
+           -Odph
+           -funbox-strict-fields
 
     Include-Dirs:
         include
