--- polyml-5.6-1-orig/sha1/build	2015-09-09 03:05:54.000000000 +1000
+++ polyml-5.6-1/sha1/build	2016-12-28 19:10:36.148579115 +1100
@@ -32,34 +32,34 @@
 
 case "$target" in
   x86-linux)
-    CFLAGS="-fPIC -I. -m32"
-    LDFLAGS="-fPIC -m32 -shared"
+    CFLAGS="-O2 -march=native -ggdb -fvar-tracking-assignments -fvar-tracking -pipe -fPIC -I. -m32"
+    LDFLAGS="-Wl,-O1 -Wl,--as-needed -fPIC -m32 -shared"
     library="$target/libsha1.so"
     test_sha1="test_sha1"
     ;;
   x86_64-linux)
-    CFLAGS="-fPIC -I. -m64"
-    LDFLAGS="-fPIC -m64 -shared"
+    CFLAGS="-O2 -march=native -ggdb -fvar-tracking-assignments -fvar-tracking -pipe -fPIC -I. -m64"
+    LDFLAGS="-Wl,-O1 -Wl,--as-needed -fPIC -m64 -shared"
     library="$target/libsha1.so"
     test_sha1="test_sha1"
     ;;
   x86-darwin)
     LD=libtool
-    CFLAGS="-fPIC -I. -m32"
-    LDFLAGS="-dynamic -lc"
+    CFLAGS="-O2 -march=native -ggdb -fvar-tracking-assignments -fvar-tracking -pipe -fPIC -I. -m32"
+    LDFLAGS="-Wl,-O1 -Wl,--as-needed -dynamic -lc"
     library="$target/libsha1.so"
     test_sha1="test_sha1"
     ;;
   x86_64-darwin)
     LD=libtool
-    CFLAGS="-fPIC -I. -m64"
-    LDFLAGS="-dynamic -lc"
+    CFLAGS="-O2 -march=native -ggdb -fvar-tracking-assignments -fvar-tracking -pipe -fPIC -I. -m64"
+    LDFLAGS="-Wl,-O1 -Wl,--as-needed -dynamic -lc"
     library="$target/libsha1.so"
     test_sha1="test_sha1"
     ;;
   x86-cygwin)
-    CFLAGS="-I. -m32"
-    LDFLAGS="-shared"
+    CFLAGS="-O2 -march=native -ggdb -fvar-tracking-assignments -fvar-tracking -pipe -I. -m32"
+    LDFLAGS="-Wl,-O1 -Wl,--as-needed -shared"
     library="$target/sha1.dll"
     test_sha1="test_sha1.exe"
     ;;
@@ -105,10 +105,6 @@
 $CC $CFLAGS -o $test_sha1 test_sha1.c -ldl
 [ "$?" -ne 0 ] && { exit 1; }
 
-echo "Running tests ..."
-./$test_sha1 $library
-[ "$?" -ne 0 ] && { exit 1; }
-
-rm test_sha1 sha1.o
+exit 0
 
 fi
\ No newline at end of file
