Modify AddStdCatalog to use the versioned wxstd32-gtk3.mo.
--- a/src/common/translation.cpp
+++ b/src/common/translation.cpp
@@ -1409,7 +1409,7 @@ bool wxTranslations::AddStdCatalog()
     if ( AddCatalog("wxstd-" wxSTRINGIZE(wxMAJOR_VERSION) "." wxSTRINGIZE(wxMINOR_VERSION)) )
         return true;

-    if ( AddCatalog(wxS("wxstd")) )
+    if ( AddCatalog(wxS("wxstd32-gtk3")) )
         return true;

     return false;
