From 91023ef967c680db9307b5c58762e8872346167e Mon Sep 17 00:00:00 2001
From: Paul Varner <fuzzyray@gentoo.org>
Date: Fri, 24 Oct 2014 12:59:59 -0500
Subject: [PATCH 06/14] Fix gawk warning escape sequence \. treated as plain.
 Bug 526400

---
 bin/revdep-rebuild.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/revdep-rebuild.sh b/bin/revdep-rebuild.sh
index 3179a83..01a0454 100755
--- a/bin/revdep-rebuild.sh
+++ b/bin/revdep-rebuild.sh
@@ -842,7 +842,7 @@ main_checks() {
 			done < <(
 				# Regexify LD_LIBRARY_MASK. Exclude it from the search.
 				LD_LIBRARY_MASK="${LD_LIBRARY_MASK//$'\n'/|}"
-				gawk -v ldmask="(${LD_LIBRARY_MASK//./\\\.})" '
+				gawk -v ldmask="(${LD_LIBRARY_MASK//./\\\\.})" '
 					/no version information available/ && $0 !~ ldmask {
 						gsub(/[()]/, "", $NF)
 						if (seen[$NF]++)  next
-- 
2.1.3

