diff -urN krename-4.0.9.old/src/krename.desktop krename-4.0.9/src/krename.desktop
--- krename-4.0.9.old/src/krename.desktop	2007-03-18 17:54:10.000000000 +0100
+++ krename-4.0.9/src/krename.desktop	2014-04-29 18:37:34.171237934 +0200
@@ -1,8 +1,7 @@
 [Desktop Entry]
 Type=Application
-Exec=krename -caption "%c" %i %m %U
-Icon=krename.png
-MiniIcon=krename.png
+Exec=krename -caption "%c" %i %U
+Icon=krename
 DocPath=krename/index.html
 Comment=A batch renamer
 Comment[de]=Ein Batch Umbenenner
