--- epm.orig	2012-12-27 21:21:51.288350173 -0700
+++ epm	2012-12-27 21:26:51.429905384 -0700
@@ -7,13 +7,13 @@
 # Global vars
 (my $version = '$Revision: 1.2 $') =~ s/.*?(\d.*\d).*/EPM version $1/;
 my $verbose = 0;
-my $dbpath = '/var/db/pkg';
+my $dbpath = '@GENTOO_PORTAGE_EPREFIX@/var/db/pkg';
 my $pkgregex =
     '^.+?\/'.					# group (ignored)
     '(.+?)'.					# name
     '-(\d+(?:\.\d+)*\w*)'.			# version, eg 1.23.4a
     '((?:(?:_alpha|_beta|_pre|_rc)\d*)?)'.	# special suffix
-    '((?:-r\d+)?)$';				# revision, eg r12
+    '((?:-r\d+)?(?:\.\d+)?)$';			# revision, eg r12 or r01.1
 my $root = '/';
 my %opt = (
     'dbpath' => \$dbpath,
@@ -267,11 +267,11 @@
     if ($opt{'c'}) {
 	# Read in CONFIG_PROTECT from /etc/make.{global,conf}
 	my (@CONFIG_PROTECT) = split ' ',
-	    `. /etc/make.globals; . /etc/make.conf; echo \$CONFIG_PROTECT`;
+	    `. @GENTOO_PORTAGE_EPREFIX@/etc/make.globals; .  @GENTOO_PORTAGE_EPREFIX@/etc/make.conf; echo \$CONFIG_PROTECT`;
 	die "CONFIG_PROTECT is empty" unless @CONFIG_PROTECT;
 	my ($confprotre) = join '|', @CONFIG_PROTECT;
 	@files = grep {
-	    (split ' ', $_, 2)[1] =~ /^($confprotre)/o } @files;
+	    (split ' ', $_, 2)[1] =~ m!^@GENTOO_PORTAGE_EPREFIX@($confprotre)!o } @files;
     }
 
     # Trim @files if doc files requested
