commit 8b54a592731e65dec02bbfc83c994fe373bc0dca
Author: Raymond Wooninck <tittiatcoke@gmail.com>
Date:   Tue Jan 24 20:55:28 2017 +0100

    Fix build with -fno-operator-names

diff --git a/DB/ImageInfo.cpp b/DB/ImageInfo.cpp
index 59c2c6f9..5eadbb10 100644
--- a/DB/ImageInfo.cpp
+++ b/DB/ImageInfo.cpp
@@ -190,7 +190,7 @@ void ImageInfo::rotate( int degrees, RotationMode mode )
     m_dirty = true;
     m_angle = ( m_angle + degrees ) % 360;
 
-    if (degrees == 90 or degrees == 270) {
+    if (degrees == 90 || degrees == 270) {
         m_size.transpose();
     }
 
