--- libsvgtiny-0.1.2/Makefile
+++ libsvgtiny-0.1.2/Makefile
@@ -15,14 +15,8 @@
 WARNFLAGS := -Wall -W -Wundef -Wpointer-arith -Wcast-align \
 	-Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes \
 	-Wmissing-declarations -Wnested-externs -pedantic
-# BeOS/Haiku/AmigaOS standard library headers create warnings
-ifneq ($(TARGET),beos)
-  ifneq ($(TARGET),amiga)
-    WARNFLAGS := $(WARNFLAGS) -Werror
-  endif
-endif
 
-CFLAGS := -D_BSD_SOURCE -I$(CURDIR)/include/ \
+CFLAGS := -D_DEFAULT_SOURCE -I$(CURDIR)/include/ \
 	-I$(CURDIR)/src $(WARNFLAGS) $(CFLAGS)
 ifneq ($(GCCVER),2)
   CFLAGS := $(CFLAGS) -std=c99
