diff --git a/thrust/type_traits/is_contiguous_iterator.h b/thrust/type_traits/is_contiguous_iterator.h
index 252ed69..c542929 100644
--- a/thrust/type_traits/is_contiguous_iterator.h
+++ b/thrust/type_traits/is_contiguous_iterator.h
@@ -131,7 +131,7 @@ struct is_libcxx_wrap_iter : false_type {};
 #if defined(_LIBCPP_VERSION)
 template <typename Iterator>
 struct is_libcxx_wrap_iter<
-#  if _LIBCPP_VERSION < 14000 || THRUST_DEVICE_COMPILER == THRUST_DEVICE_COMPILER_HIP
+#  if _LIBCPP_VERSION < 14000
   _VSTD::__wrap_iter<Iterator>
 #  else
   std::__wrap_iter<Iterator>
