https://github.com/ROCmSoftwarePlatform/rocThrust/issues/180

--- orig/thrust/system/hip/detail/guarded_driver_types.h
+++ rocThrust-rocm-4.3.0/thrust/system/hip/detail/guarded_driver_types.h
@@ -47,7 +47,7 @@
 #  endif // __DRIVER_TYPES_H__
 #endif // __GNUC__
 
-#include <hip/hcc_detail/host_defines.h>
+#include <hip/amd_detail/host_defines.h>
 
 #if !defined(__GNUC__) || ((10000 * __GNUC__ + 100 * __GNUC_MINOR__ + __GNUC_PATCHLEVEL__) >= 40500)
 #  ifdef THRUST_HOST_NEEDS_RESTORATION
