commit fb4234cd72ceebb200bef13234ea9c36c9a62046
Author: Kevin Funk <kevin@kfunk.org>
Date:   Sun Nov 3 22:09:56 2013 +0100

    Check if kdevplatform has JSON support
    
    BUG: 327095

diff --git a/languages/cpp/tests/CMakeLists.txt b/languages/cpp/tests/CMakeLists.txt
index 84ce3ee..fbdd4db 100644
--- a/languages/cpp/tests/CMakeLists.txt
+++ b/languages/cpp/tests/CMakeLists.txt
@@ -55,7 +55,9 @@ target_link_libraries(buddiestest
 ########### next target ###############
 
 find_package(QJSON)
-if(QJSON_FOUND)
+# Also check that kdevplatform is built with JSON support
+# see: https://bugs.kde.org/show_bug.cgi?id=327095
+if(QJSON_FOUND AND KDEVPLATFORM_JSONTESTS_LIBRARIES)
   set(cppfilestest_SRCS test_cppfiles.cpp)
   kde4_add_unit_test(cppfilestest ${cppfilestest_SRCS})
   configure_file("testfilepaths.h.cmake" "testfilepaths.h" ESCAPE_QUOTES)
@@ -67,7 +69,7 @@ if(QJSON_FOUND)
       ${KDEVPLATFORM_JSONTESTS_LIBRARIES}
       kdev4cppduchain
   )
-endif(QJSON_FOUND)
+endif()
 
 ########### next target ###############
 
