Patch by Andrej Kacian <ticho@gentoo.org> from bug #177587


diff -ur lcd-stuff-0.1.2-orig/src/mpd.c lcd-stuff-0.1.2/src/mpd.c
--- lcd-stuff-0.1.2-orig/src/mpd.c	2007-05-10 23:26:55.000000000 +0200
+++ lcd-stuff-0.1.2/src/mpd.c	2007-05-10 23:27:59.000000000 +0200
@@ -119,14 +119,13 @@
             case MPD_DATA_TYPE_PLAYLIST:
                 g_ptr_array_add(array, g_path_get_basename(data->playlist));
                 g_free(data->playlist);
+                data->playlist = NULL;
                 break;
 
             case MPD_DATA_TYPE_DIRECTORY:
-                g_free(data->directory);
-                break;
-
             case MPD_INFO_ENTITY_TYPE_SONG:
                 g_free(data->directory);
+                data->directory = NULL;
                 break;
 
             default:
