--- a/configure.ac
+++ b/configure.ac
@@ -46,16 +46,15 @@
 		])
 fi
 
-CPPFLAGS="$CPPFLAGS $WX_CPPFLAGS"
-CXXFLAGS="$CXXFLAGS $WX_CXXFLAGS_ONLY"
-CFLAGS="$CFLAGS $WX_CFLAGS_ONLY"
-LIBS="$LIBS $WX_LIBS"
+#CPPFLAGS="$CPPFLAGS $WX_CPPFLAGS"
+#CXXFLAGS="$CXXFLAGS $WX_CXXFLAGS_ONLY"
+#CFLAGS="$CFLAGS $WX_CFLAGS_ONLY"
+#LIBS="$LIBS $WX_LIBS"
 
 AM_CPPFLAGS="$AM_CPPFLAGS $GL_CFLAGS $GLU_CFLAGS"
-AM_LDFLAGS="$AM_LDFLAGS $GL_LIBS $GLU_LIBS"
 AC_SUBST([AM_CPPFLAGS])
 AC_SUBST([AM_LDFLAGS])
-AC_SUBST([$PREFIX])
+#AC_SUBST([$PREFIX])
 
 AC_OUTPUT
 
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,8 +1,10 @@
-fe3d_CPPFLAGS	= -DFE_DATA_DIR="\"@datadir@/@PACKAGE@/\""
+fe3d_CPPFLAGS	= -DFE_DATA_DIR="\"@datadir@/@PACKAGE@/\"" @WX_CPPFLAGS@
 INCLUDES		= -I$(top_srcdir)/include -I$(top_srcdir)/src/include
 bin_PROGRAMS	= fe3d
 fe3d_SOURCES	= fe_config.cpp fe_core.cpp  fe_draw.cpp fe_events.cpp fe_host.cpp fe_hosts.cpp fe_match.cpp fe_node.cpp fe_os_rgx.cpp fe_port.cpp fe_root_node.cpp fe_tools.cpp geo_cache.cpp gl_util.cpp libfe3d.cpp nmap_parser.cpp tex_cache.cpp wxfe3d_app.cpp wxfe3d_frame.cpp wxfe3d_glcanvas.cpp ui_nmap_ctrl.cpp ./gui/fe3d_ui.cpp ./regexp/Matcher.cpp ./regexp/Pattern.cpp ./regexp/WCMatcher.cpp ./regexp/WCPattern.cpp
-
+fe3d_CFLAGS 	= @WX_CFLAGS@
+fe3d_CXXFLAGS 	= @WX_CXXFLAGS_ONLY@
+fe3d_LDADD 	= @GLU_LIBS@ @WX_LIBS@
 fe3ddir		= @datadir@/@PACKAGE@
 fe3d_imgdir  	= @datadir@/@PACKAGE@/img
 fe3d_DATA	= $(top_srcdir)/data/*
