--- a/CMakeLists.txt	2020-08-22 23:59:26.674785914 +0200
+++ b/CMakeLists.txt	2020-08-23 00:01:18.491783097 +0200
@@ -253,8 +253,8 @@
 
 install( TARGETS amdrocclr_static
   EXPORT rocclr
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib64
+  LIBRARY DESTINATION lib64
   COMPONENT applications)
 
 install (
@@ -262,31 +262,31 @@
    DESTINATION include)
 install (
    DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/elf"
-   DESTINATION include
+   DESTINATION include/rocclr
    FILES_MATCHING PATTERN "*.h*")
 install (
    DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/compiler"
-   DESTINATION include
+   DESTINATION include/rocclr
    FILES_MATCHING PATTERN "*.h*")
 install (
    DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/utils"
-   DESTINATION include
+   DESTINATION include/rocclr
    FILES_MATCHING PATTERN "*.h*")
 install (
    DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/platform"
-   DESTINATION include
+   DESTINATION include/rocclr
    FILES_MATCHING PATTERN "*.h*")
 install (
    DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/thread"
-   DESTINATION include
+   DESTINATION include/rocclr
    FILES_MATCHING PATTERN "*.h*")
 install (
    DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/os"
-   DESTINATION include
+   DESTINATION include/rocclr
    FILES_MATCHING PATTERN "*.h*")
 install (
    DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/device"
-   DESTINATION include
+   DESTINATION include/rocclr
    FILES_MATCHING PATTERN "*.h*")
 
 # Generate the install-tree package.
