From 59dceeb53196063c568201d0c567bea7887b1cc4 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Sat, 3 Jul 2021 20:32:11 +0200
Subject: [PATCH] Remove unused KService dep

---
 src/CMakeLists.txt     | 1 -
 src/statusnotifier.cpp | 1 -
 2 files changed, 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1c176188..6ec2120e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -96,7 +96,6 @@ target_link_libraries(
     DrKonqiInternal
     KF5::I18n
     KF5::CoreAddons
-    KF5::Service
     KF5::ConfigWidgets
     KF5::JobWidgets
     KF5::KIOCore
diff --git a/src/statusnotifier.cpp b/src/statusnotifier.cpp
index 41271874..c5f795c0 100644
--- a/src/statusnotifier.cpp
+++ b/src/statusnotifier.cpp
@@ -17,7 +17,6 @@
 #include <KIdleTime>
 #include <KLocalizedString>
 #include <KNotification>
-#include <KService>
 #include <KStatusNotifierItem>
 
 #include "crashedapplication.h"
-- 
GitLab

