--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -222,28 +222,6 @@ ENDIF()
 OPTION(FORCE_UNSUPPORTED_COMPILER "Disable compiler version checks" OFF)
 MARK_AS_ADVANCED(WITHOUT_SERVER FORCE_UNSUPPORTED_COMPILER)
 
-# Use 'uname -r' and 'rpm -qf /' to figure out host system.
-# For Docker images we cannot trust uname, so use rpm instead.
-IF(UNIX)
-  FIND_PROGRAM(MY_UNAME uname /bin /usr/bin /usr/local/bin /sbin)
-  IF(MY_UNAME)
-    EXECUTE_PROCESS(COMMAND ${MY_UNAME} -s
-      OUTPUT_VARIABLE MY_HOST_SYSTEM_NAME
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-      RESULT_VARIABLE MY_UNAME_RESULT
-      )
-    EXEC_PROGRAM(${MY_UNAME} ARGS -r OUTPUT_VARIABLE MY_HOST_SYSTEM_VERSION)
-  ENDIF()
-  FIND_PROGRAM(MY_RPM rpm /bin /usr/bin)
-  IF(MY_RPM)
-    EXECUTE_PROCESS(COMMAND ${MY_RPM} -qf /
-      OUTPUT_VARIABLE MY_HOST_FILESYSTEM_NAME
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-      RESULT_VARIABLE MY_RPM_RESULT
-      )
-  ENDIF()
-ENDIF()
-
 # See comments above, about CMAKE_MINIMUM_REQUIRED VERSION and macOS 11.
 IF(APPLE)
   IF(MY_HOST_SYSTEM_VERSION VERSION_GREATER_EQUAL 20)
