--- m4/kpse-freetype2-flags.m4	2018-09-19 10:19:06.437789178 +0100
+++ m4/kpse-freetype2-flags.m4	2018-09-19 10:23:48.556050046 +0100
@@ -21,17 +21,10 @@
 
 # KPSE_FREETYPE2_OPTIONS([WITH-SYSTEM])
 # -------------------------------------
-AC_DEFUN([KPSE_FREETYPE2_OPTIONS], [_KPSE_LIB_OPTIONS([freetype2], [$1], [freetype-config])])
+AC_DEFUN([KPSE_FREETYPE2_OPTIONS], [_KPSE_LIB_OPTIONS([freetype2], [$1], [pkg-config])])
 
 # KPSE_FREETYPE2_SYSTEM_FLAGS
 # ---------------------------
 AC_DEFUN([KPSE_FREETYPE2_SYSTEM_FLAGS], [dnl
-AC_REQUIRE([AC_CANONICAL_HOST])[]dnl
-AC_CHECK_TOOL([FT2_CONFIG], [freetype-config], [false])[]dnl
-if $FT2_CONFIG --ftversion >/dev/null 2>&1; then
-  FREETYPE2_INCLUDES=`$FT2_CONFIG --cflags`
-  FREETYPE2_LIBS=`$FT2_CONFIG --libs`
-elif test "x$need_freetype2:$with_system_freetype2" = xyes:yes; then
-  AC_MSG_ERROR([did not find freetype-config required for system freetype2 library])
-fi
-]) # KPSE_FREETYPE2_SYSTEM_FLAGS
+_KPSE_PKG_CONFIG_FLAGS([freetype2], [freetype2])])
+
