--- a/src/IVtkVTK/IVtkVTK_ShapeData.cxx
+++ b/src/IVtkVTK/IVtkVTK_ShapeData.cxx
@@ -80,9 +80,9 @@ void IVtkVTK_ShapeData::InsertVertex (const IVtk_IdType theShapeID,
   vtkIdType aPointIdVTK = thePointId;
   myPolyData->InsertNextCell (VTK_VERTEX, 1, &aPointIdVTK);
   const vtkIdType aShapeIDVTK = theShapeID;
-  mySubShapeIDs->InsertNextTupleValue (&aShapeIDVTK);
+  mySubShapeIDs->InsertNextTypedTuple (&aShapeIDVTK);
   const vtkIdType aType = theMeshType;
-  myMeshTypes->InsertNextTupleValue (&aType);
+  myMeshTypes->InsertNextTypedTuple (&aType);
 }
 
 //================================================================
@@ -97,9 +97,9 @@ void IVtkVTK_ShapeData::InsertLine (const IVtk_IdType   theShapeID,
   vtkIdType aPoints[2] = { thePointId1, thePointId2 };
   myPolyData->InsertNextCell (VTK_LINE, 2, aPoints);
   const vtkIdType aShapeIDVTK = theShapeID;
-  mySubShapeIDs->InsertNextTupleValue (&aShapeIDVTK);
+  mySubShapeIDs->InsertNextTypedTuple (&aShapeIDVTK);
   const vtkIdType aType = theMeshType;
-  myMeshTypes->InsertNextTupleValue (&aType);
+  myMeshTypes->InsertNextTypedTuple (&aType);
 }
 
 //================================================================
@@ -124,9 +124,9 @@ void IVtkVTK_ShapeData::InsertLine (const IVtk_IdType       theShapeID,
 
     myPolyData->InsertNextCell (VTK_POLY_LINE, anIdList);
     const vtkIdType aShapeIDVTK = theShapeID;
-    mySubShapeIDs->InsertNextTupleValue (&aShapeIDVTK);
+    mySubShapeIDs->InsertNextTypedTuple (&aShapeIDVTK);
     const vtkIdType aType = theMeshType;
-    myMeshTypes->InsertNextTupleValue (&aType);
+    myMeshTypes->InsertNextTypedTuple (&aType);
     anIdList->Delete();
   }
 }
@@ -144,7 +144,7 @@ void IVtkVTK_ShapeData::InsertTriangle (const IVtk_IdType   theShapeID,
   vtkIdType aPoints[3] = { thePointId1, thePointId2, thePointId3 };
   myPolyData->InsertNextCell (VTK_TRIANGLE, 3, aPoints);
   const vtkIdType aShapeIDVTK = theShapeID;
-  mySubShapeIDs->InsertNextTupleValue (&aShapeIDVTK);
+  mySubShapeIDs->InsertNextTypedTuple (&aShapeIDVTK);
   const vtkIdType aType = theMeshType;
-  myMeshTypes->InsertNextTupleValue (&aType);
+  myMeshTypes->InsertNextTypedTuple (&aType);
 }
