--- opencascade-6.8.0/Makefile.am
+++ opencascade-6.8.0/Makefile.am
@@ -82,27 +82,27 @@
 	cd $(top_builddir)/adm/lin/amk && $(MAKE) $(AM_MAKEFLAGS) Draw
 
 install-exec-local:
-	$(INSTALL) -d $(prefix)/$(platform)
-	if [ -e $(prefix)/bin -a ! -e $(prefix)/$(platform)/bin ]; then \
-		cd $(prefix)/$(platform) && ln -s ../bin bin; \
+	$(INSTALL) -d $(DESTDIR)$(prefix)/$(platform)
+	if [ -e $(DESTDIR)$(prefix)/bin -a ! -e $(DESTDIR)$(prefix)/$(platform)/bin ]; then \
+		cd $(DESTDIR)$(prefix)/$(platform) && ln -s ../bin bin; \
 	fi
-	if [ -e $(prefix)/lib -a ! -e $(prefix)/$(platform)/lib ]; then \
-		cd $(prefix)/$(platform) && ln -s ../lib lib; \
+	if [ -e $(DESTDIR)$(prefix)/lib -a ! -e $(DESTDIR)$(prefix)/$(platform)/lib ]; then \
+		cd $(DESTDIR)$(prefix)/$(platform) && ln -s ../lib lib; \
 	fi
-	buildd=`pwd`; cd $(top_srcdir); sourced=`pwd`; cd $(prefix); installd=`pwd`; cd $$buildd; \
+	buildd=`pwd`; cd $(top_srcdir); sourced=`pwd`; cd $(DESTDIR)$(prefix); installd=`pwd`; cd $$buildd; \
 	if [ "$$installd" != "$$sourced" ]; then \
-		$(INSTALL) -d $(prefix)/inc; \
-		cp -frL $(top_srcdir)/inc $(prefix); \
-		cp -frL $$buildd/config.h $(prefix); \
+		$(INSTALL) -d $(DESTDIR)$(prefix)/inc; \
+		cp -frL $(top_srcdir)/inc $(DESTDIR)$(prefix); \
+		cp -frL $$buildd/config.h $(DESTDIR)$(prefix); \
 		for d in  $(FoundationClasses_DIRS) $(ModelingData_DIRS) $(ModelingAlgorithms_DIRS) $(Visualization_DIRS) $(ApplicationFramework_DIRS) $(DataExchange_DIRS) $(Draw_DIRS); do \
-			$(INSTALL) -d $(prefix)/src/$$d; \
-			cp -frL $(top_srcdir)/src/$$d $(prefix)/src; \
+			$(INSTALL) -d $(DESTDIR)$(prefix)/src/$$d; \
+			cp -frL $(top_srcdir)/src/$$d $(DESTDIR)$(prefix)/src; \
 		done; \
 	fi
-	if [ -e $(prefix)/inc/config.h ]; then \
-		rm -f $(prefix)/inc/config.h; \
+	if [ -e $(DESTDIR)$(prefix)/inc/config.h ]; then \
+		rm -f $(DESTDIR)$(prefix)/inc/config.h; \
 	fi
-	cd $(prefix)/inc && ln -s ../config.h config.h
-	cd $(top_srcdir) && cp *.sh $(prefix)
+	cd $(DESTDIR)$(prefix)/inc && ln -s ../config.h config.h
+	cd $(top_srcdir) && cp *.sh $(DESTDIR)$(prefix)
 
-.PHONY: FoundationClasses ModelingData ModelingAlgorithms Visualization ApplicationFramework DataExchange Draw
\ No newline at end of file
+.PHONY: FoundationClasses ModelingData ModelingAlgorithms Visualization ApplicationFramework DataExchange Draw
