--- opencascade-6.5.5/ros/inc/InterfaceGraphic_X11.hxx
+++ opencascade-6.5.5/ros/inc/InterfaceGraphic_X11.hxx
@@ -25,6 +25,7 @@
 #  include <X11/Xlib.h>
 #  include <X11/Xutil.h>
 #  include <X11/Xatom.h>
+#define GLX_GLXEXT_LEGACY
 #  include <GL/glx.h>
 
 #  if defined (sun) || defined (SUNOS) || defined (__alpha) || defined (DECOSF1) || defined (sgi) || defined (IRIX) || defined (__hpux)|| defined (HPUX) 
--- opencascade-6.5.5/ros/src/InterfaceGraphic/InterfaceGraphic_X11.hxx
+++ opencascade-6.5.5/ros/src/InterfaceGraphic/InterfaceGraphic_X11.hxx
@@ -25,6 +25,7 @@
 #  include <X11/Xlib.h>
 #  include <X11/Xutil.h>
 #  include <X11/Xatom.h>
+#define GLX_GLXEXT_LEGACY
 #  include <GL/glx.h>
 
 #  if defined (sun) || defined (SUNOS) || defined (__alpha) || defined (DECOSF1) || defined (sgi) || defined (IRIX) || defined (__hpux)|| defined (HPUX) 
--- opencascade-6.5.5/ros/src/OpenGl/OpenGl_Context.cxx
+++ opencascade-6.5.5/ros/src/OpenGl/OpenGl_Context.cxx
@@ -38,6 +38,7 @@
 #elif defined(__APPLE__) && !defined(MACOSX_USE_GLX)
   #include <dlfcn.h>
 #else
+  #define GLX_GLXEXT_LEGACY
   #include <GL/glx.h> // glXGetProcAddress()
 #endif
 
