diff -ruN openslp-1.2.1.orig/configure.in openslp-1.2.1/configure.in
--- openslp-1.2.1.orig/configure.in	2005-02-15 21:06:02.000000000 +0100
+++ openslp-1.2.1/configure.in	2012-01-24 23:49:13.000000000 +0100
@@ -153,7 +153,6 @@
         OPTFLAGS="-O"
     fi
 fi
-CFLAGS="$CFLAGS $OPTFLAGS"
 
 dnl ***********************************************************************
 dnl Checks for header files.
