--- openslp-1.0.11/configure.in.orig	2003-03-11 05:20:05.000000000 +0100
+++ openslp-1.0.11/configure.in	2003-04-18 00:34:27.000000000 +0200
@@ -120,9 +120,6 @@
     if test X"$debug" = X"yes"; then
        CFLAGS="$CFLAGS -Werror"
     fi
-    if test X"$debug" = X"no"; then 
-       OPTFLAGS="-O3"
-    fi
     
 elif $CC -V 2>&1 | grep "WorkShop Compilers"; then
     dnl Allow C++ style comments
@@ -161,9 +158,9 @@
 dnl ***********************************************************************
 dnl Checks for libraries.
 dnl ***********************************************************************
-AC_CHECK_LIB(resolv, inet_aton)
-AC_CHECK_LIB(socket, main)
-AC_CHECK_LIB(nsl, gethostbyname)
+AC_SEARCH_LIBS(inet_aton, resolv)
+AC_SEARCH_LIBS(socket, socket)
+AC_SEARCH_LIBS(gethostbyname, nsl)
 AC_CHECK_LIB(m, main)
 AC_CHECK_FUNCS(ceil log10 strncasecmp strcasecmp )
 AC_OUTPUT(Makefile common/Makefile libslpattr/Makefile libslp/Makefile \
