From ef8efc5b80c9395c01f23e5acd28bb3ee7810b54 Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentnl@gentoo.org>
Date: Sun, 8 Oct 2017 13:08:20 +1300
Subject: Fix useless localization of scalar assignment

Bug: https://rt.cpan.org/Ticket/Display.html?id=73873
---
 lib/DBI/Format.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/DBI/Format.pm b/lib/DBI/Format.pm
index 49cd7dd..bc985f9 100755
--- a/lib/DBI/Format.pm
+++ b/lib/DBI/Format.pm
@@ -374,7 +374,7 @@ sub header {
 
 sub row {
     my($self, $rowref) = @_;
-		local( $^W = 0 );
+		local ($^W) = 0;
     my @row = @$rowref;
 	my $fh = $self->{'fh'};
 	print $fh (join($self->{sep}, @row), "\n");
-- 
2.14.1

