=== modified file 'fbtruetype/Makefile'
--- fbtruetype/Makefile	2011-10-01 14:10:17 +0000
+++ fbtruetype/Makefile	2011-10-01 14:15:02 +0000
@@ -23,8 +23,8 @@
 OBJECTS  = fbtruetype.o messages.o console.o ttf.o luxisri.o
 
 LDFLAGS  += -L$(LIBDIR)
-LIBSS    = $(LIBDIR)/libfreetype.a $(LIBDIR)/libz.a -lm
-LIBSD    = -lfreetype -lm
+LIBSS    = $(shell pkg-config freetype2 --libs --static) -lm
+LIBSD    = $(shell pkg-config freetype2 --libs) -lm
 
 CFLAGS   += -I/usr/include/freetype2
 COMPILE  = $(CC) $(CFLAGS)

