diff --git a/getloadavg.c b/getloadavg.c
index cf5869f..c7e2b27 100644
--- a/getloadavg.c
+++ b/getloadavg.c
@@ -69,8 +69,12 @@ Boston, MA  02110-1301  USA */
 #include <config.h>
 #endif
 
-#include "lisp.h"
-#include "sysfile.h" /* for encapsulated open, close, read, write */
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+//#include "lisp.h"
+//#include "sysfile.h" /* for encapsulated open, close, read, write */
 
 #ifndef HAVE_GETLOADAVG

