diff -ur dvbstreamer-1.1.orig/configure.in dvbstreamer-1.1/configure.in
--- dvbstreamer-1.1.orig/configure.in	2008-10-31 16:49:00.000000000 +0200
+++ dvbstreamer-1.1/configure.in	2010-02-03 12:30:56.000000000 +0200
@@ -23,7 +23,6 @@
 
 AM_INIT_AUTOMAKE("dvbstreamer", $SPEC_VERSION)
 
-AC_CONFIG_SUBDIRS(libltdl)
 AM_CONFIG_HEADER(config.h)
 
 AC_ISC_POSIX
@@ -35,12 +34,8 @@
 dnl ---------------------------------------------------------------------------
 dnl Libltdl setup
 dnl ---------------------------------------------------------------------------
-AC_LIBTOOL_WIN32_DLL dnl Make sure under cygwin we can load plugins !!!
-AC_LIBTOOL_DLOPEN
-AC_LIBLTDL_CONVENIENCE
-AC_PROG_LIBTOOL
-AC_SUBST(INCLTDL)
-AC_SUBST(LIBLTDL)
+LT_PREREQ([2.2.6])
+LT_INIT([dlopen win32-dll disable-static])
 
 dnl ---------------------------------------------------------------------------
 dnl Work out the libdir name (copied from xine-lib configure.ac)
diff -ur dvbstreamer-1.1.orig/Makefile.am dvbstreamer-1.1/Makefile.am
--- dvbstreamer-1.1.orig/Makefile.am	2008-10-31 16:25:03.000000000 +0200
+++ dvbstreamer-1.1/Makefile.am	2010-02-03 12:25:57.000000000 +0200
@@ -1,6 +1,6 @@
 ## DVBStreamer main Makefile
 
-SUBDIRS = libltdl include include/dvbpsi src/dvbpsi src src/plugins
+SUBDIRS = include include/dvbpsi src/dvbpsi src src/plugins
 
 dvbstreamerdocdir = ${prefix}/doc/DVBStreamer
 dvbstreamerdoc_DATA = \
diff -ur dvbstreamer-1.1.orig/src/Makefile.am dvbstreamer-1.1/src/Makefile.am
--- dvbstreamer-1.1.orig/src/Makefile.am	2008-10-31 16:25:02.000000000 +0200
+++ dvbstreamer-1.1/src/Makefile.am	2010-02-03 12:33:30.000000000 +0200
@@ -1,7 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = @INCLTDL@
-
 AM_CFLAGS =\
      -I$(top_srcdir)/include -Wall -Werror -D_GNU_SOURCE 
 
@@ -104,7 +102,7 @@
 dvbstreamer_LDFLAGS = -rdynamic -Wl,-whole-archive -Wl,dvbpsi/libdvbpsi.a -Wl,-no-whole-archive 
 
 dvbstreamer_LDADD = \
-	  -lpthread -lsqlite3 -lreadline @GETTIME_LIB@ @ICONV_LIB@ @READLINE_TERMCAP@ @LIBLTDL@
+	  -lpthread -lsqlite3 -lreadline @GETTIME_LIB@ @ICONV_LIB@ @READLINE_TERMCAP@ -lltdl
 
 #
 # dvbctrl
diff -ur dvbstreamer-1.1.orig/src/pluginmgr.c dvbstreamer-1.1/src/pluginmgr.c
--- dvbstreamer-1.1.orig/src/pluginmgr.c	2008-10-31 16:25:02.000000000 +0200
+++ dvbstreamer-1.1/src/pluginmgr.c	2010-02-03 12:27:53.000000000 +0200
@@ -22,9 +22,9 @@
 */
 #include <stdlib.h>
 #include <string.h>
+#include <ltdl.h>
 
 #include "config.h"
-#include "ltdl.h"
 #include "list.h"
 #include "deliverymethod.h"
 #include "patprocessor.h"
