From ca340cbfc518cc4b0655840453198620215535df Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl-gentoo@posteo.net>
Date: Tue, 25 Apr 2023 09:27:40 +0200
Subject: [PATCH] add missing #include statements for gcc-13

Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>
--- a/include/core_api/session.h
+++ b/include/core_api/session.h
@@ -25,6 +25,7 @@
 
 #include <yafray_constants.h>
 #include <utilities/threadUtils.h>
+#include <string>
 
 __BEGIN_YAFRAY
 
-- 
2.40.0

