--- bustle-0.2.3-orig/bustle.cabal	2013-04-03 15:15:36.278582601 +1100
+++ bustle-0.2.3/bustle.cabal	2013-04-03 15:21:14.334592030 +1100
@@ -27,30 +27,30 @@
   if flag(PostCabalizedGtk2HS)
     -- Since gtk2hs 0.11, pango is a separate package, and its module names
     -- have changed.
-    Build-Depends:  mtl, base >= 4 && < 5, containers, parsec,
+    Build-Depends:  haskell98, mtl, base >= 4 && < 5, containers, parsec,
                     filepath, process,
-                    gtk > 0.11, glade, pango, glib, cairo
+                    gtk > 0.11 && < 0.13, glade < 0.13, pango < 0.13, glib < 0.13, cairo < 0.13
   else
-    Build-Depends:  mtl, base >= 4 && < 5, containers, parsec,
+    Build-Depends:  haskell98, mtl, base >= 4 && < 5, containers, parsec,
                     filepath, process,
-                    gtk > 0.10 && < 0.11, glade, pango, glib, cairo
+                    gtk > 0.10 && < 0.11, glade < 0.13, pango < 0.13, glib < 0.13, cairo < 0.13
 
 Executable bustle-count
   Main-is:        bustle-count.hs
   Other-modules:  Bustle.Parser, Bustle.Stats, Bustle.Types,
                   Bustle.Noninteractive
-  Build-Depends:  base >= 4 && < 5, containers, parsec
+  Build-Depends:  haskell98, base >= 4 && < 5, containers, parsec
   Ghc-options:    -Wall -fno-warn-unused-imports -fno-warn-unused-do-bind
 
 Executable bustle-time
   Main-is:        bustle-time.hs
   Other-modules:  Bustle.Parser, Bustle.Stats, Bustle.Types,
                   Bustle.Noninteractive
-  Build-Depends:  base >= 4 && < 5, containers, parsec
+  Build-Depends:  haskell98, base >= 4 && < 5, containers, parsec
   Ghc-options:    -Wall -fno-warn-unused-imports -fno-warn-unused-do-bind
 
 Executable bustle-dot
   Main-is:        bustle-dot.hs
   Other-modules:  Bustle.Parser, Bustle.Types
-  Build-Depends:  base >= 4 && < 5, containers, parsec
+  Build-Depends:  haskell98, base >= 4 && < 5, containers, parsec
   Ghc-options:    -Wall -fno-warn-unused-imports -fno-warn-unused-do-bind
