--- a/dump1090.c
+++ b/dump1090.c
@@ -2233,8 +2233,8 @@
         struct stat sbuf;
         int fd = -1;
 
-        if (stat("gmap.html",&sbuf) != -1 &&
-            (fd = open("gmap.html",O_RDONLY)) != -1)
+        if (stat("/usr/share/dump1090/gmap.html",&sbuf) != -1 &&
+            (fd = open("/usr/share/dump1090/gmap.html",O_RDONLY)) != -1)
         {
             content = malloc(sbuf.st_size);
             if (read(fd,content,sbuf.st_size) == -1) {
