From 99a7f336635aaeb666ea8dc791743177a030dc18 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Wed, 28 Dec 2022 15:53:08 +0100
Subject: [PATCH] Workaround blosc2 detection via blosc2 package

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 966be028..2708deb9 100755
--- a/setup.py
+++ b/setup.py
@@ -262,11 +262,11 @@ if __name__ == "__main__":
 
     debug = "--debug" in sys.argv
 
-    blosc2_inc, blosc2_lib = get_blosc2_directories()
+    blosc2_inc, blosc2_lib = "", ""
 
     # Global variables
     lib_dirs = [blosc2_lib]
-    inc_dirs = [Path("hdf5-blosc/src"), Path("hdf5-blosc2/src"), blosc2_inc]
+    inc_dirs = [Path("hdf5-blosc/src"), Path("hdf5-blosc2/src")]
     optional_libs = []
     copy_libs = []
 
-- 
2.39.0

