--- data/gtorrentviewer.desktop.in.old	2014-11-22 17:48:55.859683842 +0100
+++ data/gtorrentviewer.desktop.in	2014-11-22 17:49:25.400108137 +0100
@@ -3,11 +3,9 @@
 _GenericName=Torrent Viewer
 _Comment=Viewer and Editor for .torrent files
 Exec=gtorrentviewer %U
-Icon=gtorrentviewer.png
+Icon=gtorrentviewer
 StartupNotify=false
 Terminal=false
 Type=Application
-Encoding=UTF-8
-Categories=Application;Network;
+Categories=Network;
 MimeType=application/x-bittorrent;
-Version=0.2
