diff -up xbase64-3.1.2/bin/dbfutil1.cpp.gcc7 xbase64-3.1.2/bin/dbfutil1.cpp
--- xbase64-3.1.2/bin/dbfutil1.cpp.gcc7	2017-02-20 09:59:36.135561103 -0500
+++ xbase64-3.1.2/bin/dbfutil1.cpp	2017-02-20 10:00:32.743218194 -0500
@@ -157,7 +157,8 @@ void  MyClass::SetFilter()
   memset( Expression, 0x00, 512 );
   while( !strlen( Expression )){
     std::cout << "Enter filter expression (like AMOUNT<5)" << std::endl;
-    gets( Expression );
+    fgets( Expression, sizeof(Expression), stdin );
+    strtok( Expression, "\n" );
   }
   if( xbf )
     delete xbf;
@@ -247,7 +248,8 @@ void MyClass::ProcessExpression()
   while( !strstr( exprsn, "QUIT" ) && !strstr( exprsn, "quit" )){
 
     std::cout << ">";
-    gets( exprsn );
+    fgets( exprsn, sizeof(exprsn), stdin );
+    strtok( exprsn, "\n" );
 
     if( strstr( exprsn, "HELP" ) || strstr( exprsn, "help" )){
       std::cout << "** Command Help ***"    << std::endl << std::endl;
diff -up xbase64-3.1.2/xbase64/xbase64.h.gcc7 xbase64-3.1.2/xbase64/xbase64.h
--- xbase64-3.1.2/xbase64/xbase64.h.gcc7	2017-02-20 09:49:07.063609482 -0500
+++ xbase64-3.1.2/xbase64/xbase64.h	2017-02-20 09:57:36.586618723 -0500
@@ -53,6 +53,7 @@
 #endif
 
 #include <string.h>
+#include <cstdio>
 
 #if defined(__WIN32__)
 #include "windows.h"
