--- a/doc/screen.texinfo
+++ b/doc/screen.texinfo
@@ -5796,8 +5796,8 @@
 ============
 
 @example
-     Thomas Renninger <treen@suse.com>,
-     Axel Beckert <abe@deuxchevaux.org>,
+     Thomas Renninger <treen@@suse.com>,
+     Axel Beckert <abe@@deuxchevaux.org>,
      Ken Beal <kbeal@@amber.ssd.csd.harris.com>,
      Rudolf Koenig <rfkoenig@@informatik.uni-erlangen.de>,
      Toerless Eckert <eckert@@informatik.uni-erlangen.de>,
