Revert commit 5460f5d28c01a9a58e021eb1dffef2965e629d58

--- b/screen.c
+++ a/screen.c
@@ -670,12 +670,6 @@ int main(int ac, char** av)
                 Panic(0, "-L: logfile name can not start with \"-\" symbol");
               if (strlen(screenlogfile) > PATH_MAX)
                 Panic(0, "-L: logfile name too long. (max. %d char)", PATH_MAX);
-
-              FILE *w_check;
-              if ((w_check = fopen(screenlogfile, "w")) == NULL)
-                Panic(0, "-L: logfile name access problem");
-              else
-                fclose(w_check);
             }
             nwin_options.Lflag = 1;
             break;
