--- libpri-1.4.13/Makefile.orig	2012-09-26 17:46:23.000000000 +0200
+++ libpri-1.4.13/Makefile	2012-10-10 08:04:44.000000000 +0200
@@ -71,7 +71,7 @@
 CFLAGS += -fPIC $(ALERTING) $(LIBPRI_OPT) $(COVERAGE_CFLAGS)
 INSTALL_PREFIX=$(DESTDIR)
 INSTALL_BASE=/usr
-libdir?=$(INSTALL_BASE)/lib
+LIBDIR=$(INSTALL_PREFIX)$(INSTALL_BASE)/lib
 ifneq ($(findstring Darwin,$(OSARCH)),)
   SOFLAGS=$(LDFLAGS) -dynamic -bundle -Xlinker -macosx_version_min -Xlinker 10.4 -Xlinker -undefined -Xlinker dynamic_lookup -force_flat_namespace
   ifeq ($(shell /usr/bin/sw_vers -productVersion | cut -c1-4),10.6)
@@ -145,27 +145,27 @@
 	fi
 
 install: $(STATIC_LIBRARY) $(DYNAMIC_LIBRARY)
-	mkdir -p $(INSTALL_PREFIX)$(libdir)
+	mkdir -p $(LIBDIR)
 	mkdir -p $(INSTALL_PREFIX)$(INSTALL_BASE)/include
 ifneq (${OSARCH},SunOS)
 	install -m 644 libpri.h $(INSTALL_PREFIX)$(INSTALL_BASE)/include
-	install -m 755 $(DYNAMIC_LIBRARY) $(INSTALL_PREFIX)$(libdir)
-	#if [ -x /usr/sbin/sestatus ] && ( /usr/sbin/sestatus | grep "SELinux status:" | grep -q "enabled"); then /sbin/restorecon -v $(INSTALL_PREFIX)$(libdir)/$(DYNAMIC_LIBRARY); fi
-	( cd $(INSTALL_PREFIX)$(libdir) ; ln -sf $(DYNAMIC_LIBRARY) libpri.so)
-	install -m 644 $(STATIC_LIBRARY) $(INSTALL_PREFIX)$(libdir)
-	if test $$(id -u) = 0; then $(LDCONFIG) $(LDCONFIG_FLAGS) $(INSTALL_PREFIX)$(libdir); fi
+	install -m 755 $(DYNAMIC_LIBRARY) $(LIBDIR)
+	if [ -x /usr/sbin/sestatus ] && ( /usr/sbin/sestatus | grep "SELinux status:" | grep -q "enabled"); then /sbin/restorecon -v $(LIBDIR)/$(DYNAMIC_LIBRARY); fi
+	( cd $(LIBDIR) ; ln -sf libpri.so.$(SONAME) libpri.so)
+	install -m 644 $(STATIC_LIBRARY) $(LIBDIR)
+	if test $$(id -u) = 0; then $(LDCONFIG) $(LDCONFIG_FLAGS) $(LIBDIR); fi
 else
 	install -f $(INSTALL_PREFIX)$(INSTALL_BASE)/include -m 644 libpri.h
-	install -f $(INSTALL_PREFIX)$(libdir) -m 755 $(DYNAMIC_LIBRARY)
-	( cd $(INSTALL_PREFIX)$(libdir) ; ln -sf $(DYNAMIC_LIBRARY) libpri.so)
-	install -f $(INSTALL_PREFIX)$(libdir) -m 644 $(STATIC_LIBRARY)
+	install -f $(LIBDIR) -m 755 $(DYNAMIC_LIBRARY)
+	( cd $(LIBDIR) ; ln -sf libpri.so.$(SONAME) libpri.so)
+	install -f $(LIBDIR) -m 644 $(STATIC_LIBRARY)
 endif
 
 uninstall:
 	@echo "Removing Libpri"
-	rm -f $(INSTALL_PREFIX)$(libdir)/$(STATIC_LIBRARY)
-	rm -f $(INSTALL_PREFIX)$(libdir)/libpri.so
-	rm -f $(INSTALL_PREFIX)$(libdir)/$(DYNAMIC_LIBRARY)
+	rm -f $(LIBDIR)/libpri.so.$(SONAME)
+	rm -f $(LIBDIR)/libpri.so
+	rm -f $(LIBDIR)/libpri.a
 	rm -f $(INSTALL_PREFIX)$(INSTALL_BASE)/include/libpri.h
 
 pritest: pritest.o $(STATIC_LIBRARY)
