Index: minitube-2.4/src/mainwindow.cpp
===================================================================
--- minitube-2.4.orig/src/mainwindow.cpp
+++ minitube-2.4/src/mainwindow.cpp
@@ -206,8 +206,6 @@ void MainWindow::lazyInit() {
     connect(mouseTimer, SIGNAL(timeout()), SLOT(hideMouse()));
 
     JsFunctions::instance();
-
-    checkForUpdate();
 }
 
 void MainWindow::changeEvent(QEvent* event) {
