--- LPRng-3.8.28/src/common/krb5_auth.c.org	2004-09-24 22:19:57.000000000 +0200
+++ LPRng-3.8.28/src/common/krb5_auth.c	2006-10-13 09:19:44.000000000 +0200
@@ -922,7 +922,7 @@
 		nstored = 0;
 	}
 	
-	if ((cc = Read_fd_len_timeout(transfer_timeout, fd, len_buf, 4)) != 4) {
+	if ((cc = Read_fd_len_timeout(transfer_timeout, fd, (char*)len_buf, 4)) != 4) {
 		/* XXX can't read enough, pipe must have closed */
 		return(0);
 	}
