From c769f79831c43f1cad0b4f8831276da93d28f479 Mon Sep 17 00:00:00 2001
From: Daniel O'Connor <daniel.oconnor@gmail.com>
Date: Sun, 6 Nov 2011 13:25:04 +1030
Subject: [PATCH] &new vs new

---
 Console/Getargs.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Console/Getargs.php b/Console/Getargs.php
index 8ba21c6..c6826ea 100644
--- a/Console/Getargs.php
+++ b/Console/Getargs.php
@@ -224,7 +224,7 @@ class Console_Getargs
     function &factory($config = array(), $arguments = null)
     {
         // Create the options object.
-        $obj = & new Console_Getargs_Options();
+        $obj = new Console_Getargs_Options();
 
         // Try to set up the arguments.
         $err = $obj->init($config, $arguments);
