--- FusionIcon/interface_qt4/main.py.orig	2012-06-05 17:54:13.000000000 +0400
+++ FusionIcon/interface_qt4/main.py		2012-06-05 18:01:13.000000000 +0400
@@ -19,6 +19,7 @@
 import sys, os, time
 from PyQt4 import QtGui, QtCore
 from FusionIcon.start import wms, apps, options, decorators, init
+import subprocess
 
 class Build(QtGui.QApplication):
 	def reload_wm(self):
@@ -63,9 +64,9 @@
 				actionWD.setChecked(True)
 		self.Tray.menu = QtGui.QMenu()
 		if 'ccsm' in apps:
-			self.Tray.menu.addAction(apps['ccsm'].label, lambda: run(['ccsm']))
+			self.Tray.menu.addAction(apps['ccsm'].label, lambda: subprocess.call(['ccsm']))
 		if 'emerald theme manager' in apps:
-			self.Tray.menu.addAction(apps['emerald theme manager'].label, lambda: run(apps['emerald theme manager'].command))
+			self.Tray.menu.addAction(apps['emerald theme manager'].label, lambda: subprocess.call(apps['emerald theme manager'].command))
 		if 'ccsm' in apps or 'emerald theme manager' in apps:
 			self.Tray.menu.addSeparator()
 		self.Tray.menu.addAction("Reload Window Manager", self.reload_wm)
