--- a/pythran/tests/__init__.py
+++ b/pythran/tests/__init__.py
@@ -71,7 +71,7 @@ class TestEnv(unittest.TestCase):
 
     module = pytest.mark.module
     # default options used for the c++ compiler
-    PYTHRAN_CXX_FLAGS = ['-O0', '-Wall', '-Werror', '-UNDEBUG',
+    PYTHRAN_CXX_FLAGS = ['-O0', '-Wall', '-UNDEBUG',
                          '-Wno-unused-function',
                          '-Wno-int-in-bool-context',
                          '-Wno-unknown-warning-option',
