--- a/CMakeLists.txt	2021-11-02 01:45:56.488999411 -1000
+++ b/CMakeLists.txt	2021-11-02 01:51:53.343219957 -1000
@@ -756,9 +756,9 @@
   # looked for them. This is a nasty fix which might fail miserably. Assumption:
   # All builds using GTK uses unicode and wxWidgets 3.0
   if (GTK3_FOUND)
-    list(APPEND wxWidgets_LIBRARIES "-lwx_gtk3u_aui-3.0")
+    list(APPEND wxWidgets_LIBRARIES "-lwx_gtk3u_aui-3.0-gtk3")
     if (OPENGL_FOUND)
-      list(APPEND wxWidgets_LIBRARIES "-lwx_gtk3u_gl-3.0")
+      list(APPEND wxWidgets_LIBRARIES "-lwx_gtk3u_gl-3.0-gtk3")
     endif ()
   endif ()
 
