--- a/setup.py	2023-12-07 20:38:21.247250326 +0100
+++ b/setup.py	2023-12-07 20:38:30.707272148 +0100
@@ -455,14 +455,6 @@
     check_submodules()
     check_pydep("yaml", "pyyaml")
 
-    build_caffe2(
-        version=version,
-        cmake_python_library=cmake_python_library,
-        build_python=True,
-        rerun_cmake=RERUN_CMAKE,
-        cmake_only=CMAKE_ONLY,
-        cmake=cmake,
-    )
 
     if CMAKE_ONLY:
         report(
