diff --git a/Makefile b/Makefile
index fc44fac..6c87e89 100644
--- a/Makefile
+++ b/Makefile
@@ -202,17 +202,17 @@ _perftest:
 #
 # LUA_LIBDIR and LUA_SHAREDIR are used by the .rockspec (don't change the names!)
 #
-DESTDIR=/usr/local
-LUA_LIBDIR=$(DESTDIR)/lib/lua/5.1
-LUA_SHAREDIR=$(DESTDIR)/share/lua/5.1
+PREFIX=/usr/local
+LUA_LIBDIR=$(PREFIX)/lib/lua/5.1
+LUA_SHAREDIR=$(PREFIX)/share/lua/5.1
 
 #
 # AKa 17-Oct: changed to use 'install -m 644' and 'cp -p'
 #
 install: $(_TARGET_SO) src/lanes.lua
-	mkdir -p $(LUA_LIBDIR) $(LUA_LIBDIR)/lanes $(LUA_SHAREDIR)
-	install -m 644 $(_TARGET_SO) $(LUA_LIBDIR)/lanes
-	cp -p src/lanes.lua $(LUA_SHAREDIR)
+	mkdir -p $(DESTDIR)$(LUA_LIBDIR)/lanes $(DESTDIR)$(LUA_SHAREDIR)
+	install -m 644 $(_TARGET_SO) $(DESTDIR)$(LUA_LIBDIR)/lanes
+	cp -p src/lanes.lua $(DESTDIR)$(LUA_SHAREDIR)
 
 
 #--- Packaging ---
