--- a/src/include/CMakeLists.txt
+++ b/src/include/CMakeLists.txt
@@ -22,6 +22,9 @@
 check_include_file(time.h       HAVE_TIME_H)
 check_include_file(unistd.h     HAVE_UNISTD_H)
 
+set (ELEKTRA_DEPRECATED "__attribute__ ((deprecated))")
+set (ELEKTRA_UNUSED "__attribute__ ((unused))")
+
 configure_file (
 	"${CMAKE_CURRENT_SOURCE_DIR}/kdb.h"
 	"${CMAKE_CURRENT_BINARY_DIR}/kdb.h"
--- a/src/include/kdbconfig.h
+++ b/src/include/kdbconfig.h
@@ -30,6 +30,12 @@
   * to search for user configuration. */
 #define KDB_DB_HOME              "@KDB_DB_HOME@"
 
+/** Declares a parameter as unused. */
+#define ELEKTRA_UNUSED           @ELEKTRA_UNUSED@
+
+/** Declares an API as deprecated. */
+#define ELEKTRA_DEPRECATED       @ELEKTRA_DEPRECATED@
+
 /* disable debug output messages */
 #define DEBUG @DEBUG@
 
