diff -p -up iscsitarget-1.4.19-1mdv2010.1/kernel/conn.c.2.6.33 iscsitarget-1.4.19-1mdv2010.1/kernel/conn.c
--- iscsitarget-1.4.19-1mdv2010.1/kernel/conn.c.2.6.33	2010-02-21 13:07:29.000000000 +0100
+++ iscsitarget-1.4.19-1mdv2010.1/kernel/conn.c	2010-02-21 13:41:38.641365087 +0100
@@ -43,7 +43,7 @@ void conn_info_show(struct seq_file *seq
 		switch (sk->sk_family) {
 		case AF_INET:
 			snprintf(buf, sizeof(buf),
-				 "%u.%u.%u.%u", NIPQUAD(inet_sk(sk)->daddr));
+				 "%u.%u.%u.%u", NIPQUAD(inet_sk(sk)->inet_daddr));
 			break;
 		case AF_INET6:
 			snprintf(buf, sizeof(buf), "[%pI6]",
