--- laszip-src-2.2.0/configure.ac
+++ laszip-src-2.2.0/configure.ac
@@ -21,9 +21,6 @@
 
 m4_define([debug_default],[no])
 
-CFLAGS="-Wall -Wno-long-long -pedantic $CFLAGS"
-CXXFLAGS="-Wall -Wno-long-long -pedantic -std=c++98 $CXXFLAGS"
-
 dnl #########################################################################
 dnl Checks for programs.
 dnl #########################################################################
@@ -68,8 +65,8 @@
     CXXFLAGS="$CXXFLAGS -g -DDEBUG"
     AC_MSG_RESULT(yes)
 else
-    CFLAGS="$CFLAGS -O3 -DNDEBUG"
-    CXXFLAGS="$CXXFLAGS -O3 -DNDEBUG"
+    CFLAGS="$CFLAGS -DNDEBUG"
+    CXXFLAGS="$CXXFLAGS -DNDEBUG"
     AC_MSG_RESULT(no)
 fi
 
--- laszip-src-2.2.0/src/Makefile.am
+++ laszip-src-2.2.0/src/Makefile.am
@@ -1,6 +1,6 @@
 SUBDIRS = . 
 
-AM_CPPFLAGS = -I. -I../include -I../include/laszip
+AM_CPPFLAGS = -I. -I$(top_srcdir)/include -I$(top_srcdir)/include/laszip
 noinst_LTLIBRARIES = liblibrary.la 
 
 liblibrary_la_SOURCES =				arithmeticmodel.cpp \
--- laszip-src-2.2.0/tools/Makefile.am
+++ laszip-src-2.2.0/tools/Makefile.am
@@ -1,5 +1,5 @@
 
-AM_CPPFLAGS = -I../include -I../include/laszip
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/include/laszip
 LDADD = ../liblaszip.la 
 
 laszippertest_SOURCES = laszippertest.cpp
--- laszip-src-2.2.0/include/laszip/Makefile.am
+++ laszip-src-2.2.0/include/laszip/Makefile.am
@@ -1,4 +1,4 @@
-laszipdir = $(includedir)/
+laszipdir = $(includedir)/laszip/
 
 laszip_HEADERS =		lasunzipper.hpp \
 						laszip.hpp \
