http://bugs.gentoo.org/227503

--- a/v/includex/v/vpsdc.h
+++ b/v/includex/v/vpsdc.h
@@ -14,7 +14,9 @@
 
 #include <v/vdc.h>
 #include <v/vprinter.h>
-#include <fstream.h>
+#include <fstream>
+
+using namespace std;
 
     class vPostScriptDC : public vDC
       {
--- a/v/includex/v/vpsprntr.h
+++ b/v/includex/v/vpsprntr.h
@@ -14,7 +14,7 @@
 
 #include <v/v_defs.h>
 #include <v/vmodald.h>
-#include <fstream.h>
+#include <fstream>
 
 #define vPaperLetter 0
 #define vPaperLegal 1
@@ -31,6 +31,8 @@
 #define vPaperQuarto 12
 #define vPaper10x14 13
 
+using namespace std;
+
 #define vPaperDefault vPaperLetter		// Index to default paper
 
     class vPSPrinter : public vModalDialog
--- a/v/srcx/vpsprntr.cxx
+++ b/v/srcx/vpsprntr.cxx
@@ -103,7 +103,7 @@ static unsigned char prompt_bits[] = {
     _height = PaperHeights[_paperType];
     _portrait = 1;
     _useColor = 0;
-    _pstream = 0;
+    vPSPrinter::_pstream = 0;
 
     ++_instances;		// bump reference counter
   }
--- a/rs232int.cpp
+++ b/rs232int.cpp
@@ -145,7 +145,7 @@ int RS232Interface::OpenSerial(int no)
 		snprintf(lockname, MAXPATH, LOCK_DIR "/LCK..ttyS%d", no);
 		UserDebug1(UserApp2, "RS232Interface::OpenSerial() now lock the device %s\n", lockname);
 
-		fd = open ((const char *)lockname,O_RDWR|O_EXCL|O_CREAT);
+		fd = open ((const char *)lockname,O_RDWR|O_EXCL|O_CREAT,0664);
 		if (fd < 0)
 		{
 			fd = open ((const char *)lockname,O_RDONLY);
