--- gettext-0.21-rc1/gettext-tools/src/Makefile.in
+++ gettext-0.21-rc1/gettext-tools/src/Makefile.in
@@ -4106,9 +4106,16 @@
 
 # How to get the include files of libtextstyle.
 @USE_INSTALLED_LIBTEXTSTYLE_FALSE@textstyle.h textstyle/stdbool.h textstyle/version.h textstyle/woe32dll.h:
-@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	here=`pwd`; \
-@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	cd ../../libtextstyle/lib && \
-@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	  $(MAKE) install-nobase_includeHEADERS install-nobase_nodist_includeHEADERS includedir="$$here"
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	cd "$(abs_top_builddir)/../libtextstyle/lib" && $(MAKE) $@
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	if test -f "$(abs_top_builddir)/../libtextstyle/lib/$@"; then \
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@		file="$(abs_top_builddir)/../libtextstyle/lib/$@"; \
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	else \
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@		file="$(abs_top_srcdir)/../libtextstyle/lib/$@"; \
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	fi; \
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	target_dir="$(abs_builddir)/$@"; \
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	target_dir="$${target_dir%/*}"; \
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	$(MKDIR_P) "$${target_dir}"; \
+@USE_INSTALLED_LIBTEXTSTYLE_FALSE@	$(INSTALL_HEADER) "$$file" $@
 
 # No need to install libgettextsrc.a, except on AIX.
 install-exec-hook: install-exec-clean
