http://bugs.gentoo.org/327769

--- NetSSL_OpenSSL/src/SecureSocketImpl.cpp
+++ NetSSL_OpenSSL/src/SecureSocketImpl.cpp
@@ -430,7 +430,11 @@
 
 			if (!strcmp(extstr, "subjectAltName"))
 			{
+				#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+				const X509V3_EXT_METHOD* meth = X509V3_EXT_get(ext);
+				#else
 				X509V3_EXT_METHOD* meth = X509V3_EXT_get(ext);
+				#endif
 				if (!meth)
 					break;
 
