diff -Naur poco-1.3.3p1-all.orig/components poco-1.3.3p1-all/components
--- poco-1.3.3p1-all.orig/components	2008-10-20 19:40:05.000000000 +0200
+++ poco-1.3.3p1-all/components	2008-10-20 20:11:19.000000000 +0200
@@ -1,12 +1,6 @@
-CppUnit
 Foundation
 XML
 Util
 Net
-NetSSL_OpenSSL
 Data
-Data/SQLite
-Data/ODBC
-Data/MySQL
-Crypto
 Zip
diff -Naur poco-1.3.3p1-all.orig/Data/ODBC/Makefile poco-1.3.3p1-all/Data/ODBC/Makefile
--- poco-1.3.3p1-all.orig/Data/ODBC/Makefile	2008-10-20 19:40:05.000000000 +0200
+++ poco-1.3.3p1-all/Data/ODBC/Makefile	2008-10-20 19:54:07.000000000 +0200
@@ -11,19 +11,19 @@
 ifeq ($(POCO_CONFIG),MinGW)
 # Lack of unixODBC or iODBC is not an error for MinGW platform since it uses Windows odbc32.lib
 else
-ifeq (0, $(shell test -e /usr/lib/libodbc$(SHAREDLIBLINKEXT); echo $$?))
+ifeq (unixodbc, $(GENTOO_ODBC))
 SYSLIBS += -lodbc -lodbcinst
 COMMONFLAGS += -DPOCO_UNIXODBC
 else
-ifeq (0, $(shell test -h /usr/lib/libodbc$(SHAREDLIBLINKEXT); echo $$?))
+ifeq (unixodbc, $(GENTOO_ODBC))
 SYSLIBS += -lodbc -lodbcinst
 COMMONFLAGS += -DPOCO_UNIXODBC
 else
-ifeq (0, $(shell test -e /usr/lib/libiodbc$(SHAREDLIBLINKEXT); echo $$?))
+ifeq (iodbc, $(GENTOO_ODBC))
 SYSLIBS += -liodbc -liodbcinst
 COMMONFLAGS += -DPOCO_IODBC
 else
-ifeq (0, $(shell test -h /usr/lib/libiodbc$(SHAREDLIBLINKEXT); echo $$?))
+ifeq (iodbc, $(GENTOO_ODBC))
 SYSLIBS += -liodbc -liodbcinst
 COMMONFLAGS += -DPOCO_IODBC
 else
diff -Naur poco-1.3.3p1-all.orig/Makefile poco-1.3.3p1-all/Makefile
--- poco-1.3.3p1-all.orig/Makefile	2008-10-20 19:40:05.000000000 +0200
+++ poco-1.3.3p1-all/Makefile	2008-10-20 20:12:09.000000000 +0200
@@ -21,7 +21,7 @@
 
 .PHONY: all libexecs cppunit tests samples clean distclean install
 
-all: libexecs tests samples
+all: libexecs
 
 INSTALLDIR = $(DESTDIR)$(POCO_PREFIX)
 COMPONENTS = Foundation XML Util Net NetSSL_OpenSSL Data Data/SQLite Data/ODBC Data/MySQL Crypto Zip
@@ -31,7 +31,7 @@
 
 install: libexecs
 	mkdir -p $(INSTALLDIR)/include/Poco
-	mkdir -p $(INSTALLDIR)/lib
+	mkdir -p $(INSTALLDIR)/$(LIBDIR)
 	mkdir -p $(INSTALLDIR)/bin
 	for comp in $(COMPONENTS) ; do \
 		if [ -d "$(POCO_BASE)/$$comp/include" ] ; then \
@@ -41,11 +41,11 @@
 			find $(POCO_BUILD)/$$comp/bin -perm -700 -type f -exec cp -f {} $(INSTALLDIR)/bin \; ; \
 		fi ; \
 	done
-	find $(POCO_BUILD)/lib -name "libPoco*" -type f -exec cp -f {} $(INSTALLDIR)/lib \;
-	find $(POCO_BUILD)/lib -name "libPoco*" -type l -exec cp -Rf {} $(INSTALLDIR)/lib \;
+	find $(POCO_BUILD)/$(LIBDIR) -name "libPoco*" -type f -exec cp -f {} $(INSTALLDIR)/$(LIBDIR) \;
+	find $(POCO_BUILD)/$(LIBDIR) -name "libPoco*" -type l -exec cp -Rf {} $(INSTALLDIR)/$(LIBDIR) \;
 
-libexecs =  Foundation-libexec XML-libexec Util-libexec Net-libexec NetSSL_OpenSSL-libexec Data-libexec Data/SQLite-libexec Data/ODBC-libexec Data/MySQL-libexec Crypto-libexec Zip-libexec
-tests    =  Foundation-tests XML-tests Util-tests Net-tests NetSSL_OpenSSL-tests Data-tests Data/SQLite-tests Data/ODBC-tests Data/MySQL-tests Crypto-tests Zip-tests
+libexecs =  Foundation-libexec XML-libexec Util-libexec Net-libexec Data-libexec Zip-libexec
+tests    =  Foundation-tests XML-tests Util-tests Net-tests Data-tests Zip-tests
 samples  =  Foundation-samples XML-samples Util-samples Net-samples NetSSL_OpenSSL-samples Data-samples Crypto-samples Zip-samples
 
 .PHONY: $(libexecs)
