--- a/cmake/Dependencies.cmake	2024-04-29 18:37:34.005639858 +0200
+++ b/cmake/Dependencies.cmake	2024-04-29 18:39:29.126587738 +0200
@@ -1235,7 +1235,7 @@
   if(CUDNN_VERSION VERSION_LESS 8.5)
     message(FATAL_ERROR "PyTorch needs CuDNN-8.5 or above, but found ${CUDNN_VERSION}. Builds are still possible with `USE_CUDNN=0`")
   endif()
-  set(CUDNN_FRONTEND_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR}/../third_party/cudnn_frontend/include)
+  set(CUDNN_FRONTEND_INCLUDE_DIR /opt/cuda/include)
   target_include_directories(torch::cudnn INTERFACE ${CUDNN_FRONTEND_INCLUDE_DIR})
 endif()
 
