Unset rpath to support blas-lapack-switch
Bug: https://bugs.gentoo.org/921129
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -10,7 +10,6 @@ endif(APPLE)
 set(CMAKE_SKIP_BUILD_RPATH  FALSE)
 # Don't use the install-rpath during the build phase
 set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-set(CMAKE_INSTALL_RPATH "${_rpath_portable_origin}")
 # Automatically add all linked folders that are NOT in the build directory to
 # the rpath (per library?)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
