--- gtk/gtkrc.c.~1.40.2.17.~	2001-07-10 20:06:35.000000000 +0200
+++ gtk/gtkrc.c	2003-03-05 16:56:10.000000000 +0100
@@ -453,7 +453,7 @@
 
   if (!initted)
     {
-      char *locale = setlocale (LC_CTYPE, NULL);
+      char *locale = g_strdup (setlocale (LC_CTYPE, NULL));
       char *p;
       
       initted = TRUE;
@@ -558,6 +558,7 @@
 
 	  if (tmp) g_free (tmp);
 	}
+      g_free (locale);
     }
   
   i = 0;
