diff --git a/shibokenmodule/files.dir/shibokensupport/signature/parser.py b/shibokenmodule/files.dir/shibokensupport/signature/parser.py
index be82a4f92..66d79ad27 100644
--- a/shibokenmodule/files.dir/shibokensupport/signature/parser.py
+++ b/shibokenmodule/files.dir/shibokensupport/signature/parser.py
@@ -69,7 +69,7 @@ def _get_flag_enum_option():
         flag = getattr(sys, sysname)
         if not isinstance(flag, int):
             flag = True
-    p = f"\n    *** Python is at version {'.'.join(map(str, pyminver))} now."
+    p = f"\n    *** Python is at version {'.'.join(map(str, pyminver or (0,)))} now."
     # PYSIDE-1797: Emit a warning when we may remove pep384_issue33738.cpp
     if pyminver and pyminver >= (3, 8):
         warnings.warn(f"{p} The file pep384_issue33738.cpp should be removed ASAP! ***")
-- 
cgit v1.2.1
