diff -Naur ez-ipupdate-3.0.11b8.orig/example-3322.conf ez-ipupdate-3.0.11b8/example-3322.conf
--- ez-ipupdate-3.0.11b8.orig/example-3322.conf	1970-01-01 01:00:00.000000000 +0100
+++ ez-ipupdate-3.0.11b8/example-3322.conf	2011-01-23 19:30:43.785652588 +0100
@@ -0,0 +1,19 @@
+#!/usr/local/bin/ez-ipupdate -c
+#
+# example config file for ez-ipupdate
+#
+# this file is actually executable!
+#
+
+service-type=qdns
+#service-type=qdns-static
+user=myuserid:mypassword
+host=mydomain.whatever.com
+interface=eth1
+max-interval=2073600
+
+# uncomment this once you have everything working how you want and you are
+# ready to have ez-ipupdate running in the background all the time. to stop it
+# you can use "killall -QUIT ez-ipupdate" under linux.
+#daemon
+
diff -Naur ez-ipupdate-3.0.11b8.orig/ez-ipupdate.c ez-ipupdate-3.0.11b8/ez-ipupdate.c
--- ez-ipupdate-3.0.11b8.orig/ez-ipupdate.c	2011-01-23 19:29:17.698793394 +0100
+++ ez-ipupdate-3.0.11b8/ez-ipupdate.c	2011-01-23 19:29:17.799792055 +0100
@@ -103,6 +103,11 @@
 #define HEIPV6TB_DEFAULT_PORT "80"
 #define HEIPV6TB_REQUEST "/index.cgi"
 
+#define QDNS_DEFAULT_SERVER "members.3322.org"
+#define QDNS_DEFAULT_PORT "80"
+#define QDNS_REQUEST "/dyndns/update"
+#define QDNS_STAT_REQUEST "/dyndns/update"
+
 #define DEFAULT_TIMEOUT 120
 #define DEFAULT_UPDATE_PERIOD 120
 #define DEFAULT_RESOLV_PERIOD 30
@@ -514,6 +519,26 @@
     HEIPV6TB_DEFAULT_PORT,
     HEIPV6TB_REQUEST
   },
+  { "qdns",
+    { "qdns", 0, 0, },
+    DYNDNS_init,
+    DYNDNS_update_entry,
+    DYNDNS_check_info,
+    DYNDNS_fields_used,
+    QDNS_DEFAULT_SERVER,
+    QDNS_DEFAULT_PORT,
+    QDNS_REQUEST
+  },
+  { "qdns-static",
+    { "qdns-static", "qdns-stat", "statdns", },
+    DYNDNS_init,
+    DYNDNS_update_entry,
+    DYNDNS_check_info,
+    DYNDNS_STAT_fields_used,
+    QDNS_DEFAULT_SERVER,
+    QDNS_DEFAULT_PORT,
+    QDNS_STAT_REQUEST
+  },
 };
 
 static struct service_t *service = NULL;
