From c2e07357abc86fc0e7715ef21dc3d279e3002819 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Tue, 29 Aug 2017 08:55:26 -0400
Subject: [PATCH] Another parallel make fix

Bug: https://bugs.gentoo.org/629278
---
 font/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/font/Makefile b/font/Makefile
index 1cd7ce3..2c85e64 100644
--- a/font/Makefile
+++ b/font/Makefile
@@ -236,7 +236,7 @@ psf: bdf
 #
 # Print coverage of scripts in Basic Multilingual Plane in .txt file.
 #
-coverage:
+coverage: hex
 	sort $(HEXDIR)/*.hex | \
 		$(BINDIR)/unicoverage > $(COMPILED_DIR)/coverage.txt
 
-- 
2.14.1

