--- mserv-0.35.orig/mserv/defconf.c	2003-08-28 20:10:17.000000000 -0700
+++ mserv-0.35/mserv/defconf.c	2003-08-28 23:04:14.000000000 -0700
@@ -16,11 +16,11 @@
 # Define player invokation methods\n\
 # mservplay is our special wrapper, the first parameter is a 'nice' level\n\
 # play is part of sox\n\
-prog_mpg123=/usr/local/bin/mpg123 -b 1024\n\
-prog_freeamp=/usr/local/bin/freeamp -ui mpg123\n\
-prog_mservplay=/usr/local/bin/mservplay 0 mpg123 -b 1024\n\
-prog_play=/usr/local/bin/play\n\
-prog_ogg123=/usr/local/bin/ogg123\n\
+prog_mpg123=/usr/bin/mpg123 -b 1024\n\
+prog_freeamp=/usr/bin/freeamp -ui mpg123\n\
+prog_mservplay=/usr/bin/mservplay 0 mpg123 -b 1024\n\
+prog_play=/usr/bin/play\n\
+prog_ogg123=/usr/bin/ogg123\n\
 \n\
 # Set players for each file extension we want to support, unknown extensions\n\
 # are ignored by mserv\n\
--- mserv-0.35.orig/webclient/index.html	2003-08-28 20:10:17.000000000 -0700
+++ mserv-0.35/webclient/index.html	2003-08-29 17:02:10.000000000 -0700
@@ -9,7 +9,7 @@
 <tr><td>
 <center>
 <table>
-<form method=post action="main.cgi">
+<form method=post action="/cgi-bin/mserv/main.cgi">
 <tr>
   <td>User</td>
   <td><input type=text name=user size=10 maxsize=10></td>
@@ -23,6 +23,6 @@
 </center>
 </td></tr>
 </table>
-<p><a href="main.cgi">GUEST ACCESS</a></p>
+<p><a href="/cgi-bin/mserv/main.cgi">GUEST ACCESS</a></p>
 </center>
 </html>
--- mserv-0.35.orig/webclient/margin.cgi	2003-08-28 20:10:17.000000000 -0700
+++ mserv-0.35/webclient/margin.cgi	2003-08-30 16:14:27.000000000 -0700
@@ -15,11 +15,11 @@
 <head>
   <title>Mserv web interface - margin</title>
 </head>
-<body bgcolor="#ffffff" background="back.gif">
+<body bgcolor="#ffffff" background="/mserv/back.gif">
 <p align=center><center>
 <table>
 <tr><td>
-<p align=center><img src="mserv7.gif" alt="Mserv"></p>
+<p align=center><img src="/mserv/mserv7.gif" alt="Mserv"></p>
 <p><font size="2"><a target=main href="mserv.cgi?user=$user&pass=$pass&page=status">Status</a>
 <br><a target=main href="mserv.cgi?user=$user&pass=$pass&page=albums">Albums</a>
 <br><a target=main href="mserv.cgi?user=$user&pass=$pass&page=queue">Queue</a>
--- mserv-0.35.orig/webclient/mserv.cgi	2003-08-28 20:10:17.000000000 -0700
+++ mserv-0.35/webclient/mserv.cgi	2003-08-30 16:14:47.000000000 -0700
@@ -3,7 +3,7 @@
 $guestuser = 'guest';
 $guestpass = 'guest';
 $host = 'localhost:4444';
-$mservcmd = '/usr/local/bin/mservcmd';
+$mservcmd = '/usr/bin/mservcmd';
 
 use POSIX;
 use CGI;
@@ -34,7 +34,7 @@
 <head>
   <title>Mserv</title>
 </head>
-<body bgcolor="#ffffff" background="back.gif">
+<body bgcolor="#ffffff" background="/mserv/back.gif">
 EOF
 
 if ($page eq 'status') {
