 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1fda8ee..e2170b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -615,8 +615,8 @@ if(PYTHON OR PYTHON_MODULE)
 		else()
 			set(PythonLibs_FIND_VERSION 2)
 		endif()
-		find_package(PythonLibs)
 		include(FindPythonInterp)
+		find_package(PythonLibs)
 	endif()
 	set(USE_PYTHON ${PYTHONLIBS_FOUND})
 	if(PYTHONLIBS_FOUND)
