--- a/autoconf/configure.in
+++ b/autoconf/configure.in
@@ -59,7 +59,6 @@
 AC_CHECK_HEADERS(sys/wait.h sys/time.h)
 AC_CHECK_HEADERS(sys/statvfs.h sys/vfs.h sys/mount.h ustat.h)
 AC_CHECK_HEADERS(machine/soundcard.h linux/soundcard.h linux/radio.h)
-AC_CONFIG_HEADER(config.h)
 
 dnl# Check for X
 
@@ -117,10 +116,6 @@
 AC_SUBST(with_gnustep_lib)
 AC_SUBST(with_afterdir)
 
-dnl# Write results
-
-AC_CONFIG_HEADER(config.h)
-
 dnl# Common parts of the Makefile
 MAKEFILEDEFINES=./autoconf/Makefile.defines
 MAKEFILECOMMON=./autoconf/Makefile.common
