--- a/UnitTests/CMakeLists.txt	2019-01-08 04:36:18.000000000 -0800
+++ a/UnitTests/CMakeLists.txt	2019-05-21 13:00:47.785964900 -0700
@@ -1,7 +1,5 @@
-ADD_SUBDIRECTORY (googletest EXCLUDE_FROM_ALL)
 enable_testing()
 	
-SET(gtest_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/googletest/googletest")
 add_definitions( -DTESTFILESPATH="${CMAKE_CURRENT_SOURCE_DIR}/../TestFiles")
 add_definitions( -DLTESTFILESPATH=L"${CMAKE_CURRENT_SOURCE_DIR}/../TestFiles")
 add_definitions( -DLOUTFILESPATH=L"${CMAKE_BINARY_DIR}/")
