diff --git a/Makefile_LINUX_gcc42 b/Makefile_LINUX_gcc42
index 52c5286..5c894a0 100644
--- a/Makefile_LINUX_gcc42
+++ b/Makefile_LINUX_gcc42
@@ -250,7 +250,10 @@
 
 
 # Version string
-VERSION = 0.9.0
+MAJOR = 0
+MINOR = 9
+MICRO = 0
+VERSION = ${MAJOR}.${MINOR}.${MICRO}
 
 
 #
@@ -282,7 +285,7 @@ SHAR	= /usr/bin/shar
 #
 # Program to use to create archives
 #
-AR	= /usr/bin/ar
+AR	= ar
 
 #
 # Program to use to add an index to an archive
@@ -292,13 +295,13 @@ RANLIB  = /usr/bin/ranlib
 #
 # Program to use to decompress a data file
 #
-DECOMPRESS = /usr/bin/bunzip2
+DECOMPRESS = bunzip2
 
 
 #
 # Program to use to compress a data file
 #
-COMPRESS = /usr/bin/bzip2
+COMPRESS = bzip2
 
 #
 # Program to use to generate a signature
@@ -486,9 +489,9 @@ F90SOURCE = $(SRC)/fcb_atol_wcnt.f90     \
 		    $(SRC)/fcb_read_image.f90    \
 		    $(SRC)/fcb_read_line.f90     \
 		    $(SRC)/fcb_read_xds_i2.f90   \
-		    $(SRC)/fcb_skip_whitespace.f90 \
-		    $(EXAMPLES)/test_fcb_read_image.f90 \
-		    $(EXAMPLES)/test_xds_binary.f90
+		    $(SRC)/fcb_skip_whitespace.f90
+#		    $(EXAMPLES)/test_fcb_read_image.f90 \
+#		    $(EXAMPLES)/test_xds_binary.f90
 		    
 		   
 #
@@ -663,7 +666,7 @@ all::	$(BIN)/test_xds_binary   \
 		$(BIN)/test_fcb_read_image
 endif
 
-shared:	$(SOLIB)/libcbf.so $(SOLIB)/libfcb.so $(SOLIB)/libimg.so
+shared:	$(SOLIB)/libcbf.so.${VERSION} $(SOLIB)/libfcb.so.${VERSION} $(SOLIB)/libimg.so.${VERSION}
 
 javawrapper: shared $(JAVADIR) $(JAVADIR)/cbflib-$(VERSION).jar $(SOLIB)/libcbf_wrap.so
 
@@ -846,10 +849,11 @@ ifneq ($(RANLIB),)
 	$(RANLIB) $@
 endif
 
-$(SOLIB)/libcbf.so: $(SOURCE) $(HEADERS) $(COMMONDEP) $(SOLIB)
+$(SOLIB)/libcbf.so.${VERSION}: $(SOURCE) $(HEADERS) $(COMMONDEP) $(SOLIB)
 	$(CC) $(CFLAGS) $(SOCFLAGS) $(INCLUDES) $(WARNINGS) -c $(SOURCE)
 	mv *.o $(SOLIB)
-	$(CC) -o $@ $(SOLIB)/*.o $(SOLDFLAGS) $(EXTRALIBS)
+	$(CC) $(LDFLAGS) -Wl,-soname,libcbf.so.$(VERSION) $(SOCFLAGS) $(SOLDFLAGS) -o $@ $(SOLIB)/*.o $(EXTRALIBS)
+	ln -sf libcbf.so.$(VERSION) $(SOLIB)/libcbf.so
 
 #
 # IMG library
@@ -862,9 +866,10 @@ ifneq ($(RANLIB),)
 endif
 	rm img.o
 	
-$(SOLIB)/libimg.so: $(SOURCE) $(HEADERS) $(COMMONDEP) $(SOLIB)
+$(SOLIB)/libimg.so.${VERSION}: $(SOURCE) $(HEADERS) $(COMMONDEP) $(SOLIB)
 	$(CC) $(CFLAGS) $(SOCFLAGS) $(INCLUDES) $(WARNINGS) -c $(EXAMPLES)/img.c
-	$(CC) -o $@ img.o $(SOLDFLAGS)
+	$(CC) $(LDFLAGS) -Wl,-soname,libimg.so.$(VERSION) $(SOCFLAGS) $(SOLDFLAGS) -o $@ img.o
+	ln -sf libimg.so.$(VERSION) $(SOLIB)/libimg.so
 	rm img.o
 
 #
@@ -888,10 +893,11 @@ else
 	echo "Define F90C to build $(LIB)/libfcb.a"
 endif
 
-$(SOLIB)/libfcb.so: $(F90SOURCE) $(HEADERS) $(COMMONDEP) $(SOLIB)
+$(SOLIB)/libfcb.so.${VERSION}: $(F90SOURCE) $(HEADERS) $(COMMONDEP) $(SOLIB)
 ifneq ($(F90C),)
 	$(F90C) $(F90FLAGS) $(SOCFLAGS) $(INCLUDES) $(WARNINGS) -c $(F90SOURCE)
-	$(F90C) $(F90FLAGS) -o $@ *.o $(SOLDFLAGS)
+	$(F90C) $(LDFLAGS) -Wl,-soname,libfcb.so.$(VERSION) $(SOCFLAGS) $(SOLDFLAGS) $(F90FLAGS) -o $@ *.o
+	ln -sf libfcb.so.$(VERSION) $(SOLIB)/libfcb.so
 	rm *.o
 else
 	echo "Define F90C to build $(SOLIB)/libfcb.so"
