--- a/mozilla/memory/jemalloc/src/configure.ac	2014-10-11 05:06:41.000000000 -0400
+++ b/mozilla/memory/jemalloc/src/configure.ac	2014-10-16 13:41:10.809799170 -0400
@@ -887,7 +887,7 @@
 dnl Check whether the BSD/SUSv1 sbrk() exists.  If not, disable DSS support.
 AC_CHECK_FUNC([sbrk], [have_sbrk="1"], [have_sbrk="0"])
 if test "x$have_sbrk" = "x1" ; then
-  if test "x$sbrk_deprecated" == "x1" ; then
+  if test "x$sbrk_deprecated" = "x1" ; then
     AC_MSG_RESULT([Disabling dss allocation because sbrk is deprecated])
     enable_dss="0"
   else
@@ -1180,7 +1180,7 @@
 		printf("%d\n", rv);
 	}
 ], [je_cv_gcc_builtin_ffsl])
-if test "x${je_cv_gcc_builtin_ffsl}" == "xyes" ; then
+if test "x${je_cv_gcc_builtin_ffsl}" = "xyes" ; then
   AC_DEFINE([JEMALLOC_INTERNAL_FFSL], [__builtin_ffsl])
   AC_DEFINE([JEMALLOC_INTERNAL_FFS], [__builtin_ffs])
 else
@@ -1194,7 +1194,7 @@
 		printf("%d\n", rv);
 	}
   ], [je_cv_function_ffsl])
-  if test "x${je_cv_function_ffsl}" == "xyes" ; then
+  if test "x${je_cv_function_ffsl}" = "xyes" ; then
     AC_DEFINE([JEMALLOC_INTERNAL_FFSL], [ffsl])
     AC_DEFINE([JEMALLOC_INTERNAL_FFS], [ffs])
   else
