--- a/mozilla/js/xpconnect/src/XPCWrappedNative.cpp.ORIG	2018-07-09 21:54:43.000000000 +0200
+++ b/mozilla/js/xpconnect/src/XPCWrappedNative.cpp	2019-05-04 17:49:23.893560494 +0200
@@ -2218,15 +2218,16 @@
                 const char* fmt = (i == 0) ?
                                     "(%s" : (i == count-1) ?
                                         ", %s)" : ", %s";
+                
+                if (!name) {
+                    return nullptr;
+                }
                 name = JS_sprintf_append(name, fmt,
                                          array[i]->GetNameString());
             }
         }
     }
 
-    if (!name) {
-        return nullptr;
-    }
     const char* fmt = "[xpconnect wrapped %s" FMT_ADDR FMT_STR(" (native")
         FMT_ADDR FMT_STR(")") "]";
     if (si) {
