--- src/support/xml_io.cxx.orig	2016-09-29 17:43:03.299747587 +0000
+++ src/support/xml_io.cxx	2016-09-29 17:43:41.639222816 +0000
@@ -85,7 +85,7 @@
 
 bool ptt_on = false;
 
-int ignore = 0; // skip next "ignore" read loops
+int xml_ignore = 0; // skip next "xml_ignore" read loops
 
 int try_count = CHECK_UPDATE_COUNT;
 
@@ -155,7 +155,7 @@
 
 	try {
 		execute(rig_set_modes, modes, res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -176,7 +176,7 @@
 
 	try {
 		execute(rig_set_bandwidths, bandwidths, res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -190,7 +190,7 @@
 	try {
 		XmlRpcValue res;
 		execute(rig_set_name, XmlRpcValue(selrig->name_), res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -204,7 +204,7 @@
 	try {
 		XmlRpcValue res;
 		execute((PTT ? main_set_tx : main_set_rx), XmlRpcValue(), res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -219,7 +219,7 @@
 		xmlvfo.freq = freq;
 		XmlRpcValue f((double)freq), res;
 		execute(rig_set_frequency, f, res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -233,7 +233,7 @@
 	try {
 		XmlRpcValue mval((int)val), res;
 		execute(rig_set_smeter, mval, res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -247,7 +247,7 @@
 	try {
 		XmlRpcValue mval((int)val), res;
 		execute(rig_set_pwrmeter, mval, res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -261,7 +261,7 @@
 	try {
 		XmlRpcValue i(freq), res;
 		execute(rig_set_notch, i, res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 		if (freq == 0)
 			xml_notch_on = false;
@@ -299,7 +299,7 @@
 		xmlvfo.imode = md;
 		XmlRpcValue mode(selrig->modes_[md]), res;
 		execute(rig_set_mode, mode, res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -315,7 +315,7 @@
 		int selbw = (bw > 0x80) ? (bw >> 8 & 0x7F) : bw;
 		XmlRpcValue bandwidth(selrig->bandwidths_[selbw]), res;
 		execute(rig_set_bandwidth, bandwidth, res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -329,7 +329,7 @@
 	try {
 		XmlRpcValue sideband(selrig->get_modetype(vfo.imode) == 'U' ? "USB" : "LSB"), res;
 		execute(main_set_wf_sideband, sideband, res);
-		ignore = 2;
+		xml_ignore = 2;
 		err_count = 5;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		err_count--;
@@ -501,7 +501,7 @@
 		LOG_ERROR("Connected");
 		Fl::awake(set_fldigi_connect, (void *)1);
 		err_count = 5;
-		ignore = 2;
+		xml_ignore = 2;
 	} catch (const XmlRpc::XmlRpcException& e) {
 		throw e;
 	}
@@ -526,8 +526,8 @@
 
 static void get_fldigi_status()
 {
-	if (ignore) {
-		--ignore;
+	if (xml_ignore) {
+		--xml_ignore;
 		return;
 	}
 	XmlRpcValue status;
