--- common.h
+++ common.h
@@ -38,7 +38,7 @@
 #define STDCALL 
 #endif
 
-#include <iostream.h>
+#include <iostream>
 #include "includes/base/myDisplayLists.h"
 #include "myTimeStats.h"
 
--- includes/base/myMatrix.h
+++ includes/base/myMatrix.h
@@ -57,7 +57,7 @@
 #include "my3dPoint.h"
 #include "myVector.h"
 
-#include <iostream.h>
+#include <iostream>
 #include <math.h>
 
 /**
--- includes/osd/menu.h
+++ includes/osd/menu.h
@@ -21,6 +21,7 @@
 #define POOPMUP_MENU_H
 
 #include <stdio.h>
+#include <cstring>
 #include "../base/myColor.h"
 #include "formElem/button.h"
 #include "formElem/intSelector.h"
--- includes/osd/onScreenDisplay.h
+++ includes/osd/onScreenDisplay.h
@@ -21,6 +21,7 @@
 #define ONSCREENDISPLAY_H
 
 #include <stdio.h>
+#include <cstring>
 #include "../base/my3dPoint.h"
 #include "../base/myColor.h"
 
