diff -urN wmsvencd-0.5.0.orig/Makefile wmsvencd-0.5.0/Makefile
--- wmsvencd-0.5.0.orig/Makefile	1999-05-08 09:54:08.000000000 +0200
+++ wmsvencd-0.5.0/Makefile	2003-09-18 03:56:41.448394312 +0200
@@ -1,11 +1,13 @@
 BINDIR = /usr/local/bin
 MANDIR = /usr/local/man/man1
 CFLAGS = -O6 -fno-strength-reduce
+LIBDIR = -L/usr/X11R6/lib
+LIBS = -lXpm -lXext -lX11
 
 all: wmsvencd
 
 wmsvencd: cdctl.o wmsvencd.o
-	c++ -o wmsvencd $(CFLAGS) -L/usr/X11R6/lib wmsvencd.o cdctl.o -lXpm -lXext -lX11
+	c++ -o wmsvencd $(CFLAGS) $(LIBDIR) wmsvencd.o cdctl.o $(LIBS)
 
 wmsvencd.o: wmsvencd.cc
 	c++ $(CFLAGS) -I/usr/X11R6/include -I/include -c wmsvencd.cc -o wmsvencd.o
@@ -21,4 +23,4 @@
 install:
 	mkdir -p $(BINDIR)
 	install -c -s -m 0755 wmsvencd $(BINDIR)
-	install -c -m 0644 wmsvencd.1x $(MANDIR)
+	install -c -m 0644 wmsvencd.1x $(MANDIR)
\ No newline at end of file
diff -urN wmsvencd-0.5.0.orig/cdctl.cc wmsvencd-0.5.0/cdctl.cc
--- wmsvencd-0.5.0.orig/cdctl.cc	1999-05-08 05:51:29.000000000 +0200
+++ wmsvencd-0.5.0/cdctl.cc	2003-09-18 03:55:06.393844800 +0200
@@ -35,6 +35,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
