From db50d885a87985310e65159a705b45de8646dc26 Mon Sep 17 00:00:00 2001
From: AIZAWA Hina <hina@bouhime.com>
Date: Fri, 23 Mar 2018 22:50:51 +0900
Subject: [PATCH] Add supporting LibreSSL 2.7

Signed-off-by: AIZAWA Hina <hina@bouhime.com>
---
 deps/neverbleed/neverbleed.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deps/neverbleed/neverbleed.c b/deps/neverbleed/neverbleed.c
index 29b35a9..42356a6 100644
--- a/deps/neverbleed/neverbleed.c
+++ b/deps/neverbleed/neverbleed.c
@@ -547,7 +547,7 @@ static int sign_stub(struct expbuf_t *buf)
     return 0;
 }
 
-#if !OPENSSL_1_1_API
+#if !OPENSSL_1_1_API && (!defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER < 0x2070000fL)
 
 static void RSA_get0_key(const RSA *rsa, const BIGNUM **n, const BIGNUM **e, const BIGNUM **d)
 {
