diff -ruN grok-0.9.2.orig/Makefile grok-0.9.2/Makefile
--- grok-0.9.2.orig/Makefile	2018-05-05 17:26:35.451938040 +0200
+++ grok-0.9.2/Makefile	2018-05-05 17:29:03.414780593 +0200
@@ -93,12 +93,12 @@
 
 install: libgrok.$(LIBSUFFIX) grok discogrok $(GROKHEADER)
 	install -d $(DESTDIR)$(PREFIX)/bin
-	install -d $(DESTDIR)$(PREFIX)/lib
+	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR)
 	install -d $(DESTDIR)$(PREFIX)/include
 	install -m 755 grok $(DESTDIR)$(PREFIX)/bin
 	install -m 755 discogrok $(DESTDIR)$(PREFIX)/bin
-	install -m 644 libgrok.$(LIBSUFFIX) $(DESTDIR)$(PREFIX)/lib
-	ln -s libgrok.$(LIBSUFFIX) $(DESTDIR)$(PREFIX)/lib/libgrok.$(LIBSUFFIX).1
+	install -m 644 libgrok.$(LIBSUFFIX) $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	ln -s libgrok.$(LIBSUFFIX) $(DESTDIR)$(PREFIX)/$(LIBDIR)/libgrok.$(LIBSUFFIX).1
 	for header in $(GROKHEADER); do \
 		install -m 644 $$header $(DESTDIR)$(PREFIX)/include; \
 	done 
@@ -109,7 +109,7 @@
 uninstall:
 	rm -f $(DESTDIR)$(PREFIX)/bin/grok
 	rm -f $(DESTDIR)$(PREFIX)/bin/discogrok
-	rm -f $(DESTDIR)$(PREFIX)/lib/libgrok.$(LIBSUFFIX)
+	rm -f $(DESTDIR)$(PREFIX)/$(LIBDIR)/libgrok.$(LIBSUFFIX)
 	for header in $(GROKHEADER); do \
 		rm -f $(DESTDIR)$(PREFIX)/include/$$header; \
 	done 
