--- rstudio-1.2.1335-orig/src/cpp/CMakeLists.txt	2019-03-28 16:08:20.000000000 +1100
+++ rstudio-1.2.1335/src/cpp/CMakeLists.txt	2019-04-09 17:58:59.926559001 +1000
@@ -117,7 +117,7 @@
          add_definitions(-O0)
          add_definitions(-D_GLIBCXX_ASSERTIONS)
       endif()
-      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pie -Wl,-z,relro,-z,now")
+      set(CMAKE_EXE_LINKER_FLAGS "-pie -Wl,-z,relro,-z,now ${CMAKE_EXE_LINKER_FLAGS}")
    endif()
 
    # other useful gcc diagnostics
