--- rstudio-0.99.473-orig/src/cpp/desktop/CMakeLists.txt	2015-08-12 09:58:43.000000000 +1000
+++ rstudio-0.99.473/src/cpp/desktop/CMakeLists.txt	2015-08-30 23:16:11.396175587 +1000
@@ -93,7 +93,12 @@
    find_package(Qt5DBus REQUIRED)
 endif()
 
-
+find_path(QTSINGLEAPPLIB_INCLUDE_DIR NAMES QtSingleApplication
+  PATH_SUFFIXES QtSolutions
+  PATHS
+  /usr/include/qt5)
+find_library(QTSINGLEAPPLIB Qt5Solutions_SingleApplication-2.6
+  libQt5Solutions_SingleApplication-2.6)
 
 # disable clang warnings for qt sources
 if(APPLE)
@@ -113,7 +118,6 @@
 
 set(MOC_DESKTOP_HEADER_FILES ${DESKTOP_HEADER_FILES})
 list(REMOVE_ITEM MOC_DESKTOP_HEADER_FILES
-  ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qtsingleapplication/qtlockedfile.h
   ${CMAKE_CURRENT_SOURCE_DIR}/DesktopDetectRHome.hpp
   ${CMAKE_CURRENT_SOURCE_DIR}/DesktopOptions.hpp
   ${CMAKE_CURRENT_SOURCE_DIR}/DesktopRVersion.hpp
@@ -175,8 +179,6 @@
   DesktopSubMenu.cpp
   DesktopSynctex.cpp
   DesktopNetworkAccessManager.cpp
-  3rdparty/qtsingleapplication/qtsingleapplication.cpp
-  3rdparty/qtsingleapplication/qtlocalpeer.cpp
 )
 
 if(WIN32)
@@ -219,6 +221,7 @@
   ${CORE_SOURCE_DIR}/include
   ${CMAKE_CURRENT_SOURCE_DIR}
   ${CMAKE_CURRENT_BINARY_DIR}
+  ${QTSINGLEAPPLIB_INCLUDE_DIR}
   ${TESTS_INCLUDE_DIR}
 )
 
@@ -332,6 +335,7 @@
 
    # set link dependencies
    target_link_libraries(rstudio
+      ${QTSINGLEAPPLIB}
       ${QT_LIBRARIES}
       rstudio-core
       ${LINK_FLAGS}
--- rstudio-0.99.473-orig/src/cpp/desktop/DesktopPosixApplication.hpp	2015-08-12 09:58:43.000000000 +1000
+++ rstudio-0.99.473/src/cpp/desktop/DesktopPosixApplication.hpp	2015-08-30 21:41:21.616934106 +1000
@@ -16,7 +16,7 @@
 #ifndef DESKTOP_POSIX_APPLICATION_HPP
 #define DESKTOP_POSIX_APPLICATION_HPP
 
-#include "3rdparty/qtsingleapplication/QtSingleApplication"
+#include <QtSingleApplication>
 
 namespace rstudio {
 namespace desktop {
