Don't install test files
https://bugs.gentoo.org/886031

--- a/setup.py
+++ b/setup.py
@@ -40,7 +40,7 @@ setuptools.setup(name = "volatility3",
                      '': ['development', 'development.*'],
                      'development': ['*']
                  },
-                 packages = setuptools.find_namespace_packages(exclude = ["development", "development.*"]),
+                 packages = setuptools.find_namespace_packages(exclude = ["development", "development.*", "test", "test.*"]),
                  entry_points = {
                      'console_scripts': [
                          'vol = volatility3.cli:main',
