diff -Naur webserver-1.2.103.orig/admin/SystemStats.py webserver-1.2.103/admin/SystemStats.py
--- webserver-1.2.103.orig/admin/SystemStats.py	2013-01-26 17:52:44.000000000 -0500
+++ webserver-1.2.103/admin/SystemStats.py	2013-07-15 07:27:11.000000000 -0400
@@ -39,7 +39,7 @@
     global _stats
 
     if not _stats:
-        if sys.platform == 'linux2':
+        if sys.platform == 'linux2' or sys.platform == 'linux3':
             _stats = System_stats__Linux()
         elif sys.platform == 'darwin':
             _stats = System_stats__Darwin()
diff -Naur webserver-1.2.103.orig/admin/util.py webserver-1.2.103/admin/util.py
--- webserver-1.2.103.orig/admin/util.py	2013-01-26 17:52:44.000000000 -0500
+++ webserver-1.2.103/admin/util.py	2013-07-15 07:29:12.000000000 -0400
@@ -341,7 +341,7 @@
 def os_get_document_root():
     if sys.platform == 'darwin':
         return "/Library/WebServer/Documents"
-    elif sys.platform == 'linux2':
+    elif sys.platform == 'linux2' or sys.platform == 'linux3':
         if os.path.exists ("/etc/redhat-release"):
             return '/var/www'
         elif os.path.exists ("/etc/fedora-release"):
diff -Naur webserver-1.2.103.orig/admin/wizards/php.py webserver-1.2.103/admin/wizards/php.py
--- webserver-1.2.103.orig/admin/wizards/php.py	2013-01-26 17:52:44.000000000 -0500
+++ webserver-1.2.103/admin/wizards/php.py	2013-07-15 07:29:47.000000000 -0400
@@ -607,7 +607,7 @@
         first_group = str(root_group)
 
     # Systems
-    if sys.platform == 'linux2':
+    if sys.platform == 'linux2' or sys.platform == 'linux3':
         if os.getuid() == 0:
             return root_group
         return first_group
