--- gnfs/poly/stage2/root_sieve_deg45_x.c	2011-08-23 20:59:59.000000000 +0800
+++ gnfs/poly/stage2/root_sieve_deg45_x.c.new	2012-11-30 17:09:22.553116526 +0800
 		return;
 	}
 
@@ -585,7 +585,7 @@
 		if (obj->flags & MSIEVE_FLAG_STOP_SIEVING)
 			break;
 	}
-	printf("\n");
+	printf(",");
 }
 
 /*-------------------------------------------------------------------------*/
@@ -604,7 +604,7 @@
 	xline_heap_t xline_heap;
 	uint32 cutoff_score;
 
-	printf("L %u\n", (uint32)lattice_size);
+	/* printf("L %u\n", (uint32)lattice_size); */
 
 	mpz_set_ui(xy->y_base, (unsigned long)0);
 	xy->y_blocks = 0;
@@ -621,7 +621,7 @@
 		x->x_blocks = line_max - line_min;
 		x->curr_score = 0;
 		root_sieve_line(rs);
-		printf("\n");
+		printf("_");
 		return;
 	}
 
@@ -664,5 +664,5 @@
 			break;
 	}
 
-	printf("\n");
+	printf("*");
 }
--- gnfs/poly/stage2/root_sieve_deg5_xy.c	2011-07-15 09:49:00.000000000 +0800
+++ gnfs/poly/stage2/root_sieve_deg5_xy.c.new	2012-11-30 17:09:51.462454813 +0800
@@ -290,8 +290,8 @@
 	}
 	xy->y_blocks = y_blocks;
 
-	printf("\n%.0lf %u %u\n", (double)lattice_size, 
-			y_blocks, num_lattices);
+	/* printf("\n%.0lf %u %u\n", (double)lattice_size, 
+			y_blocks, num_lattices); */
 
 	sieve_x_run_deg5(rs);
 }
