diff --git a/src/lib/libderiv/Makefile.library b/src/lib/libderiv/Makefile.library
index ee152ef..8debfe8 100644
--- a/src/lib/libderiv/Makefile.library
+++ b/src/lib/libderiv/Makefile.library
@@ -23,7 +23,7 @@ $(TOPDIR)/lib/$(NAME).a: $(LIBOBJ)
 
 # this is how shared library is made
 $(TOPDIR)/lib/$(NAME).la: $(LIBOBJ)
-	$(LTLINK) $(CXX) -o $@ $^ $(LTLINKLIBOPTS)
+	$(LTLINK) $(CXX) $(LDFLAGS) -o $@ $^ $(LTLINKLIBOPTS)
 
 $(TOPDIR)/include/$(NAME)/$(NAME).h: $(NAME).h
 	$(INSTALL) $(INSTALLDIROPT) $(TOPDIR)/include/$(NAME)
diff --git a/src/lib/libint/Makefile.library b/src/lib/libint/Makefile.library
index 23e2918..01c4225 100644
--- a/src/lib/libint/Makefile.library
+++ b/src/lib/libint/Makefile.library
@@ -41,7 +41,7 @@ $(TOPDIR)/lib/$(NAME).a: $(LIBOBJ)
 
 # this is how shared library is made
 $(TOPDIR)/lib/$(NAME).la: $(LIBOBJ)
-	$(LTLINK) $(CXX) -o $@ $^ $(LTLINKLIBOPTS)
+	$(LTLINK) $(CXX) $(LDFLAGS) -o $@ $^ $(LTLINKLIBOPTS)
 
 $(TOPDIR)/include/$(NAME)/$(NAME).h: $(NAME).h
 	$(INSTALL) $(INSTALLDIROPT) $(TOPDIR)/include/$(NAME)
diff --git a/src/lib/libr12/Makefile.library b/src/lib/libr12/Makefile.library
index 1510807..26d47a4 100644
--- a/src/lib/libr12/Makefile.library
+++ b/src/lib/libr12/Makefile.library
@@ -47,7 +47,7 @@ $(TOPDIR)/lib/$(NAME).a: $(LIBOBJ)
 
 # this is how shared library is made
 $(TOPDIR)/lib/$(NAME).la: $(LIBOBJ)
-	$(LTLINK) $(CXX) -o $@ $^ $(LTLINKLIBOPTS)
+	$(LTLINK) $(CXX) $(LDFLAGS) -o $@ $^ $(LTLINKLIBOPTS)
 
 $(TOPDIR)/include/$(NAME)/$(NAME).h: $(NAME).h
 	$(INSTALL) $(INSTALLDIROPT) $(TOPDIR)/include/$(NAME)
