--- a/lib/cconv.c
+++ b/lib/cconv.c
@@ -790,7 +790,7 @@
 static int parseLine(uchar *, uchar **, int);
 static FILE *openfile(char *);
 static int doinclude(uchar *, Files *, void (*)());
-static uchar *getline(uchar *, int, Files *, void (*)());
+static uchar *getLine(uchar *, int, Files *, void (*)());
 static int readRuleFile(ccRule, char *);
 static int registMode(ccRule, int, uchar **);
 static int newMode(ccRule, Files *, _strbufRec *, _funcbufRec *,
@@ -1035,8 +1035,8 @@
 	return 0;
 }
 
-/* getline -- 1行読み込む (その際 include の処理を行なう) */
-static uchar *getline(line, linesize, files, efunc)
+/* getLine -- 1行読み込む (その際 include の処理を行なう) */
+static uchar *getLine(line, linesize, files, efunc)
 uchar *line;
 int linesize;
 Files *files;
@@ -1100,7 +1100,7 @@
 	rule->nmode = 0;
 	rule->initialmode = -1;
 
-	while (getline(line, sizeof(line), &files, efunc)) {
+	while (getLine(line, sizeof(line), &files, efunc)) {
 		(void)Strcpy(tmp, line);
 		if ((argc = parseLine(tmp, argv, 20)) == 0)
 			continue;
@@ -1269,7 +1269,7 @@
 	cdbuf.cdbuf = NULL;
 
 	/* ルールを読んでストアする */
-	while (getline(line, sizeof(line), files, efunc)) {
+	while (getLine(line, sizeof(line), files, efunc)) {
 		/* '#' で始まる行はコメント */
 		if (*line == '\0' || *line == '\n' || *line == '#')
 			continue;
--- a/lib/wnnlib.c
+++ b/lib/wnnlib.c
@@ -1367,7 +1367,7 @@
 
 	/* 変換文字列の長さのチェック */
 	clp = buf->clauseInfo + cl;
-	len = jl_kanji_len(buf->wnn, cl);
+	len = jl_kanji_len(buf->wnn, cl, -1);
 	diff = len - ((clp + 1)->dispp - clp->dispp);
 	newlen = (buf->displayEnd - buf->displayBuf) + diff;
 	if (newlen > buf->bufferSize) {
