diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 47a050e2..5dd5ff3f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -123,10 +123,8 @@ endif(STATIC_LINKING)
 
 target_link_libraries(libbpftrace parser resources runtime aot ast arch cxxdemangler_llvm)
 
-if (LIBBPF_BTF_DUMP_FOUND)
-  target_include_directories(runtime PUBLIC ${LIBBPF_INCLUDE_DIRS})
-  target_link_libraries(runtime ${LIBBPF_LIBRARIES})
-endif(LIBBPF_BTF_DUMP_FOUND)
+target_include_directories(runtime PUBLIC ${LIBBPF_INCLUDE_DIRS})
+target_link_libraries(runtime ${LIBBPF_LIBRARIES})
 
 if(HAVE_BFD_DISASM)
   if(STATIC_LINKING)
