diff --git a/man/man8/CMakeLists.txt b/man/man8/CMakeLists.txt
index 45cfe5d..d13229e 100644
--- a/man/man8/CMakeLists.txt
+++ b/man/man8/CMakeLists.txt
@@ -1,12 +1,12 @@
-find_program(GZIP gzip)
+find_program(CAT cat)
 file(GLOB FILES *.8)
 set(GZFILES "")
 foreach(FIL ${FILES})
   get_filename_component(NAME ${FIL} NAME)
-  add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.gz
-    COMMAND ${GZIP} -c ${FIL} > ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.gz
+  add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${NAME}
+    COMMAND ${CAT} ${FIL} > ${CMAKE_CURRENT_BINARY_DIR}/${NAME}
     DEPENDS ${FIL})
-  list(APPEND GZFILES "${CMAKE_CURRENT_BINARY_DIR}/${NAME}.gz")
+  list(APPEND GZFILES "${CMAKE_CURRENT_BINARY_DIR}/${NAME}")
 endforeach()
 add_custom_target(man ALL DEPENDS ${GZFILES})
-install(FILES ${GZFILES} DESTINATION man/man8)
+install(FILES ${GZFILES} DESTINATION share/man/man8)
