From d940136d0ae2a98ea72d842df0099bebcd361426 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Fri, 2 Nov 2018 12:58:08 +0100
Subject: [PATCH] Fix build without Qt5DBus

---
 CMakeLists.txt     | 2 +-
 src/CMakeLists.txt | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c1de26e3e..eaf794c27 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -304,7 +304,7 @@ endif ()
 if(WIN32)
   set(QT_LIBRARIES Qt5::Core Qt5::OpenGL Qt5::Sql Qt5::Network Qt5::Xml Qt5::Widgets Qt5::Concurrent)
 else(WIN32)
-  set(QT_LIBRARIES Qt5::Core Qt5::OpenGL Qt5::Sql Qt5::Network Qt5::Xml Qt5::Widgets Qt5::Concurrent Qt5::X11Extras Qt5::DBus)
+  set(QT_LIBRARIES Qt5::Core Qt5::OpenGL Qt5::Sql Qt5::Network Qt5::Xml Qt5::Widgets Qt5::Concurrent Qt5::X11Extras)
 endif(WIN32)
 
 # Remove GLU and GL from the link line - they're not really required
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 120f09949..88f1ac6c3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1249,6 +1249,10 @@ target_link_libraries(clementine_lib
   Qocoa
 )
 
+if (HAVE_DBUS)
+  target_link_libraries(clementine_lib Qt5::DBus)
+endif ()
+
 if(ENABLE_VISUALISATIONS)
   target_link_libraries(clementine_lib ${LIBPROJECTM_LIBRARIES})
 endif(ENABLE_VISUALISATIONS)
-- 
2.19.1

