From e0c1bf6570537086628d455cab22c802d663344d Mon Sep 17 00:00:00 2001
From: Laurent Montel <montel@kde.org>
Date: Mon, 15 Jan 2018 13:50:15 +0100
Subject: Fix Bug 388953 - Apply filters on folder doesn't work anymore

FIXED-IN: 5.7.2
BUG: 388953
---
 src/kmmainwidget.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/kmmainwidget.cpp b/src/kmmainwidget.cpp
index d0edc48..ab0996e 100644
--- a/src/kmmainwidget.cpp
+++ b/src/kmmainwidget.cpp
@@ -3310,7 +3310,7 @@ void KMMainWidget::setupActions()
     {
         mApplyAllFiltersFolderAction = new QAction(QIcon::fromTheme(QStringLiteral("view-filter")), i18n("Apply All Filters"), this);
         actionCollection()->addAction(QStringLiteral("apply_filters_folder"), mApplyAllFiltersFolderAction);
-        connect(mServerSideSubscription, &QAction::triggered,
+        connect(mApplyAllFiltersFolderAction, &QAction::triggered,
                 this, [this] { slotApplyFiltersOnFolder(/* recursive */ false);
                 });
     }
@@ -3318,7 +3318,7 @@ void KMMainWidget::setupActions()
     {
         mApplyAllFiltersFolderRecursiveAction = new QAction(QIcon::fromTheme(QStringLiteral("view-filter")), i18n("Apply All Filters"), this);
         actionCollection()->addAction(QStringLiteral("apply_filters_folder_recursive"), mApplyAllFiltersFolderRecursiveAction);
-        connect(mServerSideSubscription, &QAction::triggered,
+        connect(mApplyAllFiltersFolderRecursiveAction, &QAction::triggered,
                 this, [this] { slotApplyFiltersOnFolder(/* recursive */ true);
                 });
     }
-- 
cgit v0.11.2

