From 201eac97eba591e16f0108e2605b9977e2db9a03 Mon Sep 17 00:00:00 2001
From: Allen Winter <winter@kde.org>
Date: Thu, 18 Jul 2019 15:35:24 -0400
Subject: summaryview_part.h - fix build by including QMap

---
 src/kontactplugin/summary/summaryview_part.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kontactplugin/summary/summaryview_part.h b/src/kontactplugin/summary/summaryview_part.h
index 542335d..b116e23 100644
--- a/src/kontactplugin/summary/summaryview_part.h
+++ b/src/kontactplugin/summary/summaryview_part.h
@@ -25,7 +25,7 @@
 #define SUMMARYVIEW_PART_H
 
 #include <KParts/ReadOnlyPart>
-#include <kparts/readonlypart.h>
+#include <QMap>
 
 class DropWidget;
 
-- 
cgit v1.1
