--- util/rarian-sk-preinstall.cpp.old	2008-09-01 20:40:21.000000000 +0200
+++ util/rarian-sk-preinstall.cpp	2010-01-30 11:29:39.000000000 +0100
@@ -96,8 +96,8 @@
     }
     if (i == 1) {
       /* Normal path.  Add file:/ to the start */
-      new_url = (char *) malloc (sizeof(char) * (strlen (input) + 7));
-      sprintf (new_url, "file:/%s", input);
+      new_url = (char *) malloc (sizeof(char) * (strlen (input) + 8));
+      sprintf (new_url, "file://%s", input);
     } else {
       /* Don't know what to do.  Just copy and append file: to it */
       new_url = (char *) malloc (sizeof(char) * (strlen(input) + 6));
