diff --git gerbera/src/file_request_handler.cc src/file_request_handler.cc
index 2fade4a9..de33617b 100644
--- gerbera/src/file_request_handler.cc
+++ gerbera/src/file_request_handler.cc
@@ -282,8 +282,8 @@ void FileRequestHandler::get_info(IN const char* filename, OUT UpnpFileInfo* inf
                     header = header + "CaptionInfo.sec: " + url;
                 }
             }
-#endif
         }
+#endif
     }
 
     if (!string_ok(mimeType))
