diff --git a/libslab/app-shell.c b/libslab/app-shell.c
index 3c33230..5f4af23 100644
--- a/libslab/app-shell.c
+++ b/libslab/app-shell.c
@@ -987,8 +987,8 @@ static void handle_menu_action_performed (Tile * launcher, TileEvent * event, Ti
 		return TRUE;
 	}
 
-	/* discard mate-keyring-manager if CASA is installed */
-	if ((exec && !strcmp (exec, "mate-keyring-manager"))
+	/* discard gnome-keyring-manager if CASA is installed */
+	if ((exec && !strcmp (exec, "gnome-keyring-manager"))
 		&& (path = g_find_program_in_path ("CASAManager.sh")))
 	{
 		g_free (path);
