diff --git a/src/gui/general/ThornStyle.cpp b/src/gui/general/ThornStyle.cpp
index 4374607..de426f1 100644
--- a/src/gui/general/ThornStyle.cpp
+++ b/src/gui/general/ThornStyle.cpp
@@ -34,6 +34,7 @@
 #include <QLabel>
 #include <QLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QRadioButton>
 #include <QStyleFactory>
 #include <QStyleOption>
diff --git a/src/gui/widgets/Panner.cpp b/src/gui/widgets/Panner.cpp
index ceba4ea..ead23fa 100644
--- a/src/gui/widgets/Panner.cpp
+++ b/src/gui/widgets/Panner.cpp
@@ -24,6 +24,7 @@
 #include "misc/Debug.h"
 #include "base/Profiler.h"
 
+#include <QPainterPath>
 #include <QPolygon>
 #include <QMouseEvent>
 
