diff -ur a/Makefile b/Makefile
--- a/Makefile	2021-04-08 10:05:46.064519573 -0500
+++ b/Makefile	2021-04-08 10:51:26.906128397 -0500
@@ -97,6 +97,7 @@
 
 zfuncs.o: zfuncs.cc  zfuncs.h
 	$(CXX) zfuncs.cc $(CFLAGS)                   \
+	-D DOCDIR=\"$(DOCDIR)\"                      \
 
 # install application files in standard locations
 PREFIX ?= /usr
@@ -122,8 +123,7 @@
 	cp -f -R  data/* $(DESTDIR)$(DATADIR)
 	cp -f -R  images/* $(DESTDIR)$(IMAGEDIR)
 	# man page  (21.29) 
-	gzip -fk -9 man/fotoxx.man
-	mv -f man/fotoxx.man.gz $(DESTDIR)$(MANDIR)/fotoxx.1.gz
+	cp man/fotoxx.man $(DESTDIR)$(MANDIR)/fotoxx.1
 	# desktop file and icon
 	cp -f fotoxx.desktop $(DESTDIR)$(MENUDIR)
 	cp -f fotoxx.png $(DESTDIR)$(ICONDIR)
diff -ur a/zfuncs.cc b/zfuncs.cc
--- a/zfuncs.cc	2021-04-08 10:05:46.062519630 -0500
+++ b/zfuncs.cc	2021-04-08 11:09:07.977258807 -0500
@@ -5615,7 +5615,7 @@
 
    strncatv(zdatadir,199,zprefix,"/share/",zappname,"/data",null);               //  /prefix/share/appname/data
    strncatv(zimagedir,199,zprefix,"/share/",zappname,"/images",null);            //  /prefix/share/appname/images
-   strncatv(zdocdir,199,zprefix,"/share/doc/",zappname,null);                    //  /prefix/share/doc/appname
+   strncatv(zdocdir,199,zprefix,"/share/doc/",zappvers,null);                    //  /prefix/share/doc/appname
    
    ch_time = zstrdup(build_date_time,"zinitapp");
    if (ch_time[4] == ' ') ch_time[4] = '0';                                      //  replace month day ' d' with '0d'
