diff -r -U1 fotoxx-14.10.2.orig/Makefile fotoxx-14.10.2/Makefile
--- fotoxx-14.10.2.orig/Makefile	2014-10-09 15:17:09.000000000 +0700
+++ fotoxx-14.10.2/Makefile	2014-10-13 20:44:55.061728475 +0700
@@ -16,3 +16,3 @@
 LOCALESDIR = $(SHAREDIR)/locales
-DOCDIR = $(PREFIX)/share/doc/fotoxx
+DOCDIR = $(PREFIX)/share/doc/$(PF)
 MANDIR = $(PREFIX)/share/man/man1
@@ -90,4 +90,2 @@
 	rm -f fotoxx.1.gz
-	# menu (desktop) file
-	cp -f desktop $(DESTDIR)$(MENUFILE)
 
