diff -ruN driver.org//stk11xx-usb.c driver/stk11xx-usb.c
--- driver.org//stk11xx-usb.c	2011-01-06 21:51:53.674999852 +0200
+++ driver/stk11xx-usb.c	2011-01-06 21:56:44.418999851 +0200
@@ -844,7 +844,12 @@
 	}
 
 	// Init mutexes, spinlock, etc.
+
+#ifndef init_MUTEX
+	sema_init(&dev->mutex,1);
+#else
 	init_MUTEX(&dev->mutex);
+#endif
 	mutex_init(&dev->modlock);
 	spin_lock_init(&dev->spinlock);
 	init_waitqueue_head(&dev->wait_frame);
