respect userflag for optimization
--- a/external_libraries/tcplay/CMakeLists.txt
+++ b/external_libraries/tcplay/CMakeLists.txt
@@ -17,7 +17,6 @@ set (SRCS_LINUX  crypto-gcrypt.c)
 set (CFLAGS_LINUX  "-D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE")
 set (CFLAGS_WARN "-w")
 set (CFLAGS_DEBUG  "-O0 -g3 -DDEBUG")
-set (CFLAGS_OPT    "-O3")
 set (CFLAGS_VER    "-DMAJ_VER=${VERSION_MAJOR} -DMIN_VER=${VERSION_MINOR}")
 set (CFLAGS_COMMON "-std=c99 -fPIC  ${CFLAGS_LINUX} ${CFLAGS_WARN} ${CFLAGS_VER}")
 if ($ENV{DEBUG})
