--- a/src/server/cpulimit-unix.c	2025-04-12 09:08:31.099116998 +0200
+++ b/src/server/cpulimit-unix.c	2025-04-12 09:11:45.097592212 +0200
@@ -42,10 +42,11 @@
   }
 }
 
-void wallclock_timelimit_reached() {
+void wallclock_timelimit_reached(int signum) {
   fprintf(stderr,
           "Why3cpulimit: wallclock timelimit %d reached, killing command\n",
           wallclock_timelimit);
+  (void)signum;
 }
 
 
