--- a/test/CMakeLists.txt	2023-04-12 07:53:37.904871157 +0200
+++ b/test/CMakeLists.txt	2023-04-12 07:53:56.597572954 +0200
@@ -39,7 +39,7 @@
     QuantizationHelpers.cc
     TestUtils.cc)
   set_target_properties(${TESTNAME} PROPERTIES
-          CXX_STANDARD 11
+          CXX_STANDARD 14
           CXX_EXTENSIONS NO)
   #To compile test files with AVX2 turned on
   #For static build, defining FBGEMM_STATIC to avoid generating
