diff -urN a/src/Bfm.java b/src/Bfm.java
--- a/src/Bfm.java	2005-02-22 03:07:39.000000000 +0100
+++ b/src/Bfm.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import java.awt.*;
 import javax.swing.*;
 import com.sun.j3d.utils.universe.*;
diff -urN a/src/Blood.java b/src/Blood.java
--- a/src/Blood.java	2005-01-08 23:45:41.000000000 +0100
+++ b/src/Blood.java	2013-06-10 08:37:16.978094948 +0200
@@ -26,6 +26,8 @@
     A big thank to Andrew for letting us use his code!
 */
 
+package net.webhop.bfm;
+
 import javax.media.j3d.*;
 import javax.vecmath.*;
 import com.sun.j3d.utils.image.*;
diff -urN a/src/DroppedDynamite.java b/src/DroppedDynamite.java
--- a/src/DroppedDynamite.java	2004-12-31 14:23:44.000000000 +0100
+++ b/src/DroppedDynamite.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/Dynamite.java b/src/Dynamite.java
--- a/src/Dynamite.java	2005-02-18 21:24:29.000000000 +0100
+++ b/src/Dynamite.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/Explotion.java b/src/Explotion.java
--- a/src/Explotion.java	2005-02-20 23:55:12.000000000 +0100
+++ b/src/Explotion.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/FSDefault.java b/src/FSDefault.java
--- a/src/FSDefault.java	2005-01-04 16:02:08.000000000 +0100
+++ b/src/FSDefault.java	2013-06-10 08:37:16.978094948 +0200
@@ -20,6 +20,8 @@
 /* FS stands for FileShape but is actually an transformgroup
 	 containing the visuall shape of the FileObjet*/
 
+package net.webhop.bfm;
+
 import java.awt.*;
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
diff -urN a/src/FSDocument.java b/src/FSDocument.java
--- a/src/FSDocument.java	2005-01-04 16:02:08.000000000 +0100
+++ b/src/FSDocument.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import java.awt.*;
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
diff -urN a/src/FSPicture.java b/src/FSPicture.java
--- a/src/FSPicture.java	2005-02-22 02:29:54.000000000 +0100
+++ b/src/FSPicture.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import java.awt.*;
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
diff -urN a/src/FSVideo.java b/src/FSVideo.java
--- a/src/FSVideo.java	2005-01-04 16:02:08.000000000 +0100
+++ b/src/FSVideo.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import java.awt.*;
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
diff -urN a/src/FileObject.java b/src/FileObject.java
--- a/src/FileObject.java	2005-02-20 23:55:12.000000000 +0100
+++ b/src/FileObject.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import java.awt.*;
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
diff -urN a/src/GrapplingHook.java b/src/GrapplingHook.java
--- a/src/GrapplingHook.java	2005-01-03 06:03:50.000000000 +0100
+++ b/src/GrapplingHook.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/Hud.java b/src/Hud.java
--- a/src/Hud.java	2005-02-21 19:14:18.000000000 +0100
+++ b/src/Hud.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import java.awt.*;
 import javax.media.j3d.*;
diff -urN a/src/OpenDialog.java b/src/OpenDialog.java
--- a/src/OpenDialog.java	2005-02-21 18:36:44.000000000 +0100
+++ b/src/OpenDialog.java	2013-06-10 08:37:16.978094948 +0200
@@ -16,6 +16,9 @@
     along with this program; if not, write to the Free Software
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
+
+package net.webhop.bfm;
+
 import javax.swing.plaf.*;
 import java.awt.*;
 import java.awt.event.*;
diff -urN a/src/Pattern.java b/src/Pattern.java
--- a/src/Pattern.java	2004-12-31 14:22:27.000000000 +0100
+++ b/src/Pattern.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import javax.vecmath.*;
 
 public abstract class Pattern {
diff -urN a/src/PatternCircle.java b/src/PatternCircle.java
--- a/src/PatternCircle.java	2004-12-31 14:22:27.000000000 +0100
+++ b/src/PatternCircle.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import javax.vecmath.*;
 
 public class PatternCircle extends Pattern {
diff -urN a/src/PhysicsHandler.java b/src/PhysicsHandler.java
--- a/src/PhysicsHandler.java	2005-02-21 18:36:44.000000000 +0100
+++ b/src/PhysicsHandler.java	2013-06-10 08:37:16.978094948 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import javax.media.j3d.*;
 import javax.vecmath.*;
 import java.io.*;
diff -urN a/src/Player.java b/src/Player.java
--- a/src/Player.java	2005-02-21 18:30:54.000000000 +0100
+++ b/src/Player.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.universe.*; 
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/Rocket.java b/src/Rocket.java
--- a/src/Rocket.java	2004-12-31 14:22:27.000000000 +0100
+++ b/src/Rocket.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/RocketLauncher.java b/src/RocketLauncher.java
--- a/src/RocketLauncher.java	2005-01-08 01:22:40.000000000 +0100
+++ b/src/RocketLauncher.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/Room.java b/src/Room.java
--- a/src/Room.java	2005-02-20 23:55:12.000000000 +0100
+++ b/src/Room.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import javax.media.j3d.*;
 import javax.vecmath.*;
 import java.io.*;
diff -urN a/src/RoomObject.java b/src/RoomObject.java
--- a/src/RoomObject.java	2005-02-20 23:55:12.000000000 +0100
+++ b/src/RoomObject.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import javax.media.j3d.*;
 
 public abstract class RoomObject extends BranchGroup {
diff -urN a/src/Shotgun.java b/src/Shotgun.java
--- a/src/Shotgun.java	2005-01-08 01:22:40.000000000 +0100
+++ b/src/Shotgun.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/SniperRifle.java b/src/SniperRifle.java
--- a/src/SniperRifle.java	2005-02-20 18:16:14.000000000 +0100
+++ b/src/SniperRifle.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/Spear.java b/src/Spear.java
--- a/src/Spear.java	2005-01-03 02:00:41.000000000 +0100
+++ b/src/Spear.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
 import javax.vecmath.*;
diff -urN a/src/Wall.java b/src/Wall.java
--- a/src/Wall.java	2005-01-03 20:40:55.000000000 +0100
+++ b/src/Wall.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import java.awt.*;
 import com.sun.j3d.utils.geometry.*;
 import javax.media.j3d.*;
diff -urN a/src/Weapon.java b/src/Weapon.java
--- a/src/Weapon.java	2005-02-21 18:36:44.000000000 +0100
+++ b/src/Weapon.java	2013-06-10 08:37:16.974761615 +0200
@@ -17,6 +17,8 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+package net.webhop.bfm;
+
 import javax.media.j3d.*;
 import javax.vecmath.*;
 import com.sun.j3d.utils.picking.*;
