--- gtk-engines-2.19.0/themes/Clearlooks/gtk-2.0/gtkrc.orig	2011-11-27 14:25:46.523593240 -0800
+++ gtk-engines-2.19.0/themes/Clearlooks/gtk-2.0/gtkrc	2011-11-27 14:25:34.400658810 -0800
@@ -207,8 +207,8 @@
 }
 
 style "menu" {
-	xthickness = 0
-	ythickness = 0
+	xthickness = 2
+	ythickness = 2
 
 	bg[NORMAL]        = shade (1.08, @bg_color)
 
@@ -229,7 +229,7 @@
 # 2. The line should go to the edges (ie. no border at the left/right)
 style "separator_menu_item" {
 	xthickness = 1
-	ythickness = 0
+	ythickness = 1
 
 	GtkSeparatorMenuItem::horizontal-padding = 0
 	GtkWidget::wide-separators = 1
