diff --git a/src/main/java/org/apache/rat/Report.java b/src/main/java/org/apache/rat/Report.java
index de4a2e7..ecb8ddf 100644
--- a/src/main/java/org/apache/rat/Report.java
+++ b/src/main/java/org/apache/rat/Report.java
@@ -21,7 +21,7 @@ package org.apache.rat;
 import org.apache.commons.cli.*;
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.filefilter.*;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.rat.api.RatException;
 import org.apache.rat.report.IReportable;
 import org.apache.rat.report.RatReport;
diff --git a/src/main/java/org/apache/rat/document/impl/guesser/GuessUtils.java b/src/main/java/org/apache/rat/document/impl/guesser/GuessUtils.java
index d56a672..d6b7f4d 100644
--- a/src/main/java/org/apache/rat/document/impl/guesser/GuessUtils.java
+++ b/src/main/java/org/apache/rat/document/impl/guesser/GuessUtils.java
@@ -20,7 +20,7 @@ package org.apache.rat.document.impl.guesser;
 
 import java.util.Locale;
 
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
 
 public class GuessUtils {
     
diff --git a/src/main/java/org/apache/rat/report/claim/impl/xml/SimpleXmlClaimReporter.java b/src/main/java/org/apache/rat/report/claim/impl/xml/SimpleXmlClaimReporter.java
index 02e21d6..5001dff 100644
--- a/src/main/java/org/apache/rat/report/claim/impl/xml/SimpleXmlClaimReporter.java
+++ b/src/main/java/org/apache/rat/report/claim/impl/xml/SimpleXmlClaimReporter.java
@@ -18,7 +18,7 @@
  */ 
 package org.apache.rat.report.claim.impl.xml;
 
-import org.apache.commons.lang.time.DateFormatUtils;
+import org.apache.commons.lang3.time.DateFormatUtils;
 import org.apache.rat.api.Document;
 import org.apache.rat.api.MetaData;
 import org.apache.rat.api.RatException;
