Description: D4FilterClauseTest breaks on arm*, powerpc, s390x.
Author: Alastair McKinstry <mckinstry@debian.org>
Last-Updated: 2016-08-31
Forwarded: no

Index: libdap-3.18.0/unit-tests/Makefile.am
===================================================================
--- libdap-3.18.0.orig/unit-tests/Makefile.am
+++ libdap-3.18.0/unit-tests/Makefile.am
@@ -74,7 +74,7 @@ UNIT_TESTS = marshT arrayT attrTableT st
 if DAP4_DEFINED
 UNIT_TESTS += D4MarshallerTest D4UnMarshallerTest D4DimensionsTest \
 	D4EnumDefsTest D4GroupTest D4ParserSax2Test D4AttributesTest D4EnumTest \
-	chunked_iostream_test D4AsyncDocTest DMRTest D4FilterClauseTest \
+	chunked_iostream_test D4AsyncDocTest DMRTest  \
 	D4SequenceTest
 endif
 
