diff -ru lua-5.1.1.orig/src/Makefile lua-5.1.1/src/Makefile
--- lua-5.1.1.orig/src/Makefile	2006-11-21 07:19:31 +0000
+++ lua-5.1.1/src/Makefile	2006-11-21 07:19:52 +0000
@@ -196,7 +196,7 @@
             -rpath $(RPATH) $(LDFLAGS) -o $(LIB_NAME) $(LIB_OBJS) $(LIB_LIBS)
 
 $(LUA_T): $(LUA_O:.o=.lo) $(LIB_NAME)
-	$(LIBTOOL) --mode=link $(CC) -export-dynamic $(LDFLAGS) -o $@ $(LUA_O:.o=.lo) $(LIB_NAME) $(LUA_LIBS)
+	$(LIBTOOL) --mode=link $(CC) -static -export-dynamic $(LDFLAGS) -o $@ $(LUA_O:.o=.lo) $(LIB_NAME) $(LUA_LIBS)
 
 $(LUAC_T): $(LUAC_O:.o=.lo) $(LIB_NAME)
 	$(LIBTOOL) --mode=link $(CC) -static -o $@ $(LUAC_O:.o=.lo) $(LIB_NAME)
