diff -urN eqe-1.3.0.old/Makefile eqe-1.3.0/Makefile
--- eqe-1.3.0.old/Makefile	2006-02-04 15:54:54.000000000 +0100
+++ eqe-1.3.0/Makefile	2006-02-14 10:52:04.000000000 +0100
@@ -1,7 +1,7 @@
 # Ronan Le Hy, 2005
 # eqe installer
 
-PREFIX=/usr/local
+PREFIX=/usr
 DESTDIR=
 PACKAGE_NAME=eqe
 VERSION=1.3.0
@@ -21,19 +21,17 @@
 	pod2man doc/eqedit.pod $(DESTDIR)/$(PREFIX)/share/man/man1/eqedit.1	
 	gzip -f --best $(DESTDIR)/$(PREFIX)/share/man/man1/eqe.1
 	gzip -f --best $(DESTDIR)/$(PREFIX)/share/man/man1/eqedit.1
-	mkdir -p $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/
-	cp changelog $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/changelog
-	gzip -f --best $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/changelog
-	cp README $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/README
-	gzip -f --best $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/README
-	cp FAQ $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/FAQ
-	gzip -f --best $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/FAQ
-	cp TODO $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/TODO
-	cp copyright $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/copyright
-	cp GPL.txt $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/GPL.txt
+	mkdir -p $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/
+	cp changelog $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/changelog
+	gzip -f --best $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/changelog
+	cp README $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/README
+	gzip -f --best $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/README
+	cp FAQ $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/FAQ
+	gzip -f --best $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/FAQ
+	cp TODO $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/TODO
+	gzip -f --best $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/TODO
 	chmod 644 $(DESTDIR)/$(PREFIX)/share/man/man1/eqe.1.gz $(DESTDIR)/$(PREFIX)/share/man/man1/eqedit.1.gz
-	chmod 644 $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/README.gz $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/changelog.gz $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/TODO $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/copyright
-	chmod 644 $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)/GPL.txt
+	chmod 644 $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/README.gz $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/changelog.gz $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/TODO.gz $(DESTDIR)/$(PREFIX)/share/doc/$(PACKAGE_NAME)-$(VERSION)/FAQ.gz
 
 install-templates:
 	mkdir -p $(DESTDIR)/$(PREFIX)/share/eqe/
