--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -3,7 +3,7 @@ LIBWPS_DIR=libwps
 INC=-I$(LIBWPS_DIR)
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-CONFDIR=@sysconfdir@/@target@
+CONFDIR=/var/db/reaver
 CFLAGS=-DCONF_DIR='"$(CONFDIR)"' -DREAVER_DATABASE='"$(CONFDIR)/reaver.db"' @CFLAGS@
 LDFLAGS=$(LIBWPS_DIR)/*.o wps/*.o tls/bignum.o tls/libtls.a utils/libutils.a crypto/libcrypto.a lwe/libiw.a @LDFLAGS@
 
