commit 11d35e62fe13b47c5b34139559e650dfe6ebbd84
Author: hasufell <julian.ospald@googlemail.com>
Date:   Sat Jun 8 13:40:30 2013 +0200

    CMAKE: use shared lib from Irrlicht

diff --git a/cmake/Modules/FindIrrlicht.cmake b/cmake/Modules/FindIrrlicht.cmake
index bd00422..4632993 100644
--- a/cmake/Modules/FindIrrlicht.cmake
+++ b/cmake/Modules/FindIrrlicht.cmake
@@ -50,10 +50,11 @@ else()
 		/usr/include/irrlicht
 	)
 
-	FIND_LIBRARY(IRRLICHT_LIBRARY NAMES libIrrlicht.a Irrlicht
+	FIND_LIBRARY(IRRLICHT_LIBRARY NAMES libIrrlicht.so Irrlicht
 		PATHS
 		/usr/local/lib
 		/usr/lib
+		/usr/lib64
 	)
 endif()
 
