Index: cpuid-20110305/Makefile
===================================================================
--- cpuid-20110305.orig/Makefile
+++ cpuid-20110305/Makefile
@@ -1,7 +1,3 @@
-CFLAGS=-g
-ifneq (,$(findstring arch=i386,$(CFLAGS)))
-CISA=-m32
-endif
 CFL=$(CFLAGS) $(CISA) -Wall -Wshadow -Wcast-align -Wredundant-decls -Wbad-function-cast -Wcast-qual -Wwrite-strings -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -D_FILE_OFFSET_BITS=64 -DVERSION=$(VERSION)
 
 PACKAGE=cpuid
@@ -36,14 +32,14 @@ BUILDROOT=
 default: $(PROG) $(PROG).man.gz
 
 $(PROG): cpuid.c Makefile
-	$(CC) $(CFL) -o $@ cpuid.c
+	$(CC) $(CFL) $(LDFLAGS) -o $@ cpuid.c
 
 $(PROG).man.gz: $(PROG).man
 	gzip < $< > $@
 
 install: $(PROG) $(PROG).man.gz
-	install -D -s -m 755 $(PROG)        $(BUILDROOT)/usr/bin/$(PROG)
-	install -D    -m 444 $(PROG).man.gz $(BUILDROOT)/usr/share/man/man1/$(PROG).1.gz
+	install -D -m 755 $(PROG)        $(BUILDROOT)/usr/bin/$(PROG)
+	install -D -m 444 $(PROG).man.gz $(BUILDROOT)/usr/share/man/man1/$(PROG).1.gz
 
 clean:
 	rm -f $(PROG) $(PROG).i386 $(PROG).x86_64
