From 746f5f6f0c29867ac97516d00fdf58d3496a6687 Mon Sep 17 00:00:00 2001
From: Chris Rowett <rowett@yahoo.com>
Date: Wed, 11 Jul 2018 12:44:22 +0100
Subject: [PATCH] bugfix: clear mousedown flag in mousecapturelost event
 handler

---
 gui-wx/wxview.cpp | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

--- a/gui-wx/wxview.cpp
+++ b/gui-wx/wxview.cpp
@@ -81,9 +81,7 @@ EVT_MIDDLE_DCLICK    (           PatternView::OnMouseDown)
 EVT_LEFT_UP          (           PatternView::OnMouseUp)
 EVT_RIGHT_UP         (           PatternView::OnMouseUp)
 EVT_MIDDLE_UP        (           PatternView::OnMouseUp)
-#if wxCHECK_VERSION(2, 8, 0)
 EVT_MOUSE_CAPTURE_LOST (         PatternView::OnMouseCaptureLost)
-#endif
 EVT_MOTION           (           PatternView::OnMouseMotion)
 EVT_ENTER_WINDOW     (           PatternView::OnMouseEnter)
 EVT_LEAVE_WINDOW     (           PatternView::OnMouseExit)
@@ -2901,18 +2899,15 @@ void PatternView::OnMouseUp(wxMouseEvent& event)
 
 // -----------------------------------------------------------------------------
 
-#if wxCHECK_VERSION(2, 8, 0)
-
 // mouse capture can be lost on Windows before mouse-up event
 void PatternView::OnMouseCaptureLost(wxMouseCaptureLostEvent& WXUNUSED(event))
 {
+    mouseisdown = false;
     if (drawingcells || selectingcells || movingview || clickedcontrol > NO_CONTROL) {
         StopDraggingMouse();
     }
 }
 
-#endif
-
 // -----------------------------------------------------------------------------
 
 void PatternView::OnMouseMotion(wxMouseEvent& event)
-- 
2.19.2

