--- src/wiliki.cgi.orig	2008-01-11 08:52:19.000000000 -0500
+++ src/wiliki.cgi	2009-01-30 11:54:41.000000000 -0500
@@ -52,13 +52,13 @@
 (define (main args)
   (wiliki-main
    (make <wiliki>
-     :db-path "/home/shiro/data/wikidata.dbm"
-     :top-page "WiLiKi"
-     :title "MyWiliki"
-     :description "Shiro's Wiliki Site"
-     :style-sheet "wiliki.css"
-     :language 'jp
-     :charsets '((jp . euc-jp) (en . euc-jp))
+     :db-path "/var/www/localhost/wiliki-0.6.1/data/wikidata.dbm"
+     :top-page "Gentoo WiLiKi"
+     :title "Gentoo Wiliki"
+     :description "Gentoo Wiliki Site"
+     :style-sheet "/wiliki/wiliki.css"
+     :language 'en
+     :charsets '((en . utf-8) (jp . euc-jp) (en . euc-jp))
      :image-urls '((#/^http:\/\/sourceforge.net\/sflogo/ allow))
      :debug-level 0
      )))
