commit d3a2c46bf255c428220342c05a46f9191d94077c
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Wed Mar 16 21:03:12 2016 +0100

    Remove unused deps

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3c341ef..08ed990 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,26 +6,20 @@ set (QT_MIN_VERSION "5.3.0")
 find_package(ECM 1.7.0 REQUIRED CONFIG)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 
-find_package(Qt5 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Widgets Qml Quick QuickWidgets Svg Test)
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Widgets Qml Quick Svg)
 find_package(KF5 REQUIRED COMPONENTS 
     CoreAddons
     Config
-    ItemModels
     WidgetsAddons
-    WindowSystem
     Config
     DBusAddons
     I18n
     ConfigWidgets
-    TextWidgets
     XmlGui
-    KIO
-    NotifyConfig
-    NewStuff
-    KDELibs4Support)
+    KDELibs4Support
+)
 
 find_package(KF5KDEGames 4.9.0 REQUIRED)
-find_package(Phonon4Qt5 CONFIG REQUIRED)
 
 include(FeatureSummary)
 include(ECMAddAppIcon)
@@ -60,7 +54,12 @@ file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/*-apps-kjumpingcube.png")
 ecm_add_app_icon(kjumpingcube_SRCS ICONS ${ICONS_SRCS})
 add_executable(kjumpingcube ${kjumpingcube_SRCS})
 
-target_link_libraries(kjumpingcube  KF5::KIOCore KF5KDEGames KF5::XmlGui KF5::KDELibs4Support Qt5::Svg)
+target_link_libraries(kjumpingcube
+    KF5KDEGames
+    KF5::XmlGui
+    KF5::KDELibs4Support
+    Qt5::Svg
+)
 
 install(TARGETS kjumpingcube  ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
 
