--- Makefile.bak	2008-07-08 21:34:10.000000000 +0200
+++ Makefile	2008-07-08 21:35:34.000000000 +0200
@@ -20,13 +20,13 @@
 #
 
 PROGNAME = ratproxy
-CFLAGS	 = -Wall -O3 -Wno-pointer-sign -D_GNU_SOURCE
-LDFLAGS  = -lcrypto -lssl
+CFLAGS	+= -Wall -Wno-pointer-sign -D_GNU_SOURCE
+LIBS	= -lcrypto -lssl
 
 all: $(PROGNAME) flare-check
 
 $(PROGNAME): $(PROGNAME).c http.c mime.c ssl.c http.h mime.h ssl.h nlist.h config.h debug.h types.h string-inl.h
-	$(CC) $(PROGNAME).c -o $(PROGNAME)  $(CFLAGS) http.c mime.c ssl.c $(LDFLAGS)
+	$(CC) $(PROGNAME).c -o $(PROGNAME)  $(CFLAGS) $(LDFLAGS) http.c mime.c ssl.c $(LIBS)
 
 flare-check:
 	@flare-dist/flare 2>&1 | grep -qF Igor || ( \
