--- gkrellweather-2.0.8/GrabWeather	2009-01-11 13:15:02.000000000 -0500
+++ gkrellweather-2.0.8-r1/GrabWeather	2016-08-29 00:02:35.345041005 -0400
@@ -12,7 +12,8 @@
 
 $ReportDir  = '.wmWeatherReports';
 #$WeatherSrc = 'ftp://weather.noaa.gov/data/observations/metar/decoded';
-$WeatherSrc = 'http://weather.noaa.gov/pub/data/observations/metar/decoded';
+#$WeatherSrc = 'http://weather.noaa.gov/pub/data/observations/metar/decoded';
+$WeatherSrc = 'http://tgftp.nws.noaa.gov/data/observations/metar/decoded/';
 
 use strict; 
 use vars qw( $ReportDir $WeatherSrc );
