--- spatialindex-src-1.8.0/configure.ac
+++ spatialindex-src-1.8.0/configure.ac
@@ -49,10 +49,10 @@
 AC_ARG_ENABLE(debug, [  --enable-debug=[no/yes] turn on debugging [default=$debug_default]],, enable_debug=$debug_default)
 
 if test "x$enable_debug" = "xyes"; then
-	CXXFLAGS="$CXXFLAGS -g -DDEBUG"
+	CXXFLAGS="$CXXFLAGS -DDEBUG"
 	AC_MSG_RESULT(checking wether debug information is enabled... yes)
 else
-	CXXFLAGS="$CXXFLAGS -O2 -DNDEBUG"
+	CXXFLAGS="$CXXFLAGS -DNDEBUG"
 	AC_MSG_RESULT(checking wether debug information is enabled... no)
 fi
 
