--- a/FDDL/mdd.h
+++ b/FDDL/mdd.h
@@ -29,10 +29,11 @@
 //#define BRIEF_DEBUG
 
 #ifndef FDDL_MDD_H
 #   define FDDL_MDD_H 1
 
+#   include <cstdio>
 #   include <iostream>
 #   include <assert.h>
 
 #   include <FDDL/mddtypes.h>
 #   include <FDDL/caches.h>
--- a/src/structures.h
+++ b/src/structures.h
@@ -28,10 +28,11 @@
 
 #ifndef __STRUCTURES_H
 #define __STRUCTURES_H
 
 #include <stdlib.h>
+#include <cstdio>
 
 //Linked list of IP addresses
 class address
 {
 public:
