--- scripts/Makefile.nml.old
+++ scripts/Makefile.nml
@@ -14,11 +14,11 @@
 
 %.src.dep: $(SRC_DIR)/%.pnml
 	$(_E) "[DEP] $@"
-	$(_V) $(CC) -C -M -MF $@ -E -MT $(patsubst %.src.dep,%.nml,$@) - < $<
+	$(_V) $(CC) -P -C -M -MF $@ -E -MT $(patsubst %.src.dep,%.nml,$@) - < $<
 
 %.nml: $(SRC_DIR)/%.pnml $(TAG_FILENAME)
 	$(_E) "[CPP] $@"
-	$(_V) $(CC) -D REPO_REVISION=$(REPO_REVISION) $(CC_FLAGS) $< > $@
+	$(_V) $(CC) -P -D REPO_REVISION=$(REPO_REVISION) $(CC_FLAGS) $< > $@
 
 %.gfx.dep: %.nml
 	$(_E) "[DEP] $@"
