Index: pycapnp-0.6.4/setup.py
===================================================================
--- pycapnp-0.6.4.orig/setup.py
+++ pycapnp-0.6.4/setup.py
@@ -49,7 +49,7 @@ try:
     changelog = '\nChangelog\n=============\n' + changelog
     long_description += changelog
 except (IOError, ImportError):
-    if sys.argv[2] == 'sdist':
+    if len(sys.argv) > 2 and sys.argv[2] == 'sdist':
         raise
     long_description = ''
 
