--- a/src/misc.c
+++ b/src/misc.c
@@ -33,7 +33,7 @@
 
 char *search_vendor(unsigned char mac[6])
 {
-	char tmac[6];
+	char tmac[7];
 	int i = 0;
 	
 	sprintf(tmac, "%02x%02x%02x", mac[0], mac[1], mac[2]);
@@ -42,7 +42,7 @@
     for (i=0; i<6; i++)
         tmac[i] = toupper(tmac[i]);
 	
-	for (i=0; i<8436; i++)
+	for (i=0; i<NUM_OUI_RECORDS; i++)
 	{
 		if (strcmp(oui_table[i].prefix, tmac) == 0)
 			return oui_table[i].vendor;
