diff --git a/com/jclark/xml/sax/Driver.java b/com/jclark/xml/sax/Driver.java
index 2b0d90a..d1efd3e 100644
--- a/com/jclark/xml/sax/Driver.java
+++ b/com/jclark/xml/sax/Driver.java
@@ -160,9 +160,9 @@ public class Driver extends ApplicationImpl
 
     DTD dtd = event.getDTD();
 
-    for (Enumeration enum = dtd.entityNames(DTD.NOTATION);
-	 enum.hasMoreElements(); ) {
-      String name = (String)enum.nextElement();
+    for (Enumeration enumx = dtd.entityNames(DTD.NOTATION);
+	 enumx.hasMoreElements(); ) {
+      String name = (String)enumx.nextElement();
       Entity entity = dtd.getEntity(DTD.NOTATION, name);
       String systemId = entity.getSystemId();
       if (systemId != null) {
@@ -173,9 +173,9 @@ public class Driver extends ApplicationImpl
       }
       dtdHandler.notationDecl(name, entity.getPublicId(), systemId);
     }
-    for (Enumeration enum = dtd.entityNames(DTD.GENERAL_ENTITY);
-	 enum.hasMoreElements();) {
-      String name = (String)enum.nextElement();
+    for (Enumeration enumx = dtd.entityNames(DTD.GENERAL_ENTITY);
+	 enumx.hasMoreElements();) {
+      String name = (String)enumx.nextElement();
       Entity entity = dtd.getEntity(DTD.GENERAL_ENTITY, name);
       String notationName = entity.getNotationName();
       if (notationName != null) {
