--- a/CMakeLists.txt	2017-07-26 00:54:48.021345174 -0300
+++ b/CMakeLists.txt	2017-07-26 00:54:39.510156990 -0300
@@ -140,7 +140,7 @@
     if (PORTABLE)
         install(TARGETS "libopenrct2" LIBRARY DESTINATION "bin")
     else ()
-        install(TARGETS "libopenrct2" LIBRARY DESTINATION "lib")
+        install(TARGETS "libopenrct2" LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
     endif ()
 endif ()
 install(TARGETS "openrct2" RUNTIME DESTINATION "bin")
