--- a/src/openrct2/config/Config.cpp
+++ b/src/openrct2/config/Config.cpp
@@ -638,6 +638,7 @@
         log_verbose("config_find_rct1_path(...)");
 
         static constexpr u8string_view searchLocations[] = {
+            R"(/usr/share/openrct2/rct1_data)",
             R"(C:\Program Files\Steam\steamapps\common\Rollercoaster Tycoon Deluxe)",
             R"(C:\Program Files (x86)\Steam\steamapps\common\Rollercoaster Tycoon Deluxe)",
             R"(C:\GOG Games\RollerCoaster Tycoon Deluxe)",
@@ -683,6 +684,7 @@
         log_verbose("config_find_rct2_path(...)");
 
         static constexpr u8string_view searchLocations[] = {
+            R"(/usr/share/openrct2/rct2_data)",
             R"(C:\Program Files\Steam\steamapps\common\Rollercoaster Tycoon 2)",
             R"(C:\Program Files (x86)\Steam\steamapps\common\Rollercoaster Tycoon 2)",
             R"(C:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack)",
