From d4943c80cded6325284be8b58003fe63dd6e8255 Mon Sep 17 00:00:00 2001
From: Raimar <no@email.de>
Date: Thu, 21 Aug 2014 23:21:28 +0200
Subject: [PATCH 1/1] fix crash after click on empty part of panel

---
 src/panel.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/panel.c b/src/panel.c
index 6b2ede6..87903d8 100644
--- a/src/panel.c
+++ b/src/panel.c
@@ -233,7 +233,7 @@ static gboolean lxpanel_button_press(GtkWidget *widget, GdkEventButton *event)
         gtk_menu_popup(popup, NULL, NULL, NULL, NULL, event->button, event->time);
         return TRUE;
     }
-    return GTK_WIDGET_CLASS(lxpanel_parent_class)->button_press_event(widget, event);
+    return FALSE;
 }
 
 static void lxpanel_class_init(PanelToplevelClass *klass)
-- 
1.7.10.4

