Index: work/src/qtsingleapplication/src/qtlocalpeer.cpp
===================================================================
--- work.orig/src/qtsingleapplication/src/qtlocalpeer.cpp
+++ work/src/qtsingleapplication/src/qtlocalpeer.cpp
@@ -100,7 +100,11 @@ QtLocalPeer::QtLocalPeer(QObject* parent
         socketName += QLatin1Char('-') + QString::number(sessionId, 16);
     }
 #else
+#if __GNUC__ >= 4 && __GNUC_MINOR__ <=6
     socketName += QLatin1Char('-') + QString::number(::getuid(), 16);
+#else
+    socketName += QLatin1Char('-') + QString::number(QtLP_Private::getuid(), 16);
+#endif
 #endif
 
     server = new QLocalServer(this);
