--- a/cmake/compiler_settings.cmake	2020-12-12 15:31:17.703362345 +0100
+++ b/cmake/compiler_settings.cmake	2020-12-12 15:31:53.641363625 +0100
@@ -53,7 +53,7 @@
 endmacro()
 
 macro(set_gcc_warnings)
-    list(APPEND AWS_COMPILER_WARNINGS "-Wall" "-Werror" "-pedantic" "-Wextra")
+    list(APPEND AWS_COMPILER_WARNINGS "-Wall" "-pedantic" "-Wextra")
     if(COMPILER_CLANG)
         if(PLATFORM_ANDROID)
             # when using clang with libc and API lower than 21 we need to include Android support headers and ignore the gnu-include-next warning.
--- a/cspell.json	2020-12-12 15:31:17.703362345 +0100
+++ b/cspell.json	2020-12-12 15:31:42.488363228 +0100
@@ -13,7 +13,7 @@
         // CMake
         "cmake", "DCMAKE", "AUTORUN", "DEPS", "STREQUAL", "RPATH", "STREQUAL", "DTARGET", "TOOLCHAIN", "DANDROID", "DBUILD", "DGIT", "SYSROOT", "DSTATIC", "BINDIR", "LIBDIR", "INCLUDEDIR", "DHAS",
         // Compiler and linker
-        "Wpedantic", "Wextra", "Werror", "xldscope", "Wtype", "Wunused", "RTTI", "ffunction", "fdata", "fsanitize", "pathconf", "unistd", "umask",
+        "Wpedantic", "Wextra", "xldscope", "Wtype", "Wunused", "RTTI", "ffunction", "fdata", "fsanitize", "pathconf", "unistd", "umask",
         // Android NDK
         "JNIEXPORT", "jint", "JNICALL", "jobject", "jclass", "jmethod", "Ljava",
         // XML parser
--- a/CI/install-test/CMakeLists.txt	2020-12-12 15:31:17.704362345 +0100
+++ b/CI/install-test/CMakeLists.txt	2020-12-12 15:32:06.433364080 +0100
@@ -4,4 +4,4 @@
 find_package(AWSSDK REQUIRED COMPONENTS s3)
 add_executable(${PROJECT_NAME} "main.cpp")
 target_link_libraries(${PROJECT_NAME} ${AWSSDK_LINK_LIBRARIES})
-target_compile_options(${PROJECT_NAME} PRIVATE "-Wall" "-Werror")
+target_compile_options(${PROJECT_NAME} PRIVATE "-Wall")
