--- a/scripts/build_example.py	(original)
+++ b/scripts/build_example.py	(refactored)
@@ -56,13 +56,13 @@
     cmake_call = cmake_call + " -DCMAKE_PREFIX_PATH=" + sdkInstallDir
     cmake_call = cmake_call + " -DAWSSDK_ROOT_DIR=" + sdkInstallDir
     cmake_call = cmake_call + " " + sourceDir
-    print("cmake call = " + cmake_call)
+    print(("cmake call = " + cmake_call))
     subprocess.check_call(cmake_call, shell = True)
     
     build_call = buildEnvironment[ platform ][ 'global_build_call' ]
     build_call = build_call + buildEnvironment[ platform ][ 'configs' ].replace("??", buildType)
     build_call = build_call + buildEnvironment[ platform ][ 'parallel_option' ].replace("??", parallel)
-    print("build call = " + build_call)
+    print(("build call = " + build_call))
     subprocess.check_call(build_call, shell = True)
 
 def Main():
