--- a/configure.ac
+++ b/configure.ac
@@ -84,14 +84,14 @@
 #
 
 filechk="yes"
-AC_CHECK_FILE(/usr/lib/libnet.a,, filechk="no")
+AC_CHECK_FILE(/usr/lib/libnet.so,, filechk="no")
 if test "$filechk" = "no"; then
- AC_CHECK_FILE(/usr/local/lib/libnet.a, filechk="yes" ; LDFLAGS="-L/usr/local/lib"; CPPFLAGS="-I/usr/local/include")
+ AC_CHECK_FILE(/usr/local/lib/libnet.so, filechk="yes" ; LDFLAGS="-L/usr/local/lib"; CPPFLAGS="-I/usr/local/include")
 fi
 
 if test "$filechk" = "no"; then AC_MSG_ERROR([
 
-Libnet archive file (libnet.a) not found in /usr or /usr/local!
+Libnet archive file (libnet.so) not found in /usr or /usr/local!
 
 Libnet-1.1.x Packet Shaping Library is required.
 You can download it from official web site: http://www.packetfactory.net/libnet
@@ -111,14 +111,14 @@
 #
 
 filechk="yes"
-AC_CHECK_FILE(/usr/lib/libpcap.a,, filechk="no")
+AC_CHECK_FILE(/usr/lib/libpcap.so,, filechk="no")
 if test "$filechk" = "no"; then
- AC_CHECK_FILE(/usr/local/lib/libpcap.a, filechk="yes" ; LDFLAGS="-L/usr/local/lib"; CPPFLAGS="-I/usr/local/include")
+ AC_CHECK_FILE(/usr/local/lib/libpcap.so, filechk="yes" ; LDFLAGS="-L/usr/local/lib"; CPPFLAGS="-I/usr/local/include")
 fi
 
 if test "$filechk" = "no"; then AC_MSG_ERROR([
 
-Libpcap archive file (libpcap.a) not found in /usr or /usr/local!
+Libpcap archive file (libpcap.so) not found in /usr or /usr/local!
 
 Libpcap-0.7.1 Packet Capture Library is required. 
 You can download it from official web site: http://www.tcpdump.org/#current
@@ -137,16 +137,18 @@
 # -- libncurses support --
 #
 
+PKG_CHECK_MODULES(ncurses,ncurses,[LIBS="$LIBS $ncurses_LIBS"],)
+
 filechk="yes"
-AC_CHECK_FILE(/usr/lib/libncurses.a,, filechk="no")
+AC_CHECK_FILE(/usr/lib/libncurses.so,, filechk="no")
 if test "$filechk" = "no"; then
- AC_CHECK_FILE(/usr/local/lib/libncurses.a, filechk="yes" ; LDFLAGS="-L/usr/local/lib"; CPPFLAGS="-I/usr/local/include")
+ AC_CHECK_FILE(/usr/local/lib/libncurses.so, filechk="yes" ; LDFLAGS="-L/usr/local/lib"; CPPFLAGS="-I/usr/local/include")
 fi
 
 if test "$filechk" = "no"; 
 then AC_MSG_WARN([
 
-Libncurses archive file (libncurses.a) not found in /usr or /usr/local!
+Libncurses archive file (libncurses.so) not found in /usr or /usr/local!
 
 Ncurses library (*) not found in your system.
 You can download it from official web site: http://www.gnu.org/software/ncurses/ncurses.html
@@ -174,15 +176,15 @@
 #
 
 filechk="yes"
-AC_CHECK_FILE(/usr/lib/libmenu.a,, filechk="no")
+AC_CHECK_FILE(/usr/lib/libmenu.so,, filechk="no")
 if test "$filechk" = "no"; then
- AC_CHECK_FILE(/usr/local/lib/libmenu.a, filechk="yes" ; LDFLAGS="-L/usr/local/lib"; CPPFLAGS="-I/usr/local/include")
+ AC_CHECK_FILE(/usr/local/lib/libmenu.so, filechk="yes" ; LDFLAGS="-L/usr/local/lib"; CPPFLAGS="-I/usr/local/include")
 fi
 
 if test "$filechk" = "no"; 
 then AC_MSG_WARN([
 
-Menu archive file (libncurses.a) not found in /usr or /usr/local!
+Menu archive file (libncurses.so) not found in /usr or /usr/local!
 
 
 Menu library (*) not found in your system.
