diff --git a/samples/Makefile.am b/samples/Makefile.am
index 4558d66..6fc6d77 100644
--- a/samples/Makefile.am
+++ b/samples/Makefile.am
@@ -16,9 +16,9 @@ samplessysconf_DATA = \
 	swtpm-localca.options
 
 install-data-local:
-	$(MKDIR_P) $(DESTDIR)$(localstatedir)/lib/swtpm-localca
+	$(MKDIR_P) $(DESTDIR)$(localstatedir)/swtpm-localca
 	if test -z $(DESTDIR); then \
-		chown @TSS_USER@:@TSS_GROUP@ $(DESTDIR)$(localstatedir)/lib/swtpm-localca || true; \
+		chown @TSS_USER@:@TSS_GROUP@ $(DESTDIR)$(localstatedir)/swtpm-localca || true; \
 	fi
 
 EXTRA_DIST= \
diff --git a/samples/swtpm-localca.conf.in b/samples/swtpm-localca.conf.in
index 1f0b48d..63f1fed 100644
--- a/samples/swtpm-localca.conf.in
+++ b/samples/swtpm-localca.conf.in
@@ -1,4 +1,4 @@
-statedir = @LOCALSTATEDIR@/lib/swtpm-localca
-signingkey = @LOCALSTATEDIR@/lib/swtpm-localca/signkey.pem
-issuercert = @LOCALSTATEDIR@/lib/swtpm-localca/issuercert.pem
-certserial = @LOCALSTATEDIR@/lib/swtpm-localca/certserial
+statedir = @LOCALSTATEDIR@/swtpm-localca
+signingkey = @LOCALSTATEDIR@/swtpm-localca/signkey.pem
+issuercert = @LOCALSTATEDIR@/swtpm-localca/issuercert.pem
+certserial = @LOCALSTATEDIR@/swtpm-localca/certserial
