diff -aurN a/autoconf/pecl/libcurl.m4 b/autoconf/pecl/libcurl.m4
--- a/autoconf/pecl/libcurl.m4	2016-12-12 04:04:21.000000000 -0500
+++ b/autoconf/pecl/libcurl.m4	2017-01-06 09:07:17.575660038 -0500
@@ -87,7 +87,7 @@
 	AC_REQUIRE([PECL_HAVE_LIBCURL_CA])dnl
 	PECL_HAVE_LIBCURL_FEATURE([SSL], [
 		PECL_HAVE_LIBCURL_SSLLIB([OpenSSL], [openssl/ssl.h openssl/crypto.h], [ssl crypto])
-		PECL_HAVE_LIBCURL_SSLLIB([GnuTLS], [gnutls.h gcrypt.h], [gnutls gcrypt])
+		PECL_HAVE_LIBCURL_SSLLIB([GnuTLS], [gnutls/gnutls.h gcrypt.h], [gnutls gcrypt])
 		PECL_HAVE_LIBCURL_SSLLIB([NSS])
 		PECL_HAVE_LIBCURL_SSLLIB([SecureTransport])
 		PECL_HAVE_LIBCURL_SSLLIB([GSKit])
diff -aurN a/src/php_http_client_curl.c b/src/php_http_client_curl.c
--- a/src/php_http_client_curl.c	2016-12-12 04:04:21.000000000 -0500
+++ b/src/php_http_client_curl.c	2017-01-06 09:09:01.631772852 -0500
@@ -21,7 +21,7 @@
 #	include <openssl/ssl.h>
 #endif
 #if PHP_HTTP_HAVE_LIBCURL_GNUTLS
-#	include <gnutls.h>
+#	include <gnutls/gnutls.h>
 #endif
 
 typedef struct php_http_client_curl_handler {
