diff --git a/openpam-20130907/lib/libpam/openpam_constants.c b/openpam-20130907/lib/libpam/openpam_constants.c
index 1cdd810..f66e469 100644
--- a/openpam-20130907/lib/libpam/openpam_constants.c
+++ b/openpam-20130907/lib/libpam/openpam_constants.c
@@ -135,8 +135,8 @@ const char *openpam_policy_path[] = {
 };
 
 const char *openpam_module_path[] = {
-#ifdef OPENPAM_MODULES_DIRECTORY
-	OPENPAM_MODULES_DIRECTORY,
+#ifdef OPENPAM_MODULES_DIR
+	OPENPAM_MODULES_DIR,
 #else
 	"/usr/lib",
 	"/usr/local/lib",
