diff -NrU5 subtitleripper.orig/spudec.c subtitleripper/spudec.c
--- subtitleripper.orig/spudec.c	2009-05-20 03:06:23.000000000 -0600
+++ subtitleripper/spudec.c	2009-05-20 03:07:47.000000000 -0600
@@ -34,17 +34,18 @@
 
 #ifdef _HAVE_ZLIB_
 #include <zlib.h>
 #endif
 
+#ifdef _HAVE_PNG_
+#include <png.h>
+#endif
+
 #ifdef _HAVE_LIB_PPM_
 #include <netpbm/ppm.h>
 #endif
 
-#ifdef _HAVE_PNG_
-#include "png.h"
-#endif
 
 static void spudec_handle_rest(spudec_handle_t *this);
 
 // read a big endian 16 bit value
 static inline unsigned int get_be16(const unsigned char *p)
