--- ace/Makefile.am.orig	2008-10-22 22:28:29.000000000 +0300
+++ ace/Makefile.am	2008-10-22 22:29:50.000000000 +0300
@@ -876,6 +876,7 @@
   Recursive_Thread_Mutex.inl \
   Recyclable.h \
   Recyclable.inl \
+  Refcountable.h \
   Refcountable_T.cpp \
   Refcountable_T.h \
   Refcountable_T.inl \
@@ -1054,6 +1055,7 @@
   Timeprobe_T.h \
   Timer_Hash_T.cpp \
   Timer_Hash_T.h \
+  Timer_Heap.h \
   Timer_Heap_T.cpp \
   Timer_Heap_T.h \
   Timer_List_T.cpp \
@@ -1061,6 +1063,7 @@
   Timer_Queue_Adapters.cpp \
   Timer_Queue_Adapters.h \
   Timer_Queue_Adapters.inl \
+  Timer_Queue.h \
   Timer_Queue_T.cpp \
   Timer_Queue_T.h \
   Timer_Queue_T.inl \
