--- ifp-gnome.py
+++ ifp-gnome.py
@@ -21,7 +21,7 @@
 
 class GiRiverApp:
 	def __init__(self):
-		gladefile="ifp-gnome.glade"
+		gladefile="/usr/share/ifp-gnome/ifp-gnome.glade"
 		windowname="window1"
 		progresswinname="winProgress"
 
@@ -46,7 +46,7 @@
 		self.wTree.signal_autoconnect(dic)
 		self.mainWindow = self.wTree.get_widget(windowname)
 		self.mainWindow.hide()
-		self.mainWindow.set_icon_from_file('ifp-gnome.png')
+		self.mainWindow.set_icon_from_file('/usr/share/ifp-gnome/ifp-gnome.png')
 
 		# Progress Window
 		self.progressWindow = self.progTree.get_widget(progresswinname)
@@ -55,7 +55,7 @@
 
 		
 		self.progressWindow.hide()
-		self.progressWindow.set_icon_from_file('ifp-gnome.png')
+		self.progressWindow.set_icon_from_file('/usr/share/ifp-gnome/ifp-gnome.png')
 
 		# Transfer Window
 		self.interrupted = False
@@ -67,7 +67,7 @@
 		self.verDialog.signal_autoconnect(dictt)
 		self.verifyWindow = self.verDialog.get_widget("winTransfer")
 		self.verifyWindow.hide()
-		self.verifyWindow.set_icon_from_file('ifp-gnome.png')
+		self.verifyWindow.set_icon_from_file('/usr/share/ifp-gnome/ifp-gnome.png')
 
 		self.lblBattery = self.wTree.get_widget("lblBattery")
 		self.lblSpace = self.wTree.get_widget("lblSpace")
@@ -752,7 +752,7 @@
 		gnome.ui.About("iFP-Gnome","0.7", "Copyright 2006 Billy Charlton",
              "A GNOME file manager for some iRiver Flash devices",
              ["Billy Charlton <billy@trigby.com>"],[],"",
-             gtk.gdk.pixbuf_new_from_file("ifp-gnome.png")).show()
+             gtk.gdk.pixbuf_new_from_file("/usr/share/ifp-gnome/ifp-gnome.png")).show()
 
 
 # MAIN ENTRY POINT ---------------------
