From ccb0d5e98b30577fe27016b5bc774388f02db876 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sat, 19 Sep 2020 08:00:51 +0200
Subject: [PATCH] Remove tests relying on py<3.9 behavior

---
 tests/test_clean.py | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/tests/test_clean.py b/tests/test_clean.py
index 1cd58df..2b2f7c5 100644
--- a/tests/test_clean.py
+++ b/tests/test_clean.py
@@ -499,31 +499,16 @@ def test_attributes_list():
             {"protocols": ["http"]},
             '<a href="example.com">valid</a>',
         ),
-        (
-            '<a href="example.com:8000">valid</a>',
-            {"protocols": ["http"]},
-            '<a href="example.com:8000">valid</a>',
-        ),
         (
             '<a href="localhost">valid</a>',
             {"protocols": ["http"]},
             '<a href="localhost">valid</a>',
         ),
-        (
-            '<a href="localhost:8000">valid</a>',
-            {"protocols": ["http"]},
-            '<a href="localhost:8000">valid</a>',
-        ),
         (
             '<a href="192.168.100.100">valid</a>',
             {"protocols": ["http"]},
             '<a href="192.168.100.100">valid</a>',
         ),
-        (
-            '<a href="192.168.100.100:8000">valid</a>',
-            {"protocols": ["http"]},
-            '<a href="192.168.100.100:8000">valid</a>',
-        ),
         # Disallow implicit http if disallowed
         ('<a href="example.com">foo</a>', {"protocols": []}, "<a>foo</a>"),
         ('<a href="example.com:8000">foo</a>', {"protocols": []}, "<a>foo</a>"),
-- 
2.28.0

