diff --git a/tests/unit/auth_test.py b/tests/unit/auth_test.py
index 10b77e9..c40f439 100644
--- a/tests/unit/auth_test.py
+++ b/tests/unit/auth_test.py
@@ -38,12 +38,12 @@ class LoadAuthTestCase(TestCase):
 
         # Test a case with valid params
         with patch('salt.utils.format_call') as format_call_mock:
-            expected_ret = call('fake_func_str', {
+            expected_ret = [call('fake_func_str', {
                 'username': 'test_user',
                 'test_password': '',
                 'show_timeout': False,
                 'eauth': 'pam'
-            }, expected_extra_kws=auth.AUTH_INTERNAL_KEYWORDS)
+            }, expected_extra_kws=auth.AUTH_INTERNAL_KEYWORDS)]
             ret = self.lauth.load_name(valid_eauth_load)
             format_call_mock.assert_has_calls(expected_ret)
 
@@ -53,12 +53,12 @@ class LoadAuthTestCase(TestCase):
                             'test_password': '',
                             'eauth': 'pam'}
         with patch('salt.utils.format_call') as format_call_mock:
-            expected_ret = call('fake_groups_function_str', {
+            expected_ret = [call('fake_groups_function_str', {
                 'username': 'test_user',
                 'test_password': '',
                 'show_timeout': False,
                 'eauth': 'pam'
-                }, expected_extra_kws=auth.AUTH_INTERNAL_KEYWORDS)
+                }, expected_extra_kws=auth.AUTH_INTERNAL_KEYWORDS)]
             self.lauth.get_groups(valid_eauth_load)
             format_call_mock.assert_has_calls(expected_ret)
 
