diff --git a/tests/pytests/unit/modules/dockermod/test_module.py b/tests/pytests/unit/modules/dockermod/test_module.py
index 8fb7806497..584d416412 100644
--- a/tests/pytests/unit/modules/dockermod/test_module.py
+++ b/tests/pytests/unit/modules/dockermod/test_module.py
@@ -354,10 +354,6 @@ def test_update_mine():
         mine_mock.assert_called_once()
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_list_networks():
     """
     test list networks.
@@ -378,10 +374,6 @@ def test_list_networks():
     client.networks.assert_called_once_with(names=["foo"], ids=["01234"])
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_create_network():
     """
     test create network.
@@ -422,10 +414,6 @@ def test_create_network():
     )
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_remove_network():
     """
     test remove network.
@@ -444,10 +432,6 @@ def test_remove_network():
     client.remove_network.assert_called_once_with("foo")
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_inspect_network():
     """
     test inspect network.
@@ -466,10 +450,6 @@ def test_inspect_network():
     client.inspect_network.assert_called_once_with("foo")
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_connect_container_to_network():
     """
     test connect_container_to_network
@@ -491,10 +471,6 @@ def test_connect_container_to_network():
     client.connect_container_to_network.assert_called_once_with("container", "foo")
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_disconnect_container_from_network():
     """
     test disconnect_container_from_network
@@ -513,10 +489,6 @@ def test_disconnect_container_from_network():
     client.disconnect_container_from_network.assert_called_once_with("container", "foo")
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_list_volumes():
     """
     test list volumes.
@@ -539,10 +511,6 @@ def test_list_volumes():
     )
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_create_volume():
     """
     test create volume.
@@ -569,10 +537,6 @@ def test_create_volume():
     )
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_remove_volume():
     """
     test remove volume.
@@ -591,10 +555,6 @@ def test_remove_volume():
     client.remove_volume.assert_called_once_with("foo")
 
 
-@pytest.mark.skipif(
-    docker_mod.docker.version_info < (1, 5, 0),
-    reason="docker module must be installed to run this test or is too old. >=1.5.0",
-)
 def test_inspect_volume():
     """
     test inspect volume.
