--- salt-3002.5/tests/unit/modules/test_cmdmod.py	2021-02-26 11:01:33.407219444 -0800
+++ salt-3002.5-python3_7/tests/unit/modules/test_cmdmod.py	2021-02-26 11:01:47.709124416 -0800
@@ -560,6 +560,7 @@
 
         self.assertEqual(ret["stdout"], salt.utils.stringutils.to_unicode(stdout))
 
+    @skipIf(True, "doesn't work in sandbox")
     def test_run_all_output_loglevel_debug(self):
         """
         Test that specifying debug for loglevel
--- salt-3002.5//tests/unit/utils/test_thin.py	2021-02-26 13:30:47.841179546 -0800
+++ salt-3002.5-python3_7//tests/unit/utils/test_thin.py	2021-02-26 13:41:33.681174531 -0800
@@ -1275,6 +1275,7 @@
     @skipIf(
         salt.utils.platform.is_windows(), "salt-ssh does not deploy to/from windows"
     )
+    @skipIf(True, "does not work with network-sandbox")
     def test_thin_dir(self):
         """
         Test the thin dir to make sure salt-call can run
--- salt-3002.5/tests/unit/modules/test_cmdmod.py	2021-02-26 13:30:48.014178373 -0800
+++ salt-3002.5-python3_7/tests/unit/modules/test_cmdmod.py	2021-02-26 13:43:22.395441008 -0800
@@ -439,6 +439,7 @@
 
     @skipIf(salt.utils.platform.is_windows(), "Do not run on Windows")
     @skipIf(salt.utils.platform.is_darwin(), "Do not run on MacOS")
+    @skipIf(True, "does not work with network-sandbox")
     def test_run_cwd_in_combination_with_runas(self):
         """
         cmd.run executes command in the cwd directory
