diff -r -U2 asymptote-2.41.orig/doc/FAQ/bfnnconv.pl asymptote-2.41/doc/FAQ/bfnnconv.pl
--- asymptote-2.41.orig/doc/FAQ/bfnnconv.pl	2017-03-22 14:56:46.000000000 +0700
+++ asymptote-2.41/doc/FAQ/bfnnconv.pl	2017-10-31 23:44:05.511045390 +0700
@@ -62,5 +62,5 @@
 open(U,">$prefix.xrefdb-new");
 
-for $x (@outputs) { require("m-$x.pl"); }
+for $x (@outputs) { require("./m-$x.pl"); }
 
 &call('init');
