--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,7 @@
 
     license="GPLv3+",
 
-    data_files=[('identicurse',['README', 'conf/config.json'])],
+    data_files=[('share/identicurse',['conf/config.json'])],
     packages=find_packages('src'),
     package_dir={'': 'src'},
     include_package_data=True,
--- a/src/identicurse/identicurse.py
+++ b/src/identicurse/identicurse.py
@@ -156,7 +156,7 @@
             else:
                 import getpass, time
                 # no config yet, so let's build one
-                config.config.load(os.path.join(self.path, "config.json"))
+                config.config.load(os.path.join("/", "usr", "share", "identicurse", "config.json"))
                 print "No config was found, so we will now run through a few quick questions to set up a basic config for you (which will be saved as %s so you can manually edit it later). If the default (where defaults are available, they're stated in []) is already fine for any question, just press Enter without typing anything, and the default will be used." % (config.config.filename)
                 print "This version of IdentiCurse supports OAuth login. Using OAuth to log in means that you do not need to enter your username and password."
                 use_oauth = raw_input("Use OAuth [Y/n]? ").upper()
