From a193e48dd01ad3aef9dd998c1247f649719f0a80 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Sun, 1 Aug 2021 07:48:51 +0100
Subject: [PATCH] Drop bashisms from configure.ac

---
 configure.ac | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9ae1292..106ea57 100644
--- a/configure.ac
+++ b/configure.ac
@@ -218,18 +218,18 @@ selected_for_media_art="no  (disabled)"
 # Choose between backends (GdkPixbuf/Qt/etc)
 ##################################################################
 
-if test "x$enable_qt" == "xyes" && test "x$enable_gdkpixbuf" == "xyes"; then
+if test "x$enable_qt" = "xyes" && test "x$enable_gdkpixbuf" = "xyes"; then
    AC_MSG_ERROR([Can not enable both Qt and GdkPixbuf backends, please pick one])
 fi
 
 if test "x$enable_qt" != "xno" && test "x$enable_gdkpixbuf" != "xyes"; then
-   if test "x$with_qt_version" == "xauto" || test "x$with_qt_version" == "x5"; then
+   if test "x$with_qt_version" = "xauto" || test "x$with_qt_version" = "x5"; then
        PKG_CHECK_MODULES(QT5,
                          [Qt5Gui >= $QT5_REQUIRED],
                          [have_qt5=yes],
                          [have_qt5=no])
    fi
-   if test "x$with_qt_version" == "xauto" || test "x$with_qt_version" == "x4"; then
+   if test "x$with_qt_version" = "xauto" || test "x$with_qt_version" = "x4"; then
        PKG_CHECK_MODULES(QT4,
                          [QtGui >= $QT4_REQUIRED],
                          [have_qt4=yes],
-- 
2.32.0

