--- process-conduit-1.1.0.0-orig/process-conduit.cabal	2014-04-03 18:54:59.000000000 +1100
+++ process-conduit-1.1.0.0/process-conduit.cabal	2014-07-02 09:45:32.738952743 +1000
@@ -34,6 +34,7 @@
                      , text             >= 0.11
                      , process          >= 1.0
                      , conduit          >= 1.1
+                     , conduit-extra    >= 1.1
                      , resourcet        >= 1.1
                      , shakespeare-text >= 1.0
                      , shakespeare
@@ -48,4 +49,6 @@
                      , bytestring
                      , hspec           >= 1.3
                      , conduit
+                     , conduit-extra
                      , process-conduit
+                     , resourcet
--- process-conduit-1.1.0.0-orig/test.hs	2014-04-03 18:54:59.000000000 +1100
+++ process-conduit-1.1.0.0/test.hs	2014-07-02 09:51:31.862391926 +1000
@@ -7,6 +7,7 @@
 import Data.Conduit
 import qualified Data.Conduit.Binary as CB
 import Test.Hspec
+import Control.Monad.Trans.Resource (runResourceT)
 
 main :: IO ()
 main = hspec $ do
