--- src/r_bsp.h.old	2006-05-06 02:07:04.000000000 +0200
+++ src/r_bsp.h	2006-05-06 02:08:48.000000000 +0200
@@ -41,11 +41,6 @@
 extern line_t   *linedef;
 extern sector_t *frontsector;
 extern sector_t *backsector;
-extern int      rw_x;
-extern int      rw_stopx;
-extern boolean  segtextured;
-extern boolean  markfloor;      /* false if the back side is the same plane */
-extern boolean  markceiling;
 
 /* old code -- killough:
  * extern drawseg_t drawsegs[MAXDRAWSEGS];
