From 5d70cd5ec5ac8460ff5d3d6b0b47b9c60e1f3ca2 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Tue, 10 Apr 2018 21:59:32 +0200
Subject: [PATCH] Fix build with Qt-5.11

---
 avogadro/qtplugins/gamessinput/gamessinput.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/avogadro/qtplugins/gamessinput/gamessinput.cpp b/avogadro/qtplugins/gamessinput/gamessinput.cpp
index 779d2d4..0b2e106 100644
--- a/avogadro/qtplugins/gamessinput/gamessinput.cpp
+++ b/avogadro/qtplugins/gamessinput/gamessinput.cpp
@@ -24,9 +24,9 @@
 
 #include <molequeue/client/jobobject.h>
 
-#include <QtCore/QDebug>
-
-#include <QtWidgets/QMessageBox>
+#include <QAction>
+#include <QDebug>
+#include <QMessageBox>
 
 namespace Avogadro {
 namespace Core {
-- 
2.17.0

