From bccc569b2c049e6eb1c43b45ba0e25bb39868913 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sandro=20Knau=C3=9F?= <sknauss@kde.org>
Date: Wed, 25 Jan 2017 23:35:22 +0100
Subject: [PATCH] import QImage directly instead of usage via KContacts

---
 resources/openxchange/oxa/contactutils.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/resources/openxchange/oxa/contactutils.cpp b/resources/openxchange/oxa/contactutils.cpp
index dea8188..06bb34b 100644
--- a/resources/openxchange/oxa/contactutils.cpp
+++ b/resources/openxchange/oxa/contactutils.cpp
@@ -28,6 +28,7 @@
 #include <Akonadi/Contact/ContactGroupExpandJob>
 
 #include <QtCore/QBuffer>
+#include <QtGui/QImage>
 #include <QtXml/QDomElement>
 
 using namespace OXA;
-- 
2.10.2

