From a6b370873e406f9921a50359ed8ebf4714776411 Mon Sep 17 00:00:00 2001
From: Jean-Guillaume Dumas <Jean-Guillaume.Dumas@imag.fr>
Date: Wed, 11 Oct 2023 14:19:52 +0200
Subject: [PATCH] cstdint required for uint64_t

---
 src/kernel/system/givpower.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/kernel/system/givpower.h b/src/kernel/system/givpower.h
index 5644264d..080e114d 100644
--- a/src/kernel/system/givpower.h
+++ b/src/kernel/system/givpower.h
@@ -16,6 +16,8 @@
 #ifndef __GIVARO_power_H
 #define __GIVARO_power_H
 
+#include <cstdint>
+
 namespace Givaro {
 
     // -------------------------------------------------------------

From c7744bb133496cd7ac04688f345646d505e1bf52 Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Thu, 19 Jan 2023 09:12:22 -0500
Subject: [PATCH] Add missing #include <cstdint> for (u)int64_t

Fixes failure to compile on GCC 13.
---
 src/library/poly1/givdegree.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/library/poly1/givdegree.h b/src/library/poly1/givdegree.h
index 3753a425..eb85a0dd 100644
--- a/src/library/poly1/givdegree.h
+++ b/src/library/poly1/givdegree.h
@@ -19,6 +19,8 @@
 #ifndef __GIVARO_poly1degree_H
 #define __GIVARO_poly1degree_H
 
+#include <cstdint>
+
 #include <iostream>
 
 namespace Givaro {
