diff --git a/CMakeLists.txt b/CMakeLists.txt
index e2946ba..039d72a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,9 +41,9 @@ include(GenerateExportHeader)
 include(CMakePackageConfigHelpers)
 
 # Set the version information here
-set(VERSION_INFO_MAJOR_VERSION 0) # increment major on api compatibility changes
-set(VERSION_INFO_MINOR_VERSION 6) # increment minor on feature-level changes
-set(VERSION_INFO_PATCH_VERSION git) # increment patch for bug fixes and docs
+set(VERSION_INFO_MAJOR_VERSION 0 CACHE STRING "Increment major on api compatibility changes")
+set(VERSION_INFO_MINOR_VERSION 6 CACHE STRING "Increment minor on feature-level changes")
+set(VERSION_INFO_PATCH_VERSION git CACHE STRING "increment patch for bug fixes and docs")
 include(Version) # setup version info
 
 ########################################################################
