--- a/data/desktop/desktop.sh
+++ b/data/desktop/desktop.sh
@@ -18,17 +18,14 @@ echo "[Desktop Action Play]"
 echo "Exec=parole --play"
 echo "_Name=Play/Pause"
 echo "Icon=media-playback-start-symbolic"
-echo "OnlyShowIn=Unity;"
 echo ""
 echo "[Desktop Action Previous]"
 echo "Exec=parole --previous"
 echo "_Name=Previous Track"
 echo "Icon=media-skip-backward-symbolic"
-echo "OnlyShowIn=Unity;"
 echo ""
 echo "[Desktop Action Next]"
 echo "Exec=parole --next"
 echo "_Name=Next Track"
 echo "Icon=media-skip-forward-symbolic"
-echo "OnlyShowIn=Unity;"
 echo ""
-- 
2.26.2

