diff -Nur hdf-4.2.7-patch1.orig/configure.ac hdf-4.2.7-patch1/configure.ac
--- hdf-4.2.7-patch1.orig/configure.ac	2012-03-18 03:45:46.000000000 +0000
+++ hdf-4.2.7-patch1/configure.ac	2012-04-13 03:04:17.000000000 +0100
@@ -200,20 +200,6 @@
 dnl Build static libraries by default. Furthermore, fortran shared libraries
 dnl are unsupported. Disallow a user from enabling both shared libraries and
 dnl fortran.
-if test "X${enable_shared}" != "Xyes"; then
-    enable_shared="no"
-fi
-
-if test "X${enable_shared}" = "Xyes"; then
-    if test "X${BUILD_FORTRAN}" = "Xyes"; then
-        AC_MSG_ERROR([Cannot build shared fortran libraries. Please configure with --disable-fortran flag.])
-    fi
-fi
-
-AC_PROG_INSTALL
-AC_PROG_LN_S
-
-AC_CHECK_PROGS([AR], [ar xar], [:], [$PATH])
 
 AC_CHECK_PROGS([YACC], ['bison -y' byacc yacc], [none], [])
 
@@ -404,6 +390,8 @@
 
     if test -z "$HAVE_SZIP"; then
       AC_MSG_ERROR([couldn't find szlib library])
+    else
+      USE_COMP_SZIP="yes"
     fi
     ;;
   no)
@@ -447,7 +435,7 @@
 
     AC_CHECK_LIB([sz], [SZ_BufftoBuffCompress],, [unset HAVE_SZIP])
 
-    if test -z "$HAVE_SZIP"; then
+    if test "X$HAVE_SZIP" != "Xyes"; then
       AC_MSG_ERROR([couldn't find szlib library])
     else
       USE_COMP_SZIP="yes"
@@ -703,7 +691,7 @@
 dnl ======================================================================
 dnl Libtool initialization
 dnl ======================================================================
-LT_INIT(dlopen disable-shared) 
+LT_INIT(dlopen) 
 LT_OUTPUT
 
 dnl ======================================================================
@@ -781,11 +769,6 @@
 dnl ------------------------------------------------------------------------
 dnl Check to see if libtool has enabled shared libraries. Set a conditional
 dnl as some Makefiles will build based on availability of shared libraries.
-if (./libtool --features | grep '^enable shared libraries' > /dev/null); then
-  enable_shared=yes
-else
-  enable_shared=no
-fi
 
 dnl ------------------------------------------------------------------------
 dnl Specify shared library extension the host machine should recognize.
@@ -840,23 +823,8 @@
 dnl name of the compiler and the reported version number.
 AC_SUBST([F77_VERSION])
 dnl Strip anything that looks like a flag off of $F77
-F77_NOFLAGS=`echo $F77 | sed 's/ -.*//'`
-
-if `echo $F77_NOFLAGS | grep ^/ >/dev/null 2>&1`; then
-  F77_VERSION="$F77"
-else
-  F77_VERSION="$F77";
-  for x in `echo $PATH | sed -e 's/:/ /g'`; do
-    if test -x $x/$F77_NOFLAGS; then
-      F77_VERSION="$x/$F77"
-      break
-    fi
-  done
-fi
-if test -n "$fc_version_info"; then
-  F77_VERSION="$F77_VERSION ( $fc_version_info)"
-fi
 
+F77_VERSION="$F77";
 
 dnl This part doesn't work yet since HDF4 config files do not contain
 dnl information for fortran_vendor and fortran_version. 
diff -Nur hdf-4.2.7-patch1.orig/hdf/test/fortestF.f hdf-4.2.7-patch1/hdf/test/fortestF.f
--- hdf-4.2.7-patch1.orig/hdf/test/fortestF.f	2012-03-18 03:44:01.000000000 +0000
+++ hdf-4.2.7-patch1/hdf/test/fortestF.f	2012-04-13 02:20:45.000000000 +0100
@@ -203,7 +203,7 @@
 
 	if (cmd .NE. 'Test' .AND. cmd .NE. 'test') then
 	 print *, 'Unknown Command: ', cmd, param
-	 print *, 'Try one of "Skip", "Test", "Verbosity" or "Cleanup"'
+	 print *, 'Try one of Skip, Test, Verbosity or Cleanup'
 	 retcode = -1
 	 return
 	endif
diff -Nur hdf-4.2.7-patch1.orig/mfhdf/libsrc/Makefile.am hdf-4.2.7-patch1/mfhdf/libsrc/Makefile.am
--- hdf-4.2.7-patch1.orig/mfhdf/libsrc/Makefile.am	2012-03-18 03:44:00.000000000 +0000
+++ hdf-4.2.7-patch1/mfhdf/libsrc/Makefile.am	2012-04-13 02:20:45.000000000 +0100
@@ -32,10 +32,11 @@
 else
 libmfhdf_la_SOURCES = $(CSOURCES)
 endif
+libmfhdf_la_LIBADD = $(top_builddir)/hdf/src/libdf.la
 
 if HDF_BUILD_XDR
 XDRLIB = ../xdr/libxdr.la
-libmfhdf_la_LIBADD = $(XDRLIB)
+libmfhdf_la_LIBADD += $(XDRLIB)
 endif
 
 if HDF_BUILD_NETCDF
