--- a/configure.ac
+++ b/configure.ac
@@ -270,6 +270,11 @@
    LIBS="$LIBS $V_PCAPDEP"
 fi
 
+AC_ARG_WITH(wrappers,
+   [  --with-wrappers         build with libwrappers suuport],
+   with_wrappers="$withval",
+   with_wrappers="yes")
+if test "x$with_wrappers" != "xno"; then
 AC_QOSIENT_TCPWRAP(V_WRAPDEP, V_INCLS)
 if test ! -z "$V_WRAPDEP"; then
    if test -f $V_WRAPDEP; then
@@ -293,6 +298,7 @@
    AC_DEFINE([HAVE_TCP_WRAPPER], [], [Using Tcp wrappers])
    WRAPLIBS="$V_WRAPDEP"
 fi
+fi
 
 AC_CHECK_FUNCS(xdrmem_create)
 if test "$ac_cv_func_xdrmem_create" = yes; then
