https://bugs.gentoo.org/show_bug.cgi?id=175630
JTRiley

--- ccp4-6.0.2.orig/src/refmac5_/make_unix_ccp4.f	2005-09-06 07:29:56.000000000 -0400
+++ ccp4-6.0.2.new/src/refmac5_/make_unix_ccp4.f	2007-05-02 20:03:46.000000000 -0400
@@ -97,7 +97,7 @@
 
       READ(CID,'(I2,1X,I2,1X,I2,2X)') IH(4),IH(3),IH(2)
 
-      CALL IDATE(IMON,IDAY,IYR)
+      CALL UIDATE(IMON,IDAY,IYR)
       IF(IYR.GE.99) THEN
         IYR=IYR+1900
       ELSE
