--- ccp4-5.99.5/lib/clipper/configure.orig	2006-01-07 16:54:13.000000000 -0800
+++ ccp4-5.99.5/lib/clipper/configure	2006-01-07 16:54:31.000000000 -0800
@@ -21751,7 +21751,7 @@
 
 if test "x$mccp4_prefix" != x; then
  # ie. mccp4=thing was given (thing is what we're checking for)
- ac_MCCP4_CXXFLAGS="-I$mccp4_prefix/include"
+ ac_MCCP4_CXXFLAGS="-I$mccp4_prefix/include -I$mccp4_prefix/include/mccp4"
  ac_MCCP4_LDOPTS="-L$mccp4_prefix/lib -lmccp4"
 else
  # treat as standard lib/include
