diff -up rocksndiamonds-4.1.4.1/src/main.c.gcc10 rocksndiamonds-4.1.4.1/src/main.c
--- rocksndiamonds-4.1.4.1/src/main.c.gcc10	2019-05-20 15:24:46.000000000 -0400
+++ rocksndiamonds-4.1.4.1/src/main.c	2020-01-31 05:56:37.154931291 -0500
@@ -67,7 +67,7 @@ short			ExplodeDelay[MAX_LEV_FIELDX][MAX
 int			RunnerVisit[MAX_LEV_FIELDX][MAX_LEV_FIELDY];
 int			PlayerVisit[MAX_LEV_FIELDX][MAX_LEV_FIELDY];
 
-int			GfxFrame[MAX_LEV_FIELDX][MAX_LEV_FIELDY];
+// int			GfxFrame[MAX_LEV_FIELDX][MAX_LEV_FIELDY];
 int 			GfxRandom[MAX_LEV_FIELDX][MAX_LEV_FIELDY];
 int 			GfxElement[MAX_LEV_FIELDX][MAX_LEV_FIELDY];
 int			GfxAction[MAX_LEV_FIELDX][MAX_LEV_FIELDY];
@@ -132,7 +132,7 @@ struct LevelInfo	level, level_template;
 struct PlayerInfo	stored_player[MAX_PLAYERS], *local_player = NULL;
 struct HiScore		highscore[MAX_SCORE_ENTRIES];
 struct TapeInfo		tape;
-struct SetupInfo	setup;
+// struct SetupInfo	setup;
 struct GameInfo		game;
 struct GlobalInfo	global;
 struct BorderInfo	border;
