diff --git a/setup.py b/setup.py
index 4e7645c..aa3895a 100644
--- a/setup.py
+++ b/setup.py
@@ -180,10 +180,10 @@ setup(name='setools',
                 'build_lex': LexCommand,
                 'build_ext': BuildExtCommand,
                 'build_qhc': QtHelpCommand},
-      packages=['setools', 'setools.diff', 'setools.policyrep', 'setoolsgui', 'setoolsgui.apol'],
-      scripts=['apol', 'sediff', 'seinfo', 'seinfoflow', 'sesearch', 'sedta'],
+      packages=['setools', 'setools.diff', 'setools.policyrep'],
+      scripts=['sediff', 'seinfo', 'seinfoflow', 'sesearch', 'sedta'],
       data_files=[(join(sys.prefix, 'share/man/man1'), glob.glob("man/*.1"))],
-      package_data={'': ['*.ui', '*.qhc'], 'setools': ['perm_map']},
+      package_data={'setools': ['perm_map']},
       ext_modules=ext_py_mods,
       test_suite='tests',
       license='GPLv2+, LGPLv2.1+',
