diff -u MUSTANG_v.3.orig/src/ew_2.cpp MUSTANG_v.3/src/ew_2.cpp
--- MUSTANG_v.3.orig/src/ew_2.cpp	2008-01-15 11:18:34.000000000 -0700
+++ MUSTANG_v.3/src/ew_2.cpp	2008-12-09 10:47:22.000000000 -0700
@@ -36,6 +36,7 @@
 using std::ifstream;
 
 #include<math.h>
+#include <cstdlib>
 #include "macros.h"
 #include "globals.h"
 #include "ew.h"
diff -u MUSTANG_v.3.orig/src/multiple_superposition.cpp MUSTANG_v.3/src/multiple_superposition.cpp
--- MUSTANG_v.3.orig/src/multiple_superposition.cpp	2008-01-15 11:18:34.000000000 -0700
+++ MUSTANG_v.3/src/multiple_superposition.cpp	2008-12-09 10:49:00.000000000 -0700
@@ -51,6 +51,7 @@
 #include "alloc_routines.h"
 #include "de_alloc_routines.h"
 #include <cmath>
+#include <cstdlib>
 #define THRESH 0.0001
 
 void MOVE_TO_CENTER_OF_GRAVITY( int NSTRUC , int NRES , float ***R , float **CMs )
diff -u MUSTANG_v.3.orig/src/multiple_superposition.h MUSTANG_v.3/src/multiple_superposition.h
--- MUSTANG_v.3.orig/src/multiple_superposition.h	2008-01-15 11:18:34.000000000 -0700
+++ MUSTANG_v.3/src/multiple_superposition.h	2008-12-09 10:48:11.000000000 -0700
@@ -25,3 +25,4 @@
 #define MULT_SUPERPOSE
 void  M_SUPERPOSE( int , int , float***, float** , float*** ) ;
 #endif
+#include <cstdlib>
diff -u MUSTANG_v.3.orig/src/pdb_ripper.h MUSTANG_v.3/src/pdb_ripper.h
--- MUSTANG_v.3.orig/src/pdb_ripper.h	2008-01-15 11:18:34.000000000 -0700
+++ MUSTANG_v.3/src/pdb_ripper.h	2008-12-09 10:46:47.000000000 -0700
@@ -29,4 +29,5 @@
 void PARSE_ENTIRE_PDB_STRUCTURE( char * , int ) ;
 
 #endif
+#include <cstdlib>
 
diff -u MUSTANG_v.3.orig/src/progress_align.h MUSTANG_v.3/src/progress_align.h
--- MUSTANG_v.3.orig/src/progress_align.h	2008-01-15 11:18:34.000000000 -0700
+++ MUSTANG_v.3/src/progress_align.h	2008-12-09 10:47:50.000000000 -0700
@@ -25,3 +25,4 @@
 #define PROG_ALGN_H
 void PROGRESSIVE_ALIGNMENT_USING_EXTENDED_EDGE_WEIGHTS() ;
 #endif
+#include <cstdlib>
diff -u MUSTANG_v.3.orig/src/read_structures.h MUSTANG_v.3/src/read_structures.h
--- MUSTANG_v.3.orig/src/read_structures.h	2008-01-15 11:18:34.000000000 -0700
+++ MUSTANG_v.3/src/read_structures.h	2008-12-09 10:46:22.000000000 -0700
@@ -26,3 +26,4 @@
 void READ_STRUCTURES( char **paths );
 void READ_ENTIRE_PDBS( char **paths) ;
 #endif
+#include <cstdlib>
