diff -Naur synapse-0.2.10.orig/src/core/desktop-file-service.vala synapse-0.2.10/src/core/desktop-file-service.vala
--- synapse-0.2.10.orig/src/core/desktop-file-service.vala	2012-03-18 20:02:41.000000000 +0100
+++ synapse-0.2.10/src/core/desktop-file-service.vala	2012-12-26 18:07:55.962276823 +0100
@@ -125,7 +125,11 @@
         }
 
         name = app_info.get_name ();
-        exec = app_info.get_executable ();
+        exec = app_info.get_commandline ();
+        if (exec == null)
+        {
+          throw new DesktopFileError.UNINTERESTING_ENTRY ("Unable to get exec for %s".printf (name));
+        }
 
         // check for hidden desktop files
         if (keyfile.has_key (GROUP, "Hidden") &&
