diff -ur jaxws-si.old/src/tools/wscompile/src/com/sun/tools/ws/util/xml/XmlUtil.java jaxws-si/src/tools/wscompile/src/com/sun/tools/ws/util/xml/XmlUtil.java
--- jaxws-si.old/src/tools/wscompile/src/com/sun/tools/ws/util/xml/XmlUtil.java	2008-10-05 12:32:27.000000000 +0300
+++ jaxws-si/src/tools/wscompile/src/com/sun/tools/ws/util/xml/XmlUtil.java	2008-10-05 12:32:42.000000000 +0300
@@ -44,7 +44,6 @@
 import org.w3c.dom.NodeList;
 import org.w3c.dom.Text;
 
-import com.sun.xml.messaging.saaj.util.ByteInputStream;
 import com.sun.tools.ws.util.WSDLParseException;
 import com.sun.xml.ws.util.xml.NamedNodeMapIterator;
 import com.sun.xml.ws.util.xml.NodeListIterator;
