--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -345,8 +345,9 @@
                      COMMAND ${GZIP} ARGS -9 -n -c ${PROJECT_SOURCE_DIR}/docs/pwsafe.1 > pwsafe.1.gz
                      WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
   ADD_CUSTOM_TARGET(manpage ALL DEPENDS ${CMAKE_BINARY_DIR}/pwsafe.1.gz)
-  install (FILES ${CMAKE_BINARY_DIR}/pwsafe.1.gz DESTINATION "share/man/man1")
-  install (FILES "install/desktop/pwsafe.desktop" DESTINATION "share/applications")
+  install (FILES ${PROJECT_SOURCE_DIR}/docs/pwsafe.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1 RENAME passwordsafe.1)
+  install (FILES "install/desktop/pwsafe.desktop" DESTINATION "share/applications" RENAME "passwordsafe.desktop")
+
   install (FILES "install/metainfo/org.pwsafe.pwsafe.metainfo.xml" DESTINATION "share/metainfo")
   install (FILES "install/graphics/48x48/pwsafe.png" DESTINATION "share/icons/hicolor/48x48/apps/")
   install (FILES "install/graphics/128x128/pwsafe.png" DESTINATION "share/icons/hicolor/128x128/apps/")
