From: Mike Hommey <mh@glandium.org>
Date: Wed, 25 Feb 2015 10:41:37 +0900
Subject: Make powerpc not use static page sizes in mozjemalloc

Closes: #763900
---
 memory/mozjemalloc/jemalloc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/memory/mozjemalloc/jemalloc.c b/memory/mozjemalloc/jemalloc.c
index ee9e288..2e27e69 100644
--- a/memory/mozjemalloc/jemalloc.c
+++ b/memory/mozjemalloc/jemalloc.c
@@ -1089,7 +1089,7 @@ static const bool config_recycle = false;
  * controlling the malloc behavior are defined as compile-time constants
  * for best performance and cannot be altered at runtime.
  */
-#if !defined(__ia64__) && !defined(__sparc__) && !defined(__mips__)
+#if !defined(__ia64__) && !defined(__sparc__) && !defined(__mips__) && !defined(__powerpc__)
 #define MALLOC_STATIC_SIZES 1
 #endif
 
