diff --git a/python/mach/mach/site.py b/python/mach/mach/site.py
index 8fef9bfaf8..61c3101c11 100644
--- a/python/mach/mach/site.py
+++ b/python/mach/mach/site.py
@@ -940,6 +940,9 @@ def _assert_pip_check(topsrcdir, pthfile_lines, virtualenv_name):
     If there's an incompatibility, raise an exception and allow it to bubble up since
     it will require user intervention to resolve.
     """
+
+    return True
+
     if os.environ.get(
         f"MACH_SYSTEM_ASSERTED_COMPATIBLE_WITH_{virtualenv_name.upper()}_SITE", None
     ):
