Index: mp3_check-1.98/Makefile
===================================================================
--- mp3_check-1.98.orig/Makefile
+++ mp3_check-1.98/Makefile
@@ -2,7 +2,7 @@
 
 CC ?= gcc
 LOCALBASE ?= /usr/local
-CFLAGS += -Wall -Werror
+#CFLAGS += -Wall -Werror
 OPT_FLAGS = -L${LOCALBASE}/lib 
 OBJS = 	mp3_check.o \
 	mp3_check_misc.o
@@ -16,8 +16,7 @@ norm: 	$(OBJS) 
 	$(CC) -g $(CFLAGS) $(OBJS) -o mp3_check
 
 trim: 	$(OBJS)
-	$(CC) $(CFLAGS) $(OPT_FLAGS) $(OBJS) -o mp3_check
-	strip mp3_check
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OPT_FLAGS) $(OBJS) -o mp3_check
 	cat MILESTONE.INTRO
 
 install: all
