--- a/Makefile	2018-10-12 00:25:16.000000000 +0200
+++ b/Makefile	2018-10-26 17:59:44.458689842 +0200
@@ -378,7 +378,6 @@
 install-server: install-server-generic
 
 install-common: build
-	./init/adduser.sh ${OSSEC_USER} ${OSSEC_USER_MAIL} ${OSSEC_USER_REM} ${OSSEC_GROUP} ${PREFIX}
 	install -d -m 0550 -o root -g ${OSSEC_GROUP} ${PREFIX}/
 	install -d -m 0750 -o ${OSSEC_USER} -g ${OSSEC_GROUP} ${PREFIX}/logs
 	install -m 0660 -o ${OSSEC_USER} -g ${OSSEC_GROUP} /dev/null ${PREFIX}/logs/ossec.log
@@ -1166,7 +1165,7 @@
 #### test ##########
 ####################
 
-CFLAGS_TEST = -g -O0 --coverage
+CFLAGS_TEST = -g -O0
 
 LDFLAGS_TEST = -lcheck -lm -pthread -lrt
 
@@ -1177,7 +1176,7 @@
 
 test_programs = test_os_zlib test_os_xml test_os_regex test_os_crypto test_shared
 
-.PHONY: test run_tests build_tests test_valgrind test_coverage
+.PHONY: test run_tests build_tests test_valgrind
 
 test: build_tests
 	${MAKE} run_tests
