--- configure.in
+++ configure.in
@@ -267,10 +267,7 @@
 AC_MSG_CHECKING(linker flags used to link Python)
 AC_SUBST(LDFLAGS)
 PyLFS=`grep "^LINKFORSHARED=" ${PyLIBPL}/Makefile | cut -f2 -d= | tr '\011\012\015' '   '`
-PyLDFLAGS=`grep "^LDFLAGS=" ${PyLIBPL}/Makefile | cut -f2 -d= | tr '\011\012\015' '   '`
-LDFLAGS="${LDFLAGS} ${PyLFS} ${PyLDFLAGS}"
-LDFLAGS="${LDFLAGS} ${PY_LDFLAGS}"
-AC_MSG_RESULT($PY_LDFLAGS)
+LDFLAGS="${LDFLAGS} ${PyLFS}"
 
 AC_MSG_CHECKING(where Python include files are)
 AC_SUBST(INCLUDES)
