diff --git a/configure.ac b/configure.ac
index ecbc3bb..5973ff5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -135,7 +135,6 @@ AM_CONDITIONAL([ENABLE_INTEGRATION],[test "x$enable_integration" = "xyes"])
 #   these macros are defined in m4/flags.m4
 AX_ADD_COMPILER_FLAG([-Wall])
 AX_ADD_COMPILER_FLAG([-Wextra])
-AX_ADD_COMPILER_FLAG([-Werror])
 AX_ADD_COMPILER_FLAG([-std=gnu99])
 AX_ADD_COMPILER_FLAG([-Wformat])
 AX_ADD_COMPILER_FLAG([-Wformat-security])
@@ -143,19 +142,9 @@ AX_ADD_COMPILER_FLAG([-Wno-missing-braces])
 # work around for Glib usage of function pointers type casting
 #   https://bugzilla.gnome.org/show_bug.cgi?id=793272
 AX_ADD_COMPILER_FLAG([-Wno-cast-function-type])
-AX_ADD_COMPILER_FLAG([-fdata-sections])
-AX_ADD_COMPILER_FLAG([-ffunction-sections])
-AX_ADD_COMPILER_FLAG([-fstack-protector-all])
-AX_ADD_COMPILER_FLAG([-fpic])
-AX_ADD_COMPILER_FLAG([-fPIC])
 AX_ADD_PREPROC_FLAG([-D_GNU_SOURCE])
 AX_ADD_PREPROC_FLAG([-U_FORTIFY_SOURCE])
 AX_ADD_PREPROC_FLAG([-D_FORTIFY_SOURCE=2])
-AX_ADD_LINK_FLAG([-Wl,--gc-sections])
-AX_ADD_LINK_FLAG([-Wl,--no-undefined])
-AX_ADD_LINK_FLAG([-Wl,-z,noexecstack])
-AX_ADD_LINK_FLAG([-Wl,-z,now])
-AX_ADD_LINK_FLAG([-Wl,-z,relro])
 
 AC_SUBST([PATH])
 
From 353c5e962f46e3739b1e92cbf6b1f3eb500157fc Mon Sep 17 00:00:00 2001
From: Alon Bar-Lev <alon.barlev@gmail.com>
Date: Thu, 17 Jan 2019 23:56:50 +0200
Subject: [PATCH] tests: run without X

Solves:

$ DISPLAY= make check

FAIL: test/tss2-tcti-tabrmd_unit

** (process:17633): CRITICAL **: 23:53:22.403: failed to allocate dbus proxy object:
Error spawning command line ?dbus-launch --autolaunch=4f9189e347d5cdf343

Signed-off-by: Alon Bar-Lev <alon.barlev@gmail.com>

https://github.com/tpm2-software/tpm2-abrmd/pull/577
---
 .travis.yml | 2 +-
 Makefile.am | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index b32a5fb..2598f13 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -76,7 +76,8 @@ XFAIL_TESTS = \
 TEST_EXTENSIONS = .int
 AM_TESTS_ENVIRONMENT = \
     TEST_FUNC_LIB=$(srcdir)/scripts/int-test-funcs.sh \
-    PATH=./src:$(PATH)
+    PATH=./src:$(PATH) \
+    dbus-launch
 INT_LOG_COMPILER = $(srcdir)/scripts/int-test-setup.sh
 INT_LOG_FLAGS = --tabrmd-tcti=$(TABRMD_TCTI)
 
-- 
2.19.2

