From 160ea833ef64916906c9334f7da12db3208293eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20K=2E=20H=C3=BCttel?= <dilfridge@gentoo.org>
Date: Wed, 16 Oct 2019 17:00:25 +1300
Subject: Avoid clobbering CFLAGS in Makefile.pl

Bug: https://bugs.gentoo.org/658000
---
 Makefile.PL | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.PL b/Makefile.PL
index 34f1008..6ede3a6 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -15,7 +15,6 @@ test_requires 'Test::More' => '0.47';
 requires_external_cc();
 
 cc_lib_links('crack');
-cc_optimize_flags('-g -Wall -Werror');
 
 auto_install();
 WriteAll();
-- 
2.27.0

