--- a/oam/CMakeLists.txt
+++ b/oam/CMakeLists.txt
@@ -82,15 +82,6 @@ set_property(TARGET ${OAM_TARGET} PROPERTY
 set_property(TARGET ${OAM_TARGET} PROPERTY
              VERSION "${SO_VERSION_STRING}")
 
-## If the library is a release, strip the target library
-if ("${CMAKE_BUILD_TYPE}" STREQUAL Release)
-    if(${BUILD_SHARED_LIBS}) #striping only for .so
-        add_custom_command(
-             TARGET ${OAM_TARGET}
-             POST_BUILD COMMAND ${CMAKE_STRIP} lib${OAM_TARGET}.so)
-    endif()
-endif ()
-
 # use the target_include_directories() command to specify the include directories for the target
 target_include_directories(${OAM_TARGET}
                            PUBLIC
--- a/rocm_smi/CMakeLists.txt
+++ b/rocm_smi/CMakeLists.txt
@@ -103,15 +103,6 @@ set_property(TARGET ${ROCM_SMI_TARGET} PROPERTY
 set_property(TARGET ${ROCM_SMI_TARGET} PROPERTY
              VERSION "${SO_VERSION_STRING}")
 
-## If the library is a release, strip the target library
-if ("${CMAKE_BUILD_TYPE}" STREQUAL Release)
-  if(${BUILD_SHARED_LIBS}) #stripping only for .so
-    add_custom_command(
-          TARGET ${ROCM_SMI_TARGET}
-          POST_BUILD COMMAND ${CMAKE_STRIP} lib${ROCM_SMI_TARGET}.so.${SO_VERSION_STRING})
-  endif()
-endif ()
-
 #file reorganization changes
 #rocm_smi.py moved to libexec/rocm_smi. so creating rocm-smi symlink
 file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
