--- a/meson.build
+++ b/meson.build
@@ -6,7 +6,7 @@
     default_options: [
         'warning_level=2'
     ],
-    meson_version: '>=0.63.0'
+    meson_version: '>=0.64.0'
 )
 
 cpp = meson.get_compiler('cpp')
@@ -69,7 +69,7 @@
 endif # Platform detection
 
 pymod = import('python')
-python = pymod.find_installation(get_option('python'), required: true)
+python = pymod.find_installation(get_option('python'), required: true, pure: false)
 
 # Generate _rtmidi extension source
 subdir('src')
--- a/rtmidi/meson.build
+++ b/rtmidi/meson.build
@@ -78,6 +78,5 @@
 python.install_sources(
     python_sources,
     version_py,
-    pure: true,
     subdir: 'rtmidi',
 )
