Index: 2ManDVD/videowrapper.cpp
===================================================================
--- 2ManDVD.orig/videowrapper.cpp
+++ 2ManDVD/videowrapper.cpp
@@ -99,11 +99,11 @@ int videowrapper::load_video(QString fil
 
     img_convert_ctx = sws_getContext(pCodecCtx->width,
                     pCodecCtx->height, pCodecCtx->pix_fmt, pCodecCtx->width,
-                    pCodecCtx->height, PIX_FMT_RGB24, SWS_BICUBIC, NULL,
+                    pCodecCtx->height, AV_PIX_FMT_RGB24, SWS_BICUBIC, NULL,
                     NULL, NULL);
 
     // *** ON DETERMINE LA TAILLE DU BUFFER ***
-    numBytes=avpicture_get_size(PIX_FMT_RGB24, pCodecCtx->width,
+    numBytes=avpicture_get_size(AV_PIX_FMT_RGB24, pCodecCtx->width,
                                 pCodecCtx->height);
     buffer=(uint8_t *)av_malloc(numBytes*sizeof(uint8_t));
     old_frame = -1;
@@ -120,10 +120,10 @@ QImage videowrapper::read_frame(int64_t
     if ( num_frame != old_frame )
     {
         // *** ALLOCATE VIDEO FRAME ***
-        pFrameRGB=avcodec_alloc_frame();
-        pFrame=avcodec_alloc_frame();
+        pFrameRGB=av_frame_alloc();
+        pFrame=av_frame_alloc();
 
-        avpicture_fill((AVPicture *)pFrameRGB, buffer, PIX_FMT_RGB24,
+        avpicture_fill((AVPicture *)pFrameRGB, buffer, AV_PIX_FMT_RGB24,
                         pCodecCtx->width, pCodecCtx->height);
         av_seek_frame(pFormatCtx,videoStream,0, AVSEEK_FLAG_FRAME);
         compte_frame = -1;
