Index: 2ManDVD/tetrahedron.h
===================================================================
--- 2ManDVD.orig/tetrahedron.h
+++ 2ManDVD/tetrahedron.h
@@ -83,7 +83,6 @@ public:
     float scale_factor;
     float wg_x;
     float wdecal;
-    GLfloat light_position[];
 
     GLfloat anag_x;
     GLfloat anag_y;
@@ -427,6 +426,8 @@ private:
     void dropEvent(QDropEvent *);
     void dragEnterEvent(QDragEnterEvent *);
     Qt::DropActions supportedDropActions () const;
+public:
+    GLfloat light_position[];
 };
 
 #endif
