--- a/libpcap_stuff.c
+++ b/libpcap_stuff.c
@@ -17,7 +17,7 @@
 #include <stdlib.h>
 #include <sys/ioctl.h>
 #include <pcap.h>
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 
 #include "globals.h"
 
--- a/script.c
+++ b/script.c
@@ -24,7 +24,7 @@
 
 #include <sys/ioctl.h>
 #include <pcap.h>
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 
 #include "release.h"
 #include "hping2.h"
--- a/configure
+++ b/configure
@@ -93,7 +93,7 @@
 	else
 		USE_TCL=""
 		TCL_LIB=""
-		echo "==> WARNING: no Tcl header files found!"
+		echo "==> WARNING: no Tcl header files found."
 	fi
 fi
 if [ -n $USE_TCL ]
--- a/Makefile.in
+++ b/Makefile.in
@@ -52,9 +52,6 @@
 hping3: byteorder.h $(OBJ)
 	$(CC) -o hping3 $(CCOPT) $(DEBUG) $(OBJ) -L/usr/local/lib $(PCAP) @SOLARISLIB@ @TCL_LIB@
 	@echo
-	./hping3 -v
-	@echo "use \`make strip' to strip hping3 binary"
-	@echo "use \`make install' to install hping3"
 
 hping3-static: byteorder.h $(OBJ)
 	$(CC) -static -o hping3-static $(CCOPT) $(DEBUG) $(OBJ) -L/usr/local/lib $(PCAP) @SOLARISLIB@ @TCL_LIB@ -ldl
