diff --git a/elogv b/elogv
index 85d507b..bef7ae9 100755
--- a/elogv
+++ b/elogv
@@ -544,7 +544,7 @@ if __name__ == "__main__":
     if port_settings['PORT_LOGDIR']:
         elogdir = os.path.join(port_settings['PORT_LOGDIR'],"elog")
     else:
-        elogdir = os.path.join(os.sep,"var","log","portage","elog")
+        elogdir = os.path.join(os.sep,'@GENTOO_PORTAGE_EPREFIX@',"var","log","portage","elog")
 
     # Launch curses interface
     try:
