From 2e3812d2c1d1f54861fb79f5c2b0197de96a00a3 Mon Sep 17 00:00:00 2001
From: Arvin Schnell <aschnell@suse.de>
Date: Wed, 6 Sep 2017 09:49:38 +0200
Subject: [PATCH] - added include statement

---
 client/cleanup.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/client/cleanup.cc b/client/cleanup.cc
index 0dbfc03..d7f2f68 100644
--- a/client/cleanup.cc
+++ b/client/cleanup.cc
@@ -21,6 +21,7 @@
  */
 
 
+#include <iostream>
 #include <vector>
 
 #include "dbus/DBusMessage.h"
