diff -Naurp gpe-login-0.95/configure.ac gpe-login-0.95.orig/configure.ac
--- gpe-login-0.95/configure.ac	2008-08-13 18:12:22.000000000 +0000
+++ gpe-login-0.95.orig/configure.ac	2010-04-02 03:38:59.000000000 +0000
@@ -4,11 +4,34 @@ AC_PREREQ(2.59)
 AC_INIT(gpe-login,0.95,gpe-list@linuxtogo.org)
 AC_CONFIG_SRCDIR([gpe-login.c])
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
+AM_CONFIG_HEADER([config.h:config.h.in])
 
 AC_PROG_CC
 AC_STDC_HEADERS
 AC_PROG_INSTALL
 
+AC_ARG_WITH([min-valid-uid],
+            [AS_HELP_STRING([--with-min-valid-uid=N],
+              [Set the minimal UID for valid users (default: 100).])],
+            [],
+            [with_min_valid_uid=100])
+
+AS_IF([test "x$with_min_valid_uid" != "x"], [
+  AC_DEFINE_UNQUOTED([MIN_UID], [$with_min_valid_uid],
+            [Minimal UID for valid users.])
+])
+
+AC_ARG_WITH([min-valid-gid],
+            [AS_HELP_STRING([--with-min-valid-gid=N],
+              [Set the minimal GID for valid users (default: 100).])],
+            [],
+            [with_min_valid_gid=100])
+
+AS_IF([test "x$with_min_valid_gid" != "x"], [
+  AC_DEFINE_UNQUOTED([MIN_GID], [$with_min_valid_gid],
+            [Minimal GID for valid users.])
+])
+
 
 DEP_MODULES="glib-2.0 >= 2.2 libgpewidget"
 DEP_MODULES_LOCK="x11"
diff -Naurp gpe-login-0.95/gpe-login.c gpe-login-0.95.orig/gpe-login.c
--- gpe-login-0.95/gpe-login.c	2010-04-02 03:41:41.000000000 +0000
+++ gpe-login-0.95.orig/gpe-login.c	2010-04-02 03:40:35.000000000 +0000
@@ -44,6 +44,13 @@
 #include <gpe/spacing.h>
 #include <gpe/translabel.h>
 
+#ifdef HAVE_CONFIG_H 
+#include <config.h> 
+#else
+#define MIN_GID 100
+#define MIN_UID 100
+#endif /* HAVE_CONFIG_H */ 
+
 #include "gpe-ownerinfo.h"
 
 #define _(x) gettext(x)
@@ -348,7 +355,7 @@ slurp_passwd (GtkWidget *menu)
 	  && strcmp (pw->pw_passwd, "*"))
 	root_password_set = TRUE;
 
-      if (pw->pw_uid < 100 || pw->pw_uid >= 65534)
+      if (pw->pw_uid < MIN_UID || pw->pw_uid >= 65534)
 	continue;
 
       have_users = TRUE;
@@ -717,8 +724,8 @@ enter_newuser_callback (GtkWidget *widge
   char salt[2];
   char buf[256];
   char home[80];
-  gid_t gid = 100;
-  uid_t uid = 100;
+  gid_t gid = MIN_UID;
+  uid_t uid = MIN_GID;
   FILE *fp;
 
   username = gtk_entry_get_text (GTK_ENTRY (entry_username));
