--- gpe-login.c	2008-08-13 19:08:28.000000000 +0100
+++ /home/miknix/Community/gpe-login.c	2009-04-07 14:16:16.000000000 +0100
@@ -1,6 +1,9 @@
 /*
  * Copyright (C) 2002, 2003, 2005 Philip Blundell <philb@gnu.org>
  *
+ * Add gentoo logo widget
+ * Copyright (C) 2009, Angelo Arrifano <miknix@gentoo.org>
+ *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License
  * as published by the Free Software Foundation; either version
@@ -1584,6 +1587,8 @@ main (int argc, char *argv[])
   FILE *cfp;
   GdkCursor *cursor;
   GtkWidget *socket_frame;
+  GtkWidget *logo;
+
 
   /* gchar *gpe_catindent = gpe_get_catindent ();  */
   /* guint gpe_catspacing = gpe_get_catspacing (); */
@@ -2042,6 +2047,14 @@ main (int argc, char *argv[])
          gtk_socket_steal (GTK_SOCKET (socket), xkbd_xid);
     }
 
+  /* Show up a beauty gentoo powered logo */
+#define GENTOO_LOGO_DIR "/usr/share/pixmaps/gpe-login-gentoo.png"
+  if (! access(GENTOO_LOGO_DIR, R_OK)) {
+     logo = gtk_image_new_from_file(GENTOO_LOGO_DIR);
+     gtk_box_pack_end (GTK_BOX (vbox2), logo, FALSE, FALSE, 0);
+     gtk_widget_show(logo);
+  }
+
   gtk_main ();
 
   cleanup_children ();
