--- base-orig/runtime/objs/makefile	2012-04-18 10:28:08.000000000 +1000
+++ base/runtime/objs/makefile	2018-03-10 10:21:07.417776270 +1100
@@ -12,6 +12,7 @@
 AR =		ar
 ARFLAGS =	rcv
 RANLIB =	ranlib
+PAXMARK =	true
 
 #
 # dummy version
@@ -114,6 +115,7 @@
 
 $(RUNTIME):	$(VERSION) main.o $(OBJS) $(ALL_LIBS)
 	$(CC) -o $(RUNTIME) $(CFLAGS) $(LDFLAGS) main.o $(OBJS) $(ALL_LIBS) $(LD_LIBS) -lm
+	$(PAXMARK) -m $(RUNTIME)
 
 $(RUNTIME_A):	$(VERSION) main.o $(OBJS) $(ALL_LIBS)
 	rm -f $(RUNTIME_A)
--- base-orig/runtime/objs/mk.alpha32-dunix	2000-06-02 04:34:03.000000000 +1000
+++ base/runtime/objs/mk.alpha32-dunix	2018-03-10 11:31:32.572918457 +1100
@@ -10,6 +10,7 @@
 LDFLAGS =	-non_shared -T 20000000 -D 40000000
 AS =		as -nocpp
 AR =		/usr/ucb/ar
+PAXMARK =	true
 XOBJS =	
 XLIBS =	
 LD_LIBS =
@@ -20,4 +21,4 @@
 RUNTIME =	run.alpha32-dunix
 
 all:
-	(make RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" AS="$(AS)" AR="$(AR)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
+	(make RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" AS="$(AS)" AR="$(AR)" PAXMARK="$(PAXMARK)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
--- base-orig/runtime/objs/mk.alpha32x-osf1	2000-06-02 04:34:03.000000000 +1000
+++ base/runtime/objs/mk.alpha32x-osf1	2018-03-10 11:31:32.573918461 +1100
@@ -8,6 +8,7 @@
 LDFLAGS =	-non_shared -T 20000000 -D 40000000
 AS =		as -nocpp
 AR =		/usr/ucb/ar
+PAXMARK = 	true
 XOBJS =	
 XLIBS =	
 LD_LIBS =
@@ -18,4 +19,4 @@
 RUNTIME =	run.alpha32x-osf1
 
 all:
-	(make RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" AS="$(AS)" AR="$(AR)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
+	(make RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" AS="$(AS)" AR="$(AR)" PAXMARK="$(PAXMARK)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
--- base-orig/runtime/objs/mk.amd64-darwin	2017-08-23 04:41:35.000000000 +1000
+++ base/runtime/objs/mk.amd64-darwin	2018-03-10 11:31:32.575918471 +1100
@@ -14,6 +14,7 @@
 CC =		cc -m64 -std=c99
 CFLAGS =	-g -O2 -D_DARWIN_C_SOURCE $(SDK)
 CPP =           cc -x assembler-with-cpp -E -P -std=c99
+PAXMARK =	true
 XOBJS =
 XLIBS =         ../c-libs/dl/libunix-dynload.a
 LD_LIBS =       -ldl
@@ -28,5 +29,5 @@
 RUNTIMEX =	runx.amd64-darwin
 
 all:
-	($(MAKE) RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CFLAGS="$(CFLAGS)" CPP="$(CPP)" AS="$(AS)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
-	($(MAKE) RUNTIME="$(RUNTIMEX)" VERSION="$(VERSION)" CC="$(CC)" CFLAGS="$(CFLAGS)" CPP="$(CPP)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" LDFLAGS="-Xlinker -r" $(RUNTIMEX))
+	($(MAKE) RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CFLAGS="$(CFLAGS)" CPP="$(CPP)" AS="$(AS)" PAXMARK="$(PAXMARK)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
+	($(MAKE) RUNTIME="$(RUNTIMEX)" VERSION="$(VERSION)" CC="$(CC)" CFLAGS="$(CFLAGS)" CPP="$(CPP)" PAXMARK="$(PAXMARK)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" LDFLAGS="-Xlinker -r" $(RUNTIMEX))
--- base-orig/runtime/objs/mk.hppa-hpux	2000-06-02 04:34:03.000000000 +1000
+++ base/runtime/objs/mk.hppa-hpux	2018-03-10 11:31:32.577918480 +1100
@@ -16,6 +16,7 @@
 
 AS =		/usr/ccs/bin/as 
 RANLIB =	ar ts
+PAXMARK = 	true
 
 XOBJS =
 XLIBS =
@@ -27,4 +28,4 @@
 RUNTIME =	run.hppa-hpux
 
 all:
-	(make  RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CPP="$(CPP)" CFLAGS="$(CFLAGS)" AS="$(AS)" RANLIB="$(RANLIB)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
+	(make  RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CPP="$(CPP)" CFLAGS="$(CFLAGS)" AS="$(AS)" RANLIB="$(RANLIB)" PAXMARK="$(PAXMARK)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
--- base-orig/runtime/objs/mk.hppa-hpux11	2005-07-21 00:30:18.000000000 +1000
+++ base/runtime/objs/mk.hppa-hpux11	2018-03-10 11:31:32.579918490 +1100
@@ -17,6 +17,7 @@
 
 AS =		/usr/ccs/bin/as 
 RANLIB =	ar ts
+PAXMARK = 	true
 
 XOBJS =
 XLIBS =
@@ -28,4 +29,4 @@
 RUNTIME =	run.hppa-hpux11
 
 all:
-	(make  RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CPP="$(CPP)" CFLAGS="$(CFLAGS)" AS="$(AS)" RANLIB="$(RANLIB)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
+	(make  RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CPP="$(CPP)" CFLAGS="$(CFLAGS)" AS="$(AS)" RANLIB="$(RANLIB)" PAXMARK="$(PAXMARK)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
--- base-orig/runtime/objs/mk.hppa-hpux9	2000-06-02 04:34:03.000000000 +1000
+++ base/runtime/objs/mk.hppa-hpux9	2018-03-10 11:31:32.581918499 +1100
@@ -13,6 +13,7 @@
 
 AS =		/bin/as
 RANLIB =	ar ts
+PAXMARK =	true
 
 XOBJS =
 XLIBS =
@@ -24,4 +25,4 @@
 RUNTIME =	run.hppa-hpux9
 
 all:
-	(make CHECK_HEAP=check-heap.o RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CPP="$(CPP)" CFLAGS="$(CFLAGS)" AS="$(AS)" RANLIB="$(RANLIB)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
+	(make CHECK_HEAP=check-heap.o RUNTIME="$(RUNTIME)" VERSION="$(VERSION)" CC="$(CC)" CPP="$(CPP)" CFLAGS="$(CFLAGS)" AS="$(AS)" RANLIB="$(RANLIB)" PAXMARK="$(PAXMARK)" TARGET=$(TARGET) DEFS="$(DEFS)" XOBJS="$(XOBJS)" XLIBS="$(XLIBS)" LD_LIBS="$(LD_LIBS)" $(RUNTIME))
