diff -Naur usr.orig/share/applications/swarmplayer.desktop usr/share/applications/swarmplayer.desktop
--- usr.orig/share/applications/swarmplayer.desktop	2013-05-09 20:47:46.000000000 -0400
+++ usr/share/applications/swarmplayer.desktop	2013-05-09 20:48:34.000000000 -0400
@@ -5,4 +5,4 @@
 Icon=swarmplayer
 Terminal=false
 Type=Application
-Categories=Application;Network;P2P
+Categories=Network;P2P;
diff -Naur usr.orig/share/applications/tribler.desktop usr/share/applications/tribler.desktop
--- usr.orig/share/applications/tribler.desktop	2013-05-09 20:47:46.000000000 -0400
+++ usr/share/applications/tribler.desktop	2013-05-09 20:49:03.000000000 -0400
@@ -5,5 +5,5 @@
 Icon=tribler
 Terminal=false
 Type=Application
-Categories=Application;Network;P2P
-MimeType=x-scheme-handler/ppsp;x-scheme-handler/tswift;x-scheme-handler/magnet
\ No newline at end of file
+Categories=Network;P2P;
+MimeType=x-scheme-handler/ppsp;x-scheme-handler/tswift;x-scheme-handler/magnet;
