diff -Naur usr.orig/bin/tribler usr/bin/tribler
--- usr.orig/bin/tribler	2012-02-07 10:33:48.000000000 -0500
+++ usr/bin/tribler	2012-02-11 12:19:05.000000000 -0500
@@ -6,5 +6,7 @@
 export PYTHONPATH="$PYTHONPATH":$_TRIBLERPATH
 
 echo "Starting Tribler..."
+STATEDIR="$HOME/.Tribler"
+[ ! -d $STATEDIR ] && mkdir -p $STATEDIR
 cd $_TRIBLERPATH
-exec python -O Tribler/Main/tribler.py "$@" > /tmp/$USER-tribler.log 2>&1
+exec python2 -O Tribler/Main/tribler.py "$@" > $STATEDIR/tribler.log 2>&1
