--- a/OpenRGB.pro
+++ b/OpenRGB.pro
@@ -46,9 +46,6 @@ TEMPLATE    = app
 #-----------------------------------------------------------------------------------------------#
 win32:BUILDDATE = $$system(date /t)
 unix:BUILDDATE  = $$system(date -R -d "@${SOURCE_DATE_EPOCH:-$(date +%s)}")
-GIT_COMMIT_ID   = $$system(git log -n 1 --pretty=format:"%H")
-GIT_COMMIT_DATE = $$system(git log -n 1 --pretty=format:"%ci")
-GIT_BRANCH      = $$system(git branch --show-current)
 
 DEFINES +=                                                                                      \
     VERSION_STRING=\\"\"\"$$VERSION\\"\"\"                                                      \
