diff -Naur Perl-Tags-0.32/t/05_vim.t Perl-Tags-0.32b/t/05_vim.t
--- Perl-Tags-0.32/t/05_vim.t	2014-05-11 12:11:34.000000000 +0000
+++ Perl-Tags-0.32b/t/05_vim.t	2016-03-19 04:25:04.831985917 +0000
@@ -37,6 +37,7 @@
     system vim =>
              -u => 't/_vimrc',        # use our vimrc to add Perl::Tags settings etc.
              -S => 't/session.vim',   # use our session file to make modfications to file
+             '--noplugin',            # Don't load 3rd party plugins which can break
             '-n',                     # don't use swapfile
             $temp;
 
