Don't override hardened toolchain defaults.
--- a/configure.ac
+++ b/configure.ac
@@ -311,8 +311,6 @@ my_CFLAGS="\
 -Wshadow \
 -Wstrict-prototypes \
 -Wtype-limits \
--fstack-protector \
--D_FORTIFY_SOURCE=2 \
 -DNO_COMPAT_ISAL_CRYPTO_API_2_24
 "
 AC_SUBST([my_CFLAGS])
--- a/make.inc
+++ b/make.inc
@@ -51,7 +51,7 @@ DEBUG      = -g
 DEBUG_nasm = -g
 
 # Default arch= build options
-CFLAGS_    = -Wall -Wchar-subscripts -Wformat-security -Wnested-externs -Wpointer-arith -Wshadow -Wstrict-prototypes -Wtype-limits -fstack-protector -D_FORTIFY_SOURCE=2 -DNO_COMPAT_ISAL_CRYPTO_API_2_24
+CFLAGS_    = -Wall -Wchar-subscripts -Wformat-security -Wnested-externs -Wpointer-arith -Wshadow -Wstrict-prototypes -Wtype-limits -DNO_COMPAT_ISAL_CRYPTO_API_2_24
 ASFLAGS_   = -f elf64
 ARFLAGS_   = cr $@
 STRIP_gcc  = strip -d -R .comment $@
