From 1c64add35ed4918222ceb4fe93b7bef0e2b4e41c Mon Sep 17 00:00:00 2001
From: stolk <b.stolk@gmail.com>
Date: Wed, 1 Apr 2020 10:59:16 -0700
Subject: [PATCH] Do not return a dangling pointer!

Signed-off-by: Ross Charles Campbell <rossbridger.cc@gmail.com>
---
 Extras/ConvexDecomposition/cd_wavefront.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Extras/ConvexDecomposition/cd_wavefront.cpp b/Extras/ConvexDecomposition/cd_wavefront.cpp
index b33080964..53b5784df 100644
--- a/Extras/ConvexDecomposition/cd_wavefront.cpp
+++ b/Extras/ConvexDecomposition/cd_wavefront.cpp
@@ -437,7 +437,7 @@ const char **InPlaceParser::GetArglist(char *line, int &count)  // convert sourc
 {
 	const char **ret = 0;
 
-	const char *argv[MAXARGS];
+	static const char *argv[MAXARGS];
 	int argc = 0;
 
 	char *foo = line;
-- 
2.28.0

