--- dnsval-1.13/libsres/Makefile.in
+++ dnsval-1.13/libsres/Makefile.in
@@ -63,7 +63,7 @@
 	$(RM) -rf $(LT_DIR)
 
 $(LIBRES): $(LOBJ)
-	$(LIBTOOLLD) -o $@ $(LOBJ) 
+	$(LIBTOOLLD) -o $@ $(LOBJ) $(LDFLAGS) -lssl
 
 install:
 	$(MKPATH) $(DESTDIR)$(libdir)
--- dnsval-1.13/libval/Makefile.in
+++ dnsval-1.13/libval/Makefile.in
@@ -60,7 +60,7 @@
 	$(RM) -rf $(LT_DIR)
 
 $(LIBVAL): $(LOBJ)
-	$(LIBTOOLLD) -o $@ $(LOBJ)
+	$(LIBTOOLLD) -o $@ $(LOBJ) $(LDFLAGS) -lssl -lpthread
 
 install:
 	$(MKPATH) $(DESTDIR)$(libdir)
