diff -ruN flightgear-2020.3.11/src/Navaids/PositionedOctree.cxx fgfs-my/src/Navaids/PositionedOctree.cxx
--- flightgear-2020.3.11/src/Navaids/PositionedOctree.cxx	2021-07-29 11:35:40.000000000 +0200
+++ fgfs-my/src/Navaids/PositionedOctree.cxx	2021-09-16 23:01:20.989931176 +0200
@@ -148,7 +148,7 @@
 
 void Leaf::insertChild(FGPositioned::Type ty, PositionedID id)
 {
-  assert(childrenLoaded);
+  assert(_childrenLoaded);
   children.insert(children.end(), TypedPositioned(ty, id));
 }
 
