--- abiword-3.0.1/plugin-configure.m4.orig	2016-04-10 14:22:18.271356761 +0100
+++ abiword-3.0.1/plugin-configure.m4	2016-04-10 14:24:22.493303398 +0100
@@ -1355,7 +1355,7 @@
 AC_SUBST([BMP_LIBS])
 
 
-wpg_pkgs="$gsf_req libwpg-0.3 libwpd-0.10 librevenge-0.0 librevenge-stream-0.0"
+wpg_pkgs="libwpg-0.3 $gsf_req"
 wpg_deps="no"
 
 if test "$enable_wpg" != ""; then
@@ -1469,8 +1469,8 @@
 AC_SUBST([AIKSAURUS_LIBS])
 
 
-wordperfect_pkgs="libwpg-0.3 libwpd-0.10 librevenge-0.0 librevenge-stream-0.0 $gsf_req"
-wordperfect_wps_pkgs='libwps-0.4'
+wordperfect_pkgs="libwpd-0.10 $gsf_req"
+wordperfect_wps_pkgs='libwps-0.3'
 wordperfect_deps="no"
 
 WORDPERFECT_CFLAGS=
@@ -1499,7 +1499,7 @@
 
 PKG_CHECK_EXISTS([ $wordperfect_wps_pkgs ],
 [
-	wp_deps_pkgs="$wp_deps_pkgs $wordperfect_wps_pkgs"
+	wp_deps_pkgs="$wordperfect_wps_pkgs $wp_deps_pkgs"
 	WPS_DEFINE=" -DHAVE_LIBWPS"
 ])
 
