--- a/resources/templates/CMakeLists.txt
+++ b/resources/templates/CMakeLists.txt
@@ -5,7 +5,7 @@ ${CMAKE_SOURCE_DIR}/scribus
 
 # Gentoo has an option for no templates. Let's make the ebuild simpler and open up the option to others, but leave as a default for us. Make the directory in any case as our source will look for it.
 if (WANT_NOTEMPLATES)
-  install(CODE "file(MAKE_DIRECTORY ${ENV}${CMAKE_INSTALL_PREFIX}/${TEMPLATEDIR})")
+  install(CODE "file(MAKE_DIRECTORY ${ENV}./${CMAKE_INSTALL_PREFIX}/${TEMPLATEDIR})")
 else (WANT_NOTEMPLATES)
 
 install(FILES
