Index: DFBSee-0.7.4/src/rotate.c
===================================================================
--- DFBSee-0.7.4.orig/src/rotate.c
+++ DFBSee-0.7.4/src/rotate.c
@@ -110,10 +110,10 @@ rotate (void         *dest,
       switch (bpp)
         {
         case 2:
-          ROTATE_LEFT (__u16 *)
+          ROTATE_LEFT (uint16_t *)
           break;
         case 4:
-          ROTATE_LEFT (__u32 *)
+          ROTATE_LEFT (uint32_t *)
           break;
         default:
           break;
@@ -125,10 +125,10 @@ rotate (void         *dest,
       switch (bpp)
         {        
         case 2:
-          ROTATE_RIGHT (__u16 *)
+          ROTATE_RIGHT (uint16_t *)
           break;
         case 4:
-          ROTATE_RIGHT (__u32 *)
+          ROTATE_RIGHT (uint32_t *)
           break;
         default:
           break;
@@ -139,10 +139,10 @@ rotate (void         *dest,
       switch (bpp)
         {
         case 2:
-          ROTATE_HALF (__u16 *)
+          ROTATE_HALF (uint16_t *)
           break;
         case 4:
-           ROTATE_HALF (__u32 *)
+           ROTATE_HALF (uint32_t *)
           break;
         default:
           break;
