http://bugs.gentoo.org/311573

--- src/mlview-tree-view.h
+++ src/mlview-tree-view.h
@@ -178,7 +178,7 @@
 
 	enum MlViewStatus build_app_edit_menu () ;
 
-	GtkDialog * TreeView::get_expand_tree_dialog ()  ;
+	GtkDialog * get_expand_tree_dialog ()  ;
 
 	void clear_completion_popup_submenus () ;
 
--- src/mlview-source-view.h
+++ src/mlview-source-view.h
@@ -53,7 +53,7 @@
 
 	enum MlViewStatus set_default_options () ;
 
-	MlViewXMLDocument* SourceView::get_document () ;
+	MlViewXMLDocument* get_document () ;
 
 	GtkUIManager * get_ui_manager () ;
 
--- src/mlview-validator.cc
+++ src/mlview-validator.cc
@@ -25,6 +25,7 @@
 
 #include "mlview-validator.h"
 #include "mlview-safe-ptr-utils.h"
+#include <memory>
 
 #define MESSAGE_LINE_LENGTH 55
 namespace mlview
--- src/mlview-plugin.h
+++ src/mlview-plugin.h
@@ -46,12 +46,12 @@
 		
 		//forbid assignation/copy
 		Plugin& operator= (Plugin const&) ;
-		Plugin::Plugin (Plugin const&) ;
+		Plugin (Plugin const&) ;
 
 	public:
 		Plugin (const PluginDescriptor &a_descr) ;
 
-		Plugin::Plugin (const UString &a_url) ;
+		Plugin (const UString &a_url) ;
 		
 		virtual ~Plugin () ;
 		
--- src/mlview-editor.cc
+++ src/mlview-editor.cc
@@ -157,7 +157,7 @@
 			contextual_menu (NULL)
 	{}
 
-	static void EditorPriv::schemas_window_destroy_cb
+	static void schemas_window_destroy_cb
 	(GtkWidget *a_widget, struct DocumentWindowData *a_win) ;
 
 	static void validation_window_destroy_cb
@@ -406,17 +406,13 @@
 	         gtk_dialog_run (GTK_DIALOG (dialog));
 
 
-	switch (button) {
-	case GTK_RESPONSE_OK:
+	if (button == GTK_RESPONSE_OK) {
 		sel_menu_item = gtk_menu_get_active(GTK_MENU(menu));
 
 		result = (ViewDescriptor *)g_object_get_data
 		         (G_OBJECT(sel_menu_item), "mlview_view_desc");
-		break;
-
-	default:
+	} else {
 		result = NULL ;
-		break;
 	}
 	gtk_widget_destroy (dialog);
 	return result;
