diff --git a/SConstruct b/SConstruct
index df92e41659..2c387d5a24 100644
--- a/SConstruct
+++ b/SConstruct
@@ -2831,8 +2831,6 @@ def doConfigure(myenv):
     if use_system_version_of_library("boost"):
         if not conf.CheckCXXHeader( "boost/filesystem/operations.hpp" ):
             myenv.ConfError("can't find boost headers")
-        if not conf.CheckBoostMinVersion():
-            myenv.ConfError("system's version of boost is too old. version 1.49 or better required")
 
         # Note that on Windows with using-system-boost builds, the following
         # FindSysLibDep calls do nothing useful (but nothing problematic either)
