From eb57ef5ad6abeaf8d9daf4914f95bcd86670340a Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Mon, 6 Apr 2020 13:00:31 +0200
Subject: [PATCH] Drop dependency on pytest-runner

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index d9887a8..8a63a13 100644
--- a/setup.py
+++ b/setup.py
@@ -29,7 +29,7 @@ class PyTest(TestCommand):
 
 
 setuptools.setup(
-    setup_requires=['pbr>=1.10', 'setuptools>=17.1', 'pytest-runner'],
+    setup_requires=['pbr>=1.10', 'setuptools>=17.1'],
     pbr=True,
     cmdclass={'test': PyTest},
     test_suite='tests')
-- 
2.24.1

