diff -ur aacgain-1.8.orig/mpeg4ip/include/mpeg4ip.h aacgain-1.8/mpeg4ip/include/mpeg4ip.h
--- aacgain-1.8.orig/mpeg4ip/include/mpeg4ip.h	2008-11-08 16:39:36.000000000 +0200
+++ aacgain-1.8/mpeg4ip/include/mpeg4ip.h	2009-07-10 20:59:51.000000000 +0300
@@ -120,6 +120,7 @@
 #endif
 #include <sys/param.h>
 
+/*
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -127,6 +128,7 @@
 #ifdef __cplusplus
 }
 #endif
+*/
 
 #define OPEN_RDWR O_RDWR
 #define OPEN_CREAT O_CREAT 
