From 0be981476e019e9fddb5529a73aadf004e94656b Mon Sep 17 00:00:00 2001
From: tomty89 <tom.ty89@gmail.com>
Date: Tue, 12 Feb 2019 18:55:09 +0800
Subject: [PATCH] ip/ffmpeg: enable opus support (#865)

---
 ip/ffmpeg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ip/ffmpeg.c b/ip/ffmpeg.c
index 04b916cc..427257c7 100644
--- a/ip/ffmpeg.c
+++ b/ip/ffmpeg.c
@@ -519,7 +519,7 @@ const int ip_priority = 30;
 const char *const ip_extensions[] = {
 	"aa", "aac", "ac3", "aif", "aifc", "aiff", "ape", "au", "fla", "flac",
 	"m4a", "m4b", "mka", "mkv", "mp+", "mp2", "mp3", "mp4", "mpc", "mpp",
-	"ogg", "shn", "tak", "tta", "wav", "webm", "wma", "wv",
+	"ogg", "opus", "shn", "tak", "tta", "wav", "webm", "wma", "wv",
 #ifdef USE_FALLBACK_IP
 	"*",
 #endif
