main: raise SHM unit limit

Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

diff -Nuar --exclude '*~' omnisync-1.0.orig/main.c omnisync-1.0/main.c
--- omnisync-1.0.orig/main.c	2009-01-02 01:03:20.000000000 -0800
+++ omnisync-1.0/main.c	2018-06-01 14:32:34.630066839 -0700
@@ -346,8 +346,8 @@
 				unit_nr = atoi(optarg);
 				if (unit_nr < 0)
 					error_exit("-u requires a positive value\n");
-				if (unit_nr > 3)
-					error_exit("NTPd normally supports only 4 shared memory devices, still continuing though\n");
+				if (unit_nr > 254)
+					error_exit("NTPd normally supports only 254 shared memory devices\n");
 				break;
 
 			case 'v':
