diff -uNr asterisk-13.18.2.ORIG/configure.ac asterisk-13.18.2/configure.ac
--- asterisk-13.18.2.ORIG/configure.ac	2017-11-14 11:26:06.179850146 +0000
+++ asterisk-13.18.2/configure.ac	2017-11-14 11:27:46.256844611 +0000
@@ -2503,7 +2503,7 @@
     fi
 fi
 
-for ver in 2.0 2.2 2.4 2.6 3.0; do
+for ver in 3.0 2.6 2.4 2.2 2.0; do
 	AST_PKG_CONFIG_CHECK([GMIME], gmime-$ver)
 	if test "$PBX_GMIME" = 1; then
 		break;
