--- a/setup.py
+++ b/setup.py
@@ -35,7 +35,7 @@ management tasks, including:
 """
 
 requires = [
-    'boto==2.30.0',
+    'boto>=2.30.0',
     'crcmod>=1.7',
     'gcs-oauth2-boto-plugin>=1.8',
     'httplib2>=0.8',
@@ -44,9 +44,7 @@ requires = [
     'python-gflags>=2.0',
     'retry_decorator>=1.0.0',
     'six>=1.8.0',
-    # Not using 1.02 because of:
-    #   https://code.google.com/p/socksipy-branch/issues/detail?id=3
-    'SocksiPy-branch==1.01',
+    'SocksiPy-branch>=1.01',
 ]
 
 dependency_links = [
